<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4.6 Advanced Multi-Group Comparisons | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="4.6 Advanced Multi-Group Comparisons | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4.6 Advanced Multi-Group Comparisons | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  <meta name="twitter:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  

<meta name="author" content="Rager Lab" />


<meta name="date" content="2024-10-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="Icons_Used_Throughout/Favicon.png" type="image/x-icon" />
<link rel="prev" href="multi-group-and-multi-variable-comparisons-and-visualizations.html"/>
<link rel="next" href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>
<link href="libs/table1-1.0/table1_defaults.css" rel="stylesheet" />
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-FGFKVX2X3G"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-FGFKVX2X3G');
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="Icons_Used_Throughout/TAME 2.0 Logo.png"><br></a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Chapter 1 Introductory <br>Data Science</b></span></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html"><i class="fa fa-check"></i>1.1 FAIR Data Management Practices</a>
<ul>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-training-module"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-fair"><i class="fa fa-check"></i>Introduction to FAIR</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#breaking-down-fair-letter-by-letter"><i class="fa fa-check"></i>Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#what-does-this-mean-for-you"><i class="fa fa-check"></i>What Does This Mean for You?</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#data-repositories-for-sharing-of-data"><i class="fa fa-check"></i>Data Repositories for Sharing of Data</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#recent-shifts-in-regulatory-policies-for-data-sharing"><i class="fa fa-check"></i>Recent Shifts in Regulatory Policies for Data Sharing</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#additional-training-resources-on-fair"><i class="fa fa-check"></i>Additional Training Resources on FAIR</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html"><i class="fa fa-check"></i>1.2 Data Sharing through Online Repositories</a>
<ul>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#an-overview-and-example-with-the-dataverse-repository"><i class="fa fa-check"></i>An Overview and Example with the Dataverse Repository</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#introduction-to-training-module-1"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#data-repositories"><i class="fa fa-check"></i>Data Repositories</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#the-dataverse-project"><i class="fa fa-check"></i>The Dataverse Project</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#what-is-a-dataverse"><i class="fa fa-check"></i>What is a Dataverse?</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#metadata"><i class="fa fa-check"></i>Metadata</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataverse"><i class="fa fa-check"></i>Creating a Dataverse</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataset"><i class="fa fa-check"></i>Creating a Dataset</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#concluding-remarks-1"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html"><i class="fa fa-check"></i>1.3 File Management using Github</a>
<ul>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#introduction-to-training-module-2"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-an-account"><i class="fa fa-check"></i>Creating an Account</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-a-repository"><i class="fa fa-check"></i>Creating a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#uploading-code"><i class="fa fa-check"></i>Uploading Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#adding-subfolders-in-a-repository"><i class="fa fa-check"></i>Adding Subfolders in a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-code"><i class="fa fa-check"></i>Updating Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-repository-titles-and-structure-to-support-a-manuscript"><i class="fa fa-check"></i>Updating Repository Titles and Structure to Support a Manuscript</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#tracking-code-changes-using-github-branches"><i class="fa fa-check"></i>Tracking Code Changes using Github Branches</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#concluding-remarks-2"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html"><i class="fa fa-check"></i>1.4 Data Wrangling in Excel</a>
<ul>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#introduction-to-training-module-3"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#save-a-copy-of-the-soon-to-be-organized-and-cleaned-dataset-as-a-new-file"><i class="fa fa-check"></i>Save a Copy of the Soon-To-Be Organized and Cleaned Dataset as a New File</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-extraneous-white-space"><i class="fa fa-check"></i>Remove Extraneous White Space</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#replace-missing-data-with-na"><i class="fa fa-check"></i>Replace Missing Data with “NA”</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-metadata-tab"><i class="fa fa-check"></i>Create a Metadata Tab</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#abbreviate-and-capitalize-categorical-data"><i class="fa fa-check"></i>Abbreviate and Capitalize Categorical Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#alphabetize-sort-the-data-by-the-categorical-variable-of-interest"><i class="fa fa-check"></i>Alphabetize (Sort) the Data by the Categorical Variable of Interest</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-new-subject-number-column"><i class="fa fa-check"></i>Create a New Subject Number Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-special-symbols-and-dashes"><i class="fa fa-check"></i>Remove Special Symbols and Dashes</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#bold-all-column-names-and-center-all-data"><i class="fa fa-check"></i>Bold all Column Names and Center all Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-subject-identifier-column"><i class="fa fa-check"></i>Create a Subject Identifier Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#separate-subject-demographic-data-from-experimental-measurements"><i class="fa fa-check"></i>Separate Subject Demographic Data from Experimental Measurements</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#convert-data-from-wide-to-long-format"><i class="fa fa-check"></i>Convert Data from Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#pivoting-data-from-a-wide-to-long-format"><i class="fa fa-check"></i>Pivoting Data from a Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#generating-summary-level-statistics-with-pivot-tables"><i class="fa fa-check"></i>Generating Summary-Level Statistics with Pivot Tables</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#excel-vs.-r-which-should-you-use"><i class="fa fa-check"></i>Excel vs. R: Which Should You Use?</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#concluding-remarks-3"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 2 Coding in R</b></span></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html"><i class="fa fa-check"></i>2.1 Downloading and Programming in R</a>
<ul>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-training-module-4"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#general-introduction-and-installation-of-r-and-rstudio"><i class="fa fa-check"></i>General Introduction and Installation of R and RStudio</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-r-packages"><i class="fa fa-check"></i>Introduction to R Packages</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#scripting-basics"><i class="fa fa-check"></i>Scripting Basics</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#code-troubleshooting"><i class="fa fa-check"></i>Code Troubleshooting</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#concluding-remarks-4"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html"><i class="fa fa-check"></i>2.2 Coding “Best” Practices</a>
<ul>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#introduction-to-training-module-5"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#scripting-file-types"><i class="fa fa-check"></i>Scripting File Types</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-headers-and-annotation"><i class="fa fa-check"></i>Script Headers and Annotation</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#coding-style"><i class="fa fa-check"></i>Coding Style</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-organization"><i class="fa fa-check"></i>Script Organization</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#concluding-remarks-5"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html"><i class="fa fa-check"></i>2.3 Data Manipulation and Reshaping</a>
<ul>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-training-module-6"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#data-manipulation-using-base-r"><i class="fa fa-check"></i>Data Manipulation Using Base R</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-tidyverse"><i class="fa fa-check"></i>Introduction to Tidyverse</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#concluding-remarks-6"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html"><i class="fa fa-check"></i>2.4 Improving Coding Efficiencies</a>
<ul>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#introduction-to-training-module-7"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#loops"><i class="fa fa-check"></i>Loops</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#functions"><i class="fa fa-check"></i>Functions</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#list-operations"><i class="fa fa-check"></i>List operations</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#concluding-remarks-7"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#additional-resources-2"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="part"><span><b>Chapter 3 Basics of <br>Data Analysis and Visualizations</b></span></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html"><i class="fa fa-check"></i>3.1 Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-data-visualizations"><i class="fa fa-check"></i>Introduction to Data Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-training-module-8"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#density-plot-visualization"><i class="fa fa-check"></i>Density Plot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#boxplot-visualization"><i class="fa fa-check"></i>Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#correlation-visualizations"><i class="fa fa-check"></i>Correlation Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#heatmap-visualization"><i class="fa fa-check"></i>Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#concluding-remarks-8"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html"><i class="fa fa-check"></i>3.2 Improving Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-data-visulization-conventions"><i class="fa fa-check"></i>Introduction to Data Visulization Conventions</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-training-module-9"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-boxplot-visualization"><i class="fa fa-check"></i>Creating an Improved Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-heatmap-visualization"><i class="fa fa-check"></i>Creating an Improved Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-multi-plot-figures"><i class="fa fa-check"></i>Creating Multi-Plot Figures</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#concluding-remarks-9"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html"><i class="fa fa-check"></i>3.3 Normality Tests and Data Transformations</a>
<ul>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#introduction-to-training-module-10"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#what-is-a-normal-distribution"><i class="fa fa-check"></i>What is a Normal Distribution?</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#qualitative-assessment-of-normality"><i class="fa fa-check"></i>Qualitative Assessment of Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#quantitative-normality-assessment"><i class="fa fa-check"></i>Quantitative Normality Assessment</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#data-transformation"><i class="fa fa-check"></i>Data Transformation</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#additional-considerations-regarding-normality"><i class="fa fa-check"></i>Additional Considerations Regarding Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#concluding-remarks-10"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html"><i class="fa fa-check"></i>3.4 Intoduction to Statistical Tests</a>
<ul>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#introduction-to-training-module-11"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#assessing-normality-homogeneity-of-variance"><i class="fa fa-check"></i>Assessing Normality &amp; Homogeneity of Variance</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#two-group-visualizations-and-statistical-comparisons-using-the-t-test"><i class="fa fa-check"></i>Two-Group Visualizations and Statistical Comparisons using the T-Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#three-group-visualizations-and-statistical-comparisons-using-an-anova"><i class="fa fa-check"></i>Three-Group Visualizations and Statistical Comparisons using an ANOVA</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#regression-modeling-and-visualization-linear-and-logistic-regressions"><i class="fa fa-check"></i>Regression Modeling and Visualization: Linear and Logistic Regressions</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#statistical-evaluations-of-categorical-data-using-the-chi-squared-test-and-fishers-exact-test"><i class="fa fa-check"></i>Statistical Evaluations of Categorical Data using the Chi-Squared Test and Fisher’s Exact Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#concluding-remarks-11"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 4 Converting Wet Lab Data into Dry Lab Analyses</b></span></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html"><i class="fa fa-check"></i>4.1 Overview of Experimental Design and Example Data</a>
<ul>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#introduction-to-training-module-12"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#replicates"><i class="fa fa-check"></i>Replicates</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#orientation-to-example-data-for-chapter-4"><i class="fa fa-check"></i>Orientation to Example Data for Chapter 4</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#concluding-remarks-12"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html"><i class="fa fa-check"></i>4.2 Data Import, Processing, and Summary Statistics</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#introduction-to-training-module-13"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#data-import"><i class="fa fa-check"></i>Data Import</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#handling-missing-values"><i class="fa fa-check"></i>Handling Missing Values</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#averaging-replicates"><i class="fa fa-check"></i>Averaging Replicates</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#descriptive-statistics"><i class="fa fa-check"></i>Descriptive Statistics</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#normality-assessment-and-data-transformation"><i class="fa fa-check"></i>Normality Assessment and Data Transformation</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#concluding-remarks-13"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html"><i class="fa fa-check"></i>4.3 Data Import from PDF Sources</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#introduction-to-training-module-14"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-from-many-single-pdfs-with-the-same-formatting"><i class="fa fa-check"></i>Importing Data from Many Single PDFs with the Same Formatting</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-stored-in-pdf-tables"><i class="fa fa-check"></i>Importing Data Stored in PDF Tables</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#concluding-remarks-14"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html"><i class="fa fa-check"></i>4.4 Two Group Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#introduction-to-training-module-15"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#overview-of-two-group-statistical-tests"><i class="fa fa-check"></i>Overview of Two Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#statistical-vs.-biological-significance"><i class="fa fa-check"></i>Statistical vs. Biological Significance</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#unpaired-test-example"><i class="fa fa-check"></i>Unpaired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#paired-test-example"><i class="fa fa-check"></i>Paired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#visualizing-results"><i class="fa fa-check"></i>Visualizing Results</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#concluding-remarks-15"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html"><i class="fa fa-check"></i>4.5 Multi-Group and Multi-Variable Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#introduction-to-training-module-16"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#overview-of-multi-group-statistical-tests"><i class="fa fa-check"></i>Overview of Multi-Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#multi-group-analysis-example"><i class="fa fa-check"></i>Multi-Group Analysis Example</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#visualization-of-multi-group-statistical-results"><i class="fa fa-check"></i>Visualization of Multi-Group Statistical Results</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#concluding-remarks-16"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html"><i class="fa fa-check"></i>4.6 Advanced Multi-Group Comparisons</a>
<ul>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#introduction-to-training-module-17"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#advanced-multi-group-comparisons-1"><i class="fa fa-check"></i>Advanced Multi-Group Comparisons</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#two-way-anova-example"><i class="fa fa-check"></i>Two-way ANOVA Example</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#ancova-example"><i class="fa fa-check"></i>ANCOVA Example</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#concluding-remarks-17"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 5 Machine Learning &amp; Artificial Intelligence</b></span></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><i class="fa fa-check"></i>5.1 Introduction to Artificial Intelligence, Machine Learning, and Predictive Modeling for Environmental Health</a>
<ul>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#introduction-to-training-module-18"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#general-historical-context-and-taxonomy-of-modern-aiml"><i class="fa fa-check"></i>General Historical Context and Taxonomy of Modern AI/ML</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#application-of-machine-learning-in-environmental-health-science"><i class="fa fa-check"></i>Application of Machine Learning in Environmental Health Science</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#concluding-remarks-18"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html"><i class="fa fa-check"></i>5.2 Supervised Machine Learning</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-training-module-19"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-machine-learning"><i class="fa fa-check"></i>Types of Machine Learning</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-supervised-machine-learning-algorithms"><i class="fa fa-check"></i>Types of Supervised Machine Learning Algorithms</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#training-supervised-machine-learning-models"><i class="fa fa-check"></i>Training Supervised Machine Learning Models</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#assessing-classification-based-model-performance"><i class="fa fa-check"></i>Assessing Classification-Based Model Performance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-activity-and-example-dataset"><i class="fa fa-check"></i>Introduction to Activity and Example Dataset</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#testing-for-differences-in-predictor-variables-across-the-outcome-classes"><i class="fa fa-check"></i>Testing for Differences in Predictor Variables across the Outcome Classes</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#predicting-ias-detection-with-a-random-forest-rf-model"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#class-imbalance"><i class="fa fa-check"></i>Class Imbalance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#concluding-remarks-19"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html"><i class="fa fa-check"></i>5.3 Supervised Machine Learning Model Interpretation</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-training-module-20"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance"><i class="fa fa-check"></i>Variable Importance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary"><i class="fa fa-check"></i>Decision Boundary</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-example-dataset-and-activity"><i class="fa fa-check"></i>Introduction to Example Dataset and Activity</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#predicting-ias-detection-with-a-random-forest-rf-model-1"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance-plot"><i class="fa fa-check"></i>Variable Importance Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot"><i class="fa fa-check"></i>Decision Boundary Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot-incorporating-smote"><i class="fa fa-check"></i>Decision Boundary Plot Incorporating SMOTE</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#concluding-remarks-20"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><i class="fa fa-check"></i>5.4 Unsupervised Machine Learning Part 1: K-Means Clustering &amp; PCA</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-training-module-21"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-unsupervised-machine-learning"><i class="fa fa-check"></i>Introduction to Unsupervised Machine Learning</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-example-data"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#identifying-clusters-of-chemicals-through-k-means"><i class="fa fa-check"></i>Identifying Clusters of Chemicals through <em>K</em>-Means</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#principal-component-analysis-pca-1"><i class="fa fa-check"></i>Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#incorporating-k-means-into-pca-for-predictive-modeling"><i class="fa fa-check"></i>Incorporating <em>K</em>-Means into PCA for Predictive Modeling</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#concluding-remarks-21"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><i class="fa fa-check"></i>5.5 Unsupervised Machine Learning Part 2: Additional Clustering Applications</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-training-module-22"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#k-means-clustering-1"><i class="fa fa-check"></i><em>K</em>-Means Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-example-data-1"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-1"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#clustering-plot"><i class="fa fa-check"></i>Clustering Plot</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-visualization"><i class="fa fa-check"></i>Hierarchical Clustering Visualization</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#variable-contributions"><i class="fa fa-check"></i>Variable Contributions</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#concluding-remarks-22"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 6 Applications in Toxicology &amp; Exposure Science</b></span></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html"><i class="fa fa-check"></i>6.1 Descriptive Cohort Analyses</a>
<ul>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#introduction-to-training-module-23"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#participant-exploration"><i class="fa fa-check"></i>Participant Exploration</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#chemical-detection-filtering"><i class="fa fa-check"></i>Chemical Detection Filtering</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#outlier-identification"><i class="fa fa-check"></i>Outlier Identification</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#summary-statistics-tables"><i class="fa fa-check"></i>Summary Statistics Tables</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#demographics-table"><i class="fa fa-check"></i>Demographics Table</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#concluding-remarks-23"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html"><i class="fa fa-check"></i>6.2 -Omics and System Biology: Transcriptomic Applications</a>
<ul>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-training-module-24"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-the-field-of--omics"><i class="fa fa-check"></i>Introduction to the Field of “-Omics”</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-transcriptomics"><i class="fa fa-check"></i>Introduction to Transcriptomics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#formatting-data-for-downstream-statistics"><i class="fa fa-check"></i>Formatting Data for Downstream Statistics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#transcriptomics-data-qaqc"><i class="fa fa-check"></i>Transcriptomics Data QA/QC</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#controlling-for-sources-of-sample-heterogeneity"><i class="fa fa-check"></i>Controlling for Sources of Sample Heterogeneity</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e.g.-biomass-smoke-exposure"><i class="fa fa-check"></i>Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure)</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-ma-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using MA Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-volcano-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using Volcano Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis"><i class="fa fa-check"></i>Interpretting Findings at the Systems Level through Pathway Enrichment Analysis</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#concluding-remarks-24"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><i class="fa fa-check"></i>6.3 Mixtures Analysis Methods Part 1: Overview and Example with Quantile G-Computation</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-training-module-25"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#overview-of-mixtures-analysis"><i class="fa fa-check"></i>Overview of Mixtures Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-example-data-2"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#mixtures-model-with-standard-qqcomp"><i class="fa fa-check"></i>Mixtures Model with Standard qqcomp</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#concluding-remarks-25"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#additional-resources-11"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><i class="fa fa-check"></i>6.4 Mixtures Analysis Methods Part 2: Bayesian Kernel Machine Regression</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-training-module-26"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-example-data-3"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-bkmr"><i class="fa fa-check"></i>Introduction to BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#run-bkmr"><i class="fa fa-check"></i>Run BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#concluding-remarks-26"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#additional-resources-12"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html"><i class="fa fa-check"></i>6.5 Mixtures Analysis Methods Part 3: Sufficient Similarity</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-training-module-27"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-toxicant-and-dataset"><i class="fa fa-check"></i>Introduction to Toxicant and Dataset</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#workspace-preparation-and-data-import-9"><i class="fa fa-check"></i>Workspace Preparation and Data Import</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#chemistry-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Chemistry-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#toxicity-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Toxicity-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#comparing-chemistry-vs.-toxicity-sufficient-similarity-analyses"><i class="fa fa-check"></i>Comparing Chemistry vs. Toxicity Sufficient Similarity Analyses</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#concluding-remarks-27"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i>6.6 Toxicokinetic Modeling</a>
<ul>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-training-module-28"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-toxicokinetic-modeling"><i class="fa fa-check"></i>Introduction to Toxicokinetic Modeling</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#data-and-models-used-in-toxicokinetic-modeling-tk"><i class="fa fa-check"></i>Data and Models used in Toxicokinetic Modeling (TK)</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-toxicokinetics"><i class="fa fa-check"></i>Reverse Toxicokinetics</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#capturing-population-variability-in-toxicokinetics-and-uncertainty-in-chemical-specific-parameters"><i class="fa fa-check"></i>Capturing Population Variability in Toxicokinetics, and Uncertainty in Chemical-Specific Parameters</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-tk-calculating-administered-equivalent-doses-for-toxcast-bioactive-concentrations"><i class="fa fa-check"></i>Reverse TK: Calculating Administered Equivalent Doses for ToxCast Bioactive Concentrations</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#comparing-equivalent-doses-estimated-to-elicit-toxicity-hazard-to-external-exposure-estimates-exposure-for-chemical-prioritization-by-bioactivity-exposure-ratios-bers"><i class="fa fa-check"></i>Comparing Equivalent Doses Estimated to Elicit Toxicity (Hazard) to External Exposure Estimates (Exposure), for Chemical Prioritization by Bioactivity-Exposure Ratios (BERs)</a></li>
</ul></li>
<li class="part"><span><b>Chapter 7 Environmental Health Database Mining</b></span></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html"><i class="fa fa-check"></i>7.1 Comparative Toxicogenomics Database</a>
<ul>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#introduction-to-training-module-30"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#ctd-data-in-r"><i class="fa fa-check"></i>CTD Data in R</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#identifying-genes-under-epigenetic-control"><i class="fa fa-check"></i>Identifying Genes Under Epigenetic Control</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#concluding-remarks-30"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html"><i class="fa fa-check"></i>7.2 Gene Expression Omnibus</a>
<ul>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-training-module-31"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-geo"><i class="fa fa-check"></i>Introduction to GEO</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-example-data-4"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#geo-data-in-r"><i class="fa fa-check"></i>GEO Data in R</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#visualizing-data"><i class="fa fa-check"></i>Visualizing Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#statistical-analyses"><i class="fa fa-check"></i>Statistical Analyses</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#concluding-remarks-31"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html"><i class="fa fa-check"></i>7.3 CompTox Dashboard Data through APIs</a>
<ul>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#introduction-to-training-module-32"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#training-modules-environmental-health-questions-24"><i class="fa fa-check"></i>Training Module’s Environmental Health Questions</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#script-preparations-12"><i class="fa fa-check"></i>Script Preparations</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#introduction-to-comptox-chemicals-dashboard"><i class="fa fa-check"></i>Introduction to CompTox Chemicals Dashboard</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#cctes-ctx-application-programming-interfaces-apis-for-automated-batch-search-of-the-ccd"><i class="fa fa-check"></i>CCTE’s CTX Application Programming Interfaces (APIs) for Automated Batch Search of the CCD</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#comparing-physico-chemical-properties-between-two-important-environmental-contaminant-lists"><i class="fa fa-check"></i>Comparing Physico-chemical Properties between Two Important Environmental Contaminant Lists</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#hazard-data-genotoxicity"><i class="fa fa-check"></i>Hazard Data: Genotoxicity</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#hazard-resource"><i class="fa fa-check"></i>Hazard Resource</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#concluding-remarks-32"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html"><i class="fa fa-check"></i>7.4 Database Integration: Air Quality, Mortality, and Environmental Justice Data</a>
<ul>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#introduction-to-training-module-33"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#introduction-to-exposure-and-health-databases"><i class="fa fa-check"></i>Introduction to Exposure and Health Databases</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#analyzing-relationships-between-pm2.5-and-mortality"><i class="fa fa-check"></i>Analyzing Relationships between PM<sub>2.5</sub> and Mortality</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#environmental-justice-considerations"><i class="fa fa-check"></i>Environmental Justice Considerations</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#concluding-remarks-33"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="advanced-multi-group-comparisons" class="section level1 hasAnchor">
<h1>4.6 Advanced Multi-Group Comparisons<a href="advanced-multi-group-comparisons.html#advanced-multi-group-comparisons" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This training module was developed by Elise Hickman, Alexis Payton, and Julia E. Rager.</p>
<p>All input files (script, data, and figures) can be downloaded from the <a href="https://github.com/UNCSRP/TAME2">UNC-SRP TAME2 GitHub website</a>.</p>
<div id="introduction-to-training-module-17" class="section level2 hasAnchor">
<h2>Introduction to Training Module<a href="advanced-multi-group-comparisons.html#introduction-to-training-module-17" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In the previous module, we covered how to apply multi-group statistical testing, in which we tested for significant differences in endpoints across different values for one independent variable. In this module, we will build on the concepts introduced previously to test for significant differences in endpoints while considering two or more independent variables. We will review relevant statistical approaches and demonstrate how to apply these tests using the same example dataset as in previous modules in this chapter. As a reminder, this dataset includes concentrations of inflammatory biomarkers secreted by airway epithelial cells after exposure to different concentrations of acrolein.</p>
<div id="training-modules-environmental-health-questions-10" class="section level3 hasAnchor">
<h3>Training Module’s Environmental Health Questions<a href="advanced-multi-group-comparisons.html#training-modules-environmental-health-questions-10" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This training module was specifically developed to answer the following environmental health questions:</p>
<ol style="list-style-type: decimal">
<li><p>Are there significant differences in inflammatory biomarker concentrations between sex and different doses of acrolein?</p></li>
<li><p>Are there significant differences in inflammatory biomarker concentrations across different doses of acrolein after controlling for sex and age?</p></li>
</ol>
</div>
<div id="workspace-preparation-and-data-import-6" class="section level3 hasAnchor">
<h3>Workspace Preparation and Data Import<a href="advanced-multi-group-comparisons.html#workspace-preparation-and-data-import-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Here, we will import the processed data that we generated at the end of TAME 2.0 Module 4.2, introduced in <strong>TAME 2.0 Module 4.1 Overview of Experimental Design and Example Data</strong> and associated demographic data. These data represent log<sub>2</sub> concentrations of inflammatory biomarkers secreted by airway epithelial cells after exposure to four different concentrations of acrolein (plus filtered air as a control). We will also load packages that will be needed for the analysis, including previously introduced packages such as <em>openxlsx</em>, <em>tidyverse</em>, <em>DT</em>, <em>ggpubr</em>, and <em>rstatix</em>.</p>
<div id="cleaning-the-global-environment-5" class="section level4 hasAnchor">
<h4>Cleaning the global environment<a href="advanced-multi-group-comparisons.html#cleaning-the-global-environment-5" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb452"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb452-1"><a href="advanced-multi-group-comparisons.html#cb452-1" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list=</span><span class="fu">ls</span>())</span></code></pre></div>
</div>
<div id="loading-r-packages-required-for-this-session-4" class="section level4 hasAnchor">
<h4>Loading R packages required for this session<a href="advanced-multi-group-comparisons.html#loading-r-packages-required-for-this-session-4" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb453"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb453-1"><a href="advanced-multi-group-comparisons.html#cb453-1" tabindex="-1"></a><span class="fu">library</span>(openxlsx)</span>
<span id="cb453-2"><a href="advanced-multi-group-comparisons.html#cb453-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb453-3"><a href="advanced-multi-group-comparisons.html#cb453-3" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb453-4"><a href="advanced-multi-group-comparisons.html#cb453-4" tabindex="-1"></a><span class="fu">library</span>(rstatix)</span>
<span id="cb453-5"><a href="advanced-multi-group-comparisons.html#cb453-5" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb453-6"><a href="advanced-multi-group-comparisons.html#cb453-6" tabindex="-1"></a><span class="fu">library</span>(multcomp)</span>
<span id="cb453-7"><a href="advanced-multi-group-comparisons.html#cb453-7" tabindex="-1"></a><span class="fu">library</span>(pander)</span>
<span id="cb453-8"><a href="advanced-multi-group-comparisons.html#cb453-8" tabindex="-1"></a></span>
<span id="cb453-9"><a href="advanced-multi-group-comparisons.html#cb453-9" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>()) <span class="co"># Set graphing theme</span></span></code></pre></div>
</div>
<div id="set-your-working-directory-6" class="section level4 hasAnchor">
<h4>Set your working directory<a href="advanced-multi-group-comparisons.html#set-your-working-directory-6" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb454"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb454-1"><a href="advanced-multi-group-comparisons.html#cb454-1" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/filepath to where your input files are&quot;</span>)</span></code></pre></div>
</div>
<div id="importing-example-dataset-5" class="section level4 hasAnchor">
<h4>Importing example dataset<a href="advanced-multi-group-comparisons.html#importing-example-dataset-5" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb455"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb455-1"><a href="advanced-multi-group-comparisons.html#cb455-1" tabindex="-1"></a>biomarker_data <span class="ot">&lt;-</span> <span class="fu">read.xlsx</span>(<span class="st">&quot;Module4_6_Input/Module4_6_InputData1.xlsx&quot;</span>)</span>
<span id="cb455-2"><a href="advanced-multi-group-comparisons.html#cb455-2" tabindex="-1"></a>demographic_data <span class="ot">&lt;-</span> <span class="fu">read.xlsx</span>(<span class="st">&quot;Module4_6_Input/Module4_6_InputData2.xlsx&quot;</span>)</span>
<span id="cb455-3"><a href="advanced-multi-group-comparisons.html#cb455-3" tabindex="-1"></a></span>
<span id="cb455-4"><a href="advanced-multi-group-comparisons.html#cb455-4" tabindex="-1"></a><span class="co"># View data</span></span>
<span id="cb455-5"><a href="advanced-multi-group-comparisons.html#cb455-5" tabindex="-1"></a><span class="fu">datatable</span>(biomarker_data)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-a75a5827a6196abfe175" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-a75a5827a6196abfe175">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80"],["D10","D10","D10","D10","D10","D11","D11","D11","D11","D11","D12","D12","D12","D12","D12","D13","D13","D13","D13","D13","D14","D14","D14","D14","D14","D15","D15","D15","D15","D15","D16","D16","D16","D16","D16","D1","D1","D1","D1","D1","D2","D2","D2","D2","D2","D3","D3","D3","D3","D3","D4","D4","D4","D4","D4","D5","D5","D5","D5","D5","D6","D6","D6","D6","D6","D7","D7","D7","D7","D7","D8","D8","D8","D8","D8","D9","D9","D9","D9","D9"],["0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4"],[3.71808758396052,3.7235585614721,3.70597790168252,3.76553474636298,3.6724253419715,3.83491342970104,3.93073733756289,3.92884403671257,3.80632405739003,3.81762325751143,3.36036427654347,3.27649666564036,3.57167680997093,3.54101915313356,3.10097764772482,3.35755200461808,3.33628338786443,3.31469652565629,3.50589092972996,3.69041709553247,3.31614574229336,3.43562859405209,3.57773093149008,3.87578006306849,3.83086375675176,3.42894634476948,3.49825086752783,3.57288966842058,3.63807383718072,3.6915341649192,3.38681094647222,3.49825086752783,3.58856473740135,3.66220549965362,3.69376571221778,3.76871365703048,3.77820857639809,3.74953426766926,3.84096670448742,3.50589092972996,3.53480866123175,3.60880924267552,3.58856473740135,3.2585189247113,3.75167794568758,3.58014548442338,3.73876783680019,3.61823865559545,3.58736499093646,3.39643353125099,3.52732060787705,3.50842865253186,3.62175885670973,3.23266075679027,4.13996056954546,3.16671544496642,3.18269229751619,3.13422093976063,3.7279204545632,3.799087306074,2.86195536414487,3.08746284125034,3.64961545906341,3.66220549965362,3.64616265715789,3.04614178164472,3.1811025507538,3.27202318906105,3.54101915313356,3.42088657497553,3.37712374912949,3.57531233068744,3.58976348698498,3.83289001416474,3.8972404255748,3.21878116778407,3.30888505741176,3.51853513898218,3.5765221379205,3.7279204545632],[8.68422263328295,8.773963368433559,8.561326136453919,9.710324307072939,7.55028504908841,10.2222394489088,10.1590009172826,9.94747622373011,9.81610366284939,8.03815081026938,9.27052894238072,9.05077375519723,8.99740532549677,8.27635707925924,7.61889934790406,8.15790268278964,8.089053172156429,7.41515020542639,7.37920505996061,4.10265813136374,7.56094417806009,7.5894638929561,7.9047851297316,7.83857365266832,6.50969584193339,9.254131033541951,9.422148885880191,9.165032591502291,9.09402488708305,8.202809492125009,8.68422263328295,8.773963368433559,8.77360079528254,8.60099074240823,7.71163256182322,10.1099481540673,10.3357141430951,9.686097776967401,10.0151641270811,10.1498741117182,8.88962564051876,9.317005526387209,9.03875441393145,8.60099074240823,4.87479696589866,9.23160520814165,9.677085350559841,9.23208509872283,9.583628142323651,9.560179957764181,8.34837407519116,8.351469324267081,8.77360079528254,9.33695280016051,7.70355756192103,3.9429835981871,4.12018602753186,4.24868662687383,5.4110872267986,5.17871464117544,7.43212427264156,6.85112436288675,7.38076407498511,6.96220195703608,8.36382813184704,8.12861303870471,8.0503384381673,7.86530024272912,7.3639595068907,8.90926307937532,11.4238722598684,11.7288813640812,11.4570986742783,10.878112237463,8.52575569282949,7.01077983875324,7.32804655627074,7.03452387533439,7.27537959625362,4.70099449416827],[15.0940690235359,15.1807289234454,15.1191518828101,16.3458996676138,16.7241460980745,15.0028356680746,15.5465951732949,15.4894995973234,16.169128959283,16.0996069083821,13.9520295507825,14.0492691434224,13.1502883461063,14.8168226706095,16.3560077655342,14.30563571243,14.2750718138596,14.2603087281481,14.3170903457307,16.6038859325635,13.9567554773633,14.1371656727442,14.4431119040824,15.1950634099335,16.3613300730261,14.4305440239947,14.6030328264996,14.6583565484435,15.3569594653045,16.5368999542995,14.3291992954213,14.4239037658366,14.5587322032289,15.1950634099335,16.3717050181953,15.1620702745062,15.5056676741312,15.1429673906615,15.4079647745043,16.9008155214109,14.6079451777076,14.6753941971444,13.1497550568455,14.9206710618732,17.2934204566841,14.5987126147926,14.8967320984425,14.6970451993658,15.0929012215711,17.1264130896721,14.542155252148,14.4584290430224,14.8614654668171,15.9922404459517,15.3426552970282,14.1001424218123,13.9841362654046,14.0738746130993,15.2370937904406,16.3510586187167,13.4034767756747,13.4654490662827,13.7572472746771,15.5483372770406,16.2848127161883,13.5861396108168,13.8384840329337,13.8699889688218,14.4670316946933,17.0361832734254,14.1156143059506,14.4239037658366,14.4497857465895,15.0182445232662,15.1484821432167,14.3291992954213,14.1946037445951,13.1529914376826,14.708543993667,16.3449160294487],[0.189033824390017,-0.3040061868901,0.286881147788162,0.298658315564515,1.0703893278914,2.3305584000308,1.37851162325373,1.30451104180995,0.978195629681652,1.64154602908752,0.526068811667588,0.238786859587116,0.713695814843359,1.50589092972996,0.333423733725192,1.24488705912353,-0.3040061868901,1.56559717585423,1.09761079662642,0.855989697308481,1.06350294230616,0.20163386116965,0.895302621333307,1.1243281350022,1.37295209791183,1.27500704749987,0.632268215499513,1.16992500144231,0.910732661902913,1.16992500144231,0.8155754288625729,0.333423733725192,0.748461233004036,0.903038270112912,1.13750352374994,0.367371065648529,0.137503523749935,0.464668267003444,0.910732661902913,0.454175893185802,1.35614381022528,0.298658315564515,0.632268215499513,0.748461233004036,1.40599235967584,0.31034012061215,0.124328135002202,0.53605290024021,0.495695162624069,0.321928094887362,0.879705766282288,0.298658315564515,1.01435529297707,1.19534759832222,2.03210084316702,0.933572638261024,0.20163386116965,2.68706068833989,1.13750352374994,0.84799690655495,1.86393845042397,1.41142624572647,0.992768430768924,1.0071955014042,0.443606651475615,0.650764559116902,0.8718436485093179,1.19534759832222,0.16349873228288,1.17632277264046,1.01435529297707,0.941106310946431,1.04264433740849,0.8718436485093179,1.57531233068744,2.51349074558812,0.604071323668861,0.411426245726465,0.584962500721156,1.47508488294878],[1.73552217729654,1.55581615506164,1.93734439215023,2.08406426478847,2.83390207666916,3.23572705983806,3.24031432933371,3.15218341905505,3.16510798514454,3.48284828306847,1.63226821549951,1.91073266190291,2.35049724708413,3.24031432933371,1.57046293102604,1.32192809488736,1.20789285164133,1.20789285164133,2.75274859140713,2.59931779369823,1.91456452349394,1.72246602447109,1.96347412397489,2.81352468929781,3.27202318906105,2.08746284125034,2.02502879449152,2.37016428054021,2.63923216324928,2.94485844580754,1.83187724119167,1.91073266190291,2.0496307677246,2.64154602908752,2.93168305705981,1.86393845042397,1.790772037862,1.86393845042397,1.88362081628567,2.88362081628567,1.83187724119167,2.0071955014042,2.02147972741045,2.48542682717024,2.98913900736823,2.52606881166759,1.49569516262407,1.85598969730848,1.50080205305716,2.06695024392463,2.47508488294878,2.31034012061215,2.39231742277876,2.99457972421575,3.73226919950145,2.20476675065461,2.34482849699744,1.96347412397489,2.40053792958373,2.85798099512757,1.790772037862,2.11103131238874,2.39506279951758,3.0232553523003,2.16671544496642,1.56559717585423,1.32192809488736,3.59693514238723,1.73552217729654,2.75702324650746,2.53106949272595,2.48284828306847,2.7158933705476,2.99819550315325,3.47118746038699,1.0703893278914,1.33342373372519,1.97819562968165,2.58496250072116,3.08406426478847],[10.4475275409266,10.4320198468125,10.4523853997561,10.7211590148677,11.2458081856384,9.02901108665907,9.22583403553784,9.412739192116421,9.91233549843777,9.316688823436939,9.428025110276749,9.412146396540679,9.620146497339491,10.9260809830959,7.3039633562453,9.247642742898019,9.32424806586749,10.2153509830024,11.2601080099627,10.8557639471917,10.6537676442859,10.6473145095471,10.8149028512154,11.0078962674403,10.9341110109735,9.933439857942711,10.009954550333,10.2267734234482,10.7809497987497,10.646036617491,9.797969484059131,9.8490610025615,10.2090268302306,10.7211590148677,10.8557639471917,9.696480751820189,9.72723977748481,9.49325503512067,10.0461417816447,11.0193335777127,10.534828110721,10.8215188973143,10.8050102485124,11.7900226112658,11.031563370024,9.35887093968562,9.18935511254853,9.033670771393091,9.36430978186441,11.0217850685075,8.283551423174311,8.22211866974429,8.617467465294199,9.414283550503709,8.973180744513019,9.425320822759881,9.308885057411761,9.61402881967061,10.4624204732526,10.7035402591663,10.7662472988462,10.9813603523445,11.5188793273505,11.3227900730377,9.89590846480624,10.3333342781343,10.3711558797312,10.374854737178,10.7157046495212,11.7319596151152,9.92478281821723,9.8490610025615,10.0477374043091,10.6360535917989,9.40224465525408,9.797969484059131,10.0853263897027,10.4360035526972,11.191225812124,10.9299462991725]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Donor<\/th>\n      <th>Dose<\/th>\n      <th>IL1B<\/th>\n      <th>IL6<\/th>\n      <th>IL8<\/th>\n      <th>IL10<\/th>\n      <th>TNFa<\/th>\n      <th>VEGF<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4,5,6,7,8]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Donor","targets":1},{"name":"Dose","targets":2},{"name":"IL1B","targets":3},{"name":"IL6","targets":4},{"name":"IL8","targets":5},{"name":"IL10","targets":6},{"name":"TNFa","targets":7},{"name":"VEGF","targets":8}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb456"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb456-1"><a href="advanced-multi-group-comparisons.html#cb456-1" tabindex="-1"></a><span class="fu">datatable</span>(demographic_data)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-e059adcf85330407ce46" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-e059adcf85330407ce46">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16"],["D1","D2","D3","D4","D5","D6","D7","D8","D9","D10","D11","D12","D13","D14","D15","D16"],[22,35,27,60,43,38,26,36,40,44,46,52,58,65,37,25],["M","F","M","F","F","M","M","F","F","M","M","M","M","M","F","F"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Donor<\/th>\n      <th>Age_yr<\/th>\n      <th>Sex<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Donor","targets":1},{"name":"Age_yr","targets":2},{"name":"Sex","targets":3}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
</div>
<div id="advanced-multi-group-comparisons-1" class="section level2 hasAnchor">
<h2>Advanced Multi-Group Comparisons<a href="advanced-multi-group-comparisons.html#advanced-multi-group-comparisons-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="two-way-anova" class="section level3 hasAnchor">
<h3>Two-way ANOVA<a href="advanced-multi-group-comparisons.html#two-way-anova" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The first test that we’ll introduce is a <strong>two-way ANOVA</strong>. This test involves testing for mean differences in a continuous dependent variable across two categorical independent variables. (As a refresher, a one-way ANOVA uses a single independent variable to compare mean differences between groups.) Subjects or samples can be matched based upon their between-group factors (i.e., exposure duration) and/or their within-group factors (i.e., batch effects). Models that include both between-group and within-group factors are known as <strong>mixed two-way ANOVAs</strong>.</p>
<p>Like other parametric tests, two-way ANOVAs assume:</p>
<ul>
<li>Homogeneity of variance</li>
<li>Independent observations</li>
<li>Normal distribution</li>
</ul>
</div>
<div id="ancova" class="section level3 hasAnchor">
<h3>ANCOVA<a href="advanced-multi-group-comparisons.html#ancova" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>An <strong>Analysis of Covariances (ANCOVA)</strong> tests for mean differences in a continuous dependent variable and at least one categorical independent variable. It also includes another variable, known as a covariate, that needs to be controlled or adjusted for to more accurately capture the relationship between the independent and dependent variables. Potential covariates can include either between-group factors like exposure duration and/or within-group factors like batch effects or sex. Note that if the dataset has a smaller sample size, stratification of the dataset based on that covariate is another option to determine its effects rather than adjusting for it using an ANCOVA.</p>
<p>ANCOVAs have the same assumptions listed above.</p>
<p><strong>Note</strong>: It is possible to run <em>two-way ANCOVA</em> models, where the model contains two independent variables and at least one covariate to be adjusted for.</p>
<p><br></p>
</div>
</div>
<div id="two-way-anova-example" class="section level2 hasAnchor">
<h2>Two-way ANOVA Example<a href="advanced-multi-group-comparisons.html#two-way-anova-example" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Our first environmental health question can be answered using a two-way ANOVA. We can test three different null hypotheses using this test:</p>
<ol style="list-style-type: decimal">
<li>There is no difference in average biomarker concentrations based on sex.</li>
<li>There is no difference in average biomarker concentrations based on dose.</li>
<li>The effect of sex on average biomarker concentration does not depend on the effect of dose and vice versa.</li>
</ol>
<p><br></p>
<p>The first step would be to check that the assumptions (independence, homogeneity of variance, and normal distribution) have been met, but this was done previously in <strong>TAME 2.0 Module 4.4 Two Group Comparisions and Visualizations</strong>.</p>
<p>To run our two-way ANOVA, we will use the <code>anova_test()</code> function from the <em>rstatix</em> package. This function allows us to define subject identifiers for matching between-subject factor variables (such as sex - factors that differ between subjects) and within-subject factors (such as dose - factors that are measured within each subject). Since we have both between- and within- subject factors, we will specifically be running a two-way mixed ANOVA.</p>
<p>First, we need to add our demographic data to our biomarker data so that these variables can be incorporated into the analysis. Also, we need to convert <code>Dose</code> into a factor to specify the levels.</p>
<div class="sourceCode" id="cb457"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb457-1"><a href="advanced-multi-group-comparisons.html#cb457-1" tabindex="-1"></a>biomarker_data <span class="ot">&lt;-</span> biomarker_data <span class="sc">%&gt;%</span></span>
<span id="cb457-2"><a href="advanced-multi-group-comparisons.html#cb457-2" tabindex="-1"></a>  <span class="fu">left_join</span>(demographic_data, <span class="at">by =</span> <span class="st">&quot;Donor&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb457-3"><a href="advanced-multi-group-comparisons.html#cb457-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Dose =</span> <span class="fu">factor</span>(Dose, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;0&quot;</span>, <span class="st">&quot;0.6&quot;</span>, <span class="st">&quot;1&quot;</span>, <span class="st">&quot;2&quot;</span>, <span class="st">&quot;4&quot;</span>)))</span>
<span id="cb457-4"><a href="advanced-multi-group-comparisons.html#cb457-4" tabindex="-1"></a></span>
<span id="cb457-5"><a href="advanced-multi-group-comparisons.html#cb457-5" tabindex="-1"></a><span class="co"># viewing data</span></span>
<span id="cb457-6"><a href="advanced-multi-group-comparisons.html#cb457-6" tabindex="-1"></a><span class="fu">datatable</span>(biomarker_data)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-4f79b006aac172293a47" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-4f79b006aac172293a47">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80"],["D10","D10","D10","D10","D10","D11","D11","D11","D11","D11","D12","D12","D12","D12","D12","D13","D13","D13","D13","D13","D14","D14","D14","D14","D14","D15","D15","D15","D15","D15","D16","D16","D16","D16","D16","D1","D1","D1","D1","D1","D2","D2","D2","D2","D2","D3","D3","D3","D3","D3","D4","D4","D4","D4","D4","D5","D5","D5","D5","D5","D6","D6","D6","D6","D6","D7","D7","D7","D7","D7","D8","D8","D8","D8","D8","D9","D9","D9","D9","D9"],["0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4"],[3.71808758396052,3.7235585614721,3.70597790168252,3.76553474636298,3.6724253419715,3.83491342970104,3.93073733756289,3.92884403671257,3.80632405739003,3.81762325751143,3.36036427654347,3.27649666564036,3.57167680997093,3.54101915313356,3.10097764772482,3.35755200461808,3.33628338786443,3.31469652565629,3.50589092972996,3.69041709553247,3.31614574229336,3.43562859405209,3.57773093149008,3.87578006306849,3.83086375675176,3.42894634476948,3.49825086752783,3.57288966842058,3.63807383718072,3.6915341649192,3.38681094647222,3.49825086752783,3.58856473740135,3.66220549965362,3.69376571221778,3.76871365703048,3.77820857639809,3.74953426766926,3.84096670448742,3.50589092972996,3.53480866123175,3.60880924267552,3.58856473740135,3.2585189247113,3.75167794568758,3.58014548442338,3.73876783680019,3.61823865559545,3.58736499093646,3.39643353125099,3.52732060787705,3.50842865253186,3.62175885670973,3.23266075679027,4.13996056954546,3.16671544496642,3.18269229751619,3.13422093976063,3.7279204545632,3.799087306074,2.86195536414487,3.08746284125034,3.64961545906341,3.66220549965362,3.64616265715789,3.04614178164472,3.1811025507538,3.27202318906105,3.54101915313356,3.42088657497553,3.37712374912949,3.57531233068744,3.58976348698498,3.83289001416474,3.8972404255748,3.21878116778407,3.30888505741176,3.51853513898218,3.5765221379205,3.7279204545632],[8.68422263328295,8.773963368433559,8.561326136453919,9.710324307072939,7.55028504908841,10.2222394489088,10.1590009172826,9.94747622373011,9.81610366284939,8.03815081026938,9.27052894238072,9.05077375519723,8.99740532549677,8.27635707925924,7.61889934790406,8.15790268278964,8.089053172156429,7.41515020542639,7.37920505996061,4.10265813136374,7.56094417806009,7.5894638929561,7.9047851297316,7.83857365266832,6.50969584193339,9.254131033541951,9.422148885880191,9.165032591502291,9.09402488708305,8.202809492125009,8.68422263328295,8.773963368433559,8.77360079528254,8.60099074240823,7.71163256182322,10.1099481540673,10.3357141430951,9.686097776967401,10.0151641270811,10.1498741117182,8.88962564051876,9.317005526387209,9.03875441393145,8.60099074240823,4.87479696589866,9.23160520814165,9.677085350559841,9.23208509872283,9.583628142323651,9.560179957764181,8.34837407519116,8.351469324267081,8.77360079528254,9.33695280016051,7.70355756192103,3.9429835981871,4.12018602753186,4.24868662687383,5.4110872267986,5.17871464117544,7.43212427264156,6.85112436288675,7.38076407498511,6.96220195703608,8.36382813184704,8.12861303870471,8.0503384381673,7.86530024272912,7.3639595068907,8.90926307937532,11.4238722598684,11.7288813640812,11.4570986742783,10.878112237463,8.52575569282949,7.01077983875324,7.32804655627074,7.03452387533439,7.27537959625362,4.70099449416827],[15.0940690235359,15.1807289234454,15.1191518828101,16.3458996676138,16.7241460980745,15.0028356680746,15.5465951732949,15.4894995973234,16.169128959283,16.0996069083821,13.9520295507825,14.0492691434224,13.1502883461063,14.8168226706095,16.3560077655342,14.30563571243,14.2750718138596,14.2603087281481,14.3170903457307,16.6038859325635,13.9567554773633,14.1371656727442,14.4431119040824,15.1950634099335,16.3613300730261,14.4305440239947,14.6030328264996,14.6583565484435,15.3569594653045,16.5368999542995,14.3291992954213,14.4239037658366,14.5587322032289,15.1950634099335,16.3717050181953,15.1620702745062,15.5056676741312,15.1429673906615,15.4079647745043,16.9008155214109,14.6079451777076,14.6753941971444,13.1497550568455,14.9206710618732,17.2934204566841,14.5987126147926,14.8967320984425,14.6970451993658,15.0929012215711,17.1264130896721,14.542155252148,14.4584290430224,14.8614654668171,15.9922404459517,15.3426552970282,14.1001424218123,13.9841362654046,14.0738746130993,15.2370937904406,16.3510586187167,13.4034767756747,13.4654490662827,13.7572472746771,15.5483372770406,16.2848127161883,13.5861396108168,13.8384840329337,13.8699889688218,14.4670316946933,17.0361832734254,14.1156143059506,14.4239037658366,14.4497857465895,15.0182445232662,15.1484821432167,14.3291992954213,14.1946037445951,13.1529914376826,14.708543993667,16.3449160294487],[0.189033824390017,-0.3040061868901,0.286881147788162,0.298658315564515,1.0703893278914,2.3305584000308,1.37851162325373,1.30451104180995,0.978195629681652,1.64154602908752,0.526068811667588,0.238786859587116,0.713695814843359,1.50589092972996,0.333423733725192,1.24488705912353,-0.3040061868901,1.56559717585423,1.09761079662642,0.855989697308481,1.06350294230616,0.20163386116965,0.895302621333307,1.1243281350022,1.37295209791183,1.27500704749987,0.632268215499513,1.16992500144231,0.910732661902913,1.16992500144231,0.8155754288625729,0.333423733725192,0.748461233004036,0.903038270112912,1.13750352374994,0.367371065648529,0.137503523749935,0.464668267003444,0.910732661902913,0.454175893185802,1.35614381022528,0.298658315564515,0.632268215499513,0.748461233004036,1.40599235967584,0.31034012061215,0.124328135002202,0.53605290024021,0.495695162624069,0.321928094887362,0.879705766282288,0.298658315564515,1.01435529297707,1.19534759832222,2.03210084316702,0.933572638261024,0.20163386116965,2.68706068833989,1.13750352374994,0.84799690655495,1.86393845042397,1.41142624572647,0.992768430768924,1.0071955014042,0.443606651475615,0.650764559116902,0.8718436485093179,1.19534759832222,0.16349873228288,1.17632277264046,1.01435529297707,0.941106310946431,1.04264433740849,0.8718436485093179,1.57531233068744,2.51349074558812,0.604071323668861,0.411426245726465,0.584962500721156,1.47508488294878],[1.73552217729654,1.55581615506164,1.93734439215023,2.08406426478847,2.83390207666916,3.23572705983806,3.24031432933371,3.15218341905505,3.16510798514454,3.48284828306847,1.63226821549951,1.91073266190291,2.35049724708413,3.24031432933371,1.57046293102604,1.32192809488736,1.20789285164133,1.20789285164133,2.75274859140713,2.59931779369823,1.91456452349394,1.72246602447109,1.96347412397489,2.81352468929781,3.27202318906105,2.08746284125034,2.02502879449152,2.37016428054021,2.63923216324928,2.94485844580754,1.83187724119167,1.91073266190291,2.0496307677246,2.64154602908752,2.93168305705981,1.86393845042397,1.790772037862,1.86393845042397,1.88362081628567,2.88362081628567,1.83187724119167,2.0071955014042,2.02147972741045,2.48542682717024,2.98913900736823,2.52606881166759,1.49569516262407,1.85598969730848,1.50080205305716,2.06695024392463,2.47508488294878,2.31034012061215,2.39231742277876,2.99457972421575,3.73226919950145,2.20476675065461,2.34482849699744,1.96347412397489,2.40053792958373,2.85798099512757,1.790772037862,2.11103131238874,2.39506279951758,3.0232553523003,2.16671544496642,1.56559717585423,1.32192809488736,3.59693514238723,1.73552217729654,2.75702324650746,2.53106949272595,2.48284828306847,2.7158933705476,2.99819550315325,3.47118746038699,1.0703893278914,1.33342373372519,1.97819562968165,2.58496250072116,3.08406426478847],[10.4475275409266,10.4320198468125,10.4523853997561,10.7211590148677,11.2458081856384,9.02901108665907,9.22583403553784,9.412739192116421,9.91233549843777,9.316688823436939,9.428025110276749,9.412146396540679,9.620146497339491,10.9260809830959,7.3039633562453,9.247642742898019,9.32424806586749,10.2153509830024,11.2601080099627,10.8557639471917,10.6537676442859,10.6473145095471,10.8149028512154,11.0078962674403,10.9341110109735,9.933439857942711,10.009954550333,10.2267734234482,10.7809497987497,10.646036617491,9.797969484059131,9.8490610025615,10.2090268302306,10.7211590148677,10.8557639471917,9.696480751820189,9.72723977748481,9.49325503512067,10.0461417816447,11.0193335777127,10.534828110721,10.8215188973143,10.8050102485124,11.7900226112658,11.031563370024,9.35887093968562,9.18935511254853,9.033670771393091,9.36430978186441,11.0217850685075,8.283551423174311,8.22211866974429,8.617467465294199,9.414283550503709,8.973180744513019,9.425320822759881,9.308885057411761,9.61402881967061,10.4624204732526,10.7035402591663,10.7662472988462,10.9813603523445,11.5188793273505,11.3227900730377,9.89590846480624,10.3333342781343,10.3711558797312,10.374854737178,10.7157046495212,11.7319596151152,9.92478281821723,9.8490610025615,10.0477374043091,10.6360535917989,9.40224465525408,9.797969484059131,10.0853263897027,10.4360035526972,11.191225812124,10.9299462991725],[44,44,44,44,44,46,46,46,46,46,52,52,52,52,52,58,58,58,58,58,65,65,65,65,65,37,37,37,37,37,25,25,25,25,25,22,22,22,22,22,35,35,35,35,35,27,27,27,27,27,60,60,60,60,60,43,43,43,43,43,38,38,38,38,38,26,26,26,26,26,36,36,36,36,36,40,40,40,40,40],["M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","F","F","F","F","F","F","F","F","F","F","M","M","M","M","M","F","F","F","F","F","M","M","M","M","M","F","F","F","F","F","F","F","F","F","F","M","M","M","M","M","M","M","M","M","M","F","F","F","F","F","F","F","F","F","F"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Donor<\/th>\n      <th>Dose<\/th>\n      <th>IL1B<\/th>\n      <th>IL6<\/th>\n      <th>IL8<\/th>\n      <th>IL10<\/th>\n      <th>TNFa<\/th>\n      <th>VEGF<\/th>\n      <th>Age_yr<\/th>\n      <th>Sex<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4,5,6,7,8,9]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Donor","targets":1},{"name":"Dose","targets":2},{"name":"IL1B","targets":3},{"name":"IL6","targets":4},{"name":"IL8","targets":5},{"name":"IL10","targets":6},{"name":"TNFa","targets":7},{"name":"VEGF","targets":8},{"name":"Age_yr","targets":9},{"name":"Sex","targets":10}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Then, we can demonstrate how to run the two-way ANOVA and what the results look like by running the test on just one of our variables (IL-1<span class="math inline">\(\beta\)</span>).</p>
<div class="sourceCode" id="cb458"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb458-1"><a href="advanced-multi-group-comparisons.html#cb458-1" tabindex="-1"></a><span class="fu">get_anova_table</span>(<span class="fu">anova_test</span>(<span class="at">data =</span> biomarker_data, </span>
<span id="cb458-2"><a href="advanced-multi-group-comparisons.html#cb458-2" tabindex="-1"></a>           <span class="at">dv =</span> IL1B, </span>
<span id="cb458-3"><a href="advanced-multi-group-comparisons.html#cb458-3" tabindex="-1"></a>           <span class="at">wid =</span> Donor, </span>
<span id="cb458-4"><a href="advanced-multi-group-comparisons.html#cb458-4" tabindex="-1"></a>           <span class="at">between =</span> Sex, </span>
<span id="cb458-5"><a href="advanced-multi-group-comparisons.html#cb458-5" tabindex="-1"></a>           <span class="at">within =</span> Dose))</span></code></pre></div>
<pre><code>## ANOVA Table (type III tests)
## 
##     Effect  DFn  DFd     F     p p&lt;.05      ges
## 1      Sex 1.00 14.0 0.014 0.909       0.000488
## 2     Dose 2.42 33.9 6.632 0.002     * 0.192000
## 3 Sex:Dose 2.42 33.9 2.799 0.065       0.091000</code></pre>
<p>The column names are described below:</p>
<ul>
<li><code>Effect</code>: the name of the variable tested</li>
<li><code>DFn</code>: degrees of freedom in the numerator</li>
<li><code>Dfd</code>: degrees of freedom in the denominator</li>
<li><code>F</code>: F distribution test</li>
<li><code>p</code>: p-value</li>
<li><code>p&lt;.05</code>: denotes whether the p-value is significant</li>
<li><code>ges</code>: generalized effect size</li>
</ul>
<p>Based on the table above, there are significant differences in IL-1<span class="math inline">\(\beta\)</span> concentrations based on dose (p-value = 0.02). There are no significant differences in IL-1<span class="math inline">\(\beta\)</span> between the sexes nor are there significant differences in IL-1<span class="math inline">\(\beta\)</span> with an interaction between sex and dose.</p>
<p>Similar to previous modules, we now want to apply our two-way ANOVA to each of our variables of interest. To do this, we can use a for loop that will:</p>
<ol style="list-style-type: decimal">
<li>Loop through each column in the data and apply the test to each column.</li>
<li>Pull out statistics we are interested in (for example, p-value) and bind the results from each column together into a results dataframe.</li>
</ol>
<div class="sourceCode" id="cb460"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb460-1"><a href="advanced-multi-group-comparisons.html#cb460-1" tabindex="-1"></a><span class="co"># Create a vector with the names of the variables you want to run the test on</span></span>
<span id="cb460-2"><a href="advanced-multi-group-comparisons.html#cb460-2" tabindex="-1"></a>endpoints <span class="ot">&lt;-</span> <span class="fu">colnames</span>(biomarker_data <span class="sc">%&gt;%</span> dplyr<span class="sc">::</span><span class="fu">select</span>(IL1B<span class="sc">:</span>VEGF))</span>
<span id="cb460-3"><a href="advanced-multi-group-comparisons.html#cb460-3" tabindex="-1"></a></span>
<span id="cb460-4"><a href="advanced-multi-group-comparisons.html#cb460-4" tabindex="-1"></a><span class="co"># Create data frame to store results</span></span>
<span id="cb460-5"><a href="advanced-multi-group-comparisons.html#cb460-5" tabindex="-1"></a>twoway_aov_res <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Factor =</span> <span class="fu">c</span>(<span class="st">&quot;Dose&quot;</span>, <span class="st">&quot;Sex&quot;</span>, <span class="st">&quot;Sex:Dose&quot;</span>))</span>
<span id="cb460-6"><a href="advanced-multi-group-comparisons.html#cb460-6" tabindex="-1"></a></span>
<span id="cb460-7"><a href="advanced-multi-group-comparisons.html#cb460-7" tabindex="-1"></a><span class="co"># Run for loop</span></span>
<span id="cb460-8"><a href="advanced-multi-group-comparisons.html#cb460-8" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(endpoints)) {</span>
<span id="cb460-9"><a href="advanced-multi-group-comparisons.html#cb460-9" tabindex="-1"></a>  </span>
<span id="cb460-10"><a href="advanced-multi-group-comparisons.html#cb460-10" tabindex="-1"></a>  <span class="co"># Assign a name to the endpoint variable</span></span>
<span id="cb460-11"><a href="advanced-multi-group-comparisons.html#cb460-11" tabindex="-1"></a>  endpoint <span class="ot">&lt;-</span> endpoints[i]</span>
<span id="cb460-12"><a href="advanced-multi-group-comparisons.html#cb460-12" tabindex="-1"></a>  </span>
<span id="cb460-13"><a href="advanced-multi-group-comparisons.html#cb460-13" tabindex="-1"></a>  <span class="co"># Run two-way mixed ANOVA and store results in res_aov</span></span>
<span id="cb460-14"><a href="advanced-multi-group-comparisons.html#cb460-14" tabindex="-1"></a>  res_aov <span class="ot">&lt;-</span> <span class="fu">anova_test</span>(<span class="at">data =</span> biomarker_data,</span>
<span id="cb460-15"><a href="advanced-multi-group-comparisons.html#cb460-15" tabindex="-1"></a>                        <span class="at">dv =</span> <span class="fu">paste0</span>(endpoint),</span>
<span id="cb460-16"><a href="advanced-multi-group-comparisons.html#cb460-16" tabindex="-1"></a>                        <span class="at">wid =</span> Donor,</span>
<span id="cb460-17"><a href="advanced-multi-group-comparisons.html#cb460-17" tabindex="-1"></a>                        <span class="at">between =</span> Sex,</span>
<span id="cb460-18"><a href="advanced-multi-group-comparisons.html#cb460-18" tabindex="-1"></a>                        <span class="at">within =</span> Dose)</span>
<span id="cb460-19"><a href="advanced-multi-group-comparisons.html#cb460-19" tabindex="-1"></a>  </span>
<span id="cb460-20"><a href="advanced-multi-group-comparisons.html#cb460-20" tabindex="-1"></a>  <span class="co"># Extract the results we are interested in (from the ANOVA table)</span></span>
<span id="cb460-21"><a href="advanced-multi-group-comparisons.html#cb460-21" tabindex="-1"></a>  res_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">get_anova_table</span>(res_aov)) <span class="sc">%&gt;%</span></span>
<span id="cb460-22"><a href="advanced-multi-group-comparisons.html#cb460-22" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(Effect, p)) <span class="sc">%&gt;%</span></span>
<span id="cb460-23"><a href="advanced-multi-group-comparisons.html#cb460-23" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="st">&quot;Factor&quot;</span> <span class="ot">=</span> <span class="st">&quot;Effect&quot;</span>)</span>
<span id="cb460-24"><a href="advanced-multi-group-comparisons.html#cb460-24" tabindex="-1"></a>  </span>
<span id="cb460-25"><a href="advanced-multi-group-comparisons.html#cb460-25" tabindex="-1"></a>  <span class="co"># Rename columns in the results dataframe so that the output is more nicely formatted</span></span>
<span id="cb460-26"><a href="advanced-multi-group-comparisons.html#cb460-26" tabindex="-1"></a>  <span class="fu">names</span>(res_df)[<span class="fu">names</span>(res_df) <span class="sc">==</span> <span class="st">&#39;p&#39;</span>] <span class="ot">&lt;-</span> <span class="fu">noquote</span>(<span class="fu">paste0</span>(endpoint))</span>
<span id="cb460-27"><a href="advanced-multi-group-comparisons.html#cb460-27" tabindex="-1"></a>  </span>
<span id="cb460-28"><a href="advanced-multi-group-comparisons.html#cb460-28" tabindex="-1"></a>  <span class="co"># Bind the results to the results dataframe</span></span>
<span id="cb460-29"><a href="advanced-multi-group-comparisons.html#cb460-29" tabindex="-1"></a>  twoway_aov_res <span class="ot">&lt;-</span> <span class="fu">merge</span>(twoway_aov_res, res_df, <span class="at">by =</span> <span class="st">&quot;Factor&quot;</span>, <span class="at">all.y =</span> <span class="cn">TRUE</span>)</span>
<span id="cb460-30"><a href="advanced-multi-group-comparisons.html#cb460-30" tabindex="-1"></a>}</span>
<span id="cb460-31"><a href="advanced-multi-group-comparisons.html#cb460-31" tabindex="-1"></a></span>
<span id="cb460-32"><a href="advanced-multi-group-comparisons.html#cb460-32" tabindex="-1"></a><span class="co"># View results</span></span>
<span id="cb460-33"><a href="advanced-multi-group-comparisons.html#cb460-33" tabindex="-1"></a><span class="fu">datatable</span>(twoway_aov_res)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-aa54af27a9d1eda3f6f4" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-aa54af27a9d1eda3f6f4">{"x":{"filter":"none","vertical":false,"data":[["1","2","3"],["Dose","Sex","Sex:Dose"],[0.002,0.909,0.065],[0.007,0.5590000000000001,0.297],[1.7e-10,0.403,0.603],[0.001,0.157,0.635],[6.36e-08,0.317,0.422],[0.008999999999999999,0.819,0.6899999999999999]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Factor<\/th>\n      <th>IL1B<\/th>\n      <th>IL6<\/th>\n      <th>IL8<\/th>\n      <th>IL10<\/th>\n      <th>TNFa<\/th>\n      <th>VEGF<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Factor","targets":1},{"name":"IL1B","targets":2},{"name":"IL6","targets":3},{"name":"IL8","targets":4},{"name":"IL10","targets":5},{"name":"TNFa","targets":6},{"name":"VEGF","targets":7}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>An appropriate title for this table could be:</p>
<p>“<strong>Figure X. Statistical test results for differences in cytokine concentrations.</strong> A two-way ANOVA was performed using sex and dose as independent variables to test for statistical differences in concentration across 6 cytokines.”</p>
<p>From this table, dose is the only variable with significant differences in concentrations in all 6 biomarkers (p-value &lt; 0.05).</p>
<p>Although we know that dose has significant differences overall, an ANOVA test doesn’t tell us which doses of acrolein differ from each other or the directionality of each biomarker’s change in concentration after exposure to each dose. Therefore, we need to use a post-hoc test. One common post-hoc test following a one-way or two-way ANOVA is a Tukey’s HSD. However, there is no way to pass the output of the <code>anova_test()</code> function to the <code>TukeyHSD()</code> function. A good alternative is a pairwise t-test with a Bonferroni correction. Our data are paired in that there are repeated measures (doses) on each subject.</p>
<div class="sourceCode" id="cb461"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb461-1"><a href="advanced-multi-group-comparisons.html#cb461-1" tabindex="-1"></a><span class="co"># Create data frame to store results</span></span>
<span id="cb461-2"><a href="advanced-multi-group-comparisons.html#cb461-2" tabindex="-1"></a>twoway_aov_pairedt <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Comparison =</span> <span class="fu">c</span>(<span class="st">&quot;0_0.6&quot;</span>, <span class="st">&quot;0_1&quot;</span>, <span class="st">&quot;0_2&quot;</span>, <span class="st">&quot;0_4&quot;</span>, <span class="st">&quot;0.6_1&quot;</span>, <span class="st">&quot;0.6_2&quot;</span>, <span class="st">&quot;0.6_4&quot;</span>, <span class="st">&quot;1_2&quot;</span>, <span class="st">&quot;1_4&quot;</span>, <span class="st">&quot;2_4&quot;</span>))  </span>
<span id="cb461-3"><a href="advanced-multi-group-comparisons.html#cb461-3" tabindex="-1"></a></span>
<span id="cb461-4"><a href="advanced-multi-group-comparisons.html#cb461-4" tabindex="-1"></a><span class="co"># Run for loop</span></span>
<span id="cb461-5"><a href="advanced-multi-group-comparisons.html#cb461-5" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(endpoints)) {</span>
<span id="cb461-6"><a href="advanced-multi-group-comparisons.html#cb461-6" tabindex="-1"></a>  </span>
<span id="cb461-7"><a href="advanced-multi-group-comparisons.html#cb461-7" tabindex="-1"></a>  <span class="co"># Assign a name to the endpoint variable.</span></span>
<span id="cb461-8"><a href="advanced-multi-group-comparisons.html#cb461-8" tabindex="-1"></a>  endpoint <span class="ot">&lt;-</span> endpoints[i]</span>
<span id="cb461-9"><a href="advanced-multi-group-comparisons.html#cb461-9" tabindex="-1"></a>  </span>
<span id="cb461-10"><a href="advanced-multi-group-comparisons.html#cb461-10" tabindex="-1"></a>  <span class="co"># Run pairwise t-tests</span></span>
<span id="cb461-11"><a href="advanced-multi-group-comparisons.html#cb461-11" tabindex="-1"></a>  res_df <span class="ot">&lt;-</span> biomarker_data <span class="sc">%&gt;%</span></span>
<span id="cb461-12"><a href="advanced-multi-group-comparisons.html#cb461-12" tabindex="-1"></a>    <span class="fu">pairwise_t_test</span>(<span class="fu">as.formula</span>(<span class="fu">paste0</span>(<span class="fu">paste0</span>(endpoint), <span class="st">&quot;~&quot;</span>, <span class="st">&quot;Dose&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)),</span>
<span id="cb461-13"><a href="advanced-multi-group-comparisons.html#cb461-13" tabindex="-1"></a>                    <span class="at">paired =</span> <span class="cn">TRUE</span>, </span>
<span id="cb461-14"><a href="advanced-multi-group-comparisons.html#cb461-14" tabindex="-1"></a>                    <span class="at">p.adjust.method =</span> <span class="st">&quot;bonferroni&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb461-15"><a href="advanced-multi-group-comparisons.html#cb461-15" tabindex="-1"></a>  <span class="fu">unite</span>(Comparison, group1, group2, <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>, <span class="at">remove =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb461-16"><a href="advanced-multi-group-comparisons.html#cb461-16" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Comparison, p.adj) </span>
<span id="cb461-17"><a href="advanced-multi-group-comparisons.html#cb461-17" tabindex="-1"></a>  </span>
<span id="cb461-18"><a href="advanced-multi-group-comparisons.html#cb461-18" tabindex="-1"></a>  <span class="co"># Rename columns in the results data frame so that the output is more nicely formatted.</span></span>
<span id="cb461-19"><a href="advanced-multi-group-comparisons.html#cb461-19" tabindex="-1"></a>  <span class="fu">names</span>(res_df)[<span class="fu">names</span>(res_df) <span class="sc">==</span> <span class="st">&#39;p.adj&#39;</span>] <span class="ot">&lt;-</span> <span class="fu">noquote</span>(<span class="fu">paste0</span>(endpoint))</span>
<span id="cb461-20"><a href="advanced-multi-group-comparisons.html#cb461-20" tabindex="-1"></a>  </span>
<span id="cb461-21"><a href="advanced-multi-group-comparisons.html#cb461-21" tabindex="-1"></a>  <span class="co"># Bind the results to the results data frame. </span></span>
<span id="cb461-22"><a href="advanced-multi-group-comparisons.html#cb461-22" tabindex="-1"></a>  twoway_aov_pairedt <span class="ot">&lt;-</span> <span class="fu">merge</span>(twoway_aov_pairedt, res_df, <span class="at">by =</span> <span class="st">&quot;Comparison&quot;</span>, <span class="at">all.y =</span> <span class="cn">TRUE</span>)</span>
<span id="cb461-23"><a href="advanced-multi-group-comparisons.html#cb461-23" tabindex="-1"></a>}</span>
<span id="cb461-24"><a href="advanced-multi-group-comparisons.html#cb461-24" tabindex="-1"></a></span>
<span id="cb461-25"><a href="advanced-multi-group-comparisons.html#cb461-25" tabindex="-1"></a><span class="co"># View results</span></span>
<span id="cb461-26"><a href="advanced-multi-group-comparisons.html#cb461-26" tabindex="-1"></a><span class="fu">datatable</span>(twoway_aov_pairedt)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-ba258ad33c54d80581b2" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ba258ad33c54d80581b2">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10"],["0.6_1","0.6_2","0.6_4","0_0.6","0_1","0_2","0_4","1_2","1_4","2_4"],[0.62,0.454,0.238,0.033,0.07000000000000001,0.105,0.056,1,1,1],[1,1,0.112,1,1,1,0.121,1,0.132,0.081],[1,0.000796,3e-07,0.109,1,6.770000000000001e-05,4.97e-08,5.97e-05,4.75e-06,0.001],[0.094,0.138,0.011,0.003,1,1,1,1,1,1],[0.447,0.001,0.000122,1,1,0.037,0.000354,0.977,0.032,1],[0.067,0.000111,0.603,1,0.059,0.000115,0.38,0.000127,1,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Comparison<\/th>\n      <th>IL1B<\/th>\n      <th>IL6<\/th>\n      <th>IL8<\/th>\n      <th>IL10<\/th>\n      <th>TNFa<\/th>\n      <th>VEGF<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Comparison","targets":1},{"name":"IL1B","targets":2},{"name":"IL6","targets":3},{"name":"IL8","targets":4},{"name":"IL10","targets":5},{"name":"TNFa","targets":6},{"name":"VEGF","targets":7}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>An appropriate title for this table could be:</p>
<p>“<strong>Figure X. Post hoc testing for differences in cytokine concentrations.</strong> Paired t-tests were run as a post hoc test using dose as an independent variable to test for statistical differences in concentration across 6 cytokines.”</p>
<p>Note that this table and the two-way ANOVA table would likely be put into supplemental material for a publication. Before including this table in supplemental material, it would be best to clean it up (make the two comparison groups more clear, round all results to the same number of decimals) as demonstrated in <strong>TAME 2.0 Module 4.5 Multi-Group Comparisons and Visualizations</strong>.</p>
<div id="answer-to-environmental-health-question-1-7" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 1<a href="advanced-multi-group-comparisons.html#answer-to-environmental-health-question-1-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #1</strong></em>: Are there significant differences in inflammatory biomarker concentrations between sex and different doses of acrolein?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: Based on the two-way ANOVA and post-hoc t-tests, there are only significant differences in cytokine concentrations based on dose (p adj &lt; 0.05). All biomarkers, with the exception of IL-6, had at least 1 significantly different concentration when comparing doses.</p>
</div>
</div>
<div id="visualizing-two-way-anova-results" class="section level3 hasAnchor">
<h3>Visualizing Two-Way ANOVA Results<a href="advanced-multi-group-comparisons.html#visualizing-two-way-anova-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Since our overall p-values associated with dose were significant for a number of mediators, we will proceed with creating our final figures with our endpoints by dose, showing the overall two-way ANOVA p-value and the pairwise comparisons from our post hoc paired pairwise t-tests.</p>
<p>To facilitate plotting in a faceted panel, we’ll first pivot our <code>biomarker_data</code> dataframe longer.</p>
<div class="sourceCode" id="cb462"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb462-1"><a href="advanced-multi-group-comparisons.html#cb462-1" tabindex="-1"></a>biomarker_data_long <span class="ot">&lt;-</span> biomarker_data <span class="sc">%&gt;%</span></span>
<span id="cb462-2"><a href="advanced-multi-group-comparisons.html#cb462-2" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(Age_yr, Sex)) <span class="sc">%&gt;%</span></span>
<span id="cb462-3"><a href="advanced-multi-group-comparisons.html#cb462-3" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(Donor, Dose), <span class="at">names_to =</span> <span class="st">&quot;Variable&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;Value&quot;</span>)</span>
<span id="cb462-4"><a href="advanced-multi-group-comparisons.html#cb462-4" tabindex="-1"></a></span>
<span id="cb462-5"><a href="advanced-multi-group-comparisons.html#cb462-5" tabindex="-1"></a><span class="fu">datatable</span>(biomarker_data_long)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-6fc8f6e54b854e8ce3e7" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-6fc8f6e54b854e8ce3e7">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291","292","293","294","295","296","297","298","299","300","301","302","303","304","305","306","307","308","309","310","311","312","313","314","315","316","317","318","319","320","321","322","323","324","325","326","327","328","329","330","331","332","333","334","335","336","337","338","339","340","341","342","343","344","345","346","347","348","349","350","351","352","353","354","355","356","357","358","359","360","361","362","363","364","365","366","367","368","369","370","371","372","373","374","375","376","377","378","379","380","381","382","383","384","385","386","387","388","389","390","391","392","393","394","395","396","397","398","399","400","401","402","403","404","405","406","407","408","409","410","411","412","413","414","415","416","417","418","419","420","421","422","423","424","425","426","427","428","429","430","431","432","433","434","435","436","437","438","439","440","441","442","443","444","445","446","447","448","449","450","451","452","453","454","455","456","457","458","459","460","461","462","463","464","465","466","467","468","469","470","471","472","473","474","475","476","477","478","479","480"],["D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D10","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D11","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D12","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D13","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D14","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D15","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D16","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D1","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D2","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D3","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D4","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D5","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D6","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D7","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D8","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9","D9"],["0","0","0","0","0","0","0.6","0.6","0.6","0.6","0.6","0.6","1","1","1","1","1","1","2","2","2","2","2","2","4","4","4","4","4","4","0","0","0","0","0","0","0.6","0.6","0.6","0.6","0.6","0.6","1","1","1","1","1","1","2","2","2","2","2","2","4","4","4","4","4","4","0","0","0","0","0","0","0.6","0.6","0.6","0.6","0.6","0.6","1","1","1","1","1","1","2","2","2","2","2","2","4","4","4","4","4","4","0","0","0","0","0","0","0.6","0.6","0.6","0.6","0.6","0.6","1","1","1","1","1","1","2","2","2","2","2","2","4","4","4","4","4","4","0","0","0","0","0","0","0.6","0.6","0.6","0.6","0.6","0.6","1","1","1","1","1","1","2","2","2","2","2","2","4","4","4","4","4","4","0","0","0","0","0","0","0.6","0.6","0.6","0.6","0.6","0.6","1","1","1","1","1","1","2","2","2","2","2","2","4","4","4","4","4","4","0","0","0","0","0","0","0.6","0.6","0.6","0.6","0.6","0.6","1","1","1","1","1","1","2","2","2","2","2","2","4","4","4","4","4","4","0","0","0","0","0","0","0.6","0.6","0.6","0.6","0.6","0.6","1","1","1","1","1","1","2","2","2","2","2","2","4","4","4","4","4","4","0","0","0","0","0","0","0.6","0.6","0.6","0.6","0.6","0.6","1","1","1","1","1","1","2","2","2","2","2","2","4","4","4","4","4","4","0","0","0","0","0","0","0.6","0.6","0.6","0.6","0.6","0.6","1","1","1","1","1","1","2","2","2","2","2","2","4","4","4","4","4","4","0","0","0","0","0","0","0.6","0.6","0.6","0.6","0.6","0.6","1","1","1","1","1","1","2","2","2","2","2","2","4","4","4","4","4","4","0","0","0","0","0","0","0.6","0.6","0.6","0.6","0.6","0.6","1","1","1","1","1","1","2","2","2","2","2","2","4","4","4","4","4","4","0","0","0","0","0","0","0.6","0.6","0.6","0.6","0.6","0.6","1","1","1","1","1","1","2","2","2","2","2","2","4","4","4","4","4","4","0","0","0","0","0","0","0.6","0.6","0.6","0.6","0.6","0.6","1","1","1","1","1","1","2","2","2","2","2","2","4","4","4","4","4","4","0","0","0","0","0","0","0.6","0.6","0.6","0.6","0.6","0.6","1","1","1","1","1","1","2","2","2","2","2","2","4","4","4","4","4","4","0","0","0","0","0","0","0.6","0.6","0.6","0.6","0.6","0.6","1","1","1","1","1","1","2","2","2","2","2","2","4","4","4","4","4","4"],["IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF"],[3.71808758396052,8.68422263328295,15.0940690235359,0.189033824390017,1.73552217729654,10.4475275409266,3.7235585614721,8.773963368433559,15.1807289234454,-0.3040061868901,1.55581615506164,10.4320198468125,3.70597790168252,8.561326136453919,15.1191518828101,0.286881147788162,1.93734439215023,10.4523853997561,3.76553474636298,9.710324307072939,16.3458996676138,0.298658315564515,2.08406426478847,10.7211590148677,3.6724253419715,7.55028504908841,16.7241460980745,1.0703893278914,2.83390207666916,11.2458081856384,3.83491342970104,10.2222394489088,15.0028356680746,2.3305584000308,3.23572705983806,9.02901108665907,3.93073733756289,10.1590009172826,15.5465951732949,1.37851162325373,3.24031432933371,9.22583403553784,3.92884403671257,9.94747622373011,15.4894995973234,1.30451104180995,3.15218341905505,9.412739192116421,3.80632405739003,9.81610366284939,16.169128959283,0.978195629681652,3.16510798514454,9.91233549843777,3.81762325751143,8.03815081026938,16.0996069083821,1.64154602908752,3.48284828306847,9.316688823436939,3.36036427654347,9.27052894238072,13.9520295507825,0.526068811667588,1.63226821549951,9.428025110276749,3.27649666564036,9.05077375519723,14.0492691434224,0.238786859587116,1.91073266190291,9.412146396540679,3.57167680997093,8.99740532549677,13.1502883461063,0.713695814843359,2.35049724708413,9.620146497339491,3.54101915313356,8.27635707925924,14.8168226706095,1.50589092972996,3.24031432933371,10.9260809830959,3.10097764772482,7.61889934790406,16.3560077655342,0.333423733725192,1.57046293102604,7.3039633562453,3.35755200461808,8.15790268278964,14.30563571243,1.24488705912353,1.32192809488736,9.247642742898019,3.33628338786443,8.089053172156429,14.2750718138596,-0.3040061868901,1.20789285164133,9.32424806586749,3.31469652565629,7.41515020542639,14.2603087281481,1.56559717585423,1.20789285164133,10.2153509830024,3.50589092972996,7.37920505996061,14.3170903457307,1.09761079662642,2.75274859140713,11.2601080099627,3.69041709553247,4.10265813136374,16.6038859325635,0.855989697308481,2.59931779369823,10.8557639471917,3.31614574229336,7.56094417806009,13.9567554773633,1.06350294230616,1.91456452349394,10.6537676442859,3.43562859405209,7.5894638929561,14.1371656727442,0.20163386116965,1.72246602447109,10.6473145095471,3.57773093149008,7.9047851297316,14.4431119040824,0.895302621333307,1.96347412397489,10.8149028512154,3.87578006306849,7.83857365266832,15.1950634099335,1.1243281350022,2.81352468929781,11.0078962674403,3.83086375675176,6.50969584193339,16.3613300730261,1.37295209791183,3.27202318906105,10.9341110109735,3.42894634476948,9.254131033541951,14.4305440239947,1.27500704749987,2.08746284125034,9.933439857942711,3.49825086752783,9.422148885880191,14.6030328264996,0.632268215499513,2.02502879449152,10.009954550333,3.57288966842058,9.165032591502291,14.6583565484435,1.16992500144231,2.37016428054021,10.2267734234482,3.63807383718072,9.09402488708305,15.3569594653045,0.910732661902913,2.63923216324928,10.7809497987497,3.6915341649192,8.202809492125009,16.5368999542995,1.16992500144231,2.94485844580754,10.646036617491,3.38681094647222,8.68422263328295,14.3291992954213,0.8155754288625729,1.83187724119167,9.797969484059131,3.49825086752783,8.773963368433559,14.4239037658366,0.333423733725192,1.91073266190291,9.8490610025615,3.58856473740135,8.77360079528254,14.5587322032289,0.748461233004036,2.0496307677246,10.2090268302306,3.66220549965362,8.60099074240823,15.1950634099335,0.903038270112912,2.64154602908752,10.7211590148677,3.69376571221778,7.71163256182322,16.3717050181953,1.13750352374994,2.93168305705981,10.8557639471917,3.76871365703048,10.1099481540673,15.1620702745062,0.367371065648529,1.86393845042397,9.696480751820189,3.77820857639809,10.3357141430951,15.5056676741312,0.137503523749935,1.790772037862,9.72723977748481,3.74953426766926,9.686097776967401,15.1429673906615,0.464668267003444,1.86393845042397,9.49325503512067,3.84096670448742,10.0151641270811,15.4079647745043,0.910732661902913,1.88362081628567,10.0461417816447,3.50589092972996,10.1498741117182,16.9008155214109,0.454175893185802,2.88362081628567,11.0193335777127,3.53480866123175,8.88962564051876,14.6079451777076,1.35614381022528,1.83187724119167,10.534828110721,3.60880924267552,9.317005526387209,14.6753941971444,0.298658315564515,2.0071955014042,10.8215188973143,3.58856473740135,9.03875441393145,13.1497550568455,0.632268215499513,2.02147972741045,10.8050102485124,3.2585189247113,8.60099074240823,14.9206710618732,0.748461233004036,2.48542682717024,11.7900226112658,3.75167794568758,4.87479696589866,17.2934204566841,1.40599235967584,2.98913900736823,11.031563370024,3.58014548442338,9.23160520814165,14.5987126147926,0.31034012061215,2.52606881166759,9.35887093968562,3.73876783680019,9.677085350559841,14.8967320984425,0.124328135002202,1.49569516262407,9.18935511254853,3.61823865559545,9.23208509872283,14.6970451993658,0.53605290024021,1.85598969730848,9.033670771393091,3.58736499093646,9.583628142323651,15.0929012215711,0.495695162624069,1.50080205305716,9.36430978186441,3.39643353125099,9.560179957764181,17.1264130896721,0.321928094887362,2.06695024392463,11.0217850685075,3.52732060787705,8.34837407519116,14.542155252148,0.879705766282288,2.47508488294878,8.283551423174311,3.50842865253186,8.351469324267081,14.4584290430224,0.298658315564515,2.31034012061215,8.22211866974429,3.62175885670973,8.77360079528254,14.8614654668171,1.01435529297707,2.39231742277876,8.617467465294199,3.23266075679027,9.33695280016051,15.9922404459517,1.19534759832222,2.99457972421575,9.414283550503709,4.13996056954546,7.70355756192103,15.3426552970282,2.03210084316702,3.73226919950145,8.973180744513019,3.16671544496642,3.9429835981871,14.1001424218123,0.933572638261024,2.20476675065461,9.425320822759881,3.18269229751619,4.12018602753186,13.9841362654046,0.20163386116965,2.34482849699744,9.308885057411761,3.13422093976063,4.24868662687383,14.0738746130993,2.68706068833989,1.96347412397489,9.61402881967061,3.7279204545632,5.4110872267986,15.2370937904406,1.13750352374994,2.40053792958373,10.4624204732526,3.799087306074,5.17871464117544,16.3510586187167,0.84799690655495,2.85798099512757,10.7035402591663,2.86195536414487,7.43212427264156,13.4034767756747,1.86393845042397,1.790772037862,10.7662472988462,3.08746284125034,6.85112436288675,13.4654490662827,1.41142624572647,2.11103131238874,10.9813603523445,3.64961545906341,7.38076407498511,13.7572472746771,0.992768430768924,2.39506279951758,11.5188793273505,3.66220549965362,6.96220195703608,15.5483372770406,1.0071955014042,3.0232553523003,11.3227900730377,3.64616265715789,8.36382813184704,16.2848127161883,0.443606651475615,2.16671544496642,9.89590846480624,3.04614178164472,8.12861303870471,13.5861396108168,0.650764559116902,1.56559717585423,10.3333342781343,3.1811025507538,8.0503384381673,13.8384840329337,0.8718436485093179,1.32192809488736,10.3711558797312,3.27202318906105,7.86530024272912,13.8699889688218,1.19534759832222,3.59693514238723,10.374854737178,3.54101915313356,7.3639595068907,14.4670316946933,0.16349873228288,1.73552217729654,10.7157046495212,3.42088657497553,8.90926307937532,17.0361832734254,1.17632277264046,2.75702324650746,11.7319596151152,3.37712374912949,11.4238722598684,14.1156143059506,1.01435529297707,2.53106949272595,9.92478281821723,3.57531233068744,11.7288813640812,14.4239037658366,0.941106310946431,2.48284828306847,9.8490610025615,3.58976348698498,11.4570986742783,14.4497857465895,1.04264433740849,2.7158933705476,10.0477374043091,3.83289001416474,10.878112237463,15.0182445232662,0.8718436485093179,2.99819550315325,10.6360535917989,3.8972404255748,8.52575569282949,15.1484821432167,1.57531233068744,3.47118746038699,9.40224465525408,3.21878116778407,7.01077983875324,14.3291992954213,2.51349074558812,1.0703893278914,9.797969484059131,3.30888505741176,7.32804655627074,14.1946037445951,0.604071323668861,1.33342373372519,10.0853263897027,3.51853513898218,7.03452387533439,13.1529914376826,0.411426245726465,1.97819562968165,10.4360035526972,3.5765221379205,7.27537959625362,14.708543993667,0.584962500721156,2.58496250072116,11.191225812124,3.7279204545632,4.70099449416827,16.3449160294487,1.47508488294878,3.08406426478847,10.9299462991725]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Donor<\/th>\n      <th>Dose<\/th>\n      <th>Variable<\/th>\n      <th>Value<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":4},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Donor","targets":1},{"name":"Dose","targets":2},{"name":"Variable","targets":3},{"name":"Value","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Then, we will create an annotation dataframe for adding our overall two-way ANOVA p-values. This dataframe needs to contain a column for our variables (to match with our variable column in our <code>biomarker_data_long</code> dataframe) and the p-value for annotation. We can extract these from our <code>two_way_aov_res</code> dataframe generated above.</p>
<div class="sourceCode" id="cb463"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb463-1"><a href="advanced-multi-group-comparisons.html#cb463-1" tabindex="-1"></a>overall_dose_pvals <span class="ot">&lt;-</span> twoway_aov_res <span class="sc">%&gt;%</span></span>
<span id="cb463-2"><a href="advanced-multi-group-comparisons.html#cb463-2" tabindex="-1"></a>  <span class="co"># Transpose dataframe</span></span>
<span id="cb463-3"><a href="advanced-multi-group-comparisons.html#cb463-3" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&quot;Factor&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb463-4"><a href="advanced-multi-group-comparisons.html#cb463-4" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb463-5"><a href="advanced-multi-group-comparisons.html#cb463-5" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;Variable&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb463-6"><a href="advanced-multi-group-comparisons.html#cb463-6" tabindex="-1"></a>  <span class="co"># Keep only the dose results and rename them to p-value</span></span>
<span id="cb463-7"><a href="advanced-multi-group-comparisons.html#cb463-7" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">c</span>(Variable, Dose)) <span class="sc">%&gt;%</span></span>
<span id="cb463-8"><a href="advanced-multi-group-comparisons.html#cb463-8" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at">P Value</span><span class="st">`</span> <span class="ot">=</span> Dose)</span>
<span id="cb463-9"><a href="advanced-multi-group-comparisons.html#cb463-9" tabindex="-1"></a></span>
<span id="cb463-10"><a href="advanced-multi-group-comparisons.html#cb463-10" tabindex="-1"></a><span class="fu">datatable</span>(overall_dose_pvals)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-fc5fd0fad487551e38b9" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-fc5fd0fad487551e38b9">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6"],["IL1B","IL6","IL8","IL10","TNFa","VEGF"],[0.002,0.007,1.7e-10,0.001,6.36e-08,0.008999999999999999]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Variable<\/th>\n      <th>P Value<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Variable","targets":1},{"name":"P Value","targets":2}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>We now have our p-values for each biomarker. Next, we’ll make a column where our p-values are formatted with “p =” for annotation on the graph.</p>
<div class="sourceCode" id="cb464"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb464-1"><a href="advanced-multi-group-comparisons.html#cb464-1" tabindex="-1"></a>overall_dose_pvals <span class="ot">&lt;-</span> overall_dose_pvals <span class="sc">%&gt;%</span></span>
<span id="cb464-2"><a href="advanced-multi-group-comparisons.html#cb464-2" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">P Value</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">formatC</span>(<span class="st">`</span><span class="at">P Value</span><span class="st">`</span>, <span class="at">format =</span> <span class="st">&quot;e&quot;</span>, <span class="at">digits =</span> <span class="dv">2</span>),</span>
<span id="cb464-3"><a href="advanced-multi-group-comparisons.html#cb464-3" tabindex="-1"></a>         <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">&quot;p = &quot;</span>, <span class="st">`</span><span class="at">P Value</span><span class="st">`</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)) </span>
<span id="cb464-4"><a href="advanced-multi-group-comparisons.html#cb464-4" tabindex="-1"></a></span>
<span id="cb464-5"><a href="advanced-multi-group-comparisons.html#cb464-5" tabindex="-1"></a><span class="fu">datatable</span>(overall_dose_pvals)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-d8f4ec423c8a31cd4830" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-d8f4ec423c8a31cd4830">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6"],["IL1B","IL6","IL8","IL10","TNFa","VEGF"],["2.00e-03","7.00e-03","1.70e-10","1.00e-03","6.36e-08","9.00e-03"],["p = 2.00e-03","p = 7.00e-03","p = 1.70e-10","p = 1.00e-03","p = 6.36e-08","p = 9.00e-03"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Variable<\/th>\n      <th>P Value<\/th>\n      <th>label<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Variable","targets":1},{"name":"P Value","targets":2},{"name":"label","targets":3}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Finally, we’ll add a column indicating where to add the labels on the y-axis. This will be different for each variable because each variable is on a different scale. We can approach this by computing the maximum value of each variable, then increasing that by 10% to add some space on top of the points.</p>
<div class="sourceCode" id="cb465"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb465-1"><a href="advanced-multi-group-comparisons.html#cb465-1" tabindex="-1"></a>sig_labs_y <span class="ot">&lt;-</span> biomarker_data <span class="sc">%&gt;%</span></span>
<span id="cb465-2"><a href="advanced-multi-group-comparisons.html#cb465-2" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(IL1B<span class="sc">:</span>VEGF, \(x) <span class="fu">max</span>(x))) <span class="sc">%&gt;%</span></span>
<span id="cb465-3"><a href="advanced-multi-group-comparisons.html#cb465-3" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb465-4"><a href="advanced-multi-group-comparisons.html#cb465-4" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;Variable&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb465-5"><a href="advanced-multi-group-comparisons.html#cb465-5" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">&quot;y_pos&quot;</span> <span class="ot">=</span> <span class="st">&quot;V1&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb465-6"><a href="advanced-multi-group-comparisons.html#cb465-6" tabindex="-1"></a>  <span class="co"># moving the significance asterisks higher on the y axis</span></span>
<span id="cb465-7"><a href="advanced-multi-group-comparisons.html#cb465-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">y_pos =</span> y_pos <span class="sc">*</span> <span class="fl">1.1</span>)</span>
<span id="cb465-8"><a href="advanced-multi-group-comparisons.html#cb465-8" tabindex="-1"></a></span>
<span id="cb465-9"><a href="advanced-multi-group-comparisons.html#cb465-9" tabindex="-1"></a>sig_labs_y</span></code></pre></div>
<pre><code>##   Variable     y_pos
## 1     IL1B  4.553957
## 2      IL6 12.901770
## 3      IL8 19.022763
## 4     IL10  2.955767
## 5     TNFa  4.105496
## 6     VEGF 12.969025</code></pre>
<div class="sourceCode" id="cb467"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb467-1"><a href="advanced-multi-group-comparisons.html#cb467-1" tabindex="-1"></a>overall_dose_pvals <span class="ot">&lt;-</span> overall_dose_pvals <span class="sc">%&gt;%</span></span>
<span id="cb467-2"><a href="advanced-multi-group-comparisons.html#cb467-2" tabindex="-1"></a>  <span class="fu">left_join</span>(sig_labs_y, <span class="at">by =</span> <span class="st">&quot;Variable&quot;</span>)</span>
<span id="cb467-3"><a href="advanced-multi-group-comparisons.html#cb467-3" tabindex="-1"></a></span>
<span id="cb467-4"><a href="advanced-multi-group-comparisons.html#cb467-4" tabindex="-1"></a><span class="fu">datatable</span>(overall_dose_pvals)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-37481a384870b93f33b9" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-37481a384870b93f33b9">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6"],["IL1B","IL6","IL8","IL10","TNFa","VEGF"],["2.00e-03","7.00e-03","1.70e-10","1.00e-03","6.36e-08","9.00e-03"],["p = 2.00e-03","p = 7.00e-03","p = 1.70e-10","p = 1.00e-03","p = 6.36e-08","p = 9.00e-03"],[4.553956626500006,12.90176950048932,19.02276250235251,2.955766757173879,4.105496119451596,12.96902487239238]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Variable<\/th>\n      <th>P Value<\/th>\n      <th>label<\/th>\n      <th>y_pos<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":4},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Variable","targets":1},{"name":"P Value","targets":2},{"name":"label","targets":3},{"name":"y_pos","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Now, we’ll use the <code>biomarker_data</code> dataframe to plot our individual points and boxplots (similar to the plotting demonstrated in previous TAME Chapter 4 modules) and our <code>overall_dose_pvals</code> dataframe to add our p value annotation.</p>
<div class="sourceCode" id="cb468"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb468-1"><a href="advanced-multi-group-comparisons.html#cb468-1" tabindex="-1"></a><span class="co"># Create clean labels for the graph titles</span></span>
<span id="cb468-2"><a href="advanced-multi-group-comparisons.html#cb468-2" tabindex="-1"></a>new_labels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;IL10&quot;</span> <span class="ot">=</span> <span class="st">&quot;IL-10&quot;</span>, <span class="st">&quot;IL1B&quot;</span> <span class="ot">=</span> <span class="st">&quot;IL-1\u03B2 &quot;</span>, <span class="st">&quot;IL6&quot;</span> <span class="ot">=</span> <span class="st">&quot;IL-6&quot;</span>, <span class="st">&quot;IL8&quot;</span> <span class="ot">=</span> <span class="st">&quot;IL-8&quot;</span>, </span>
<span id="cb468-3"><a href="advanced-multi-group-comparisons.html#cb468-3" tabindex="-1"></a>                <span class="st">&quot;TNFa&quot;</span> <span class="ot">=</span> <span class="st">&quot;TNF-\u03b1&quot;</span>, <span class="st">&quot;VEGF&quot;</span> <span class="ot">=</span> <span class="st">&quot;VEGF&quot;</span>)</span>
<span id="cb468-4"><a href="advanced-multi-group-comparisons.html#cb468-4" tabindex="-1"></a></span>
<span id="cb468-5"><a href="advanced-multi-group-comparisons.html#cb468-5" tabindex="-1"></a><span class="co"># Make graph</span></span>
<span id="cb468-6"><a href="advanced-multi-group-comparisons.html#cb468-6" tabindex="-1"></a><span class="fu">ggplot</span>(biomarker_data_long, <span class="fu">aes</span>(<span class="at">x =</span> Dose, <span class="at">y =</span> Value)) <span class="sc">+</span></span>
<span id="cb468-7"><a href="advanced-multi-group-comparisons.html#cb468-7" tabindex="-1"></a>  <span class="co"># outlier.shape = NA removes outliers</span></span>
<span id="cb468-8"><a href="advanced-multi-group-comparisons.html#cb468-8" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> Dose), <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb468-9"><a href="advanced-multi-group-comparisons.html#cb468-9" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">1.5</span>, <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="fl">0.15</span>), <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb468-10"><a href="advanced-multi-group-comparisons.html#cb468-10" tabindex="-1"></a>  <span class="co"># Add label</span></span>
<span id="cb468-11"><a href="advanced-multi-group-comparisons.html#cb468-11" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> overall_dose_pvals, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fl">1.3</span>, <span class="at">y =</span> y_pos, <span class="at">label =</span> label,</span>
<span id="cb468-12"><a href="advanced-multi-group-comparisons.html#cb468-12" tabindex="-1"></a>                                       <span class="at">size =</span> <span class="dv">5</span>)) <span class="sc">+</span></span>
<span id="cb468-13"><a href="advanced-multi-group-comparisons.html#cb468-13" tabindex="-1"></a>  <span class="co"># Adding padding y axis </span></span>
<span id="cb468-14"><a href="advanced-multi-group-comparisons.html#cb468-14" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.1</span>))) <span class="sc">+</span></span>
<span id="cb468-15"><a href="advanced-multi-group-comparisons.html#cb468-15" tabindex="-1"></a></span>
<span id="cb468-16"><a href="advanced-multi-group-comparisons.html#cb468-16" tabindex="-1"></a>  <span class="co"># Faceting by each biomarker</span></span>
<span id="cb468-17"><a href="advanced-multi-group-comparisons.html#cb468-17" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Variable, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">scales =</span> <span class="st">&quot;free_y&quot;</span>, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">variable =</span> new_labels)) <span class="sc">+</span></span>
<span id="cb468-18"><a href="advanced-multi-group-comparisons.html#cb468-18" tabindex="-1"></a>  </span>
<span id="cb468-19"><a href="advanced-multi-group-comparisons.html#cb468-19" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>, <span class="co"># Removing legend</span></span>
<span id="cb468-20"><a href="advanced-multi-group-comparisons.html#cb468-20" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">&quot;bold&quot;</span>, <span class="at">size =</span> <span class="fu">rel</span>(<span class="fl">1.3</span>)),</span>
<span id="cb468-21"><a href="advanced-multi-group-comparisons.html#cb468-21" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.75</span>),</span>
<span id="cb468-22"><a href="advanced-multi-group-comparisons.html#cb468-22" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="dv">2</span>),</span>
<span id="cb468-23"><a href="advanced-multi-group-comparisons.html#cb468-23" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb468-24"><a href="advanced-multi-group-comparisons.html#cb468-24" tabindex="-1"></a>        <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>)) <span class="sc">+</span> </span>
<span id="cb468-25"><a href="advanced-multi-group-comparisons.html#cb468-25" tabindex="-1"></a>  </span>
<span id="cb468-26"><a href="advanced-multi-group-comparisons.html#cb468-26" tabindex="-1"></a>  <span class="co"># Changing axes labels</span></span>
<span id="cb468-27"><a href="advanced-multi-group-comparisons.html#cb468-27" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Acrolein (ppm)&quot;</span>, <span class="at">y =</span> <span class="fu">expression</span>(<span class="fu">bold</span>(Log[<span class="dv">2</span>]<span class="sc">*</span><span class="st">&quot;(Concentration (pg/ml))&quot;</span>)))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-373-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>It’s a bit more difficult to add the pairwise t test results to the boxplots comparing each treatment group to each other as was done similarly in <strong>TAME 2.0 Module 4.5 Multi-Group Comparisons and Visualizations</strong>, so that addition to the figure was omitted here.</p>
<p><br></p>
</div>
</div>
<div id="ancova-example" class="section level2 hasAnchor">
<h2>ANCOVA Example<a href="advanced-multi-group-comparisons.html#ancova-example" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In the following ANCOVA example, we’ll still investigate potential differences in cytokine concentrations as result of varying doses of acrolein. However, this time we’ll adjust for sex and age to answer our second environmental health question: <strong>Are there significant differences in inflammatory biomarker concentrations across different doses of acrolein after controlling for sex and age?</strong>.</p>
<p>Let’s first demonstrate how to run an ANCOVA and what the results look like by running the test on just one of our variables (IL-1<span class="math inline">\(\beta\)</span>). The <code>Anova()</code> function was specifically designed to run type II or III ANOVA tests, which have different approaches to dealing with interactions terms and unbalanced datasets. For more information on Type I, II, III ANOVA tests, check out <a href="https://md.psych.bio.uni-goettingen.de/mv/unit/lm_cat/lm_cat_unbal_ss_explained.html">Anova – Type I/II/III SS explained</a>. For the purposes of this example just know that isn’t much of a difference between the type I, II, or III results.</p>
<div class="sourceCode" id="cb469"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb469-1"><a href="advanced-multi-group-comparisons.html#cb469-1" tabindex="-1"></a>anova_test <span class="ot">=</span> <span class="fu">aov</span>(IL1B <span class="sc">~</span> Dose <span class="sc">+</span> Sex <span class="sc">+</span> Age_yr, <span class="at">data =</span> biomarker_data)</span>
<span id="cb469-2"><a href="advanced-multi-group-comparisons.html#cb469-2" tabindex="-1"></a>type3_anova <span class="ot">=</span> <span class="fu">Anova</span>(anova_test, <span class="at">type =</span> <span class="st">&#39;III&#39;</span>)</span>
<span id="cb469-3"><a href="advanced-multi-group-comparisons.html#cb469-3" tabindex="-1"></a>type3_anova</span></code></pre></div>
<pre><code>## Anova Table (Type III tests)
## 
## Response: IL1B
##             Sum Sq Df   F value    Pr(&gt;F)    
## (Intercept) 56.720  1 1092.9386 &lt; 2.2e-16 ***
## Dose         0.762  4    3.6697  0.008862 ** 
## Sex          0.002  1    0.0318  0.858954    
## Age_yr       0.000  1    0.0000  0.994833    
## Residuals    3.788 73                        
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</code></pre>
<p>Based on the table above, there are significant differences in IL-1<span class="math inline">\(\beta\)</span> concentrations in dose after adjusting for sex and age (p-value = 0.009).</p>
<p>Now we’ll run ANCOVA tests across all of our biomarkers.</p>
<div class="sourceCode" id="cb471"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb471-1"><a href="advanced-multi-group-comparisons.html#cb471-1" tabindex="-1"></a><span class="co"># Create data frame to store results</span></span>
<span id="cb471-2"><a href="advanced-multi-group-comparisons.html#cb471-2" tabindex="-1"></a>ancova_res <span class="ot">=</span> <span class="fu">data.frame</span>()</span>
<span id="cb471-3"><a href="advanced-multi-group-comparisons.html#cb471-3" tabindex="-1"></a></span>
<span id="cb471-4"><a href="advanced-multi-group-comparisons.html#cb471-4" tabindex="-1"></a><span class="co"># Add row names to data frame so that it will be able to add ANCOVA results</span></span>
<span id="cb471-5"><a href="advanced-multi-group-comparisons.html#cb471-5" tabindex="-1"></a>rownames <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;(Intercept)&quot;</span>, <span class="st">&quot;Dose&quot;</span>, <span class="st">&quot;Sex&quot;</span>, <span class="st">&quot;Age_yr&quot;</span>)</span>
<span id="cb471-6"><a href="advanced-multi-group-comparisons.html#cb471-6" tabindex="-1"></a>ancova_res <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span>(rownames))</span>
<span id="cb471-7"><a href="advanced-multi-group-comparisons.html#cb471-7" tabindex="-1"></a></span>
<span id="cb471-8"><a href="advanced-multi-group-comparisons.html#cb471-8" tabindex="-1"></a><span class="co"># Assign row names </span></span>
<span id="cb471-9"><a href="advanced-multi-group-comparisons.html#cb471-9" tabindex="-1"></a>ancova_res <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(ancova_res[, <span class="sc">-</span><span class="dv">1</span>], <span class="at">row.names =</span> ancova_res<span class="sc">$</span>rownames)</span>
<span id="cb471-10"><a href="advanced-multi-group-comparisons.html#cb471-10" tabindex="-1"></a></span>
<span id="cb471-11"><a href="advanced-multi-group-comparisons.html#cb471-11" tabindex="-1"></a><span class="co"># Perform ANCOVA over all columns</span></span>
<span id="cb471-12"><a href="advanced-multi-group-comparisons.html#cb471-12" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">3</span><span class="sc">:</span><span class="dv">8</span>) {</span>
<span id="cb471-13"><a href="advanced-multi-group-comparisons.html#cb471-13" tabindex="-1"></a>  </span>
<span id="cb471-14"><a href="advanced-multi-group-comparisons.html#cb471-14" tabindex="-1"></a>  fit <span class="ot">=</span> <span class="fu">aov</span>(<span class="fu">as.formula</span>(<span class="fu">paste0</span>(<span class="fu">names</span>(biomarker_data)[i], <span class="st">&quot;~ Dose + Sex + Age_yr&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)),</span>
<span id="cb471-15"><a href="advanced-multi-group-comparisons.html#cb471-15" tabindex="-1"></a>            biomarker_data)</span>
<span id="cb471-16"><a href="advanced-multi-group-comparisons.html#cb471-16" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(car<span class="sc">::</span><span class="fu">Anova</span>(fit, <span class="at">type =</span> <span class="st">&quot;III&quot;</span>))</span>
<span id="cb471-17"><a href="advanced-multi-group-comparisons.html#cb471-17" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">subset</span>(res, <span class="at">select =</span> Pr..F.)</span>
<span id="cb471-18"><a href="advanced-multi-group-comparisons.html#cb471-18" tabindex="-1"></a>  <span class="fu">names</span>(res)[<span class="fu">names</span>(res) <span class="sc">==</span> <span class="st">&#39;Pr..F.&#39;</span>] <span class="ot">&lt;-</span> <span class="fu">noquote</span>(<span class="fu">paste0</span>(<span class="fu">names</span>(biomarker_data[i])))</span>
<span id="cb471-19"><a href="advanced-multi-group-comparisons.html#cb471-19" tabindex="-1"></a>  ancova_res <span class="ot">&lt;-</span> <span class="fu">transform</span>(<span class="fu">merge</span>(ancova_res, res, <span class="at">by =</span> <span class="dv">0</span>), <span class="at">row.names =</span> Row.names, <span class="at">Row.names =</span> <span class="cn">NULL</span>)</span>
<span id="cb471-20"><a href="advanced-multi-group-comparisons.html#cb471-20" tabindex="-1"></a></span>
<span id="cb471-21"><a href="advanced-multi-group-comparisons.html#cb471-21" tabindex="-1"></a>}</span>
<span id="cb471-22"><a href="advanced-multi-group-comparisons.html#cb471-22" tabindex="-1"></a></span>
<span id="cb471-23"><a href="advanced-multi-group-comparisons.html#cb471-23" tabindex="-1"></a><span class="co"># Transpose for easy viewing, keep columns of interest, and apply BH adjustment</span></span>
<span id="cb471-24"><a href="advanced-multi-group-comparisons.html#cb471-24" tabindex="-1"></a>ancova_res <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">t</span>(ancova_res)) <span class="sc">%&gt;%</span></span>
<span id="cb471-25"><a href="advanced-multi-group-comparisons.html#cb471-25" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Dose) <span class="sc">%&gt;%</span></span>
<span id="cb471-26"><a href="advanced-multi-group-comparisons.html#cb471-26" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), \(x) <span class="fu">format</span>(<span class="fu">p.adjust</span>(x, <span class="st">&quot;BH&quot;</span>), <span class="at">scientific =</span> <span class="cn">TRUE</span>)))</span>
<span id="cb471-27"><a href="advanced-multi-group-comparisons.html#cb471-27" tabindex="-1"></a></span>
<span id="cb471-28"><a href="advanced-multi-group-comparisons.html#cb471-28" tabindex="-1"></a><span class="co"># View results</span></span>
<span id="cb471-29"><a href="advanced-multi-group-comparisons.html#cb471-29" tabindex="-1"></a><span class="fu">datatable</span>(ancova_res)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-ae1379e00d53b85dd7b6" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ae1379e00d53b85dd7b6">{"x":{"filter":"none","vertical":false,"data":[["IL1B","IL6","IL8","IL10","TNFa","VEGF"],["1.329371e-02","1.358496e-01","2.634994e-16","7.091243e-03","1.897664e-05","1.796588e-02"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Dose<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Dose","targets":1}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Looking at the table above, there are statistically differences in all cytokine concentrations with the exception of IL-6 based on dose (p adj &lt; 0.05). To determine what doses were significantly different from one another we’ll need to run Tukey’s post hoc tests.</p>
<div class="sourceCode" id="cb472"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb472-1"><a href="advanced-multi-group-comparisons.html#cb472-1" tabindex="-1"></a><span class="co"># Create results data frame with a column showing the comparisons (extracted from single run vs for loop)</span></span>
<span id="cb472-2"><a href="advanced-multi-group-comparisons.html#cb472-2" tabindex="-1"></a>tukey_res <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Comparison =</span> <span class="fu">c</span>(<span class="st">&quot;0.6 - 0&quot;</span>, <span class="st">&quot;1 - 0&quot;</span>, <span class="st">&quot;2 - 0&quot;</span>, <span class="st">&quot;4 - 0&quot;</span>, <span class="st">&quot;1 - 0.6&quot;</span>, <span class="st">&quot;2 - 0.6&quot;</span>, </span>
<span id="cb472-3"><a href="advanced-multi-group-comparisons.html#cb472-3" tabindex="-1"></a><span class="st">&quot;4 - 0.6&quot;</span>, <span class="st">&quot;2 - 1&quot;</span>, <span class="st">&quot;4 - 1&quot;</span>, <span class="st">&quot;4 - 2&quot;</span>))</span>
<span id="cb472-4"><a href="advanced-multi-group-comparisons.html#cb472-4" tabindex="-1"></a></span>
<span id="cb472-5"><a href="advanced-multi-group-comparisons.html#cb472-5" tabindex="-1"></a><span class="co"># Perform Tukey&#39;s test</span></span>
<span id="cb472-6"><a href="advanced-multi-group-comparisons.html#cb472-6" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">3</span><span class="sc">:</span><span class="dv">8</span>) {</span>
<span id="cb472-7"><a href="advanced-multi-group-comparisons.html#cb472-7" tabindex="-1"></a>  </span>
<span id="cb472-8"><a href="advanced-multi-group-comparisons.html#cb472-8" tabindex="-1"></a>  <span class="co"># need to run ANCOVA first</span></span>
<span id="cb472-9"><a href="advanced-multi-group-comparisons.html#cb472-9" tabindex="-1"></a>  fit <span class="ot">=</span> <span class="fu">aov</span>(<span class="fu">as.formula</span>(<span class="fu">paste0</span>(<span class="fu">names</span>(biomarker_data)[i], <span class="st">&quot;~ Dose + Sex + Age_yr&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)),</span>
<span id="cb472-10"><a href="advanced-multi-group-comparisons.html#cb472-10" tabindex="-1"></a>            biomarker_data)</span>
<span id="cb472-11"><a href="advanced-multi-group-comparisons.html#cb472-11" tabindex="-1"></a>  </span>
<span id="cb472-12"><a href="advanced-multi-group-comparisons.html#cb472-12" tabindex="-1"></a>  <span class="co"># Tukey&#39;s</span></span>
<span id="cb472-13"><a href="advanced-multi-group-comparisons.html#cb472-13" tabindex="-1"></a>  posthoc <span class="ot">&lt;-</span> <span class="fu">summary</span>(<span class="fu">glht</span>(fit, <span class="at">linfct =</span> <span class="fu">mcp</span>(<span class="at">Dose =</span> <span class="st">&quot;Tukey&quot;</span>)), <span class="at">test =</span> <span class="fu">adjusted</span>(<span class="st">&quot;BH&quot;</span>))</span>
<span id="cb472-14"><a href="advanced-multi-group-comparisons.html#cb472-14" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">summary</span>(posthoc)<span class="sc">$</span>test</span>
<span id="cb472-15"><a href="advanced-multi-group-comparisons.html#cb472-15" tabindex="-1"></a>  </span>
<span id="cb472-16"><a href="advanced-multi-group-comparisons.html#cb472-16" tabindex="-1"></a>  <span class="co"># Formatting the df with the Tukey&#39;s values</span></span>
<span id="cb472-17"><a href="advanced-multi-group-comparisons.html#cb472-17" tabindex="-1"></a>  res_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cbind</span> (res<span class="sc">$</span>coefficients, res<span class="sc">$</span>sigma, res<span class="sc">$</span>tstat, res<span class="sc">$</span>pvalues))</span>
<span id="cb472-18"><a href="advanced-multi-group-comparisons.html#cb472-18" tabindex="-1"></a>  <span class="fu">colnames</span>(res_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Estimate&quot;</span>, <span class="st">&quot;Std.Error&quot;</span>, <span class="st">&quot;t.value&quot;</span>, <span class="st">&quot;Pr(&gt;|t|)&quot;</span>)</span>
<span id="cb472-19"><a href="advanced-multi-group-comparisons.html#cb472-19" tabindex="-1"></a>  res_df <span class="ot">&lt;-</span> <span class="fu">round</span>(res_df[<span class="dv">4</span>],<span class="dv">4</span>)</span>
<span id="cb472-20"><a href="advanced-multi-group-comparisons.html#cb472-20" tabindex="-1"></a>  <span class="fu">names</span>(res_df)[<span class="fu">names</span>(res_df) <span class="sc">==</span> <span class="st">&#39;Pr(&gt;|t|)&#39;</span>] <span class="ot">&lt;-</span> <span class="fu">noquote</span>(<span class="fu">paste0</span>(<span class="fu">names</span>(biomarker_data[i])))</span>
<span id="cb472-21"><a href="advanced-multi-group-comparisons.html#cb472-21" tabindex="-1"></a>  res_df <span class="ot">&lt;-</span> res_df <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;Comparison&quot;</span>)</span>
<span id="cb472-22"><a href="advanced-multi-group-comparisons.html#cb472-22" tabindex="-1"></a>  </span>
<span id="cb472-23"><a href="advanced-multi-group-comparisons.html#cb472-23" tabindex="-1"></a>  tukey_res <span class="ot">&lt;-</span> <span class="fu">left_join</span>(tukey_res, res_df, <span class="at">by =</span> <span class="st">&quot;Comparison&quot;</span>)</span>
<span id="cb472-24"><a href="advanced-multi-group-comparisons.html#cb472-24" tabindex="-1"></a>}</span>
<span id="cb472-25"><a href="advanced-multi-group-comparisons.html#cb472-25" tabindex="-1"></a></span>
<span id="cb472-26"><a href="advanced-multi-group-comparisons.html#cb472-26" tabindex="-1"></a><span class="fu">datatable</span>(tukey_res)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-41559414d0f317cae8ca" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-41559414d0f317cae8ca">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10"],["0.6 - 0","1 - 0","2 - 0","4 - 0","1 - 0.6","2 - 0.6","4 - 0.6","2 - 1","4 - 1","4 - 2"],[0.8887,0.2988,0.0534,0.0114,0.8383,0.3527,0.1225,0.9247,0.6414,0.9797],[0.9999,1,1,0.2255,0.9992,0.9998,0.171,1,0.2692,0.2356],[0.9676,0.9996,0.0005,0,0.9151,0.0046,0,0.0003,0,0],[0.0062,0.9782,0.7684,1,0.0331,0.1375,0.0064,0.9755,0.9792,0.7722],[0.9985000000000001,0.6271,0.0211,0.0001,0.4478,0.0092,0,0.4304,0.0131,0.5193],[0.9998,0.879,0.0269,0.2523,0.9397,0.0431,0.3405,0.2355,0.8016,0.8628]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Comparison<\/th>\n      <th>IL1B<\/th>\n      <th>IL6<\/th>\n      <th>IL8<\/th>\n      <th>IL10<\/th>\n      <th>TNFa<\/th>\n      <th>VEGF<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Comparison","targets":1},{"name":"IL1B","targets":2},{"name":"IL6","targets":3},{"name":"IL8","targets":4},{"name":"IL10","targets":5},{"name":"TNFa","targets":6},{"name":"VEGF","targets":7}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="answer-to-environmental-health-question-2-8" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 2<a href="advanced-multi-group-comparisons.html#answer-to-environmental-health-question-2-8" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #2</strong></em>: Are there significant differences in inflammatory biomarker concentrations across different doses of acrolein after controlling for sex and age?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: Based on the ANCOVA tests, there are significant differences resulting from various doses of acrolein (p adj &lt; 0.05) across all cytokine concentrations with the exception of IL-6. All biomarkers, with the exception of IL-6, had at least 1 significantly different biomarker concentration when comparing doses.</p>
</div>
</div>
<div id="visualizing-ancova-results" class="section level3 hasAnchor">
<h3>Visualizing ANCOVA Results<a href="advanced-multi-group-comparisons.html#visualizing-ancova-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Before graphing these results, we first need to think about which ones we want to display. For simplicity’s sake, we will demonstrate graphing only comparisons that are with the control(“0”) group and that are significant. To do this, we’ll:</p>
<ol style="list-style-type: decimal">
<li>Separate our <code>Comparison</code> column into a <code>group1</code> and <code>group2</code> column.</li>
<li>Filter to comparisons including only the 0 group.</li>
<li>Pivot the dataframe longer, to match the format of our data used as input for facet plotting.</li>
<li>Filter to only p-values that are less that 0.05.</li>
</ol>
<div class="sourceCode" id="cb473"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb473-1"><a href="advanced-multi-group-comparisons.html#cb473-1" tabindex="-1"></a>tukey_res_forgraph <span class="ot">&lt;-</span> tukey_res <span class="sc">%&gt;%</span></span>
<span id="cb473-2"><a href="advanced-multi-group-comparisons.html#cb473-2" tabindex="-1"></a>  <span class="fu">separate</span>(Comparison, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&quot;group1&quot;</span>, <span class="st">&quot;group2&quot;</span>), <span class="at">sep =</span> <span class="st">&quot; - &quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb473-3"><a href="advanced-multi-group-comparisons.html#cb473-3" tabindex="-1"></a>  <span class="fu">filter</span>(group2 <span class="sc">==</span> <span class="st">&quot;0&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb473-4"><a href="advanced-multi-group-comparisons.html#cb473-4" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="sc">-</span>group2) <span class="sc">%&gt;%</span></span>
<span id="cb473-5"><a href="advanced-multi-group-comparisons.html#cb473-5" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">!</span>group1, <span class="at">names_to =</span> <span class="st">&quot;Variable&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;P Value&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb473-6"><a href="advanced-multi-group-comparisons.html#cb473-6" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="st">`</span><span class="at">P Value</span><span class="st">`</span> <span class="sc">&lt;</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb473-7"><a href="advanced-multi-group-comparisons.html#cb473-7" tabindex="-1"></a>  <span class="co"># rounding the p values to 4 digits for readability</span></span>
<span id="cb473-8"><a href="advanced-multi-group-comparisons.html#cb473-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">P Value</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">round</span>(<span class="st">`</span><span class="at">P Value</span><span class="st">`</span>, <span class="dv">4</span>))</span>
<span id="cb473-9"><a href="advanced-multi-group-comparisons.html#cb473-9" tabindex="-1"></a></span>
<span id="cb473-10"><a href="advanced-multi-group-comparisons.html#cb473-10" tabindex="-1"></a><span class="fu">datatable</span>(tukey_res_forgraph)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-2a121e9b8885a5a3b8e0" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-2a121e9b8885a5a3b8e0">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7"],["0.6","2","2","2","4","4","4"],["IL10","IL8","TNFa","VEGF","IL1B","IL8","TNFa"],[0.0062,0.0005,0.0211,0.0269,0.0114,0,0.0001]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>group1<\/th>\n      <th>Variable<\/th>\n      <th>P Value<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":3},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"group1","targets":1},{"name":"Variable","targets":2},{"name":"P Value","targets":3}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Next, we can take a few steps to add columns to the dataframe that will aid in graphing:</p>
<ol style="list-style-type: decimal">
<li>Add a column for significance stars.</li>
<li>Add a column to indicate the y position for the significance annotation (similar to the above example with the two-way ANOVA).</li>
</ol>
<div class="sourceCode" id="cb474"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb474-1"><a href="advanced-multi-group-comparisons.html#cb474-1" tabindex="-1"></a><span class="co"># Add column for significance stars</span></span>
<span id="cb474-2"><a href="advanced-multi-group-comparisons.html#cb474-2" tabindex="-1"></a>tukey_res_forgraph <span class="ot">&lt;-</span> tukey_res_forgraph <span class="sc">%&gt;%</span></span>
<span id="cb474-3"><a href="advanced-multi-group-comparisons.html#cb474-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">p.signif =</span> <span class="fu">ifelse</span>(<span class="st">`</span><span class="at">P Value</span><span class="st">`</span> <span class="sc">&lt;</span> <span class="fl">0.0001</span>, <span class="st">&quot;****&quot;</span>, </span>
<span id="cb474-4"><a href="advanced-multi-group-comparisons.html#cb474-4" tabindex="-1"></a>                           <span class="fu">ifelse</span>(<span class="st">`</span><span class="at">P Value</span><span class="st">`</span> <span class="sc">&lt;</span> <span class="fl">0.001</span>, <span class="st">&quot;***&quot;</span>, </span>
<span id="cb474-5"><a href="advanced-multi-group-comparisons.html#cb474-5" tabindex="-1"></a>                                  <span class="fu">ifelse</span>(<span class="st">`</span><span class="at">P Value</span><span class="st">`</span> <span class="sc">&lt;</span> <span class="fl">0.01</span>, <span class="st">&quot;**&quot;</span>, </span>
<span id="cb474-6"><a href="advanced-multi-group-comparisons.html#cb474-6" tabindex="-1"></a>                                         <span class="fu">ifelse</span>(<span class="st">`</span><span class="at">P Value</span><span class="st">`</span> <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="st">&quot;*&quot;</span>, <span class="cn">NA</span>)))))</span>
<span id="cb474-7"><a href="advanced-multi-group-comparisons.html#cb474-7" tabindex="-1"></a></span>
<span id="cb474-8"><a href="advanced-multi-group-comparisons.html#cb474-8" tabindex="-1"></a><span class="co"># Calculate y positions to plot significance stars</span></span>
<span id="cb474-9"><a href="advanced-multi-group-comparisons.html#cb474-9" tabindex="-1"></a>sig_labs_y_tukey <span class="ot">&lt;-</span> biomarker_data <span class="sc">%&gt;%</span></span>
<span id="cb474-10"><a href="advanced-multi-group-comparisons.html#cb474-10" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(IL1B<span class="sc">:</span>VEGF, \(x) <span class="fu">max</span>(x))) <span class="sc">%&gt;%</span></span>
<span id="cb474-11"><a href="advanced-multi-group-comparisons.html#cb474-11" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb474-12"><a href="advanced-multi-group-comparisons.html#cb474-12" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;Variable&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb474-13"><a href="advanced-multi-group-comparisons.html#cb474-13" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">&quot;y_pos&quot;</span> <span class="ot">=</span> <span class="st">&quot;V1&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb474-14"><a href="advanced-multi-group-comparisons.html#cb474-14" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">y_pos =</span> y_pos <span class="sc">*</span> <span class="fl">1.15</span>)</span>
<span id="cb474-15"><a href="advanced-multi-group-comparisons.html#cb474-15" tabindex="-1"></a></span>
<span id="cb474-16"><a href="advanced-multi-group-comparisons.html#cb474-16" tabindex="-1"></a>sig_labs_y_tukey</span></code></pre></div>
<pre><code>##   Variable     y_pos
## 1     IL1B  4.760955
## 2      IL6 13.488214
## 3      IL8 19.887434
## 4     IL10  3.090120
## 5     TNFa  4.292110
## 6     VEGF 13.558526</code></pre>
<div class="sourceCode" id="cb476"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb476-1"><a href="advanced-multi-group-comparisons.html#cb476-1" tabindex="-1"></a><span class="co"># Join y positions to tukey_res</span></span>
<span id="cb476-2"><a href="advanced-multi-group-comparisons.html#cb476-2" tabindex="-1"></a>tukey_res_forgraph <span class="ot">&lt;-</span> tukey_res_forgraph <span class="sc">%&gt;%</span></span>
<span id="cb476-3"><a href="advanced-multi-group-comparisons.html#cb476-3" tabindex="-1"></a>  <span class="fu">left_join</span>(sig_labs_y_tukey, <span class="at">by =</span> <span class="st">&quot;Variable&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb476-4"><a href="advanced-multi-group-comparisons.html#cb476-4" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">&quot;Dose&quot;</span> <span class="ot">=</span> <span class="st">&quot;group1&quot;</span>)</span>
<span id="cb476-5"><a href="advanced-multi-group-comparisons.html#cb476-5" tabindex="-1"></a></span>
<span id="cb476-6"><a href="advanced-multi-group-comparisons.html#cb476-6" tabindex="-1"></a><span class="fu">datatable</span>(tukey_res_forgraph)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-f50fa4245f1894576d98" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f50fa4245f1894576d98">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7"],["0.6","2","2","2","4","4","4"],["IL10","IL8","TNFa","VEGF","IL1B","IL8","TNFa"],[0.0062,0.0005,0.0211,0.0269,0.0114,0,0.0001],["**","***","*","*","*","****","***"],[3.090119791590873,19.88743352518671,4.292109579426667,13.55852600295567,4.760954654977279,19.88743352518671,4.292109579426667]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Dose<\/th>\n      <th>Variable<\/th>\n      <th>P Value<\/th>\n      <th>p.signif<\/th>\n      <th>y_pos<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,5]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Dose","targets":1},{"name":"Variable","targets":2},{"name":"P Value","targets":3},{"name":"p.signif","targets":4},{"name":"y_pos","targets":5}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>We also need to prepare our overall p-values from our ANCOVA for display:</p>
<div class="sourceCode" id="cb477"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb477-1"><a href="advanced-multi-group-comparisons.html#cb477-1" tabindex="-1"></a>ancova_res_forgraphing <span class="ot">&lt;-</span> ancova_res <span class="sc">%&gt;%</span></span>
<span id="cb477-2"><a href="advanced-multi-group-comparisons.html#cb477-2" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">`</span><span class="at">P Value</span><span class="st">`</span> <span class="ot">=</span> Dose) <span class="sc">%&gt;%</span></span>
<span id="cb477-3"><a href="advanced-multi-group-comparisons.html#cb477-3" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;Variable&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb477-4"><a href="advanced-multi-group-comparisons.html#cb477-4" tabindex="-1"></a>  <span class="fu">left_join</span>(sig_labs_y, <span class="at">by =</span> <span class="st">&quot;Variable&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb477-5"><a href="advanced-multi-group-comparisons.html#cb477-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">P Value</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">formatC</span>(<span class="fu">as.numeric</span>(<span class="st">`</span><span class="at">P Value</span><span class="st">`</span>), <span class="at">format =</span> <span class="st">&quot;e&quot;</span>, <span class="at">digits =</span> <span class="dv">2</span>),</span>
<span id="cb477-6"><a href="advanced-multi-group-comparisons.html#cb477-6" tabindex="-1"></a>         <span class="at">label =</span> <span class="fu">paste</span>(<span class="st">&quot;p = &quot;</span>, <span class="st">`</span><span class="at">P Value</span><span class="st">`</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span></code></pre></div>
<p>Now, we are ready to make our graph! We will use similar code to the above, this time adding in our significance stars over specific columns.</p>
<div class="sourceCode" id="cb478"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb478-1"><a href="advanced-multi-group-comparisons.html#cb478-1" tabindex="-1"></a><span class="co"># Make graph</span></span>
<span id="cb478-2"><a href="advanced-multi-group-comparisons.html#cb478-2" tabindex="-1"></a><span class="fu">ggplot</span>(biomarker_data_long, <span class="fu">aes</span>(<span class="at">x =</span> Dose, <span class="at">y =</span> Value)) <span class="sc">+</span></span>
<span id="cb478-3"><a href="advanced-multi-group-comparisons.html#cb478-3" tabindex="-1"></a>  <span class="co"># outlier.shape = NA removes outliers</span></span>
<span id="cb478-4"><a href="advanced-multi-group-comparisons.html#cb478-4" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> Dose), <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb478-5"><a href="advanced-multi-group-comparisons.html#cb478-5" tabindex="-1"></a>  <span class="co"># Changing box plot colors</span></span>
<span id="cb478-6"><a href="advanced-multi-group-comparisons.html#cb478-6" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;#BFBFBF&quot;</span>, <span class="st">&quot;#D5A298&quot;</span>, <span class="st">&quot;#E38273&quot;</span>, <span class="st">&quot;#EB5F4E&quot;</span>, <span class="st">&quot;#EE2B2B&quot;</span>)) <span class="sc">+</span></span>
<span id="cb478-7"><a href="advanced-multi-group-comparisons.html#cb478-7" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">1.5</span>, <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="fl">0.15</span>), <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb478-8"><a href="advanced-multi-group-comparisons.html#cb478-8" tabindex="-1"></a>  <span class="co"># Add overall ANCOVA label</span></span>
<span id="cb478-9"><a href="advanced-multi-group-comparisons.html#cb478-9" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> ancova_res_forgraphing, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fl">1.3</span>, <span class="at">y =</span> y_pos <span class="sc">*</span> <span class="fl">1.15</span>, <span class="at">label =</span> label, <span class="at">size =</span> <span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb478-10"><a href="advanced-multi-group-comparisons.html#cb478-10" tabindex="-1"></a>  <span class="co"># Add tukey annotation </span></span>
<span id="cb478-11"><a href="advanced-multi-group-comparisons.html#cb478-11" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> tukey_res_forgraph, <span class="fu">aes</span>(<span class="at">x =</span> Dose, <span class="at">y =</span> y_pos, <span class="at">label =</span> p.signif, <span class="at">size =</span> <span class="dv">10</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb478-12"><a href="advanced-multi-group-comparisons.html#cb478-12" tabindex="-1"></a>  </span>
<span id="cb478-13"><a href="advanced-multi-group-comparisons.html#cb478-13" tabindex="-1"></a>  <span class="co"># Faceting by each biomarker</span></span>
<span id="cb478-14"><a href="advanced-multi-group-comparisons.html#cb478-14" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> Variable, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">scales =</span> <span class="st">&quot;free_y&quot;</span>, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">Variable =</span> new_labels)) <span class="sc">+</span></span>
<span id="cb478-15"><a href="advanced-multi-group-comparisons.html#cb478-15" tabindex="-1"></a>  <span class="co"># Removing legend</span></span>
<span id="cb478-16"><a href="advanced-multi-group-comparisons.html#cb478-16" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb478-17"><a href="advanced-multi-group-comparisons.html#cb478-17" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">&quot;bold&quot;</span>, <span class="at">size =</span> <span class="fu">rel</span>(<span class="fl">1.5</span>)),</span>
<span id="cb478-18"><a href="advanced-multi-group-comparisons.html#cb478-18" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.75</span>),</span>
<span id="cb478-19"><a href="advanced-multi-group-comparisons.html#cb478-19" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="dv">2</span>),</span>
<span id="cb478-20"><a href="advanced-multi-group-comparisons.html#cb478-20" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb478-21"><a href="advanced-multi-group-comparisons.html#cb478-21" tabindex="-1"></a>        <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>)) <span class="sc">+</span> </span>
<span id="cb478-22"><a href="advanced-multi-group-comparisons.html#cb478-22" tabindex="-1"></a>  </span>
<span id="cb478-23"><a href="advanced-multi-group-comparisons.html#cb478-23" tabindex="-1"></a>  <span class="co"># Changing axes labels</span></span>
<span id="cb478-24"><a href="advanced-multi-group-comparisons.html#cb478-24" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;Acrolein (ppm)&quot;</span>, <span class="at">y =</span> <span class="fu">expression</span>(<span class="fu">bold</span>(Log[<span class="dv">2</span>]<span class="sc">*</span><span class="st">&quot;(Concentration (pg/ml))&quot;</span>)))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-380-1.png" width="1152" style="display: block; margin: auto;" />
An appropriate title for this figure could be:</p>
<p>“<strong>Figure X. Acrolein exposure increases inflammatory cytokine secretion in most primary human bronchial epithelial cells.</strong> Overall p-values from ANCOVA tests adjusting for age and sex are in the left-hand corner. Tukey’s post hoc tests were subsequently run and significant Benjamini-Hochberg adjusted p-values are denoted with asterisks compared to the control (0ppm) dose only. p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001, <em>n</em> = 16 per group.”</p>
<p><br></p>
</div>
</div>
<div id="concluding-remarks-17" class="section level2 hasAnchor">
<h2>Concluding Remarks<a href="advanced-multi-group-comparisons.html#concluding-remarks-17" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this module, we introduced advanced multi-group comparisons using two-way ANOVA and ANCOVA tests. These overall effect tests along with post-hoc testing were used on an example dataset to provide a basis for publication-ready tables and figures to present these results. This training module provides code and text for advanced multi-group comparisons necessary to answer more complex research questions.</p>
<p><br></p>
<div id="additional-resources-6" class="section level3 hasAnchor">
<h3>Additional Resources<a href="advanced-multi-group-comparisons.html#additional-resources-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><a href="https://www.scribbr.com/statistics/two-way-anova/">Two-Way ANOVA</a></li>
<li><a href="https://www.datanovia.com/en/lessons/repeated-measures-anova-in-r/">Repeated Measure ANOVA in R</a></li>
<li><a href="https://ibecav.github.io/ancova_example/">ANCOVA Example</a></li>
<li><a href="https://cran.r-project.org/web/packages/fANCOVA/fANCOVA.pdf">Nonparametric ANOVA RDocumentation</a></li>
<li><a href="https://www.rdocumentation.org/packages/sm/versions/2.2-6.0/topics/sm.ancova">Nonparametric ANCOVA RDocumentation</a></li>
</ul>
<p><br></p>
<p><label class="tykfont">
Test Your Knowledge
</label></p>
<div class="tyk">
<p>Functional endpoints from these cultures were also measured. These endpoints were: 1) Membrane Permeability (MemPerm), 2) Trans-Epithelial Electrical Resistance (TEER), 3) Ciliary Beat Frequency (CBF), and 4) Expression of Mucin (MUC5AC). These data were already processed and tested for normality (see Test Your Knowledge for <strong>TAME 2.0 Module 4.2 Data Import, Processing, and Summary Statistics</strong>), with results indicating that two of the endpoints are normally distributed and two non-normally distributed.
Using the data found in “Module4_5_TYKInput.xlsx”, answer the following research question: Are there significant differences in functional endpoints based on doses of acrolein and sex after adjusting for age? To streamline the analysis, we’ll only include doses of acrolein at 0, 1, and 4ppm.</p>
<p><strong>Hint</strong>: You’ll need to run a two-way ANCOVA. Given that some of the assumptions for parametric tests (i.e., normality and homogeneity of variance) and the size of the data is on the smaller side, we likely wouldn’t run a parametric test. However, we’ll do so here just to illustrate an example of how to run a two-way ANCOVA.</p>
</div>

</div>
</div>
</div>



            </section>

          </div>
        </div>
      </div>
<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
