<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4.2 Data Import, Processing, and Summary Statistics | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="4.2 Data Import, Processing, and Summary Statistics | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4.2 Data Import, Processing, and Summary Statistics | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  <meta name="twitter:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  

<meta name="author" content="Rager Lab" />


<meta name="date" content="2024-12-06" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="Icons_Used_Throughout/Favicon.png" type="image/x-icon" />
<link rel="prev" href="overview-of-experimental-design-and-example-data.html"/>
<link rel="next" href="data-import-from-pdf-sources.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>
<link href="libs/table1-1.0/table1_defaults.css" rel="stylesheet" />
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-FGFKVX2X3G"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-FGFKVX2X3G');
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="Icons_Used_Throughout/TAME 2.0 Logo.png"><br></a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Chapter 1 Introductory <br>Data Science</b></span></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html"><i class="fa fa-check"></i>1.1 FAIR Data Management Practices</a>
<ul>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-training-module"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-fair"><i class="fa fa-check"></i>Introduction to FAIR</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#breaking-down-fair-letter-by-letter"><i class="fa fa-check"></i>Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#what-does-this-mean-for-you"><i class="fa fa-check"></i>What Does This Mean for You?</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#data-repositories-for-sharing-of-data"><i class="fa fa-check"></i>Data Repositories for Sharing of Data</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#recent-shifts-in-regulatory-policies-for-data-sharing"><i class="fa fa-check"></i>Recent Shifts in Regulatory Policies for Data Sharing</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#additional-training-resources-on-fair"><i class="fa fa-check"></i>Additional Training Resources on FAIR</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html"><i class="fa fa-check"></i>1.2 Data Sharing through Online Repositories</a>
<ul>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#an-overview-and-example-with-the-dataverse-repository"><i class="fa fa-check"></i>An Overview and Example with the Dataverse Repository</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#introduction-to-training-module-1"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#data-repositories"><i class="fa fa-check"></i>Data Repositories</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#the-dataverse-project"><i class="fa fa-check"></i>The Dataverse Project</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#what-is-a-dataverse"><i class="fa fa-check"></i>What is a Dataverse?</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#metadata"><i class="fa fa-check"></i>Metadata</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataverse"><i class="fa fa-check"></i>Creating a Dataverse</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataset"><i class="fa fa-check"></i>Creating a Dataset</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#concluding-remarks-1"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html"><i class="fa fa-check"></i>1.3 File Management using Github</a>
<ul>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#introduction-to-training-module-2"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-an-account"><i class="fa fa-check"></i>Creating an Account</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-a-repository"><i class="fa fa-check"></i>Creating a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#uploading-code"><i class="fa fa-check"></i>Uploading Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#adding-subfolders-in-a-repository"><i class="fa fa-check"></i>Adding Subfolders in a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-code"><i class="fa fa-check"></i>Updating Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-repository-titles-and-structure-to-support-a-manuscript"><i class="fa fa-check"></i>Updating Repository Titles and Structure to Support a Manuscript</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#tracking-code-changes-using-github-branches"><i class="fa fa-check"></i>Tracking Code Changes using Github Branches</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#concluding-remarks-2"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html"><i class="fa fa-check"></i>1.4 Data Wrangling in Excel</a>
<ul>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#introduction-to-training-module-3"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#save-a-copy-of-the-soon-to-be-organized-and-cleaned-dataset-as-a-new-file"><i class="fa fa-check"></i>Save a Copy of the Soon-To-Be Organized and Cleaned Dataset as a New File</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-extraneous-white-space"><i class="fa fa-check"></i>Remove Extraneous White Space</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#replace-missing-data-with-na"><i class="fa fa-check"></i>Replace Missing Data with “NA”</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-metadata-tab"><i class="fa fa-check"></i>Create a Metadata Tab</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#abbreviate-and-capitalize-categorical-data"><i class="fa fa-check"></i>Abbreviate and Capitalize Categorical Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#alphabetize-sort-the-data-by-the-categorical-variable-of-interest"><i class="fa fa-check"></i>Alphabetize (Sort) the Data by the Categorical Variable of Interest</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-new-subject-number-column"><i class="fa fa-check"></i>Create a New Subject Number Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-special-symbols-and-dashes"><i class="fa fa-check"></i>Remove Special Symbols and Dashes</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#bold-all-column-names-and-center-all-data"><i class="fa fa-check"></i>Bold all Column Names and Center all Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-subject-identifier-column"><i class="fa fa-check"></i>Create a Subject Identifier Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#separate-subject-demographic-data-from-experimental-measurements"><i class="fa fa-check"></i>Separate Subject Demographic Data from Experimental Measurements</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#convert-data-from-wide-to-long-format"><i class="fa fa-check"></i>Convert Data from Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#pivoting-data-from-a-wide-to-long-format"><i class="fa fa-check"></i>Pivoting Data from a Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#generating-summary-level-statistics-with-pivot-tables"><i class="fa fa-check"></i>Generating Summary-Level Statistics with Pivot Tables</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#excel-vs.-r-which-should-you-use"><i class="fa fa-check"></i>Excel vs. R: Which Should You Use?</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#concluding-remarks-3"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 2 Coding in R</b></span></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html"><i class="fa fa-check"></i>2.1 Downloading and Programming in R</a>
<ul>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-training-module-4"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#general-introduction-and-installation-of-r-and-rstudio"><i class="fa fa-check"></i>General Introduction and Installation of R and RStudio</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-r-packages"><i class="fa fa-check"></i>Introduction to R Packages</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#scripting-basics"><i class="fa fa-check"></i>Scripting Basics</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#code-troubleshooting"><i class="fa fa-check"></i>Code Troubleshooting</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#concluding-remarks-4"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html"><i class="fa fa-check"></i>2.2 Coding “Best” Practices</a>
<ul>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#introduction-to-training-module-5"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#scripting-file-types"><i class="fa fa-check"></i>Scripting File Types</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-headers-and-annotation"><i class="fa fa-check"></i>Script Headers and Annotation</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#coding-style"><i class="fa fa-check"></i>Coding Style</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-organization"><i class="fa fa-check"></i>Script Organization</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#concluding-remarks-5"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html"><i class="fa fa-check"></i>2.3 Data Manipulation and Reshaping</a>
<ul>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-training-module-6"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#data-manipulation-using-base-r"><i class="fa fa-check"></i>Data Manipulation Using Base R</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-tidyverse"><i class="fa fa-check"></i>Introduction to Tidyverse</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#concluding-remarks-6"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html"><i class="fa fa-check"></i>2.4 Improving Coding Efficiencies</a>
<ul>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#introduction-to-training-module-7"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#loops"><i class="fa fa-check"></i>Loops</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#functions"><i class="fa fa-check"></i>Functions</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#list-operations"><i class="fa fa-check"></i>List operations</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#concluding-remarks-7"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#additional-resources-2"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="part"><span><b>Chapter 3 Basics of Data <br>Analysis and Visualizations</b></span></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html"><i class="fa fa-check"></i>3.1 Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-data-visualizations"><i class="fa fa-check"></i>Introduction to Data Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-training-module-8"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#density-plot-visualization"><i class="fa fa-check"></i>Density Plot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#boxplot-visualization"><i class="fa fa-check"></i>Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#correlation-visualizations"><i class="fa fa-check"></i>Correlation Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#heatmap-visualization"><i class="fa fa-check"></i>Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#concluding-remarks-8"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html"><i class="fa fa-check"></i>3.2 Improving Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-data-visulization-conventions"><i class="fa fa-check"></i>Introduction to Data Visulization Conventions</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-training-module-9"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-boxplot-visualization"><i class="fa fa-check"></i>Creating an Improved Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-heatmap-visualization"><i class="fa fa-check"></i>Creating an Improved Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-multi-plot-figures"><i class="fa fa-check"></i>Creating Multi-Plot Figures</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#concluding-remarks-9"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html"><i class="fa fa-check"></i>3.3 Normality Tests and Data Transformations</a>
<ul>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#introduction-to-training-module-10"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#what-is-a-normal-distribution"><i class="fa fa-check"></i>What is a Normal Distribution?</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#qualitative-assessment-of-normality"><i class="fa fa-check"></i>Qualitative Assessment of Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#quantitative-normality-assessment"><i class="fa fa-check"></i>Quantitative Normality Assessment</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#data-transformation"><i class="fa fa-check"></i>Data Transformation</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#additional-considerations-regarding-normality"><i class="fa fa-check"></i>Additional Considerations Regarding Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#concluding-remarks-10"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html"><i class="fa fa-check"></i>3.4 Intoduction to Statistical Tests</a>
<ul>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#introduction-to-training-module-11"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#assessing-normality-homogeneity-of-variance"><i class="fa fa-check"></i>Assessing Normality &amp; Homogeneity of Variance</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#two-group-visualizations-and-statistical-comparisons-using-the-t-test"><i class="fa fa-check"></i>Two-Group Visualizations and Statistical Comparisons using the T-Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#three-group-visualizations-and-statistical-comparisons-using-an-anova"><i class="fa fa-check"></i>Three-Group Visualizations and Statistical Comparisons using an ANOVA</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#regression-modeling-and-visualization-linear-and-logistic-regressions"><i class="fa fa-check"></i>Regression Modeling and Visualization: Linear and Logistic Regressions</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#statistical-evaluations-of-categorical-data-using-the-chi-squared-test-and-fishers-exact-test"><i class="fa fa-check"></i>Statistical Evaluations of Categorical Data using the Chi-Squared Test and Fisher’s Exact Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#concluding-remarks-11"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 4 Converting Wet <br>Lab Data into Dry Lab <br>Analyses</b></span></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html"><i class="fa fa-check"></i>4.1 Overview of Experimental Design and Example Data</a>
<ul>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#introduction-to-training-module-12"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#replicates"><i class="fa fa-check"></i>Replicates</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#orientation-to-example-data-for-chapter-4"><i class="fa fa-check"></i>Orientation to Example Data for Chapter 4</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#concluding-remarks-12"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html"><i class="fa fa-check"></i>4.2 Data Import, Processing, and Summary Statistics</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#introduction-to-training-module-13"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#data-import"><i class="fa fa-check"></i>Data Import</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#handling-missing-values"><i class="fa fa-check"></i>Handling Missing Values</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#averaging-replicates"><i class="fa fa-check"></i>Averaging Replicates</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#descriptive-statistics"><i class="fa fa-check"></i>Descriptive Statistics</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#normality-assessment-and-data-transformation"><i class="fa fa-check"></i>Normality Assessment and Data Transformation</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#concluding-remarks-13"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html"><i class="fa fa-check"></i>4.3 Data Import from PDF Sources</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#introduction-to-training-module-14"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-from-many-single-pdfs-with-the-same-formatting"><i class="fa fa-check"></i>Importing Data from Many Single PDFs with the Same Formatting</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-stored-in-pdf-tables"><i class="fa fa-check"></i>Importing Data Stored in PDF Tables</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#concluding-remarks-14"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html"><i class="fa fa-check"></i>4.4 Two Group Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#introduction-to-training-module-15"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#overview-of-two-group-statistical-tests"><i class="fa fa-check"></i>Overview of Two Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#statistical-vs.-biological-significance"><i class="fa fa-check"></i>Statistical vs. Biological Significance</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#unpaired-test-example"><i class="fa fa-check"></i>Unpaired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#paired-test-example"><i class="fa fa-check"></i>Paired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#visualizing-results"><i class="fa fa-check"></i>Visualizing Results</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#concluding-remarks-15"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html"><i class="fa fa-check"></i>4.5 Multi-Group and Multi-Variable Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#introduction-to-training-module-16"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#overview-of-multi-group-statistical-tests"><i class="fa fa-check"></i>Overview of Multi-Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#multi-group-analysis-example"><i class="fa fa-check"></i>Multi-Group Analysis Example</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#visualization-of-multi-group-statistical-results"><i class="fa fa-check"></i>Visualization of Multi-Group Statistical Results</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#concluding-remarks-16"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html"><i class="fa fa-check"></i>4.6 Advanced Multi-Group Comparisons</a>
<ul>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#introduction-to-training-module-17"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#advanced-multi-group-comparisons-1"><i class="fa fa-check"></i>Advanced Multi-Group Comparisons</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#two-way-anova-example"><i class="fa fa-check"></i>Two-way ANOVA Example</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#ancova-example"><i class="fa fa-check"></i>ANCOVA Example</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#concluding-remarks-17"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 5 Machine Learning <br>&amp; Artificial Intelligence</b></span></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><i class="fa fa-check"></i>5.1 Introduction to Artificial Intelligence, Machine Learning, and Predictive Modeling for Environmental Health</a>
<ul>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#introduction-to-training-module-18"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#general-historical-context-and-taxonomy-of-modern-aiml"><i class="fa fa-check"></i>General Historical Context and Taxonomy of Modern AI/ML</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#application-of-machine-learning-in-environmental-health-science"><i class="fa fa-check"></i>Application of Machine Learning in Environmental Health Science</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#concluding-remarks-18"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html"><i class="fa fa-check"></i>5.2 Supervised Machine Learning</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-training-module-19"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-machine-learning"><i class="fa fa-check"></i>Types of Machine Learning</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-supervised-machine-learning-algorithms"><i class="fa fa-check"></i>Types of Supervised Machine Learning Algorithms</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#training-supervised-machine-learning-models"><i class="fa fa-check"></i>Training Supervised Machine Learning Models</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#assessing-classification-based-model-performance"><i class="fa fa-check"></i>Assessing Classification-Based Model Performance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-activity-and-example-dataset"><i class="fa fa-check"></i>Introduction to Activity and Example Dataset</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#testing-for-differences-in-predictor-variables-across-the-outcome-classes"><i class="fa fa-check"></i>Testing for Differences in Predictor Variables across the Outcome Classes</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#predicting-ias-detection-with-a-random-forest-rf-model"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#class-imbalance"><i class="fa fa-check"></i>Class Imbalance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#concluding-remarks-19"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html"><i class="fa fa-check"></i>5.3 Supervised Machine Learning Model Interpretation</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-training-module-20"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance"><i class="fa fa-check"></i>Variable Importance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary"><i class="fa fa-check"></i>Decision Boundary</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-example-dataset-and-activity"><i class="fa fa-check"></i>Introduction to Example Dataset and Activity</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#predicting-ias-detection-with-a-random-forest-rf-model-1"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance-plot"><i class="fa fa-check"></i>Variable Importance Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot"><i class="fa fa-check"></i>Decision Boundary Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot-incorporating-smote"><i class="fa fa-check"></i>Decision Boundary Plot Incorporating SMOTE</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#concluding-remarks-20"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><i class="fa fa-check"></i>5.4 Unsupervised Machine Learning Part 1: K-Means Clustering &amp; PCA</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-training-module-21"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-unsupervised-machine-learning"><i class="fa fa-check"></i>Introduction to Unsupervised Machine Learning</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-example-data"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#identifying-clusters-of-chemicals-through-k-means"><i class="fa fa-check"></i>Identifying Clusters of Chemicals through <em>K</em>-Means</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#principal-component-analysis-pca-1"><i class="fa fa-check"></i>Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#incorporating-k-means-into-pca-for-predictive-modeling"><i class="fa fa-check"></i>Incorporating <em>K</em>-Means into PCA for Predictive Modeling</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#concluding-remarks-21"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><i class="fa fa-check"></i>5.5 Unsupervised Machine Learning Part 2: Additional Clustering Applications</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-training-module-22"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#k-means-clustering-1"><i class="fa fa-check"></i><em>K</em>-Means Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-example-data-1"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-1"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#clustering-plot"><i class="fa fa-check"></i>Clustering Plot</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-visualization"><i class="fa fa-check"></i>Hierarchical Clustering Visualization</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#variable-contributions"><i class="fa fa-check"></i>Variable Contributions</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#concluding-remarks-22"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 6 Applications in <br>Toxicology &amp; Exposure <br>Science</b></span></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html"><i class="fa fa-check"></i>6.1 Descriptive Cohort Analyses</a>
<ul>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#introduction-to-training-module-23"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#participant-exploration"><i class="fa fa-check"></i>Participant Exploration</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#chemical-detection-filtering"><i class="fa fa-check"></i>Chemical Detection Filtering</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#outlier-identification"><i class="fa fa-check"></i>Outlier Identification</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#summary-statistics-tables"><i class="fa fa-check"></i>Summary Statistics Tables</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#demographics-table"><i class="fa fa-check"></i>Demographics Table</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#concluding-remarks-23"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html"><i class="fa fa-check"></i>6.2 -Omics and System Biology: Transcriptomic Applications</a>
<ul>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-training-module-24"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-the-field-of--omics"><i class="fa fa-check"></i>Introduction to the Field of “-Omics”</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-transcriptomics"><i class="fa fa-check"></i>Introduction to Transcriptomics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#formatting-data-for-downstream-statistics"><i class="fa fa-check"></i>Formatting Data for Downstream Statistics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#transcriptomics-data-qaqc"><i class="fa fa-check"></i>Transcriptomics Data QA/QC</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#controlling-for-sources-of-sample-heterogeneity"><i class="fa fa-check"></i>Controlling for Sources of Sample Heterogeneity</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e.g.-biomass-smoke-exposure"><i class="fa fa-check"></i>Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure)</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-ma-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using MA Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-volcano-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using Volcano Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis"><i class="fa fa-check"></i>Interpretting Findings at the Systems Level through Pathway Enrichment Analysis</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#concluding-remarks-24"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><i class="fa fa-check"></i>6.3 Mixtures Analysis Methods Part 1: Overview and Example with Quantile G-Computation</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-training-module-25"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#overview-of-mixtures-analysis"><i class="fa fa-check"></i>Overview of Mixtures Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-example-data-2"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#mixtures-model-with-standard-qqcomp"><i class="fa fa-check"></i>Mixtures Model with Standard qqcomp</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#concluding-remarks-25"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#additional-resources-11"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><i class="fa fa-check"></i>6.4 Mixtures Analysis Methods Part 2: Bayesian Kernel Machine Regression</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-training-module-26"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-example-data-3"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-bkmr"><i class="fa fa-check"></i>Introduction to BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#run-bkmr"><i class="fa fa-check"></i>Run BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#concluding-remarks-26"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#additional-resources-12"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html"><i class="fa fa-check"></i>6.5 Mixtures Analysis Methods Part 3: Sufficient Similarity</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-training-module-27"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-toxicant-and-dataset"><i class="fa fa-check"></i>Introduction to Toxicant and Dataset</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#workspace-preparation-and-data-import-9"><i class="fa fa-check"></i>Workspace Preparation and Data Import</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#chemistry-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Chemistry-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#toxicity-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Toxicity-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#comparing-chemistry-vs.-toxicity-sufficient-similarity-analyses"><i class="fa fa-check"></i>Comparing Chemistry vs. Toxicity Sufficient Similarity Analyses</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#concluding-remarks-27"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i>6.6 Toxicokinetic Modeling</a>
<ul>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-training-module-28"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-toxicokinetic-modeling"><i class="fa fa-check"></i>Introduction to Toxicokinetic Modeling</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#data-and-models-used-in-toxicokinetic-modeling-tk"><i class="fa fa-check"></i>Data and Models used in Toxicokinetic Modeling (TK)</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-toxicokinetics"><i class="fa fa-check"></i>Reverse Toxicokinetics</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#capturing-population-variability-in-toxicokinetics-and-uncertainty-in-chemical-specific-parameters"><i class="fa fa-check"></i>Capturing Population Variability in Toxicokinetics, and Uncertainty in Chemical-Specific Parameters</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-tk-calculating-administered-equivalent-doses-for-toxcast-bioactive-concentrations"><i class="fa fa-check"></i>Reverse TK: Calculating Administered Equivalent Doses for ToxCast Bioactive Concentrations</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#comparing-equivalent-doses-estimated-to-elicit-toxicity-hazard-to-external-exposure-estimates-exposure-for-chemical-prioritization-by-bioactivity-exposure-ratios-bers"><i class="fa fa-check"></i>Comparing Equivalent Doses Estimated to Elicit Toxicity (Hazard) to External Exposure Estimates (Exposure), for Chemical Prioritization by Bioactivity-Exposure Ratios (BERs)</a></li>
</ul></li>
<li class="part"><span><b>Chapter 7 Environmental<br> Health Database Mining</b></span></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html"><i class="fa fa-check"></i>7.1 Comparative Toxicogenomics Database</a>
<ul>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#introduction-to-training-module-30"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#ctd-data-in-r"><i class="fa fa-check"></i>CTD Data in R</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#identifying-genes-under-epigenetic-control"><i class="fa fa-check"></i>Identifying Genes Under Epigenetic Control</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#concluding-remarks-30"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html"><i class="fa fa-check"></i>7.2 Gene Expression Omnibus</a>
<ul>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-training-module-31"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-geo"><i class="fa fa-check"></i>Introduction to GEO</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-example-data-4"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#geo-data-in-r"><i class="fa fa-check"></i>GEO Data in R</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#visualizing-data"><i class="fa fa-check"></i>Visualizing Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#statistical-analyses"><i class="fa fa-check"></i>Statistical Analyses</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#concluding-remarks-31"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html"><i class="fa fa-check"></i>7.3 CompTox Dashboard Data through APIs</a>
<ul>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#introduction-to-training-module-32"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#training-modules-environmental-health-questions-24"><i class="fa fa-check"></i>Training Module’s Environmental Health Questions</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#script-preparations-12"><i class="fa fa-check"></i>Script Preparations</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#introduction-to-comptox-chemicals-dashboard"><i class="fa fa-check"></i>Introduction to CompTox Chemicals Dashboard</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#cctes-ctx-application-programming-interfaces-apis-for-automated-batch-search-of-the-ccd"><i class="fa fa-check"></i>CCTE’s CTX Application Programming Interfaces (APIs) for Automated Batch Search of the CCD</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#comparing-physico-chemical-properties-between-two-important-environmental-contaminant-lists"><i class="fa fa-check"></i>Comparing Physico-chemical Properties between Two Important Environmental Contaminant Lists</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#hazard-data-genotoxicity"><i class="fa fa-check"></i>Hazard Data: Genotoxicity</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#hazard-resource"><i class="fa fa-check"></i>Hazard Resource</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#concluding-remarks-32"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html"><i class="fa fa-check"></i>7.4 Database Integration: Air Quality, Mortality, and Environmental Justice Data</a>
<ul>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#introduction-to-training-module-33"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#introduction-to-exposure-and-health-databases"><i class="fa fa-check"></i>Introduction to Exposure and Health Databases</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#analyzing-relationships-between-pm2.5-and-mortality"><i class="fa fa-check"></i>Analyzing Relationships between PM<sub>2.5</sub> and Mortality</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#environmental-justice-considerations"><i class="fa fa-check"></i>Environmental Justice Considerations</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#concluding-remarks-33"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data-import-processing-and-summary-statistics" class="section level1 hasAnchor">
<h1>4.2 Data Import, Processing, and Summary Statistics<a href="data-import-processing-and-summary-statistics.html#data-import-processing-and-summary-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This training module was developed by Elise Hickman, Alexis Payton, Sarah Miller, and Julia E. Rager.</p>
<p>All input files (script, data, and figures) can be downloaded from the <a href="https://github.com/UNCSRP/TAME2">UNC-SRP TAME2 GitHub website</a>.</p>
<div id="introduction-to-training-module-13" class="section level2 hasAnchor">
<h2>Introduction to Training Module<a href="data-import-processing-and-summary-statistics.html#introduction-to-training-module-13" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The first steps in any scripted analysis of wet-bench data include importing the data, cleaning the data to prepare for analyses, and conducting preliminary data exploration steps, such as addressing missing values, calculating summary statistics, and assessing normality. Although less exciting than diving right into the statistical analysis, these steps are crucial in guiding downstream analyses and ensuring accurate results. In this module, we will discuss each of these steps and work through them using an example dataset (introduced in <strong>TAME 2.0 Module 4.1 Overview of Experimental Design and Example Data</strong> of inflammatory markers secreted by airway epithelial cells after exposure to different concentrations of acrolein.</p>
<div id="training-modules-environmental-health-questions-6" class="section level3 hasAnchor">
<h3>Training Module’s Environmental Health Questions<a href="data-import-processing-and-summary-statistics.html#training-modules-environmental-health-questions-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This training module was specifically developed to answer the following environmental health questions:</p>
<ol style="list-style-type: decimal">
<li><p>What is the mean concentration of each inflammatory biomarker by acrolein concentration?</p></li>
<li><p>Are our data normally distributed?</p></li>
</ol>
<p><br></p>
</div>
</div>
<div id="data-import" class="section level2 hasAnchor">
<h2>Data Import<a href="data-import-processing-and-summary-statistics.html#data-import" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>First, we need to import our data. Data can be imported into R from many different file formats, including .csv (as demonstrated in previous chapters), .txt, .xlsx, and .pdf. Often, data are formatted in Excel prior to import, and the <a href="https://ycphs.github.io/openxlsx/"><em>openxlsx</em></a> package provides helpful functions that allow the user to import data from Excel, create workbooks for storing results generated in R, and export data from R to Excel workbooks. Below, we will use the <code>read.xlsx()</code> function to import our data directly from Excel. Other useful packages include <a href="https://github.com/ropensci/pdftools"><em>pdftools</em></a> (PDF import), <a href="https://cran.r-project.org/web/packages/tm/vignettes/tm.pdf"><em>tm</em></a> (text mining of PDFs), and <a href="https://cran.r-project.org/web/packages/plater/vignettes/plater-basics.html"><em>plater</em></a> (plate reader formatted data import).
<img src="Module4_2_Input/Module4_2_Image1.png" width="850px" style="display: block; margin: auto;" /></p>
<div id="workspace-preparation-and-data-import-1" class="section level3 hasAnchor">
<h3>Workspace Preparation and Data Import<a href="data-import-processing-and-summary-statistics.html#workspace-preparation-and-data-import-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="set-working-directory" class="section level4 hasAnchor">
<h4>Set working directory<a href="data-import-processing-and-summary-statistics.html#set-working-directory" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>In preparation, first let’s set our working directory to the folder path that contains our input files:</p>
<div class="sourceCode" id="cb322"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb322-1"><a href="data-import-processing-and-summary-statistics.html#cb322-1" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/filepath to where your input files are&quot;</span>)</span></code></pre></div>
</div>
<div id="installing-required-r-packages-4" class="section level4 hasAnchor">
<h4>Installing required R packages<a href="data-import-processing-and-summary-statistics.html#installing-required-r-packages-4" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>If you already have these packages installed, you can skip this step, or you can run the below code which checks installation status for you</p>
<div class="sourceCode" id="cb323"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb323-1"><a href="data-import-processing-and-summary-statistics.html#cb323-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;table1&quot;</span>))</span>
<span id="cb323-2"><a href="data-import-processing-and-summary-statistics.html#cb323-2" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;table1&quot;</span>);</span>
<span id="cb323-3"><a href="data-import-processing-and-summary-statistics.html#cb323-3" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;vtable&quot;</span>))</span>
<span id="cb323-4"><a href="data-import-processing-and-summary-statistics.html#cb323-4" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;vtable&quot;</span>);</span>
<span id="cb323-5"><a href="data-import-processing-and-summary-statistics.html#cb323-5" tabindex="-1"></a><span class="co"># some packages need to be installed through Bioconductor/ BiocManager</span></span>
<span id="cb323-6"><a href="data-import-processing-and-summary-statistics.html#cb323-6" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">require</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="at">quietly =</span> <span class="cn">TRUE</span>))</span>
<span id="cb323-7"><a href="data-import-processing-and-summary-statistics.html#cb323-7" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb323-8"><a href="data-import-processing-and-summary-statistics.html#cb323-8" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;pcaMethods&quot;</span>)</span>
<span id="cb323-9"><a href="data-import-processing-and-summary-statistics.html#cb323-9" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;impute&quot;</span>)</span>
<span id="cb323-10"><a href="data-import-processing-and-summary-statistics.html#cb323-10" tabindex="-1"></a>BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;imputeLCMD&quot;</span>)</span></code></pre></div>
</div>
<div id="load-required-packages" class="section level4 hasAnchor">
<h4>Load required packages<a href="data-import-processing-and-summary-statistics.html#load-required-packages" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>And load required packages:</p>
<div class="sourceCode" id="cb324"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb324-1"><a href="data-import-processing-and-summary-statistics.html#cb324-1" tabindex="-1"></a><span class="fu">library</span>(openxlsx) <span class="co"># for importing Excel files</span></span>
<span id="cb324-2"><a href="data-import-processing-and-summary-statistics.html#cb324-2" tabindex="-1"></a><span class="fu">library</span>(DT) <span class="co"># for easier viewing of data tables</span></span>
<span id="cb324-3"><a href="data-import-processing-and-summary-statistics.html#cb324-3" tabindex="-1"></a><span class="fu">library</span>(tidyverse) <span class="co"># for data cleaning and graphing</span></span>
<span id="cb324-4"><a href="data-import-processing-and-summary-statistics.html#cb324-4" tabindex="-1"></a><span class="fu">library</span>(imputeLCMD) <span class="co"># for data imputation with QRILC</span></span>
<span id="cb324-5"><a href="data-import-processing-and-summary-statistics.html#cb324-5" tabindex="-1"></a><span class="fu">library</span>(table1) <span class="co"># for summary table</span></span>
<span id="cb324-6"><a href="data-import-processing-and-summary-statistics.html#cb324-6" tabindex="-1"></a><span class="fu">library</span>(vtable) <span class="co"># for summary table</span></span>
<span id="cb324-7"><a href="data-import-processing-and-summary-statistics.html#cb324-7" tabindex="-1"></a><span class="fu">library</span>(ggpubr) <span class="co"># for making Q-Q plots with ggplot </span></span></code></pre></div>
</div>
<div id="import-example-datasets" class="section level4 hasAnchor">
<h4>Import example datasets<a href="data-import-processing-and-summary-statistics.html#import-example-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Next, let’s read in our example datasets:</p>
<div class="sourceCode" id="cb325"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb325-1"><a href="data-import-processing-and-summary-statistics.html#cb325-1" tabindex="-1"></a>biomarker_data <span class="ot">&lt;-</span> <span class="fu">read.xlsx</span>(<span class="st">&quot;Module4_2_Input/Module4_2_InputData1.xlsx&quot;</span>)</span>
<span id="cb325-2"><a href="data-import-processing-and-summary-statistics.html#cb325-2" tabindex="-1"></a>demographic_data <span class="ot">&lt;-</span> <span class="fu">read.xlsx</span>(<span class="st">&quot;Module4_2_Input/Module4_2_InputData2.xlsx&quot;</span>)</span></code></pre></div>
</div>
<div id="view-example-datasets" class="section level4 hasAnchor">
<h4>View example datasets<a href="data-import-processing-and-summary-statistics.html#view-example-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>First, let’s preview our example data. Using the <code>datatable()</code> function from the <em>DT</em> package allows us to interactively scroll through our biomarker data.</p>
<div class="sourceCode" id="cb326"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb326-1"><a href="data-import-processing-and-summary-statistics.html#cb326-1" tabindex="-1"></a><span class="fu">datatable</span>(biomarker_data)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-0c7682d03e8d04050c57" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-0c7682d03e8d04050c57">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160"],["D1","D2","D3","D4","D5","D6","D7","D8","D9","D10","D11","D12","D13","D14","D15","D16","D1","D2","D3","D4","D5","D6","D7","D8","D9","D10","D11","D12","D13","D14","D15","D16","D1","D2","D3","D4","D5","D6","D7","D8","D9","D10","D11","D12","D13","D14","D15","D16","D1","D2","D3","D4","D5","D6","D7","D8","D9","D10","D11","D12","D13","D14","D15","D16","D1","D2","D3","D4","D5","D6","D7","D8","D9","D10","D11","D12","D13","D14","D15","D16","D1","D2","D3","D4","D5","D6","D7","D8","D9","D10","D11","D12","D13","D14","D15","D16","D1","D2","D3","D4","D5","D6","D7","D8","D9","D10","D11","D12","D13","D14","D15","D16","D1","D2","D3","D4","D5","D6","D7","D8","D9","D10","D11","D12","D13","D14","D15","D16","D1","D2","D3","D4","D5","D6","D7","D8","D9","D10","D11","D12","D13","D14","D15","D16","D1","D2","D3","D4","D5","D6","D7","D8","D9","D10","D11","D12","D13","D14","D15","D16"],[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,0.6,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4,4],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2],[13.26,11.02,11.32,11.18,8.69,6.71,7.87,10.11,8.869999999999999,12.39,14.11,9.73,9.83,9.75,10.34571428571429,10.11,12.42,11.1,12.54,10.24,8,7.28,7.88,10.93,8.720000000000001,12.06,14.36,8.59,9.07,9.949999999999999,10.22428571428572,10.22428571428572,12.55,11.59,11.61,12.02,8.35,11.76,9.81,11.59,11.13,12.73,14.78,11.45,9.43,11.8,11.47142857142857,11.59,13.23,8.43,12.53,3.13,12.25,11.37,10.33,12.99,10.96,12.66,12.73,10.61,10.31,13.72,11.08928571428571,11.37,10.63,12.81,9.880000000000001,17.28,13.6,11.93,10.66,14.16,12.77,12.46,13.6,7.86,12.42,13.84,12.42142857142857,12.46,12,10.15,10.6,9.880000000000001,7.26,5.83,6.65,8.68,7.74,11.92,12.43,8.800000000000001,8.67,8.16,9.197857142857142,8.800000000000001,13.02,11.3,12.16,10.51,8.16,7.71,8.26,10.91,9.09,12.36,14.15,8.789999999999999,9.130000000000001,9.69,10.37428571428572,10.37428571428572,12.34,10.47,10.94,10.61,7.22,11.33,7.52,10.48,9.789999999999999,11.37,13.67,10.34,8.470000000000001,10.07,10.33,10.47,13.44,8.699999999999999,9.5,13.68,12.26,11.94,10.94,13.5,10.9,12.53,13.25,10.68,10.41,13.64,11.81214285714286,11.94,10.09,12.14,9.18,15.99,12.24,11.11,8.76,13.65,11.74,11.05,12.6,7.31,11.41,12.61,11.42,11.42],[1100.47,472.1,611.87,328.26,14.84,174.38,277.68,2772.77,128.48,414.77,1187.36,615.6799999999999,283.72,187.51,612.135,414.77,1298.94,637.35,828.29,321.39,15.2,114.23,264.92,3417.01,158.27,438.2,1154.57,532.6,269.57,192.66,688.8000000000001,438.2,833.9299999999999,524.11,614.99,439.01,18.2,165.19,228.98,2801.85,131.09,380.11,989.41,508.88,170.14,242.6,574.8921428571429,439.01,1044.69,381.44,772.01,641.17,40.57,123.33,161.79,1859.58,155.81,854.5700000000001,899.76,304.68,163.61,228.69,545.1214285714285,381.44,1123.37,28.83,751.36,219.63,34.57,323.07,481.13,357.62,24.57,190.44,259.75,193.53,16.79,88.91,292.3978571428572,219.63,1107.72,474.48,588.45,321.59,13.91,169.02,280.06,2720.11,127.44,405.93,1199.72,617.52,285.5,188.15,607.1142857142858,405.93,1283.64,636.29,806.98,329.85,17.58,114.66,263.26,3369.52,161.08,435.3,1130.06,526.09,273.04,190.55,681.2785714285714,435.3,811.62,525.76,585.74,434.27,17.83,166.14,235.38,2819.04,129.11,373.4,983.37,511.28,169.22,234.69,571.2035714285714,434.27,1022.94,393.13,760.5700000000001,650.22,42.54,124.06,165.67,1902.58,152.02,818.87,901.15,313.42,167.34,227.12,545.8307142857144,393.13,1146.83,27.84,756.48,195.28,35.88,333.79,478.45,377.49,25.45,182.44,263.98,197.61,15.58,91.31999999999999,294.8871428571429,197.61],[36292.57,25439.34,24654.6,23825.7,17504.9,11099.84,12415.64,18131.15,20614.12,35493.56,33949.25,15885.92,20604.15,16270.8,22298.68142857143,20614.12,46107.81,25876.96,30163.14,22211.63,15943.24,11196.96,14713.98,21742.66,18827.56,36444.95,48042.38,16594.76,19994.08,17964.85,24701.78285714286,21742.66,36166.34,null,26696.07,29626.69,17143.72,13923.51,15166.47,22525.33,null,35901.82,46541.14,null,20075.02,22585.19,26031.93636363637,24308.56318181818,42729.62,31695.42,34751.24,64650.82,38096.75,47679.53,22894.16,33252.23,26798.35,81542.88,74197.75999999999,29052.07,19997.71,36989.15,41737.69214285714,36989.15,123431.3,161270.2,144296.4,41552.38,83413.5,78688.66,137315.2,37062.96,85823.24000000001,108839.5,69392.60000000001,84240.08,102113.6,84952.55,95885.15500000003,85823.24000000001,37033.11,24499.93,24966.09,23887.29,17616.43,10569.15,12178.42,17368.88,20550.84,34456.03,31713.68,15808.44,19893.89,15527.56,21862.12428571429,20550.84,46936.91,26452.63,30843.75,22811.07,16464.42,11423.21,14581.4,22215.35,18670.46,37835.25,47679.75,17309.63,19654.98,18069.5,25067.73642857143,22215.35,36194.8,null,26424.8,29906.94,17343.9,13767.75,14775.73,22228.43,null,35274.58,45466.64,null,19170.36,21962.01,25683.26727272727,23955.84863636364,44222.37,30332.27,35141.75,65716.10000000001,39143.03,48158.27,22397.82,33115.8,26747.61,85040.21000000001,73174.89999999999,28667.48,20823.16,38032.79,42193.82571428572,38032.79,121294,159996.4,141851.8,41551.05,83766.34,80988.08,131484.6,35575.33,80646.31,107679.3,71046.63,83514.25999999999,97088.25,83421.81,94278.86857142857,83766.34],[0.32,2.89,0.21,1.19,0.48,1.48,1.04,0.88,1.39,0.22,1.35,0.78,0.43,1.94,1.042857142857143,1.04,0.08,0.15,0.07000000000000001,0.37,0.16,1.93,1.64,1.13,1.02,null,1.09,0.23,null,0.16,0.6691666666666666,0.37,0.5,0.48,0.61,1.76,0.29,0.75,2.4,0.9399999999999999,0.34,0.29,1.39,0.7,4.12,1.05,1.115714285714285,0.75,0.04,0.62,0.65,2.05,1.96,0.98,0.08,0.73,0.47,0.27,1,1.77,1.39,1.6,0.9721428571428572,0.9721428571428572,0.59,1.7,0.3,3.66,0.89,0.48,1.28,1.99,1.83,1.2,2.31,0.36,0.9,1.57,1.361428571428571,1.28,0.26,0.23,0.27,0.48,1.34,3.8,0.1,1.16,8.029999999999999,0.05,6.71,0.1,2.31,0.25,1.792142857142857,0.48,0.11,0.31,0.11,0.09,0.15,1.39,0.02,0.72,0.01,null,2.11,0.12,null,0.14,0.4399999999999999,0.14,0.26,0.62,0.3,0.29,10.6,1.23,0.18,1.18,0.31,0.16,1.55,0.58,null,0.67,1.379230769230769,0.6,1.73,0.73,0.17,0.54,0.44,1.05,0.17,0.93,0.54,0.19,0.9399999999999999,1.91,0.9,0.76,0.7857142857142857,0.76,0.16,1.6,0.2,2.53,0.72,0.25,1.24,1.96,1.73,1,1.92,0.16,0.73,1.61,1.129285714285714,1.129285714285714],[2.21,2.14,7.36,3.53,3.63,1.93,1.1,3.72,0.74,1.98,7.27,2,1.01,1.61,2.873571428571428,2.14,2.56,2.72,1.61,3.65,3.5,2.51,1.11,4.26,1.12,1.89,8.25,2.53,1.03,2.11,2.775000000000001,2.53,2.51,2.73,2.15,3.29,2.42,4.01,18.78,4.49,2.34,2.15,6.57,3.64,0.86,1.88,4.13,2.73,2.41,3.98,1.62,4.92,4,6.71,1.74,6.21,4.33,2.92,7.89,9.460000000000001,5.34,5.7,4.802142857142857,4.802142857142857,3.54,6.52,2.61,11.85,5.8,2.94,4.27,9.1,7.22,6.1,8.85,2.02,4.68,7.04,5.895714285714286,5.895714285714286,3.07,2.98,2.15,5.58,3.59,2.98,2.82,5.83,1.47,2.67,9.56,2.21,2,3.93,3.631428571428572,2.98,2.37,3.32,2.03,4.26,4.67,4.14,1.88,4.92,1.91,1.99,8.65,3,1.58,2.49,3.372142857142857,3,2.78,3.38,3.08,5.21,3.38,4.52,3.42,6.65,3.55,3.51,9.210000000000001,4.56,1.77,3.93,4.210714285714285,3.55,2.97,5.22,2.05,9.029999999999999,4.57,7.55,2.91,7.77,5.68,3.57,8.050000000000001,7.44,6.14,6.36,5.664999999999999,5.68,9.220000000000001,7.36,3.77,12.72,6.7,4.03,7.25,11.08,7.75,6.16,11.51,1.92,5.44,10.29,7.514285714285713,7.36],[840.15,1473.95,659.11,317.44,684.41,1747.14,1283.13,963.13,888.54,1403.65,523.91,688.66,607.05,1619.15,978.5299999999999,888.54,843.51,1797.04,594.4400000000001,296.94,634.76,1991.14,1316.97,911.71,1087.69,1366.48,593.3,686.04,627.85,1592.17,1024.288571428571,911.71,724.58,1756.42,533.08,393.74,789.4299999999999,2897.94,1286.39,1055.57,1376.92,1397.72,678.89,793.83,1204.1,1823.47,1193.72,1193.72,1055.99,3508.22,641.35,670.59,1398.74,2525.07,1666.68,1555.25,2306.87,1672.74,942.61,1938.45,2418.53,2049.55,1739.331428571429,1672.74,2087.9,2103.69,2113.62,511.01,1674,954.04,3401.2,677.6,1948.54,2433.39,638.5,159.97,1875.89,1953.13,1609.462857142857,1875.89,817.28,1491.14,652.09,303.76,688.6900000000001,1734.19,1295.18,978.83,889.84,1387.21,518.89,687.02,606.72,1600.92,975.1257142857141,889.84,849.7,1820.32,571.1900000000001,298.29,631.72,2050.29,1329.88,930.85,1083.1,1394.52,602.22,674.5700000000001,652.22,1613.52,1035.885,930.85,714.8099999999999,1819.75,513.11,389.76,775.8200000000001,2968.98,1367.26,1059.32,1391.71,1402.56,682.28,778.09,1171.59,1777.33,1200.883571428572,1171.59,1056.56,3572.19,674.8,692.03,1421.1,2595.99,1694.72,1625.5,2367.68,1701.4,982.65,1950.98,2484.71,2066.93,1777.66,1701.4,2061.36,2080.9,2042.7,492.12,1659.16,949.41,3399.82,673.67,1951.32,2421.46,634.86,154.07,1828.42,1958.01,1593.377142857142,1828.42]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Donor<\/th>\n      <th>Dose<\/th>\n      <th>Replicate<\/th>\n      <th>IL1B<\/th>\n      <th>IL6<\/th>\n      <th>IL8<\/th>\n      <th>IL10<\/th>\n      <th>TNFa<\/th>\n      <th>VEGF<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7,8,9]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Donor","targets":1},{"name":"Dose","targets":2},{"name":"Replicate","targets":3},{"name":"IL1B","targets":4},{"name":"IL6","targets":5},{"name":"IL8","targets":6},{"name":"IL10","targets":7},{"name":"TNFa","targets":8},{"name":"VEGF","targets":9}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>We can see that our biomarker data are arranged with samples in rows and sample information and biomarker measurements in the columns.</p>
<div class="sourceCode" id="cb327"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb327-1"><a href="data-import-processing-and-summary-statistics.html#cb327-1" tabindex="-1"></a><span class="fu">datatable</span>(demographic_data)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-5028538cdf1acf899bfe" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-5028538cdf1acf899bfe">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16"],["D1","D2","D3","D4","D5","D6","D7","D8","D9","D10","D11","D12","D13","D14","D15","D16"],[22,35,27,60,43,38,26,36,40,44,46,52,58,65,37,25],["M","F","M","F","F","M","M","F","F","M","M","M","M","M","F","F"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Donor<\/th>\n      <th>Age_yr<\/th>\n      <th>Sex<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Donor","targets":1},{"name":"Age_yr","targets":2},{"name":"Sex","targets":3}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Our demographic data provide information about the donors that our cells came from, matching to the <code>Donor</code> column in our biomarker data.</p>
<p><br></p>
</div>
</div>
</div>
<div id="handling-missing-values" class="section level2 hasAnchor">
<h2>Handling Missing Values<a href="data-import-processing-and-summary-statistics.html#handling-missing-values" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Next, we will investigate whether we have missing values and which variables and donors have missing values.</p>
<div class="sourceCode" id="cb328"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb328-1"><a href="data-import-processing-and-summary-statistics.html#cb328-1" tabindex="-1"></a><span class="co"># Calculate the total number of NAs per variable</span></span>
<span id="cb328-2"><a href="data-import-processing-and-summary-statistics.html#cb328-2" tabindex="-1"></a>biomarker_data <span class="sc">%&gt;%</span> </span>
<span id="cb328-3"><a href="data-import-processing-and-summary-statistics.html#cb328-3" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(IL1B<span class="sc">:</span>VEGF, <span class="sc">~</span><span class="fu">sum</span>(<span class="fu">is.na</span>(.))))</span></code></pre></div>
<pre><code>##   IL1B IL6 IL8 IL10 TNFa VEGF
## 1    0   0   6    5    0    0</code></pre>
<div class="sourceCode" id="cb330"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb330-1"><a href="data-import-processing-and-summary-statistics.html#cb330-1" tabindex="-1"></a><span class="co"># Calculate the number of missing values per subject</span></span>
<span id="cb330-2"><a href="data-import-processing-and-summary-statistics.html#cb330-2" tabindex="-1"></a>biomarker_data <span class="sc">%&gt;%</span></span>
<span id="cb330-3"><a href="data-import-processing-and-summary-statistics.html#cb330-3" tabindex="-1"></a>  <span class="fu">group_by</span>(Donor) <span class="sc">%&gt;%</span></span>
<span id="cb330-4"><a href="data-import-processing-and-summary-statistics.html#cb330-4" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(IL1B<span class="sc">:</span>VEGF, <span class="sc">~</span><span class="fu">sum</span>(<span class="fu">is.na</span>(.))))</span></code></pre></div>
<pre><code>## # A tibble: 16 × 7
##    Donor  IL1B   IL6   IL8  IL10  TNFa  VEGF
##    &lt;chr&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt; &lt;int&gt;
##  1 D1        0     0     0     0     0     0
##  2 D10       0     0     0     2     0     0
##  3 D11       0     0     0     0     0     0
##  4 D12       0     0     2     0     0     0
##  5 D13       0     0     0     3     0     0
##  6 D14       0     0     0     0     0     0
##  7 D15       0     0     0     0     0     0
##  8 D16       0     0     0     0     0     0
##  9 D2        0     0     2     0     0     0
## 10 D3        0     0     0     0     0     0
## 11 D4        0     0     0     0     0     0
## 12 D5        0     0     0     0     0     0
## 13 D6        0     0     0     0     0     0
## 14 D7        0     0     0     0     0     0
## 15 D8        0     0     0     0     0     0
## 16 D9        0     0     2     0     0     0</code></pre>
<p>Here, we can see that we do have a few missing values. What should we do with these values?</p>
<div id="missing-values-and-data-imputation" class="section level3 hasAnchor">
<h3>Missing Values and Data Imputation<a href="data-import-processing-and-summary-statistics.html#missing-values-and-data-imputation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="missing-values" class="section level4 hasAnchor">
<h4>Missing values<a href="data-import-processing-and-summary-statistics.html#missing-values" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Before deciding what to do about our missing values, it’s important to understand why they are missing. There are a few different types of missing values that could be present in a dataset:</p>
<ol style="list-style-type: decimal">
<li><p><strong>Missing completely at random (MCAR):</strong> has nothing to do with the experimental unit being studied (e.g., a sample is damaged or lost in the lab)</p></li>
<li><p><strong>Missing at random (MAR):</strong> there may be a systematic difference between missing and measured values, but they can be explained by observed differences in the data or experimental unit</p></li>
<li><p><strong>Missing not at random (MNAR):</strong> data are missing due to factors that are not observed/measured (e.g., measurement for a specific endpoint is below the limit of detection (LOD) of an assay)</p></li>
</ol>
<p>We know from the researchers who generated this dataset that the values are missing because these specific proteins were below the limit of detection for the assay for certain samples; therefore, our data are missing not at random. This can help us with our choice of imputation method, described below.</p>
</div>
<div id="imputation" class="section level4 hasAnchor">
<h4>Imputation<a href="data-import-processing-and-summary-statistics.html#imputation" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Imputation is the assignment of a value to a missing data point by inferring that value from other properties of the dataset or externally defined limits. Whether or not you should impute your data is not a one-size-fits-all approach and may vary depending on your field, experimental design, the type of data, and the type of missing values in your dataset. Two questions you can ask yourself when deciding whether or not to impute data are:</p>
<ol style="list-style-type: decimal">
<li><p>Is imputation needed for downstream analyses? <em>Some analyses are not permissive to including NAs or 0s; others are.</em></p></li>
<li><p>Will imputing values bias my analyses unnecessarily? <em>If so, consider analyzing subsets of the data that are complete separately.</em></p></li>
</ol>
<p>There are many different imputation methods (too many to cover them all in this module); here, we will introduce a few that we use most often. We encourage you to explore these in more depth and to understand typical imputation workflows for your lab, data type, and/or discipline.</p>
<ul>
<li><p>For variables where imputed values are expected to be generally bound by the existing range of data (e.g., MCAR): <a href="https://rpubs.com/lmorgan95/MissForest">missForest</a></p></li>
<li><p>For variables with samples below the limit of detection for the assay, such as for mass spectrometry or ELISAs (e.g., MNAR)</p>
<ul>
<li>Replace non-detects with the limit of detection divided by the square root of 2</li>
<li><a href="https://www.nature.com/articles/s41598-017-19120-0">Quantile Regression Imputation of Left-Censored Data (QRILC)</a></li>
<li><a href="https://github.com/WandeRum/GSimp">GSimp</a> (can also be used to impute values above a specific threshold)</li>
</ul></li>
</ul>
<p>If you do impute missing values, make sure to include both your raw and imputed data, along with detailed information about the imputation method, within your manuscript, supplemental information, and/or GitHub. You can even present summary statistics for both raw and imputed data for additional transparency.</p>
</div>
</div>
<div id="imputation-of-our-data" class="section level3 hasAnchor">
<h3>Imputation of Our Data<a href="data-import-processing-and-summary-statistics.html#imputation-of-our-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Before imputing our data, it is a good idea to implement a background filter that checks to see if a certain percentage of values for each variable are missing. For variables with a very high percentage of missing values, imputation can be unreliable because there is not enough information for the imputation algorithm to reference. The threshold for what this percentage should be can vary by study design and the extent to which your data are subset into groups that may have differing biomarker profiles; however, a common threshold we frequently use is to remove variables with missing data for 25% or more of samples.</p>
<p>We can use the following code to calculate the percentage values missing for each endpoint:</p>
<div class="sourceCode" id="cb332"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb332-1"><a href="data-import-processing-and-summary-statistics.html#cb332-1" tabindex="-1"></a>biomarker_data <span class="sc">%&gt;%</span> </span>
<span id="cb332-2"><a href="data-import-processing-and-summary-statistics.html#cb332-2" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(IL1B<span class="sc">:</span>VEGF, <span class="sc">~</span><span class="fu">sum</span>(<span class="fu">is.na</span>(.))<span class="sc">/</span><span class="fu">nrow</span>(biomarker_data)<span class="sc">*</span><span class="dv">100</span>))</span></code></pre></div>
<pre><code>##   IL1B IL6  IL8  IL10 TNFa VEGF
## 1    0   0 3.75 3.125    0    0</code></pre>
<p>Here, we can see that only about 3-4% of values are missing for our variables with missing data, so we will proceed to imputation with our dataset as-is.</p>
<p>We will impute values using QRILC, which pulls from the left side of the data distribution (the lower values) to impute missing values. We will write a function that will apply QRILC imputation to our dataframe. This function takes a dataframe with missing values as input and returns a dataframe with QRILC imputed values in place of NAs as output.</p>
<div class="sourceCode" id="cb334"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb334-1"><a href="data-import-processing-and-summary-statistics.html#cb334-1" tabindex="-1"></a>QRILC_imputation <span class="ot">=</span> <span class="cf">function</span>(df){</span>
<span id="cb334-2"><a href="data-import-processing-and-summary-statistics.html#cb334-2" tabindex="-1"></a>    <span class="co"># Normalize data before applying QRILC per QRILC documentation</span></span>
<span id="cb334-3"><a href="data-import-processing-and-summary-statistics.html#cb334-3" tabindex="-1"></a>    <span class="do">## Select only numeric columns, psuedo log2 transform, and convert to a matrix</span></span>
<span id="cb334-4"><a href="data-import-processing-and-summary-statistics.html#cb334-4" tabindex="-1"></a>    <span class="do">### 4 comes from there being 3 metadata columns before the numeric data starts</span></span>
<span id="cb334-5"><a href="data-import-processing-and-summary-statistics.html#cb334-5" tabindex="-1"></a>    QRILC_prep <span class="ot">=</span> df[,<span class="dv">4</span><span class="sc">:</span><span class="fu">dim</span>(df)[<span class="dv">2</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb334-6"><a href="data-import-processing-and-summary-statistics.html#cb334-6" tabindex="-1"></a>         <span class="fu">mutate_all</span>(., <span class="cf">function</span>(x) <span class="fu">log2</span>(x <span class="sc">+</span> <span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb334-7"><a href="data-import-processing-and-summary-statistics.html#cb334-7" tabindex="-1"></a>         <span class="fu">as.matrix</span>()</span>
<span id="cb334-8"><a href="data-import-processing-and-summary-statistics.html#cb334-8" tabindex="-1"></a>    </span>
<span id="cb334-9"><a href="data-import-processing-and-summary-statistics.html#cb334-9" tabindex="-1"></a>    <span class="co"># QRILC imputation</span></span>
<span id="cb334-10"><a href="data-import-processing-and-summary-statistics.html#cb334-10" tabindex="-1"></a>    imputed_QRILC_object <span class="ot">=</span> <span class="fu">impute.QRILC</span>(QRILC_prep, <span class="at">tune.sigma =</span> <span class="fl">0.1</span>)</span>
<span id="cb334-11"><a href="data-import-processing-and-summary-statistics.html#cb334-11" tabindex="-1"></a>    QRILC_log2_df <span class="ot">=</span> <span class="fu">data.frame</span>(imputed_QRILC_object[<span class="dv">1</span>]) </span>
<span id="cb334-12"><a href="data-import-processing-and-summary-statistics.html#cb334-12" tabindex="-1"></a>    </span>
<span id="cb334-13"><a href="data-import-processing-and-summary-statistics.html#cb334-13" tabindex="-1"></a>    <span class="co"># Converting back the original scale</span></span>
<span id="cb334-14"><a href="data-import-processing-and-summary-statistics.html#cb334-14" tabindex="-1"></a>    QRILC_df <span class="ot">=</span> QRILC_log2_df <span class="sc">%&gt;%</span></span>
<span id="cb334-15"><a href="data-import-processing-and-summary-statistics.html#cb334-15" tabindex="-1"></a>        <span class="fu">mutate_all</span>(., <span class="cf">function</span>(x) <span class="dv">2</span><span class="sc">^</span>x <span class="sc">-</span> <span class="dv">1</span>)</span>
<span id="cb334-16"><a href="data-import-processing-and-summary-statistics.html#cb334-16" tabindex="-1"></a>    </span>
<span id="cb334-17"><a href="data-import-processing-and-summary-statistics.html#cb334-17" tabindex="-1"></a>    <span class="co"># Adding back in metadata columns</span></span>
<span id="cb334-18"><a href="data-import-processing-and-summary-statistics.html#cb334-18" tabindex="-1"></a>    QRILC_df <span class="ot">=</span> <span class="fu">cbind</span>(<span class="at">Donor =</span> df<span class="sc">$</span>Donor,</span>
<span id="cb334-19"><a href="data-import-processing-and-summary-statistics.html#cb334-19" tabindex="-1"></a>                     <span class="at">Dose =</span> df<span class="sc">$</span>Dose,</span>
<span id="cb334-20"><a href="data-import-processing-and-summary-statistics.html#cb334-20" tabindex="-1"></a>                     <span class="at">Replicate =</span> df<span class="sc">$</span>Replicate,</span>
<span id="cb334-21"><a href="data-import-processing-and-summary-statistics.html#cb334-21" tabindex="-1"></a>                     QRILC_df)</span>
<span id="cb334-22"><a href="data-import-processing-and-summary-statistics.html#cb334-22" tabindex="-1"></a>    </span>
<span id="cb334-23"><a href="data-import-processing-and-summary-statistics.html#cb334-23" tabindex="-1"></a>   <span class="fu">return</span>(QRILC_df)</span>
<span id="cb334-24"><a href="data-import-processing-and-summary-statistics.html#cb334-24" tabindex="-1"></a>}</span></code></pre></div>
<p>Now we can apply the <code>QRILC_imputation()</code> function to our dataframe. We use the function <code>set.seed()</code> to ensure that the QRILC function generates the same numbers each time we run the script. For more on setting seeds, see <a href="https://www.statology.org/set-seed-in-r/">here</a>.</p>
<div class="sourceCode" id="cb335"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb335-1"><a href="data-import-processing-and-summary-statistics.html#cb335-1" tabindex="-1"></a><span class="co"># Set random seed to ensure reproducibility in results</span></span>
<span id="cb335-2"><a href="data-import-processing-and-summary-statistics.html#cb335-2" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1104</span>)</span>
<span id="cb335-3"><a href="data-import-processing-and-summary-statistics.html#cb335-3" tabindex="-1"></a></span>
<span id="cb335-4"><a href="data-import-processing-and-summary-statistics.html#cb335-4" tabindex="-1"></a><span class="co"># Apply function</span></span>
<span id="cb335-5"><a href="data-import-processing-and-summary-statistics.html#cb335-5" tabindex="-1"></a>biomarker_data_imp <span class="ot">&lt;-</span> <span class="fu">QRILC_imputation</span>(biomarker_data)</span></code></pre></div>
<p><br></p>
</div>
</div>
<div id="averaging-replicates" class="section level2 hasAnchor">
<h2>Averaging Replicates<a href="data-import-processing-and-summary-statistics.html#averaging-replicates" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The last step we need to take before our data are ready for analysis is averaging the two technical replicates for each donor and dose. We will do this by creating an ID column that represents the donor and dose together and using that column to group and average the data. This results in a dataframe where our rows contain data representing each biological replicate exposed to each of the five concentrations of acrolein.</p>
<div class="sourceCode" id="cb336"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb336-1"><a href="data-import-processing-and-summary-statistics.html#cb336-1" tabindex="-1"></a>biomarker_data_imp_avg <span class="ot">&lt;-</span> biomarker_data_imp <span class="sc">%&gt;%</span></span>
<span id="cb336-2"><a href="data-import-processing-and-summary-statistics.html#cb336-2" tabindex="-1"></a>  </span>
<span id="cb336-3"><a href="data-import-processing-and-summary-statistics.html#cb336-3" tabindex="-1"></a>  <span class="co"># Create an ID column that represents the donor and dose</span></span>
<span id="cb336-4"><a href="data-import-processing-and-summary-statistics.html#cb336-4" tabindex="-1"></a>  <span class="fu">unite</span>(Donor_Dose, Donor, Dose, <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb336-5"><a href="data-import-processing-and-summary-statistics.html#cb336-5" tabindex="-1"></a>  </span>
<span id="cb336-6"><a href="data-import-processing-and-summary-statistics.html#cb336-6" tabindex="-1"></a>  <span class="co"># Average replicates with each unique Donor_Dose</span></span>
<span id="cb336-7"><a href="data-import-processing-and-summary-statistics.html#cb336-7" tabindex="-1"></a>  <span class="fu">group_by</span>(Donor_Dose) <span class="sc">%&gt;%</span></span>
<span id="cb336-8"><a href="data-import-processing-and-summary-statistics.html#cb336-8" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="fu">across</span>(IL1B<span class="sc">:</span>VEGF, mean)) <span class="sc">%&gt;%</span></span>
<span id="cb336-9"><a href="data-import-processing-and-summary-statistics.html#cb336-9" tabindex="-1"></a>  </span>
<span id="cb336-10"><a href="data-import-processing-and-summary-statistics.html#cb336-10" tabindex="-1"></a>  <span class="co"># Round results to the same number of significant figures as the original data</span></span>
<span id="cb336-11"><a href="data-import-processing-and-summary-statistics.html#cb336-11" tabindex="-1"></a>   <span class="fu">mutate</span>(<span class="fu">across</span>(IL1B<span class="sc">:</span>VEGF, \(x) <span class="fu">round</span>(x, <span class="dv">2</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb336-12"><a href="data-import-processing-and-summary-statistics.html#cb336-12" tabindex="-1"></a></span>
<span id="cb336-13"><a href="data-import-processing-and-summary-statistics.html#cb336-13" tabindex="-1"></a>  <span class="co"># Separate back out the Donor_Dose column</span></span>
<span id="cb336-14"><a href="data-import-processing-and-summary-statistics.html#cb336-14" tabindex="-1"></a>  <span class="fu">separate</span>(Donor_Dose, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&quot;Donor&quot;</span>, <span class="st">&quot;Dose&quot;</span>), <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb336-15"><a href="data-import-processing-and-summary-statistics.html#cb336-15" tabindex="-1"></a></span>
<span id="cb336-16"><a href="data-import-processing-and-summary-statistics.html#cb336-16" tabindex="-1"></a><span class="co"># View new dataframe</span></span>
<span id="cb336-17"><a href="data-import-processing-and-summary-statistics.html#cb336-17" tabindex="-1"></a><span class="fu">datatable</span>(biomarker_data_imp_avg)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-bda90b7e1fd90be724e5" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-bda90b7e1fd90be724e5">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80"],["D10","D10","D10","D10","D10","D11","D11","D11","D11","D11","D12","D12","D12","D12","D12","D13","D13","D13","D13","D13","D14","D14","D14","D14","D14","D15","D15","D15","D15","D15","D16","D16","D16","D16","D16","D1","D1","D1","D1","D1","D2","D2","D2","D2","D2","D3","D3","D3","D3","D3","D4","D4","D4","D4","D4","D5","D5","D5","D5","D5","D6","D6","D6","D6","D6","D7","D7","D7","D7","D7","D8","D8","D8","D8","D8","D9","D9","D9","D9","D9"],["0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4"],[12.16,12.21,12.05,12.6,11.75,13.27,14.25,14.23,12.99,13.1,9.27,8.69,10.89,10.64,7.58,9.25,9.1,8.949999999999999,10.36,11.91,8.960000000000001,9.82,10.94,13.68,13.23,9.77,10.3,10.9,11.45,11.92,9.460000000000001,10.3,11.03,11.66,11.94,12.63,12.72,12.45,13.33,10.36,10.59,11.2,11.03,8.57,12.47,10.96,12.35,11.28,11.02,9.529999999999999,10.53,10.38,11.31,8.4,16.63,7.98,8.08,7.78,12.25,12.92,6.27,7.5,11.55,11.66,11.52,7.26,8.07,8.66,10.64,9.710000000000001,9.390000000000001,10.92,11.04,13.25,13.9,8.31,8.91,10.46,10.93,12.25],[410.35,436.75,376.76,836.72,186.44,1193.54,1142.31,986.39,900.45,261.86,616.6,529.34,510.08,309.05,195.57,284.61,271.3,169.68,165.48,16.18,187.83,191.6,238.65,227.9,90.12,609.62,685.04,573.05,545.48,293.64,410.35,436.75,436.64,387.29,208.62,1104.09,1291.29,822.77,1033.82,1135.1,473.29,636.8200000000001,524.9400000000001,387.29,28.34,600.16,817.64,600.36,766.29,753.92,324.92,325.62,436.64,645.7,207.45,14.38,16.39,18.01,41.55,35.22,171.7,114.45,165.66,123.69,328.43,278.87,264.09,232.18,163.73,479.79,2746.44,3393.26,2810.45,1881.08,367.56,127.96,159.68,130.1,153.92,25.01],[34974.79,37140.1,35588.2,83291.53999999999,108259.4,32831.47,47861.07,46003.89,73686.33,70219.61,15847.18,16952.19,9090.41,28859.78,83877.17,20249.02,19824.53,19622.69,20410.44,99600.92,15899.18,18017.18,22273.6,37510.97,84187.17999999999,22080.4,24884.76,25857.6,41965.76,95082.00999999999,20582.48,21979,24132.21,37510.97,84794.78999999999,36662.84,46522.36,36180.57,43476,122362.65,24969.64,26164.79,9087.049999999999,31013.84,160633.3,24810.35,30503.45,26560.43,34946.49,143074.1,23856.5,22511.35,29766.81,65183.46,41551.72,17560.67,16203.83,17243.81,38619.89,83589.92,10834.49,11310.08,13845.63,47918.9,79838.37,12297.03,14647.69,14971.1,22645.99,134399.9,17750.02,21979,22376.88,33184.02,36319.14,20582.48,18749.01,9107.459999999999,26772.98,83234.77],[0.14,-0.19,0.22,0.23,1.1,4.03,1.6,1.47,0.97,2.12,0.44,0.18,0.64,1.84,0.26,1.37,-0.19,1.96,1.14,0.8100000000000001,1.09,0.15,0.86,1.18,1.59,1.42,0.55,1.25,0.88,1.25,0.76,0.26,0.68,0.87,1.2,0.29,0.1,0.38,0.88,0.37,1.56,0.23,0.55,0.68,1.65,0.24,0.09,0.45,0.41,0.25,0.84,0.23,1.02,1.29,3.09,0.91,0.15,5.44,1.2,0.8,2.64,1.66,0.99,1.01,0.36,0.57,0.83,1.29,0.12,1.26,1.02,0.92,1.06,0.83,1.98,4.71,0.52,0.33,0.5,1.78],[2.33,1.94,2.83,3.24,6.13,8.42,8.449999999999999,7.89,7.97,10.18,2.1,2.76,4.1,8.449999999999999,1.97,1.5,1.31,1.31,5.74,5.06,2.77,2.3,2.9,6.03,8.66,3.25,3.07,4.17,5.23,6.7,2.56,2.76,3.14,5.24,6.63,2.64,2.46,2.64,2.69,6.38,2.56,3.02,3.06,4.6,6.94,4.76,1.82,2.62,1.83,3.19,4.56,3.96,4.25,6.97,12.29,3.61,4.08,2.9,4.28,6.25,2.46,3.32,4.26,7.13,3.49,1.96,1.5,11.1,2.33,5.76,4.78,4.59,5.57,6.99,10.09,1.1,1.52,2.94,5,7.48],[1395.43,1380.5,1400.14,1687.07,2427.43,521.4,597.76,680.58,962.63,636.6799999999999,687.84,680.3,785.96,1944.71,157.02,606.88,640.03,1187.85,2451.62,1852.15,1610.03,1602.84,1800.4,2058.24,1955.57,976.83,1030.09,1197.3,1758.5,1601.42,889.1900000000001,921.28,1182.65,1687.07,1852.15,828.72,846.6,719.7,1056.28,2074.63,1482.54,1808.68,1788.09,3540.2,2092.3,655.6,582.8099999999999,523.09,658.08,2078.16,310.6,297.61,391.75,681.3099999999999,501.57,686.55,633.24,782.63,1409.92,1666.58,1740.66,2020.71,2933.46,2560.53,951.72,1289.16,1323.43,1326.83,1680.7,3400.51,970.98,921.28,1057.45,1590.37,675.64,889.1900000000001,1085.39,1384.32,2337.27,1949.93]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Donor<\/th>\n      <th>Dose<\/th>\n      <th>IL1B<\/th>\n      <th>IL6<\/th>\n      <th>IL8<\/th>\n      <th>IL10<\/th>\n      <th>TNFa<\/th>\n      <th>VEGF<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4,5,6,7,8]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Donor","targets":1},{"name":"Dose","targets":2},{"name":"IL1B","targets":3},{"name":"IL6","targets":4},{"name":"IL8","targets":5},{"name":"IL10","targets":6},{"name":"TNFa","targets":7},{"name":"VEGF","targets":8}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br></p>
</div>
<div id="descriptive-statistics" class="section level2 hasAnchor">
<h2>Descriptive Statistics<a href="data-import-processing-and-summary-statistics.html#descriptive-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Generating descriptive statistics (e.g., mean, median, mode, range, standard deviation) can be helpful for understanding the general distribution of your data and for reporting results either in the main body of a manuscript/report (for small datasets) or in the supplementary material (for larger datasets). There are a number of different approaches that can be used to calculate summary statistics, including functions that are part of base R and that are part of packages. Here, we will demonstrate a few different ways to efficiently calculate descriptive statistics across our dataset.</p>
<div id="method-1---tidyverse-and-basic-functions" class="section level3 hasAnchor">
<h3>Method #1 - Tidyverse and Basic Functions<a href="data-import-processing-and-summary-statistics.html#method-1---tidyverse-and-basic-functions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The mean, or average of data points, is one of the most commonly reported summary statistics and is often reported as mean ± standard deviation to demonstrate the spread in the data. Here, we will make a table of mean ± standard deviation for each of our biomarkers across each of the dose groups using <em>tidyverse</em> functions.</p>
<div class="sourceCode" id="cb337"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb337-1"><a href="data-import-processing-and-summary-statistics.html#cb337-1" tabindex="-1"></a><span class="co"># Calculate means</span></span>
<span id="cb337-2"><a href="data-import-processing-and-summary-statistics.html#cb337-2" tabindex="-1"></a>biomarker_group_means <span class="ot">&lt;-</span> biomarker_data_imp_avg <span class="sc">%&gt;%</span></span>
<span id="cb337-3"><a href="data-import-processing-and-summary-statistics.html#cb337-3" tabindex="-1"></a>  <span class="fu">group_by</span>(Dose) <span class="sc">%&gt;%</span></span>
<span id="cb337-4"><a href="data-import-processing-and-summary-statistics.html#cb337-4" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(IL1B<span class="sc">:</span>VEGF, \(x) <span class="fu">mean</span>(x))) </span>
<span id="cb337-5"><a href="data-import-processing-and-summary-statistics.html#cb337-5" tabindex="-1"></a></span>
<span id="cb337-6"><a href="data-import-processing-and-summary-statistics.html#cb337-6" tabindex="-1"></a><span class="co"># View data</span></span>
<span id="cb337-7"><a href="data-import-processing-and-summary-statistics.html#cb337-7" tabindex="-1"></a><span class="fu">datatable</span>(biomarker_group_means)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-d559c1de0d5645782923" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-d559c1de0d5645782923">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5"],["0","0.6","1","2","4"],[9.75375,10.3,10.909375,11.464375,11.92],[597.1693749999999,669.520625,564.5225,535.59,288.328125],[21986.78375,24703.149375,22606.77125,41687.335,94439.059375],[1.376875,0.443125,1.161875,0.876875,1.241875],[3.21,3.05375,4.105,5.2325,6.7],[971.35,1023.284375,1196.3875,1754.03125,1617.09125]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Dose<\/th>\n      <th>IL1B<\/th>\n      <th>IL6<\/th>\n      <th>IL8<\/th>\n      <th>IL10<\/th>\n      <th>TNFa<\/th>\n      <th>VEGF<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Dose","targets":1},{"name":"IL1B","targets":2},{"name":"IL6","targets":3},{"name":"IL8","targets":4},{"name":"IL10","targets":5},{"name":"TNFa","targets":6},{"name":"VEGF","targets":7}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>You’ll notice that there are a lot of decimal places in our calculated means, while in our original data, there are only two decimal places. We can add a step to round the data to our above code chunk to produce cleaner results.</p>
<div class="sourceCode" id="cb338"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb338-1"><a href="data-import-processing-and-summary-statistics.html#cb338-1" tabindex="-1"></a><span class="co"># Calculate means</span></span>
<span id="cb338-2"><a href="data-import-processing-and-summary-statistics.html#cb338-2" tabindex="-1"></a>biomarker_group_means <span class="ot">&lt;-</span> biomarker_data_imp_avg <span class="sc">%&gt;%</span></span>
<span id="cb338-3"><a href="data-import-processing-and-summary-statistics.html#cb338-3" tabindex="-1"></a>  <span class="fu">group_by</span>(Dose) <span class="sc">%&gt;%</span></span>
<span id="cb338-4"><a href="data-import-processing-and-summary-statistics.html#cb338-4" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(IL1B<span class="sc">:</span>VEGF, \(x) <span class="fu">mean</span>(x))) <span class="sc">%&gt;%</span></span>
<span id="cb338-5"><a href="data-import-processing-and-summary-statistics.html#cb338-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(IL1B<span class="sc">:</span>VEGF, \(x) <span class="fu">round</span>(x, <span class="dv">2</span>)))</span>
<span id="cb338-6"><a href="data-import-processing-and-summary-statistics.html#cb338-6" tabindex="-1"></a></span>
<span id="cb338-7"><a href="data-import-processing-and-summary-statistics.html#cb338-7" tabindex="-1"></a><span class="co"># View data</span></span>
<span id="cb338-8"><a href="data-import-processing-and-summary-statistics.html#cb338-8" tabindex="-1"></a><span class="fu">datatable</span>(biomarker_group_means)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-8e2f75a0425662906ca3" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-8e2f75a0425662906ca3">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5"],["0","0.6","1","2","4"],[9.75,10.3,10.91,11.46,11.92],[597.17,669.52,564.52,535.59,288.33],[21986.78,24703.15,22606.77,41687.34,94439.06],[1.38,0.44,1.16,0.88,1.24],[3.21,3.05,4.1,5.23,6.7],[971.35,1023.28,1196.39,1754.03,1617.09]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Dose<\/th>\n      <th>IL1B<\/th>\n      <th>IL6<\/th>\n      <th>IL8<\/th>\n      <th>IL10<\/th>\n      <th>TNFa<\/th>\n      <th>VEGF<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Dose","targets":1},{"name":"IL1B","targets":2},{"name":"IL6","targets":3},{"name":"IL8","targets":4},{"name":"IL10","targets":5},{"name":"TNFa","targets":6},{"name":"VEGF","targets":7}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="answer-to-environmental-health-question-1-5" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 1<a href="data-import-processing-and-summary-statistics.html#answer-to-environmental-health-question-1-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><i>With this, we can answer <strong>Environmental Health Question 1</strong>:</i> What is the mean concentration of each inflammatory biomarker by acrolein concentration?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> With the above table, we can see the mean concentrations for each of our inflammatory biomarkers by acrolein dose. IL-8 overall has the highest concentrations, followed by VEGF and IL-6. For IL-1<span class="math inline">\(\beta\)</span>, IL-8, TNF-<span class="math inline">\(\alpha\)</span>, and VEGF, it appears that the concentration of the biomarker goes up with increasing dose.</p>
</div>
<p>We can use very similar code to calculate our standard deviations:</p>
<div class="sourceCode" id="cb339"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb339-1"><a href="data-import-processing-and-summary-statistics.html#cb339-1" tabindex="-1"></a><span class="co"># Calculate means</span></span>
<span id="cb339-2"><a href="data-import-processing-and-summary-statistics.html#cb339-2" tabindex="-1"></a>biomarker_group_sds <span class="ot">&lt;-</span> biomarker_data_imp_avg <span class="sc">%&gt;%</span></span>
<span id="cb339-3"><a href="data-import-processing-and-summary-statistics.html#cb339-3" tabindex="-1"></a>  <span class="fu">group_by</span>(Dose) <span class="sc">%&gt;%</span></span>
<span id="cb339-4"><a href="data-import-processing-and-summary-statistics.html#cb339-4" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(IL1B<span class="sc">:</span>VEGF, \(x) <span class="fu">sd</span>(x))) <span class="sc">%&gt;%</span></span>
<span id="cb339-5"><a href="data-import-processing-and-summary-statistics.html#cb339-5" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(IL1B<span class="sc">:</span>VEGF, \(x) <span class="fu">round</span>(x, <span class="dv">1</span>)))</span>
<span id="cb339-6"><a href="data-import-processing-and-summary-statistics.html#cb339-6" tabindex="-1"></a></span>
<span id="cb339-7"><a href="data-import-processing-and-summary-statistics.html#cb339-7" tabindex="-1"></a><span class="co"># View data</span></span>
<span id="cb339-8"><a href="data-import-processing-and-summary-statistics.html#cb339-8" tabindex="-1"></a><span class="fu">datatable</span>(biomarker_group_sds)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-f40a05c531dafbaa65de" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f40a05c531dafbaa65de">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5"],["0","0.6","1","2","4"],[1.9,1.9,1.5,1.6,2],[658.2,810.1,652.3,474.2,297.2],[7570,10768,10647.6,17939.2,33535.4],[1.3,0.6,1.2,0.4,0.8],[1.8,1.7,2.4,2,2.7],[417,485.8,620.7,749.3,836.3]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Dose<\/th>\n      <th>IL1B<\/th>\n      <th>IL6<\/th>\n      <th>IL8<\/th>\n      <th>IL10<\/th>\n      <th>TNFa<\/th>\n      <th>VEGF<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6,7]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Dose","targets":1},{"name":"IL1B","targets":2},{"name":"IL6","targets":3},{"name":"IL8","targets":4},{"name":"IL10","targets":5},{"name":"TNFa","targets":6},{"name":"VEGF","targets":7}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Now we’ve calculated both the means and standard deviations! However, these are typically presented as mean ± standard deviation. We can merge these dataframes by executing the following steps:</p>
<ol style="list-style-type: decimal">
<li>Pivot each dataframe to a long format, with each row containing the value for one biomarker at one dose.</li>
<li>Create a variable that represents each unique row (combination of <code>Dose</code> and <code>variable</code>).</li>
<li>Join the dataframes by row.</li>
<li>Unite the two columns with mean and standard deviation, with <code>±</code> in between them.</li>
<li>Pivot the dataframe wider so that the dataframe resembles what we started with for the means and standard deviations.</li>
</ol>
<p>First, we’ll pivot each dataframe to a long format and create a variable that represents each unique row.</p>
<div class="sourceCode" id="cb340"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb340-1"><a href="data-import-processing-and-summary-statistics.html#cb340-1" tabindex="-1"></a><span class="co"># Pivot dataframes longer and create variable column for each row</span></span>
<span id="cb340-2"><a href="data-import-processing-and-summary-statistics.html#cb340-2" tabindex="-1"></a>biomarker_group_means_long <span class="ot">&lt;-</span> <span class="fu">pivot_longer</span>(biomarker_group_means, </span>
<span id="cb340-3"><a href="data-import-processing-and-summary-statistics.html#cb340-3" tabindex="-1"></a>                                           <span class="sc">!</span>Dose, <span class="at">names_to =</span> <span class="st">&quot;variable&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;mean&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb340-4"><a href="data-import-processing-and-summary-statistics.html#cb340-4" tabindex="-1"></a>  <span class="fu">unite</span>(Dose_variable, Dose, variable, <span class="at">remove =</span> <span class="cn">FALSE</span>)</span>
<span id="cb340-5"><a href="data-import-processing-and-summary-statistics.html#cb340-5" tabindex="-1"></a></span>
<span id="cb340-6"><a href="data-import-processing-and-summary-statistics.html#cb340-6" tabindex="-1"></a>biomarker_group_sds_long <span class="ot">&lt;-</span> <span class="fu">pivot_longer</span>(biomarker_group_means,</span>
<span id="cb340-7"><a href="data-import-processing-and-summary-statistics.html#cb340-7" tabindex="-1"></a>                                         <span class="sc">!</span>Dose, <span class="at">names_to =</span> <span class="st">&quot;variable&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;sd&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb340-8"><a href="data-import-processing-and-summary-statistics.html#cb340-8" tabindex="-1"></a>  <span class="fu">unite</span>(Dose_variable, Dose, variable, <span class="at">remove =</span> <span class="cn">FALSE</span>)</span>
<span id="cb340-9"><a href="data-import-processing-and-summary-statistics.html#cb340-9" tabindex="-1"></a></span>
<span id="cb340-10"><a href="data-import-processing-and-summary-statistics.html#cb340-10" tabindex="-1"></a></span>
<span id="cb340-11"><a href="data-import-processing-and-summary-statistics.html#cb340-11" tabindex="-1"></a><span class="co"># Preview what dataframe looks like</span></span>
<span id="cb340-12"><a href="data-import-processing-and-summary-statistics.html#cb340-12" tabindex="-1"></a><span class="fu">datatable</span>(biomarker_group_means_long)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-69bc5d3b326c7d25ed1d" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-69bc5d3b326c7d25ed1d">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30"],["0_IL1B","0_IL6","0_IL8","0_IL10","0_TNFa","0_VEGF","0.6_IL1B","0.6_IL6","0.6_IL8","0.6_IL10","0.6_TNFa","0.6_VEGF","1_IL1B","1_IL6","1_IL8","1_IL10","1_TNFa","1_VEGF","2_IL1B","2_IL6","2_IL8","2_IL10","2_TNFa","2_VEGF","4_IL1B","4_IL6","4_IL8","4_IL10","4_TNFa","4_VEGF"],["0","0","0","0","0","0","0.6","0.6","0.6","0.6","0.6","0.6","1","1","1","1","1","1","2","2","2","2","2","2","4","4","4","4","4","4"],["IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF"],[9.75,597.17,21986.78,1.38,3.21,971.35,10.3,669.52,24703.15,0.44,3.05,1023.28,10.91,564.52,22606.77,1.16,4.1,1196.39,11.46,535.59,41687.34,0.88,5.23,1754.03,11.92,288.33,94439.06,1.24,6.7,1617.09]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Dose_variable<\/th>\n      <th>Dose<\/th>\n      <th>variable<\/th>\n      <th>mean<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":4},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Dose_variable","targets":1},{"name":"Dose","targets":2},{"name":"variable","targets":3},{"name":"mean","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Next, we will join the mean and standard deviation datasets. Notice that we are only joining the <code>Dose_variable</code> and <code>sd</code> columns from the standard deviation dataframe to prevent duplicate columns (<code>Dose</code>, <code>variable</code>) from being included.</p>
<div class="sourceCode" id="cb341"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb341-1"><a href="data-import-processing-and-summary-statistics.html#cb341-1" tabindex="-1"></a><span class="co"># Merge the dataframes by row</span></span>
<span id="cb341-2"><a href="data-import-processing-and-summary-statistics.html#cb341-2" tabindex="-1"></a>biomarker_group_summstats <span class="ot">&lt;-</span> <span class="fu">left_join</span>(biomarker_group_means_long, </span>
<span id="cb341-3"><a href="data-import-processing-and-summary-statistics.html#cb341-3" tabindex="-1"></a>                                       biomarker_group_sds_long <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="fu">c</span>(Dose_variable, sd)), </span>
<span id="cb341-4"><a href="data-import-processing-and-summary-statistics.html#cb341-4" tabindex="-1"></a>                                       <span class="at">by =</span> <span class="st">&quot;Dose_variable&quot;</span>)</span>
<span id="cb341-5"><a href="data-import-processing-and-summary-statistics.html#cb341-5" tabindex="-1"></a></span>
<span id="cb341-6"><a href="data-import-processing-and-summary-statistics.html#cb341-6" tabindex="-1"></a><span class="co"># Preview the new dataframe</span></span>
<span id="cb341-7"><a href="data-import-processing-and-summary-statistics.html#cb341-7" tabindex="-1"></a><span class="fu">datatable</span>(biomarker_group_summstats)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-70d706548ed6efbe58c4" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-70d706548ed6efbe58c4">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30"],["0_IL1B","0_IL6","0_IL8","0_IL10","0_TNFa","0_VEGF","0.6_IL1B","0.6_IL6","0.6_IL8","0.6_IL10","0.6_TNFa","0.6_VEGF","1_IL1B","1_IL6","1_IL8","1_IL10","1_TNFa","1_VEGF","2_IL1B","2_IL6","2_IL8","2_IL10","2_TNFa","2_VEGF","4_IL1B","4_IL6","4_IL8","4_IL10","4_TNFa","4_VEGF"],["0","0","0","0","0","0","0.6","0.6","0.6","0.6","0.6","0.6","1","1","1","1","1","1","2","2","2","2","2","2","4","4","4","4","4","4"],["IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF"],[9.75,597.17,21986.78,1.38,3.21,971.35,10.3,669.52,24703.15,0.44,3.05,1023.28,10.91,564.52,22606.77,1.16,4.1,1196.39,11.46,535.59,41687.34,0.88,5.23,1754.03,11.92,288.33,94439.06,1.24,6.7,1617.09],[9.75,597.17,21986.78,1.38,3.21,971.35,10.3,669.52,24703.15,0.44,3.05,1023.28,10.91,564.52,22606.77,1.16,4.1,1196.39,11.46,535.59,41687.34,0.88,5.23,1754.03,11.92,288.33,94439.06,1.24,6.7,1617.09]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Dose_variable<\/th>\n      <th>Dose<\/th>\n      <th>variable<\/th>\n      <th>mean<\/th>\n      <th>sd<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[4,5]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Dose_variable","targets":1},{"name":"Dose","targets":2},{"name":"variable","targets":3},{"name":"mean","targets":4},{"name":"sd","targets":5}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Then, we can unite the mean and standard deviation columns and add the ± symbol between them by storing that character as a variable and pasting that variable in our <code>paste()</code> function.</p>
<div class="sourceCode" id="cb342"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb342-1"><a href="data-import-processing-and-summary-statistics.html#cb342-1" tabindex="-1"></a><span class="co"># Store plus/minus character</span></span>
<span id="cb342-2"><a href="data-import-processing-and-summary-statistics.html#cb342-2" tabindex="-1"></a>plusminus <span class="ot">&lt;-</span><span class="st">&quot;\u00b1&quot;</span></span>
<span id="cb342-3"><a href="data-import-processing-and-summary-statistics.html#cb342-3" tabindex="-1"></a><span class="fu">Encoding</span>(plusminus)<span class="ot">&lt;-</span><span class="st">&quot;UTF-8&quot;</span></span>
<span id="cb342-4"><a href="data-import-processing-and-summary-statistics.html#cb342-4" tabindex="-1"></a></span>
<span id="cb342-5"><a href="data-import-processing-and-summary-statistics.html#cb342-5" tabindex="-1"></a><span class="co"># Create new column with mean +/- standard deviation</span></span>
<span id="cb342-6"><a href="data-import-processing-and-summary-statistics.html#cb342-6" tabindex="-1"></a>biomarker_group_summstats <span class="ot">&lt;-</span> biomarker_group_summstats <span class="sc">%&gt;%</span></span>
<span id="cb342-7"><a href="data-import-processing-and-summary-statistics.html#cb342-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mean_sd =</span> <span class="fu">paste</span>(mean, plusminus, sd, <span class="at">sep =</span> <span class="st">&quot; &quot;</span>))</span>
<span id="cb342-8"><a href="data-import-processing-and-summary-statistics.html#cb342-8" tabindex="-1"></a></span>
<span id="cb342-9"><a href="data-import-processing-and-summary-statistics.html#cb342-9" tabindex="-1"></a><span class="co"># Preview the new dataframe</span></span>
<span id="cb342-10"><a href="data-import-processing-and-summary-statistics.html#cb342-10" tabindex="-1"></a><span class="fu">datatable</span>(biomarker_group_summstats)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-1010b4127e35e38f0e2f" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-1010b4127e35e38f0e2f">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30"],["0_IL1B","0_IL6","0_IL8","0_IL10","0_TNFa","0_VEGF","0.6_IL1B","0.6_IL6","0.6_IL8","0.6_IL10","0.6_TNFa","0.6_VEGF","1_IL1B","1_IL6","1_IL8","1_IL10","1_TNFa","1_VEGF","2_IL1B","2_IL6","2_IL8","2_IL10","2_TNFa","2_VEGF","4_IL1B","4_IL6","4_IL8","4_IL10","4_TNFa","4_VEGF"],["0","0","0","0","0","0","0.6","0.6","0.6","0.6","0.6","0.6","1","1","1","1","1","1","2","2","2","2","2","2","4","4","4","4","4","4"],["IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF","IL1B","IL6","IL8","IL10","TNFa","VEGF"],[9.75,597.17,21986.78,1.38,3.21,971.35,10.3,669.52,24703.15,0.44,3.05,1023.28,10.91,564.52,22606.77,1.16,4.1,1196.39,11.46,535.59,41687.34,0.88,5.23,1754.03,11.92,288.33,94439.06,1.24,6.7,1617.09],[9.75,597.17,21986.78,1.38,3.21,971.35,10.3,669.52,24703.15,0.44,3.05,1023.28,10.91,564.52,22606.77,1.16,4.1,1196.39,11.46,535.59,41687.34,0.88,5.23,1754.03,11.92,288.33,94439.06,1.24,6.7,1617.09],["9.75 ± 9.75","597.17 ± 597.17","21986.78 ± 21986.78","1.38 ± 1.38","3.21 ± 3.21","971.35 ± 971.35","10.3 ± 10.3","669.52 ± 669.52","24703.15 ± 24703.15","0.44 ± 0.44","3.05 ± 3.05","1023.28 ± 1023.28","10.91 ± 10.91","564.52 ± 564.52","22606.77 ± 22606.77","1.16 ± 1.16","4.1 ± 4.1","1196.39 ± 1196.39","11.46 ± 11.46","535.59 ± 535.59","41687.34 ± 41687.34","0.88 ± 0.88","5.23 ± 5.23","1754.03 ± 1754.03","11.92 ± 11.92","288.33 ± 288.33","94439.06 ± 94439.06","1.24 ± 1.24","6.7 ± 6.7","1617.09 ± 1617.09"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Dose_variable<\/th>\n      <th>Dose<\/th>\n      <th>variable<\/th>\n      <th>mean<\/th>\n      <th>sd<\/th>\n      <th>mean_sd<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[4,5]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Dose_variable","targets":1},{"name":"Dose","targets":2},{"name":"variable","targets":3},{"name":"mean","targets":4},{"name":"sd","targets":5},{"name":"mean_sd","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Last, we can pivot the dataframe wider to revert it to its original layout, which is easier to read.</p>
<div class="sourceCode" id="cb343"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb343-1"><a href="data-import-processing-and-summary-statistics.html#cb343-1" tabindex="-1"></a><span class="co"># Pivot dataframe wider</span></span>
<span id="cb343-2"><a href="data-import-processing-and-summary-statistics.html#cb343-2" tabindex="-1"></a>biomarker_group_summstats <span class="ot">&lt;-</span> biomarker_group_summstats <span class="sc">%&gt;%</span></span>
<span id="cb343-3"><a href="data-import-processing-and-summary-statistics.html#cb343-3" tabindex="-1"></a>  </span>
<span id="cb343-4"><a href="data-import-processing-and-summary-statistics.html#cb343-4" tabindex="-1"></a>  <span class="co"># Remove columns we don&#39;t need any more</span></span>
<span id="cb343-5"><a href="data-import-processing-and-summary-statistics.html#cb343-5" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(Dose_variable, mean, sd)) <span class="sc">%&gt;%</span></span>
<span id="cb343-6"><a href="data-import-processing-and-summary-statistics.html#cb343-6" tabindex="-1"></a>  </span>
<span id="cb343-7"><a href="data-import-processing-and-summary-statistics.html#cb343-7" tabindex="-1"></a>  <span class="co"># Pivot wider</span></span>
<span id="cb343-8"><a href="data-import-processing-and-summary-statistics.html#cb343-8" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> Dose, <span class="at">names_from =</span> <span class="st">&quot;variable&quot;</span>, <span class="at">values_from =</span> <span class="st">&quot;mean_sd&quot;</span>)</span>
<span id="cb343-9"><a href="data-import-processing-and-summary-statistics.html#cb343-9" tabindex="-1"></a></span>
<span id="cb343-10"><a href="data-import-processing-and-summary-statistics.html#cb343-10" tabindex="-1"></a><span class="co"># View final dataframe</span></span>
<span id="cb343-11"><a href="data-import-processing-and-summary-statistics.html#cb343-11" tabindex="-1"></a><span class="fu">datatable</span>(biomarker_group_summstats)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-3aaa9286154636757d26" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-3aaa9286154636757d26">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5"],["0","0.6","1","2","4"],["9.75 ± 9.75","10.3 ± 10.3","10.91 ± 10.91","11.46 ± 11.46","11.92 ± 11.92"],["597.17 ± 597.17","669.52 ± 669.52","564.52 ± 564.52","535.59 ± 535.59","288.33 ± 288.33"],["21986.78 ± 21986.78","24703.15 ± 24703.15","22606.77 ± 22606.77","41687.34 ± 41687.34","94439.06 ± 94439.06"],["1.38 ± 1.38","0.44 ± 0.44","1.16 ± 1.16","0.88 ± 0.88","1.24 ± 1.24"],["3.21 ± 3.21","3.05 ± 3.05","4.1 ± 4.1","5.23 ± 5.23","6.7 ± 6.7"],["971.35 ± 971.35","1023.28 ± 1023.28","1196.39 ± 1196.39","1754.03 ± 1754.03","1617.09 ± 1617.09"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Dose<\/th>\n      <th>IL1B<\/th>\n      <th>IL6<\/th>\n      <th>IL8<\/th>\n      <th>IL10<\/th>\n      <th>TNFa<\/th>\n      <th>VEGF<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Dose","targets":1},{"name":"IL1B","targets":2},{"name":"IL6","targets":3},{"name":"IL8","targets":4},{"name":"IL10","targets":5},{"name":"TNFa","targets":6},{"name":"VEGF","targets":7}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>These data are now in a publication-ready format that can be exported to a .txt, .csv., or .xlsx file for sharing.</p>
</div>
<div id="method-2---applying-a-list-of-functions" class="section level3 hasAnchor">
<h3>Method #2 - Applying a List of Functions<a href="data-import-processing-and-summary-statistics.html#method-2---applying-a-list-of-functions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Calculating our mean and standard deviation separately using <em>tidyverse</em> wasn’t too difficult, but what if we want to calculate other descriptive statistics, such as minimum, median, and maximum? We could use the above approach, but we would need to make a separate dataframe for each and then merge them all together. Instead, we can use the <code>map_dfr()</code> function from the <em>purrr</em> package, which is also part of <em>tidyverse.</em> This function takes a list of functions you want to apply to your data and applies these functions over specified columns in the data. Let’s see how it works:</p>
<div class="sourceCode" id="cb344"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb344-1"><a href="data-import-processing-and-summary-statistics.html#cb344-1" tabindex="-1"></a><span class="co"># Define summary functions</span></span>
<span id="cb344-2"><a href="data-import-processing-and-summary-statistics.html#cb344-2" tabindex="-1"></a>summary_functs <span class="ot">&lt;-</span> <span class="fu">lst</span>(min, median, mean, max, sd)</span>
<span id="cb344-3"><a href="data-import-processing-and-summary-statistics.html#cb344-3" tabindex="-1"></a></span>
<span id="cb344-4"><a href="data-import-processing-and-summary-statistics.html#cb344-4" tabindex="-1"></a><span class="co"># Apply functions to data, grouping by dose</span></span>
<span id="cb344-5"><a href="data-import-processing-and-summary-statistics.html#cb344-5" tabindex="-1"></a><span class="co"># .id = &quot;statistic&quot; tells the function to create a column describing which statistic that row is reporting</span></span>
<span id="cb344-6"><a href="data-import-processing-and-summary-statistics.html#cb344-6" tabindex="-1"></a>biomarker_descriptive_stats_all <span class="ot">&lt;-</span> <span class="fu">map_dfr</span>(summary_functs, </span>
<span id="cb344-7"><a href="data-import-processing-and-summary-statistics.html#cb344-7" tabindex="-1"></a>                                           <span class="sc">~</span> <span class="fu">summarize</span>(biomarker_data_imp_avg <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(Dose),</span>
<span id="cb344-8"><a href="data-import-processing-and-summary-statistics.html#cb344-8" tabindex="-1"></a>                                                       <span class="fu">across</span>(IL1B<span class="sc">:</span>VEGF, .x)), <span class="at">.id =</span> <span class="st">&quot;statistic&quot;</span>)</span>
<span id="cb344-9"><a href="data-import-processing-and-summary-statistics.html#cb344-9" tabindex="-1"></a></span>
<span id="cb344-10"><a href="data-import-processing-and-summary-statistics.html#cb344-10" tabindex="-1"></a><span class="co"># View data</span></span>
<span id="cb344-11"><a href="data-import-processing-and-summary-statistics.html#cb344-11" tabindex="-1"></a><span class="fu">datatable</span>(biomarker_descriptive_stats_all)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-16d2dfcd56cce7d5f0e7" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-16d2dfcd56cce7d5f0e7">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25"],["min","min","min","min","min","median","median","median","median","median","mean","mean","mean","mean","mean","max","max","max","max","max","sd","sd","sd","sd","sd"],["0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4"],[6.27,7.5,7.78,8.4,7.58,9.425000000000001,10.3,11.03,11.555,11.93,9.75375,10.3,10.909375,11.464375,11.92,13.27,14.25,14.23,13.68,16.63,1.899234494912797,1.905189404407516,1.513289876395134,1.56458287838431,2.044165681478224],[14.38,16.39,18.01,41.55,16.18,410.35,436.75,436.64,387.29,208.035,597.1693749999999,669.520625,564.5225,535.59,288.328125,2746.44,3393.26,2810.45,1881.08,1135.1,658.2397037398939,810.1192296896284,652.2647653266782,474.1640196950137,297.2423973621248],[10834.49,11310.08,9087.049999999999,20410.44,36319.14,20582.48,21979,22325.24,37510.97,84490.98499999999,21986.78375,24703.149375,22606.77125,41687.335,94439.059375,36662.84,47861.07,46003.89,83291.53999999999,160633.3,7570.028454551431,10768.04243681303,10647.63009995049,17939.24569853593,33535.41840978907],[0.14,-0.19,0.22,0.12,0.25,0.9650000000000001,0.23,0.925,0.88,1.225,1.376875,0.443125,1.161875,0.876875,1.241875,4.71,1.66,5.44,1.84,3.09,1.326864568572593,0.5551662663863262,1.2325,0.4307004953174151,0.7830982803369362],[1.1,1.31,1.31,1.83,1.97,2.6,2.76,3.1,5.235,6.505,3.21,3.05375,4.105,5.2325,6.7,8.42,8.449999999999999,11.1,8.449999999999999,12.29,1.76761232552088,1.733769977053857,2.387821880570939,2.004084163235999,2.690268883711564],[310.6,297.61,391.75,658.08,157.02,889.1900000000001,921.28,1185.25,1687.07,1852.15,971.35,1023.284375,1196.3875,1754.03125,1617.09125,1740.66,2020.71,2933.46,3540.2,3400.51,417.0417453924726,485.7680272444005,620.7245347763638,749.2713471622502,836.2607857829598]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>statistic<\/th>\n      <th>Dose<\/th>\n      <th>IL1B<\/th>\n      <th>IL6<\/th>\n      <th>IL8<\/th>\n      <th>IL10<\/th>\n      <th>TNFa<\/th>\n      <th>VEGF<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4,5,6,7,8]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"statistic","targets":1},{"name":"Dose","targets":2},{"name":"IL1B","targets":3},{"name":"IL6","targets":4},{"name":"IL8","targets":5},{"name":"IL10","targets":6},{"name":"TNFa","targets":7},{"name":"VEGF","targets":8}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Depending on your final goal, descriptive statistics data can then be extracted from this dataframe and cleaned up or reformatted as needed to create a publication-ready table!</p>
</div>
<div id="other-methods" class="section level3 hasAnchor">
<h3>Other Methods<a href="data-import-processing-and-summary-statistics.html#other-methods" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>There are also packages that have been developed for specifically making summary tables, such as <a href="https://cran.r-project.org/web/packages/table1/vignettes/table1-examples.html"><em>table1</em></a> and <a href="https://cran.r-project.org/web/packages/vtable/vignettes/sumtable.html"><em>vtable</em></a>. These packages can create summary tables in HTML format, which appear nicely in R Markdown and can be copied and pasted into Word. Here, we will briefly demonstrate how these packages work, and we encourage you to explore more using the package vignettes!</p>
<div id="table1" class="section level4 hasAnchor">
<h4>Table1<a href="data-import-processing-and-summary-statistics.html#table1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The <em>table1</em> package makes summary tables using the function <code>table1()</code>, which takes the columns that you want in the rows of the table on the left side of the first argument, followed by <code>|</code> and then the grouping variable. The output table can be customized in a number of ways, including what summary statistics are output and whether or not statistical comparisons are run between groups (see package vignette for more details).</p>
<div class="sourceCode" id="cb345"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb345-1"><a href="data-import-processing-and-summary-statistics.html#cb345-1" tabindex="-1"></a><span class="co"># Get names of all of the columns to include in the table</span></span>
<span id="cb345-2"><a href="data-import-processing-and-summary-statistics.html#cb345-2" tabindex="-1"></a><span class="fu">paste</span>(<span class="fu">names</span>(biomarker_data_imp_avg <span class="sc">%&gt;%</span> <span class="fu">select</span>(IL1B<span class="sc">:</span>VEGF)), <span class="at">collapse=</span><span class="st">&quot; + &quot;</span>)</span></code></pre></div>
<pre><code>## [1] &quot;IL1B + IL6 + IL8 + IL10 + TNFa + VEGF&quot;</code></pre>
<div class="sourceCode" id="cb347"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb347-1"><a href="data-import-processing-and-summary-statistics.html#cb347-1" tabindex="-1"></a><span class="co"># Make the table</span></span>
<span id="cb347-2"><a href="data-import-processing-and-summary-statistics.html#cb347-2" tabindex="-1"></a><span class="fu">table1</span>(<span class="sc">~</span> IL1B <span class="sc">+</span> IL6 <span class="sc">+</span> IL8 <span class="sc">+</span> IL10 <span class="sc">+</span> TNFa <span class="sc">+</span> VEGF <span class="sc">|</span> Dose, <span class="at">data =</span> biomarker_data_imp_avg)</span></code></pre></div>
<p><img src="Module4_2_Input/Module4_2_Image2.png" width="850px" style="display: block; margin: auto;" /></p>
</div>
<div id="vtable" class="section level4 hasAnchor">
<h4>Vtable<a href="data-import-processing-and-summary-statistics.html#vtable" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The <em>vtable</em> package includes the function <code>st()</code>, which can also be used to make HTML tables (and other output formats; see <code>out</code> argument). For example:</p>
<div class="sourceCode" id="cb348"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb348-1"><a href="data-import-processing-and-summary-statistics.html#cb348-1" tabindex="-1"></a><span class="co"># HTML output</span></span>
<span id="cb348-2"><a href="data-import-processing-and-summary-statistics.html#cb348-2" tabindex="-1"></a><span class="fu">st</span>(biomarker_data_imp_avg, <span class="at">group =</span> <span class="st">&#39;Dose&#39;</span>)</span></code></pre></div>
<table class="table" style="margin-left: auto; margin-right: auto;">
<caption>
<span id="tab:unnamed-chunk-277">Table 1: </span>Summary Statistics
</caption>
<thead>
<tr>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="1">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
Dose
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="3">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
0
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="3">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
0.6
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="3">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
1
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="3">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
2
</div>
</th>
<th style="border-bottom:hidden;padding-bottom:0; padding-left:3px;padding-right:3px;text-align: center; " colspan="3">
<div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
4
</div>
</th>
</tr>
<tr>
<th style="text-align:left;">
Variable
</th>
<th style="text-align:left;">
N
</th>
<th style="text-align:left;">
Mean
</th>
<th style="text-align:left;">
SD
</th>
<th style="text-align:left;">
N
</th>
<th style="text-align:left;">
Mean
</th>
<th style="text-align:left;">
SD
</th>
<th style="text-align:left;">
N
</th>
<th style="text-align:left;">
Mean
</th>
<th style="text-align:left;">
SD
</th>
<th style="text-align:left;">
N
</th>
<th style="text-align:left;">
Mean
</th>
<th style="text-align:left;">
SD
</th>
<th style="text-align:left;">
N
</th>
<th style="text-align:left;">
Mean
</th>
<th style="text-align:left;">
SD
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
IL1B
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
9.8
</td>
<td style="text-align:left;">
1.9
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
10
</td>
<td style="text-align:left;">
1.9
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
1.5
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
11
</td>
<td style="text-align:left;">
1.6
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
12
</td>
<td style="text-align:left;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
IL6
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
597
</td>
<td style="text-align:left;">
658
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
670
</td>
<td style="text-align:left;">
810
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
565
</td>
<td style="text-align:left;">
652
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
536
</td>
<td style="text-align:left;">
474
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
288
</td>
<td style="text-align:left;">
297
</td>
</tr>
<tr>
<td style="text-align:left;">
IL8
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
21987
</td>
<td style="text-align:left;">
7570
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
24703
</td>
<td style="text-align:left;">
10768
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
22607
</td>
<td style="text-align:left;">
10648
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
41687
</td>
<td style="text-align:left;">
17939
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
94439
</td>
<td style="text-align:left;">
33535
</td>
</tr>
<tr>
<td style="text-align:left;">
IL10
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
1.4
</td>
<td style="text-align:left;">
1.3
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
0.44
</td>
<td style="text-align:left;">
0.56
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
1.2
</td>
<td style="text-align:left;">
1.2
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
0.88
</td>
<td style="text-align:left;">
0.43
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
1.2
</td>
<td style="text-align:left;">
0.78
</td>
</tr>
<tr>
<td style="text-align:left;">
TNFa
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
3.2
</td>
<td style="text-align:left;">
1.8
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
3.1
</td>
<td style="text-align:left;">
1.7
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
4.1
</td>
<td style="text-align:left;">
2.4
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
5.2
</td>
<td style="text-align:left;">
2
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
6.7
</td>
<td style="text-align:left;">
2.7
</td>
</tr>
<tr>
<td style="text-align:left;">
VEGF
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
971
</td>
<td style="text-align:left;">
417
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
1023
</td>
<td style="text-align:left;">
486
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
1196
</td>
<td style="text-align:left;">
621
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
1754
</td>
<td style="text-align:left;">
749
</td>
<td style="text-align:left;">
16
</td>
<td style="text-align:left;">
1617
</td>
<td style="text-align:left;">
836
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb349"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb349-1"><a href="data-import-processing-and-summary-statistics.html#cb349-1" tabindex="-1"></a><span class="co"># Dataframe output</span></span>
<span id="cb349-2"><a href="data-import-processing-and-summary-statistics.html#cb349-2" tabindex="-1"></a><span class="fu">st</span>(biomarker_data_imp_avg, <span class="at">group =</span> <span class="st">&#39;Dose&#39;</span>, <span class="at">out =</span> <span class="st">&#39;return&#39;</span>)</span></code></pre></div>
<pre><code>##   Variable  N  Mean   SD   N  Mean    SD  N  Mean    SD  N  Mean    SD  N  Mean
## 1     Dose  0            0.6              1              2              4      
## 2     IL1B 16   9.8  1.9  16    10   1.9 16    11   1.5 16    11   1.6 16    12
## 3      IL6 16   597  658  16   670   810 16   565   652 16   536   474 16   288
## 4      IL8 16 21987 7570  16 24703 10768 16 22607 10648 16 41687 17939 16 94439
## 5     IL10 16   1.4  1.3  16  0.44  0.56 16   1.2   1.2 16  0.88  0.43 16   1.2
## 6     TNFa 16   3.2  1.8  16   3.1   1.7 16   4.1   2.4 16   5.2     2 16   6.7
## 7     VEGF 16   971  417  16  1023   486 16  1196   621 16  1754   749 16  1617
##      SD
## 1      
## 2     2
## 3   297
## 4 33535
## 5  0.78
## 6   2.7
## 7   836</code></pre>
<p>Similar to <em>table1</em>, see the package vignette for detailed information about how to customize tables using this package.</p>
<p><br></p>
</div>
</div>
</div>
<div id="normality-assessment-and-data-transformation" class="section level2 hasAnchor">
<h2>Normality Assessment and Data Transformation<a href="data-import-processing-and-summary-statistics.html#normality-assessment-and-data-transformation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The last step we will take before beginning to test our data for statistical differences between groups (in the next module) is to understand our data’s distribution through normality assessment. This will inform which statistical tests we will perform on our data. For more detail on normality testing, including detailed explanations of each type of normality assessment and explanations of the code underlying the following graphs and tables, see <strong>TAME 2.0 Module 3.3 Normality Tests and Data Transformations</strong>.</p>
<p>We’ll start by looking at histograms of our data for qualitative normality assessment:</p>
<div class="sourceCode" id="cb351"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb351-1"><a href="data-import-processing-and-summary-statistics.html#cb351-1" tabindex="-1"></a><span class="co"># Set theme</span></span>
<span id="cb351-2"><a href="data-import-processing-and-summary-statistics.html#cb351-2" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>())</span>
<span id="cb351-3"><a href="data-import-processing-and-summary-statistics.html#cb351-3" tabindex="-1"></a></span>
<span id="cb351-4"><a href="data-import-processing-and-summary-statistics.html#cb351-4" tabindex="-1"></a><span class="co"># Pivot data longer to prepare for plotting</span></span>
<span id="cb351-5"><a href="data-import-processing-and-summary-statistics.html#cb351-5" tabindex="-1"></a>biomarker_data_imp_avg_long <span class="ot">&lt;-</span> biomarker_data_imp_avg <span class="sc">%&gt;%</span></span>
<span id="cb351-6"><a href="data-import-processing-and-summary-statistics.html#cb351-6" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(Donor, Dose), <span class="at">names_to =</span> <span class="st">&quot;variable&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>)</span>
<span id="cb351-7"><a href="data-import-processing-and-summary-statistics.html#cb351-7" tabindex="-1"></a></span>
<span id="cb351-8"><a href="data-import-processing-and-summary-statistics.html#cb351-8" tabindex="-1"></a><span class="co"># Make figure panel of histograms</span></span>
<span id="cb351-9"><a href="data-import-processing-and-summary-statistics.html#cb351-9" tabindex="-1"></a><span class="fu">ggplot</span>(biomarker_data_imp_avg_long, <span class="fu">aes</span>(value)) <span class="sc">+</span></span>
<span id="cb351-10"><a href="data-import-processing-and-summary-statistics.html#cb351-10" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">fill =</span> <span class="st">&quot;gray40&quot;</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">binwidth =</span> <span class="cf">function</span>(x) {(<span class="fu">max</span>(x) <span class="sc">-</span> <span class="fu">min</span>(x))<span class="sc">/</span><span class="dv">25</span>}) <span class="sc">+</span></span>
<span id="cb351-11"><a href="data-import-processing-and-summary-statistics.html#cb351-11" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> variable, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>, <span class="at">nrow =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb351-12"><a href="data-import-processing-and-summary-statistics.html#cb351-12" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;# of Observations&quot;</span>, <span class="at">x =</span> <span class="st">&quot;Value&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-278-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>From these histograms, we can see that IL-1<span class="math inline">\(\beta\)</span> appears to be normally distributed, while the other endpoints do not appear to be normally distributed.</p>
<p>We can also use Q-Q plots to assess normality qualitatively:</p>
<div class="sourceCode" id="cb352"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb352-1"><a href="data-import-processing-and-summary-statistics.html#cb352-1" tabindex="-1"></a><span class="fu">ggqqplot</span>(biomarker_data_imp_avg_long, <span class="at">x =</span> <span class="st">&quot;value&quot;</span>, <span class="at">facet.by =</span> <span class="st">&quot;variable&quot;</span>, <span class="at">ggtheme =</span> <span class="fu">theme_bw</span>(), <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-279-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>With this figure panel, we can see that most of the variables have very noticeable deviations from the reference, suggesting non-normal distributions.</p>
<p>To assess normality quantitatively, we can use the Shapiro-Wilk test. Note that the null hypothesis is that the sample distribution is normal, and a significant p-value means the distribution is non-normal.</p>
<div class="sourceCode" id="cb353"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb353-1"><a href="data-import-processing-and-summary-statistics.html#cb353-1" tabindex="-1"></a><span class="co"># Apply Shapiro Wilk test to dataframe</span></span>
<span id="cb353-2"><a href="data-import-processing-and-summary-statistics.html#cb353-2" tabindex="-1"></a>shapiro_res <span class="ot">&lt;-</span>  <span class="fu">apply</span>(biomarker_data_imp_avg <span class="sc">%&gt;%</span> <span class="fu">select</span>(IL1B<span class="sc">:</span>VEGF), <span class="dv">2</span>, shapiro.test)</span>
<span id="cb353-3"><a href="data-import-processing-and-summary-statistics.html#cb353-3" tabindex="-1"></a></span>
<span id="cb353-4"><a href="data-import-processing-and-summary-statistics.html#cb353-4" tabindex="-1"></a><span class="co"># Create results dataframe</span></span>
<span id="cb353-5"><a href="data-import-processing-and-summary-statistics.html#cb353-5" tabindex="-1"></a>shapiro_res <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind.data.frame, shapiro_res)</span>
<span id="cb353-6"><a href="data-import-processing-and-summary-statistics.html#cb353-6" tabindex="-1"></a></span>
<span id="cb353-7"><a href="data-import-processing-and-summary-statistics.html#cb353-7" tabindex="-1"></a><span class="co"># Clean dataframe</span></span>
<span id="cb353-8"><a href="data-import-processing-and-summary-statistics.html#cb353-8" tabindex="-1"></a>shapiro_res <span class="ot">&lt;-</span> shapiro_res <span class="sc">%&gt;%</span> </span>
<span id="cb353-9"><a href="data-import-processing-and-summary-statistics.html#cb353-9" tabindex="-1"></a>  </span>
<span id="cb353-10"><a href="data-import-processing-and-summary-statistics.html#cb353-10" tabindex="-1"></a>  <span class="do">## Add normality conclusion</span></span>
<span id="cb353-11"><a href="data-import-processing-and-summary-statistics.html#cb353-11" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">normal =</span> <span class="fu">ifelse</span>(p.value <span class="sc">&lt;</span> <span class="fl">0.05</span>, F, T)) <span class="sc">%&gt;%</span></span>
<span id="cb353-12"><a href="data-import-processing-and-summary-statistics.html#cb353-12" tabindex="-1"></a>  </span>
<span id="cb353-13"><a href="data-import-processing-and-summary-statistics.html#cb353-13" tabindex="-1"></a>  <span class="do">## Remove columns that do not contain informative data</span></span>
<span id="cb353-14"><a href="data-import-processing-and-summary-statistics.html#cb353-14" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(p.value, normal)) </span>
<span id="cb353-15"><a href="data-import-processing-and-summary-statistics.html#cb353-15" tabindex="-1"></a></span>
<span id="cb353-16"><a href="data-import-processing-and-summary-statistics.html#cb353-16" tabindex="-1"></a><span class="co"># View cleaned up dataframe</span></span>
<span id="cb353-17"><a href="data-import-processing-and-summary-statistics.html#cb353-17" tabindex="-1"></a><span class="fu">datatable</span>(shapiro_res)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-ee572ee9031bc4e35ba4" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ee572ee9031bc4e35ba4">{"x":{"filter":"none","vertical":false,"data":[["IL1B","IL6","IL8","IL10","TNFa","VEGF"],[0.7908975299411561,1.061132292837358e-11,1.578095224843693e-09,1.646966900182463e-09,4.301352465611483e-05,0.0009741571784854114],[true,false,false,false,false,false]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>p.value<\/th>\n      <th>normal<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":1},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"p.value","targets":1},{"name":"normal","targets":2}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="answer-to-environmental-health-question-2-6" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 2<a href="data-import-processing-and-summary-statistics.html#answer-to-environmental-health-question-2-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><i>With this, we can answer <strong>Environmental Health Question 2</strong>:</i> Are our data normally distributed?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> The results from the Shapiro-Wilk test demonstrate that the IL-1<span class="math inline">\(\beta\)</span> data are normally distributed, while the other variables are non-normally distributed. These results support the conclusions we made based on our qualitative assessment above with histograms and Q-Q plots.</p>
</div>
</div>
<div id="log2-transforming-and-re-assessing-normality" class="section level3 hasAnchor">
<h3>Log<sub>2</sub> Transforming and Re-Assessing Normality<a href="data-import-processing-and-summary-statistics.html#log2-transforming-and-re-assessing-normality" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Log<sub>2</sub> transformation is a common transformation used in environmental health research and can move data closer to a normal distribution. For more on data transformation, see <strong>TAME 2.0 Module 3.3 Normality Tests and Data Transformations</strong>. We will pseudo-log<sub>2</sub> transform our data, which adds a 1 to each value before log<sub>2</sub> transformation and ensures that resulting values are positive real numbers. Let’s see if the log<sub>2</sub> data are more normally distributed than the raw data.</p>
<div class="sourceCode" id="cb354"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb354-1"><a href="data-import-processing-and-summary-statistics.html#cb354-1" tabindex="-1"></a><span class="co"># Apply log2 transformation to data</span></span>
<span id="cb354-2"><a href="data-import-processing-and-summary-statistics.html#cb354-2" tabindex="-1"></a>biomarker_data_imp_avg_log2 <span class="ot">&lt;-</span> biomarker_data_imp_avg <span class="sc">%&gt;%</span></span>
<span id="cb354-3"><a href="data-import-processing-and-summary-statistics.html#cb354-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(IL1B<span class="sc">:</span>VEGF, <span class="sc">~</span> <span class="fu">log2</span>(.x <span class="sc">+</span> <span class="dv">1</span>)))</span></code></pre></div>
<p>Make histogram panel:</p>
<div class="sourceCode" id="cb355"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb355-1"><a href="data-import-processing-and-summary-statistics.html#cb355-1" tabindex="-1"></a><span class="co"># Pivot data longer and make figure panel of histograms</span></span>
<span id="cb355-2"><a href="data-import-processing-and-summary-statistics.html#cb355-2" tabindex="-1"></a>biomarker_data_imp_avg_log2_long <span class="ot">&lt;-</span> biomarker_data_imp_avg_log2 <span class="sc">%&gt;%</span></span>
<span id="cb355-3"><a href="data-import-processing-and-summary-statistics.html#cb355-3" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(Donor, Dose), <span class="at">names_to =</span> <span class="st">&quot;variable&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>)</span>
<span id="cb355-4"><a href="data-import-processing-and-summary-statistics.html#cb355-4" tabindex="-1"></a></span>
<span id="cb355-5"><a href="data-import-processing-and-summary-statistics.html#cb355-5" tabindex="-1"></a><span class="co"># Make histogram panel</span></span>
<span id="cb355-6"><a href="data-import-processing-and-summary-statistics.html#cb355-6" tabindex="-1"></a><span class="fu">ggplot</span>(biomarker_data_imp_avg_log2_long, <span class="fu">aes</span>(value)) <span class="sc">+</span></span>
<span id="cb355-7"><a href="data-import-processing-and-summary-statistics.html#cb355-7" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">fill =</span> <span class="st">&quot;gray40&quot;</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">binwidth =</span> <span class="cf">function</span>(x) {(<span class="fu">max</span>(x) <span class="sc">-</span> <span class="fu">min</span>(x))<span class="sc">/</span><span class="dv">25</span>}) <span class="sc">+</span></span>
<span id="cb355-8"><a href="data-import-processing-and-summary-statistics.html#cb355-8" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> variable, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>) <span class="sc">+</span></span>
<span id="cb355-9"><a href="data-import-processing-and-summary-statistics.html#cb355-9" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;# of Observations&quot;</span>, <span class="at">x =</span> <span class="st">&quot;Value&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-282-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Make Q-Q plot panel:</p>
<div class="sourceCode" id="cb356"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb356-1"><a href="data-import-processing-and-summary-statistics.html#cb356-1" tabindex="-1"></a><span class="fu">ggqqplot</span>(biomarker_data_imp_avg_log2_long, <span class="at">x =</span> <span class="st">&quot;value&quot;</span>, <span class="at">facet.by =</span> <span class="st">&quot;variable&quot;</span>, <span class="at">ggtheme =</span> <span class="fu">theme_bw</span>(), <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-283-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Run Shapiro-Wilk test:</p>
<div class="sourceCode" id="cb357"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb357-1"><a href="data-import-processing-and-summary-statistics.html#cb357-1" tabindex="-1"></a><span class="co"># Apply Shapiro Wilk test</span></span>
<span id="cb357-2"><a href="data-import-processing-and-summary-statistics.html#cb357-2" tabindex="-1"></a>shapiro_res_log2 <span class="ot">&lt;-</span>  <span class="fu">apply</span>(biomarker_data_imp_avg_log2 <span class="sc">%&gt;%</span> <span class="fu">select</span>(IL1B<span class="sc">:</span>VEGF), <span class="dv">2</span>, shapiro.test)</span>
<span id="cb357-3"><a href="data-import-processing-and-summary-statistics.html#cb357-3" tabindex="-1"></a></span>
<span id="cb357-4"><a href="data-import-processing-and-summary-statistics.html#cb357-4" tabindex="-1"></a><span class="co"># Create results dataframe</span></span>
<span id="cb357-5"><a href="data-import-processing-and-summary-statistics.html#cb357-5" tabindex="-1"></a>shapiro_res_log2 <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind.data.frame, shapiro_res_log2)</span>
<span id="cb357-6"><a href="data-import-processing-and-summary-statistics.html#cb357-6" tabindex="-1"></a></span>
<span id="cb357-7"><a href="data-import-processing-and-summary-statistics.html#cb357-7" tabindex="-1"></a><span class="co"># Clean dataframe</span></span>
<span id="cb357-8"><a href="data-import-processing-and-summary-statistics.html#cb357-8" tabindex="-1"></a>shapiro_res_log2 <span class="ot">&lt;-</span> shapiro_res_log2 <span class="sc">%&gt;%</span> </span>
<span id="cb357-9"><a href="data-import-processing-and-summary-statistics.html#cb357-9" tabindex="-1"></a>  </span>
<span id="cb357-10"><a href="data-import-processing-and-summary-statistics.html#cb357-10" tabindex="-1"></a>  <span class="do">## Add normality conclusion</span></span>
<span id="cb357-11"><a href="data-import-processing-and-summary-statistics.html#cb357-11" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">normal =</span> <span class="fu">ifelse</span>(p.value <span class="sc">&lt;</span> <span class="fl">0.05</span>, F, T)) <span class="sc">%&gt;%</span></span>
<span id="cb357-12"><a href="data-import-processing-and-summary-statistics.html#cb357-12" tabindex="-1"></a>  </span>
<span id="cb357-13"><a href="data-import-processing-and-summary-statistics.html#cb357-13" tabindex="-1"></a>  <span class="do">## Remove columns that do not contain informative data</span></span>
<span id="cb357-14"><a href="data-import-processing-and-summary-statistics.html#cb357-14" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(p.value, normal)) </span>
<span id="cb357-15"><a href="data-import-processing-and-summary-statistics.html#cb357-15" tabindex="-1"></a></span>
<span id="cb357-16"><a href="data-import-processing-and-summary-statistics.html#cb357-16" tabindex="-1"></a><span class="co"># View cleaned up dataframe</span></span>
<span id="cb357-17"><a href="data-import-processing-and-summary-statistics.html#cb357-17" tabindex="-1"></a>shapiro_res_log2</span></code></pre></div>
<pre><code>##          p.value normal
## IL1B 0.250821677   TRUE
## IL6  0.001445717  FALSE
## IL8  0.017386740  FALSE
## IL10 0.018935719  FALSE
## TNFa 0.194774045   TRUE
## VEGF 0.047231367  FALSE</code></pre>
<p>The histograms and Q-Q plots demonstrate that the log<sub>2</sub> data are more normally distributed than the raw data. The results from the Shapiro-Wilk test also demonstrate that the the log<sub>2</sub> data are more normally distributed as a whole than the raw data. Overall, the p-values, even for the variables that are still non-normally distributed, are much higher.</p>
<p>So, should we proceed with the raw data or the log<sub>2</sub> data? This depends on what analyses we plan to do. In general, it is best to keep the data in as close to its raw format as possible, so if all of our analyses are available with a non-parametric test, we could use our raw data. However, some statistical tests do not have a non-parametric equivalent, in which case it would likely be best to use the log<sub>2</sub> transformed data. For subsequent modules, we will proceed with the log<sub>2</sub> data for consistency; however, choices regarding normality assessment can vary, so be sure to discuss these choices within your research group before proceeding with your analysis.</p>
<p>For more on decisions regarding normality, see <strong>TAME 2.0 Module 3.3 Normality Tests and Data Transformations</strong>. For more on parametric vs. non-parametric tests, see <strong>TAME 2.0 Module 4.4 Two Group Comparisons and Visualizations</strong> and <strong>TAME 2.0 Module 4.5 Multi-Group Comparisons and Visualizations</strong>.</p>
<p><br></p>
</div>
</div>
<div id="concluding-remarks-13" class="section level2 hasAnchor">
<h2>Concluding Remarks<a href="data-import-processing-and-summary-statistics.html#concluding-remarks-13" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Taken together, this module demonstrates important data processing steps necessary before proceeding with between-group statistical testing, including data import, handling missing values, averaging replicates, generating descriptive statistics tables, and assessing normality. Careful consideration and description of these steps in the methods section of a manuscript or report increases reproducibility of analyses and helps to improve the accuracy and statistical validity of subsequent statistical results.</p>
<p><br></p>
<p><label class="tykfont">
Test Your Knowledge
</label></p>
<div class="tyk">
<p>Functional endpoints from these cultures were also measured. These endpoints were: 1) Membrane Permeability (MemPerm), 2) Trans-Epithelial Electrical Resistance (TEER), 3) Ciliary Beat Frequency (CBF), and 4) Expression of Mucin (MUC5AC). Work through the same processes demonstrated in this module using the provided data (“Module4_2_TYKInput.xlsx”) to answer the following questions:</p>
<ol style="list-style-type: decimal">
<li>How many technical replicates are there for each dose?</li>
<li>Are there any missing values?</li>
<li>What are the average values for each endpoint by dose?</li>
<li>Are the raw data normally distributed?</li>
</ol>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="overview-of-experimental-design-and-example-data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="data-import-from-pdf-sources.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
