<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Data Manipulation and Reshaping | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="Data Manipulation and Reshaping | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Data Manipulation and Reshaping | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  <meta name="twitter:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  

<meta name="author" content="Rager Lab" />


<meta name="date" content="2024-10-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="Icons_Used_Throughout/Favicon.png" type="image/x-icon" />
<link rel="prev" href="coding-best-practices.html"/>
<link rel="next" href="improving-coding-efficiencies.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>
<link href="libs/table1-1.0/table1_defaults.css" rel="stylesheet" />
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-FGFKVX2X3G"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-FGFKVX2X3G');
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="Icons_Used_Throughout/TAME 2.0 Logo.png"><br></a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Chapter 1 Introductory <br>Data Science</b></span></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html"><i class="fa fa-check"></i>FAIR Data Management Practices</a>
<ul>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-training-module"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-fair"><i class="fa fa-check"></i>Introduction to FAIR</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#breaking-down-fair-letter-by-letter"><i class="fa fa-check"></i>Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#what-does-this-mean-for-you"><i class="fa fa-check"></i>What Does This Mean for You?</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#data-repositories-for-sharing-of-data"><i class="fa fa-check"></i>Data Repositories for Sharing of Data</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#recent-shifts-in-regulatory-policies-for-data-sharing"><i class="fa fa-check"></i>Recent Shifts in Regulatory Policies for Data Sharing</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#additional-training-resources-on-fair"><i class="fa fa-check"></i>Additional Training Resources on FAIR</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html"><i class="fa fa-check"></i>Data Sharing through Online Repositories</a>
<ul>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#an-overview-and-example-with-the-dataverse-repository"><i class="fa fa-check"></i>An Overview and Example with the Dataverse Repository</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#introduction-to-training-module-1"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#data-repositories"><i class="fa fa-check"></i>Data Repositories</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#the-dataverse-project"><i class="fa fa-check"></i>The Dataverse Project</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#what-is-a-dataverse"><i class="fa fa-check"></i>What is a Dataverse?</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#metadata"><i class="fa fa-check"></i>Metadata</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataverse"><i class="fa fa-check"></i>Creating a Dataverse</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataset"><i class="fa fa-check"></i>Creating a Dataset</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#concluding-remarks-1"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html"><i class="fa fa-check"></i>File Management using Github</a>
<ul>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#introduction-to-training-module-2"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-an-account"><i class="fa fa-check"></i>Creating an Account</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-a-repository"><i class="fa fa-check"></i>Creating a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#uploading-code"><i class="fa fa-check"></i>Uploading Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#adding-subfolders-in-a-repository"><i class="fa fa-check"></i>Adding Subfolders in a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-code"><i class="fa fa-check"></i>Updating Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-repository-titles-and-structure-to-support-a-manuscript"><i class="fa fa-check"></i>Updating Repository Titles and Structure to Support a Manuscript</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#tracking-code-changes-using-github-branches"><i class="fa fa-check"></i>Tracking Code Changes using Github Branches</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#concluding-remarks-2"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html"><i class="fa fa-check"></i>Data Wrangling in Excel</a>
<ul>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#introduction-to-training-module-3"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#save-a-copy-of-the-soon-to-be-organized-and-cleaned-dataset-as-a-new-file"><i class="fa fa-check"></i>Save a Copy of the Soon-To-Be Organized and Cleaned Dataset as a New File</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-extraneous-white-space"><i class="fa fa-check"></i>Remove Extraneous White Space</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#replace-missing-data-with-na"><i class="fa fa-check"></i>Replace Missing Data with “NA”</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-metadata-tab"><i class="fa fa-check"></i>Create a Metadata Tab</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#abbreviate-and-capitalize-categorical-data"><i class="fa fa-check"></i>Abbreviate and Capitalize Categorical Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#alphabetize-sort-the-data-by-the-categorical-variable-of-interest"><i class="fa fa-check"></i>Alphabetize (Sort) the Data by the Categorical Variable of Interest</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-new-subject-number-column"><i class="fa fa-check"></i>Create a New Subject Number Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-special-symbols-and-dashes"><i class="fa fa-check"></i>Remove Special Symbols and Dashes</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#bold-all-column-names-and-center-all-data"><i class="fa fa-check"></i>Bold all Column Names and Center all Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-subject-identifier-column"><i class="fa fa-check"></i>Create a Subject Identifier Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#separate-subject-demographic-data-from-experimental-measurements"><i class="fa fa-check"></i>Separate Subject Demographic Data from Experimental Measurements</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#convert-data-from-wide-to-long-format"><i class="fa fa-check"></i>Convert Data from Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#pivoting-data-from-a-wide-to-long-format"><i class="fa fa-check"></i>Pivoting Data from a Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#generating-summary-level-statistics-with-pivot-tables"><i class="fa fa-check"></i>Generating Summary-Level Statistics with Pivot Tables</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#excel-vs.-r-which-should-you-use"><i class="fa fa-check"></i>Excel vs. R: Which Should You Use?</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#concluding-remarks-3"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 2 Coding in R</b></span></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html"><i class="fa fa-check"></i>Downloading and Programming in R</a>
<ul>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-training-module-4"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#general-introduction-and-installation-of-r-and-rstudio"><i class="fa fa-check"></i>General Introduction and Installation of R and RStudio</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-r-packages"><i class="fa fa-check"></i>Introduction to R Packages</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#scripting-basics"><i class="fa fa-check"></i>Scripting Basics</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#code-troubleshooting"><i class="fa fa-check"></i>Code Troubleshooting</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#concluding-remarks-4"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html"><i class="fa fa-check"></i>Coding “Best” Practices</a>
<ul>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#introduction-to-training-module-5"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#scripting-file-types"><i class="fa fa-check"></i>Scripting File Types</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-headers-and-annotation"><i class="fa fa-check"></i>Script Headers and Annotation</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#coding-style"><i class="fa fa-check"></i>Coding Style</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-organization"><i class="fa fa-check"></i>Script Organization</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#concluding-remarks-5"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html"><i class="fa fa-check"></i>Data Manipulation and Reshaping</a>
<ul>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-training-module-6"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#data-manipulation-using-base-r"><i class="fa fa-check"></i>Data Manipulation Using Base R</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-tidyverse"><i class="fa fa-check"></i>Introduction to Tidyverse</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#concluding-remarks-6"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html"><i class="fa fa-check"></i>Improving Coding Efficiencies</a>
<ul>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#introduction-to-training-module-7"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#loops"><i class="fa fa-check"></i>Loops</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#functions"><i class="fa fa-check"></i>Functions</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#list-operations"><i class="fa fa-check"></i>List operations</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#concluding-remarks-7"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#additional-resources-2"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="part"><span><b>Chapter 3 Basics of <br>Data Analysis and Visualizations</b></span></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html"><i class="fa fa-check"></i>Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-data-visualizations"><i class="fa fa-check"></i>Introduction to Data Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-training-module-8"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#density-plot-visualization"><i class="fa fa-check"></i>Density Plot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#boxplot-visualization"><i class="fa fa-check"></i>Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#correlation-visualizations"><i class="fa fa-check"></i>Correlation Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#heatmap-visualization"><i class="fa fa-check"></i>Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#concluding-remarks-8"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html"><i class="fa fa-check"></i>Improving Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-data-visulization-conventions"><i class="fa fa-check"></i>Introduction to Data Visulization Conventions</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-training-module-9"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-boxplot-visualization"><i class="fa fa-check"></i>Creating an Improved Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-heatmap-visualization"><i class="fa fa-check"></i>Creating an Improved Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-multi-plot-figures"><i class="fa fa-check"></i>Creating Multi-Plot Figures</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#concluding-remarks-9"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html"><i class="fa fa-check"></i>Normality Tests and Data Transformations</a>
<ul>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#introduction-to-training-module-10"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#what-is-a-normal-distribution"><i class="fa fa-check"></i>What is a Normal Distribution?</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#qualitative-assessment-of-normality"><i class="fa fa-check"></i>Qualitative Assessment of Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#quantitative-normality-assessment"><i class="fa fa-check"></i>Quantitative Normality Assessment</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#data-transformation"><i class="fa fa-check"></i>Data Transformation</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#additional-considerations-regarding-normality"><i class="fa fa-check"></i>Additional Considerations Regarding Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#concluding-remarks-10"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html"><i class="fa fa-check"></i>Intoduction to Statistical Tests</a>
<ul>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#introduction-to-training-module-11"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#assessing-normality-homogeneity-of-variance"><i class="fa fa-check"></i>Assessing Normality &amp; Homogeneity of Variance</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#two-group-visualizations-and-statistical-comparisons-using-the-t-test"><i class="fa fa-check"></i>Two-Group Visualizations and Statistical Comparisons using the T-Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#three-group-visualizations-and-statistical-comparisons-using-an-anova"><i class="fa fa-check"></i>Three-Group Visualizations and Statistical Comparisons using an ANOVA</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#regression-modeling-and-visualization-linear-and-logistic-regressions"><i class="fa fa-check"></i>Regression Modeling and Visualization: Linear and Logistic Regressions</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#statistical-evaluations-of-categorical-data-using-the-chi-squared-test-and-fishers-exact-test"><i class="fa fa-check"></i>Statistical Evaluations of Categorical Data using the Chi-Squared Test and Fisher’s Exact Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#concluding-remarks-11"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 4 Converting Wet Lab Data into Dry Lab Analyses</b></span></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html"><i class="fa fa-check"></i>Overview of Experimental Design and Example Data</a>
<ul>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#introduction-to-training-module-12"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#replicates"><i class="fa fa-check"></i>Replicates</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#orientation-to-example-data-for-chapter-4"><i class="fa fa-check"></i>Orientation to Example Data for Chapter 4</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#concluding-remarks-12"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html"><i class="fa fa-check"></i>Data Import, Processing, and Summary Statistics</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#introduction-to-training-module-13"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#data-import"><i class="fa fa-check"></i>Data Import</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#handling-missing-values"><i class="fa fa-check"></i>Handling Missing Values</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#averaging-replicates"><i class="fa fa-check"></i>Averaging Replicates</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#descriptive-statistics"><i class="fa fa-check"></i>Descriptive Statistics</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#normality-assessment-and-data-transformation"><i class="fa fa-check"></i>Normality Assessment and Data Transformation</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#concluding-remarks-13"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html"><i class="fa fa-check"></i>Data Import from PDF Sources</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#introduction-to-training-module-14"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-from-many-single-pdfs-with-the-same-formatting"><i class="fa fa-check"></i>Importing Data from Many Single PDFs with the Same Formatting</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-stored-in-pdf-tables"><i class="fa fa-check"></i>Importing Data Stored in PDF Tables</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#concluding-remarks-14"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html"><i class="fa fa-check"></i>Two-Group Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#introduction-to-training-module-15"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#overview-of-two-group-statistical-tests"><i class="fa fa-check"></i>Overview of Two Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#statistical-vs-biological-significance"><i class="fa fa-check"></i>Statistical vs Biological Significance</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#unpaired-test-example"><i class="fa fa-check"></i>Unpaired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#paired-test-example"><i class="fa fa-check"></i>Paired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#visualizing-results"><i class="fa fa-check"></i>Visualizing Results</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#concluding-remarks-15"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html"><i class="fa fa-check"></i>Multi-Group and Multi-Variable Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#introduction-to-training-module-16"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#overview-of-multi-group-statistical-tests"><i class="fa fa-check"></i>Overview of Multi-Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#multi-group-analysis-example"><i class="fa fa-check"></i>Multi-Group Analysis Example</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#visualization-of-multi-group-statistical-results"><i class="fa fa-check"></i>Visualization of Multi-Group Statistical Results</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#concluding-remarks-16"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 5 Machine Learning &amp; Artificial Intelligence</b></span></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><i class="fa fa-check"></i>Introduction to Artificial Intelligence, Machine Learning, and Predictive Modeling for Environmental Health</a>
<ul>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#introduction-to-training-module-17"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#general-historical-context-and-taxonomy-of-modern-aiml"><i class="fa fa-check"></i>General Historical Context and Taxonomy of Modern AI/ML</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#application-of-machine-learning-in-environmental-health-science"><i class="fa fa-check"></i>Application of Machine Learning in Environmental Health Science</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#concluding-remarks-17"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html"><i class="fa fa-check"></i>Supervised Machine Learning</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-training-module-18"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-machine-learning"><i class="fa fa-check"></i>Types of Machine Learning</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-supervised-machine-learning-algorithms"><i class="fa fa-check"></i>Types of Supervised Machine Learning Algorithms</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#training-supervised-machine-learning-models"><i class="fa fa-check"></i>Training Supervised Machine Learning Models</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#assessing-classification-based-model-performance"><i class="fa fa-check"></i>Assessing Classification-Based Model Performance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-activity-and-example-dataset"><i class="fa fa-check"></i>Introduction to Activity and Example Dataset</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#testing-for-differences-in-predictor-variables-across-the-outcome-classes"><i class="fa fa-check"></i>Testing for Differences in Predictor Variables across the Outcome Classes</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#predicting-ias-detection-with-a-random-forest-rf-model"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#class-imbalance"><i class="fa fa-check"></i>Class Imbalance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#concluding-remarks-18"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html"><i class="fa fa-check"></i>Supervised Machine Learning Model Interpretation</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-training-module-19"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance"><i class="fa fa-check"></i>Variable Importance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary"><i class="fa fa-check"></i>Decision Boundary</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-example-dataset-and-activity"><i class="fa fa-check"></i>Introduction to Example Dataset and Activity</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#predicting-ias-detection-with-a-random-forest-rf-model-1"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance-plot"><i class="fa fa-check"></i>Variable Importance Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot"><i class="fa fa-check"></i>Decision Boundary Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot-incorporating-smote"><i class="fa fa-check"></i>Decision Boundary Plot Incorporating SMOTE</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#concluding-remarks-19"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><i class="fa fa-check"></i>Unsupervised Machine Learning Part 1: K-Means Clustering &amp; PCA</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-training-module-20"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-unsupervised-machine-learning"><i class="fa fa-check"></i>Introduction to Unsupervised Machine Learning</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-example-data"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#identifying-clusters-of-chemicals-through-k-means"><i class="fa fa-check"></i>Identifying Clusters of Chemicals through <em>K</em>-Means</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#principal-component-analysis-pca-1"><i class="fa fa-check"></i>Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#incorporating-k-means-into-pca-for-predictive-modeling"><i class="fa fa-check"></i>Incorporating <em>K</em>-Means into PCA for Predictive Modeling</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#concluding-remarks-20"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><i class="fa fa-check"></i>Unsupervised Machine Learning Part 2: Additional Clustering Applications</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-training-module-21"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#k-means-clustering-1"><i class="fa fa-check"></i><em>K</em>-Means Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-example-data-1"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-1"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#clustering-plot"><i class="fa fa-check"></i>Clustering Plot</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-visualization"><i class="fa fa-check"></i>Hierarchical Clustering Visualization</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#variable-contributions"><i class="fa fa-check"></i>Variable Contributions</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#concluding-remarks-21"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 6 Applications in Toxicology &amp; Exposure Science</b></span></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html"><i class="fa fa-check"></i>6.1 Descriptive Cohort Analyses</a>
<ul>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#introduction-to-training-module-22"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#participant-exploration"><i class="fa fa-check"></i>Participant Exploration</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#chemical-detection-filtering"><i class="fa fa-check"></i>Chemical Detection Filtering</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#outlier-identification"><i class="fa fa-check"></i>Outlier Identification</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#summary-statistics-tables"><i class="fa fa-check"></i>Summary Statistics Tables</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#demographics-table"><i class="fa fa-check"></i>Demographics Table</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#concluding-remarks-22"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html"><i class="fa fa-check"></i>6.2 -Omics and System Biology: Transcriptomic Applications</a>
<ul>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-training-module-23"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-the-field-of--omics"><i class="fa fa-check"></i>Introduction to the Field of “-Omics”</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-transcriptomics"><i class="fa fa-check"></i>Introduction to Transcriptomics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#formatting-data-for-downstream-statistics"><i class="fa fa-check"></i>Formatting Data for Downstream Statistics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#transcriptomics-data-qaqc"><i class="fa fa-check"></i>Transcriptomics Data QA/QC</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#controlling-for-sources-of-sample-heterogeneity"><i class="fa fa-check"></i>Controlling for Sources of Sample Heterogeneity</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e.g.-biomass-smoke-exposure"><i class="fa fa-check"></i>Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure)</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-ma-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using MA Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-volcano-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using Volcano Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis"><i class="fa fa-check"></i>Interpretting Findings at the Systems Level through Pathway Enrichment Analysis</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#concluding-remarks-23"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><i class="fa fa-check"></i>6.3 Mixtures Analysis Methods Part 1: Overview and Example with Quantile G-Computation</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-training-module-24"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#overview-of-mixtures-analysis"><i class="fa fa-check"></i>Overview of Mixtures Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-example-data-2"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#mixtures-model-with-standard-qqcomp"><i class="fa fa-check"></i>Mixtures Model with Standard qqcomp</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#concluding-remarks-24"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#additional-resources-10"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><i class="fa fa-check"></i>6.4 Mixtures Analysis Methods Part 2: Bayesian Kernel Machine Regression</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-training-module-25"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-example-data-3"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-bkmr"><i class="fa fa-check"></i>Introduction to BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#run-bkmr"><i class="fa fa-check"></i>Run BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#concluding-remarks-25"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#additional-resources-11"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html"><i class="fa fa-check"></i>6.5 Mixtures Analysis Methods Part 3: Sufficient Similarity</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-training-module-26"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-toxicant-and-dataset"><i class="fa fa-check"></i>Introduction to Toxicant and Dataset</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#workspace-preparation-and-data-import-8"><i class="fa fa-check"></i>Workspace Preparation and Data Import</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#chemistry-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Chemistry-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#toxicity-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Toxicity-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#comparing-chemistry-vs.-toxicity-sufficient-similarity-analyses"><i class="fa fa-check"></i>Comparing Chemistry vs. Toxicity Sufficient Similarity Analyses</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#concluding-remarks-26"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i>6.6 Toxicokinetic Modeling</a>
<ul>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-training-module-27"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-toxicokinetic-modeling"><i class="fa fa-check"></i>Introduction to Toxicokinetic Modeling</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#data-and-models-used-in-toxicokinetic-modeling-tk"><i class="fa fa-check"></i>Data and Models used in Toxicokinetic Modeling (TK)</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-toxicokinetics"><i class="fa fa-check"></i>Reverse Toxicokinetics</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#capturing-population-variability-in-toxicokinetics-and-uncertainty-in-chemical-specific-parameters"><i class="fa fa-check"></i>Capturing Population Variability in Toxicokinetics, and Uncertainty in Chemical-Specific Parameters</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-tk-calculating-administered-equivalent-doses-for-toxcast-bioactive-concentrations"><i class="fa fa-check"></i>Reverse TK: Calculating Administered Equivalent Doses for ToxCast Bioactive Concentrations</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#comparing-equivalent-doses-estimated-to-elicit-toxicity-hazard-to-external-exposure-estimates-exposure-for-chemical-prioritization-by-bioactivity-exposure-ratios-bers"><i class="fa fa-check"></i>Comparing Equivalent Doses Estimated to Elicit Toxicity (Hazard) to External Exposure Estimates (Exposure), for Chemical Prioritization by Bioactivity-Exposure Ratios (BERs)</a></li>
</ul></li>
<li class="part"><span><b>Chapter 7 Environmental Health Database Mining</b></span></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data-manipulation-and-reshaping" class="section level1 hasAnchor">
<h1>Data Manipulation and Reshaping<a href="data-manipulation-and-reshaping.html#data-manipulation-and-reshaping" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This training module was developed by Kyle Roell, Alexis Payton, Elise Hickman, and Julia E. Rager.</p>
<p>All input files (script, data, and figures) can be downloaded from the <a href="https://github.com/UNCSRP/TAME2">UNC-SRP TAME2 GitHub website</a>.</p>
<div id="introduction-to-training-module-6" class="section level2 hasAnchor">
<h2>Introduction to Training Module<a href="data-manipulation-and-reshaping.html#introduction-to-training-module-6" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Data within the fields of exposure science, toxicology, and public health are very rarely prepared and ready for downstream statistical analyses and visualization code. The beginning of almost any scripted analysis includes important formatting steps that make the data easier to read and work with. This can be done in several ways, including:</p>
<ul>
<li><a href="https://www.r-project.org/about.html">Base R operations and functions</a>, or</li>
<li>A collection of packages (and philosophy) known as <a href="https://www.tidyverse.org">The Tidyverse</a>.</li>
</ul>
<p>In this training tutorial we will review some of the most common ways you can organize and manipulate data, including:</p>
<ul>
<li>Merging data</li>
<li>Filtering and subsetting data</li>
<li>Pivoting data wider and longer (also known as casting and melting)</li>
</ul>
<p>These approaches will first be demonstrated using the functions available in base R. Then, the exact same approaches will be demonstrated using the functions and syntax that are part of the Tidyverse package.</p>
<p>We will demonstrate these data manipulation and organization methods using an environmentally relevant example data set from a human cohort. This dataset was generated by creating data distributions randomly pulled from our previously published cohorts, resulting in a unique data set for these training purposes. The dataset contains environmental exposure metrics from metal levels obtained using sources of drinking water and human urine samples and associated demographic data.</p>
<div id="training-modules-environmental-health-question" class="section level3 hasAnchor">
<h3>Training Module’s Environmental Health Question<a href="data-manipulation-and-reshaping.html#training-modules-environmental-health-question" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This training module was specifically developed to answer the following environmental health question using data manipulation and reshaping approaches:</p>
<p>What is the average urinary chromium concentration across different maternal education levels?</p>
<p>We’ll use base R and <em>Tidydverse</em> to answer this question, but let’s start with Base R.</p>
</div>
<div id="workspace-preparation-and-data-import" class="section level3 hasAnchor">
<h3>Workspace Preparation and Data Import<a href="data-manipulation-and-reshaping.html#workspace-preparation-and-data-import" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="set-your-working-directory" class="section level4 hasAnchor">
<h4>Set your working directory<a href="data-manipulation-and-reshaping.html#set-your-working-directory" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>In preparation, first let’s set our working directory to the folder path that contains our input files:</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="data-manipulation-and-reshaping.html#cb43-1" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/file path to where your input files are&quot;</span>)</span></code></pre></div>
<p>Note that macOS file paths use <code>/</code> as folder separators, and PC file paths use <code>\</code>.</p>
</div>
<div id="importing-example-datasets" class="section level4 hasAnchor">
<h4>Importing example datasets<a href="data-manipulation-and-reshaping.html#importing-example-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Next, let’s read in our example data sets:</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="data-manipulation-and-reshaping.html#cb44-1" tabindex="-1"></a>demographic_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Module2_3_Input/Module2_3_InputData1.csv&quot;</span>)</span>
<span id="cb44-2"><a href="data-manipulation-and-reshaping.html#cb44-2" tabindex="-1"></a>chemical_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Module2_3_Input/Module2_3_InputData2.csv&quot;</span>)</span></code></pre></div>
</div>
<div id="viewing-example-datasets" class="section level4 hasAnchor">
<h4>Viewing example datasets<a href="data-manipulation-and-reshaping.html#viewing-example-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Let’s see what these datasets look like:</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="data-manipulation-and-reshaping.html#cb45-1" tabindex="-1"></a><span class="fu">dim</span>(demographic_data)</span></code></pre></div>
<pre><code>## [1] 200   6</code></pre>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="data-manipulation-and-reshaping.html#cb47-1" tabindex="-1"></a><span class="fu">dim</span>(chemical_data)</span></code></pre></div>
<pre><code>## [1] 200   7</code></pre>
<p>The demographic data set includes 200 rows x 7 columns, while the chemical measurement data set includes 200 rows x 7 columns.</p>
<p>We can preview the demographic data frame by using the <code>head()</code> function, which displays all the columns and the first 6 rows of a data frame:</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="data-manipulation-and-reshaping.html#cb49-1" tabindex="-1"></a><span class="fu">head</span>(demographic_data)</span></code></pre></div>
<pre><code>##   ID  BMI     MAge MEdu       BW GA
## 1  1 27.7 22.99928    3 3180.058 34
## 2  2 26.8 30.05142    3 3210.823 43
## 3  3 33.2 28.04660    3 3311.551 40
## 4  4 30.1 34.81796    3 3266.844 32
## 5  5 37.4 42.68440    3 3664.088 35
## 6  6 33.3 24.94960    3 3328.988 40</code></pre>
<p>These demographic data are organized according to subject ID (first column) followed by the following subject information:</p>
<ul>
<li><code>ID</code>: subject number</li>
<li><code>BMI</code>: body mass index</li>
<li><code>MAge</code>: maternal age in years</li>
<li><code>MEdu</code>: maternal education level; 1 = “less than high school”, 2 = “high school or some college”, 3 = “college or greater”</li>
<li><code>BW</code>: body weight in grams</li>
<li><code>GA</code>: gestational age in weeks</li>
</ul>
<p>We can also preview the chemical dataframe:</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="data-manipulation-and-reshaping.html#cb51-1" tabindex="-1"></a><span class="fu">head</span>(chemical_data)</span></code></pre></div>
<pre><code>##   ID     DWAs     DWCd     DWCr       UAs       UCd      UCr
## 1  1 6.426464 1.292941 51.67987 10.192695 0.7537104 42.60187
## 2  2 7.832384 1.798535 50.10409 11.815088 0.9789506 41.30757
## 3  3 7.516569 1.288461 48.74001 10.079057 0.1903262 36.47716
## 4  4 5.906656 2.075259 50.92745  8.719123 0.9364825 42.47987
## 5  5 7.181873 2.762643 55.16882  9.436559 1.4977829 47.78528
## 6  6 9.723429 3.054057 51.14812 11.589403 1.6645837 38.26386</code></pre>
<p>These chemical data are organized according to subject ID (first column), followed by measures of:</p>
<ul>
<li><code>DWAs</code>: drinking water arsenic levels in µg/L</li>
<li><code>DWCd</code>: drinking water cadmium levels in µg/L</li>
<li><code>DWCr</code>: drinking water chromium levels in µg/L</li>
<li><code>UAs</code>: urinary arsenic levels in µg/L</li>
<li><code>UCd</code>: urinary cadmium levels in µg/L</li>
<li><code>UCr</code>: urinary chromium levels in µg/L</li>
</ul>
</div>
</div>
</div>
<div id="data-manipulation-using-base-r" class="section level2 hasAnchor">
<h2>Data Manipulation Using Base R<a href="data-manipulation-and-reshaping.html#data-manipulation-using-base-r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="merging-data-using-base-r-syntax" class="section level3 hasAnchor">
<h3>Merging Data Using Base R Syntax<a href="data-manipulation-and-reshaping.html#merging-data-using-base-r-syntax" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Merging datasets represents the joining together of two or more datasets, using a common identifier (generally some sort of ID) to connect the rows. This is useful if you have multiple datasets describing different aspects of the study, different variables, or different measures across the same samples. Samples could correspond to the same study participants, animals, cell culture samples, environmental media samples, etc, depending on the study design. In the current example, we will be joining human demographic data and environmental metals exposure data collected from drinking water and human urine samples.</p>
<p>Let’s start by merging the example demographic data with the chemical measurement data using the base R function <code>merge()</code>. To learn more about this function, you can type <code>?merge</code>, which brings up helpful information in the R console. To merge these datasets with the merge function, use the following code. The <code>by =</code> argument specifies the column used to match the rows of data.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="data-manipulation-and-reshaping.html#cb53-1" tabindex="-1"></a>full.data <span class="ot">&lt;-</span> <span class="fu">merge</span>(demographic_data, chemical_data, <span class="at">by =</span> <span class="st">&quot;ID&quot;</span>) </span>
<span id="cb53-2"><a href="data-manipulation-and-reshaping.html#cb53-2" tabindex="-1"></a><span class="fu">dim</span>(full.data) </span></code></pre></div>
<pre><code>## [1] 200  12</code></pre>
<p>This merged dataframe contains 200 rows x 12 columns. Viewing this merged dataframe, we can see that the <code>merge()</code> function retained the first column in each original dataframe (<code>ID</code>), though did not replicate it since it was used as the identifier for merging. All other columns include their original data, just merged together by the IDs in the first column.</p>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb55-1"><a href="data-manipulation-and-reshaping.html#cb55-1" tabindex="-1"></a><span class="fu">head</span>(full.data)</span></code></pre></div>
<pre><code>##   ID  BMI     MAge MEdu       BW GA     DWAs     DWCd     DWCr       UAs
## 1  1 27.7 22.99928    3 3180.058 34 6.426464 1.292941 51.67987 10.192695
## 2  2 26.8 30.05142    3 3210.823 43 7.832384 1.798535 50.10409 11.815088
## 3  3 33.2 28.04660    3 3311.551 40 7.516569 1.288461 48.74001 10.079057
## 4  4 30.1 34.81796    3 3266.844 32 5.906656 2.075259 50.92745  8.719123
## 5  5 37.4 42.68440    3 3664.088 35 7.181873 2.762643 55.16882  9.436559
## 6  6 33.3 24.94960    3 3328.988 40 9.723429 3.054057 51.14812 11.589403
##         UCd      UCr
## 1 0.7537104 42.60187
## 2 0.9789506 41.30757
## 3 0.1903262 36.47716
## 4 0.9364825 42.47987
## 5 1.4977829 47.78528
## 6 1.6645837 38.26386</code></pre>
<p>These datasets were actually quite easy to merge, since they had the same exact column identifier and number of rows. You can edit your script to include more specifics in instances when these may differ across datasets that you would like to merge. This option allows you to edit the name of the column that is used in each dataframe. Here, these are still the same “ID”, but you can see that adding the <code>by.x</code> and <code>by.y</code> arguments allows you to specify instances when different column names are used in the two datasets.</p>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb57-1"><a href="data-manipulation-and-reshaping.html#cb57-1" tabindex="-1"></a>full.data <span class="ot">&lt;-</span> <span class="fu">merge</span>(demographic_data, chemical_data, <span class="at">by.x =</span> <span class="st">&quot;ID&quot;</span>, <span class="at">by.y =</span> <span class="st">&quot;ID&quot;</span>) </span>
<span id="cb57-2"><a href="data-manipulation-and-reshaping.html#cb57-2" tabindex="-1"></a></span>
<span id="cb57-3"><a href="data-manipulation-and-reshaping.html#cb57-3" tabindex="-1"></a><span class="co"># Viewing data</span></span>
<span id="cb57-4"><a href="data-manipulation-and-reshaping.html#cb57-4" tabindex="-1"></a><span class="fu">head</span>(full.data)</span></code></pre></div>
<pre><code>##   ID  BMI     MAge MEdu       BW GA     DWAs     DWCd     DWCr       UAs
## 1  1 27.7 22.99928    3 3180.058 34 6.426464 1.292941 51.67987 10.192695
## 2  2 26.8 30.05142    3 3210.823 43 7.832384 1.798535 50.10409 11.815088
## 3  3 33.2 28.04660    3 3311.551 40 7.516569 1.288461 48.74001 10.079057
## 4  4 30.1 34.81796    3 3266.844 32 5.906656 2.075259 50.92745  8.719123
## 5  5 37.4 42.68440    3 3664.088 35 7.181873 2.762643 55.16882  9.436559
## 6  6 33.3 24.94960    3 3328.988 40 9.723429 3.054057 51.14812 11.589403
##         UCd      UCr
## 1 0.7537104 42.60187
## 2 0.9789506 41.30757
## 3 0.1903262 36.47716
## 4 0.9364825 42.47987
## 5 1.4977829 47.78528
## 6 1.6645837 38.26386</code></pre>
<p>Note that after merging datasets, it is always helpful to check that the merging was done properly before proceeding with your data analysis. Helpful checks could include viewing the merged dataset, checking the numbers of rows and columns to make sure chunks of data are not missing, and searching for values (or strings) that exist in one dataset but not the other, among other mechanisms of QA/QC.</p>
</div>
<div id="filtering-and-subsetting-data-using-base-r-syntax" class="section level3 hasAnchor">
<h3>Filtering and Subsetting Data Using Base R Syntax<a href="data-manipulation-and-reshaping.html#filtering-and-subsetting-data-using-base-r-syntax" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Filtering and subsetting data are useful tools when you need to focus on specific parts of your dataset for downstream analyses. These could represent, for example, specific samples or participants that meet certain criteria that you are interested in evaluating. It is also useful for removing unneeded variables or samples from dataframes as you are working through your script.</p>
<p>Note that in the examples that follow, we will create new dataframes that are distinguished from our original dataframe by adding sequential numbers to the end of the dataframe name (e.g., subset.data1, subset.data2, subset.data3). This style of dataframe naming is useful for the simple examples we are demonstrating, but in a full scripted analysis, we encourage the use of more descriptive dataframe names. For example, if you are subsetting your data to include only the first 100 rows, you could name that dataframe “data.first100.”</p>
<p>For this example, let’s first define a vector of columns that we want to keep in our analysis, then subset the data by keeping only the columns specified in our vector:</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="data-manipulation-and-reshaping.html#cb59-1" tabindex="-1"></a><span class="co"># Defining a vector of columns to keep in the analysis</span></span>
<span id="cb59-2"><a href="data-manipulation-and-reshaping.html#cb59-2" tabindex="-1"></a>subset.columns <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;BMI&quot;</span>, <span class="st">&quot;MAge&quot;</span>, <span class="st">&quot;MEdu&quot;</span>)</span>
<span id="cb59-3"><a href="data-manipulation-and-reshaping.html#cb59-3" tabindex="-1"></a></span>
<span id="cb59-4"><a href="data-manipulation-and-reshaping.html#cb59-4" tabindex="-1"></a><span class="co"># Subsetting the data by selecting the columns represented in the defined &#39;subset.columns&#39; vector</span></span>
<span id="cb59-5"><a href="data-manipulation-and-reshaping.html#cb59-5" tabindex="-1"></a>subset.data1 <span class="ot">&lt;-</span> full.data[,subset.columns]</span>
<span id="cb59-6"><a href="data-manipulation-and-reshaping.html#cb59-6" tabindex="-1"></a></span>
<span id="cb59-7"><a href="data-manipulation-and-reshaping.html#cb59-7" tabindex="-1"></a><span class="co"># Viewing the top of this subsetted dataframe</span></span>
<span id="cb59-8"><a href="data-manipulation-and-reshaping.html#cb59-8" tabindex="-1"></a><span class="fu">head</span>(subset.data1) </span></code></pre></div>
<pre><code>##    BMI     MAge MEdu
## 1 27.7 22.99928    3
## 2 26.8 30.05142    3
## 3 33.2 28.04660    3
## 4 30.1 34.81796    3
## 5 37.4 42.68440    3
## 6 33.3 24.94960    3</code></pre>
<p>We can also easily subset data based on row numbers. For example, to keep only the first 100 rows:</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="data-manipulation-and-reshaping.html#cb61-1" tabindex="-1"></a>subset.data2 <span class="ot">&lt;-</span> full.data[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>,]</span>
<span id="cb61-2"><a href="data-manipulation-and-reshaping.html#cb61-2" tabindex="-1"></a></span>
<span id="cb61-3"><a href="data-manipulation-and-reshaping.html#cb61-3" tabindex="-1"></a><span class="co"># Viewing the dimensions of this new dataframe</span></span>
<span id="cb61-4"><a href="data-manipulation-and-reshaping.html#cb61-4" tabindex="-1"></a><span class="fu">dim</span>(subset.data2)</span></code></pre></div>
<pre><code>## [1] 100  12</code></pre>
<p>To remove the first 100 rows, we use the same code as above, but include a <code>-</code> sign before our vector to indicate that these rows should be removed:</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="data-manipulation-and-reshaping.html#cb63-1" tabindex="-1"></a>subset.data3 <span class="ot">&lt;-</span> full.data[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>),]</span>
<span id="cb63-2"><a href="data-manipulation-and-reshaping.html#cb63-2" tabindex="-1"></a></span>
<span id="cb63-3"><a href="data-manipulation-and-reshaping.html#cb63-3" tabindex="-1"></a><span class="co"># Viewing the dimensions of this new dataframe</span></span>
<span id="cb63-4"><a href="data-manipulation-and-reshaping.html#cb63-4" tabindex="-1"></a><span class="fu">dim</span>(subset.data3)</span></code></pre></div>
<pre><code>## [1] 100  12</code></pre>
<p><strong>Conditional statements</strong> are also written to filter and subset data. A <strong>conditional statement</strong> is written to execute one block of code if the statement is true and a different block of code if the statement is false.</p>
<p>A conditional statement requires a Boolean or true/false statement that will be either <code>TRUE</code> or <code>FALSE</code>. A couple of the more commonly used functions used to create conditional statements include…</p>
<ul>
<li><code>if(){}</code> or an <strong>if statement</strong> means “execute R code when the condition is met”.</li>
<li><code>if(){} else{}</code> or an <strong>if/else statement</strong> means “execute R code when condition 1 is met, if not execute R code for condition 2”.</li>
<li><code>ifelse()</code> is a function that executes the same logic as an if/else statement. The first argument specifies a condition to be met. If that condition is met, R code in the second argument is executed, and if that condition is not met, R code in the third argument is executed.</li>
</ul>
<p>There are six comparison operators that are used to created these Boolean values:</p>
<ul>
<li><code>==</code> means “equals”.</li>
<li><code>!=</code> means “not equal”.</li>
<li><code>&lt;</code> means “less than”.</li>
<li><code>&gt;</code> means “greater than”.</li>
<li><code>&lt;=</code> means “less than or equal to”.</li>
<li><code>&gt;=</code> mean “greater than or equal to”.</li>
</ul>
<p>There are also three logical operators that are used to create these Boolean values:</p>
<ul>
<li><code>&amp;</code> means “and”.</li>
<li><code>|</code> means “or”.</li>
<li><code>!</code> means “not”.</li>
</ul>
<p>We can filter data based on conditions using the <code>subset()</code> function. For example, the following code filters for subjects whose BMI is greater than 25 and who have a college education:</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="data-manipulation-and-reshaping.html#cb65-1" tabindex="-1"></a>subset.data4 <span class="ot">&lt;-</span> <span class="fu">subset</span>(full.data, BMI <span class="sc">&gt;</span> <span class="dv">25</span> <span class="sc">&amp;</span> MEdu <span class="sc">==</span> <span class="dv">3</span>)</span></code></pre></div>
<p>Additionally, we can subset and select specific columns we would like to keep, using the <code>select</code> argument within the <code>subset()</code> function:</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="data-manipulation-and-reshaping.html#cb66-1" tabindex="-1"></a><span class="co"># Filtering for subjects whose BMI is less than 22 or greater than 27</span></span>
<span id="cb66-2"><a href="data-manipulation-and-reshaping.html#cb66-2" tabindex="-1"></a><span class="co"># Also selecting the BMI, maternal age, and maternal education columns</span></span>
<span id="cb66-3"><a href="data-manipulation-and-reshaping.html#cb66-3" tabindex="-1"></a>subset.data5 <span class="ot">&lt;-</span> <span class="fu">subset</span>(full.data, BMI <span class="sc">&lt;</span> <span class="dv">22</span> <span class="sc">|</span> BMI <span class="sc">&gt;</span> <span class="dv">27</span>, <span class="at">select =</span> subset.columns)</span></code></pre></div>
<p>For more information on the <code>subset()</code> function, see its associated <a href="https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/subset">documentation</a>.</p>
</div>
<div id="melting-and-casting-data-using-base-r-syntax" class="section level3 hasAnchor">
<h3>Melting and Casting Data using Base R Syntax<a href="data-manipulation-and-reshaping.html#melting-and-casting-data-using-base-r-syntax" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Melting and casting refers to the conversion of data to “long” or “wide” form as discussed previously in <strong>TAME 2.0 Module 1.4 Data Wrangling in Excel</strong>. You will often see data within the environmental health field in wide format, though long format is necessary for some procedures, such as plotting with <a href="https://ggplot2.tidyverse.org"><em>ggplot2</em></a> and performing certain analyses.</p>
<p>Here, we’ll illustrate some example script to melt and cast data using the <a href="https://www.rdocumentation.org/packages/reshape2/versions/1.4.4"><em>reshape2</em></a> package.
Let’s first install and load the <code>reshape2</code> package:</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="data-manipulation-and-reshaping.html#cb67-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;reshape2&quot;</span>))</span>
<span id="cb67-2"><a href="data-manipulation-and-reshaping.html#cb67-2" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;reshape2&quot;</span>);</span></code></pre></div>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="data-manipulation-and-reshaping.html#cb68-1" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span></code></pre></div>
<pre><code>## 
## Attaching package: &#39;reshape2&#39;</code></pre>
<pre><code>## The following object is masked from &#39;package:tidyr&#39;:
## 
##     smiths</code></pre>
<p>Using the fully merged dataframe, let’s remind ourselves what these data look like in the current dataframe format:</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="data-manipulation-and-reshaping.html#cb71-1" tabindex="-1"></a><span class="fu">head</span>(full.data)</span></code></pre></div>
<pre><code>##   ID  BMI     MAge MEdu       BW GA     DWAs     DWCd     DWCr       UAs
## 1  1 27.7 22.99928    3 3180.058 34 6.426464 1.292941 51.67987 10.192695
## 2  2 26.8 30.05142    3 3210.823 43 7.832384 1.798535 50.10409 11.815088
## 3  3 33.2 28.04660    3 3311.551 40 7.516569 1.288461 48.74001 10.079057
## 4  4 30.1 34.81796    3 3266.844 32 5.906656 2.075259 50.92745  8.719123
## 5  5 37.4 42.68440    3 3664.088 35 7.181873 2.762643 55.16882  9.436559
## 6  6 33.3 24.94960    3 3328.988 40 9.723429 3.054057 51.14812 11.589403
##         UCd      UCr
## 1 0.7537104 42.60187
## 2 0.9789506 41.30757
## 3 0.1903262 36.47716
## 4 0.9364825 42.47987
## 5 1.4977829 47.78528
## 6 1.6645837 38.26386</code></pre>
<p>These data are represented by single subject identifiers listed as unique IDs per row, with associated environmental measures and demographic data organized across the columns. Thus, this dataframe is currently in <strong>wide (also known as casted)</strong> format.</p>
<p>Let’s convert this dataframe to <strong>long (also known as melted)</strong> format. Here, will will specify that we want a row for each unique sample ID + variable measure pair by using <code>id = "ID"</code>:</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="data-manipulation-and-reshaping.html#cb73-1" tabindex="-1"></a>full.melted <span class="ot">&lt;-</span> <span class="fu">melt</span>(full.data, <span class="at">id =</span> <span class="st">&quot;ID&quot;</span>) </span>
<span id="cb73-2"><a href="data-manipulation-and-reshaping.html#cb73-2" tabindex="-1"></a></span>
<span id="cb73-3"><a href="data-manipulation-and-reshaping.html#cb73-3" tabindex="-1"></a><span class="co"># Viewing this new dataframe</span></span>
<span id="cb73-4"><a href="data-manipulation-and-reshaping.html#cb73-4" tabindex="-1"></a><span class="fu">head</span>(full.melted) </span></code></pre></div>
<pre><code>##   ID variable value
## 1  1      BMI  27.7
## 2  2      BMI  26.8
## 3  3      BMI  33.2
## 4  4      BMI  30.1
## 5  5      BMI  37.4
## 6  6      BMI  33.3</code></pre>
<p>You can see here that each measure that was originally contained as a unique column has been reoriented, such that the original column header is now listed throughout the second column labeled <code>variable</code>. Then, the third column contains the value of this variable.</p>
<p>Let’s see an example view of the middle of this new dataframe:</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="data-manipulation-and-reshaping.html#cb75-1" tabindex="-1"></a>full.melted[<span class="dv">1100</span><span class="sc">:</span><span class="dv">1110</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]</span></code></pre></div>
<pre><code>##       ID variable    value
## 1100 100     DWAs 7.928885
## 1101 101     DWAs 8.677403
## 1102 102     DWAs 8.115183
## 1103 103     DWAs 7.134189
## 1104 104     DWAs 8.816142
## 1105 105     DWAs 7.487227
## 1106 106     DWAs 7.541973
## 1107 107     DWAs 6.313516
## 1108 108     DWAs 6.654474
## 1109 109     DWAs 7.564429
## 1110 110     DWAs 7.357122</code></pre>
<p>Here, we can see a different variable (DWAs) now being listed. This continues throughout the entire dataframe, which has the following dimensions:</p>
<div class="sourceCode" id="cb77"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb77-1"><a href="data-manipulation-and-reshaping.html#cb77-1" tabindex="-1"></a><span class="fu">dim</span>(full.melted)</span></code></pre></div>
<pre><code>## [1] 2200    3</code></pre>
<p>Let’s now re-cast this dataframe back into wide format using the <code>dcast()</code> function. Here, we are telling the <code>dcast()</code> function to give us a sample (ID) for every variable in the column labeled <code>variable</code>. The column names from the variable column and corresponding values from the value column are then used to fill in the dataset:</p>
<div class="sourceCode" id="cb79"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb79-1"><a href="data-manipulation-and-reshaping.html#cb79-1" tabindex="-1"></a>full.cast <span class="ot">&lt;-</span> <span class="fu">dcast</span>(full.melted, ID <span class="sc">~</span> variable) </span>
<span id="cb79-2"><a href="data-manipulation-and-reshaping.html#cb79-2" tabindex="-1"></a><span class="fu">head</span>(full.cast)</span></code></pre></div>
<pre><code>##   ID  BMI     MAge MEdu       BW GA     DWAs     DWCd     DWCr       UAs
## 1  1 27.7 22.99928    3 3180.058 34 6.426464 1.292941 51.67987 10.192695
## 2  2 26.8 30.05142    3 3210.823 43 7.832384 1.798535 50.10409 11.815088
## 3  3 33.2 28.04660    3 3311.551 40 7.516569 1.288461 48.74001 10.079057
## 4  4 30.1 34.81796    3 3266.844 32 5.906656 2.075259 50.92745  8.719123
## 5  5 37.4 42.68440    3 3664.088 35 7.181873 2.762643 55.16882  9.436559
## 6  6 33.3 24.94960    3 3328.988 40 9.723429 3.054057 51.14812 11.589403
##         UCd      UCr
## 1 0.7537104 42.60187
## 2 0.9789506 41.30757
## 3 0.1903262 36.47716
## 4 0.9364825 42.47987
## 5 1.4977829 47.78528
## 6 1.6645837 38.26386</code></pre>
<p>Here, we can see that this dataframe is back in its original casted (or wide) format. Now that we’re familiar with some base R functions to reshape our data, let’s answer our original question: What is the average urinary chromium concentration for each maternal education level?</p>
<p>Although it is not necessary to calculate the average, we could first subset our data frame to only include the two columns we are interested in (MEdu and UCr):</p>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="data-manipulation-and-reshaping.html#cb81-1" tabindex="-1"></a>subset.data6 <span class="ot">&lt;-</span> full.data[,<span class="fu">c</span>(<span class="st">&quot;MEdu&quot;</span>, <span class="st">&quot;UCr&quot;</span>)]</span>
<span id="cb81-2"><a href="data-manipulation-and-reshaping.html#cb81-2" tabindex="-1"></a></span>
<span id="cb81-3"><a href="data-manipulation-and-reshaping.html#cb81-3" tabindex="-1"></a><span class="fu">head</span>(subset.data6)</span></code></pre></div>
<pre><code>##   MEdu      UCr
## 1    3 42.60187
## 2    3 41.30757
## 3    3 36.47716
## 4    3 42.47987
## 5    3 47.78528
## 6    3 38.26386</code></pre>
<p>Next, we will make a new data frame for each maternal education level:</p>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="data-manipulation-and-reshaping.html#cb83-1" tabindex="-1"></a><span class="co"># Creating new data frames based on maternal education category</span></span>
<span id="cb83-2"><a href="data-manipulation-and-reshaping.html#cb83-2" tabindex="-1"></a>data.matedu<span class="fl">.1</span> <span class="ot">&lt;-</span> <span class="fu">subset</span>(subset.data6, MEdu <span class="sc">==</span> <span class="dv">1</span>)</span>
<span id="cb83-3"><a href="data-manipulation-and-reshaping.html#cb83-3" tabindex="-1"></a>data.matedu<span class="fl">.2</span> <span class="ot">&lt;-</span> <span class="fu">subset</span>(subset.data6, MEdu <span class="sc">==</span> <span class="dv">2</span>)</span>
<span id="cb83-4"><a href="data-manipulation-and-reshaping.html#cb83-4" tabindex="-1"></a>data.matedu<span class="fl">.3</span> <span class="ot">&lt;-</span> <span class="fu">subset</span>(subset.data6, MEdu <span class="sc">==</span> <span class="dv">3</span>)</span>
<span id="cb83-5"><a href="data-manipulation-and-reshaping.html#cb83-5" tabindex="-1"></a></span>
<span id="cb83-6"><a href="data-manipulation-and-reshaping.html#cb83-6" tabindex="-1"></a><span class="co"># Previewing the first data frame to make sure our function is working as specified</span></span>
<span id="cb83-7"><a href="data-manipulation-and-reshaping.html#cb83-7" tabindex="-1"></a><span class="fu">head</span>(data.matedu<span class="fl">.1</span>)</span></code></pre></div>
<pre><code>##    MEdu      UCr
## 14    1 38.59349
## 18    1 47.77878
## 37    1 35.33980
## 63    1 34.72255
## 66    1 34.13982
## 76    1 31.38145</code></pre>
<p>Last, we can calculate the average urinary chromium concentration using each of our data frames:</p>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="data-manipulation-and-reshaping.html#cb85-1" tabindex="-1"></a><span class="fu">mean</span>(data.matedu<span class="fl">.1</span><span class="sc">$</span>UCr)</span></code></pre></div>
<pre><code>## [1] 39.88055</code></pre>
<div class="sourceCode" id="cb87"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb87-1"><a href="data-manipulation-and-reshaping.html#cb87-1" tabindex="-1"></a><span class="fu">mean</span>(data.matedu<span class="fl">.2</span><span class="sc">$</span>UCr)</span></code></pre></div>
<pre><code>## [1] 40.61807</code></pre>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="data-manipulation-and-reshaping.html#cb89-1" tabindex="-1"></a><span class="fu">mean</span>(data.matedu<span class="fl">.3</span><span class="sc">$</span>UCr)</span></code></pre></div>
<pre><code>## [1] 40.41556</code></pre>
<div class="question">
<p><i> With this, we can answer our <strong>Environmental Health Question</strong>: </i></p>
<p>What is the average urinary chromium concentration across different maternal education levels?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> The average urinary Chromium concentrations are 39.9 µg/L for participants with less than high school education, 40.6 µg/L for participants with high school or some college education, and 40.4 µg/L for participants with college education or greater.</p>
</div>
</div>
</div>
<div id="introduction-to-tidyverse" class="section level2 hasAnchor">
<h2>Introduction to Tidyverse<a href="data-manipulation-and-reshaping.html#introduction-to-tidyverse" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="https://www.tidyverse.org">Tidyverse</a> is a collection of packages that are commonly used to more efficiently organize and manipulate datasets in R. This collection of packages has its own specific type of syntax and formatting that differ slightly from base R functions. There are eight core tidyverse packages:</p>
<ul>
<li>For data visualization and exploration:
<ul>
<li><em>ggplot2</em></li>
</ul></li>
<li>For data wrangling and transformation:
<ul>
<li><em>dplyr</em></li>
<li><em>tidyr</em></li>
<li><em>stringr</em></li>
<li><em>forcats</em></li>
</ul></li>
<li>For data import and management:
<ul>
<li><em>tibble</em></li>
<li><em>readr</em></li>
</ul></li>
<li>For functional programming:
<ul>
<li><em>purr</em></li>
</ul></li>
</ul>
<p>Here, we will carry out all the of the same data organization exercises demonstrated above using packages that are part of The Tidyverse, specifically using functions that are part of the <em>dplyr</em> and <em>tidyr</em> packages.</p>
<div id="downloading-and-loading-the-tidyverse-package" class="section level3 hasAnchor">
<h3>Downloading and Loading the Tidyverse Package<a href="data-manipulation-and-reshaping.html#downloading-and-loading-the-tidyverse-package" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>If you don’t have <em>tidyverse</em> already installed, you will need to install it using:</p>
<div class="sourceCode" id="cb91"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb91-1"><a href="data-manipulation-and-reshaping.html#cb91-1" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(tidyverse)) </span>
<span id="cb91-2"><a href="data-manipulation-and-reshaping.html#cb91-2" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;tidyverse&quot;</span>)</span></code></pre></div>
<p>And then load the <em>tidyverse</em> package using:</p>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="data-manipulation-and-reshaping.html#cb92-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code></pre></div>
<p>Note that by loading the <em>tidyverse</em> package, you are also loading all of the packages included within The Tidyverse and do not need to separately load these packages.</p>
</div>
<div id="merging-data-using-tidyverse-syntax" class="section level3 hasAnchor">
<h3>Merging Data Using Tidyverse Syntax<a href="data-manipulation-and-reshaping.html#merging-data-using-tidyverse-syntax" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To merge the same example dataframes using <em>tidyverse</em>, you can run the following script:</p>
<div class="sourceCode" id="cb93"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb93-1"><a href="data-manipulation-and-reshaping.html#cb93-1" tabindex="-1"></a>full.data.tidy <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(demographic_data, chemical_data, <span class="at">by =</span> <span class="st">&quot;ID&quot;</span>)</span>
<span id="cb93-2"><a href="data-manipulation-and-reshaping.html#cb93-2" tabindex="-1"></a></span>
<span id="cb93-3"><a href="data-manipulation-and-reshaping.html#cb93-3" tabindex="-1"></a><span class="fu">head</span>(full.data.tidy)</span></code></pre></div>
<pre><code>##   ID  BMI     MAge MEdu       BW GA     DWAs     DWCd     DWCr       UAs
## 1  1 27.7 22.99928    3 3180.058 34 6.426464 1.292941 51.67987 10.192695
## 2  2 26.8 30.05142    3 3210.823 43 7.832384 1.798535 50.10409 11.815088
## 3  3 33.2 28.04660    3 3311.551 40 7.516569 1.288461 48.74001 10.079057
## 4  4 30.1 34.81796    3 3266.844 32 5.906656 2.075259 50.92745  8.719123
## 5  5 37.4 42.68440    3 3664.088 35 7.181873 2.762643 55.16882  9.436559
## 6  6 33.3 24.94960    3 3328.988 40 9.723429 3.054057 51.14812 11.589403
##         UCd      UCr
## 1 0.7537104 42.60187
## 2 0.9789506 41.30757
## 3 0.1903262 36.47716
## 4 0.9364825 42.47987
## 5 1.4977829 47.78528
## 6 1.6645837 38.26386</code></pre>
<p>Note that you can still merge dataframes that have different ID column names with the argument <code>by = c("ID.x", "ID.y")</code>. <em>tidyverse</em> also has other <code>join</code>, functions, shown in the graphic below (<a href="https://tavareshugo.github.io/r-intro-tidyverse-gapminder/08-joins/index.html">source</a>):
<img src="Module2_3_Input/Image1.svg" width="400px" style="display: block; margin: auto;" /></p>
<ul>
<li><strong>inner_join</strong> keeps only rows that have matching ID variables in both datasets</li>
<li><strong>full_join</strong> keeps the rows in both datasets</li>
<li><strong>left_join</strong> matches rows based on the ID variables in the first dataset (and omits any rows from the second dataset that do not have matching ID variables in the first dataset)</li>
<li><strong>right_join</strong> matches rows based on ID variables in the second dataset (and omits any rows from the first dataset that do not have matching ID variables in the second dataset)</li>
<li><strong>anti_join(x,y)</strong> keeps the rows that are unique to the first dataset</li>
<li><strong>anti_join(y,x)</strong> keeps the rows that are unique to the second dataset</li>
</ul>
</div>
<div id="the-pipe-operator" class="section level3 hasAnchor">
<h3>The Pipe Operator<a href="data-manipulation-and-reshaping.html#the-pipe-operator" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>One of the most important elements of Tidyverse syntax is use of the pipe operator (<code>%&gt;%</code>). The pipe operator can be used to chain multiple functions together. It takes the object (typically a dataframe) to the left of the pipe operator and passes it to the function to the right of the pipe operator. Multiple pipes can be used in chain to execute multiple data cleaning steps without the need for intermediate dataframes. The pipe operator can be used to pass data to functions within all of the Tidyverse universe packages, not just the functions demonstrated here.</p>
<p>Below, we can see the same code executed above, but this time with the pipe operator. The <code>demographic_data</code> dataframe is passed to <code>inner_join()</code> as the first argument to that function, with the following arguments remaining the same.</p>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="data-manipulation-and-reshaping.html#cb95-1" tabindex="-1"></a>full.data.tidy2 <span class="ot">&lt;-</span> demographic_data <span class="sc">%&gt;%</span> </span>
<span id="cb95-2"><a href="data-manipulation-and-reshaping.html#cb95-2" tabindex="-1"></a>  <span class="fu">inner_join</span>(chemical_data, <span class="at">by =</span> <span class="st">&quot;ID&quot;</span>)</span>
<span id="cb95-3"><a href="data-manipulation-and-reshaping.html#cb95-3" tabindex="-1"></a></span>
<span id="cb95-4"><a href="data-manipulation-and-reshaping.html#cb95-4" tabindex="-1"></a><span class="fu">head</span>(full.data.tidy2)</span></code></pre></div>
<pre><code>##   ID  BMI     MAge MEdu       BW GA     DWAs     DWCd     DWCr       UAs
## 1  1 27.7 22.99928    3 3180.058 34 6.426464 1.292941 51.67987 10.192695
## 2  2 26.8 30.05142    3 3210.823 43 7.832384 1.798535 50.10409 11.815088
## 3  3 33.2 28.04660    3 3311.551 40 7.516569 1.288461 48.74001 10.079057
## 4  4 30.1 34.81796    3 3266.844 32 5.906656 2.075259 50.92745  8.719123
## 5  5 37.4 42.68440    3 3664.088 35 7.181873 2.762643 55.16882  9.436559
## 6  6 33.3 24.94960    3 3328.988 40 9.723429 3.054057 51.14812 11.589403
##         UCd      UCr
## 1 0.7537104 42.60187
## 2 0.9789506 41.30757
## 3 0.1903262 36.47716
## 4 0.9364825 42.47987
## 5 1.4977829 47.78528
## 6 1.6645837 38.26386</code></pre>
<p>Because the pipe operator is often used in a chain, it is best practice is to start a new line after each pipe operator, with the new lines of code indented. This makes code with multiple piped steps easier to follow. However, if just one function is being executed, the pipe operator can be used on the same line as the input and function or omitted altogether (as shown in the previous two code chunks). Here is an example of placing the function to the right of the pipe operator on a new line, with placeholder functions shown as additional steps:</p>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="data-manipulation-and-reshaping.html#cb97-1" tabindex="-1"></a>full.data.tidy3 <span class="ot">&lt;-</span> demographic_data <span class="sc">%&gt;%</span> </span>
<span id="cb97-2"><a href="data-manipulation-and-reshaping.html#cb97-2" tabindex="-1"></a>  <span class="fu">inner_join</span>(chemical_data, <span class="at">by =</span> <span class="st">&quot;ID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb97-3"><a href="data-manipulation-and-reshaping.html#cb97-3" tabindex="-1"></a>  <span class="fu">additional_function_1</span>() <span class="sc">%&gt;%</span></span>
<span id="cb97-4"><a href="data-manipulation-and-reshaping.html#cb97-4" tabindex="-1"></a>  <span class="fu">additional_function_2</span>() </span></code></pre></div>
</div>
<div id="filtering-and-subsetting-data-using-tidyverse-syntax" class="section level3 hasAnchor">
<h3>Filtering and Subsetting Data Using Tidyverse Syntax<a href="data-manipulation-and-reshaping.html#filtering-and-subsetting-data-using-tidyverse-syntax" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="column-wise-functions" class="section level4 hasAnchor">
<h4>Column-wise functions<a href="data-manipulation-and-reshaping.html#column-wise-functions" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The <code>select()</code> function is used to subset columns in Tidyverse. Here, we can use our previously defined vector <code>subset.columns</code> in the <code>select()</code> function to keep only the columns in our <code>subset.columns</code> vector. The <code>all_of()</code> function tells the <code>select()</code> to keep all of the columns that match elements of the <code>subset.columns</code> vector.</p>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="data-manipulation-and-reshaping.html#cb98-1" tabindex="-1"></a>subset.tidy1 <span class="ot">&lt;-</span> full.data.tidy <span class="sc">%&gt;%</span> </span>
<span id="cb98-2"><a href="data-manipulation-and-reshaping.html#cb98-2" tabindex="-1"></a>    <span class="fu">select</span>(<span class="fu">all_of</span>(subset.columns))</span>
<span id="cb98-3"><a href="data-manipulation-and-reshaping.html#cb98-3" tabindex="-1"></a></span>
<span id="cb98-4"><a href="data-manipulation-and-reshaping.html#cb98-4" tabindex="-1"></a><span class="fu">head</span>(subset.tidy1)</span></code></pre></div>
<pre><code>##    BMI     MAge MEdu
## 1 27.7 22.99928    3
## 2 26.8 30.05142    3
## 3 33.2 28.04660    3
## 4 30.1 34.81796    3
## 5 37.4 42.68440    3
## 6 33.3 24.94960    3</code></pre>
<p>There are many different ways that <code>select()</code> can be used. See below for some examples using dummy variable names:</p>
<div class="sourceCode" id="cb100"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb100-1"><a href="data-manipulation-and-reshaping.html#cb100-1" tabindex="-1"></a><span class="co"># Select specific ranges in the dataframe</span></span>
<span id="cb100-2"><a href="data-manipulation-and-reshaping.html#cb100-2" tabindex="-1"></a>data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb100-3"><a href="data-manipulation-and-reshaping.html#cb100-3" tabindex="-1"></a>  <span class="fu">select</span>(start_column_1<span class="sc">:</span>end_column_1)</span>
<span id="cb100-4"><a href="data-manipulation-and-reshaping.html#cb100-4" tabindex="-1"></a></span>
<span id="cb100-5"><a href="data-manipulation-and-reshaping.html#cb100-5" tabindex="-1"></a>data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb100-6"><a href="data-manipulation-and-reshaping.html#cb100-6" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(start_column_1<span class="sc">:</span>end_column_1, start_column_2<span class="sc">:</span>end_column_2))</span>
<span id="cb100-7"><a href="data-manipulation-and-reshaping.html#cb100-7" tabindex="-1"></a></span>
<span id="cb100-8"><a href="data-manipulation-and-reshaping.html#cb100-8" tabindex="-1"></a><span class="co"># Select columns that match the elements in a character vector an an additional range of columns</span></span>
<span id="cb100-9"><a href="data-manipulation-and-reshaping.html#cb100-9" tabindex="-1"></a>data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb100-10"><a href="data-manipulation-and-reshaping.html#cb100-10" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(<span class="fu">all_of</span>(character_vector), start_column_1<span class="sc">:</span>end_column_1))</span></code></pre></div>
<p>To select columns that have names that contain specific strings, you can use functions such as <code>starts_with()</code>, <code>ends_with()</code>, and <code>contains()</code>. These functions allow you to ignore the case of the strings with <code>ignore.case = TRUE</code>. These arguments can be combined with specific column names and other selection ranges.</p>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="data-manipulation-and-reshaping.html#cb101-1" tabindex="-1"></a>data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb101-2"><a href="data-manipulation-and-reshaping.html#cb101-2" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">starts_with</span>(<span class="st">&quot;starting_string&quot;</span>))</span>
<span id="cb101-3"><a href="data-manipulation-and-reshaping.html#cb101-3" tabindex="-1"></a></span>
<span id="cb101-4"><a href="data-manipulation-and-reshaping.html#cb101-4" tabindex="-1"></a>data <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span></span>
<span id="cb101-5"><a href="data-manipulation-and-reshaping.html#cb101-5" tabindex="-1"></a>  <span class="fu">select</span>(other_column_to_keep, <span class="fu">starts_with</span>(<span class="st">&quot;starting_string&quot;</span>))</span></code></pre></div>
<p>To remove columns using tidyverse, you can use similar code, but include a <code>-</code> sign before the argument defining the columns.</p>
<div class="sourceCode" id="cb102"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb102-1"><a href="data-manipulation-and-reshaping.html#cb102-1" tabindex="-1"></a><span class="co"># Removing columns</span></span>
<span id="cb102-2"><a href="data-manipulation-and-reshaping.html#cb102-2" tabindex="-1"></a>subset.tidy2 <span class="ot">&lt;-</span> full.data.tidy <span class="sc">%&gt;%</span> </span>
<span id="cb102-3"><a href="data-manipulation-and-reshaping.html#cb102-3" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span><span class="fu">all_of</span>(subset.columns))</span>
<span id="cb102-4"><a href="data-manipulation-and-reshaping.html#cb102-4" tabindex="-1"></a></span>
<span id="cb102-5"><a href="data-manipulation-and-reshaping.html#cb102-5" tabindex="-1"></a><span class="co"># Viewing this new dataframe</span></span>
<span id="cb102-6"><a href="data-manipulation-and-reshaping.html#cb102-6" tabindex="-1"></a><span class="fu">head</span>(subset.tidy2) </span></code></pre></div>
<pre><code>##   ID       BW GA     DWAs     DWCd     DWCr       UAs       UCd      UCr
## 1  1 3180.058 34 6.426464 1.292941 51.67987 10.192695 0.7537104 42.60187
## 2  2 3210.823 43 7.832384 1.798535 50.10409 11.815088 0.9789506 41.30757
## 3  3 3311.551 40 7.516569 1.288461 48.74001 10.079057 0.1903262 36.47716
## 4  4 3266.844 32 5.906656 2.075259 50.92745  8.719123 0.9364825 42.47987
## 5  5 3664.088 35 7.181873 2.762643 55.16882  9.436559 1.4977829 47.78528
## 6  6 3328.988 40 9.723429 3.054057 51.14812 11.589403 1.6645837 38.26386</code></pre>
</div>
<div id="row-wise-functions" class="section level4 hasAnchor">
<h4>Row-wise functions<a href="data-manipulation-and-reshaping.html#row-wise-functions" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The <code>slice()</code> function can be used to keep or remove a certain number of rows based on their position within the dataframe. For example, we can retain only the first 100 rows using the following code:</p>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="data-manipulation-and-reshaping.html#cb104-1" tabindex="-1"></a>subset.tidy3 <span class="ot">&lt;-</span> full.data.tidy <span class="sc">%&gt;%</span> </span>
<span id="cb104-2"><a href="data-manipulation-and-reshaping.html#cb104-2" tabindex="-1"></a>    <span class="fu">slice</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>) </span>
<span id="cb104-3"><a href="data-manipulation-and-reshaping.html#cb104-3" tabindex="-1"></a></span>
<span id="cb104-4"><a href="data-manipulation-and-reshaping.html#cb104-4" tabindex="-1"></a><span class="fu">dim</span>(subset.tidy3)</span></code></pre></div>
<pre><code>## [1] 100  12</code></pre>
<p>Or, we can remove the first 100 rows:</p>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="data-manipulation-and-reshaping.html#cb106-1" tabindex="-1"></a>subset.tidy4 <span class="ot">&lt;-</span> full.data.tidy <span class="sc">%&gt;%</span> </span>
<span id="cb106-2"><a href="data-manipulation-and-reshaping.html#cb106-2" tabindex="-1"></a>    <span class="fu">slice</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>))</span>
<span id="cb106-3"><a href="data-manipulation-and-reshaping.html#cb106-3" tabindex="-1"></a></span>
<span id="cb106-4"><a href="data-manipulation-and-reshaping.html#cb106-4" tabindex="-1"></a><span class="fu">dim</span>(subset.tidy4)</span></code></pre></div>
<pre><code>## [1] 100  12</code></pre>
<p>The related functions <code>slice_min()</code> and <code>slice_max()</code> can be used to select rows with the smallest or largest values of a variable.</p>
<p>The <code>filter()</code> function can be used to keep or remove specific rows based on conditional statements. For example, we can keep only rows where BMI is greater than 25 and age is greater than 31:</p>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="data-manipulation-and-reshaping.html#cb108-1" tabindex="-1"></a>subset.tidy5 <span class="ot">&lt;-</span> full.data.tidy <span class="sc">%&gt;%</span> </span>
<span id="cb108-2"><a href="data-manipulation-and-reshaping.html#cb108-2" tabindex="-1"></a>    <span class="fu">filter</span>(BMI <span class="sc">&gt;</span> <span class="dv">25</span> <span class="sc">&amp;</span> MAge <span class="sc">&gt;</span> <span class="dv">31</span>)</span>
<span id="cb108-3"><a href="data-manipulation-and-reshaping.html#cb108-3" tabindex="-1"></a></span>
<span id="cb108-4"><a href="data-manipulation-and-reshaping.html#cb108-4" tabindex="-1"></a><span class="fu">dim</span>(subset.tidy5)</span></code></pre></div>
<pre><code>## [1] 49 12</code></pre>
</div>
<div id="combining-column-and-row-wise-functions" class="section level4 hasAnchor">
<h4>Combining column and row-wise functions<a href="data-manipulation-and-reshaping.html#combining-column-and-row-wise-functions" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Now, we can see how Tidyverse makes it easy to chain together multiple data manipulation steps. Here, we first filter rows based on values for BMI and age, then we select our columns of interest:</p>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="data-manipulation-and-reshaping.html#cb110-1" tabindex="-1"></a>subset.tidy6 <span class="ot">&lt;-</span> full.data.tidy <span class="sc">%&gt;%</span> </span>
<span id="cb110-2"><a href="data-manipulation-and-reshaping.html#cb110-2" tabindex="-1"></a>    <span class="fu">filter</span>(BMI <span class="sc">&gt;</span> <span class="dv">25</span> <span class="sc">&amp;</span> MAge <span class="sc">&gt;</span> <span class="dv">31</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb110-3"><a href="data-manipulation-and-reshaping.html#cb110-3" tabindex="-1"></a>    <span class="fu">select</span>(BMI, MAge, MEdu)</span>
<span id="cb110-4"><a href="data-manipulation-and-reshaping.html#cb110-4" tabindex="-1"></a></span>
<span id="cb110-5"><a href="data-manipulation-and-reshaping.html#cb110-5" tabindex="-1"></a><span class="fu">head</span>(subset.tidy6)</span></code></pre></div>
<pre><code>##    BMI     MAge MEdu
## 1 30.1 34.81796    3
## 2 37.4 42.68440    3
## 3 36.9 33.58589    3
## 4 33.7 33.82961    3
## 5 25.7 37.08028    3
## 6 28.4 47.85761    3</code></pre>
</div>
</div>
<div id="melting-and-casting-data-using-tidyverse-syntax" class="section level3 hasAnchor">
<h3>Melting and Casting Data Using Tidyverse Syntax<a href="data-manipulation-and-reshaping.html#melting-and-casting-data-using-tidyverse-syntax" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To melt and cast data in Tidyverse, you can use the pivot functions (i.e., <code>pivot_longer()</code> or <code>pivot_wider()</code>).</p>
<p>The first argument in the <code>pivot_longer()</code> function specifies which columns should be pivoted. This can be specified with either positive or negative selection - i.e., naming columns to pivot with a vector or range or naming columns not to pivot with a <code>-</code> sign. Here, we are telling the function to pivot all of the columns except the ID column, which we need to keep to be able to trace back which values came from which subject. The <code>names_to =</code> argument allows you to set what you want to name the column that stores the variable names (the column names in wide format). The <code>values_to =</code> argument allows you to set what you want to name the column that stores the values. We almost always call these columns “var” and “value”, respectively, but you can name them anything that makes sense for your dataset.</p>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="data-manipulation-and-reshaping.html#cb112-1" tabindex="-1"></a>full.pivotlong <span class="ot">&lt;-</span> full.data.tidy <span class="sc">%&gt;%</span> </span>
<span id="cb112-2"><a href="data-manipulation-and-reshaping.html#cb112-2" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(<span class="sc">-</span>ID, <span class="at">names_to =</span> <span class="st">&quot;var&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>)</span>
<span id="cb112-3"><a href="data-manipulation-and-reshaping.html#cb112-3" tabindex="-1"></a></span>
<span id="cb112-4"><a href="data-manipulation-and-reshaping.html#cb112-4" tabindex="-1"></a><span class="fu">head</span>(full.pivotlong, <span class="dv">15</span>)</span></code></pre></div>
<pre><code>## # A tibble: 15 × 3
##       ID var      value
##    &lt;int&gt; &lt;chr&gt;    &lt;dbl&gt;
##  1     1 BMI     27.7  
##  2     1 MAge    23.0  
##  3     1 MEdu     3    
##  4     1 BW    3180.   
##  5     1 GA      34    
##  6     1 DWAs     6.43 
##  7     1 DWCd     1.29 
##  8     1 DWCr    51.7  
##  9     1 UAs     10.2  
## 10     1 UCd      0.754
## 11     1 UCr     42.6  
## 12     2 BMI     26.8  
## 13     2 MAge    30.1  
## 14     2 MEdu     3    
## 15     2 BW    3211.</code></pre>
<p>To pivot our data back to wide format, we can use <code>pivot_wider()</code>, which will pull the column names from the column specified in the <code>names_from =</code> argument and the corresponding values from the column specified in the <code>values_from =</code> argument.</p>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="data-manipulation-and-reshaping.html#cb114-1" tabindex="-1"></a>full.pivotwide <span class="ot">&lt;-</span> full.pivotlong <span class="sc">%&gt;%</span> </span>
<span id="cb114-2"><a href="data-manipulation-and-reshaping.html#cb114-2" tabindex="-1"></a>    <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">&quot;var&quot;</span>, <span class="at">values_from =</span> <span class="st">&quot;value&quot;</span>)</span>
<span id="cb114-3"><a href="data-manipulation-and-reshaping.html#cb114-3" tabindex="-1"></a></span>
<span id="cb114-4"><a href="data-manipulation-and-reshaping.html#cb114-4" tabindex="-1"></a><span class="fu">head</span>(full.pivotwide)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 12
##      ID   BMI  MAge  MEdu    BW    GA  DWAs  DWCd  DWCr   UAs   UCd   UCr
##   &lt;int&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
## 1     1  27.7  23.0     3 3180.    34  6.43  1.29  51.7 10.2  0.754  42.6
## 2     2  26.8  30.1     3 3211.    43  7.83  1.80  50.1 11.8  0.979  41.3
## 3     3  33.2  28.0     3 3312.    40  7.52  1.29  48.7 10.1  0.190  36.5
## 4     4  30.1  34.8     3 3267.    32  5.91  2.08  50.9  8.72 0.936  42.5
## 5     5  37.4  42.7     3 3664.    35  7.18  2.76  55.2  9.44 1.50   47.8
## 6     6  33.3  24.9     3 3329.    40  9.72  3.05  51.1 11.6  1.66   38.3</code></pre>
<p>Now that we’re familiar with some <em>tidyverse</em> functions to reshape our data, let’s answer our original question: What is the average urinary Chromium concentration for each maternal education level?</p>
<p>We can use the <code>group_by()</code> function to group our dataset by education class, then the summarize function to calculate the mean of our variable of interest within each class. Note how much shorter and more efficient this code is than the code we used to calculate the same values using base R!</p>
<div class="sourceCode" id="cb116"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb116-1"><a href="data-manipulation-and-reshaping.html#cb116-1" tabindex="-1"></a>full.data <span class="sc">%&gt;%</span></span>
<span id="cb116-2"><a href="data-manipulation-and-reshaping.html#cb116-2" tabindex="-1"></a>  <span class="fu">group_by</span>(MEdu) <span class="sc">%&gt;%</span></span>
<span id="cb116-3"><a href="data-manipulation-and-reshaping.html#cb116-3" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">Avg_UCr =</span> <span class="fu">mean</span>(UCr))</span></code></pre></div>
<pre><code>## # A tibble: 3 × 2
##    MEdu Avg_UCr
##   &lt;int&gt;   &lt;dbl&gt;
## 1     1    39.9
## 2     2    40.6
## 3     3    40.4</code></pre>
<p>For more detailed and advanced examples of pivoting in Tidyverse, see the <a href="https://cran.r-project.org/web/packages/tidyr/vignettes/pivot.html">Tidyverse Pivoting Vignette</a>.</p>
</div>
</div>
<div id="concluding-remarks-6" class="section level2 hasAnchor">
<h2>Concluding Remarks<a href="data-manipulation-and-reshaping.html#concluding-remarks-6" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This training module provides an introductory level overview of data organization and manipulation basics in base R and Tidyverse, including merging, filtering, subsetting, melting, and casting, and demonstrates these methods with an environmentally relevant dataset. These methods are used regularly in scripted analyses and are important preparation steps for almost all downstream analyses and visualizations.</p>
<p><label class="tykfont">
Test Your Knowledge
</label></p>
<div class="tyk">
<p>What subjects, arranged from highest to lowest drinking water cadmium levels, had babies at at least 35 weeks and had urinary cadmium levels of at least 1.5 µg/L?</p>
<p><strong>Hint</strong>: Try using the <code>arrange()</code> function from the <em>tidyverse</em> package.</p>
</div>
<!-- ############################################################################################################################# -->
<!-- ################################################## -->
<!-- ################################################## -->
<!-- START SECTION FOUR  -->
<!-- ################################################## -->
<!-- ################################################## -->
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="coding-best-practices.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="improving-coding-efficiencies.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
