<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6.1 Descriptive Cohort Analyses | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="6.1 Descriptive Cohort Analyses | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6.1 Descriptive Cohort Analyses | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  <meta name="twitter:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  

<meta name="author" content="Rager Lab" />


<meta name="date" content="2024-12-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="Icons_Used_Throughout/Favicon.png" type="image/x-icon" />
<link rel="prev" href="unsupervised-machine-learning-part-2-additional-clustering-applications.html"/>
<link rel="next" href="omics-and-system-biology-transcriptomic-applications.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>
<link href="libs/table1-1.0/table1_defaults.css" rel="stylesheet" />
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-FGFKVX2X3G"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-FGFKVX2X3G');
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="Icons_Used_Throughout/TAME 2.0 Logo.png"><br></a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Chapter 1 Introductory <br>Data Science</b></span></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html"><i class="fa fa-check"></i>1.1 FAIR Data Management Practices</a>
<ul>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-training-module"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-fair"><i class="fa fa-check"></i>Introduction to FAIR</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#breaking-down-fair-letter-by-letter"><i class="fa fa-check"></i>Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#what-does-this-mean-for-you"><i class="fa fa-check"></i>What Does This Mean for You?</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#data-repositories-for-sharing-of-data"><i class="fa fa-check"></i>Data Repositories for Sharing of Data</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#recent-shifts-in-regulatory-policies-for-data-sharing"><i class="fa fa-check"></i>Recent Shifts in Regulatory Policies for Data Sharing</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#additional-training-resources-on-fair"><i class="fa fa-check"></i>Additional Training Resources on FAIR</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html"><i class="fa fa-check"></i>1.2 Data Sharing through Online Repositories</a>
<ul>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#an-overview-and-example-with-the-dataverse-repository"><i class="fa fa-check"></i>An Overview and Example with the Dataverse Repository</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#introduction-to-training-module-1"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#data-repositories"><i class="fa fa-check"></i>Data Repositories</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#the-dataverse-project"><i class="fa fa-check"></i>The Dataverse Project</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#what-is-a-dataverse"><i class="fa fa-check"></i>What is a Dataverse?</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#metadata"><i class="fa fa-check"></i>Metadata</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataverse"><i class="fa fa-check"></i>Creating a Dataverse</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataset"><i class="fa fa-check"></i>Creating a Dataset</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#concluding-remarks-1"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html"><i class="fa fa-check"></i>1.3 File Management using Github</a>
<ul>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#introduction-to-training-module-2"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-an-account"><i class="fa fa-check"></i>Creating an Account</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-a-repository"><i class="fa fa-check"></i>Creating a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#uploading-code"><i class="fa fa-check"></i>Uploading Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#adding-subfolders-in-a-repository"><i class="fa fa-check"></i>Adding Subfolders in a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-code"><i class="fa fa-check"></i>Updating Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-repository-titles-and-structure-to-support-a-manuscript"><i class="fa fa-check"></i>Updating Repository Titles and Structure to Support a Manuscript</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#tracking-code-changes-using-github-branches"><i class="fa fa-check"></i>Tracking Code Changes using Github Branches</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#concluding-remarks-2"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html"><i class="fa fa-check"></i>1.4 Data Wrangling in Excel</a>
<ul>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#introduction-to-training-module-3"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#save-a-copy-of-the-soon-to-be-organized-and-cleaned-dataset-as-a-new-file"><i class="fa fa-check"></i>Save a Copy of the Soon-To-Be Organized and Cleaned Dataset as a New File</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-extraneous-white-space"><i class="fa fa-check"></i>Remove Extraneous White Space</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#replace-missing-data-with-na"><i class="fa fa-check"></i>Replace Missing Data with “NA”</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-metadata-tab"><i class="fa fa-check"></i>Create a Metadata Tab</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#abbreviate-and-capitalize-categorical-data"><i class="fa fa-check"></i>Abbreviate and Capitalize Categorical Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#alphabetize-sort-the-data-by-the-categorical-variable-of-interest"><i class="fa fa-check"></i>Alphabetize (Sort) the Data by the Categorical Variable of Interest</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-new-subject-number-column"><i class="fa fa-check"></i>Create a New Subject Number Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-special-symbols-and-dashes"><i class="fa fa-check"></i>Remove Special Symbols and Dashes</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#bold-all-column-names-and-center-all-data"><i class="fa fa-check"></i>Bold all Column Names and Center all Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-subject-identifier-column"><i class="fa fa-check"></i>Create a Subject Identifier Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#separate-subject-demographic-data-from-experimental-measurements"><i class="fa fa-check"></i>Separate Subject Demographic Data from Experimental Measurements</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#convert-data-from-wide-to-long-format"><i class="fa fa-check"></i>Convert Data from Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#pivoting-data-from-a-wide-to-long-format"><i class="fa fa-check"></i>Pivoting Data from a Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#generating-summary-level-statistics-with-pivot-tables"><i class="fa fa-check"></i>Generating Summary-Level Statistics with Pivot Tables</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#excel-vs.-r-which-should-you-use"><i class="fa fa-check"></i>Excel vs. R: Which Should You Use?</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#concluding-remarks-3"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 2 Coding in R</b></span></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html"><i class="fa fa-check"></i>2.1 Downloading and Programming in R</a>
<ul>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-training-module-4"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#general-introduction-and-installation-of-r-and-rstudio"><i class="fa fa-check"></i>General Introduction and Installation of R and RStudio</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-r-packages"><i class="fa fa-check"></i>Introduction to R Packages</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#scripting-basics"><i class="fa fa-check"></i>Scripting Basics</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#code-troubleshooting"><i class="fa fa-check"></i>Code Troubleshooting</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#concluding-remarks-4"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html"><i class="fa fa-check"></i>2.2 Coding “Best” Practices</a>
<ul>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#introduction-to-training-module-5"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#scripting-file-types"><i class="fa fa-check"></i>Scripting File Types</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-headers-and-annotation"><i class="fa fa-check"></i>Script Headers and Annotation</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#coding-style"><i class="fa fa-check"></i>Coding Style</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-organization"><i class="fa fa-check"></i>Script Organization</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#concluding-remarks-5"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html"><i class="fa fa-check"></i>2.3 Data Manipulation and Reshaping</a>
<ul>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-training-module-6"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#data-manipulation-using-base-r"><i class="fa fa-check"></i>Data Manipulation Using Base R</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-tidyverse"><i class="fa fa-check"></i>Introduction to Tidyverse</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#concluding-remarks-6"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html"><i class="fa fa-check"></i>2.4 Improving Coding Efficiencies</a>
<ul>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#introduction-to-training-module-7"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#loops"><i class="fa fa-check"></i>Loops</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#functions"><i class="fa fa-check"></i>Functions</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#list-operations"><i class="fa fa-check"></i>List operations</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#concluding-remarks-7"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#additional-resources-2"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="part"><span><b>Chapter 3 Basics of Data <br>Analysis and Visualizations</b></span></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html"><i class="fa fa-check"></i>3.1 Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-data-visualizations"><i class="fa fa-check"></i>Introduction to Data Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-training-module-8"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#density-plot-visualization"><i class="fa fa-check"></i>Density Plot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#boxplot-visualization"><i class="fa fa-check"></i>Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#correlation-visualizations"><i class="fa fa-check"></i>Correlation Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#heatmap-visualization"><i class="fa fa-check"></i>Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#concluding-remarks-8"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html"><i class="fa fa-check"></i>3.2 Improving Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-data-visulization-conventions"><i class="fa fa-check"></i>Introduction to Data Visulization Conventions</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-training-module-9"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-boxplot-visualization"><i class="fa fa-check"></i>Creating an Improved Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-heatmap-visualization"><i class="fa fa-check"></i>Creating an Improved Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-multi-plot-figures"><i class="fa fa-check"></i>Creating Multi-Plot Figures</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#concluding-remarks-9"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html"><i class="fa fa-check"></i>3.3 Normality Tests and Data Transformations</a>
<ul>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#introduction-to-training-module-10"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#what-is-a-normal-distribution"><i class="fa fa-check"></i>What is a Normal Distribution?</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#qualitative-assessment-of-normality"><i class="fa fa-check"></i>Qualitative Assessment of Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#quantitative-normality-assessment"><i class="fa fa-check"></i>Quantitative Normality Assessment</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#data-transformation"><i class="fa fa-check"></i>Data Transformation</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#additional-considerations-regarding-normality"><i class="fa fa-check"></i>Additional Considerations Regarding Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#concluding-remarks-10"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html"><i class="fa fa-check"></i>3.4 Intoduction to Statistical Tests</a>
<ul>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#introduction-to-training-module-11"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#assessing-normality-homogeneity-of-variance"><i class="fa fa-check"></i>Assessing Normality &amp; Homogeneity of Variance</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#two-group-visualizations-and-statistical-comparisons-using-the-t-test"><i class="fa fa-check"></i>Two-Group Visualizations and Statistical Comparisons using the T-Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#three-group-visualizations-and-statistical-comparisons-using-an-anova"><i class="fa fa-check"></i>Three-Group Visualizations and Statistical Comparisons using an ANOVA</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#regression-modeling-and-visualization-linear-and-logistic-regressions"><i class="fa fa-check"></i>Regression Modeling and Visualization: Linear and Logistic Regressions</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#statistical-evaluations-of-categorical-data-using-the-chi-squared-test-and-fishers-exact-test"><i class="fa fa-check"></i>Statistical Evaluations of Categorical Data using the Chi-Squared Test and Fisher’s Exact Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#concluding-remarks-11"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 4 Converting Wet <br>Lab Data into Dry Lab <br>Analyses</b></span></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html"><i class="fa fa-check"></i>4.1 Overview of Experimental Design and Example Data</a>
<ul>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#introduction-to-training-module-12"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#replicates"><i class="fa fa-check"></i>Replicates</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#orientation-to-example-data-for-chapter-4"><i class="fa fa-check"></i>Orientation to Example Data for Chapter 4</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#concluding-remarks-12"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html"><i class="fa fa-check"></i>4.2 Data Import, Processing, and Summary Statistics</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#introduction-to-training-module-13"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#data-import"><i class="fa fa-check"></i>Data Import</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#handling-missing-values"><i class="fa fa-check"></i>Handling Missing Values</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#averaging-replicates"><i class="fa fa-check"></i>Averaging Replicates</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#descriptive-statistics"><i class="fa fa-check"></i>Descriptive Statistics</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#normality-assessment-and-data-transformation"><i class="fa fa-check"></i>Normality Assessment and Data Transformation</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#concluding-remarks-13"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html"><i class="fa fa-check"></i>4.3 Data Import from PDF Sources</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#introduction-to-training-module-14"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-from-many-single-pdfs-with-the-same-formatting"><i class="fa fa-check"></i>Importing Data from Many Single PDFs with the Same Formatting</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-stored-in-pdf-tables"><i class="fa fa-check"></i>Importing Data Stored in PDF Tables</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#concluding-remarks-14"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html"><i class="fa fa-check"></i>4.4 Two Group Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#introduction-to-training-module-15"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#overview-of-two-group-statistical-tests"><i class="fa fa-check"></i>Overview of Two Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#statistical-vs.-biological-significance"><i class="fa fa-check"></i>Statistical vs. Biological Significance</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#unpaired-test-example"><i class="fa fa-check"></i>Unpaired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#paired-test-example"><i class="fa fa-check"></i>Paired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#visualizing-results"><i class="fa fa-check"></i>Visualizing Results</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#concluding-remarks-15"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html"><i class="fa fa-check"></i>4.5 Multi-Group and Multi-Variable Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#introduction-to-training-module-16"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#overview-of-multi-group-statistical-tests"><i class="fa fa-check"></i>Overview of Multi-Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#multi-group-analysis-example"><i class="fa fa-check"></i>Multi-Group Analysis Example</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#visualization-of-multi-group-statistical-results"><i class="fa fa-check"></i>Visualization of Multi-Group Statistical Results</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#concluding-remarks-16"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html"><i class="fa fa-check"></i>4.6 Advanced Multi-Group Comparisons</a>
<ul>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#introduction-to-training-module-17"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#advanced-multi-group-comparisons-1"><i class="fa fa-check"></i>Advanced Multi-Group Comparisons</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#two-way-anova-example"><i class="fa fa-check"></i>Two-way ANOVA Example</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#ancova-example"><i class="fa fa-check"></i>ANCOVA Example</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#concluding-remarks-17"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 5 Machine Learning <br>&amp; Artificial Intelligence</b></span></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><i class="fa fa-check"></i>5.1 Introduction to Artificial Intelligence, Machine Learning, and Predictive Modeling for Environmental Health</a>
<ul>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#introduction-to-training-module-18"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#general-historical-context-and-taxonomy-of-modern-aiml"><i class="fa fa-check"></i>General Historical Context and Taxonomy of Modern AI/ML</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#application-of-machine-learning-in-environmental-health-science"><i class="fa fa-check"></i>Application of Machine Learning in Environmental Health Science</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#concluding-remarks-18"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html"><i class="fa fa-check"></i>5.2 Supervised Machine Learning</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-training-module-19"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-machine-learning"><i class="fa fa-check"></i>Types of Machine Learning</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-supervised-machine-learning-algorithms"><i class="fa fa-check"></i>Types of Supervised Machine Learning Algorithms</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#training-supervised-machine-learning-models"><i class="fa fa-check"></i>Training Supervised Machine Learning Models</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#assessing-classification-based-model-performance"><i class="fa fa-check"></i>Assessing Classification-Based Model Performance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-activity-and-example-dataset"><i class="fa fa-check"></i>Introduction to Activity and Example Dataset</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#testing-for-differences-in-predictor-variables-across-the-outcome-classes"><i class="fa fa-check"></i>Testing for Differences in Predictor Variables across the Outcome Classes</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#predicting-ias-detection-with-a-random-forest-rf-model"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#class-imbalance"><i class="fa fa-check"></i>Class Imbalance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#concluding-remarks-19"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html"><i class="fa fa-check"></i>5.3 Supervised Machine Learning Model Interpretation</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-training-module-20"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance"><i class="fa fa-check"></i>Variable Importance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary"><i class="fa fa-check"></i>Decision Boundary</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-example-dataset-and-activity"><i class="fa fa-check"></i>Introduction to Example Dataset and Activity</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#predicting-ias-detection-with-a-random-forest-rf-model-1"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance-plot"><i class="fa fa-check"></i>Variable Importance Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot"><i class="fa fa-check"></i>Decision Boundary Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot-incorporating-smote"><i class="fa fa-check"></i>Decision Boundary Plot Incorporating SMOTE</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#concluding-remarks-20"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><i class="fa fa-check"></i>5.4 Unsupervised Machine Learning Part 1: K-Means Clustering &amp; PCA</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-training-module-21"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-unsupervised-machine-learning"><i class="fa fa-check"></i>Introduction to Unsupervised Machine Learning</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-example-data"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#identifying-clusters-of-chemicals-through-k-means"><i class="fa fa-check"></i>Identifying Clusters of Chemicals through <em>K</em>-Means</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#principal-component-analysis-pca-1"><i class="fa fa-check"></i>Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#incorporating-k-means-into-pca-for-predictive-modeling"><i class="fa fa-check"></i>Incorporating <em>K</em>-Means into PCA for Predictive Modeling</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#concluding-remarks-21"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><i class="fa fa-check"></i>5.5 Unsupervised Machine Learning Part 2: Additional Clustering Applications</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-training-module-22"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#k-means-clustering-1"><i class="fa fa-check"></i><em>K</em>-Means Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-example-data-1"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-1"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#clustering-plot"><i class="fa fa-check"></i>Clustering Plot</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-visualization"><i class="fa fa-check"></i>Hierarchical Clustering Visualization</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#variable-contributions"><i class="fa fa-check"></i>Variable Contributions</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#concluding-remarks-22"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 6 Applications in <br>Toxicology &amp; Exposure <br>Science</b></span></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html"><i class="fa fa-check"></i>6.1 Descriptive Cohort Analyses</a>
<ul>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#introduction-to-training-module-23"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#participant-exploration"><i class="fa fa-check"></i>Participant Exploration</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#chemical-detection-filtering"><i class="fa fa-check"></i>Chemical Detection Filtering</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#outlier-identification"><i class="fa fa-check"></i>Outlier Identification</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#summary-statistics-tables"><i class="fa fa-check"></i>Summary Statistics Tables</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#demographics-table"><i class="fa fa-check"></i>Demographics Table</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#concluding-remarks-23"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html"><i class="fa fa-check"></i>6.2 -Omics and System Biology: Transcriptomic Applications</a>
<ul>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-training-module-24"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-the-field-of--omics"><i class="fa fa-check"></i>Introduction to the Field of “-Omics”</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-transcriptomics"><i class="fa fa-check"></i>Introduction to Transcriptomics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#formatting-data-for-downstream-statistics"><i class="fa fa-check"></i>Formatting Data for Downstream Statistics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#transcriptomics-data-qaqc"><i class="fa fa-check"></i>Transcriptomics Data QA/QC</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#controlling-for-sources-of-sample-heterogeneity"><i class="fa fa-check"></i>Controlling for Sources of Sample Heterogeneity</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e.g.-biomass-smoke-exposure"><i class="fa fa-check"></i>Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure)</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-ma-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using MA Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-volcano-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using Volcano Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis"><i class="fa fa-check"></i>Interpretting Findings at the Systems Level through Pathway Enrichment Analysis</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#concluding-remarks-24"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><i class="fa fa-check"></i>6.3 Mixtures Analysis Methods Part 1: Overview and Example with Quantile G-Computation</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-training-module-25"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#overview-of-mixtures-analysis"><i class="fa fa-check"></i>Overview of Mixtures Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-example-data-2"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#mixtures-model-with-standard-qqcomp"><i class="fa fa-check"></i>Mixtures Model with Standard qqcomp</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#concluding-remarks-25"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#additional-resources-11"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><i class="fa fa-check"></i>6.4 Mixtures Analysis Methods Part 2: Bayesian Kernel Machine Regression</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-training-module-26"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-example-data-3"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-bkmr"><i class="fa fa-check"></i>Introduction to BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#run-bkmr"><i class="fa fa-check"></i>Run BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#concluding-remarks-26"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#additional-resources-12"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html"><i class="fa fa-check"></i>6.5 Mixtures Analysis Methods Part 3: Sufficient Similarity</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-training-module-27"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-toxicant-and-dataset"><i class="fa fa-check"></i>Introduction to Toxicant and Dataset</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#workspace-preparation-and-data-import-9"><i class="fa fa-check"></i>Workspace Preparation and Data Import</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#chemistry-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Chemistry-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#toxicity-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Toxicity-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#comparing-chemistry-vs.-toxicity-sufficient-similarity-analyses"><i class="fa fa-check"></i>Comparing Chemistry vs. Toxicity Sufficient Similarity Analyses</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#concluding-remarks-27"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i>6.6 Toxicokinetic Modeling</a>
<ul>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-training-module-28"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-toxicokinetic-modeling"><i class="fa fa-check"></i>Introduction to Toxicokinetic Modeling</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#data-and-models-used-in-toxicokinetic-modeling-tk"><i class="fa fa-check"></i>Data and Models used in Toxicokinetic Modeling (TK)</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-toxicokinetics"><i class="fa fa-check"></i>Reverse Toxicokinetics</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#capturing-population-variability-in-toxicokinetics-and-uncertainty-in-chemical-specific-parameters"><i class="fa fa-check"></i>Capturing Population Variability in Toxicokinetics, and Uncertainty in Chemical-Specific Parameters</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-tk-calculating-administered-equivalent-doses-for-toxcast-bioactive-concentrations"><i class="fa fa-check"></i>Reverse TK: Calculating Administered Equivalent Doses for ToxCast Bioactive Concentrations</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#comparing-equivalent-doses-estimated-to-elicit-toxicity-hazard-to-external-exposure-estimates-exposure-for-chemical-prioritization-by-bioactivity-exposure-ratios-bers"><i class="fa fa-check"></i>Comparing Equivalent Doses Estimated to Elicit Toxicity (Hazard) to External Exposure Estimates (Exposure), for Chemical Prioritization by Bioactivity-Exposure Ratios (BERs)</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#concluding-remarks-28"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html"><i class="fa fa-check"></i>Chemical Read-Across for Toxicity Predictions</a>
<ul>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#introduction-to-training-module-29"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#introduction-to-activity"><i class="fa fa-check"></i>Introduction to Activity</a></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#read-across-example-analysis"><i class="fa fa-check"></i>Read-Across Example Analysis</a></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#calculating-chemical-similarities"><i class="fa fa-check"></i>Calculating Chemical Similarities</a></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#chemical-read-across-to-predict-acute-toxicity"><i class="fa fa-check"></i>Chemical Read-Across to Predict Acute Toxicity</a></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#concluding-remarks-29"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 7 Environmental<br> Health Database Mining</b></span></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html"><i class="fa fa-check"></i>7.1 Comparative Toxicogenomics Database</a>
<ul>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#introduction-to-training-module-30"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#ctd-data-in-r"><i class="fa fa-check"></i>CTD Data in R</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#identifying-genes-under-epigenetic-control"><i class="fa fa-check"></i>Identifying Genes Under Epigenetic Control</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#concluding-remarks-30"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html"><i class="fa fa-check"></i>7.2 Gene Expression Omnibus</a>
<ul>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-training-module-31"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-geo"><i class="fa fa-check"></i>Introduction to GEO</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-example-data-4"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#geo-data-in-r"><i class="fa fa-check"></i>GEO Data in R</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#visualizing-data"><i class="fa fa-check"></i>Visualizing Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#statistical-analyses"><i class="fa fa-check"></i>Statistical Analyses</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#concluding-remarks-31"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html"><i class="fa fa-check"></i>7.3 CompTox Dashboard Data through APIs</a>
<ul>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#introduction-to-training-module-32"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#training-modules-environmental-health-questions-24"><i class="fa fa-check"></i>Training Module’s Environmental Health Questions</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#script-preparations-12"><i class="fa fa-check"></i>Script Preparations</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#introduction-to-comptox-chemicals-dashboard"><i class="fa fa-check"></i>Introduction to CompTox Chemicals Dashboard</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#cctes-ctx-application-programming-interfaces-apis-for-automated-batch-search-of-the-ccd"><i class="fa fa-check"></i>CCTE’s CTX Application Programming Interfaces (APIs) for Automated Batch Search of the CCD</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#comparing-physico-chemical-properties-between-two-important-environmental-contaminant-lists"><i class="fa fa-check"></i>Comparing Physico-chemical Properties between Two Important Environmental Contaminant Lists</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#hazard-data-genotoxicity"><i class="fa fa-check"></i>Hazard Data: Genotoxicity</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#hazard-resource"><i class="fa fa-check"></i>Hazard Resource</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#concluding-remarks-32"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html"><i class="fa fa-check"></i>7.4 Database Integration: Air Quality, Mortality, and Environmental Justice Data</a>
<ul>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#introduction-to-training-module-33"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#introduction-to-exposure-and-health-databases"><i class="fa fa-check"></i>Introduction to Exposure and Health Databases</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#analyzing-relationships-between-pm2.5-and-mortality"><i class="fa fa-check"></i>Analyzing Relationships between PM<sub>2.5</sub> and Mortality</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#environmental-justice-considerations"><i class="fa fa-check"></i>Environmental Justice Considerations</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#concluding-remarks-33"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="descriptive-cohort-analyses" class="section level1 hasAnchor">
<h1>6.1 Descriptive Cohort Analyses<a href="descriptive-cohort-analyses.html#descriptive-cohort-analyses" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This training module was developed by Elise Hickman, Kyle Roell, and Julia E. Rager.</p>
<p>All input files (script, data, and figures) can be downloaded from the <a href="https://github.com/UNCSRP/TAME2">UNC-SRP TAME2 GitHub website</a>.</p>
<div id="introduction-to-training-module-23" class="section level2 hasAnchor">
<h2>Introduction to Training Module<a href="descriptive-cohort-analyses.html#introduction-to-training-module-23" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Human cohort datasets are very commonly analyzed and integrated in environmental health research. Commone research study designs that incorporate human data include clinical, epidemiological, biomonitoring, and/or biomarker study designs. These datasets represent metrics of health and exposure collected from human participants at one or many points in time. Although these datasets can lend themselves to highly complex analyses, it is important to first explore the basic dataset properties to understand data missingness, filter data appropriately, generate demographic tables and summary statistics, and identify outliers. In this module, we will work through these common steps with an example dataset and discuss additional considerations when working with human cohort datasets.</p>
<p>Our example data are derived from a study in which chemical exposure profiles were collected using silicone wristbands. Silicone wristbands are an affordable and minimally invasive method for sampling personal chemical exposure profiles. This exposure monitoring technique has been described through previous publications (see <strong>TAME 2.0 Module 3.3 Normality Tests and Data Transformations</strong>). The example workflow can also apply to other study designs, including biomonitoring and biomarker studies, which require careful consideration of chemical or biological marker detection filters, transparent reporting of descriptive statistics, and demographics tables.</p>
<div id="training-modules-environmental-health-questions-15" class="section level3 hasAnchor">
<h3>Training Module’s Environmental Health Questions<a href="descriptive-cohort-analyses.html#training-modules-environmental-health-questions-15" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>What proportion of participants wore their wristbands for all seven days?</li>
<li>How many chemicals were detected in at least 20% of participants?</li>
<li>What are the demographics of the study participants?</li>
</ol>
</div>
<div id="workspace-preparation-and-data-import-7" class="section level3 hasAnchor">
<h3>Workspace Preparation and Data Import<a href="descriptive-cohort-analyses.html#workspace-preparation-and-data-import-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb616"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb616-1"><a href="descriptive-cohort-analyses.html#cb616-1" tabindex="-1"></a><span class="co"># Load packages</span></span>
<span id="cb616-2"><a href="descriptive-cohort-analyses.html#cb616-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse) <span class="co"># for data organization and manipulation</span></span>
<span id="cb616-3"><a href="descriptive-cohort-analyses.html#cb616-3" tabindex="-1"></a><span class="fu">library</span>(janitor) <span class="co"># for data cleaning</span></span>
<span id="cb616-4"><a href="descriptive-cohort-analyses.html#cb616-4" tabindex="-1"></a><span class="fu">library</span>(openxlsx) <span class="co"># for reading in and writing out files</span></span>
<span id="cb616-5"><a href="descriptive-cohort-analyses.html#cb616-5" tabindex="-1"></a><span class="fu">library</span>(DT) <span class="co"># for displaying tables</span></span>
<span id="cb616-6"><a href="descriptive-cohort-analyses.html#cb616-6" tabindex="-1"></a><span class="fu">library</span>(table1) <span class="co"># for making tables</span></span>
<span id="cb616-7"><a href="descriptive-cohort-analyses.html#cb616-7" tabindex="-1"></a><span class="fu">library</span>(patchwork) <span class="co"># for graphing</span></span>
<span id="cb616-8"><a href="descriptive-cohort-analyses.html#cb616-8" tabindex="-1"></a><span class="fu">library</span>(purrr) <span class="co"># for summary stats</span></span>
<span id="cb616-9"><a href="descriptive-cohort-analyses.html#cb616-9" tabindex="-1"></a><span class="fu">library</span>(factoextra) <span class="co"># for PCA outlier detection</span></span>
<span id="cb616-10"><a href="descriptive-cohort-analyses.html#cb616-10" tabindex="-1"></a><span class="fu">library</span>(table1) <span class="co"># for making demographics table</span></span>
<span id="cb616-11"><a href="descriptive-cohort-analyses.html#cb616-11" tabindex="-1"></a></span>
<span id="cb616-12"><a href="descriptive-cohort-analyses.html#cb616-12" tabindex="-1"></a><span class="co"># Make sure select is calling the correct function</span></span>
<span id="cb616-13"><a href="descriptive-cohort-analyses.html#cb616-13" tabindex="-1"></a>select <span class="ot">&lt;-</span> dplyr<span class="sc">::</span>select</span>
<span id="cb616-14"><a href="descriptive-cohort-analyses.html#cb616-14" tabindex="-1"></a></span>
<span id="cb616-15"><a href="descriptive-cohort-analyses.html#cb616-15" tabindex="-1"></a><span class="co"># Set graphing theme</span></span>
<span id="cb616-16"><a href="descriptive-cohort-analyses.html#cb616-16" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>())</span></code></pre></div>
<p>First, we will import our raw chemical data and preview it.</p>
<div class="sourceCode" id="cb617"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb617-1"><a href="descriptive-cohort-analyses.html#cb617-1" tabindex="-1"></a>wrist_data <span class="ot">&lt;-</span> <span class="fu">read.xlsx</span>(<span class="st">&quot;Module6_1_Input/Module6_1_InputData1.xlsx&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb617-2"><a href="descriptive-cohort-analyses.html#cb617-2" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), \(x) <span class="fu">as.numeric</span>(x)))</span>
<span id="cb617-3"><a href="descriptive-cohort-analyses.html#cb617-3" tabindex="-1"></a></span>
<span id="cb617-4"><a href="descriptive-cohort-analyses.html#cb617-4" tabindex="-1"></a><span class="fu">datatable</span>(wrist_data[ , <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>])</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-16d386894e30177e3b22" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-16d386894e30177e3b22">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97"],[1014,1018,1019,1023,1026,1037,1039,1049,1052,1054,1061,1063,1064,1070,1071,1073,1075,1076,1079,1081,1082,1084,1085,1088,1091,1093,1094,1095,1096,1098,1099,1101,1102,1105,1110,1114,1116,1117,1118,1119,1120,1124,1126,1127,1128,1129,1130,1131,1133,1135,1144,1145,1147,1153,1154,1156,1159,1161,1163,1168,1169,1171,1172,1179,1182,1187,1188,1191,1193,1194,1196,1199,1200,1201,1202,1207,1210,1212,1215,1216,1227,1229,1250,1252,1253,1254,1255,1257,1259,1260,1261,1263,1264,1265,1266,1268,1272],[7,7,7,7,7,7,7,7,7,7,6,7,7,7,7,7,7,7,6,7,7,7,7,7,7,7,7,7,7,6,6,7,7,7,7,7,5,7,7,7,7,5,7,7,7,2,7,7,7,7,7,7,1,7,7,7,3,7,5,7,7,7,7,7,7,2,7,7,7,1,7,7,6,7,7,7,7,7,7,7,7,6,7,7,7,7,7,7,7,7,7,7,7,7,7,7,7],[134.12,211.97,268.6,272.28,197.16,210.87,60.5,40.51,205.53,330.78,86.90000000000001,111.11,150.37,835.23,390.6,67.05,212.45,631.6900000000001,302.98,143.51,158.84,221.88,1661.73,66.23999999999999,763.87,166.55,530.74,176.16,172.02,4468.06,187.98,146.88,282.6,204.61,100.58,55.22,593.86,50.97,1051.19,373.37,1007.97,633.33,71.70999999999999,658.46,540.66,34.95,37.56,54.74,15.12,110.6,601.63,252.9,17.6,163.66,58.5,107.4,16.13,865.15,349.99,198.25,40.39,92.7,62.8,1642.88,41.96,188.82,68.43000000000001,42.72,148.76,723.2,369.28,74.73999999999999,26.56,94.90000000000001,38.04,222.68,238.16,103.87,95.01000000000001,59.2,215.62,146.01,78.03,45.42,252.45,77.08,12.62,101.08,155.59,792.09,2224.63,485.22,36.92,707.99,62.25,482.52,15.35],[6.19,58.59,8.19,8.539999999999999,24.01,34.89,7.99,26.21,46.91,50.51,6.09,53.42,19.28,38.28,85.55,16.32,46.29,134.2,65.03,18.37,25.75,24.68,230.28,6.19,49.43,21.37,72.17,23.59,13.7,140.45,4.64,24.92,35.69,26.83,11.72,17.85,40.5,9.17,155.93,45.61,133.61,74.73999999999999,14.02,98.75,96.66,8.050000000000001,11.36,6.41,null,31.68,37.17,50.11,1.01,17.83,12.96,27.8,6.94,7.5,51.52,18.19,null,21.78,15.58,149.52,19.52,44.58,null,null,7.11,41.9,44.88,6.19,6.12,null,5.74,30.54,47.56,11.41,20.85,8.35,17.74,13.96,8.039999999999999,10.37,174.08,9.18,0.41,36.81,33.16,24.5,107.5,75.73999999999999,12.19,123.8,6.31,25.74,0.16],[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,0.28,null,null,null,null,null,null,null,null,null,null,null,1.27,null,2.88,null,null,0.13,3.44,null,null,0.09,null,null,null,0.9399999999999999,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,2.66,null,null,null,null,null,null,null,null,null,null,null,null,null],[37.09,96.48,86.86,77.65000000000001,59.06,89.68000000000001,17.36,18.25,128.96,104.15,25.62,33.43,55.87,264.6,189.3,17.26,75.53,322.12,119.36,47.45,65.47,71.83,653.09,22.93,246.24,64.34999999999999,205.95,54.51,64.02,1568.76,19.77,45.87,115.57,70.34999999999999,44.31,25.42,195.18,17.12,513.78,121.66,409.59,198.02,36.59,274.87,181.16,13.84,12.21,13.17,7.66,36.63,220.87,89.34,6.54,71.63,15.15,34.67,2.15,148.7,143.19,61.83,13.8,31.76,18.72,643.5599999999999,15.56,55.26,24.47,6.09,75.33,238.14,140.23,24.68,9.67,75.81999999999999,11.41,87.45,71.41,53.31,24.32,18.69,66.26000000000001,43.96,24.34,18.69,45.81,20.67,4.14,34.1,57.15,290.45,641.36,188.41,13.67,262.49,15.4,109.35,6.28]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>S_ID<\/th>\n      <th>Ndays<\/th>\n      <th>alkylOPE_2IPPDPP<\/th>\n      <th>alkylOPE_3IPPDPP<\/th>\n      <th>alkylOPE_2tBPDPP<\/th>\n      <th>alkylOPE_4IPPDPP<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"S_ID","targets":1},{"name":"Ndays","targets":2},{"name":"alkylOPE_2IPPDPP","targets":3},{"name":"alkylOPE_3IPPDPP","targets":4},{"name":"alkylOPE_2tBPDPP","targets":5},{"name":"alkylOPE_4IPPDPP","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>In this study, 97 participants wore silicone wristbands for one week, and chemical concentrations on the wristbands were measured with gas chromatography mass spectrometry. This dataframe consists of a column with a unique identifier for each participant (<code>S_ID</code>), a column describing the number of days that participant wore the wristband (<code>Ndays</code>), and subsequent columns containing the amount of each chemical detected (nanograms of chemical per gram of wristband). The chemical columns are labeled with the chemical class first (e.g., alkyl OPE, or alkyl organophosphate ester), followed by and underscore and the chemical name (e.g., 2IPPDPP). This dataset contains 110 different chemicals categorized into 8 chemical classes (listed below with their abbreviations):</p>
<ul>
<li>Brominated diphenyl ether (BDE)</li>
<li>Brominated flame retardant (BFR)</li>
<li>Organophosphate ester (OPE)</li>
<li>Polycyclic aromatic hydrocarbon (PAH)</li>
<li>Polychlorinated biphenyl (PCB)</li>
<li>Pesticide (Pest)</li>
<li>Phthalate (Phthal)</li>
<li>Alkyl organophosphate ester (alkylOPE)</li>
</ul>
<p>Through the data exploration and cleaning process, we will aim to:</p>
<ul>
<li>Understand participant behaviors</li>
<li>Filter out chemicals with low detection</li>
<li>Generate a supplemental table containing chemical detection information and summary statistics such as minimum, mean, median, and maximum</li>
<li>Identify participant outliers</li>
<li>Generate a demographics table</li>
</ul>
<p>Although these steps are somewhat specific to our example dataset, similar steps can be taken with other datasets. We recommend thinking through the structure of your data and outlining data exploration and cleaning steps prior to starting your analysis. This process can be somewhat time-consuming and tedious but is important to ensure that your data are well-suited for downstream analyses. In addition, these steps should be included in any resulting manuscript as part of the narrative relating to the study cohort and data cleaning.</p>
</div>
</div>
<div id="participant-exploration" class="section level2 hasAnchor">
<h2>Participant Exploration<a href="descriptive-cohort-analyses.html#participant-exploration" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We can use <em>tidyverse</em> functions to quickly tabulate how many days participants wore the wristbands.</p>
<div class="sourceCode" id="cb618"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb618-1"><a href="descriptive-cohort-analyses.html#cb618-1" tabindex="-1"></a>wrist_data <span class="sc">%&gt;%</span></span>
<span id="cb618-2"><a href="descriptive-cohort-analyses.html#cb618-2" tabindex="-1"></a>  </span>
<span id="cb618-3"><a href="descriptive-cohort-analyses.html#cb618-3" tabindex="-1"></a>  <span class="co"># Count number of participants for each number of days</span></span>
<span id="cb618-4"><a href="descriptive-cohort-analyses.html#cb618-4" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">count</span>(Ndays) <span class="sc">%&gt;%</span></span>
<span id="cb618-5"><a href="descriptive-cohort-analyses.html#cb618-5" tabindex="-1"></a>  </span>
<span id="cb618-6"><a href="descriptive-cohort-analyses.html#cb618-6" tabindex="-1"></a>  <span class="co"># Calculate proportion of partipants for each number of days </span></span>
<span id="cb618-7"><a href="descriptive-cohort-analyses.html#cb618-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">prop =</span> <span class="fu">prop.table</span>(n)) <span class="sc">%&gt;%</span></span>
<span id="cb618-8"><a href="descriptive-cohort-analyses.html#cb618-8" tabindex="-1"></a>  </span>
<span id="cb618-9"><a href="descriptive-cohort-analyses.html#cb618-9" tabindex="-1"></a>  <span class="co"># Arrange the table from highest to lowest number of days</span></span>
<span id="cb618-10"><a href="descriptive-cohort-analyses.html#cb618-10" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="sc">-</span>Ndays) <span class="sc">%&gt;%</span></span>
<span id="cb618-11"><a href="descriptive-cohort-analyses.html#cb618-11" tabindex="-1"></a>  </span>
<span id="cb618-12"><a href="descriptive-cohort-analyses.html#cb618-12" tabindex="-1"></a>  <span class="co"># Round the proportion column to two decimal places</span></span>
<span id="cb618-13"><a href="descriptive-cohort-analyses.html#cb618-13" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(prop, \(x) <span class="fu">round</span>(x, <span class="dv">2</span>)))</span></code></pre></div>
<pre><code>##   Ndays  n prop
## 1     7 83 0.86
## 2     6  6 0.06
## 3     5  3 0.03
## 4     3  1 0.01
## 5     2  2 0.02
## 6     1  2 0.02</code></pre>
<div id="answer-to-environmental-health-question-1-12" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 1<a href="descriptive-cohort-analyses.html#answer-to-environmental-health-question-1-12" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can now answer <strong>Environmental Health Question #1</strong></em>: What proportion of participants wore their wristbands for all seven days?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> 86% of participants wore their wristbands for all seven days.</p>
</div>
<p>Because a few participants did not wear their wristbands for all seven days, it will be important to further explore whether there are outlier participants and to normalize the chemical concentrations by number of days the wristband was worn. We can first assess whether any participants have a particularly low or high number of chemicals detected relative to the other participants.</p>
<p>We’ll prepare the data for graphing by creating a dataframe containing information about how many chemicals were detected per participant.</p>
<div class="sourceCode" id="cb620"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb620-1"><a href="descriptive-cohort-analyses.html#cb620-1" tabindex="-1"></a>wrist_det_by_participant <span class="ot">&lt;-</span> wrist_data <span class="sc">%&gt;%</span></span>
<span id="cb620-2"><a href="descriptive-cohort-analyses.html#cb620-2" tabindex="-1"></a>  </span>
<span id="cb620-3"><a href="descriptive-cohort-analyses.html#cb620-3" tabindex="-1"></a>  <span class="co"># Remove Ndays column because we don&#39;t need it for this step</span></span>
<span id="cb620-4"><a href="descriptive-cohort-analyses.html#cb620-4" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Ndays) <span class="sc">%&gt;%</span></span>
<span id="cb620-5"><a href="descriptive-cohort-analyses.html#cb620-5" tabindex="-1"></a>  </span>
<span id="cb620-6"><a href="descriptive-cohort-analyses.html#cb620-6" tabindex="-1"></a>  <span class="co"># Move S_ID to rownames so it doesn&#39;t interfere with count</span></span>
<span id="cb620-7"><a href="descriptive-cohort-analyses.html#cb620-7" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&quot;S_ID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb620-8"><a href="descriptive-cohort-analyses.html#cb620-8" tabindex="-1"></a>  </span>
<span id="cb620-9"><a href="descriptive-cohort-analyses.html#cb620-9" tabindex="-1"></a>  <span class="co"># Create a new column for number of chemicals detected</span></span>
<span id="cb620-10"><a href="descriptive-cohort-analyses.html#cb620-10" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_det =</span> <span class="fu">rowSums</span>(<span class="sc">!</span><span class="fu">is.na</span>(.))) <span class="sc">%&gt;%</span></span>
<span id="cb620-11"><a href="descriptive-cohort-analyses.html#cb620-11" tabindex="-1"></a>  </span>
<span id="cb620-12"><a href="descriptive-cohort-analyses.html#cb620-12" tabindex="-1"></a>  <span class="co"># Clean dataframe</span></span>
<span id="cb620-13"><a href="descriptive-cohort-analyses.html#cb620-13" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;S_ID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb620-14"><a href="descriptive-cohort-analyses.html#cb620-14" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(S_ID, n_det))</span>
<span id="cb620-15"><a href="descriptive-cohort-analyses.html#cb620-15" tabindex="-1"></a></span>
<span id="cb620-16"><a href="descriptive-cohort-analyses.html#cb620-16" tabindex="-1"></a><span class="fu">datatable</span>(wrist_det_by_participant)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-aabab61e3fa9290229a6" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-aabab61e3fa9290229a6">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97"],["1014","1018","1019","1023","1026","1037","1039","1049","1052","1054","1061","1063","1064","1070","1071","1073","1075","1076","1079","1081","1082","1084","1085","1088","1091","1093","1094","1095","1096","1098","1099","1101","1102","1105","1110","1114","1116","1117","1118","1119","1120","1124","1126","1127","1128","1129","1130","1131","1133","1135","1144","1145","1147","1153","1154","1156","1159","1161","1163","1168","1169","1171","1172","1179","1182","1187","1188","1191","1193","1194","1196","1199","1200","1201","1202","1207","1210","1212","1215","1216","1227","1229","1250","1252","1253","1254","1255","1257","1259","1260","1261","1263","1264","1265","1266","1268","1272"],[56,34,41,39,49,45,48,43,45,48,30,43,56,34,48,47,53,56,53,57,48,61,58,43,47,52,50,43,42,49,50,52,39,55,62,45,48,51,52,44,58,49,54,55,46,48,42,53,41,49,61,66,43,47,50,48,41,32,49,45,43,50,53,34,51,46,47,38,40,56,60,50,51,59,61,44,43,47,42,40,57,61,54,43,39,45,41,49,54,41,52,53,47,53,53,45,44]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>S_ID<\/th>\n      <th>n_det<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"S_ID","targets":1},{"name":"n_det","targets":2}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Then, we can make our histogram:</p>
<div class="sourceCode" id="cb621"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb621-1"><a href="descriptive-cohort-analyses.html#cb621-1" tabindex="-1"></a>det_per_participant_graph <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(wrist_det_by_participant, <span class="fu">aes</span>(<span class="at">x =</span> n_det)) <span class="sc">+</span></span>
<span id="cb621-2"><a href="descriptive-cohort-analyses.html#cb621-2" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, </span>
<span id="cb621-3"><a href="descriptive-cohort-analyses.html#cb621-3" tabindex="-1"></a>                 <span class="at">fill =</span> <span class="st">&quot;gray60&quot;</span>,</span>
<span id="cb621-4"><a href="descriptive-cohort-analyses.html#cb621-4" tabindex="-1"></a>                 <span class="at">alpha =</span> <span class="fl">0.7</span>,</span>
<span id="cb621-5"><a href="descriptive-cohort-analyses.html#cb621-5" tabindex="-1"></a>                 <span class="at">binwidth =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb621-6"><a href="descriptive-cohort-analyses.html#cb621-6" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Distribution of Number of Chemicals Detected Per Participant&quot;</span>) <span class="sc">+</span></span>
<span id="cb621-7"><a href="descriptive-cohort-analyses.html#cb621-7" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Number of Participants&quot;</span>) <span class="sc">+</span></span>
<span id="cb621-8"><a href="descriptive-cohort-analyses.html#cb621-8" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Number of Chemicals Detected&quot;</span>) <span class="sc">+</span></span>
<span id="cb621-9"><a href="descriptive-cohort-analyses.html#cb621-9" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">70</span>, <span class="at">by =</span> <span class="dv">10</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">70</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="fl">0.025</span>, <span class="fl">0.025</span>)) <span class="sc">+</span></span>
<span id="cb621-10"><a href="descriptive-cohort-analyses.html#cb621-10" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">15</span>, <span class="at">by =</span> <span class="dv">5</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">15</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb621-11"><a href="descriptive-cohort-analyses.html#cb621-11" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb621-12"><a href="descriptive-cohort-analyses.html#cb621-12" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> ggplot2<span class="sc">::</span><span class="fu">margin</span>(<span class="at">t =</span> <span class="dv">10</span>), <span class="at">size =</span> <span class="dv">13</span>),</span>
<span id="cb621-13"><a href="descriptive-cohort-analyses.html#cb621-13" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> ggplot2<span class="sc">::</span><span class="fu">margin</span>(<span class="at">r =</span> <span class="dv">10</span>), <span class="at">size =</span> <span class="dv">13</span>),</span>
<span id="cb621-14"><a href="descriptive-cohort-analyses.html#cb621-14" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>))</span>
<span id="cb621-15"><a href="descriptive-cohort-analyses.html#cb621-15" tabindex="-1"></a></span>
<span id="cb621-16"><a href="descriptive-cohort-analyses.html#cb621-16" tabindex="-1"></a>det_per_participant_graph</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-483-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>From this histogram, we can see that the number of chemicals detected per participant ranges from about 30-65 chemicals, with no participants standing out as being well above or below the distribution.</p>
</div>
</div>
<div id="chemical-detection-filtering" class="section level2 hasAnchor">
<h2>Chemical Detection Filtering<a href="descriptive-cohort-analyses.html#chemical-detection-filtering" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Next, we want to apply a chemical detection filter to remove chemicals from the dataset with very low detection. To start, let’s make a dataframe summarizing the percentage of participants in which each chemical was detected and graph this distribution using a histogram.</p>
<div class="sourceCode" id="cb622"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb622-1"><a href="descriptive-cohort-analyses.html#cb622-1" tabindex="-1"></a><span class="co"># Create dataframe where n_detected is the sum of the rows where there are not NA values</span></span>
<span id="cb622-2"><a href="descriptive-cohort-analyses.html#cb622-2" tabindex="-1"></a>chemical_counts <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">n_detected =</span> <span class="fu">colSums</span>(<span class="sc">!</span><span class="fu">is.na</span>(wrist_data <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(S_ID, Ndays))))) <span class="sc">%&gt;%</span></span>
<span id="cb622-3"><a href="descriptive-cohort-analyses.html#cb622-3" tabindex="-1"></a>  </span>
<span id="cb622-4"><a href="descriptive-cohort-analyses.html#cb622-4" tabindex="-1"></a>  <span class="co"># Move rownames to a column</span></span>
<span id="cb622-5"><a href="descriptive-cohort-analyses.html#cb622-5" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;class_chemical&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb622-6"><a href="descriptive-cohort-analyses.html#cb622-6" tabindex="-1"></a>  </span>
<span id="cb622-7"><a href="descriptive-cohort-analyses.html#cb622-7" tabindex="-1"></a>  <span class="co"># Add n_undetected and percentage detected and undetected columns</span></span>
<span id="cb622-8"><a href="descriptive-cohort-analyses.html#cb622-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">n_undetected =</span> <span class="fu">nrow</span>(wrist_data) <span class="sc">-</span> n_detected,</span>
<span id="cb622-9"><a href="descriptive-cohort-analyses.html#cb622-9" tabindex="-1"></a>         <span class="at">perc_detected =</span> n_detected<span class="sc">/</span><span class="fu">nrow</span>(wrist_data)<span class="sc">*</span><span class="dv">100</span>,</span>
<span id="cb622-10"><a href="descriptive-cohort-analyses.html#cb622-10" tabindex="-1"></a>         <span class="at">perc_undetected =</span> n_undetected<span class="sc">/</span><span class="fu">nrow</span>(wrist_data)<span class="sc">*</span><span class="dv">100</span>) <span class="sc">%&gt;%</span></span>
<span id="cb622-11"><a href="descriptive-cohort-analyses.html#cb622-11" tabindex="-1"></a>  </span>
<span id="cb622-12"><a href="descriptive-cohort-analyses.html#cb622-12" tabindex="-1"></a>  <span class="co"># Round percentages to two decimal places</span></span>
<span id="cb622-13"><a href="descriptive-cohort-analyses.html#cb622-13" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(perc_detected, perc_undetected), \(x) <span class="fu">round</span>(x, <span class="dv">2</span>))) </span>
<span id="cb622-14"><a href="descriptive-cohort-analyses.html#cb622-14" tabindex="-1"></a></span>
<span id="cb622-15"><a href="descriptive-cohort-analyses.html#cb622-15" tabindex="-1"></a><span class="co"># View dataframe</span></span>
<span id="cb622-16"><a href="descriptive-cohort-analyses.html#cb622-16" tabindex="-1"></a><span class="fu">datatable</span>(chemical_counts)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-b29c80d1aae649c5fd80" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-b29c80d1aae649c5fd80">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110"],["alkylOPE_2IPPDPP","alkylOPE_3IPPDPP","alkylOPE_2tBPDPP","alkylOPE_4IPPDPP","alkylOPE_B2IPPPP","alkylOPE_24DIPPDPP","alkylOPE_4tBPDPP","alkylOPE_B3IPPPP","alkylOPE_B2tBPPP","alkylOPE_B4IPPPP","alkylOPE_T3IPPP","alkylOPE_B24DIPPPP","alkylOPE_B4tBPPP","alkylOPE_T4IPPP","alkylOPE_T4tBPP","BDE_28","BDE_47","BDE_66","BDE_100","BDE_99","BDE_85","BDE_154","BDE_153","BDE_183","BDE_209","BFR_BEHTBP","BFR_EHTBB","BFR_TBBPA","BFR_BTBPE","BFR_TDBPIC","OPE_TEP","OPE_TiPP","OPE_TPrP","OPE_TiBP","OPE_TnBP","OPE_TCEP","OPE_TCPP1","OPE_TCPP2","OPE_TCPP3","OPE_TPeP","OPE_TDCPP","OPE_TPHP","OPE_EHDPP","OPE_TEHP","OPE_isodecylPP","OPE_ToCP","OPE_TmCP","OPE_TpCP","OPE_TDMPP","PAH_Naphthalene","PAH_Acenaphthylene","PAH_Acenapthene","PAH_Fluorene","PAH_Phenanthrene","PAH_Anthracene","PAH_Fluoranthene","PAH_Pyrene","PAH_Benzocphen","PAH_Benzaanth","PAH_Chrysene","PAH_712Dimethylbenz","PAH_Benzojbkfluor","PAH_Benzoepyrene","PAH_Benzoapyrene","PAH_Perylene","PAH_3Methylchol","PAH_Indeno123cd","PAH_Dibenzahanth","PAH_Benzoghiperylene","PAH_Dibenzalpyrene","PAH_Dibenzaipyrene","PAH_Dibenzahpyrene","PCB_11","PCB_28","PCB_51","PCB_52","PCB_47","PCB_68","PCB_101","PCB_118","PCB_153","PCB_138","PCB_183","Pest_DEET","Pest_Atrazine","Pest_Lindane","Pest_Malathion","Pest_Chlorpyrifos","Pest_Fipronil","Pest_Transchlordane","Pest_CisChlordane","Pest_PPDDE","Pest_Chlorfenapyr","Pest_Trifloxystrobin","Pest_CisPermethrin","Pest_TransPermethrin","Pest_Cypermethrin","Pest_Pyraclostrobin","Pest_Azoxystrobin","Pest_Fluoxastrobin","Phthal_DMP","Phthal_DEP","Phthal_DBP","Phthal_DiBP","Phthal_BBP","Phthal_DEHA","Phthal_DEHP","Phthal_DEHT","Phthal_DINP","Phthal_TOTM"],[97,92,8,97,78,65,97,6,0,87,0,0,96,2,37,11,93,11,34,89,13,22,20,0,71,96,90,0,1,2,37,0,0,97,97,45,97,97,83,6,93,97,97,95,92,93,96,33,28,0,6,6,58,93,6,95,95,4,23,30,4,49,34,7,4,6,29,1,6,0,0,0,78,27,2,13,12,8,12,11,7,7,4,93,0,13,0,46,21,62,41,33,5,0,90,92,23,2,47,0,1,97,97,6,97,97,90,97,97,97],[0,5,89,0,19,32,0,91,97,10,97,97,1,95,60,86,4,86,63,8,84,75,77,97,26,1,7,97,96,95,60,97,97,0,0,52,0,0,14,91,4,0,0,2,5,4,1,64,69,97,91,91,39,4,91,2,2,93,74,67,93,48,63,90,93,91,68,96,91,97,97,97,19,70,95,84,85,89,85,86,90,90,93,4,97,84,97,51,76,35,56,64,92,97,7,5,74,95,50,97,96,0,0,91,0,0,7,0,0,0],[100,94.84999999999999,8.25,100,80.41,67.01000000000001,100,6.19,0,89.69,0,0,98.97,2.06,38.14,11.34,95.88,11.34,35.05,91.75,13.4,22.68,20.62,0,73.2,98.97,92.78,0,1.03,2.06,38.14,0,0,100,100,46.39,100,100,85.56999999999999,6.19,95.88,100,100,97.94,94.84999999999999,95.88,98.97,34.02,28.87,0,6.19,6.19,59.79,95.88,6.19,97.94,97.94,4.12,23.71,30.93,4.12,50.52,35.05,7.22,4.12,6.19,29.9,1.03,6.19,0,0,0,80.41,27.84,2.06,13.4,12.37,8.25,12.37,11.34,7.22,7.22,4.12,95.88,0,13.4,0,47.42,21.65,63.92,42.27,34.02,5.15,0,92.78,94.84999999999999,23.71,2.06,48.45,0,1.03,100,100,6.19,100,100,92.78,100,100,100],[0,5.15,91.75,0,19.59,32.99,0,93.81,100,10.31,100,100,1.03,97.94,61.86,88.66,4.12,88.66,64.95,8.25,86.59999999999999,77.31999999999999,79.38,100,26.8,1.03,7.22,100,98.97,97.94,61.86,100,100,0,0,53.61,0,0,14.43,93.81,4.12,0,0,2.06,5.15,4.12,1.03,65.98,71.13,100,93.81,93.81,40.21,4.12,93.81,2.06,2.06,95.88,76.29000000000001,69.06999999999999,95.88,49.48,64.95,92.78,95.88,93.81,70.09999999999999,98.97,93.81,100,100,100,19.59,72.16,97.94,86.59999999999999,87.63,91.75,87.63,88.66,92.78,92.78,95.88,4.12,100,86.59999999999999,100,52.58,78.34999999999999,36.08,57.73,65.98,94.84999999999999,100,7.22,5.15,76.29000000000001,97.94,51.55,100,98.97,0,0,93.81,0,0,7.22,0,0,0]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>class_chemical<\/th>\n      <th>n_detected<\/th>\n      <th>n_undetected<\/th>\n      <th>perc_detected<\/th>\n      <th>perc_undetected<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"class_chemical","targets":1},{"name":"n_detected","targets":2},{"name":"n_undetected","targets":3},{"name":"perc_detected","targets":4},{"name":"perc_undetected","targets":5}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb623"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb623-1"><a href="descriptive-cohort-analyses.html#cb623-1" tabindex="-1"></a>det_per_chemical_graph <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(chemical_counts, <span class="fu">aes</span>(<span class="at">x =</span> perc_detected)) <span class="sc">+</span></span>
<span id="cb623-2"><a href="descriptive-cohort-analyses.html#cb623-2" tabindex="-1"></a>    <span class="fu">geom_histogram</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb623-3"><a href="descriptive-cohort-analyses.html#cb623-3" tabindex="-1"></a>                   <span class="at">fill =</span> <span class="st">&quot;gray60&quot;</span>,</span>
<span id="cb623-4"><a href="descriptive-cohort-analyses.html#cb623-4" tabindex="-1"></a>                   <span class="at">alpha =</span> <span class="fl">0.7</span>,</span>
<span id="cb623-5"><a href="descriptive-cohort-analyses.html#cb623-5" tabindex="-1"></a>                   <span class="at">binwidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb623-6"><a href="descriptive-cohort-analyses.html#cb623-6" tabindex="-1"></a>    <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">100</span>, <span class="at">by =</span> <span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="fl">0.025</span>, <span class="fl">0.025</span>)) <span class="sc">+</span></span>
<span id="cb623-7"><a href="descriptive-cohort-analyses.html#cb623-7" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">25</span>, <span class="at">by =</span> <span class="dv">5</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">25</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb623-8"><a href="descriptive-cohort-analyses.html#cb623-8" tabindex="-1"></a>    <span class="fu">ggtitle</span>(<span class="st">&quot;Distribution of Percentage Chemical Detection&quot;</span>) <span class="sc">+</span></span>
<span id="cb623-9"><a href="descriptive-cohort-analyses.html#cb623-9" tabindex="-1"></a>    <span class="fu">ylab</span>(<span class="st">&quot;Number of Chemicals&quot;</span>) <span class="sc">+</span></span>
<span id="cb623-10"><a href="descriptive-cohort-analyses.html#cb623-10" tabindex="-1"></a>    <span class="fu">xlab</span>(<span class="st">&quot;Percentage of Detection Across All Participants&quot;</span>) <span class="sc">+</span></span>
<span id="cb623-11"><a href="descriptive-cohort-analyses.html#cb623-11" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb623-12"><a href="descriptive-cohort-analyses.html#cb623-12" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> ggplot2<span class="sc">::</span><span class="fu">margin</span>(<span class="at">t =</span> <span class="dv">10</span>)),</span>
<span id="cb623-13"><a href="descriptive-cohort-analyses.html#cb623-13" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> ggplot2<span class="sc">::</span><span class="fu">margin</span>(<span class="at">r =</span> <span class="dv">10</span>)))</span>
<span id="cb623-14"><a href="descriptive-cohort-analyses.html#cb623-14" tabindex="-1"></a></span>
<span id="cb623-15"><a href="descriptive-cohort-analyses.html#cb623-15" tabindex="-1"></a>det_per_chemical_graph</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-485-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>From this histogram, we can see that many of the chemicals fall in the &lt; 15% or &gt; 90% detection range, with the others distributed evenly between 20 and 90% detection. How we choose to filter our data in part depends on the goals of our analysis. For example, if we only want to keep chemicals detected for almost all of the participants, we could set our threshold at 90% detection:</p>
<div class="sourceCode" id="cb624"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb624-1"><a href="descriptive-cohort-analyses.html#cb624-1" tabindex="-1"></a><span class="co"># Add annotation column</span></span>
<span id="cb624-2"><a href="descriptive-cohort-analyses.html#cb624-2" tabindex="-1"></a>chemical_counts <span class="ot">&lt;-</span> chemical_counts <span class="sc">%&gt;%</span></span>
<span id="cb624-3"><a href="descriptive-cohort-analyses.html#cb624-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">det_filter_90 =</span> <span class="fu">ifelse</span>(perc_detected <span class="sc">&gt;</span> <span class="dv">90</span>, <span class="st">&quot;Yes&quot;</span>, <span class="st">&quot;No&quot;</span>))</span>
<span id="cb624-4"><a href="descriptive-cohort-analyses.html#cb624-4" tabindex="-1"></a></span>
<span id="cb624-5"><a href="descriptive-cohort-analyses.html#cb624-5" tabindex="-1"></a><span class="co"># How many chemicals pass this filter?</span></span>
<span id="cb624-6"><a href="descriptive-cohort-analyses.html#cb624-6" tabindex="-1"></a><span class="fu">nrow</span>(chemical_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(det_filter_90 <span class="sc">==</span> <span class="st">&quot;Yes&quot;</span>))</span></code></pre></div>
<pre><code>## [1] 34</code></pre>
<div class="sourceCode" id="cb626"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb626-1"><a href="descriptive-cohort-analyses.html#cb626-1" tabindex="-1"></a><span class="co"># Make graph</span></span>
<span id="cb626-2"><a href="descriptive-cohort-analyses.html#cb626-2" tabindex="-1"></a>det_per_chemical_graph_90 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(chemical_counts, <span class="fu">aes</span>(<span class="at">x =</span> perc_detected, <span class="at">fill =</span> det_filter_90)) <span class="sc">+</span></span>
<span id="cb626-3"><a href="descriptive-cohort-analyses.html#cb626-3" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb626-4"><a href="descriptive-cohort-analyses.html#cb626-4" tabindex="-1"></a>                 <span class="at">alpha =</span> <span class="fl">0.7</span>,</span>
<span id="cb626-5"><a href="descriptive-cohort-analyses.html#cb626-5" tabindex="-1"></a>                 <span class="at">binwidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb626-6"><a href="descriptive-cohort-analyses.html#cb626-6" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;gray87&quot;</span>, <span class="st">&quot;gray32&quot;</span>), <span class="at">guide =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span></span>
<span id="cb626-7"><a href="descriptive-cohort-analyses.html#cb626-7" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">90</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">xend =</span> <span class="dv">90</span>, <span class="at">yend =</span> <span class="dv">25</span>), <span class="at">color =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="at">linetype =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb626-8"><a href="descriptive-cohort-analyses.html#cb626-8" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">100</span>, <span class="at">by =</span> <span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="fl">0.025</span>, <span class="fl">0.025</span>)) <span class="sc">+</span></span>
<span id="cb626-9"><a href="descriptive-cohort-analyses.html#cb626-9" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">25</span>, <span class="at">by =</span> <span class="dv">5</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">25</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb626-10"><a href="descriptive-cohort-analyses.html#cb626-10" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Distribution of Percentage Chemical Detection&quot;</span>) <span class="sc">+</span></span>
<span id="cb626-11"><a href="descriptive-cohort-analyses.html#cb626-11" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Number of Chemicals&quot;</span>) <span class="sc">+</span></span>
<span id="cb626-12"><a href="descriptive-cohort-analyses.html#cb626-12" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Percentage of Detection Across All Participants&quot;</span>) <span class="sc">+</span></span>
<span id="cb626-13"><a href="descriptive-cohort-analyses.html#cb626-13" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb626-14"><a href="descriptive-cohort-analyses.html#cb626-14" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> ggplot2<span class="sc">::</span><span class="fu">margin</span>(<span class="at">t =</span> <span class="dv">10</span>), <span class="at">size =</span> <span class="dv">13</span>),</span>
<span id="cb626-15"><a href="descriptive-cohort-analyses.html#cb626-15" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> ggplot2<span class="sc">::</span><span class="fu">margin</span>(<span class="at">r =</span> <span class="dv">10</span>), <span class="at">size =</span> <span class="dv">13</span>),</span>
<span id="cb626-16"><a href="descriptive-cohort-analyses.html#cb626-16" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>))</span>
<span id="cb626-17"><a href="descriptive-cohort-analyses.html#cb626-17" tabindex="-1"></a></span>
<span id="cb626-18"><a href="descriptive-cohort-analyses.html#cb626-18" tabindex="-1"></a>det_per_chemical_graph_90</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-486-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>However, this only keeps 34 chemicals in our dataset, which is a significant proportion of all of the chemicals measured. We could also consider setting the filter at 20% detection to maximize inclusion of as many chemicals as possible.</p>
<div class="sourceCode" id="cb627"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb627-1"><a href="descriptive-cohort-analyses.html#cb627-1" tabindex="-1"></a><span class="co"># Add annotation column</span></span>
<span id="cb627-2"><a href="descriptive-cohort-analyses.html#cb627-2" tabindex="-1"></a>chemical_counts <span class="ot">&lt;-</span> chemical_counts <span class="sc">%&gt;%</span></span>
<span id="cb627-3"><a href="descriptive-cohort-analyses.html#cb627-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">det_filter_20 =</span> <span class="fu">ifelse</span>(perc_detected <span class="sc">&gt;</span> <span class="dv">20</span>, <span class="st">&quot;Yes&quot;</span>, <span class="st">&quot;No&quot;</span>))</span>
<span id="cb627-4"><a href="descriptive-cohort-analyses.html#cb627-4" tabindex="-1"></a></span>
<span id="cb627-5"><a href="descriptive-cohort-analyses.html#cb627-5" tabindex="-1"></a><span class="co"># How many chemicals pass this filter?</span></span>
<span id="cb627-6"><a href="descriptive-cohort-analyses.html#cb627-6" tabindex="-1"></a><span class="fu">nrow</span>(chemical_counts <span class="sc">%&gt;%</span> <span class="fu">filter</span>(det_filter_20 <span class="sc">==</span> <span class="st">&quot;Yes&quot;</span>))</span></code></pre></div>
<pre><code>## [1] 62</code></pre>
<div class="sourceCode" id="cb629"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb629-1"><a href="descriptive-cohort-analyses.html#cb629-1" tabindex="-1"></a><span class="co"># Make graph</span></span>
<span id="cb629-2"><a href="descriptive-cohort-analyses.html#cb629-2" tabindex="-1"></a>det_per_chemical_graph_20 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(chemical_counts, <span class="fu">aes</span>(<span class="at">x =</span> perc_detected, <span class="at">fill =</span> det_filter_20)) <span class="sc">+</span></span>
<span id="cb629-3"><a href="descriptive-cohort-analyses.html#cb629-3" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb629-4"><a href="descriptive-cohort-analyses.html#cb629-4" tabindex="-1"></a>                 <span class="at">alpha =</span> <span class="fl">0.7</span>,</span>
<span id="cb629-5"><a href="descriptive-cohort-analyses.html#cb629-5" tabindex="-1"></a>                 <span class="at">binwidth =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb629-6"><a href="descriptive-cohort-analyses.html#cb629-6" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;gray87&quot;</span>, <span class="st">&quot;gray32&quot;</span>), <span class="at">guide =</span> <span class="st">&quot;none&quot;</span>) <span class="sc">+</span></span>
<span id="cb629-7"><a href="descriptive-cohort-analyses.html#cb629-7" tabindex="-1"></a>  <span class="fu">geom_segment</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">20</span>, <span class="at">y =</span> <span class="dv">0</span>, <span class="at">xend =</span> <span class="dv">20</span>, <span class="at">yend =</span> <span class="dv">25</span>), <span class="at">color =</span> <span class="st">&quot;firebrick&quot;</span>, <span class="at">linetype =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb629-8"><a href="descriptive-cohort-analyses.html#cb629-8" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">100</span>, <span class="at">by =</span> <span class="dv">10</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="fl">0.025</span>, <span class="fl">0.025</span>)) <span class="sc">+</span></span>
<span id="cb629-9"><a href="descriptive-cohort-analyses.html#cb629-9" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">25</span>, <span class="at">by =</span> <span class="dv">5</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">25</span>), <span class="at">expand =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb629-10"><a href="descriptive-cohort-analyses.html#cb629-10" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Distribution of Percentage Chemical Detection&quot;</span>) <span class="sc">+</span></span>
<span id="cb629-11"><a href="descriptive-cohort-analyses.html#cb629-11" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Number of Chemicals&quot;</span>) <span class="sc">+</span></span>
<span id="cb629-12"><a href="descriptive-cohort-analyses.html#cb629-12" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Percentage of Detection Across All Participants&quot;</span>) <span class="sc">+</span></span>
<span id="cb629-13"><a href="descriptive-cohort-analyses.html#cb629-13" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">16</span>),</span>
<span id="cb629-14"><a href="descriptive-cohort-analyses.html#cb629-14" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> ggplot2<span class="sc">::</span><span class="fu">margin</span>(<span class="at">t =</span> <span class="dv">10</span>), <span class="at">size =</span> <span class="dv">13</span>),</span>
<span id="cb629-15"><a href="descriptive-cohort-analyses.html#cb629-15" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> ggplot2<span class="sc">::</span><span class="fu">margin</span>(<span class="at">r =</span> <span class="dv">10</span>), <span class="at">size =</span> <span class="dv">13</span>),</span>
<span id="cb629-16"><a href="descriptive-cohort-analyses.html#cb629-16" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>))</span>
<span id="cb629-17"><a href="descriptive-cohort-analyses.html#cb629-17" tabindex="-1"></a></span>
<span id="cb629-18"><a href="descriptive-cohort-analyses.html#cb629-18" tabindex="-1"></a>det_per_chemical_graph_20</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-487-1.png" width="672" style="display: block; margin: auto;" /></p>
<div id="answer-to-environmental-health-question-2-13" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 2<a href="descriptive-cohort-analyses.html#answer-to-environmental-health-question-2-13" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can now answer <strong>Environmental Health Question #2</strong></em>: How many chemicals were detected in at least 20% of participants?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> 62 chemicals were detected in at least 20% of participants.</p>
</div>
<p>We’ll use the 20% detection filter for downstream analyses to maximize inclusion of data for our study. Note that selection of data filters is highly project- and goal- dependent, so be sure to take into consideration typical workflows for your type of data, study, or lab group.</p>
<div class="sourceCode" id="cb630"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb630-1"><a href="descriptive-cohort-analyses.html#cb630-1" tabindex="-1"></a><span class="co"># Create vector of chemicals to keep</span></span>
<span id="cb630-2"><a href="descriptive-cohort-analyses.html#cb630-2" tabindex="-1"></a>chemicals_20perc <span class="ot">&lt;-</span> chemical_counts <span class="sc">%&gt;%</span></span>
<span id="cb630-3"><a href="descriptive-cohort-analyses.html#cb630-3" tabindex="-1"></a>  <span class="fu">filter</span>(perc_detected <span class="sc">&gt;</span> <span class="dv">20</span>) <span class="sc">%&gt;%</span></span>
<span id="cb630-4"><a href="descriptive-cohort-analyses.html#cb630-4" tabindex="-1"></a>  <span class="fu">pull</span>(class_chemical)</span>
<span id="cb630-5"><a href="descriptive-cohort-analyses.html#cb630-5" tabindex="-1"></a></span>
<span id="cb630-6"><a href="descriptive-cohort-analyses.html#cb630-6" tabindex="-1"></a><span class="co"># Filter dataframe</span></span>
<span id="cb630-7"><a href="descriptive-cohort-analyses.html#cb630-7" tabindex="-1"></a>wrist_data_filtered <span class="ot">&lt;-</span> wrist_data <span class="sc">%&gt;%</span></span>
<span id="cb630-8"><a href="descriptive-cohort-analyses.html#cb630-8" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&quot;S_ID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb630-9"><a href="descriptive-cohort-analyses.html#cb630-9" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(<span class="fu">all_of</span>(chemicals_20perc))</span></code></pre></div>
<p>We can also summarize chemical detection vs. non-detection by chemical class to understand the number of chemicals in each class that were 1) detected in any participant or 2) detected in more than 20% of participants.</p>
<div class="sourceCode" id="cb631"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb631-1"><a href="descriptive-cohort-analyses.html#cb631-1" tabindex="-1"></a>chemical_count_byclass <span class="ot">&lt;-</span> chemical_counts <span class="sc">%&gt;%</span></span>
<span id="cb631-2"><a href="descriptive-cohort-analyses.html#cb631-2" tabindex="-1"></a>  <span class="fu">separate</span>(class_chemical, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&quot;class&quot;</span>, <span class="cn">NA</span>), <span class="at">remove =</span> <span class="cn">FALSE</span>, <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb631-3"><a href="descriptive-cohort-analyses.html#cb631-3" tabindex="-1"></a>  <span class="fu">group_by</span>(class) <span class="sc">%&gt;%</span></span>
<span id="cb631-4"><a href="descriptive-cohort-analyses.html#cb631-4" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_chemicals =</span> <span class="fu">n</span>(), </span>
<span id="cb631-5"><a href="descriptive-cohort-analyses.html#cb631-5" tabindex="-1"></a>            <span class="at">n_chemicals_det =</span> <span class="fu">sum</span>(n_detected <span class="sc">&gt;</span> <span class="dv">0</span>), </span>
<span id="cb631-6"><a href="descriptive-cohort-analyses.html#cb631-6" tabindex="-1"></a>            <span class="at">n_chemicals_det_20perc =</span> <span class="fu">sum</span>(perc_detected <span class="sc">&gt;=</span> <span class="dv">20</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb631-7"><a href="descriptive-cohort-analyses.html#cb631-7" tabindex="-1"></a>  <span class="fu">bind_rows</span>(<span class="fu">summarise</span>(., <span class="fu">across</span>(<span class="fu">where</span>(is.numeric), sum),</span>
<span id="cb631-8"><a href="descriptive-cohort-analyses.html#cb631-8" tabindex="-1"></a>                         <span class="fu">across</span>(<span class="fu">where</span>(is.character), <span class="sc">~</span><span class="st">&#39;Total&#39;</span>)))</span>
<span id="cb631-9"><a href="descriptive-cohort-analyses.html#cb631-9" tabindex="-1"></a></span>
<span id="cb631-10"><a href="descriptive-cohort-analyses.html#cb631-10" tabindex="-1"></a><span class="fu">datatable</span>(chemical_count_byclass)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-79973bd0be4897d8b76c" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-79973bd0be4897d8b76c">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9"],["BDE","BFR","OPE","PAH","PCB","Pest","Phthal","alkylOPE","Total"],[10,5,19,23,11,17,10,15,110],[9,4,17,19,11,13,10,12,95],[6,2,16,9,2,10,8,9,62]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>class<\/th>\n      <th>n_chemicals<\/th>\n      <th>n_chemicals_det<\/th>\n      <th>n_chemicals_det_20perc<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"class","targets":1},{"name":"n_chemicals","targets":2},{"name":"n_chemicals_det","targets":3},{"name":"n_chemicals_det_20perc","targets":4}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>From these data, we can see that, of the 62 chemicals retained by our detection filter, some classes were retained more than others. For example, of the 8 of the 10 phthalates (80%) were retained by the 20% detection filter, while only 2 of the 11 PCBs (18%) were retained.</p>
</div>
</div>
<div id="outlier-identification" class="section level2 hasAnchor">
<h2>Outlier Identification<a href="descriptive-cohort-analyses.html#outlier-identification" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Next, we will check to see if any participants are outliers based on the entire chemical signature for each participant using principal component analysis (PCA). Prior to checking for outliers, a few final data cleaning steps are required, which are beyond the scope of this specific module, though we encourage participants to research these methods as they are important in general data pre-processing. These data cleaning steps were:</p>
<ol style="list-style-type: decimal">
<li>Imputing missing values.</li>
<li>Calculating time-weighted average values by dividing each value by the number of days the participant wore the wristband.</li>
<li>Assessing normality of data with and without log2 transformation.</li>
</ol>
<p>Here, we’ll read in the fully cleaned and processed data, which contains data for all 97 participants and the 62 chemicals that passed the detection filter (imputed, time-weighted). We will also apply log2 transformation to move the data closer to a normal distribution. For more on these steps, see <strong>TAME 2.0 Module 3.3 Normality Tests and Data Transformations</strong> and <strong>TAME 2.0 Module 4.2 Data Import, Processing, and Summary Statistics</strong>.</p>
<div class="sourceCode" id="cb632"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb632-1"><a href="descriptive-cohort-analyses.html#cb632-1" tabindex="-1"></a>wrist_data_cleaned <span class="ot">&lt;-</span> <span class="fu">read.xlsx</span>(<span class="st">&quot;Module6_1_Input/Module6_1_InputData2.xlsx&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb632-2"><a href="descriptive-cohort-analyses.html#cb632-2" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&quot;S_ID&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb632-3"><a href="descriptive-cohort-analyses.html#cb632-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), \(x) <span class="fu">log2</span>(x<span class="sc">+</span><span class="dv">1</span>)))</span>
<span id="cb632-4"><a href="descriptive-cohort-analyses.html#cb632-4" tabindex="-1"></a></span>
<span id="cb632-5"><a href="descriptive-cohort-analyses.html#cb632-5" tabindex="-1"></a><span class="fu">datatable</span>(wrist_data_cleaned[ <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>])</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-7150d420a1599ab7f8d6" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-7150d420a1599ab7f8d6">{"x":{"filter":"none","vertical":false,"data":[["1014","1018","1019","1023","1026","1037","1039","1049","1052","1054","1061","1063","1064","1070","1071","1073","1075","1076","1079","1081","1082","1084","1085","1088","1091","1093","1094","1095","1096","1098","1099","1101","1102","1105","1110","1114","1116","1117","1118","1119","1120","1124","1126","1127","1128","1129","1130","1131","1133","1135","1144","1145","1147","1153","1154","1156","1159","1161","1163","1168","1169","1171","1172","1179","1182","1187","1188","1191","1193","1194","1196","1199","1200","1201","1202","1207","1210","1212","1215","1216","1227","1229","1250","1252","1253","1254","1255","1257","1259","1260","1261","1263","1264","1265","1266","1268","1272"],[4.333423733725192,4.967234494548414,5.299077155759326,5.318213529720423,4.866201501932541,4.959968823368246,3.269460674993226,2.762804379578516,4.924167769185792,5.592585176214552,3.952644190774131,4.076632386047865,4.490661808712741,6.91071553253829,5.82781902461732,3.403072908343881,4.970393537914677,6.51161713005978,5.686407145841338,4.426360611653126,4.566293289277088,5.031092670490856,7.897179908573977,3.387204963965008,6.782988851373995,4.631852632883677,6.263410058756341,4.709605737841219,4.676622041178652,9.542406387928938,5.014801602351361,4.458307002464283,5.37056287013824,4.917909073852982,3.941911161757199,3.151951567702053,6.904146867663091,3.049879658442183,7.240028051916184,5.76390473221745,7.179866448152017,6.996230547528564,3.491120112678245,6.570853217797227,6.289781780347556,4.207502459258789,2.670322405507702,3.140778655782796,1.659924558402378,4.070389327891398,6.442066715247792,5.214457901527233,4.217230716220669,4.607626220845487,3.226068079479846,4.030588319833419,2.672802467850232,6.961077551395372,6.149706479635361,4.873883489720202,2.759155833800272,3.832166677453225,3.317800209264686,7.880790459984297,2.806176730860324,6.576068579088089,3.429711598754215,2.828399469220156,4.475826056887968,9.500244366588815,5.748307877166432,3.545615416188335,2.440066293770367,3.863655319220852,2.685780000247902,5.036126503374854,5.130224877622052,3.985370311248119,3.865211853671256,3.241404389862252,4.991084476770187,4.663059923747817,3.602545110729821,2.904690526496089,5.211957803593782,3.586335842129853,1.486898214386908,3.948600847493356,4.5377397957693,6.834859268043641,8.316527213510851,6.135804546631864,2.649451227172871,6.674424331914032,3.306387243991584,6.127869072002429,1.67484800423835],[0.9140177374192157,3.228049047884462,1.117695042669755,1.150559676575381,2.147306698780294,2.581179057034193,1.09857355611553,2.246190895670381,2.945126082398769,3.038386010556979,1.010779838753243,3.109599356895612,1.908538448489999,2.693447130999554,3.724806162925611,1.736140961368167,2.928438005927705,4.334241356326216,3.56539407948718,1.857696692231609,2.226068079479847,2.178145508247282,5.083091770622303,0.9140177374192157,3.011035522412022,2.018939323442029,3.499527024215096,2.127633279725874,1.56420394055436,4.609301882270846,0.8264637450053084,2.189033824390017,2.60847133541531,2.272876349961658,1.419153607751073,1.82781902461732,3.185866545311334,1.207892851641333,4.540753536813029,2.909910222902461,4.328200468367855,3.995303605335244,1.586335842129848,3.91715893106235,3.888360566552368,2.329123596291566,1.391139231581478,0.9378824100721453,0.006882200358075422,2.466160967644511,2.657640005207824,3.028316557424019,1.007195501404204,1.826657434344513,1.511684893504931,2.313660478903761,1.728283351066404,1.050626073069967,3.498761465671852,1.847424294901913,0.01224180899302995,2.039639764907969,1.689618658940674,4.482848283068465,1.921653948280259,4.541638734436141,0.008962574093096582,0.0023666677366516,1.011291160764938,5.422905742612183,2.88975165241937,0.9140177374192157,1.01435529297707,0.002001010351719197,0.8639384504239714,2.423001822998545,2.962416817191843,1.395062799517578,1.992250500395066,1.132811828425211,1.821418673144039,1.734077314841381,1.10337773984531,1.311170926878063,4.693128479415509,1.208784780597648,0.08211862047350493,2.645833387490622,2.520332442118444,2.169925001442312,4.031848866039343,3.563158130402808,1.454927884363051,4.223863808553117,0.9270837440795626,2.225627494641826,0.03260466543192769],[2.655024649783419,3.885853226852408,3.745083633195931,3.59608324095067,3.238350142453197,3.787790645933257,1.799087306074004,1.850856560694192,4.279683534709347,3.989009216082021,2.39780296186249,2.529999376439367,3.166944935946129,5.277984747299765,4.809561440521202,1.793152723288183,3.559491813205038,5.555114966692607,4.384970773976697,2.959505221771941,3.371957067152412,3.493317947366444,6.559164009917883,2.096165451665774,5.177006568391107,3.349486602530797,4.927015997605372,3.135394148771282,3.342798533188643,8.035953756557801,2.102658131363739,2.917022499548612,4.130107178658833,3.465974464504069,2.873813198359087,2.211457263685212,5.323225936483009,1.784803080067758,6.21717531212147,4.20006486151431,5.895129476458229,5.343549952738881,2.638570376521669,5.331531204815312,4.748461233004035,2.985500430304885,1.456430691833196,1.526784256780632,1.06645827630148,2.63989364654417,5.02471226823016,3.782708096754644,2.91456452349394,3.489653027217239,1.661879871610054,2.573582272870219,0.7796099315747002,4.475270212134068,4.889376191403106,3.297610682782375,1.571156701196125,2.469141743582753,1.877463815495617,6.538183388012403,1.688340240566464,4.839455765224629,2.168550351466096,0.903038270112912,3.555991398917418,7.901711653889882,4.39457293667599,2.178145508247282,1.384970773976701,3.564552375537553,1.395062799517578,3.754123969443399,3.485610832660054,3.106970408027699,2.16165738545871,1.875780063068488,3.38759887387623,3.057739071112077,2.162578352640252,1.875780063068488,2.915384314239109,1.982895816777625,0.6703224055077025,2.553711566736716,3.196022438144717,5.409148446005036,6.533296355516261,4.803005566043158,1.562111561593117,5.266733007418178,1.678071905112638,4.054972478460384,0.9238283195145959],[2.137971855207022,1.281804209853633,2.892529806639872,1.670969789427422,2.003088181787571,3.521768674233964,0.5959128002816967,0.07204611716483361,3.393298514309864,3.460180876011132,0.1528875099196189,0.4691417435827519,2.465414810378906,3.549317122154497,3.385036534855572,0.1403277635932639,0.09900570750520225,4.516915649523936,2.961005868384137,2.073330603203521,2.026041212105623,3.256579383696495,5.941642928040543,0.1151489503301989,4.98146127243798,2.866484133932834,3.760525119995122,1.86280559206903,0.383259938151893,7.612647377658316,1.231637205813754,2.622261042144133,2.339137384919585,2.316559536944375,2.155194100865454,0.8295596582982737,4.71699089440494,0.1323751589636077,4.969341291861093,3.108643929681297,5.116685562860911,4.481428116910582,1.660576623968682,5.167231748344618,3.372355161813019,0.400537338539677,0.06681030498938294,0.05900492384945259,0.04354007229415607,1.296142717645735,3.509876778165006,3.292781749227846,0.1181967131021408,2.304928186540599,0.3285082433110756,0.6806458487764642,0.1286012747946639,2.715579638885852,3.706420039301946,0.9216539482802588,0.06695866951095436,1.973480079680626,0.4360095036360563,4.8827129118369,0.4780472968046444,1.582556003014061,0.09311353760090584,0.0726305388652612,2.271169523331474,7.056041548931431,4.0724733504907,1.228268987673115,0.7166251458820304,2.483953937771365,0.2733027412876216,1.802636373155075,0.04269410493667448,1.818500009310454,0.8155754288625726,1.084064264788475,2.108165978694354,1.507583240822408,1.057574050232183,0.6494512271728702,0.02874765125385317,0.8774638154956184,0.01605227379358606,1.268177709109754,2.070879957054735,4.849713379143971,4.898899331515598,4.01371746855318,0.3202783576682702,4.248469778077257,0.846851455886688,3.224525443529577,0.01936823751925925],[1.961887873082936,0.4284649179112969,2.908812907739547,0.4636492965206003,3.02629420553738,3.278409631720709,0.879705766282288,0.4330545760488143,2.848855398124996,3.309300939525017,0.07895778418574996,1.263892897403838,2.891974603615552,4.907027988593805,4.028316557424022,0.460966295587226,0.4048441698620299,4.970787932406166,4.303415462685624,2.807354922057604,2.883341521920094,3.241186443636156,6.473624583058259,0.1057725441850644,4.234632699593756,3.060294483960153,5.124977957728819,2.974004791467056,3.043894398175456,6.960851462107869,0.975752453753323,1.26045586173202,3.500255750100469,3.380294073252997,2.479526225730558,0.3975866804610616,4.655695286545587,0.4543947703513586,5.921721947009604,4.039138393906958,5.5962110582395,0.7749291753023114,5.238732277174424,4.842547897835537,4.439053165188782,1.213989749212523,0.2060735974773519,0.1442177496973149,0.01768079694478184,2.493866883275487,3.649286976584562,3.328918507792978,0.5245759302774816,2.833902076669163,0.3955196490940158,3.044394119358453,0.3432156581917792,0.0643915819005666,4.670387156973834,2.717461006299902,0.1729097941931751,1.791962871640621,0.5877078774599738,6.482848283068465,0.4824457268130684,4.068670810665099,0.1068209143103466,0.0254910199281306,2.750606504835592,6.871597265613853,3.92328503385919,1.037633234625005,0.4883881613019422,1.713067326468842,0.3077572766000982,0.4891440782484848,0.3520938042244764,2.740156601586612,0.4890986086494411,0.419949819700347,1.951800911742668,2.167250867143988,1.402878068038243,0.8537105577493435,0.2173043812829895,1.646820971128197,0.0384301343794547,1.776404831581897,2.065473837477283,4.239878074404202,6.143638309922037,4.624268037655686,0.4678634993368116,5.26566192452189,0.9475325801058639,3.116507346018427,0.05569436316028197],[3.369366770823444,3.309300939525017,3.150559676575381,2.384444579007471,2.980808734980162,1.711809527748466,3.068916438492522,1.612857968553181,2.341579182468736,3.47323028050687,2.3622360835409,2.57738566523478,2.467652125442265,1.806176730860324,4.200513321275737,2.94673086014031,4.729785807747392,4.265822637658565,6.302623693355712,3.327276749390043,2.922197848396367,2.694084223101268,4.611425343902314,3.421078998455761,5.374791097575609,2.69980542705992,6.534697516262709,3.165107985144537,5.727375938594897,3.518325307690862,2.379436131482027,3.39741182859701,3.394082907446429,3.474714153140532,3.040892430646901,4.075165859831739,4.977463111239467,4.15704371014558,6.066335258178118,4.470071974475053,7.648999493050686,4.539655136920405,1.900280297487691,7.150864267527825,8.341769773965783,4.358958825832329,2.970327795015949,5.091276694365222,1.008220506804967,2.875218375289975,2.648794112590391,6.368419155081396,2.726831217032493,2.673556423990145,3.629606415776003,4.906753189537567,3.070389327891398,3.443985460731011,2.156396616597665,2.673233348176257,7.810305379462709,3.486898214386914,2.839383776250243,4.603477988254004,4.197258825229056,4.274261661257048,6.216787423840199,3.853995647176393,0.9802864924257221,6.103287808412022,4.672667791352939,3.135159583281637,0.6492322222326555,2.644515556806718,3.559316994054152,5.425689479207976,5.048074227167411,3.892252231835571,1.478047296804647,1.056583528366368,9.215283370934314,1.875124739395675,3.790623114502805,4.995678329643876,6.31342726581369,1.648136698570862,0.5501970825604793,4.57089657647729,3.555640889916442,2.275432809651665,2.49788604389688,3.828834649468056,2.903589232518267,6.067595696940041,3.620251250724301,5.903107151921556,2.192645077942395]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>alkylOPE_2IPPDPP<\/th>\n      <th>alkylOPE_3IPPDPP<\/th>\n      <th>alkylOPE_4IPPDPP<\/th>\n      <th>alkylOPE_B2IPPPP<\/th>\n      <th>alkylOPE_24DIPPDPP<\/th>\n      <th>alkylOPE_4tBPDPP<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"alkylOPE_2IPPDPP","targets":1},{"name":"alkylOPE_3IPPDPP","targets":2},{"name":"alkylOPE_4IPPDPP","targets":3},{"name":"alkylOPE_B2IPPPP","targets":4},{"name":"alkylOPE_24DIPPDPP","targets":5},{"name":"alkylOPE_4tBPDPP","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>First, let’s run PCA and plot our data.</p>
<div class="sourceCode" id="cb633"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb633-1"><a href="descriptive-cohort-analyses.html#cb633-1" tabindex="-1"></a><span class="co"># Prepare dataframe</span></span>
<span id="cb633-2"><a href="descriptive-cohort-analyses.html#cb633-2" tabindex="-1"></a>wrist_data_cleaned_scaled <span class="ot">&lt;-</span> wrist_data_cleaned <span class="sc">%&gt;%</span></span>
<span id="cb633-3"><a href="descriptive-cohort-analyses.html#cb633-3" tabindex="-1"></a>  <span class="fu">scale</span>() <span class="sc">%&gt;%</span> <span class="fu">data.frame</span>()</span>
<span id="cb633-4"><a href="descriptive-cohort-analyses.html#cb633-4" tabindex="-1"></a></span>
<span id="cb633-5"><a href="descriptive-cohort-analyses.html#cb633-5" tabindex="-1"></a><span class="co"># Run PCA</span></span>
<span id="cb633-6"><a href="descriptive-cohort-analyses.html#cb633-6" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(wrist_data_cleaned_scaled)</span>
<span id="cb633-7"><a href="descriptive-cohort-analyses.html#cb633-7" tabindex="-1"></a></span>
<span id="cb633-8"><a href="descriptive-cohort-analyses.html#cb633-8" tabindex="-1"></a><span class="co"># Visualize PCA</span></span>
<span id="cb633-9"><a href="descriptive-cohort-analyses.html#cb633-9" tabindex="-1"></a>pca_chemplot <span class="ot">&lt;-</span> <span class="fu">fviz_pca_ind</span>(pca, </span>
<span id="cb633-10"><a href="descriptive-cohort-analyses.html#cb633-10" tabindex="-1"></a>             <span class="at">label =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb633-11"><a href="descriptive-cohort-analyses.html#cb633-11" tabindex="-1"></a>             <span class="at">pointsize =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb633-12"><a href="descriptive-cohort-analyses.html#cb633-12" tabindex="-1"></a><span class="fu">theme</span>(<span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">&quot;bold&quot;</span>, <span class="at">size =</span> <span class="fu">rel</span>(<span class="fl">1.1</span>)),</span>
<span id="cb633-13"><a href="descriptive-cohort-analyses.html#cb633-13" tabindex="-1"></a>      <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">linewidth =</span> <span class="fl">0.3</span>),</span>
<span id="cb633-14"><a href="descriptive-cohort-analyses.html#cb633-14" tabindex="-1"></a>      <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb633-15"><a href="descriptive-cohort-analyses.html#cb633-15" tabindex="-1"></a>      <span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb633-16"><a href="descriptive-cohort-analyses.html#cb633-16" tabindex="-1"></a>      <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb633-17"><a href="descriptive-cohort-analyses.html#cb633-17" tabindex="-1"></a>      <span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb633-18"><a href="descriptive-cohort-analyses.html#cb633-18" tabindex="-1"></a></span>
<span id="cb633-19"><a href="descriptive-cohort-analyses.html#cb633-19" tabindex="-1"></a>pca_chemplot</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-491-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>By visual inspection, it looks like there may be some outliers, so we can use a formula to detect outliers. One standard way to detect outliers is the criterion of being “more than 6 standard deviations away from the mean” (<a href="https://privefl.github.io/blog/detecting-outlier-samples-in-pca/">Source</a>).</p>
<p>We can apply this approach to our data by first creating a function to detect PCA outliers based on whether or not that participant passed a certain standard deviation cutoff.</p>
<div class="sourceCode" id="cb634"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb634-1"><a href="descriptive-cohort-analyses.html#cb634-1" tabindex="-1"></a><span class="co"># Create a function to detect PCA sample outliers. The input is the PCA results data frame and the number of standard deviations for the cutoff. The output is outlier names. </span></span>
<span id="cb634-2"><a href="descriptive-cohort-analyses.html#cb634-2" tabindex="-1"></a>outlier_detection <span class="ot">=</span> <span class="cf">function</span>(pca_df, sd){</span>
<span id="cb634-3"><a href="descriptive-cohort-analyses.html#cb634-3" tabindex="-1"></a></span>
<span id="cb634-4"><a href="descriptive-cohort-analyses.html#cb634-4" tabindex="-1"></a>    <span class="co"># getting scores</span></span>
<span id="cb634-5"><a href="descriptive-cohort-analyses.html#cb634-5" tabindex="-1"></a>    scores <span class="ot">=</span> pca_df<span class="sc">$</span>x</span>
<span id="cb634-6"><a href="descriptive-cohort-analyses.html#cb634-6" tabindex="-1"></a>    </span>
<span id="cb634-7"><a href="descriptive-cohort-analyses.html#cb634-7" tabindex="-1"></a>    <span class="co"># identifying samples that are &gt; 6 standard deviations away from the mean</span></span>
<span id="cb634-8"><a href="descriptive-cohort-analyses.html#cb634-8" tabindex="-1"></a>    outlier_indices <span class="ot">=</span> <span class="fu">apply</span>(scores, <span class="dv">2</span>, <span class="cf">function</span>(x) <span class="fu">which</span>( <span class="fu">abs</span>(x <span class="sc">-</span> <span class="fu">mean</span>(x)) <span class="sc">&gt;</span> (sd <span class="sc">*</span> <span class="fu">sd</span>(x)) )) <span class="sc">%&gt;%</span></span>
<span id="cb634-9"><a href="descriptive-cohort-analyses.html#cb634-9" tabindex="-1"></a>        <span class="fu">Reduce</span>(union, .)</span>
<span id="cb634-10"><a href="descriptive-cohort-analyses.html#cb634-10" tabindex="-1"></a>    </span>
<span id="cb634-11"><a href="descriptive-cohort-analyses.html#cb634-11" tabindex="-1"></a>    <span class="co"># getting sample names</span></span>
<span id="cb634-12"><a href="descriptive-cohort-analyses.html#cb634-12" tabindex="-1"></a>    outliers <span class="ot">=</span> <span class="fu">rownames</span>(scores)[outlier_indices]</span>
<span id="cb634-13"><a href="descriptive-cohort-analyses.html#cb634-13" tabindex="-1"></a>    </span>
<span id="cb634-14"><a href="descriptive-cohort-analyses.html#cb634-14" tabindex="-1"></a>    <span class="fu">return</span>(outliers)</span>
<span id="cb634-15"><a href="descriptive-cohort-analyses.html#cb634-15" tabindex="-1"></a>}</span>
<span id="cb634-16"><a href="descriptive-cohort-analyses.html#cb634-16" tabindex="-1"></a></span>
<span id="cb634-17"><a href="descriptive-cohort-analyses.html#cb634-17" tabindex="-1"></a><span class="co"># Call function with different standard deviation cutoffs</span></span>
<span id="cb634-18"><a href="descriptive-cohort-analyses.html#cb634-18" tabindex="-1"></a>outliers_6 <span class="ot">&lt;-</span> <span class="fu">outlier_detection</span>(pca, <span class="dv">6</span>)</span>
<span id="cb634-19"><a href="descriptive-cohort-analyses.html#cb634-19" tabindex="-1"></a>outliers_5 <span class="ot">&lt;-</span> <span class="fu">outlier_detection</span>(pca, <span class="dv">5</span>)</span>
<span id="cb634-20"><a href="descriptive-cohort-analyses.html#cb634-20" tabindex="-1"></a>outliers_4 <span class="ot">&lt;-</span> <span class="fu">outlier_detection</span>(pca, <span class="dv">4</span>)</span>
<span id="cb634-21"><a href="descriptive-cohort-analyses.html#cb634-21" tabindex="-1"></a>outliers_3 <span class="ot">&lt;-</span> <span class="fu">outlier_detection</span>(pca, <span class="dv">3</span>)</span>
<span id="cb634-22"><a href="descriptive-cohort-analyses.html#cb634-22" tabindex="-1"></a></span>
<span id="cb634-23"><a href="descriptive-cohort-analyses.html#cb634-23" tabindex="-1"></a><span class="co"># Summary data frame</span></span>
<span id="cb634-24"><a href="descriptive-cohort-analyses.html#cb634-24" tabindex="-1"></a>outlier_summary <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sd_cutoff =</span> <span class="fu">c</span>(<span class="dv">6</span>, <span class="dv">5</span>, <span class="dv">4</span>, <span class="dv">3</span>), <span class="at">n_outliers =</span> <span class="fu">c</span>(<span class="fu">length</span>(outliers_6), <span class="fu">length</span>(outliers_5), <span class="fu">length</span>(outliers_4), <span class="fu">length</span>(outliers_3)))</span>
<span id="cb634-25"><a href="descriptive-cohort-analyses.html#cb634-25" tabindex="-1"></a></span>
<span id="cb634-26"><a href="descriptive-cohort-analyses.html#cb634-26" tabindex="-1"></a>outlier_summary</span></code></pre></div>
<pre><code>##   sd_cutoff n_outliers
## 1         6          0
## 2         5          1
## 3         4          5
## 4         3         28</code></pre>
<p>From these results, we see that there are no outliers that are &gt; 6 standard deviations from the mean, so we will proceed with the dataset without filtering any participants out.</p>
</div>
<div id="summary-statistics-tables" class="section level2 hasAnchor">
<h2>Summary Statistics Tables<a href="descriptive-cohort-analyses.html#summary-statistics-tables" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now that we have explored our dataset and finished processing the data, we can make a summary table that includes descriptive statistics (minimum, mean, median, maximum) for each of our chemicals. This table would go into supplementary material when the project is submitted for publication. It is a good idea to make this table using both the raw data and the cleaned data (imputed and normalized by time-weighted average) because different readers may have different interests in the data. For example, they may want to see the raw data so that they can understand chemical detection versus non-detection and absolute minimums or maximums of detection. Or, they may want to use the cleaned data for their own analyses. This table can also include information about whether or not the chemical passed our 20% detection filter.</p>
<p>There are many ways to generate summary statistics tables in R. Here, we will demonstrate a method using the <code>map_dfr()</code> function, which takes a list of functions and applies them across columns of the data. The summary statistics are then placed in rows, with each column representing a variable.</p>
<div class="sourceCode" id="cb636"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb636-1"><a href="descriptive-cohort-analyses.html#cb636-1" tabindex="-1"></a><span class="co"># Define summary functions</span></span>
<span id="cb636-2"><a href="descriptive-cohort-analyses.html#cb636-2" tabindex="-1"></a>summary_functs <span class="ot">&lt;-</span> <span class="fu">lst</span>(min, median, mean, max)</span>
<span id="cb636-3"><a href="descriptive-cohort-analyses.html#cb636-3" tabindex="-1"></a></span>
<span id="cb636-4"><a href="descriptive-cohort-analyses.html#cb636-4" tabindex="-1"></a><span class="co"># Apply summary functions to raw data</span></span>
<span id="cb636-5"><a href="descriptive-cohort-analyses.html#cb636-5" tabindex="-1"></a>summarystats_raw <span class="ot">&lt;-</span> <span class="fu">map_dfr</span>(summary_functs, <span class="sc">~</span> <span class="fu">summarise</span>(wrist_data, <span class="fu">across</span>(<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(wrist_data), .x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)), <span class="at">.id =</span> <span class="st">&quot;statistic&quot;</span>)</span>
<span id="cb636-6"><a href="descriptive-cohort-analyses.html#cb636-6" tabindex="-1"></a></span>
<span id="cb636-7"><a href="descriptive-cohort-analyses.html#cb636-7" tabindex="-1"></a><span class="co"># View data</span></span>
<span id="cb636-8"><a href="descriptive-cohort-analyses.html#cb636-8" tabindex="-1"></a><span class="fu">datatable</span>(summarystats_raw[, <span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>])</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-a399521a8e0b82f3c6b8" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-a399521a8e0b82f3c6b8">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4"],["min","median","mean","max"],[12.62,163.66,333.6065979381443,4468.06],[0.16,24.255,39.16260869565217,230.28],[0.09,1.105,1.46125,3.44],[2.15,57.15,118.6832989690722,1568.76],[1.46,27.19,71.87576923076922,1168.32]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>statistic<\/th>\n      <th>alkylOPE_2IPPDPP<\/th>\n      <th>alkylOPE_3IPPDPP<\/th>\n      <th>alkylOPE_2tBPDPP<\/th>\n      <th>alkylOPE_4IPPDPP<\/th>\n      <th>alkylOPE_B2IPPPP<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,6]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"statistic","targets":1},{"name":"alkylOPE_2IPPDPP","targets":2},{"name":"alkylOPE_3IPPDPP","targets":3},{"name":"alkylOPE_2tBPDPP","targets":4},{"name":"alkylOPE_4IPPDPP","targets":5},{"name":"alkylOPE_B2IPPPP","targets":6}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Through a few cleaning steps, we can transpose and format these data so that they are publication-quality.</p>
<div class="sourceCode" id="cb637"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb637-1"><a href="descriptive-cohort-analyses.html#cb637-1" tabindex="-1"></a>summarystats_raw <span class="ot">&lt;-</span> summarystats_raw <span class="sc">%&gt;%</span> </span>
<span id="cb637-2"><a href="descriptive-cohort-analyses.html#cb637-2" tabindex="-1"></a>  </span>
<span id="cb637-3"><a href="descriptive-cohort-analyses.html#cb637-3" tabindex="-1"></a>  <span class="co"># Transpose dataframe and return to dataframe class</span></span>
<span id="cb637-4"><a href="descriptive-cohort-analyses.html#cb637-4" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb637-5"><a href="descriptive-cohort-analyses.html#cb637-5" tabindex="-1"></a>  </span>
<span id="cb637-6"><a href="descriptive-cohort-analyses.html#cb637-6" tabindex="-1"></a>  <span class="co"># Make the first row the column names</span></span>
<span id="cb637-7"><a href="descriptive-cohort-analyses.html#cb637-7" tabindex="-1"></a>  <span class="fu">row_to_names</span>(<span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb637-8"><a href="descriptive-cohort-analyses.html#cb637-8" tabindex="-1"></a>  </span>
<span id="cb637-9"><a href="descriptive-cohort-analyses.html#cb637-9" tabindex="-1"></a>  <span class="co"># Remove rows with NAs (those where data are completely missing)</span></span>
<span id="cb637-10"><a href="descriptive-cohort-analyses.html#cb637-10" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span></span>
<span id="cb637-11"><a href="descriptive-cohort-analyses.html#cb637-11" tabindex="-1"></a>  </span>
<span id="cb637-12"><a href="descriptive-cohort-analyses.html#cb637-12" tabindex="-1"></a>  <span class="co"># Move chemical identifier to a column</span></span>
<span id="cb637-13"><a href="descriptive-cohort-analyses.html#cb637-13" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;class_chemical&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb637-14"><a href="descriptive-cohort-analyses.html#cb637-14" tabindex="-1"></a>  </span>
<span id="cb637-15"><a href="descriptive-cohort-analyses.html#cb637-15" tabindex="-1"></a>  <span class="co"># Round data</span></span>
<span id="cb637-16"><a href="descriptive-cohort-analyses.html#cb637-16" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(min<span class="sc">:</span>max, as.numeric)) <span class="sc">%&gt;%</span></span>
<span id="cb637-17"><a href="descriptive-cohort-analyses.html#cb637-17" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), round, <span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb637-18"><a href="descriptive-cohort-analyses.html#cb637-18" tabindex="-1"></a>  </span>
<span id="cb637-19"><a href="descriptive-cohort-analyses.html#cb637-19" tabindex="-1"></a>  <span class="co"># Add a suffix to column titles so we know that these came from the raw data</span></span>
<span id="cb637-20"><a href="descriptive-cohort-analyses.html#cb637-20" tabindex="-1"></a>  <span class="fu">rename_with</span>(<span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_raw&quot;</span>), min<span class="sc">:</span>max)</span>
<span id="cb637-21"><a href="descriptive-cohort-analyses.html#cb637-21" tabindex="-1"></a></span>
<span id="cb637-22"><a href="descriptive-cohort-analyses.html#cb637-22" tabindex="-1"></a><span class="fu">datatable</span>(summarystats_raw)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-132791470986819c0a69" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-132791470986819c0a69">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95"],["alkylOPE_2IPPDPP","alkylOPE_3IPPDPP","alkylOPE_2tBPDPP","alkylOPE_4IPPDPP","alkylOPE_B2IPPPP","alkylOPE_24DIPPDPP","alkylOPE_4tBPDPP","alkylOPE_B3IPPPP","alkylOPE_B4IPPPP","alkylOPE_B4tBPPP","alkylOPE_T4IPPP","alkylOPE_T4tBPP","BDE_28","BDE_47","BDE_66","BDE_100","BDE_99","BDE_85","BDE_154","BDE_153","BDE_209","BFR_BEHTBP","BFR_EHTBB","BFR_BTBPE","BFR_TDBPIC","OPE_TEP","OPE_TiBP","OPE_TnBP","OPE_TCEP","OPE_TCPP1","OPE_TCPP2","OPE_TCPP3","OPE_TPeP","OPE_TDCPP","OPE_TPHP","OPE_EHDPP","OPE_TEHP","OPE_isodecylPP","OPE_ToCP","OPE_TmCP","OPE_TpCP","OPE_TDMPP","PAH_Acenaphthylene","PAH_Acenapthene","PAH_Fluorene","PAH_Phenanthrene","PAH_Anthracene","PAH_Fluoranthene","PAH_Pyrene","PAH_Benzocphen","PAH_Benzaanth","PAH_Chrysene","PAH_712Dimethylbenz","PAH_Benzojbkfluor","PAH_Benzoepyrene","PAH_Benzoapyrene","PAH_Perylene","PAH_3Methylchol","PAH_Indeno123cd","PAH_Dibenzahanth","PAH_Benzoghiperylene","PCB_11","PCB_28","PCB_51","PCB_52","PCB_47","PCB_68","PCB_101","PCB_118","PCB_153","PCB_138","PCB_183","Pest_DEET","Pest_Lindane","Pest_Chlorpyrifos","Pest_Fipronil","Pest_Transchlordane","Pest_CisChlordane","Pest_PPDDE","Pest_Chlorfenapyr","Pest_CisPermethrin","Pest_TransPermethrin","Pest_Cypermethrin","Pest_Pyraclostrobin","Pest_Azoxystrobin","Phthal_DMP","Phthal_DEP","Phthal_DBP","Phthal_DiBP","Phthal_BBP","Phthal_DEHA","Phthal_DEHP","Phthal_DEHT","Phthal_DINP","Phthal_TOTM"],[12.62,0.16,0.09,2.15,1.46,3.52,3.25,0.33,0.12,0.47,2.27,1,0.25,0.75,0.2,0.33,0.37,0.18,0.46,0.64,3.56,2.57,3.93,1.96,15.78,5.5,0.96,9.5,11.07,34.03,5.58,0.16,0.07000000000000001,30.69,118.37,14.36,6.12,5.8,3.47,1.11,0.18,0.12,1.37,41.59,2.35,9.44,9.32,1.98,1.86,7.61,0.8100000000000001,0.27,1.48,1.28,0.51,1.5,4.08,0.18,0.29,2.68,1.01,0.21,0.09,0.63,0.23,0.16,0.18,0.11,0.2,0.23,0.2,0.23,56.86,0.16,0.54,0.38,0.23,0.79,0.12,3.28,5.46,7.59,1.5,0.7,1.05,2193.01,141.21,302.64,5974.72,38.34,446.86,5108.58,22701.62,4575.56,113.37],[163.66,24.25,1.1,57.15,27.19,50.75,66.59,1.15,5.17,55.92,3.09,9.83,3.06,30.28,1.92,6.16,26,3.56,1.92,4.37,7.05,30.64,77.5,1.96,20.9,12.76,4.06,41.37,36.12,308.8,68,4.21,0.11,535.97,1514.11,208.01,91.48999999999999,206.57,128.8,68.40000000000001,1.31,1,2.52,203,4.4,29.07,64.47,11.26,11.32,11.69,1.95,2.44,1.55,4.04,1.55,13.76,5.58,0.22,0.54,2.68,2.62,0.79,0.58,0.64,0.95,0.41,0.27,1.65,0.39,0.31,0.5600000000000001,0.37,407.55,0.51,3.08,11.43,8.17,9.93,0.82,8.710000000000001,75.12,101.13,12.05,0.8,11.74,2193.01,918.36,2331.52,7838.85,480.31,5536.44,28574.8,199641.7,56006.37,788.74],[333.61,39.16,1.46,118.68,71.88,113.66,215.83,1.31,13.36,166.32,3.09,33.62,4.04,88.54000000000001,4.44,30.64,93.97,10.68,15.6,21.08,12.18,76.23999999999999,300.35,1.96,20.9,74.84,8.74,68.48999999999999,69.73,959.63,166.37,11.51,0.53,1009.59,4401.51,434.41,223.47,1827.52,497.42,260.93,23.45,2.74,8.800000000000001,199.31,19.52,78.72,91.16,30.83,26.57,12.92,16.5,8.98,2.79,8.09,3.79,41.36,5.71,0.25,1.19,2.68,4.36,0.95,2.42,0.64,2.58,0.8100000000000001,0.39,3.38,0.82,0.63,1.08,1.08,1650.31,1.16,11.99,157.56,36.31,28.55,2.46,9.470000000000001,770.26,975.23,318.83,0.8,30.77,2193.01,3062.91,3687.68,9453.549999999999,1589.87,9844.01,43707.18,252269.91,93268.38,2025.85],[4468.06,230.28,3.44,1568.76,1168.32,741.4400000000001,4153.78,2.55,171.2,2512.85,3.91,446.54,13.99,1172.31,20.88,295.17,1636.56,53.7,122.26,142.83,85.06999999999999,633.98,4405.03,1.96,26.02,1430.37,322.28,1319.8,565.9400000000001,18574.28,1846.51,159.44,2.61,11591.51,162754.97,11679.46,6781.39,118220.31,16628.18,9509.889999999999,548.1799999999999,11.36,32.73,424.86,272.16,1445.7,199.06,604.76,420.26,20.7,133.75,97.8,6.57,82.22,36.73,218.49,7.6,0.38,12.72,2.68,12.58,5.26,27.99,0.65,13.29,2.22,0.82,11.74,2.74,1.64,2.73,3.36,21141.35,3.61,159.13,2323.26,505.31,211.92,26.07,17.57,30258.49,39126.38,6288.38,0.9,147.03,2193.01,113015.24,28170.98,17090.48,49744.68,76445.06,327313.95,1182220.92,815284.96,45398.66]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>class_chemical<\/th>\n      <th>min_raw<\/th>\n      <th>median_raw<\/th>\n      <th>mean_raw<\/th>\n      <th>max_raw<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"class_chemical","targets":1},{"name":"min_raw","targets":2},{"name":"median_raw","targets":3},{"name":"mean_raw","targets":4},{"name":"max_raw","targets":5}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>We can apply the same steps to the cleaned data.</p>
<div class="sourceCode" id="cb638"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb638-1"><a href="descriptive-cohort-analyses.html#cb638-1" tabindex="-1"></a>summarystats_cleaned <span class="ot">&lt;-</span> <span class="fu">map_dfr</span>(summary_functs, <span class="sc">~</span> <span class="fu">summarise</span>(wrist_data_cleaned, <span class="fu">across</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">ncol</span>(wrist_data_cleaned), .x, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)), </span>
<span id="cb638-2"><a href="descriptive-cohort-analyses.html#cb638-2" tabindex="-1"></a>                                <span class="at">.id =</span> <span class="st">&quot;statistic&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb638-3"><a href="descriptive-cohort-analyses.html#cb638-3" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb638-4"><a href="descriptive-cohort-analyses.html#cb638-4" tabindex="-1"></a>  <span class="fu">row_to_names</span>(<span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb638-5"><a href="descriptive-cohort-analyses.html#cb638-5" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span></span>
<span id="cb638-6"><a href="descriptive-cohort-analyses.html#cb638-6" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;class_chemical&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb638-7"><a href="descriptive-cohort-analyses.html#cb638-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(min<span class="sc">:</span>max, as.numeric)) <span class="sc">%&gt;%</span></span>
<span id="cb638-8"><a href="descriptive-cohort-analyses.html#cb638-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), round, <span class="dv">2</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb638-9"><a href="descriptive-cohort-analyses.html#cb638-9" tabindex="-1"></a>  <span class="fu">rename_with</span>(<span class="sc">~</span><span class="fu">paste0</span>(., <span class="st">&quot;_cleaned&quot;</span>), min<span class="sc">:</span>max)</span>
<span id="cb638-10"><a href="descriptive-cohort-analyses.html#cb638-10" tabindex="-1"></a></span>
<span id="cb638-11"><a href="descriptive-cohort-analyses.html#cb638-11" tabindex="-1"></a><span class="fu">datatable</span>(summarystats_cleaned)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-b16f9fee1031fe72956f" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-b16f9fee1031fe72956f">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62"],["alkylOPE_2IPPDPP","alkylOPE_3IPPDPP","alkylOPE_4IPPDPP","alkylOPE_B2IPPPP","alkylOPE_24DIPPDPP","alkylOPE_4tBPDPP","alkylOPE_B4IPPPP","alkylOPE_B4tBPPP","alkylOPE_T4tBPP","BDE_47","BDE_100","BDE_99","BDE_154","BDE_153","BDE_209","BFR_BEHTBP","BFR_EHTBB","OPE_TEP","OPE_TiBP","OPE_TnBP","OPE_TCEP","OPE_TCPP1","OPE_TCPP2","OPE_TCPP3","OPE_TDCPP","OPE_TPHP","OPE_EHDPP","OPE_TEHP","OPE_isodecylPP","OPE_ToCP","OPE_TmCP","OPE_TpCP","OPE_TDMPP","PAH_Fluorene","PAH_Phenanthrene","PAH_Fluoranthene","PAH_Pyrene","PAH_Benzaanth","PAH_Chrysene","PAH_Benzojbkfluor","PAH_Benzoepyrene","PAH_Indeno123cd","PCB_11","PCB_28","Pest_DEET","Pest_Chlorpyrifos","Pest_Fipronil","Pest_Transchlordane","Pest_CisChlordane","Pest_PPDDE","Pest_CisPermethrin","Pest_TransPermethrin","Pest_Cypermethrin","Pest_Azoxystrobin","Phthal_DEP","Phthal_DBP","Phthal_BBP","Phthal_DEHA","Phthal_DEHP","Phthal_DEHT","Phthal_DINP","Phthal_TOTM"],[1.49,0,0.67,0.02,0.02,0.55,0,0.01,0,0,0,0,0,0,0.2,0.13,0.01,0.04,0.19,1.24,0,2.74,0.95,0,0.13,4.28,1.76,0.07000000000000001,0.05,0.01,0.01,0,0,0.03,0.15,0.11,0.11,0,0,0.01,0,0,0.01,0,1.52,0,0,0,0,0,0.32,0.42,0,0,4.4,5.47,2.7,6.24,7.94,11.89,9.35,4.31],[4.63,2.13,3.24,1.86,2.17,3.44,0.49,3.3,0.02,2.28,0.02,2.13,0.01,0.02,0.89,2.44,3.51,0.64,0.7,2.9,1.19,5.75,3.52,0.61,6.32,7.8,4.94,3.83,4.9,4.37,3.51,0.01,0.01,0.53,2.37,1.45,1.41,0.03,0.03,0.34,0.05,0.04,0.14,0,5.95,0.03,0.01,0.38,0.11,0.01,3.7,4.12,0.14,0.17,7.12,8.51,6.47,9.800000000000001,11.79,14.92,12.99,6.95],[4.76,2.22,3.37,2.09,2.37,3.76,1,3.46,0.65,2.62,0.58,2.44,0.3,0.33,1.17,2.84,3.65,1.12,0.87,3.01,1.8,5.93,3.7,0.88,6.13,7.83,5.19,4.01,4.87,4.36,3.56,0.23,0.12,0.83,2.62,1.71,1.69,0.25,0.25,0.5,0.2,0.09,0.16,0.09,6.27,0.48,0.49,1.08,0.74,0.13,3.87,4.22,0.62,0.98,7.41,8.59,6.63,9.83,11.97,14.91,13.04,7.13],[9.539999999999999,5.42,8.039999999999999,7.61,6.96,9.220000000000001,4.88,8.49,6.02,9.18,7.37,9.82,6.1,6.39,3.72,7,9.779999999999999,8.17,5.56,7.57,6.35,11.37,10.21,7.33,10.69,14.51,10.71,9.92,14.27,11.21,10.41,6.31,1.39,8.09,10.5,6.45,5.93,4.33,3.9,3.67,2.64,1.49,0.8100000000000001,2.32,11.56,4.57,8.380000000000001,6.41,5,2.24,12.08,12.45,9.81,7.21,13.98,11.97,12.8,13.41,15.65,17.37,17.05,12.66]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>class_chemical<\/th>\n      <th>min_cleaned<\/th>\n      <th>median_cleaned<\/th>\n      <th>mean_cleaned<\/th>\n      <th>max_cleaned<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"class_chemical","targets":1},{"name":"min_cleaned","targets":2},{"name":"median_cleaned","targets":3},{"name":"mean_cleaned","targets":4},{"name":"max_cleaned","targets":5}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Finally, we will merge the data from our <code>chemical_counts</code> dataframe (which contains detection information for all of our chemicals) with our summary statistics dataframes.</p>
<div class="sourceCode" id="cb639"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb639-1"><a href="descriptive-cohort-analyses.html#cb639-1" tabindex="-1"></a>summarystats_final <span class="ot">&lt;-</span> chemical_counts <span class="sc">%&gt;%</span></span>
<span id="cb639-2"><a href="descriptive-cohort-analyses.html#cb639-2" tabindex="-1"></a>  </span>
<span id="cb639-3"><a href="descriptive-cohort-analyses.html#cb639-3" tabindex="-1"></a>  <span class="co"># Remove 90% detection filter column</span></span>
<span id="cb639-4"><a href="descriptive-cohort-analyses.html#cb639-4" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>det_filter_90) <span class="sc">%&gt;%</span></span>
<span id="cb639-5"><a href="descriptive-cohort-analyses.html#cb639-5" tabindex="-1"></a>  </span>
<span id="cb639-6"><a href="descriptive-cohort-analyses.html#cb639-6" tabindex="-1"></a>  <span class="co"># Add raw summary stats</span></span>
<span id="cb639-7"><a href="descriptive-cohort-analyses.html#cb639-7" tabindex="-1"></a>  <span class="fu">left_join</span>(summarystats_raw, <span class="at">by =</span> <span class="st">&quot;class_chemical&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb639-8"><a href="descriptive-cohort-analyses.html#cb639-8" tabindex="-1"></a>  </span>
<span id="cb639-9"><a href="descriptive-cohort-analyses.html#cb639-9" tabindex="-1"></a>  <span class="co"># Add cleaned summary stats</span></span>
<span id="cb639-10"><a href="descriptive-cohort-analyses.html#cb639-10" tabindex="-1"></a>  <span class="fu">left_join</span>(summarystats_cleaned, <span class="at">by =</span> <span class="st">&quot;class_chemical&quot;</span>)</span>
<span id="cb639-11"><a href="descriptive-cohort-analyses.html#cb639-11" tabindex="-1"></a></span>
<span id="cb639-12"><a href="descriptive-cohort-analyses.html#cb639-12" tabindex="-1"></a><span class="fu">datatable</span>(summarystats_final, <span class="at">width =</span> <span class="dv">600</span>)</span></code></pre></div>
<div id="htmlwidget-00769fef61f78d79d071" style="width:600px;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-00769fef61f78d79d071">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110"],["alkylOPE_2IPPDPP","alkylOPE_3IPPDPP","alkylOPE_2tBPDPP","alkylOPE_4IPPDPP","alkylOPE_B2IPPPP","alkylOPE_24DIPPDPP","alkylOPE_4tBPDPP","alkylOPE_B3IPPPP","alkylOPE_B2tBPPP","alkylOPE_B4IPPPP","alkylOPE_T3IPPP","alkylOPE_B24DIPPPP","alkylOPE_B4tBPPP","alkylOPE_T4IPPP","alkylOPE_T4tBPP","BDE_28","BDE_47","BDE_66","BDE_100","BDE_99","BDE_85","BDE_154","BDE_153","BDE_183","BDE_209","BFR_BEHTBP","BFR_EHTBB","BFR_TBBPA","BFR_BTBPE","BFR_TDBPIC","OPE_TEP","OPE_TiPP","OPE_TPrP","OPE_TiBP","OPE_TnBP","OPE_TCEP","OPE_TCPP1","OPE_TCPP2","OPE_TCPP3","OPE_TPeP","OPE_TDCPP","OPE_TPHP","OPE_EHDPP","OPE_TEHP","OPE_isodecylPP","OPE_ToCP","OPE_TmCP","OPE_TpCP","OPE_TDMPP","PAH_Naphthalene","PAH_Acenaphthylene","PAH_Acenapthene","PAH_Fluorene","PAH_Phenanthrene","PAH_Anthracene","PAH_Fluoranthene","PAH_Pyrene","PAH_Benzocphen","PAH_Benzaanth","PAH_Chrysene","PAH_712Dimethylbenz","PAH_Benzojbkfluor","PAH_Benzoepyrene","PAH_Benzoapyrene","PAH_Perylene","PAH_3Methylchol","PAH_Indeno123cd","PAH_Dibenzahanth","PAH_Benzoghiperylene","PAH_Dibenzalpyrene","PAH_Dibenzaipyrene","PAH_Dibenzahpyrene","PCB_11","PCB_28","PCB_51","PCB_52","PCB_47","PCB_68","PCB_101","PCB_118","PCB_153","PCB_138","PCB_183","Pest_DEET","Pest_Atrazine","Pest_Lindane","Pest_Malathion","Pest_Chlorpyrifos","Pest_Fipronil","Pest_Transchlordane","Pest_CisChlordane","Pest_PPDDE","Pest_Chlorfenapyr","Pest_Trifloxystrobin","Pest_CisPermethrin","Pest_TransPermethrin","Pest_Cypermethrin","Pest_Pyraclostrobin","Pest_Azoxystrobin","Pest_Fluoxastrobin","Phthal_DMP","Phthal_DEP","Phthal_DBP","Phthal_DiBP","Phthal_BBP","Phthal_DEHA","Phthal_DEHP","Phthal_DEHT","Phthal_DINP","Phthal_TOTM"],[97,92,8,97,78,65,97,6,0,87,0,0,96,2,37,11,93,11,34,89,13,22,20,0,71,96,90,0,1,2,37,0,0,97,97,45,97,97,83,6,93,97,97,95,92,93,96,33,28,0,6,6,58,93,6,95,95,4,23,30,4,49,34,7,4,6,29,1,6,0,0,0,78,27,2,13,12,8,12,11,7,7,4,93,0,13,0,46,21,62,41,33,5,0,90,92,23,2,47,0,1,97,97,6,97,97,90,97,97,97],[0,5,89,0,19,32,0,91,97,10,97,97,1,95,60,86,4,86,63,8,84,75,77,97,26,1,7,97,96,95,60,97,97,0,0,52,0,0,14,91,4,0,0,2,5,4,1,64,69,97,91,91,39,4,91,2,2,93,74,67,93,48,63,90,93,91,68,96,91,97,97,97,19,70,95,84,85,89,85,86,90,90,93,4,97,84,97,51,76,35,56,64,92,97,7,5,74,95,50,97,96,0,0,91,0,0,7,0,0,0],[100,94.84999999999999,8.25,100,80.41,67.01000000000001,100,6.19,0,89.69,0,0,98.97,2.06,38.14,11.34,95.88,11.34,35.05,91.75,13.4,22.68,20.62,0,73.2,98.97,92.78,0,1.03,2.06,38.14,0,0,100,100,46.39,100,100,85.56999999999999,6.19,95.88,100,100,97.94,94.84999999999999,95.88,98.97,34.02,28.87,0,6.19,6.19,59.79,95.88,6.19,97.94,97.94,4.12,23.71,30.93,4.12,50.52,35.05,7.22,4.12,6.19,29.9,1.03,6.19,0,0,0,80.41,27.84,2.06,13.4,12.37,8.25,12.37,11.34,7.22,7.22,4.12,95.88,0,13.4,0,47.42,21.65,63.92,42.27,34.02,5.15,0,92.78,94.84999999999999,23.71,2.06,48.45,0,1.03,100,100,6.19,100,100,92.78,100,100,100],[0,5.15,91.75,0,19.59,32.99,0,93.81,100,10.31,100,100,1.03,97.94,61.86,88.66,4.12,88.66,64.95,8.25,86.59999999999999,77.31999999999999,79.38,100,26.8,1.03,7.22,100,98.97,97.94,61.86,100,100,0,0,53.61,0,0,14.43,93.81,4.12,0,0,2.06,5.15,4.12,1.03,65.98,71.13,100,93.81,93.81,40.21,4.12,93.81,2.06,2.06,95.88,76.29000000000001,69.06999999999999,95.88,49.48,64.95,92.78,95.88,93.81,70.09999999999999,98.97,93.81,100,100,100,19.59,72.16,97.94,86.59999999999999,87.63,91.75,87.63,88.66,92.78,92.78,95.88,4.12,100,86.59999999999999,100,52.58,78.34999999999999,36.08,57.73,65.98,94.84999999999999,100,7.22,5.15,76.29000000000001,97.94,51.55,100,98.97,0,0,93.81,0,0,7.22,0,0,0],["Yes","Yes","No","Yes","Yes","Yes","Yes","No","No","Yes","No","No","Yes","No","Yes","No","Yes","No","Yes","Yes","No","Yes","Yes","No","Yes","Yes","Yes","No","No","No","Yes","No","No","Yes","Yes","Yes","Yes","Yes","Yes","No","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","Yes","No","No","No","Yes","Yes","No","Yes","Yes","No","Yes","Yes","No","Yes","Yes","No","No","No","Yes","No","No","No","No","No","Yes","Yes","No","No","No","No","No","No","No","No","No","Yes","No","No","No","Yes","Yes","Yes","Yes","Yes","No","No","Yes","Yes","Yes","No","Yes","No","No","Yes","Yes","No","Yes","Yes","Yes","Yes","Yes","Yes"],[12.62,0.16,0.09,2.15,1.46,3.52,3.25,0.33,null,0.12,null,null,0.47,2.27,1,0.25,0.75,0.2,0.33,0.37,0.18,0.46,0.64,null,3.56,2.57,3.93,null,1.96,15.78,5.5,null,null,0.96,9.5,11.07,34.03,5.58,0.16,0.07000000000000001,30.69,118.37,14.36,6.12,5.8,3.47,1.11,0.18,0.12,null,1.37,41.59,2.35,9.44,9.32,1.98,1.86,7.61,0.8100000000000001,0.27,1.48,1.28,0.51,1.5,4.08,0.18,0.29,2.68,1.01,null,null,null,0.21,0.09,0.63,0.23,0.16,0.18,0.11,0.2,0.23,0.2,0.23,56.86,null,0.16,null,0.54,0.38,0.23,0.79,0.12,3.28,null,5.46,7.59,1.5,0.7,1.05,null,2193.01,141.21,302.64,5974.72,38.34,446.86,5108.58,22701.62,4575.56,113.37],[163.66,24.25,1.1,57.15,27.19,50.75,66.59,1.15,null,5.17,null,null,55.92,3.09,9.83,3.06,30.28,1.92,6.16,26,3.56,1.92,4.37,null,7.05,30.64,77.5,null,1.96,20.9,12.76,null,null,4.06,41.37,36.12,308.8,68,4.21,0.11,535.97,1514.11,208.01,91.48999999999999,206.57,128.8,68.40000000000001,1.31,1,null,2.52,203,4.4,29.07,64.47,11.26,11.32,11.69,1.95,2.44,1.55,4.04,1.55,13.76,5.58,0.22,0.54,2.68,2.62,null,null,null,0.79,0.58,0.64,0.95,0.41,0.27,1.65,0.39,0.31,0.5600000000000001,0.37,407.55,null,0.51,null,3.08,11.43,8.17,9.93,0.82,8.710000000000001,null,75.12,101.13,12.05,0.8,11.74,null,2193.01,918.36,2331.52,7838.85,480.31,5536.44,28574.8,199641.7,56006.37,788.74],[333.61,39.16,1.46,118.68,71.88,113.66,215.83,1.31,null,13.36,null,null,166.32,3.09,33.62,4.04,88.54000000000001,4.44,30.64,93.97,10.68,15.6,21.08,null,12.18,76.23999999999999,300.35,null,1.96,20.9,74.84,null,null,8.74,68.48999999999999,69.73,959.63,166.37,11.51,0.53,1009.59,4401.51,434.41,223.47,1827.52,497.42,260.93,23.45,2.74,null,8.800000000000001,199.31,19.52,78.72,91.16,30.83,26.57,12.92,16.5,8.98,2.79,8.09,3.79,41.36,5.71,0.25,1.19,2.68,4.36,null,null,null,0.95,2.42,0.64,2.58,0.8100000000000001,0.39,3.38,0.82,0.63,1.08,1.08,1650.31,null,1.16,null,11.99,157.56,36.31,28.55,2.46,9.470000000000001,null,770.26,975.23,318.83,0.8,30.77,null,2193.01,3062.91,3687.68,9453.549999999999,1589.87,9844.01,43707.18,252269.91,93268.38,2025.85],[4468.06,230.28,3.44,1568.76,1168.32,741.4400000000001,4153.78,2.55,null,171.2,null,null,2512.85,3.91,446.54,13.99,1172.31,20.88,295.17,1636.56,53.7,122.26,142.83,null,85.06999999999999,633.98,4405.03,null,1.96,26.02,1430.37,null,null,322.28,1319.8,565.9400000000001,18574.28,1846.51,159.44,2.61,11591.51,162754.97,11679.46,6781.39,118220.31,16628.18,9509.889999999999,548.1799999999999,11.36,null,32.73,424.86,272.16,1445.7,199.06,604.76,420.26,20.7,133.75,97.8,6.57,82.22,36.73,218.49,7.6,0.38,12.72,2.68,12.58,null,null,null,5.26,27.99,0.65,13.29,2.22,0.82,11.74,2.74,1.64,2.73,3.36,21141.35,null,3.61,null,159.13,2323.26,505.31,211.92,26.07,17.57,null,30258.49,39126.38,6288.38,0.9,147.03,null,2193.01,113015.24,28170.98,17090.48,49744.68,76445.06,327313.95,1182220.92,815284.96,45398.66],[1.49,0,null,0.67,0.02,0.02,0.55,null,null,0,null,null,0.01,null,0,null,0,null,0,0,null,0,0,null,0.2,0.13,0.01,null,null,null,0.04,null,null,0.19,1.24,0,2.74,0.95,0,null,0.13,4.28,1.76,0.07000000000000001,0.05,0.01,0.01,0,0,null,null,null,0.03,0.15,null,0.11,0.11,null,0,0,null,0.01,0,null,null,null,0,null,null,null,null,null,0.01,0,null,null,null,null,null,null,null,null,null,1.52,null,null,null,0,0,0,0,0,null,null,0.32,0.42,0,null,0,null,null,4.4,5.47,null,2.7,6.24,7.94,11.89,9.35,4.31],[4.63,2.13,null,3.24,1.86,2.17,3.44,null,null,0.49,null,null,3.3,null,0.02,null,2.28,null,0.02,2.13,null,0.01,0.02,null,0.89,2.44,3.51,null,null,null,0.64,null,null,0.7,2.9,1.19,5.75,3.52,0.61,null,6.32,7.8,4.94,3.83,4.9,4.37,3.51,0.01,0.01,null,null,null,0.53,2.37,null,1.45,1.41,null,0.03,0.03,null,0.34,0.05,null,null,null,0.04,null,null,null,null,null,0.14,0,null,null,null,null,null,null,null,null,null,5.95,null,null,null,0.03,0.01,0.38,0.11,0.01,null,null,3.7,4.12,0.14,null,0.17,null,null,7.12,8.51,null,6.47,9.800000000000001,11.79,14.92,12.99,6.95],[4.76,2.22,null,3.37,2.09,2.37,3.76,null,null,1,null,null,3.46,null,0.65,null,2.62,null,0.58,2.44,null,0.3,0.33,null,1.17,2.84,3.65,null,null,null,1.12,null,null,0.87,3.01,1.8,5.93,3.7,0.88,null,6.13,7.83,5.19,4.01,4.87,4.36,3.56,0.23,0.12,null,null,null,0.83,2.62,null,1.71,1.69,null,0.25,0.25,null,0.5,0.2,null,null,null,0.09,null,null,null,null,null,0.16,0.09,null,null,null,null,null,null,null,null,null,6.27,null,null,null,0.48,0.49,1.08,0.74,0.13,null,null,3.87,4.22,0.62,null,0.98,null,null,7.41,8.59,null,6.63,9.83,11.97,14.91,13.04,7.13],[9.539999999999999,5.42,null,8.039999999999999,7.61,6.96,9.220000000000001,null,null,4.88,null,null,8.49,null,6.02,null,9.18,null,7.37,9.82,null,6.1,6.39,null,3.72,7,9.779999999999999,null,null,null,8.17,null,null,5.56,7.57,6.35,11.37,10.21,7.33,null,10.69,14.51,10.71,9.92,14.27,11.21,10.41,6.31,1.39,null,null,null,8.09,10.5,null,6.45,5.93,null,4.33,3.9,null,3.67,2.64,null,null,null,1.49,null,null,null,null,null,0.8100000000000001,2.32,null,null,null,null,null,null,null,null,null,11.56,null,null,null,4.57,8.380000000000001,6.41,5,2.24,null,null,12.08,12.45,9.81,null,7.21,null,null,13.98,11.97,null,12.8,13.41,15.65,17.37,17.05,12.66]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>class_chemical<\/th>\n      <th>n_detected<\/th>\n      <th>n_undetected<\/th>\n      <th>perc_detected<\/th>\n      <th>perc_undetected<\/th>\n      <th>det_filter_20<\/th>\n      <th>min_raw<\/th>\n      <th>median_raw<\/th>\n      <th>mean_raw<\/th>\n      <th>max_raw<\/th>\n      <th>min_cleaned<\/th>\n      <th>median_cleaned<\/th>\n      <th>mean_cleaned<\/th>\n      <th>max_cleaned<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[2,3,4,5,7,8,9,10,11,12,13,14]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"class_chemical","targets":1},{"name":"n_detected","targets":2},{"name":"n_undetected","targets":3},{"name":"perc_detected","targets":4},{"name":"perc_undetected","targets":5},{"name":"det_filter_20","targets":6},{"name":"min_raw","targets":7},{"name":"median_raw","targets":8},{"name":"mean_raw","targets":9},{"name":"max_raw","targets":10},{"name":"min_cleaned","targets":11},{"name":"median_cleaned","targets":12},{"name":"mean_cleaned","targets":13},{"name":"max_cleaned","targets":14}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="demographics-table" class="section level2 hasAnchor">
<h2>Demographics Table<a href="descriptive-cohort-analyses.html#demographics-table" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Another important element of any analysis of human data is the demographics table. The demographics table provides key information about the study participants and can help inform downstream analyses, such as exploration of the impact of covariates on the endpoint of interest. There are many different ways to make demographics tables in R. Here, we will demonstrate making a demographics table with the <em>table1</em> package. For more on this package, including making tables with multiple groups and testing for statistical differences in demographics between groups, see the <em>table1</em> vignette <a href="https://benjaminrich.github.io/table1/vignettes/table1-examples.html">here</a>.</p>
<p>First, we’ll read in and view our demographic data:</p>
<div class="sourceCode" id="cb640"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb640-1"><a href="descriptive-cohort-analyses.html#cb640-1" tabindex="-1"></a>demo_data <span class="ot">&lt;-</span> <span class="fu">read.xlsx</span>(<span class="st">&quot;Module6_1_Input/Module6_1_InputData3.xlsx&quot;</span>)</span>
<span id="cb640-2"><a href="descriptive-cohort-analyses.html#cb640-2" tabindex="-1"></a></span>
<span id="cb640-3"><a href="descriptive-cohort-analyses.html#cb640-3" tabindex="-1"></a><span class="fu">datatable</span>(demo_data)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-03c0c800f2209c8a2f2a" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-03c0c800f2209c8a2f2a">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97"],[1014,1018,1019,1023,1026,1037,1039,1049,1052,1054,1061,1063,1064,1070,1071,1073,1075,1076,1079,1081,1082,1084,1085,1088,1091,1093,1094,1095,1096,1098,1099,1101,1102,1105,1110,1114,1116,1117,1118,1119,1120,1124,1126,1127,1128,1129,1130,1131,1133,1135,1144,1145,1147,1153,1154,1156,1159,1161,1163,1168,1169,1171,1172,1179,1182,1187,1188,1191,1193,1194,1196,1199,1200,1201,1202,1207,1210,1212,1215,1216,1227,1229,1250,1252,1253,1254,1255,1257,1259,1260,1261,1263,1264,1265,1266,1268,1272],[32.41369863,33.45205479,18.97260274,40.55068493,31.8109589,26.84931507,30.1369863,32.0739726,25.39452055,35.52876712,25.63013699,28.52054795,32.66849315,28.96438356,25.64931507,21.15342466,27.83561644,31.14246575,36.15616438,33.63835616,27.36438356,26.43287671,25.93424658,27.81369863,23.02465753,39.39452055,28.55890411,35.86849315,31.23561644,41.09863014,29.81369863,34.12054795,34.13972603,35.20547945,32.56438356,26.57534247,23.83561644,28.73424658,35.7369863,29.26027397,32.71232877,23.06849315,29.16164384,30.55890411,28.84931507,21.96164384,32.76164384,37.03013699,29.81369863,34.40821918,34.55068493,26.52054795,31.25753425,30.66575342,32.76164384,18.95890411,30.82465753,24.24109589,32.97260274,22.45479452,39.19452055,36.91232877,37.10684932,44.84657534,26.19178082,24.76986301,34.56986301,35.53150685,30.62465753,27.40821918,20.17260274,30.48493151,39.29315068,46.03835616,41.09315068,31.87945205,30.39178082,27.60273973,35.02465753,21.40821918,32.94246575,28.30410959,31.67671233,29.82465753,24.05205479,32.2739726,35.63835616,25.12054795,34.29041096,33.89315068,32.18082192,26.24109589,39.08219178,33.71506849,41.05479452,29.43561644,34.54794521],["F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F"],["Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman","Woman"],["Four-Year Degree","Four-Year Degree","High School or GED","Professional Degree or PhD","High School or GED","Master's Degree","Professional Degree or PhD","Master's Degree","Four-Year Degree","Four-Year Degree","Four-Year Degree","Master's Degree","Master's Degree","Master's Degree","Master's Degree","High School or GED","Four-Year Degree","High School or GED","Four-Year Degree","Four-Year Degree","Four-Year Degree","Master's Degree","High School or GED","High School or GED","Associate Degree","Four-Year Degree","High School or GED","Four-Year Degree","Four-Year Degree","Four-Year Degree","Associate Degree","High School or GED","Associate Degree","Professional Degree or PhD","Four-Year Degree","High School or GED","High School or GED","High School or GED","Four-Year Degree","Four-Year Degree","Associate Degree","High School or GED","Master's Degree","Professional Degree or PhD","Four-Year Degree","High School or GED","Four-Year Degree","Professional Degree or PhD","Professional Degree or PhD","Professional Degree or PhD","Master's Degree","Associate Degree","Four-Year Degree","High School or GED","Associate Degree","High School or GED","Four-Year Degree","High School or GED","Professional Degree or PhD","High School or GED","Four-Year Degree","Four-Year Degree","Professional Degree or PhD","High School or GED","High School or GED","High School or GED","Professional Degree or PhD","High School or GED","Four-Year Degree","Four-Year Degree","Four-Year Degree","Four-Year Degree","High School or GED","High School or GED","Professional Degree or PhD","High School or GED","High School or GED","Four-Year Degree","High School or GED","High School or GED","High School or GED","Four-Year Degree","Master's Degree","Four-Year Degree","Master's Degree","Four-Year Degree","Master's Degree","Four-Year Degree","Professional Degree or PhD","Four-Year Degree","Four-Year Degree","Four-Year Degree","Master's Degree","Four-Year Degree","Master's Degree","Master's Degree","Four-Year Degree"],["No","No","No","No","No","No","No","No","No","No","No","No","No","No","No","Yes","No","No","No","No","No","No","No","No","No","No","Yes","No","No","No","Yes","Yes","No","No","No","Yes","No","Yes","No","No","No","No","No","No","No","Yes","No","No","No","No","No","No","No","Yes","No","Yes","No","No","No","Yes","Yes","No","No","No","No","No","No","No","No","Yes","Yes","No","No","No","No","Yes","No","No","No","No","No","No","No","No","No","No","No","No","No","Yes","No","No","No","No","No","No","No"],["White","White","Black","Black","White","White","White","White","White","White","White","White","White","White","Biracial/Multiracial","Other","Black","White","White","White","Black","White","Biracial/Multiracial","Black","White","White","White","Other","White","White","Other","Other","Black","White","White","Biracial/Multiracial","White","Other","White","White","Black","White","White","Biracial/Multiracial","White","Other","White","White","White","White","White","White","White","White","White","Other","White","Black","Black","Other","Other","Biracial/Multiracial","Other","Black","White","Black","White","Black","White","White","White","White","White","White","White","Black","Black","White","Black","Black","Black","White","White","White","Black","White","White","Black","White","Other","White","White","White","White","White","White","White"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>S_ID<\/th>\n      <th>mat_age_birth<\/th>\n      <th>pc_sex<\/th>\n      <th>pc_gender<\/th>\n      <th>pc_ed<\/th>\n      <th>pc_latino_hispanic<\/th>\n      <th>pc_race_cleaned<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[1,2]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"S_ID","targets":1},{"name":"mat_age_birth","targets":2},{"name":"pc_sex","targets":3},{"name":"pc_gender","targets":4},{"name":"pc_ed","targets":5},{"name":"pc_latino_hispanic","targets":6},{"name":"pc_race_cleaned","targets":7}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Then, we can create new labels for our variables so that they are more nicely formatted and more intuitive for display in the table.</p>
<div class="sourceCode" id="cb641"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb641-1"><a href="descriptive-cohort-analyses.html#cb641-1" tabindex="-1"></a><span class="co"># Create new labels for the demographics table</span></span>
<span id="cb641-2"><a href="descriptive-cohort-analyses.html#cb641-2" tabindex="-1"></a><span class="fu">label</span>(demo_data<span class="sc">$</span>mat_age_birth) <span class="ot">&lt;-</span> <span class="st">&quot;Age at Childbirth&quot;</span></span>
<span id="cb641-3"><a href="descriptive-cohort-analyses.html#cb641-3" tabindex="-1"></a><span class="fu">label</span>(demo_data<span class="sc">$</span>pc_sex) <span class="ot">&lt;-</span> <span class="st">&quot;Sex&quot;</span></span>
<span id="cb641-4"><a href="descriptive-cohort-analyses.html#cb641-4" tabindex="-1"></a><span class="fu">label</span>(demo_data<span class="sc">$</span>pc_gender) <span class="ot">&lt;-</span> <span class="st">&quot;Gender&quot;</span></span>
<span id="cb641-5"><a href="descriptive-cohort-analyses.html#cb641-5" tabindex="-1"></a><span class="fu">label</span>(demo_data<span class="sc">$</span>pc_latino_hispanic) <span class="ot">&lt;-</span> <span class="st">&quot;Latino or Hispanic&quot;</span></span>
<span id="cb641-6"><a href="descriptive-cohort-analyses.html#cb641-6" tabindex="-1"></a><span class="fu">label</span>(demo_data<span class="sc">$</span>pc_race_cleaned) <span class="ot">&lt;-</span> <span class="st">&quot;Race&quot;</span></span>
<span id="cb641-7"><a href="descriptive-cohort-analyses.html#cb641-7" tabindex="-1"></a><span class="fu">label</span>(demo_data<span class="sc">$</span>pc_ed) <span class="ot">&lt;-</span> <span class="st">&quot;Educational Attainment&quot;</span></span></code></pre></div>
<p>Our demographics data also had “F” for female in the sex column. We can change this to “Female” so that the demographics table is more readable.</p>
<div class="sourceCode" id="cb642"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb642-1"><a href="descriptive-cohort-analyses.html#cb642-1" tabindex="-1"></a>demo_data <span class="ot">&lt;-</span> demo_data <span class="sc">%&gt;%</span></span>
<span id="cb642-2"><a href="descriptive-cohort-analyses.html#cb642-2" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pc_sex =</span> dplyr<span class="sc">::</span><span class="fu">recode</span>(pc_sex, <span class="st">&quot;F&quot;</span> <span class="ot">=</span> <span class="st">&quot;Female&quot;</span>))</span>
<span id="cb642-3"><a href="descriptive-cohort-analyses.html#cb642-3" tabindex="-1"></a></span>
<span id="cb642-4"><a href="descriptive-cohort-analyses.html#cb642-4" tabindex="-1"></a><span class="fu">label</span>(demo_data<span class="sc">$</span>pc_sex) <span class="ot">&lt;-</span> <span class="st">&quot;Sex&quot;</span></span></code></pre></div>
<p>Now, let’s make the table. The first argument in the formula is all of the columns you want to include in the table, followed by the input dataframe.</p>
<div class="sourceCode" id="cb643"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb643-1"><a href="descriptive-cohort-analyses.html#cb643-1" tabindex="-1"></a><span class="fu">table1</span>(<span class="sc">~</span> mat_age_birth <span class="sc">+</span> pc_sex <span class="sc">+</span> pc_gender <span class="sc">+</span> pc_latino_hispanic <span class="sc">+</span> pc_race_cleaned <span class="sc">+</span> pc_ed, <span class="at">data =</span> demo_data)</span></code></pre></div>
<div class="Rtable1"><table class="Rtable1">
<thead>
<tr>
<th class='rowlabel firstrow lastrow'></th>
<th class='firstrow lastrow'><span class='stratlabel'>Overall<br><span class='stratn'>(N=97)</span></span></th>
</tr>
</thead>
<tbody>
<tr>
<td class='rowlabel firstrow'>Age at Childbirth</td>
<td class='firstrow'></td>
</tr>
<tr>
<td class='rowlabel'>Mean (SD)</td>
<td>31.0 (5.54)</td>
</tr>
<tr>
<td class='rowlabel lastrow'>Median [Min, Max]</td>
<td class='lastrow'>31.1 [19.0, 46.0]</td>
</tr>
<tr>
<td class='rowlabel firstrow'>Sex</td>
<td class='firstrow'></td>
</tr>
<tr>
<td class='rowlabel lastrow'>Female</td>
<td class='lastrow'>97 (100%)</td>
</tr>
<tr>
<td class='rowlabel firstrow'>Gender</td>
<td class='firstrow'></td>
</tr>
<tr>
<td class='rowlabel lastrow'>Woman</td>
<td class='lastrow'>97 (100%)</td>
</tr>
<tr>
<td class='rowlabel firstrow'>Latino or Hispanic</td>
<td class='firstrow'></td>
</tr>
<tr>
<td class='rowlabel'>No</td>
<td>82 (84.5%)</td>
</tr>
<tr>
<td class='rowlabel lastrow'>Yes</td>
<td class='lastrow'>15 (15.5%)</td>
</tr>
<tr>
<td class='rowlabel firstrow'>Race</td>
<td class='firstrow'></td>
</tr>
<tr>
<td class='rowlabel'>Biracial/Multiracial</td>
<td>5 (5.2%)</td>
</tr>
<tr>
<td class='rowlabel'>Black</td>
<td>19 (19.6%)</td>
</tr>
<tr>
<td class='rowlabel'>Other</td>
<td>11 (11.3%)</td>
</tr>
<tr>
<td class='rowlabel lastrow'>White</td>
<td class='lastrow'>62 (63.9%)</td>
</tr>
<tr>
<td class='rowlabel firstrow'>Educational Attainment</td>
<td class='firstrow'></td>
</tr>
<tr>
<td class='rowlabel'>Associate Degree</td>
<td>6 (6.2%)</td>
</tr>
<tr>
<td class='rowlabel'>Four-Year Degree</td>
<td>36 (37.1%)</td>
</tr>
<tr>
<td class='rowlabel'>High School or GED</td>
<td>28 (28.9%)</td>
</tr>
<tr>
<td class='rowlabel'>Master's Degree</td>
<td>15 (15.5%)</td>
</tr>
<tr>
<td class='rowlabel lastrow'>Professional Degree or PhD</td>
<td class='lastrow'>12 (12.4%)</td>
</tr>
</tbody>
</table>
</div>
<p><br></p>
<p>There are a couple of steps we could take to clean up the table:</p>
<ol style="list-style-type: decimal">
<li>Change the rendering for our continuous variable (age) to just mean (SD).</li>
<li>Order educational attainment so that it progresses from least to most education.</li>
</ol>
<p>We can change the rendering for our continuous variable by defining our own rendering function (as demonstrated in the package’s vignette).</p>
<div class="sourceCode" id="cb644"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb644-1"><a href="descriptive-cohort-analyses.html#cb644-1" tabindex="-1"></a><span class="co"># Create function for custom table so that Mean (SD) is shown for continuous variables</span></span>
<span id="cb644-2"><a href="descriptive-cohort-analyses.html#cb644-2" tabindex="-1"></a>my.render.cont <span class="ot">&lt;-</span> <span class="cf">function</span>(x) {</span>
<span id="cb644-3"><a href="descriptive-cohort-analyses.html#cb644-3" tabindex="-1"></a>  <span class="fu">with</span>(<span class="fu">stats.apply.rounding</span>(<span class="fu">stats.default</span>(x), <span class="at">digits=</span><span class="dv">2</span>), </span>
<span id="cb644-4"><a href="descriptive-cohort-analyses.html#cb644-4" tabindex="-1"></a>       <span class="fu">c</span>(<span class="st">&quot;&quot;</span>, <span class="st">&quot;Mean (SD)&quot;</span><span class="ot">=</span><span class="fu">sprintf</span>(<span class="st">&quot;%s (&amp;plusmn; %s)&quot;</span>, MEAN, SD)))</span>
<span id="cb644-5"><a href="descriptive-cohort-analyses.html#cb644-5" tabindex="-1"></a>}</span></code></pre></div>
<p>We can order the education attainment by changing it to a factor and defining the levels.</p>
<div class="sourceCode" id="cb645"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb645-1"><a href="descriptive-cohort-analyses.html#cb645-1" tabindex="-1"></a>demo_data <span class="ot">&lt;-</span> demo_data <span class="sc">%&gt;%</span></span>
<span id="cb645-2"><a href="descriptive-cohort-analyses.html#cb645-2" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pc_ed =</span> <span class="fu">factor</span>(pc_ed, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;High School or GED&quot;</span>, <span class="st">&quot;Associate Degree&quot;</span>, <span class="st">&quot;Four-Year Degree&quot;</span>, </span>
<span id="cb645-3"><a href="descriptive-cohort-analyses.html#cb645-3" tabindex="-1"></a>                                          <span class="st">&quot;Master&#39;s Degree&quot;</span>, <span class="st">&quot;Professional Degree or PhD&quot;</span>)))</span>
<span id="cb645-4"><a href="descriptive-cohort-analyses.html#cb645-4" tabindex="-1"></a></span>
<span id="cb645-5"><a href="descriptive-cohort-analyses.html#cb645-5" tabindex="-1"></a><span class="fu">label</span>(demo_data<span class="sc">$</span>pc_ed) <span class="ot">&lt;-</span> <span class="st">&quot;Educational Attainment&quot;</span></span></code></pre></div>
<p>Then, we can make our final table.</p>
<div class="sourceCode" id="cb646"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb646-1"><a href="descriptive-cohort-analyses.html#cb646-1" tabindex="-1"></a><span class="fu">table1</span>(<span class="sc">~</span> mat_age_birth <span class="sc">+</span> pc_sex <span class="sc">+</span> pc_gender <span class="sc">+</span> pc_latino_hispanic <span class="sc">+</span> pc_race_cleaned <span class="sc">+</span> pc_ed, </span>
<span id="cb646-2"><a href="descriptive-cohort-analyses.html#cb646-2" tabindex="-1"></a>       <span class="at">data =</span> demo_data,</span>
<span id="cb646-3"><a href="descriptive-cohort-analyses.html#cb646-3" tabindex="-1"></a>       <span class="at">render.continuous =</span> my.render.cont)</span></code></pre></div>
<div class="Rtable1"><table class="Rtable1">
<thead>
<tr>
<th class='rowlabel firstrow lastrow'></th>
<th class='firstrow lastrow'><span class='stratlabel'>Overall<br><span class='stratn'>(N=97)</span></span></th>
</tr>
</thead>
<tbody>
<tr>
<td class='rowlabel firstrow'>Age at Childbirth</td>
<td class='firstrow'></td>
</tr>
<tr>
<td class='rowlabel lastrow'>Mean (SD)</td>
<td class='lastrow'>31 (&plusmn; 5.5)</td>
</tr>
<tr>
<td class='rowlabel firstrow'>Sex</td>
<td class='firstrow'></td>
</tr>
<tr>
<td class='rowlabel lastrow'>Female</td>
<td class='lastrow'>97 (100%)</td>
</tr>
<tr>
<td class='rowlabel firstrow'>Gender</td>
<td class='firstrow'></td>
</tr>
<tr>
<td class='rowlabel lastrow'>Woman</td>
<td class='lastrow'>97 (100%)</td>
</tr>
<tr>
<td class='rowlabel firstrow'>Latino or Hispanic</td>
<td class='firstrow'></td>
</tr>
<tr>
<td class='rowlabel'>No</td>
<td>82 (84.5%)</td>
</tr>
<tr>
<td class='rowlabel lastrow'>Yes</td>
<td class='lastrow'>15 (15.5%)</td>
</tr>
<tr>
<td class='rowlabel firstrow'>Race</td>
<td class='firstrow'></td>
</tr>
<tr>
<td class='rowlabel'>Biracial/Multiracial</td>
<td>5 (5.2%)</td>
</tr>
<tr>
<td class='rowlabel'>Black</td>
<td>19 (19.6%)</td>
</tr>
<tr>
<td class='rowlabel'>Other</td>
<td>11 (11.3%)</td>
</tr>
<tr>
<td class='rowlabel lastrow'>White</td>
<td class='lastrow'>62 (63.9%)</td>
</tr>
<tr>
<td class='rowlabel firstrow'>Educational Attainment</td>
<td class='firstrow'></td>
</tr>
<tr>
<td class='rowlabel'>High School or GED</td>
<td>28 (28.9%)</td>
</tr>
<tr>
<td class='rowlabel'>Associate Degree</td>
<td>6 (6.2%)</td>
</tr>
<tr>
<td class='rowlabel'>Four-Year Degree</td>
<td>36 (37.1%)</td>
</tr>
<tr>
<td class='rowlabel'>Master's Degree</td>
<td>15 (15.5%)</td>
</tr>
<tr>
<td class='rowlabel lastrow'>Professional Degree or PhD</td>
<td class='lastrow'>12 (12.4%)</td>
</tr>
</tbody>
</table>
</div>
<div id="answer-to-environmental-health-question-3-7" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 3<a href="descriptive-cohort-analyses.html#answer-to-environmental-health-question-3-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can now answer <strong>Environmental Health Question #3</strong></em>: What are the demographics of the study participants?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> The study participants were all females who identified as women and were, on average, 31 years old when they gave birth. Participants were mostly non-latino/non-hispanic and White. Participants were spread across educational attainment levels, with the smallest education attainment group being those with an associate degree and the largest being those with a four-year degree.</p>
</div>
</div>
</div>
<div id="concluding-remarks-23" class="section level2 hasAnchor">
<h2>Concluding Remarks<a href="descriptive-cohort-analyses.html#concluding-remarks-23" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In conclusion, this training module serves as an introduction to human cohort data exploration and preliminary analysis, including data filtering, summary statistics, and multivariate outlier detection. These methods are an important step at the beginning of human cohort analyses, and the concepts introduced in this module can be applied to a wide variety of datasets.</p>
<p><br></p>
<p><label class="tykfont">
Test Your Knowledge
</label></p>
<div class="tyk">
<p>Using a more expanded demographics file (“Module6_1_TYKInput.xlsx”), create a demographics table with:</p>
<ul>
<li>The two new variables (home location and home type) included</li>
<li>The table split by which site the participant visited</li>
<li>Variable names and values presented in a publication-quality format (first letters capitalized, spaces between words, no underscores)</li>
</ul>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="omics-and-system-biology-transcriptomic-applications.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
