<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6.4 Mixtures Analysis Methods Part 2: Bayesian Kernel Machine Regression | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="6.4 Mixtures Analysis Methods Part 2: Bayesian Kernel Machine Regression | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6.4 Mixtures Analysis Methods Part 2: Bayesian Kernel Machine Regression | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  <meta name="twitter:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  

<meta name="author" content="Rager Lab" />


<meta name="date" content="2024-10-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="Icons_Used_Throughout/Favicon.png" type="image/x-icon" />
<link rel="prev" href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"/>
<link rel="next" href="mixtures-analysis-methods-part-3-sufficient-similarity.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>
<link href="libs/table1-1.0/table1_defaults.css" rel="stylesheet" />
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-FGFKVX2X3G"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-FGFKVX2X3G');
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="Icons_Used_Throughout/TAME 2.0 Logo.png"><br></a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Chapter 1 Introductory <br>Data Science</b></span></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html"><i class="fa fa-check"></i>FAIR Data Management Practices</a>
<ul>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-training-module"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-fair"><i class="fa fa-check"></i>Introduction to FAIR</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#breaking-down-fair-letter-by-letter"><i class="fa fa-check"></i>Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#what-does-this-mean-for-you"><i class="fa fa-check"></i>What Does This Mean for You?</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#data-repositories-for-sharing-of-data"><i class="fa fa-check"></i>Data Repositories for Sharing of Data</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#recent-shifts-in-regulatory-policies-for-data-sharing"><i class="fa fa-check"></i>Recent Shifts in Regulatory Policies for Data Sharing</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#additional-training-resources-on-fair"><i class="fa fa-check"></i>Additional Training Resources on FAIR</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html"><i class="fa fa-check"></i>Data Sharing through Online Repositories</a>
<ul>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#an-overview-and-example-with-the-dataverse-repository"><i class="fa fa-check"></i>An Overview and Example with the Dataverse Repository</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#introduction-to-training-module-1"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#data-repositories"><i class="fa fa-check"></i>Data Repositories</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#the-dataverse-project"><i class="fa fa-check"></i>The Dataverse Project</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#what-is-a-dataverse"><i class="fa fa-check"></i>What is a Dataverse?</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#metadata"><i class="fa fa-check"></i>Metadata</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataverse"><i class="fa fa-check"></i>Creating a Dataverse</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataset"><i class="fa fa-check"></i>Creating a Dataset</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#concluding-remarks-1"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html"><i class="fa fa-check"></i>File Management using Github</a>
<ul>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#introduction-to-training-module-2"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-an-account"><i class="fa fa-check"></i>Creating an Account</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-a-repository"><i class="fa fa-check"></i>Creating a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#uploading-code"><i class="fa fa-check"></i>Uploading Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#adding-subfolders-in-a-repository"><i class="fa fa-check"></i>Adding Subfolders in a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-code"><i class="fa fa-check"></i>Updating Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-repository-titles-and-structure-to-support-a-manuscript"><i class="fa fa-check"></i>Updating Repository Titles and Structure to Support a Manuscript</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#tracking-code-changes-using-github-branches"><i class="fa fa-check"></i>Tracking Code Changes using Github Branches</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#concluding-remarks-2"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html"><i class="fa fa-check"></i>Data Wrangling in Excel</a>
<ul>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#introduction-to-training-module-3"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#save-a-copy-of-the-soon-to-be-organized-and-cleaned-dataset-as-a-new-file"><i class="fa fa-check"></i>Save a Copy of the Soon-To-Be Organized and Cleaned Dataset as a New File</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-extraneous-white-space"><i class="fa fa-check"></i>Remove Extraneous White Space</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#replace-missing-data-with-na"><i class="fa fa-check"></i>Replace Missing Data with “NA”</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-metadata-tab"><i class="fa fa-check"></i>Create a Metadata Tab</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#abbreviate-and-capitalize-categorical-data"><i class="fa fa-check"></i>Abbreviate and Capitalize Categorical Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#alphabetize-sort-the-data-by-the-categorical-variable-of-interest"><i class="fa fa-check"></i>Alphabetize (Sort) the Data by the Categorical Variable of Interest</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-new-subject-number-column"><i class="fa fa-check"></i>Create a New Subject Number Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-special-symbols-and-dashes"><i class="fa fa-check"></i>Remove Special Symbols and Dashes</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#bold-all-column-names-and-center-all-data"><i class="fa fa-check"></i>Bold all Column Names and Center all Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-subject-identifier-column"><i class="fa fa-check"></i>Create a Subject Identifier Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#separate-subject-demographic-data-from-experimental-measurements"><i class="fa fa-check"></i>Separate Subject Demographic Data from Experimental Measurements</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#convert-data-from-wide-to-long-format"><i class="fa fa-check"></i>Convert Data from Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#pivoting-data-from-a-wide-to-long-format"><i class="fa fa-check"></i>Pivoting Data from a Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#generating-summary-level-statistics-with-pivot-tables"><i class="fa fa-check"></i>Generating Summary-Level Statistics with Pivot Tables</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#excel-vs.-r-which-should-you-use"><i class="fa fa-check"></i>Excel vs. R: Which Should You Use?</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#concluding-remarks-3"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 2 Coding in R</b></span></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html"><i class="fa fa-check"></i>Downloading and Programming in R</a>
<ul>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-training-module-4"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#general-introduction-and-installation-of-r-and-rstudio"><i class="fa fa-check"></i>General Introduction and Installation of R and RStudio</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-r-packages"><i class="fa fa-check"></i>Introduction to R Packages</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#scripting-basics"><i class="fa fa-check"></i>Scripting Basics</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#code-troubleshooting"><i class="fa fa-check"></i>Code Troubleshooting</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#concluding-remarks-4"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html"><i class="fa fa-check"></i>Coding “Best” Practices</a>
<ul>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#introduction-to-training-module-5"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#scripting-file-types"><i class="fa fa-check"></i>Scripting File Types</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-headers-and-annotation"><i class="fa fa-check"></i>Script Headers and Annotation</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#coding-style"><i class="fa fa-check"></i>Coding Style</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-organization"><i class="fa fa-check"></i>Script Organization</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#concluding-remarks-5"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html"><i class="fa fa-check"></i>Data Manipulation and Reshaping</a>
<ul>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-training-module-6"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#data-manipulation-using-base-r"><i class="fa fa-check"></i>Data Manipulation Using Base R</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-tidyverse"><i class="fa fa-check"></i>Introduction to Tidyverse</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#concluding-remarks-6"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html"><i class="fa fa-check"></i>Improving Coding Efficiencies</a>
<ul>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#introduction-to-training-module-7"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#loops"><i class="fa fa-check"></i>Loops</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#functions"><i class="fa fa-check"></i>Functions</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#list-operations"><i class="fa fa-check"></i>List operations</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#concluding-remarks-7"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#additional-resources-2"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="part"><span><b>Chapter 3 Basics of <br>Data Analysis and Visualizations</b></span></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html"><i class="fa fa-check"></i>Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-data-visualizations"><i class="fa fa-check"></i>Introduction to Data Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-training-module-8"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#density-plot-visualization"><i class="fa fa-check"></i>Density Plot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#boxplot-visualization"><i class="fa fa-check"></i>Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#correlation-visualizations"><i class="fa fa-check"></i>Correlation Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#heatmap-visualization"><i class="fa fa-check"></i>Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#concluding-remarks-8"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html"><i class="fa fa-check"></i>Improving Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-data-visulization-conventions"><i class="fa fa-check"></i>Introduction to Data Visulization Conventions</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-training-module-9"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-boxplot-visualization"><i class="fa fa-check"></i>Creating an Improved Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-heatmap-visualization"><i class="fa fa-check"></i>Creating an Improved Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-multi-plot-figures"><i class="fa fa-check"></i>Creating Multi-Plot Figures</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#concluding-remarks-9"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html"><i class="fa fa-check"></i>Normality Tests and Data Transformations</a>
<ul>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#introduction-to-training-module-10"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#what-is-a-normal-distribution"><i class="fa fa-check"></i>What is a Normal Distribution?</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#qualitative-assessment-of-normality"><i class="fa fa-check"></i>Qualitative Assessment of Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#quantitative-normality-assessment"><i class="fa fa-check"></i>Quantitative Normality Assessment</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#data-transformation"><i class="fa fa-check"></i>Data Transformation</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#additional-considerations-regarding-normality"><i class="fa fa-check"></i>Additional Considerations Regarding Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#concluding-remarks-10"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html"><i class="fa fa-check"></i>Intoduction to Statistical Tests</a>
<ul>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#introduction-to-training-module-11"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#assessing-normality-homogeneity-of-variance"><i class="fa fa-check"></i>Assessing Normality &amp; Homogeneity of Variance</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#two-group-visualizations-and-statistical-comparisons-using-the-t-test"><i class="fa fa-check"></i>Two-Group Visualizations and Statistical Comparisons using the T-Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#three-group-visualizations-and-statistical-comparisons-using-an-anova"><i class="fa fa-check"></i>Three-Group Visualizations and Statistical Comparisons using an ANOVA</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#regression-modeling-and-visualization-linear-and-logistic-regressions"><i class="fa fa-check"></i>Regression Modeling and Visualization: Linear and Logistic Regressions</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#statistical-evaluations-of-categorical-data-using-the-chi-squared-test-and-fishers-exact-test"><i class="fa fa-check"></i>Statistical Evaluations of Categorical Data using the Chi-Squared Test and Fisher’s Exact Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#concluding-remarks-11"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 4 Converting Wet Lab Data into Dry Lab Analyses</b></span></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html"><i class="fa fa-check"></i>Overview of Experimental Design and Example Data</a>
<ul>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#introduction-to-training-module-12"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#replicates"><i class="fa fa-check"></i>Replicates</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#orientation-to-example-data-for-chapter-4"><i class="fa fa-check"></i>Orientation to Example Data for Chapter 4</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#concluding-remarks-12"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html"><i class="fa fa-check"></i>Data Import, Processing, and Summary Statistics</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#introduction-to-training-module-13"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#data-import"><i class="fa fa-check"></i>Data Import</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#handling-missing-values"><i class="fa fa-check"></i>Handling Missing Values</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#averaging-replicates"><i class="fa fa-check"></i>Averaging Replicates</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#descriptive-statistics"><i class="fa fa-check"></i>Descriptive Statistics</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#normality-assessment-and-data-transformation"><i class="fa fa-check"></i>Normality Assessment and Data Transformation</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#concluding-remarks-13"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html"><i class="fa fa-check"></i>Data Import from PDF Sources</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#introduction-to-training-module-14"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-from-many-single-pdfs-with-the-same-formatting"><i class="fa fa-check"></i>Importing Data from Many Single PDFs with the Same Formatting</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-stored-in-pdf-tables"><i class="fa fa-check"></i>Importing Data Stored in PDF Tables</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#concluding-remarks-14"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html"><i class="fa fa-check"></i>Two-Group Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#introduction-to-training-module-15"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#overview-of-two-group-statistical-tests"><i class="fa fa-check"></i>Overview of Two Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#statistical-vs-biological-significance"><i class="fa fa-check"></i>Statistical vs Biological Significance</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#unpaired-test-example"><i class="fa fa-check"></i>Unpaired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#paired-test-example"><i class="fa fa-check"></i>Paired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#visualizing-results"><i class="fa fa-check"></i>Visualizing Results</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#concluding-remarks-15"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html"><i class="fa fa-check"></i>Multi-Group and Multi-Variable Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#introduction-to-training-module-16"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#overview-of-multi-group-statistical-tests"><i class="fa fa-check"></i>Overview of Multi-Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#multi-group-analysis-example"><i class="fa fa-check"></i>Multi-Group Analysis Example</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#visualization-of-multi-group-statistical-results"><i class="fa fa-check"></i>Visualization of Multi-Group Statistical Results</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#concluding-remarks-16"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 5 Machine Learning &amp; Artificial Intelligence</b></span></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><i class="fa fa-check"></i>Introduction to Artificial Intelligence, Machine Learning, and Predictive Modeling for Environmental Health</a>
<ul>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#introduction-to-training-module-17"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#general-historical-context-and-taxonomy-of-modern-aiml"><i class="fa fa-check"></i>General Historical Context and Taxonomy of Modern AI/ML</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#application-of-machine-learning-in-environmental-health-science"><i class="fa fa-check"></i>Application of Machine Learning in Environmental Health Science</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#concluding-remarks-17"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html"><i class="fa fa-check"></i>Supervised Machine Learning</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-training-module-18"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-machine-learning"><i class="fa fa-check"></i>Types of Machine Learning</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-supervised-machine-learning-algorithms"><i class="fa fa-check"></i>Types of Supervised Machine Learning Algorithms</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#training-supervised-machine-learning-models"><i class="fa fa-check"></i>Training Supervised Machine Learning Models</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#assessing-classification-based-model-performance"><i class="fa fa-check"></i>Assessing Classification-Based Model Performance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-activity-and-example-dataset"><i class="fa fa-check"></i>Introduction to Activity and Example Dataset</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#testing-for-differences-in-predictor-variables-across-the-outcome-classes"><i class="fa fa-check"></i>Testing for Differences in Predictor Variables across the Outcome Classes</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#predicting-ias-detection-with-a-random-forest-rf-model"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#class-imbalance"><i class="fa fa-check"></i>Class Imbalance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#concluding-remarks-18"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html"><i class="fa fa-check"></i>Supervised Machine Learning Model Interpretation</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-training-module-19"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance"><i class="fa fa-check"></i>Variable Importance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary"><i class="fa fa-check"></i>Decision Boundary</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-example-dataset-and-activity"><i class="fa fa-check"></i>Introduction to Example Dataset and Activity</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#predicting-ias-detection-with-a-random-forest-rf-model-1"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance-plot"><i class="fa fa-check"></i>Variable Importance Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot"><i class="fa fa-check"></i>Decision Boundary Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot-incorporating-smote"><i class="fa fa-check"></i>Decision Boundary Plot Incorporating SMOTE</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#concluding-remarks-19"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><i class="fa fa-check"></i>Unsupervised Machine Learning Part 1: K-Means Clustering &amp; PCA</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-training-module-20"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-unsupervised-machine-learning"><i class="fa fa-check"></i>Introduction to Unsupervised Machine Learning</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-example-data"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#identifying-clusters-of-chemicals-through-k-means"><i class="fa fa-check"></i>Identifying Clusters of Chemicals through <em>K</em>-Means</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#principal-component-analysis-pca-1"><i class="fa fa-check"></i>Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#incorporating-k-means-into-pca-for-predictive-modeling"><i class="fa fa-check"></i>Incorporating <em>K</em>-Means into PCA for Predictive Modeling</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#concluding-remarks-20"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><i class="fa fa-check"></i>Unsupervised Machine Learning Part 2: Additional Clustering Applications</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-training-module-21"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#k-means-clustering-1"><i class="fa fa-check"></i><em>K</em>-Means Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-example-data-1"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-1"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#clustering-plot"><i class="fa fa-check"></i>Clustering Plot</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-visualization"><i class="fa fa-check"></i>Hierarchical Clustering Visualization</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#variable-contributions"><i class="fa fa-check"></i>Variable Contributions</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#concluding-remarks-21"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 6 Applications in Toxicology &amp; Exposure Science</b></span></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html"><i class="fa fa-check"></i>6.1 Descriptive Cohort Analyses</a>
<ul>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#introduction-to-training-module-22"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#participant-exploration"><i class="fa fa-check"></i>Participant Exploration</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#chemical-detection-filtering"><i class="fa fa-check"></i>Chemical Detection Filtering</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#outlier-identification"><i class="fa fa-check"></i>Outlier Identification</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#summary-statistics-tables"><i class="fa fa-check"></i>Summary Statistics Tables</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#demographics-table"><i class="fa fa-check"></i>Demographics Table</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#concluding-remarks-22"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html"><i class="fa fa-check"></i>6.2 -Omics and System Biology: Transcriptomic Applications</a>
<ul>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-training-module-23"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-the-field-of--omics"><i class="fa fa-check"></i>Introduction to the Field of “-Omics”</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-transcriptomics"><i class="fa fa-check"></i>Introduction to Transcriptomics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#formatting-data-for-downstream-statistics"><i class="fa fa-check"></i>Formatting Data for Downstream Statistics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#transcriptomics-data-qaqc"><i class="fa fa-check"></i>Transcriptomics Data QA/QC</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#controlling-for-sources-of-sample-heterogeneity"><i class="fa fa-check"></i>Controlling for Sources of Sample Heterogeneity</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e.g.-biomass-smoke-exposure"><i class="fa fa-check"></i>Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure)</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-ma-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using MA Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-volcano-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using Volcano Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis"><i class="fa fa-check"></i>Interpretting Findings at the Systems Level through Pathway Enrichment Analysis</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#concluding-remarks-23"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><i class="fa fa-check"></i>6.3 Mixtures Analysis Methods Part 1: Overview and Example with Quantile G-Computation</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-training-module-24"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#overview-of-mixtures-analysis"><i class="fa fa-check"></i>Overview of Mixtures Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-example-data-2"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#mixtures-model-with-standard-qqcomp"><i class="fa fa-check"></i>Mixtures Model with Standard qqcomp</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#concluding-remarks-24"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#additional-resources-10"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><i class="fa fa-check"></i>6.4 Mixtures Analysis Methods Part 2: Bayesian Kernel Machine Regression</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-training-module-25"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-example-data-3"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-bkmr"><i class="fa fa-check"></i>Introduction to BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#run-bkmr"><i class="fa fa-check"></i>Run BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#concluding-remarks-25"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#additional-resources-11"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html"><i class="fa fa-check"></i>6.5 Mixtures Analysis Methods Part 3: Sufficient Similarity</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-training-module-26"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-toxicant-and-dataset"><i class="fa fa-check"></i>Introduction to Toxicant and Dataset</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#workspace-preparation-and-data-import-8"><i class="fa fa-check"></i>Workspace Preparation and Data Import</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#chemistry-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Chemistry-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#toxicity-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Toxicity-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#comparing-chemistry-vs.-toxicity-sufficient-similarity-analyses"><i class="fa fa-check"></i>Comparing Chemistry vs. Toxicity Sufficient Similarity Analyses</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#concluding-remarks-26"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i>6.6 Toxicokinetic Modeling</a>
<ul>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-training-module-27"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-toxicokinetic-modeling"><i class="fa fa-check"></i>Introduction to Toxicokinetic Modeling</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#data-and-models-used-in-toxicokinetic-modeling-tk"><i class="fa fa-check"></i>Data and Models used in Toxicokinetic Modeling (TK)</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-toxicokinetics"><i class="fa fa-check"></i>Reverse Toxicokinetics</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#capturing-population-variability-in-toxicokinetics-and-uncertainty-in-chemical-specific-parameters"><i class="fa fa-check"></i>Capturing Population Variability in Toxicokinetics, and Uncertainty in Chemical-Specific Parameters</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-tk-calculating-administered-equivalent-doses-for-toxcast-bioactive-concentrations"><i class="fa fa-check"></i>Reverse TK: Calculating Administered Equivalent Doses for ToxCast Bioactive Concentrations</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#comparing-equivalent-doses-estimated-to-elicit-toxicity-hazard-to-external-exposure-estimates-exposure-for-chemical-prioritization-by-bioactivity-exposure-ratios-bers"><i class="fa fa-check"></i>Comparing Equivalent Doses Estimated to Elicit Toxicity (Hazard) to External Exposure Estimates (Exposure), for Chemical Prioritization by Bioactivity-Exposure Ratios (BERs)</a></li>
</ul></li>
<li class="part"><span><b>Chapter 7 Environmental Health Database Mining</b></span></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression" class="section level1 hasAnchor">
<h1>6.4 Mixtures Analysis Methods Part 2: Bayesian Kernel Machine Regression<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This training module was developed by Dr. Lauren Eaves, Dr. Kyle Roell, and Dr. Julia E. Rager.</p>
<p>All input files (script, data, and figures) can be downloaded from the <a href="https://github.com/UNCSRP/TAME2">UNC-SRP TAME2 GitHub website</a>.</p>
<div id="introduction-to-training-module-25" class="section level2 hasAnchor">
<h2>Introduction to Training Module<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-training-module-25" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this training module, we will continue to explore mixtures analysis method, this time with a scripted example of Bayesian Kernel Machine Regression (BKMR). Please refer to <strong>TAME 2.0 Module 6.3, Mixtures Analysis Methods Part 1: Overview and Example with Quantile G-Computation</strong> for an overview of mixtures methodologies and a scripted example using Quantile g-Computation.</p>
</div>
<div id="introduction-to-example-data-3" class="section level2 hasAnchor">
<h2>Introduction to Example Data<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-example-data-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this scripted example, we will use a dataset from the <a href="https://elgan.fpg.unc.edu/">Extremely Low Gestational Age Newborn (ELGAN) cohort</a>. Specifically, we will analyze metal mixtures assessed in cord tissue collected at delivery with neonatal inflammation measured over the first two weeks of life.</p>
<p>For more information on the cord tissue metals data, please see the following two publications:</p>
<ul>
<li>Eaves LA, Bulka CM, Rager JE, Galusha AL, Parsons PJ, O’Shea TM and Fry RC. <a href="https://pubmed.ncbi.nlm.nih.gov/36493891/">Metals mixtures modeling identifies birth weight-associated gene networks in the placentas of children born extremely preterm</a>. Chemosphere. 2022;137469. <a href="doi:10.1016/j.chemosphere.2022.137469" class="uri">doi:10.1016/j.chemosphere.2022.137469</a>.</li>
<li>Bulka CM, Eaves LA, Gardner AJ, Parsons PJ, Kyle RR, Smeester L, O”Shea TM, Fry RC. <a href="https://pubmed.ncbi.nlm.nih.gov/36405975/">Prenatal exposure to multiple metallic and metalloid trace elements and the risk of bacterial sepsis in extremely low gestational age newborns: A prospective cohort study</a>. Front Epidemiol. 2022;2. <a href="doi:10.3389/fepid.2022.958389" class="uri">doi:10.3389/fepid.2022.958389</a></li>
</ul>
<p>For more information on the neonatal inflammation data, please see the following publication:</p>
<ul>
<li>Eaves LA, Enggasser AE, Camerota M, Gogcu S, Gower WA, Hartwell H, Jackson WM, Jensen E, Joseph RM, Marsit CJ, Roell K, Santos HP Jr, Shenberger JS, Smeester L, Yanni D, Kuban KCK, O’Shea TM, Fry RC. <a href="https://pubmed.ncbi.nlm.nih.gov/35764815/">CpG methylation patterns in placenta and neonatal blood are differentially associated with neonatal inflammation</a>. Pediatr Res. June 2022. <a href="doi:10.1038/s41390-022-02150-4" class="uri">doi:10.1038/s41390-022-02150-4</a></li>
</ul>
<p>Here, we have a dataset of n=254 participants for which we have complete data on neonatal inflammation, cord tissue metals and key demographic variables that will be included as confounders in the analysis.</p>
<p>Extensive research in the ELGAN study has demonstrated that neonatal inflammation is predictive of cerebral palsy, ASD, ADHD, obesity, cognitive impairment, attention problems,cerebral white matter damage, and decreased total brain volume, among other adverse outcomes. Therefore identifying exposures that lead to neonatal inflammation and could be intervened upon to reduce the risk of neonatal inflammation is critical to improve neonatal health. Environmental exposures during pregnancy such as metals may contribute to neonatal inflammation. As is often the case in environmental health, these chemical exposures are likely co-occurring and therefore mixtures methods are needed.</p>
</div>
<div id="introduction-to-bkmr" class="section level2 hasAnchor">
<h2>Introduction to BKMR<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-bkmr" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>BKMR offers a flexible, non-parametric method to estimate:</p>
<ol style="list-style-type: decimal">
<li>The single exposure effect: odds ratio of inflammation when a single exposure is at its 75th percentile compared to its 25th percentile, with other exposures at their 50th percentile and covariates held constant</li>
<li>The overall mixture effect: odds ratio of inflammation when all exposures are fixed at their 75th percentile compared to when all of the factors are fixed to their 25th percentile;</li>
<li>The interactive effect: the difference in the single-exposure effect when all of the other exposures are fixed at their 75th percentile, as compared to when all of the other factors are fixed at their 25th percentile;</li>
</ol>
<p>There are numerous excellent summaries of BKMR, including the publications in which it was first introduced:</p>
<ul>
<li>Bobb et al. <a href="https://academic.oup.com/biostatistics/article/16/3/493/269719">Bayesian kernel machine regression for estimating the health effects of multi-pollutant mixtures</a></li>
<li>Bobb et al. <a href="https://ehjournal.biomedcentral.com/articles/10.1186/s12940-018-0413-y">Statistical software for analyzing the health effects of multiple concurrent exposures via Bayesian kernel machine regression</a></li>
</ul>
<p>And other vignettes and toolkits including:</p>
<ul>
<li>Jennifer Bobb’s <a href="https://jenfb.github.io/bkmr/overview.html">Introduction to Bayesian kernel machine regression and the bkmr R package</a></li>
<li>Andrea Bellavia’s <a href="https://bookdown.org/andreabellavia/mixtures/bayesian-kernel-machine-regression.html">Bayesian kernel machine regression</a></li>
</ul>
<p>While BKMR can do many things other methods cannot, it can require a lot of computational resources and take a long time to run. Before working with your final dataset and analysis, if very large or complex, it is often recommended to start with a smaller sample to make sure everything is working correctly before starting an analysis that make takes days to complete.</p>
<div id="training-modules-environmental-health-questions-17" class="section level3 hasAnchor">
<h3>Training Module’s <strong>Environmental Health Questions</strong><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#training-modules-environmental-health-questions-17" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This training module was specifically developed to answer the following questions, which mirror the questions in <strong>TAME 2.0 Module 6.3</strong>, but are just in a different order:</p>
<ol style="list-style-type: decimal">
<li>Which of these chemicals has the strongest effect on neonatal inflammation risk?</li>
<li>Which of these chemicals increases the risk of neonatal inflammation and which decreases the risk of neonatal inflammation?</li>
<li>What is the risk of neonatal inflammation associated with exposure to each of manganese, copper, zinc, arsenic, selenium, cadmium, mercury, lead individually?</li>
<li>What is the risk of neonatal inflammation associated with combined exposure to manganese, copper, zinc, arsenic, selenium, cadmium, mercury, lead (ie. a mixture)?
and in addition to the questions addressed in Mixtures Methods 1, we additionally can answer:</li>
<li>Are there interactions among manganese, copper, zinc, arsenic, selenium, cadmium, mercury, lead in relation to neonatal inflammation?</li>
</ol>
</div>
</div>
<div id="run-bkmr" class="section level2 hasAnchor">
<h2>Run BKMR<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#run-bkmr" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="workspace-preparation-1" class="section level3 hasAnchor">
<h3>Workspace Preparation<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#workspace-preparation-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Install packages as needed, then load the following packages:</p>
<div class="sourceCode" id="cb718"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb718-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb718-1" tabindex="-1"></a><span class="co">#load packages</span></span>
<span id="cb718-2"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb718-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb718-3"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb718-3" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb718-4"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb718-4" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb718-5"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb718-5" tabindex="-1"></a><span class="fu">library</span>(yaml)</span>
<span id="cb718-6"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb718-6" tabindex="-1"></a><span class="fu">library</span>(rmarkdown)</span>
<span id="cb718-7"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb718-7" tabindex="-1"></a><span class="fu">library</span>(broom)</span>
<span id="cb718-8"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb718-8" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb718-9"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb718-9" tabindex="-1"></a><span class="fu">library</span>(bkmr)</span></code></pre></div>
<p>Optionally, you can also create a current date variable to name output files, and create an output folder.</p>
<div class="sourceCode" id="cb719"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb719-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb719-1" tabindex="-1"></a><span class="co">#Create a current date variable to name outputfiles</span></span>
<span id="cb719-2"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb719-2" tabindex="-1"></a>cur_date <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(<span class="fu">Sys.Date</span>(),<span class="st">&quot;-&quot;</span>,<span class="st">&quot;&quot;</span>)</span>
<span id="cb719-3"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb719-3" tabindex="-1"></a></span>
<span id="cb719-4"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb719-4" tabindex="-1"></a><span class="co">#Create an output folder </span></span>
<span id="cb719-5"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb719-5" tabindex="-1"></a>Output_Folder <span class="ot">&lt;-</span> (<span class="st">&quot;Module6_4_Output/&quot;</span>)</span></code></pre></div>
</div>
<div id="data-import-2" class="section level3 hasAnchor">
<h3>Data Import<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#data-import-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb720"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb720-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb720-1" tabindex="-1"></a>cohort <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file=</span><span class="st">&quot;Module6_4_Input/Module6_4_InputData.csv&quot;</span>)</span>
<span id="cb720-2"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb720-2" tabindex="-1"></a><span class="fu">colnames</span>(cohort)</span></code></pre></div>
<pre><code>##  [1] &quot;X&quot;              &quot;id&quot;             &quot;inflam_intense&quot; &quot;race1&quot;         
##  [5] &quot;sex&quot;            &quot;gadays&quot;         &quot;magecat&quot;        &quot;medu&quot;          
##  [9] &quot;smoke&quot;          &quot;Mn_ugg&quot;         &quot;Cu_ugg&quot;         &quot;Zn_ugg&quot;        
## [13] &quot;As_ngg&quot;         &quot;Se_ugg&quot;         &quot;Cd_ngg&quot;         &quot;Hg_ngg&quot;        
## [17] &quot;Pb_ngg&quot;</code></pre>
<div class="sourceCode" id="cb722"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb722-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb722-1" tabindex="-1"></a><span class="fu">head</span>(cohort)</span></code></pre></div>
<pre><code>##   X      id inflam_intense race1 sex gadays magecat medu smoke Mn_ugg Cu_ugg
## 1 1 1100751              0     3   1    195       2    1     0  0.272  3.229
## 2 3 1100841              0     2   0    171       3    3     0  0.505  2.868
## 3 4 1100852              0     1   0    171       3    3     0  0.387  5.504
## 4 5 1100853              0     1   0    171       3    3     0  0.441  4.295
## 5 6 1100872              0     1   1    167       2    4     0  0.457 14.310
## 6 7 1100881              0     1   1    194       2    2     0  0.388  3.816
##   Zn_ugg As_ngg Se_ugg Cd_ngg Hg_ngg Pb_ngg
## 1   77.6   6.29   0.86   0.78  17.23   26.2
## 2   58.1   8.04   0.93   3.79  20.35  305.3
## 3   60.9   3.48   1.02   1.19  20.30   50.3
## 4   56.8   4.63   1.08   1.37  21.06   26.5
## 5  133.4   7.48   1.98   3.74  12.33   72.9
## 6   81.4   3.89   1.04   1.17  10.54   35.3</code></pre>
<p>The variables in this dataset include sample and demographic information and cort tissue metal exposure in <span class="math inline">\(mu\)</span>g/g or ng/g.</p>
<p><em>Sample and Demographic Variables</em></p>
<ul>
<li>“id”: unique study ID<br />
outcome:</li>
<li>“inflam_intense”: 1= high inflammation, 0=low inflammation
covariates:</li>
<li>“race1”: maternal race, 1=White, 2=Black, 0=Other<br />
</li>
<li>“sex”: neonatal sex, 0=female, 1=male<br />
</li>
<li>“gadays”: gestational age at delivery in days<br />
</li>
<li>“magecat”: maternal age, 1= &lt;21, 2=21-35, 3= &gt; 35<br />
</li>
<li>“medu”:maternal education: 1= &lt;12, 2=12, 3=13-15, 4=16, 5= &gt;16<br />
</li>
<li>“smoke”: maternal smoking while pregnant, 0=no, 1=yes</li>
</ul>
<p><em>Exposure Variables</em></p>
<ul>
<li>“Mn_ugg”<br />
</li>
<li>“Cu_ugg”<br />
</li>
<li>“Zn_ugg”<br />
</li>
<li>“As_ngg”<br />
</li>
<li>“Se_ugg”<br />
</li>
<li>“Cd_ngg”<br />
</li>
<li>“Hg_ngg”<br />
</li>
<li>“Pb_ngg”</li>
</ul>
<p>There are many steps prior to the modeling steps outlined below. These are being skipped for educational purposes. Additional steps include assessment of normality and transformations as needed, generation of a demographics table and assessing for missing data, imputation of missing data if needed, visualizing trends and distributions in the data, assessing correlations between exposures, functional form assessments, and decisions regarding what confounders to include.</p>
<p>In addition, it is highly recommended to conduct single-contaminant modeling initially to understand individual chemical relationships with the outcomes of focus before conducting mixtures assessment. For an example of this, see <strong>TAME 2.0 Module 6.3, Mixtures Analysis Methods Part 1: Overview and Example with Quantile G-Computation</strong>. BKMR, as a flexible non-parametric modeling approach, does not allow for classical null-hypothesis testing, and 95% CI are interpreted as credible intervals, not confidence intervals. One approach therefore could be to explore non-linearities and interactions within BKMR to then validate generated hypotheses using quantile g-computation.</p>
</div>
<div id="fit-the-bkmr-model" class="section level3 hasAnchor">
<h3>Fit the BKMR Model<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#fit-the-bkmr-model" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First, define a matrix/vector of the exposure mixture, outcome, and confounders/covariates. BKMR performs better when the exposures are on a similar scale and when there are not outliers. Thus, we center and scale the exposure variables first. As noted above, in a complete analysis, thorough examination of exposure variable distributions, including outliers and normality, would be conducted before any exposure-outcome modeling. For more information on normality testing, see <strong>TAME 2.0 Module 3.3, Normality Tests and Data Transformations.</strong></p>
<p>First, we’ll assign the matrix variables to their own data frame and scale the data.</p>
<div class="sourceCode" id="cb724"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb724-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb724-1" tabindex="-1"></a><span class="co">#exposure mixture variables</span></span>
<span id="cb724-2"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb724-2" tabindex="-1"></a>mixture <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(cohort[,<span class="dv">10</span><span class="sc">:</span><span class="dv">17</span>])</span>
<span id="cb724-3"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb724-3" tabindex="-1"></a>mixture <span class="ot">&lt;-</span> <span class="fu">log</span>(mixture)</span>
<span id="cb724-4"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb724-4" tabindex="-1"></a>mixture <span class="ot">&lt;-</span><span class="fu">scale</span>(mixture, <span class="at">center=</span><span class="cn">TRUE</span>)</span>
<span id="cb724-5"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb724-5" tabindex="-1"></a><span class="fu">summary</span>(mixture)</span></code></pre></div>
<pre><code>##      Mn_ugg            Cu_ugg            Zn_ugg            As_ngg       
##  Min.   :-3.3963   Min.   :-2.6253   Min.   :-2.0090   Min.   :-1.9903  
##  1st Qu.:-0.5536   1st Qu.:-0.6558   1st Qu.:-0.5211   1st Qu.:-0.6150  
##  Median :-0.1527   Median :-0.1452   Median :-0.1559   Median :-0.1327  
##  Mean   : 0.0000   Mean   : 0.0000   Mean   : 0.0000   Mean   : 0.0000  
##  3rd Qu.: 0.4048   3rd Qu.: 0.6038   3rd Qu.: 0.3180   3rd Qu.: 0.5385  
##  Max.   : 7.3059   Max.   : 4.3512   Max.   : 5.4446   Max.   : 3.8928  
##      Se_ugg              Cd_ngg            Hg_ngg             Pb_ngg       
##  Min.   :-3.468991   Min.   :-1.5007   Min.   :-2.34070   Min.   :-2.0013  
##  1st Qu.:-0.567767   1st Qu.:-0.6649   1st Qu.:-0.70797   1st Qu.:-0.6711  
##  Median : 0.005317   Median :-0.2632   Median :-0.01572   Median :-0.1036  
##  Mean   : 0.000000   Mean   : 0.0000   Mean   : 0.00000   Mean   : 0.0000  
##  3rd Qu.: 0.629320   3rd Qu.: 0.3312   3rd Qu.: 0.67767   3rd Qu.: 0.5312  
##  Max.   : 4.328620   Max.   : 5.3945   Max.   : 2.74547   Max.   : 3.8156</code></pre>
<p>Then, we’ll define the outcome variable and ensure it is the proper class and leveling.</p>
<div class="sourceCode" id="cb726"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb726-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb726-1" tabindex="-1"></a><span class="co">#outcome variable</span></span>
<span id="cb726-2"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb726-2" tabindex="-1"></a>cohort<span class="sc">$</span>inflam_intense  <span class="ot">&lt;-</span><span class="fu">as.factor</span>(cohort<span class="sc">$</span>inflam_intense)</span>
<span id="cb726-3"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb726-3" tabindex="-1"></a>cohort<span class="sc">$</span>inflam_intense <span class="ot">&lt;-</span> <span class="fu">relevel</span>(cohort<span class="sc">$</span>inflam_intense, <span class="at">ref =</span> <span class="st">&quot;0&quot;</span>)</span>
<span id="cb726-4"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb726-4" tabindex="-1"></a>y<span class="ot">&lt;-</span><span class="fu">as.numeric</span>(<span class="fu">as.character</span>(cohort<span class="sc">$</span>inflam_intense))</span></code></pre></div>
<p>Next, we’ll assign the covariates to a matrix.</p>
<div class="sourceCode" id="cb727"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb727-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb727-1" tabindex="-1"></a><span class="co">#covariates</span></span>
<span id="cb727-2"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb727-2" tabindex="-1"></a>covariates<span class="ot">&lt;-</span><span class="fu">as.matrix</span>(cohort[,<span class="dv">7</span><span class="sc">:</span><span class="dv">9</span>])</span></code></pre></div>
<p>Then, we can fit the BKMR model. Note that this script will take a few minutes to run.</p>
<div class="sourceCode" id="cb728"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb728-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb728-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">111</span>)</span>
<span id="cb728-2"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb728-2" tabindex="-1"></a>fitkm <span class="ot">&lt;-</span> <span class="fu">kmbayes</span>(<span class="at">y =</span> y, <span class="at">Z =</span> mixture, <span class="at">X =</span> covariates, <span class="at">iter =</span> <span class="dv">5000</span>, <span class="at">verbose =</span> <span class="cn">FALSE</span>, <span class="at">varsel =</span> <span class="cn">TRUE</span>, <span class="at">family=</span><span class="st">&quot;binomial&quot;</span>, <span class="at">est.h =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<pre><code>## Fitting probit regression model</code></pre>
<pre><code>## Iteration: 500 (10% completed; 18.60599 secs elapsed)</code></pre>
<pre><code>## Iteration: 1000 (20% completed; 42.14896 secs elapsed)</code></pre>
<pre><code>## Iteration: 1500 (30% completed; 1.19931 mins elapsed)</code></pre>
<pre><code>## Iteration: 2000 (40% completed; 1.67059 mins elapsed)</code></pre>
<pre><code>## Iteration: 2500 (50% completed; 2.16125 mins elapsed)</code></pre>
<pre><code>## Iteration: 3000 (60% completed; 2.64141 mins elapsed)</code></pre>
<pre><code>## Iteration: 3500 (70% completed; 3.11403 mins elapsed)</code></pre>
<pre><code>## Iteration: 4000 (80% completed; 3.54858 mins elapsed)</code></pre>
<pre><code>## Iteration: 4500 (90% completed; 4.03024 mins elapsed)</code></pre>
<pre><code>## Iteration: 5000 (100% completed; 4.49908 mins elapsed)</code></pre>
<p>For full information regarding options for the kmbayes function, refer to the BKMR reference manual: <a href="https://cran.r-project.org/web/packages/bkmr/bkmr.pdf" class="uri">https://cran.r-project.org/web/packages/bkmr/bkmr.pdf</a></p>
</div>
<div id="assess-variable-importance" class="section level3 hasAnchor">
<h3>Assess Variable Importance<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#assess-variable-importance" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>BKMR conducts a variable selection procedure and generates posterior inclusion probabilities (PIP). The larger the PIP, the more a variable is contributing to the overall exposure-outcome effect. These are relative to each other,so there is no threshold as to when a variable becomes an “important” contributor (similar to the weights in quantile g-computation).</p>
<div class="sourceCode" id="cb740"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb740-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb740-1" tabindex="-1"></a><span class="fu">ExtractPIPs</span>(fitkm)</span></code></pre></div>
<pre><code>##   variable    PIP
## 1   Mn_ugg 0.1468
## 2   Cu_ugg 0.1568
## 3   Zn_ugg 0.4760
## 4   As_ngg 0.0820
## 5   Se_ugg 0.1668
## 6   Cd_ngg 0.3204
## 7   Hg_ngg 0.1848
## 8   Pb_ngg 0.1676</code></pre>
<p>Relative to each other, the contributions to the effect of the mixture on neonatal inflammation are shown above for each component of the mixture. Note that if a variable PIP=0, BKMR will drop it from the model and the overall mixture effect will not include this exposure.</p>
</div>
<div id="answer-to-environmental-health-question-1-13" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 1<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#answer-to-environmental-health-question-1-13" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #1</strong></em>: Which of these chemicals has the strongest effect on neonatal inflammation risk?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: Based on the PIPs: Cadmium.</p>
</div>
</div>
<div id="assess-model-convergence" class="section level3 hasAnchor">
<h3>Assess Model Convergence<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#assess-model-convergence" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We can use trace plots to evaluate how the parameters in the model converge over the many iterations. We hope to see that the line moves randomly but centers around a straight line</p>
<div class="sourceCode" id="cb742"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb742-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb742-1" tabindex="-1"></a>sel<span class="ot">&lt;-</span><span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">5000</span>,<span class="at">by=</span><span class="dv">1</span>)</span>
<span id="cb742-2"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb742-2" tabindex="-1"></a><span class="fu">TracePlot</span>(<span class="at">fit =</span> fitkm, <span class="at">par =</span> <span class="st">&quot;beta&quot;</span>, <span class="at">sel=</span>sel)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-560-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Based on this plot, it looks like the burn in period is roughly 1000 iterations. We will remove these from the results.</p>
<div class="sourceCode" id="cb743"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb743-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb743-1" tabindex="-1"></a>sel<span class="ot">&lt;-</span><span class="fu">seq</span>(<span class="dv">1000</span>,<span class="dv">5000</span>,<span class="at">by=</span><span class="dv">1</span>)</span>
<span id="cb743-2"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb743-2" tabindex="-1"></a><span class="fu">TracePlot</span>(<span class="at">fit =</span> fitkm, <span class="at">par =</span> <span class="st">&quot;beta&quot;</span>, <span class="at">sel=</span>sel)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-561-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="presenting-model-results" class="section level3 hasAnchor">
<h3>Presenting Model Results<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#presenting-model-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="single-exposure-effects" class="section level4 hasAnchor">
<h4>Single exposure effects<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#single-exposure-effects" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>As described above, one way to examine single effects is to calculate the odds ratio of inflammation when a single exposure is at its 75th percentile compared to its 25th percentile, with other exposures are at their 50th percentile and covariates are held constant.</p>
<p>Here, we use the <code>PredictorResponseUnivar()</code> function to generate a dataset that details, at varying levels of each exposure (“z”), the relationship between that exposure and the outcome, holding other exposures at their 50th percentile and covariates constant. This relationship is given by a beta value, which because we have a binomial outcome and fit a probit model represents the log(odds) (“est”). The standard error for the beta value is also calculated (“se”).</p>
<div class="sourceCode" id="cb744"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb744-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb744-1" tabindex="-1"></a>pred.resp.univar <span class="ot">&lt;-</span> <span class="fu">PredictorResponseUnivar</span>(<span class="at">fit=</span>fitkm, <span class="at">sel=</span>sel, </span>
<span id="cb744-2"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb744-2" tabindex="-1"></a>                                            <span class="at">method=</span><span class="st">&quot;approx&quot;</span>, <span class="at">q.fixed =</span> <span class="fl">0.5</span>)</span>
<span id="cb744-3"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb744-3" tabindex="-1"></a></span>
<span id="cb744-4"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb744-4" tabindex="-1"></a><span class="fu">head</span>(pred.resp.univar)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 4
##   variable     z    est    se
##   &lt;fct&gt;    &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;
## 1 Mn_ugg   -3.40 -0.150 0.246
## 2 Mn_ugg   -3.18 -0.145 0.231
## 3 Mn_ugg   -2.96 -0.140 0.216
## 4 Mn_ugg   -2.74 -0.135 0.201
## 5 Mn_ugg   -2.52 -0.130 0.187
## 6 Mn_ugg   -2.30 -0.125 0.173</code></pre>
<p>We can then plot these data for each exposure to visualize the exposure-response function for each exposure.</p>
<div class="sourceCode" id="cb746"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb746-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb746-1" tabindex="-1"></a><span class="fu">ggplot</span>(pred.resp.univar, <span class="fu">aes</span>(z, est, <span class="at">ymin =</span> est <span class="sc">-</span> <span class="fl">1.96</span><span class="sc">*</span>se, </span>
<span id="cb746-2"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb746-2" tabindex="-1"></a>                             <span class="at">ymax =</span> est <span class="sc">+</span> <span class="fl">1.96</span><span class="sc">*</span>se)) <span class="sc">+</span> </span>
<span id="cb746-3"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb746-3" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;h(z)&quot;</span>) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span> variable) </span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-563-1.png" width="672" /></p>
<p>Then, we can generate a dataset that contains for each exposure (“variable”), the log(OR) (“est”) (and its standard deviation (“sd”)) corresponding to the odds of neonatal inflammation when an exposure is at its 75th compared to the odds when at the 25th percentile. The log(OR) is estimated at three levels of the other exposures (25th, 50th and 75th percentiles). We can use this dataset to identify odds ratios for neonatal inflammation (comparing the 75th to 25th percentile odds) for each exposure at differing levels of the other exposures. These odds ratios approximate risk, whereby an odds ratio &gt;1 means there is increased risk of neonatal inflammation when that exposure is at its 75th percentile compared to its 25th percentile. We can then plot these data to see the logOR for each metal in relation to neonatal inflammation at varying levels of the rest of the exposures.</p>
<div class="sourceCode" id="cb747"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb747-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb747-1" tabindex="-1"></a>risks.singvar <span class="ot">&lt;-</span> <span class="fu">SingVarRiskSummaries</span>(<span class="at">fit=</span>fitkm, <span class="at">qs.diff =</span> <span class="fu">c</span>(<span class="fl">0.25</span>, <span class="fl">0.75</span>),</span>
<span id="cb747-2"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb747-2" tabindex="-1"></a>                                      <span class="at">q.fixed =</span> <span class="fu">c</span>(<span class="fl">0.25</span>, <span class="fl">0.50</span>, <span class="fl">0.75</span>), </span>
<span id="cb747-3"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb747-3" tabindex="-1"></a>                                      <span class="at">method =</span> <span class="st">&quot;approx&quot;</span>)</span>
<span id="cb747-4"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb747-4" tabindex="-1"></a></span>
<span id="cb747-5"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb747-5" tabindex="-1"></a><span class="fu">ggplot</span>(risks.singvar, <span class="fu">aes</span>(variable, est, <span class="at">ymin =</span> est <span class="sc">-</span> <span class="fl">1.96</span><span class="sc">*</span>sd,  </span>
<span id="cb747-6"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb747-6" tabindex="-1"></a>                          <span class="at">ymax =</span> est <span class="sc">+</span> <span class="fl">1.96</span><span class="sc">*</span>sd, <span class="at">col =</span> q.fixed)) <span class="sc">+</span> </span>
<span id="cb747-7"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb747-7" tabindex="-1"></a>       <span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept=</span><span class="dv">0</span>), <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;gray&quot;</span>) <span class="sc">+</span>  </span>
<span id="cb747-8"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb747-8" tabindex="-1"></a>       <span class="fu">geom_pointrange</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.75</span>)) <span class="sc">+</span></span>
<span id="cb747-9"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb747-9" tabindex="-1"></a>       <span class="fu">coord_flip</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&quot;none&quot;</span>)<span class="sc">+</span><span class="fu">scale_x_discrete</span>(<span class="at">name=</span><span class="st">&quot;&quot;</span>) <span class="sc">+</span></span>
<span id="cb747-10"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb747-10" tabindex="-1"></a>       <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">&quot;estimate&quot;</span>) </span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-564-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="answer-to-environmental-health-question-2-14" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 2<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#answer-to-environmental-health-question-2-14" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #2</strong></em>: Which of these chemicals increases the risk of neonatal inflammation and which decreases the risk of neonatal inflammation?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: At all levels of the other exposures, lead, cadmium, selenium, arsenic and zinc reduce the odds of neonatal inflammation, while manganese and mercury appear to increase the odds of neonatal inflammation. Copper appears has a null effect. Notice that the credibility intervals however for all metals span the null meaning we are not confident in the independent effect of any of the metals.</p>
</div>
</div>
<div id="answer-to-environmental-health-question-3-10" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 3<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#answer-to-environmental-health-question-3-10" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can also answer also <strong>Environmental Health Question #3</strong></em>: What is the risk of neonatal inflammation associated with exposure to each of manganese, copper, zinc, arsenic, selenium, cadmium, mercury, lead individually?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: As an example, take manganese: when all other exposures are at their 50th percentile, the log(OR) for Mn comparing being at the 75th to the 25th percentile is 0.024, which equals an odds ratio of 1.02. From this, you should be able to calculate the odds ratios for the other metals yourself.</p>
</div>
<div id="calculating-the-overall-mixture-effect" class="section level4 hasAnchor">
<h4>Calculating the overall mixture effect<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#calculating-the-overall-mixture-effect" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Next, we can generate a dataset that details the effect (ie. log(OR) (“est”) and corresponding standard deviation (“sd”)) on neonatal inflammation of all exposures when at a particular quantile (“quantile”) compared to all exposures being at the 50th percentile. We can use this dataset to identify odds ratios for neonatal inflammation upon simultaneous exposure to the entire mixture for different quantile threshold comparisons. These odds ratios approximate risk, whereby an odds ratio &gt;1 means there is increased risk of neonatal inflammation when the entire mixture is set at the index quantile, compared to the 50th percentile. We can also plot these results to visualize the overall mixture effect dose-response relationship.</p>
<div class="sourceCode" id="cb748"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb748-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb748-1" tabindex="-1"></a>risks.overall <span class="ot">&lt;-</span> <span class="fu">OverallRiskSummaries</span>(<span class="at">fit=</span>fitkm, <span class="at">qs=</span><span class="fu">seq</span>(<span class="fl">0.25</span>, <span class="fl">0.75</span>, <span class="at">by=</span><span class="fl">0.05</span>), </span>
<span id="cb748-2"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb748-2" tabindex="-1"></a>                                      <span class="at">q.fixed =</span> <span class="fl">0.5</span>, <span class="at">method =</span> <span class="st">&quot;approx&quot;</span>,</span>
<span id="cb748-3"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb748-3" tabindex="-1"></a>                                      <span class="at">sel=</span>sel)</span>
<span id="cb748-4"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb748-4" tabindex="-1"></a></span>
<span id="cb748-5"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb748-5" tabindex="-1"></a><span class="fu">ggplot</span>(risks.overall, <span class="fu">aes</span>(quantile, est, <span class="at">ymin =</span> est <span class="sc">-</span> <span class="fl">1.96</span><span class="sc">*</span>sd, </span>
<span id="cb748-6"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb748-6" tabindex="-1"></a>                          <span class="at">ymax =</span> est <span class="sc">+</span> <span class="fl">1.96</span><span class="sc">*</span>sd)) <span class="sc">+</span>  </span>
<span id="cb748-7"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb748-7" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">00</span>, <span class="at">linetype=</span><span class="st">&quot;dashed&quot;</span>, <span class="at">color=</span><span class="st">&quot;gray&quot;</span>) <span class="sc">+</span> </span>
<span id="cb748-8"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb748-8" tabindex="-1"></a>  <span class="fu">geom_pointrange</span>() <span class="sc">+</span> <span class="fu">scale_y_continuous</span>(<span class="at">name=</span><span class="st">&quot;estimate&quot;</span>) </span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-565-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="answer-to-environmental-health-question-4-7" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 4<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#answer-to-environmental-health-question-4-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #4</strong></em>: What is the risk of neonatal inflammation associated with combined exposure to manganese, copper, zinc, arsenic, selenium, cadmium, mercury, lead (ie. a mixture)?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: When every exposure is at its 25th percentile concentration compared to their 50th percentile concentration, the odds ratio for neonatal inflammation is 1.11 (exp(0.10073680)). When every exposure is at its 75th percentile concentration compared to their 50th percentile concentration, the odds ratio for neonatal inflammation is 0.87 (exp(-0.12000889)).</p>
</div>
</div>
<div id="evaluating-interactive-effects" class="section level3 hasAnchor">
<h3>Evaluating interactive effects<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#evaluating-interactive-effects" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To understand bivariate interactions, we can generate a dataset that for each pairing of exposures details at varying levels of both exposures, the log(odds) (“est”, and associated standard deviation (“sd”)) of neonatal inflammation when all the other exposures are held constant. These plots can be tricky to interpret, so another way of looking at these results is to take “cross sections” at specific quantiles of the second exposure (see next step).</p>
<div class="sourceCode" id="cb749"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb749-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb749-1" tabindex="-1"></a>pred.resp.bivar  <span class="ot">&lt;-</span> <span class="fu">PredictorResponseBivar</span>(<span class="at">fit=</span>fitkm,  <span class="at">min.plot.dist =</span> <span class="dv">1</span>, </span>
<span id="cb749-2"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb749-2" tabindex="-1"></a>                                           <span class="at">sel=</span>sel,  <span class="at">method=</span><span class="st">&quot;approx&quot;</span>)</span>
<span id="cb749-3"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb749-3" tabindex="-1"></a></span>
<span id="cb749-4"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb749-4" tabindex="-1"></a><span class="fu">ggplot</span>(pred.resp.bivar, <span class="fu">aes</span>(z1, z2, <span class="at">fill =</span> est)) <span class="sc">+</span> </span>
<span id="cb749-5"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb749-5" tabindex="-1"></a>  <span class="fu">geom_raster</span>() <span class="sc">+</span> </span>
<span id="cb749-6"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb749-6" tabindex="-1"></a>  <span class="fu">facet_grid</span>(variable2 <span class="sc">~</span> variable1) <span class="sc">+</span></span>
<span id="cb749-7"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb749-7" tabindex="-1"></a>  <span class="fu">scale_fill_gradientn</span>(<span class="at">colours=</span><span class="fu">c</span>(<span class="st">&quot;#0000FFFF&quot;</span>,<span class="st">&quot;#FFFFFFFF&quot;</span>,<span class="st">&quot;#FF0000FF&quot;</span>)) <span class="sc">+</span></span>
<span id="cb749-8"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb749-8" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;expos1&quot;</span>) <span class="sc">+</span></span>
<span id="cb749-9"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb749-9" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;expos2&quot;</span>) <span class="sc">+</span></span>
<span id="cb749-10"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb749-10" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;h(expos1, expos2)&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-566-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Next, we generate a dataset that includes for each pairing of exposures, the log(odds) (“est” and associated standard deviation “sd”) of neonatal inflammation at varying concentrations (“z1”) of the first exposure (“variable 1”) when the second exposure (“variable 2” is at its 25th, 50th and 75th percentile (“quantile”).</p>
<div class="sourceCode" id="cb750"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb750-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb750-1" tabindex="-1"></a>pred.resp.bivar.levels <span class="ot">&lt;-</span> <span class="fu">PredictorResponseBivarLevels</span>(<span class="at">pred.resp.df=</span> </span>
<span id="cb750-2"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb750-2" tabindex="-1"></a>                        pred.resp.bivar,   <span class="at">Z =</span> mixture, <span class="at">both_pairs=</span><span class="cn">TRUE</span>, </span>
<span id="cb750-3"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb750-3" tabindex="-1"></a>                        <span class="at">qs =</span> <span class="fu">c</span>(<span class="fl">0.25</span>, <span class="fl">0.5</span>, <span class="fl">0.75</span>))</span>
<span id="cb750-4"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb750-4" tabindex="-1"></a></span>
<span id="cb750-5"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb750-5" tabindex="-1"></a><span class="fu">ggplot</span>(pred.resp.bivar.levels, <span class="fu">aes</span>(z1, est)) <span class="sc">+</span> </span>
<span id="cb750-6"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb750-6" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="fu">aes</span>(<span class="at">col =</span> quantile), <span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>) <span class="sc">+</span> </span>
<span id="cb750-7"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb750-7" tabindex="-1"></a> <span class="fu">facet_grid</span>(variable2 <span class="sc">~</span> variable1) <span class="sc">+</span></span>
<span id="cb750-8"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb750-8" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;h(expos1 | quantiles of expos2)&quot;</span>) <span class="sc">+</span></span>
<span id="cb750-9"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb750-9" tabindex="-1"></a> <span class="fu">xlab</span>(<span class="st">&quot;expos1&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-567-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>There is evidence of an interactive effect between two exposures when the exposure-response function for exposure 1 varies in form between the different quantiles of exposure 2. You can also zoom in on one plot, for example:</p>
<div class="sourceCode" id="cb751"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb751-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb751-1" tabindex="-1"></a>HgCd <span class="ot">&lt;-</span> pred.resp.bivar.levels <span class="sc">%&gt;%</span> </span>
<span id="cb751-2"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb751-2" tabindex="-1"></a>  <span class="fu">filter</span>(variable1<span class="sc">==</span><span class="st">&quot;Hg_ngg&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb751-3"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb751-3" tabindex="-1"></a>  <span class="fu">filter</span>(variable2<span class="sc">==</span><span class="st">&quot;Cd_ngg&quot;</span>)</span>
<span id="cb751-4"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb751-4" tabindex="-1"></a></span>
<span id="cb751-5"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb751-5" tabindex="-1"></a><span class="fu">ggplot</span>(HgCd, <span class="fu">aes</span>(z1, est)) <span class="sc">+</span> </span>
<span id="cb751-6"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb751-6" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="fu">aes</span>(<span class="at">col =</span> quantile), <span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>) <span class="sc">+</span> </span>
<span id="cb751-7"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb751-7" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;h(expos1 | quantiles of expos2)&quot;</span>) <span class="sc">+</span></span>
<span id="cb751-8"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb751-8" tabindex="-1"></a> <span class="fu">xlab</span>(<span class="st">&quot;expos1&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-568-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb752"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb752-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb752-1" tabindex="-1"></a>CdHg <span class="ot">&lt;-</span> pred.resp.bivar.levels <span class="sc">%&gt;%</span> </span>
<span id="cb752-2"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb752-2" tabindex="-1"></a>  <span class="fu">filter</span>(variable1<span class="sc">==</span><span class="st">&quot;Cd_ngg&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb752-3"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb752-3" tabindex="-1"></a>  <span class="fu">filter</span>(variable2<span class="sc">==</span><span class="st">&quot;Hg_ngg&quot;</span>)</span>
<span id="cb752-4"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb752-4" tabindex="-1"></a></span>
<span id="cb752-5"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb752-5" tabindex="-1"></a><span class="fu">ggplot</span>(CdHg, <span class="fu">aes</span>(z1, est)) <span class="sc">+</span> </span>
<span id="cb752-6"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb752-6" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="fu">aes</span>(<span class="at">col =</span> quantile), <span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>) <span class="sc">+</span> </span>
<span id="cb752-7"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb752-7" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;h(expos1 | quantiles of expos2)&quot;</span>) <span class="sc">+</span></span>
<span id="cb752-8"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb752-8" tabindex="-1"></a> <span class="fu">xlab</span>(<span class="st">&quot;expos1&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-568-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>To visualize interactions between one exposure and the rest of the exposure components, we generate a dataset that details the difference in each exposure’s (“variable”) log(OR) comparing 75th to 25th percentile (“est”, and associated standard deviation “sd”) when the other exposure components are at their 75th versus 25th percentile. Perhaps more intuitively, these estimates represent the blue - red points plotted in the second figure under the single exposure effects section.</p>
<div class="sourceCode" id="cb753"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb753-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb753-1" tabindex="-1"></a>risks.int <span class="ot">&lt;-</span> <span class="fu">SingVarIntSummaries</span>(<span class="at">fit=</span>fitkm, <span class="at">qs.diff =</span> <span class="fu">c</span>(<span class="fl">0.25</span>, <span class="fl">0.75</span>),</span>
<span id="cb753-2"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb753-2" tabindex="-1"></a>                                 <span class="at">qs.fixed =</span> <span class="fu">c</span>(<span class="fl">0.25</span>, <span class="fl">0.75</span>))</span>
<span id="cb753-3"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb753-3" tabindex="-1"></a></span>
<span id="cb753-4"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb753-4" tabindex="-1"></a></span>
<span id="cb753-5"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb753-5" tabindex="-1"></a><span class="fu">ggplot</span>(risks.int, <span class="fu">aes</span>(variable, est, <span class="at">ymin =</span> est <span class="sc">-</span> <span class="fl">1.96</span><span class="sc">*</span>sd, </span>
<span id="cb753-6"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb753-6" tabindex="-1"></a>                      <span class="at">ymax =</span> est <span class="sc">+</span> <span class="fl">1.96</span><span class="sc">*</span>sd)) <span class="sc">+</span> </span>
<span id="cb753-7"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb753-7" tabindex="-1"></a>  <span class="fu">geom_pointrange</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.75</span>)) <span class="sc">+</span> </span>
<span id="cb753-8"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb753-8" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">lty =</span> <span class="dv">2</span>, <span class="at">col =</span> <span class="st">&quot;brown&quot;</span>) <span class="sc">+</span> <span class="fu">coord_flip</span>()</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-569-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="answer-to-environmental-health-question-5-5" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 5<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#answer-to-environmental-health-question-5-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #5</strong></em>: Are there interactions among manganese, copper, zinc, arsenic, selenium, cadmium, mercury, lead in relation to neonatal inflammation?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: There do not appear to be any single exposure and rest of mixture interactions (previous plot); however, there is suggestive evidence of a bivariate interaction between cadmium and mercury.</p>
</div>
</div>
</div>
<div id="concluding-remarks-25" class="section level2 hasAnchor">
<h2>Concluding Remarks<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#concluding-remarks-25" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In conclusion, this module extends upon <strong>TAME 2.0 Module 6.3, Mixtures Analysis Methods Part 1: Overview and Example with Quantile G-Computation</strong>. In this scripted example, we used a dataset from a human population study (n=246) of cord tissue metals and examined the outcome of neonatal inflammation. We found that increasing the entire mixture of metals reduced the risk of neonatal inflammation; however, certain metals increased the risk and others decreased the risk. There was also a suggestive interactive effect found between cadmium and mercury.</p>
</div>
<div id="additional-resources-11" class="section level2 hasAnchor">
<h2>Additional Resources<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#additional-resources-11" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The field of mixtures is vast, with many different approaches and example studies to learn from as analysts lead in their own analyses. Some resources that can be helpful include the following reviews:</p>
<ul>
<li>Our recent review on mixtures methodologies, particularly in the field of sufficient similarity, titled <a href="https://www.sciencedirect.com/science/article/abs/pii/S2468202023000323?via%3Dihub">Wrangling whole mixtures risk assessment: Recent advances in determining sufficient similarity</a></li>
<li>Two more general, epidemiology-focused reviews on mixtures questions and methodologies, titled <a href="https://link.springer.com/article/10.1007/s40572-019-00229-5">Complex Mixtures, Complex Analyses: an Emphasis on Interpretable Results</a> and <a href="https://pubmed.ncbi.nlm.nih.gov/30643709/">Environmental exposure mixtures: questions and methods to address them</a></li>
<li><a href="https://bookdown.org/andreabellavia/mixtures/preface.html">A helpful online toolkit</a> for mixtures analyses generated by Andrea Bellavia, PhD</li>
</ul>
<p>Some helpful mixtures case studies using BKMR include the following:</p>
<ul>
<li><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6402346/">Prenatal metal concentrations and childhood cardio-metabolic risk using Bayesian Kernel Machine Regression to assess mixture and interaction effects</a></li>
<li><a href="https://ehp.niehs.nih.gov/doi/full/10.1289/EHP11998">Associations between Phthalate Metabolite Concentrations in Follicular Fluid and Reproductive Outcomes among Women Undergoing in Vitro Fertilization/Intracytoplasmic Sperm Injection Treatment</a></li>
<li><a href="https://ehp.niehs.nih.gov/doi/full/10.1289/EHP12597">Associations of Prenatal Per- and Polyfluoroalkyl Substance (PFAS) Exposures with Offspring Adiposity and Body Composition at 16–20 Years of Age: Project Viva</a></li>
</ul>
<p><label class="tykfont">
Test Your Knowledge
</label></p>
<div class="tyk">
<p>Using the simulated dataset within the bkmr package (see below code for how to call and store this dataset), answer the key environmental health questions using BKMR.</p>
<ol style="list-style-type: decimal">
<li>Which of these chemicals has the strongest effect on the outcome?</li>
<li>Which of these chemicals increases the outome and which decreases the outcome?</li>
<li>What is the effect on the outcome with exposure to each of the chemicals individually?</li>
<li>What is the effect on the outcome associated with combined exposure to all chemicals?</li>
<li>Are there interactions among the chemicals relation to the outcome?</li>
</ol>
<p>Note that the outcome (y) variable is a continuous variable here, rather than binary as in the scripted example.</p>
</div>
<div class="sourceCode" id="cb754"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb754-1"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb754-1" tabindex="-1"></a><span class="co"># Set seed for reproducibility</span></span>
<span id="cb754-2"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb754-2" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">111</span>)</span>
<span id="cb754-3"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb754-3" tabindex="-1"></a></span>
<span id="cb754-4"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb754-4" tabindex="-1"></a><span class="co"># Create a dataset with 100 participants and 4 mixtures components</span></span>
<span id="cb754-5"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb754-5" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">SimData</span>(<span class="at">n =</span> <span class="dv">100</span>, <span class="at">M =</span> <span class="dv">4</span>)</span>
<span id="cb754-6"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb754-6" tabindex="-1"></a></span>
<span id="cb754-7"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb754-7" tabindex="-1"></a><span class="co"># Save outcome variable (y)</span></span>
<span id="cb754-8"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb754-8" tabindex="-1"></a>y <span class="ot">&lt;-</span> data<span class="sc">$</span>y</span>
<span id="cb754-9"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb754-9" tabindex="-1"></a></span>
<span id="cb754-10"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb754-10" tabindex="-1"></a><span class="co"># Save mixtures variables (Z and X)</span></span>
<span id="cb754-11"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb754-11" tabindex="-1"></a>Z <span class="ot">&lt;-</span> data<span class="sc">$</span>Z</span>
<span id="cb754-12"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#cb754-12" tabindex="-1"></a>X <span class="ot">&lt;-</span> data<span class="sc">$</span>X</span></code></pre></div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
