<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6.3 Mixtures Analysis Methods Part 1: Overview and Example with Quantile G-Computation | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="6.3 Mixtures Analysis Methods Part 1: Overview and Example with Quantile G-Computation | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6.3 Mixtures Analysis Methods Part 1: Overview and Example with Quantile G-Computation | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  <meta name="twitter:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  

<meta name="author" content="Rager Lab" />


<meta name="date" content="2025-03-12" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="Icons_Used_Throughout/Favicon.png" type="image/x-icon" />
<link rel="prev" href="omics-and-system-biology-transcriptomic-applications.html"/>
<link rel="next" href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>
<link href="libs/table1-1.0/table1_defaults.css" rel="stylesheet" />
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-FGFKVX2X3G"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-FGFKVX2X3G');
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="Icons_Used_Throughout/TAME 2.0 Logo.png"><br></a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Chapter 1 Introductory <br>Data Science</b></span></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html"><i class="fa fa-check"></i>1.1 FAIR Data Management Practices</a>
<ul>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-training-module"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-fair"><i class="fa fa-check"></i>Introduction to FAIR</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#breaking-down-fair-letter-by-letter"><i class="fa fa-check"></i>Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#what-does-this-mean-for-you"><i class="fa fa-check"></i>What Does This Mean for You?</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#data-repositories-for-sharing-of-data"><i class="fa fa-check"></i>Data Repositories for Sharing of Data</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#recent-shifts-in-regulatory-policies-for-data-sharing"><i class="fa fa-check"></i>Recent Shifts in Regulatory Policies for Data Sharing</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#additional-training-resources-on-fair"><i class="fa fa-check"></i>Additional Training Resources on FAIR</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html"><i class="fa fa-check"></i>1.2 Data Sharing through Online Repositories</a>
<ul>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#an-overview-and-example-with-the-dataverse-repository"><i class="fa fa-check"></i>An Overview and Example with the Dataverse Repository</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#introduction-to-training-module-1"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#data-repositories"><i class="fa fa-check"></i>Data Repositories</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#the-dataverse-project"><i class="fa fa-check"></i>The Dataverse Project</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#what-is-a-dataverse"><i class="fa fa-check"></i>What is a Dataverse?</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#metadata"><i class="fa fa-check"></i>Metadata</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataverse"><i class="fa fa-check"></i>Creating a Dataverse</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataset"><i class="fa fa-check"></i>Creating a Dataset</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#concluding-remarks-1"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html"><i class="fa fa-check"></i>1.3 File Management using Github</a>
<ul>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#introduction-to-training-module-2"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-an-account"><i class="fa fa-check"></i>Creating an Account</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-a-repository"><i class="fa fa-check"></i>Creating a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#uploading-code"><i class="fa fa-check"></i>Uploading Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#adding-subfolders-in-a-repository"><i class="fa fa-check"></i>Adding Subfolders in a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-code"><i class="fa fa-check"></i>Updating Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-repository-titles-and-structure-to-support-a-manuscript"><i class="fa fa-check"></i>Updating Repository Titles and Structure to Support a Manuscript</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#tracking-code-changes-using-github-branches"><i class="fa fa-check"></i>Tracking Code Changes using Github Branches</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#concluding-remarks-2"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html"><i class="fa fa-check"></i>1.4 Data Wrangling in Excel</a>
<ul>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#introduction-to-training-module-3"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#save-a-copy-of-the-soon-to-be-organized-and-cleaned-dataset-as-a-new-file"><i class="fa fa-check"></i>Save a Copy of the Soon-To-Be Organized and Cleaned Dataset as a New File</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-extraneous-white-space"><i class="fa fa-check"></i>Remove Extraneous White Space</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#replace-missing-data-with-na"><i class="fa fa-check"></i>Replace Missing Data with “NA”</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-metadata-tab"><i class="fa fa-check"></i>Create a Metadata Tab</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#abbreviate-and-capitalize-categorical-data"><i class="fa fa-check"></i>Abbreviate and Capitalize Categorical Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#alphabetize-sort-the-data-by-the-categorical-variable-of-interest"><i class="fa fa-check"></i>Alphabetize (Sort) the Data by the Categorical Variable of Interest</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-new-subject-number-column"><i class="fa fa-check"></i>Create a New Subject Number Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-special-symbols-and-dashes"><i class="fa fa-check"></i>Remove Special Symbols and Dashes</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#bold-all-column-names-and-center-all-data"><i class="fa fa-check"></i>Bold all Column Names and Center all Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-subject-identifier-column"><i class="fa fa-check"></i>Create a Subject Identifier Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#separate-subject-demographic-data-from-experimental-measurements"><i class="fa fa-check"></i>Separate Subject Demographic Data from Experimental Measurements</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#convert-data-from-wide-to-long-format"><i class="fa fa-check"></i>Convert Data from Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#pivoting-data-from-a-wide-to-long-format"><i class="fa fa-check"></i>Pivoting Data from a Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#generating-summary-level-statistics-with-pivot-tables"><i class="fa fa-check"></i>Generating Summary-Level Statistics with Pivot Tables</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#excel-vs.-r-which-should-you-use"><i class="fa fa-check"></i>Excel vs. R: Which Should You Use?</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#concluding-remarks-3"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 2 Coding in R</b></span></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html"><i class="fa fa-check"></i>2.1 Downloading and Programming in R</a>
<ul>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-training-module-4"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#general-introduction-and-installation-of-r-and-rstudio"><i class="fa fa-check"></i>General Introduction and Installation of R and RStudio</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-r-packages"><i class="fa fa-check"></i>Introduction to R Packages</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#scripting-basics"><i class="fa fa-check"></i>Scripting Basics</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#code-troubleshooting"><i class="fa fa-check"></i>Code Troubleshooting</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#concluding-remarks-4"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html"><i class="fa fa-check"></i>2.2 Coding “Best” Practices</a>
<ul>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#introduction-to-training-module-5"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#scripting-file-types"><i class="fa fa-check"></i>Scripting File Types</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-headers-and-annotation"><i class="fa fa-check"></i>Script Headers and Annotation</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#coding-style"><i class="fa fa-check"></i>Coding Style</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-organization"><i class="fa fa-check"></i>Script Organization</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#concluding-remarks-5"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html"><i class="fa fa-check"></i>2.3 Data Manipulation and Reshaping</a>
<ul>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-training-module-6"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#data-manipulation-using-base-r"><i class="fa fa-check"></i>Data Manipulation Using Base R</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-tidyverse"><i class="fa fa-check"></i>Introduction to Tidyverse</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#concluding-remarks-6"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html"><i class="fa fa-check"></i>2.4 Improving Coding Efficiencies</a>
<ul>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#introduction-to-training-module-7"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#loops"><i class="fa fa-check"></i>Loops</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#functions"><i class="fa fa-check"></i>Functions</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#list-operations"><i class="fa fa-check"></i>List operations</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#concluding-remarks-7"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#additional-resources-2"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="part"><span><b>Chapter 3 Basics of Data <br>Analysis and Visualizations</b></span></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html"><i class="fa fa-check"></i>3.1 Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-data-visualizations"><i class="fa fa-check"></i>Introduction to Data Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-training-module-8"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#density-plot-visualization"><i class="fa fa-check"></i>Density Plot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#boxplot-visualization"><i class="fa fa-check"></i>Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#correlation-visualizations"><i class="fa fa-check"></i>Correlation Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#heatmap-visualization"><i class="fa fa-check"></i>Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#concluding-remarks-8"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html"><i class="fa fa-check"></i>3.2 Improving Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-data-visulization-conventions"><i class="fa fa-check"></i>Introduction to Data Visulization Conventions</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-training-module-9"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-boxplot-visualization"><i class="fa fa-check"></i>Creating an Improved Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-heatmap-visualization"><i class="fa fa-check"></i>Creating an Improved Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-multi-plot-figures"><i class="fa fa-check"></i>Creating Multi-Plot Figures</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#concluding-remarks-9"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html"><i class="fa fa-check"></i>3.3 Normality Tests and Data Transformations</a>
<ul>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#introduction-to-training-module-10"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#what-is-a-normal-distribution"><i class="fa fa-check"></i>What is a Normal Distribution?</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#qualitative-assessment-of-normality"><i class="fa fa-check"></i>Qualitative Assessment of Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#quantitative-normality-assessment"><i class="fa fa-check"></i>Quantitative Normality Assessment</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#data-transformation"><i class="fa fa-check"></i>Data Transformation</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#additional-considerations-regarding-normality"><i class="fa fa-check"></i>Additional Considerations Regarding Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#concluding-remarks-10"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html"><i class="fa fa-check"></i>3.4 Intoduction to Statistical Tests</a>
<ul>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#introduction-to-training-module-11"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#assessing-normality-homogeneity-of-variance"><i class="fa fa-check"></i>Assessing Normality &amp; Homogeneity of Variance</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#two-group-visualizations-and-statistical-comparisons-using-the-t-test"><i class="fa fa-check"></i>Two-Group Visualizations and Statistical Comparisons using the T-Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#three-group-visualizations-and-statistical-comparisons-using-an-anova"><i class="fa fa-check"></i>Three-Group Visualizations and Statistical Comparisons using an ANOVA</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#regression-modeling-and-visualization-linear-and-logistic-regressions"><i class="fa fa-check"></i>Regression Modeling and Visualization: Linear and Logistic Regressions</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#statistical-evaluations-of-categorical-data-using-the-chi-squared-test-and-fishers-exact-test"><i class="fa fa-check"></i>Statistical Evaluations of Categorical Data using the Chi-Squared Test and Fisher’s Exact Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#concluding-remarks-11"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 4 Converting Wet <br>Lab Data into Dry Lab <br>Analyses</b></span></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html"><i class="fa fa-check"></i>4.1 Overview of Experimental Design and Example Data</a>
<ul>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#introduction-to-training-module-12"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#replicates"><i class="fa fa-check"></i>Replicates</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#orientation-to-example-data-for-chapter-4"><i class="fa fa-check"></i>Orientation to Example Data for Chapter 4</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#concluding-remarks-12"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html"><i class="fa fa-check"></i>4.2 Data Import, Processing, and Summary Statistics</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#introduction-to-training-module-13"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#data-import"><i class="fa fa-check"></i>Data Import</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#handling-missing-values"><i class="fa fa-check"></i>Handling Missing Values</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#averaging-replicates"><i class="fa fa-check"></i>Averaging Replicates</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#descriptive-statistics"><i class="fa fa-check"></i>Descriptive Statistics</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#normality-assessment-and-data-transformation"><i class="fa fa-check"></i>Normality Assessment and Data Transformation</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#concluding-remarks-13"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html"><i class="fa fa-check"></i>4.3 Data Import from PDF Sources</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#introduction-to-training-module-14"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-from-many-single-pdfs-with-the-same-formatting"><i class="fa fa-check"></i>Importing Data from Many Single PDFs with the Same Formatting</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-stored-in-pdf-tables"><i class="fa fa-check"></i>Importing Data Stored in PDF Tables</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#concluding-remarks-14"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html"><i class="fa fa-check"></i>4.4 Two Group Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#introduction-to-training-module-15"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#overview-of-two-group-statistical-tests"><i class="fa fa-check"></i>Overview of Two Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#statistical-vs.-biological-significance"><i class="fa fa-check"></i>Statistical vs. Biological Significance</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#unpaired-test-example"><i class="fa fa-check"></i>Unpaired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#paired-test-example"><i class="fa fa-check"></i>Paired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#visualizing-results"><i class="fa fa-check"></i>Visualizing Results</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#concluding-remarks-15"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html"><i class="fa fa-check"></i>4.5 Multi-Group and Multi-Variable Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#introduction-to-training-module-16"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#overview-of-multi-group-statistical-tests"><i class="fa fa-check"></i>Overview of Multi-Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#multi-group-analysis-example"><i class="fa fa-check"></i>Multi-Group Analysis Example</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#visualization-of-multi-group-statistical-results"><i class="fa fa-check"></i>Visualization of Multi-Group Statistical Results</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#concluding-remarks-16"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html"><i class="fa fa-check"></i>4.6 Advanced Multi-Group Comparisons</a>
<ul>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#introduction-to-training-module-17"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#advanced-multi-group-comparisons-1"><i class="fa fa-check"></i>Advanced Multi-Group Comparisons</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#two-way-anova-example"><i class="fa fa-check"></i>Two-way ANOVA Example</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#ancova-example"><i class="fa fa-check"></i>ANCOVA Example</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#concluding-remarks-17"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 5 Machine Learning <br>&amp; Artificial Intelligence</b></span></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><i class="fa fa-check"></i>5.1 Introduction to Artificial Intelligence, Machine Learning, and Predictive Modeling for Environmental Health</a>
<ul>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#introduction-to-training-module-18"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#general-historical-context-and-taxonomy-of-modern-aiml"><i class="fa fa-check"></i>General Historical Context and Taxonomy of Modern AI/ML</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#application-of-machine-learning-in-environmental-health-science"><i class="fa fa-check"></i>Application of Machine Learning in Environmental Health Science</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#concluding-remarks-18"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html"><i class="fa fa-check"></i>5.2 Supervised Machine Learning</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-training-module-19"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-machine-learning"><i class="fa fa-check"></i>Types of Machine Learning</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-supervised-machine-learning-algorithms"><i class="fa fa-check"></i>Types of Supervised Machine Learning Algorithms</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#training-supervised-machine-learning-models"><i class="fa fa-check"></i>Training Supervised Machine Learning Models</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#assessing-classification-based-model-performance"><i class="fa fa-check"></i>Assessing Classification-Based Model Performance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-activity-and-example-dataset"><i class="fa fa-check"></i>Introduction to Activity and Example Dataset</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#testing-for-differences-in-predictor-variables-across-the-outcome-classes"><i class="fa fa-check"></i>Testing for Differences in Predictor Variables across the Outcome Classes</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#predicting-ias-detection-with-a-random-forest-rf-model"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#class-imbalance"><i class="fa fa-check"></i>Class Imbalance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#concluding-remarks-19"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html"><i class="fa fa-check"></i>5.3 Supervised Machine Learning Model Interpretation</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-training-module-20"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance"><i class="fa fa-check"></i>Variable Importance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary"><i class="fa fa-check"></i>Decision Boundary</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-example-dataset-and-activity"><i class="fa fa-check"></i>Introduction to Example Dataset and Activity</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#predicting-ias-detection-with-a-random-forest-rf-model-1"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance-plot"><i class="fa fa-check"></i>Variable Importance Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot"><i class="fa fa-check"></i>Decision Boundary Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot-incorporating-smote"><i class="fa fa-check"></i>Decision Boundary Plot Incorporating SMOTE</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#concluding-remarks-20"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><i class="fa fa-check"></i>5.4 Unsupervised Machine Learning Part 1: K-Means Clustering &amp; PCA</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-training-module-21"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-unsupervised-machine-learning"><i class="fa fa-check"></i>Introduction to Unsupervised Machine Learning</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-example-data"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#identifying-clusters-of-chemicals-through-k-means"><i class="fa fa-check"></i>Identifying Clusters of Chemicals through <em>K</em>-Means</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#principal-component-analysis-pca-1"><i class="fa fa-check"></i>Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#incorporating-k-means-into-pca-for-predictive-modeling"><i class="fa fa-check"></i>Incorporating <em>K</em>-Means into PCA for Predictive Modeling</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#concluding-remarks-21"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><i class="fa fa-check"></i>5.5 Unsupervised Machine Learning Part 2: Additional Clustering Applications</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-training-module-22"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#k-means-clustering-1"><i class="fa fa-check"></i><em>K</em>-Means Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-example-data-1"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-1"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#clustering-plot"><i class="fa fa-check"></i>Clustering Plot</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-visualization"><i class="fa fa-check"></i>Hierarchical Clustering Visualization</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#variable-contributions"><i class="fa fa-check"></i>Variable Contributions</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#concluding-remarks-22"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 6 Applications in <br>Toxicology &amp; Exposure <br>Science</b></span></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html"><i class="fa fa-check"></i>6.1 Descriptive Cohort Analyses</a>
<ul>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#introduction-to-training-module-23"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#participant-exploration"><i class="fa fa-check"></i>Participant Exploration</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#chemical-detection-filtering"><i class="fa fa-check"></i>Chemical Detection Filtering</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#outlier-identification"><i class="fa fa-check"></i>Outlier Identification</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#summary-statistics-tables"><i class="fa fa-check"></i>Summary Statistics Tables</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#demographics-table"><i class="fa fa-check"></i>Demographics Table</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#concluding-remarks-23"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html"><i class="fa fa-check"></i>6.2 -Omics and System Biology: Transcriptomic Applications</a>
<ul>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-training-module-24"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-the-field-of--omics"><i class="fa fa-check"></i>Introduction to the Field of “-Omics”</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-transcriptomics"><i class="fa fa-check"></i>Introduction to Transcriptomics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#formatting-data-for-downstream-statistics"><i class="fa fa-check"></i>Formatting Data for Downstream Statistics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#transcriptomics-data-qaqc"><i class="fa fa-check"></i>Transcriptomics Data QA/QC</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#controlling-for-sources-of-sample-heterogeneity"><i class="fa fa-check"></i>Controlling for Sources of Sample Heterogeneity</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e.g.-biomass-smoke-exposure"><i class="fa fa-check"></i>Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure)</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-ma-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using MA Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-volcano-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using Volcano Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis"><i class="fa fa-check"></i>Interpretting Findings at the Systems Level through Pathway Enrichment Analysis</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#concluding-remarks-24"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><i class="fa fa-check"></i>6.3 Mixtures Analysis Methods Part 1: Overview and Example with Quantile G-Computation</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-training-module-25"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#overview-of-mixtures-analysis"><i class="fa fa-check"></i>Overview of Mixtures Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-example-data-2"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#mixtures-model-with-standard-qqcomp"><i class="fa fa-check"></i>Mixtures Model with Standard qqcomp</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#concluding-remarks-25"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#additional-resources-11"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><i class="fa fa-check"></i>6.4 Mixtures Analysis Methods Part 2: Bayesian Kernel Machine Regression</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-training-module-26"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-example-data-3"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-bkmr"><i class="fa fa-check"></i>Introduction to BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#run-bkmr"><i class="fa fa-check"></i>Run BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#concluding-remarks-26"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#additional-resources-12"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html"><i class="fa fa-check"></i>6.5 Mixtures Analysis Methods Part 3: Sufficient Similarity</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-training-module-27"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-toxicant-and-dataset"><i class="fa fa-check"></i>Introduction to Toxicant and Dataset</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#workspace-preparation-and-data-import-9"><i class="fa fa-check"></i>Workspace Preparation and Data Import</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#chemistry-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Chemistry-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#toxicity-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Toxicity-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#comparing-chemistry-vs.-toxicity-sufficient-similarity-analyses"><i class="fa fa-check"></i>Comparing Chemistry vs. Toxicity Sufficient Similarity Analyses</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#concluding-remarks-27"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i>6.6 Toxicokinetic Modeling</a>
<ul>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-training-module-28"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-toxicokinetic-modeling"><i class="fa fa-check"></i>Introduction to Toxicokinetic Modeling</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#data-and-models-used-in-toxicokinetic-modeling-tk"><i class="fa fa-check"></i>Data and Models used in Toxicokinetic Modeling (TK)</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-toxicokinetics"><i class="fa fa-check"></i>Reverse Toxicokinetics</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#capturing-population-variability-in-toxicokinetics-and-uncertainty-in-chemical-specific-parameters"><i class="fa fa-check"></i>Capturing Population Variability in Toxicokinetics, and Uncertainty in Chemical-Specific Parameters</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-tk-calculating-administered-equivalent-doses-for-toxcast-bioactive-concentrations"><i class="fa fa-check"></i>Reverse TK: Calculating Administered Equivalent Doses for ToxCast Bioactive Concentrations</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#comparing-equivalent-doses-estimated-to-elicit-toxicity-hazard-to-external-exposure-estimates-exposure-for-chemical-prioritization-by-bioactivity-exposure-ratios-bers"><i class="fa fa-check"></i>Comparing Equivalent Doses Estimated to Elicit Toxicity (Hazard) to External Exposure Estimates (Exposure), for Chemical Prioritization by Bioactivity-Exposure Ratios (BERs)</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#concluding-remarks-28"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html"><i class="fa fa-check"></i>Chemical Read-Across for Toxicity Predictions</a>
<ul>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#introduction-to-training-module-29"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#introduction-to-activity"><i class="fa fa-check"></i>Introduction to Activity</a></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#read-across-example-analysis"><i class="fa fa-check"></i>Read-Across Example Analysis</a></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#calculating-chemical-similarities"><i class="fa fa-check"></i>Calculating Chemical Similarities</a></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#chemical-read-across-to-predict-acute-toxicity"><i class="fa fa-check"></i>Chemical Read-Across to Predict Acute Toxicity</a></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#concluding-remarks-29"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 7 Environmental<br> Health Database Mining</b></span></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html"><i class="fa fa-check"></i>7.1 Comparative Toxicogenomics Database</a>
<ul>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#introduction-to-training-module-30"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#ctd-data-in-r"><i class="fa fa-check"></i>CTD Data in R</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#identifying-genes-under-epigenetic-control"><i class="fa fa-check"></i>Identifying Genes Under Epigenetic Control</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#concluding-remarks-30"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html"><i class="fa fa-check"></i>7.2 Gene Expression Omnibus</a>
<ul>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-training-module-31"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-geo"><i class="fa fa-check"></i>Introduction to GEO</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-example-data-4"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#geo-data-in-r"><i class="fa fa-check"></i>GEO Data in R</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#visualizing-data"><i class="fa fa-check"></i>Visualizing Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#statistical-analyses"><i class="fa fa-check"></i>Statistical Analyses</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#concluding-remarks-31"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html"><i class="fa fa-check"></i>7.3 CompTox Dashboard Data through APIs</a>
<ul>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#introduction-to-training-module-32"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#training-modules-environmental-health-questions-24"><i class="fa fa-check"></i>Training Module’s Environmental Health Questions</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#script-preparations-12"><i class="fa fa-check"></i>Script Preparations</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#introduction-to-comptox-chemicals-dashboard"><i class="fa fa-check"></i>Introduction to CompTox Chemicals Dashboard</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#cctes-ctx-application-programming-interfaces-apis-for-automated-batch-search-of-the-ccd"><i class="fa fa-check"></i>CCTE’s CTX Application Programming Interfaces (APIs) for Automated Batch Search of the CCD</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#comparing-physico-chemical-properties-between-two-important-environmental-contaminant-lists"><i class="fa fa-check"></i>Comparing Physico-chemical Properties between Two Important Environmental Contaminant Lists</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#hazard-data-genotoxicity"><i class="fa fa-check"></i>Hazard Data: Genotoxicity</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#hazard-resource"><i class="fa fa-check"></i>Hazard Resource</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#concluding-remarks-32"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation" class="section level1 hasAnchor">
<h1>6.3 Mixtures Analysis Methods Part 1: Overview and Example with Quantile G-Computation<a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This training module was developed by Dr. Lauren Eaves, Dr. Kyle Roell, and Dr. Julia E. Rager.</p>
<p>All input files (script, data, and figures) can be downloaded from the <a href="https://github.com/UNCSRP/TAME2">UNC-SRP TAME2 GitHub website</a>.</p>
<div id="introduction-to-training-module-25" class="section level2 hasAnchor">
<h2>Introduction to Training Module<a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-training-module-25" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Historically, toxicology and epidemiology studies have largely focused on analyzing relationships between one chemical and one outcome at a time. This is still important in identifying the degree to which a single chemical exposure is associated with a disease outcome (e.g., <a href="https://sph.unc.edu/superfund-pages/srp/">UNC Superfund Research Program’s</a> focus on inorganic arsenic exposure and its influence on metabolic disease). However, we are exposed, everyday, to many different stressors in our environment. It is therefore critical to deconvolute what co-occurring stressors (i.e., mixtures) in our environment impact human health! The field of mixtures research continues to grow to address this need, with the goals of developing methods to study environmental exposures using approaches to that better capture the mixture of exposures humans experience in real life. In this module, we will provide an overview of mixtures analysis methods and demonstrate how to use one of these methods, quantile g-computation, to analyzing chemical mixtures in a large geospatial epidemiologic study.</p>
</div>
<div id="overview-of-mixtures-analysis" class="section level2 hasAnchor">
<h2>Overview of Mixtures Analysis<a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#overview-of-mixtures-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="mixtures-methods-relevance-and-challenges" class="section level3 hasAnchor">
<h3>Mixtures Methods Relevance and Challenges<a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#mixtures-methods-relevance-and-challenges" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><strong>Mixtures approaches are recently becoming more routine in environmental health because methodological advancements are just now making mixtures research more feasible.</strong> These advancements parallel the following:</p>
<ul>
<li>Advances in the ability to measure many different chemicals (e.g., through suspect screening and non-targeted chemical analysis approaches) and stressors (e.g., through improved collection and storage of survey data and clinical data) in our environment</li>
<li>Improvements in data science to organize, store, and analyze big data</li>
<li>Developments in statistical methodologies to parse relationships within these data</li>
</ul>
<p>Though statistical methodologies are still evolving, we will be discussing our current knowledge in this module.</p>
<p><strong>Some challenges that data analysts may experience when analyzing data from mixtures studies include the following:</strong></p>
<ol style="list-style-type: decimal">
<li>Size of mixture:</li>
</ol>
<ul>
<li>As the number of components evaluated increases, your available analysis methods and statistical power may decrease</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Correlated data structure:</li>
</ol>
<ul>
<li>Statistical challenge of collinearity: If data include large amounts of collinearity, this may dampen the observed effects from components that are highly correlated (e.g., may commonly co-occur) to other components</li>
<li>Methodological challenge of co-occurring contaminant confounding: Co-occurring contaminant confounding may make it difficult to discern what is the true driver of the observed effect.</li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>Data analysis method selection:</li>
</ol>
<ul>
<li>There are many different methods to choose from!</li>
<li>A critical rule to address this challenge is to, first and foremost, <em>lay out your study’s question</em>. This question will then help guide your method selection, as discussed below.</li>
</ul>
</div>
<div id="overview-of-mixtures-methods" class="section level3 hasAnchor">
<h3>Overview of Mixtures Methods<a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#overview-of-mixtures-methods" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>There are many methods that can be implemented to also elucidate relationships between individual chemicals/chemical groups in complex mixtures and their resulting toxicity/health effects. Some of the more common methods used in mixtures analyses, as identified by our team, are summarized in the below figure according to potential questions that could be asked in a study. Two of the methods, specifically quantile based g-computation (qgcomp) and bayesian kernel machine regression (BKMR) are highlighted as example mixtures scripted activities (qgcomp in this script and BKMR in Mixtures Methods 2). Throughout TAME 2.0 training materials, other methods are included such as Principal Component Analysis (PCA), K-means clustering, hierarchical clustering, and predictive modeling / machine learning (e.g., Random Forest modeling and variable selection). The following figure provides an overview of the types of questions that can be asked regarding mixtures and models that are commonly used to answer these questions:</p>
<p><img src="Module6_3_Input/Module6_3_Mixtures_Methods_Overview.png" width="1289" style="display: block; margin: auto;" /></p>
<p>In this module, we will be using quantile based g-computation to analyze our data. This method is used for analysis of a total mixture effect as opposed to individual effects of mixture components. It is similar to previous, popular methods such as weighted quantile sum (WQS) regression, but does not assume directional homogeneity. It also provides access to models for non-additive and non-linear effects of the individual mixture components and overall mixture. Additionally, it runs very quickly and does not require as much computationally as other methods, making it an accessible option for those without access to many computational resources.</p>
</div>
</div>
<div id="introduction-to-example-data-2" class="section level2 hasAnchor">
<h2>Introduction to Example Data<a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-example-data-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This script outlines single-contaminant (logistic regression) and multi-contaminant modeling approaches (Quantile G-Computation (qgcomp)). The workflow follows the steps used to generate results published in <a href="https://pubmed.ncbi.nlm.nih.gov/37845729/">Eaves et al. 2023</a>. This study examined the relationship between metals in private well water and the risk of preterm birth. The study population was all singleton, non-anomalous births in NC between 2003-2015. Pregnancies were assigned tract-level metal exposure based on maternal residence at delivery. The relationship with single metal exposure was examined with logistic regression and metal mixtures with qgcomp.</p>
<p>For more info on qgcomp, see <a href=":https://ehp.niehs.nih.gov/doi/full/10.1289/EHP5838">Keil et al. 2020</a> and the associated <a href="https://cran.r-project.org/web/packages/qgcomp/vignettes/qgcomp-vignette.html">vignette</a>.</p>
<p>Note that for educational purposes, in this example we are using a randomly sampled dataset of 100,000 births, rather than the full dataset of &gt;1.3million (ie. using less than 10% of the full study population). Therefore the actual results of the analysis outlined below do not match the results published in the paper.</p>
<div id="training-modules-environmental-health-questions-17" class="section level3 hasAnchor">
<h3>Training Module’s Environmental Health Questions<a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#training-modules-environmental-health-questions-17" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This training module was specifically developed to answer the following questions:</p>
<ol style="list-style-type: decimal">
<li><p>What is the risk of preterm birth associated with exposure to each of arsenic, lead, cadmium, chromium, manganese, copper and zinc via private well water individually?</p></li>
<li><p>What is the risk of preterm birth associated with combined exposure to arsenic, lead, cadmium, chromium, manganese, copper and zinc (ie. a mixture) via private well water?</p></li>
<li><p>Which of these chemicals has the strongest effect on preterm birth risk?</p></li>
<li><p>Which of these chemicals increases the risk of preterm birth and which decreases the risk of preterm birth?</p></li>
</ol>
</div>
<div id="workspace-preparation" class="section level3 hasAnchor">
<h3>Workspace Preparation<a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#workspace-preparation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Install packages as needed, then load the following packages:</p>
<div class="sourceCode" id="cb729"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb729-1"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb729-1" tabindex="-1"></a><span class="co">#load packages</span></span>
<span id="cb729-2"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb729-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb729-3"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb729-3" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb729-4"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb729-4" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb729-5"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb729-5" tabindex="-1"></a><span class="fu">library</span>(yaml)</span>
<span id="cb729-6"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb729-6" tabindex="-1"></a><span class="fu">library</span>(rmarkdown)</span>
<span id="cb729-7"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb729-7" tabindex="-1"></a><span class="fu">library</span>(broom)</span>
<span id="cb729-8"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb729-8" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb729-9"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb729-9" tabindex="-1"></a><span class="fu">library</span>(qgcomp)</span></code></pre></div>
<p>Optionally, you can also create a current date variable to name output files, and create an output folder.</p>
<div class="sourceCode" id="cb730"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb730-1"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb730-1" tabindex="-1"></a><span class="co"># Create a current date variable to name outputfiles</span></span>
<span id="cb730-2"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb730-2" tabindex="-1"></a>cur_date <span class="ot">&lt;-</span> <span class="fu">str_replace_all</span>(<span class="fu">Sys.Date</span>(),<span class="st">&quot;-&quot;</span>,<span class="st">&quot;&quot;</span>)</span>
<span id="cb730-3"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb730-3" tabindex="-1"></a></span>
<span id="cb730-4"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb730-4" tabindex="-1"></a><span class="co">#Create an output folder </span></span>
<span id="cb730-5"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb730-5" tabindex="-1"></a>Output_Folder <span class="ot">&lt;-</span> (<span class="st">&quot;Module6_3_Output/&quot;</span>)</span></code></pre></div>
</div>
<div id="data-import-1" class="section level3 hasAnchor">
<h3>Data Import<a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#data-import-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb731"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb731-1"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb731-1" tabindex="-1"></a>cohort <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file=</span><span class="st">&quot;Module6_3_Input/Module6_3_InputData.csv&quot;</span>)</span>
<span id="cb731-2"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb731-2" tabindex="-1"></a><span class="fu">colnames</span>(cohort)</span></code></pre></div>
<pre><code>##  [1] &quot;ID&quot;                             &quot;year&quot;                          
##  [3] &quot;sex&quot;                            &quot;lmpdate&quot;                       
##  [5] &quot;racegp&quot;                         &quot;mage&quot;                          
##  [7] &quot;mothed&quot;                         &quot;bwt&quot;                           
##  [9] &quot;calcega&quot;                        &quot;clinega&quot;                       
## [11] &quot;plural&quot;                         &quot;delivery&quot;                      
## [13] &quot;rescity&quot;                        &quot;zipcode&quot;                       
## [15] &quot;state&quot;                          &quot;ga&quot;                            
## [17] &quot;preterm&quot;                        &quot;mage_cat&quot;                      
## [19] &quot;concep_date&quot;                    &quot;concep_month&quot;                  
## [21] &quot;season_concep&quot;                  &quot;smoke&quot;                         
## [23] &quot;Arsenic_perc&quot;                   &quot;Cadmium_perc&quot;                  
## [25] &quot;Calcium_perc&quot;                   &quot;Chromium_perc&quot;                 
## [27] &quot;Copper_perc&quot;                    &quot;Iron_perc&quot;                     
## [29] &quot;Lead_perc&quot;                      &quot;Manganese_perc&quot;                
## [31] &quot;Magnesium_perc&quot;                 &quot;Zinc_perc&quot;                     
## [33] &quot;Sodium_perc&quot;                    &quot;Arsenic_limit&quot;                 
## [35] &quot;Cadmium_limit&quot;                  &quot;Chromium_limit&quot;                
## [37] &quot;Copper_limit&quot;                   &quot;Iron_limit&quot;                    
## [39] &quot;Lead_limit&quot;                     &quot;Manganese_limit&quot;               
## [41] &quot;Zinc_limit&quot;                     &quot;Arsenic.Mean_avg&quot;              
## [43] &quot;Cadmium.Mean_avg&quot;               &quot;Chromium.Mean_avg&quot;             
## [45] &quot;Copper.Mean_avg&quot;                &quot;Iron.Mean_avg&quot;                 
## [47] &quot;Lead.Mean_avg&quot;                  &quot;Manganese.Mean_avg&quot;            
## [49] &quot;Zinc.Mean_avg&quot;                  &quot;percent_lowincome_2010&quot;        
## [51] &quot;percent_lowincome_2015&quot;         &quot;percent_lowincome_20102015_avg&quot;
## [53] &quot;Nitr&quot;                           &quot;mage_sq&quot;                       
## [55] &quot;Nitr_perc&quot;                      &quot;pov_perc&quot;</code></pre>
<div class="sourceCode" id="cb733"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb733-1"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb733-1" tabindex="-1"></a><span class="fu">head</span>(cohort)</span></code></pre></div>
<pre><code>##        ID year sex    lmpdate racegp mage mothed  bwt calcega clinega plural
## 1 9098140 2009   2       &lt;NA&gt;      1   38      2 3175      NA      39      1
## 2 4105090 2004   1 2004-02-28      5   25      1 3175      37      37      1
## 3 3015971 2003   1 2002-06-17      3   14      1 3119      37      38      1
## 4 3103126 2003   1 2003-02-17      3   32      2 3544      39      40      1
## 5 9030473 2009   2 2008-06-15      2   33      3 3317      40      39      1
## 6 6024870 2006   2 2005-06-07      2   25      1 3430      40      39      1
##   delivery     rescity zipcode state ga preterm mage_cat concep_date
## 1        2      DALLAS   28034    37 39       0        3  2009-01-12
## 2        1 WAYNESVILLE   28786    37 37       0        2  2004-03-03
## 3        1 GIBSONVILLE   27249    37 38       0        1  2002-06-14
## 4        2      GARNER   27529    37 40       0        3  2003-02-10
## 5        2  GREENVILLE   27858    37 39       0        3  2008-06-25
## 6        2        EDEN   27288    37 39       0        2  2005-06-16
##   concep_month season_concep smoke Arsenic_perc Cadmium_perc Calcium_perc
## 1            1             1     0            2            2            1
## 2            3             2     0            0            0            0
## 3            6             3     0            0            0            1
## 4            2             1     0            0            2            1
## 5            6             3     0            0            0            2
## 6            6             3     1            1            0            1
##   Chromium_perc Copper_perc Iron_perc Lead_perc Manganese_perc Magnesium_perc
## 1             1           0         2         1              1              0
## 2             1           2         1         1              0              0
## 3             2           1         1         1              1              1
## 4             1           1         1         1              1              1
## 5             0           0         0         0              0              0
## 6             0           0         1         0              2              1
##   Zinc_perc Sodium_perc Arsenic_limit Cadmium_limit Chromium_limit Copper_limit
## 1         0           0             0             0              0            0
## 2         1           0             0             0              0            0
## 3         1           1             0             0              0            0
## 4         1           1             0             0              0            0
## 5         0           2             0             0              0            0
## 6         0           0             0             0              0            0
##   Iron_limit Lead_limit Manganese_limit Zinc_limit Arsenic.Mean_avg
## 1          0          0               0          0       1.50407414
## 2          1          0               0          0       0.08686698
## 3          1          0               0          0       0.10619964
## 4          1          0               0          0       0.07614543
## 5          1          0               0          0       0.01302664
## 6          1          0               0          0       0.40594302
##   Cadmium.Mean_avg Chromium.Mean_avg Copper.Mean_avg Iron.Mean_avg
## 1      0.100876988        0.62645522      13.4649631     5199.2885
## 2      0.006774432        0.62262246     105.3293541      488.2481
## 3      0.003405709        3.53144521      36.6958692     1472.2966
## 4      0.120147641        1.00173292      58.6573330      699.0566
## 5      0.002553201        0.00910553       0.3058865      275.1736
## 6      0.001376889        0.18293440      15.1655788      985.0538
##   Lead.Mean_avg Manganese.Mean_avg Zinc.Mean_avg percent_lowincome_2010
## 1     1.5978593           38.06972      54.29592              13.777468
## 2     2.1463184           13.21209     253.92295              14.787742
## 3     2.6009340           65.45814     541.73708              17.495030
## 4     3.1221232           36.06194     511.58471               1.957295
## 5     0.0305281           30.04164      13.40628              22.706422
## 6     0.5590962          149.49924      21.95994              20.715677
##   percent_lowincome_2015 percent_lowincome_20102015_avg     Nitr mage_sq
## 1              18.332801                      16.055134 521.0576    1444
## 2              18.355821                      16.571782 388.9087     625
## 3               9.235569                      13.365300 506.7652     196
## 4               6.642066                       4.299681       NA    1024
## 5              13.564499                      18.135460 388.9087    1089
## 6              29.359165                      25.037421 388.9087     625
##   Nitr_perc pov_perc
## 1         0        2
## 2         0        2
## 3         0        1
## 4        NA        0
## 5         0        2
## 6         0        3</code></pre>
<p>Note: there are many steps prior to the modeling steps outlined below. These are being skipped for educational purposes. Additional steps include assessment of normality and transformations as needed, generation of a demographics table and assessing for missing data, imputation of missing data if needed, visualizing trends and distributions in the data, functional form assessments, decisions regarding what confounders to include etc.</p>
<p>The following are the metals of interest: arsenic, lead, cadmium, chromium, manganese, copper, zinc.</p>
<p>For each metal there are three exposure variables:</p>
<ol style="list-style-type: decimal">
<li><code>[metal]_perc</code>: 0: less than or equal to the 50th percentile, 1: above the 50th percentile and less than or equal to the 90th percentile, 3: above the 90th percentile</li>
<li><code>[metal]_limit</code>: 0: &lt;25% f well water tests for a given metal exceeded EPA regulatory standard, 1: 25% or over of well water tests for a given metal exceeded EPA regulatory standard</li>
<li><code>[metal].Mean_avg</code>: the mean concentration of the metal in the tract (ppb).
Please see the Eaves et al. 2023 paper linked above for further information on these variables.</li>
</ol>
<p>Other variables of interest (outcome and covariates) in this dataset:</p>
<ul>
<li><code>preterm</code>: 0= 37 weeks gestational age or greater, 1= less than 37 weeks gestational age</li>
<li><code>mage</code>: maternal age in years, continuous</li>
<li><code>sex</code>: sex of baby at birth: 1=M, 2=F</li>
<li><code>racegp</code>: maternal race ethnicity: 1=white non-Hispanic, 2=Black non-Hispanic, 3=Hispanic, 4=Asian/Pacific Islander, 5=American Indian, 6=other/unknown</li>
<li><code>smoke</code>: maternal smoking in pregnany: 0=non-smoker, 1=smoker</li>
<li><code>season_conep</code>: season of conception: 1=winter (Dec, Jan, Feb), 2=spring (Mar, Apr, May), 3=summer (June, Jul, Aug), 4=fall (Sept, Oct, Nov)</li>
<li><code>mothed</code>: mother’s education: 1=&lt;HS, 2=HS, 3=more than HS</li>
<li><code>pov_perc</code>: ACS estimates for poverty rate in tract: 0=less than or equal to 25th percentile, 1= greater than 25th percentile and less than or equal to 50th percentile, 2= greater than 50th percentile and less than or equal to 75th percentile, 3= greater than 75th percentile</li>
<li><code>Nitr_perc</code>: average of nitrites and nitrates in well water: 0= less than or equal to the 50th percentile, 1= above the 50th percentile and less than or equal to the 90th percentile, 2: above the 90th percentile</li>
</ul>
<div id="check-variable-formats" class="section level4 hasAnchor">
<h4>Check variable formats<a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#check-variable-formats" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Ensure that the outcome variable is binomial (factor) and has the correct reference level. Ensure that the exposure variables are categorical (factors). Ensure that covariates are in the correct variable format</p>
<div class="sourceCode" id="cb735"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb735-1"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-1" tabindex="-1"></a><span class="co">#outcome variable</span></span>
<span id="cb735-2"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-2" tabindex="-1"></a>cohort <span class="ot">&lt;-</span> cohort <span class="sc">%&gt;%</span> </span>
<span id="cb735-3"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">preterm =</span> <span class="fu">as.factor</span>(preterm)) </span>
<span id="cb735-4"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-4" tabindex="-1"></a>cohort<span class="sc">$</span>preterm <span class="ot">&lt;-</span> <span class="fu">relevel</span>(cohort<span class="sc">$</span>preterm, <span class="at">ref =</span> <span class="st">&quot;0&quot;</span>)</span>
<span id="cb735-5"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-5" tabindex="-1"></a></span>
<span id="cb735-6"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-6" tabindex="-1"></a><span class="co">#exposure variables</span></span>
<span id="cb735-7"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-7" tabindex="-1"></a>cohort <span class="ot">&lt;-</span> cohort <span class="sc">%&gt;%</span> </span>
<span id="cb735-8"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Arsenic_perc=</span><span class="fu">as.factor</span>(Arsenic_perc)) <span class="sc">%&gt;%</span> </span>
<span id="cb735-9"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-9" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Cadmium_perc=</span><span class="fu">as.factor</span>(Cadmium_perc)) <span class="sc">%&gt;%</span> </span>
<span id="cb735-10"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-10" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Chromium_perc=</span><span class="fu">as.factor</span>(Chromium_perc)) <span class="sc">%&gt;%</span> </span>
<span id="cb735-11"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-11" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Copper_perc=</span><span class="fu">as.factor</span>(Copper_perc)) <span class="sc">%&gt;%</span> </span>
<span id="cb735-12"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-12" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Lead_perc=</span><span class="fu">as.factor</span>(Lead_perc)) <span class="sc">%&gt;%</span> </span>
<span id="cb735-13"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-13" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Manganese_perc=</span><span class="fu">as.factor</span>(Manganese_perc)) <span class="sc">%&gt;%</span> </span>
<span id="cb735-14"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-14" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Zinc_perc=</span><span class="fu">as.factor</span>(Zinc_perc)) <span class="sc">%&gt;%</span> </span>
<span id="cb735-15"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-15" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Arsenic_limit=</span><span class="fu">as.factor</span>(Arsenic_limit)) <span class="sc">%&gt;%</span> </span>
<span id="cb735-16"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-16" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Cadmium_limit=</span><span class="fu">as.factor</span>(Cadmium_limit)) <span class="sc">%&gt;%</span> </span>
<span id="cb735-17"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-17" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Chromium_limit=</span><span class="fu">as.factor</span>(Chromium_limit)) <span class="sc">%&gt;%</span> </span>
<span id="cb735-18"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-18" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Copper_limit=</span><span class="fu">as.factor</span>(Copper_limit)) <span class="sc">%&gt;%</span> </span>
<span id="cb735-19"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-19" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Lead_limit=</span><span class="fu">as.factor</span>(Lead_limit)) <span class="sc">%&gt;%</span> </span>
<span id="cb735-20"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-20" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Manganese_limit=</span><span class="fu">as.factor</span>(Manganese_limit)) <span class="sc">%&gt;%</span> </span>
<span id="cb735-21"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-21" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Zinc_limit=</span><span class="fu">as.factor</span>(Zinc_limit)) </span>
<span id="cb735-22"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-22" tabindex="-1"></a></span>
<span id="cb735-23"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-23" tabindex="-1"></a></span>
<span id="cb735-24"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-24" tabindex="-1"></a><span class="co">#ensure covariates are in correct variable type form </span></span>
<span id="cb735-25"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-25" tabindex="-1"></a>cohort <span class="ot">&lt;-</span> cohort <span class="sc">%&gt;%</span> </span>
<span id="cb735-26"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-26" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">racegp =</span> <span class="fu">as.factor</span>(racegp)) <span class="sc">%&gt;%</span> </span>
<span id="cb735-27"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-27" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mage =</span> <span class="fu">as.numeric</span>(mage)) <span class="sc">%&gt;%</span> </span>
<span id="cb735-28"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-28" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mage_sq =</span> <span class="fu">as.numeric</span>(mage_sq)) <span class="sc">%&gt;%</span> </span>
<span id="cb735-29"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-29" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">smoke =</span> <span class="fu">as.numeric</span>(smoke)) <span class="sc">%&gt;%</span> </span>
<span id="cb735-30"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-30" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">season_concep =</span> <span class="fu">as.factor</span>(season_concep)) <span class="sc">%&gt;%</span> </span>
<span id="cb735-31"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-31" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mothed =</span> <span class="fu">as.numeric</span>(mothed)) <span class="sc">%&gt;%</span> </span>
<span id="cb735-32"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-32" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Nitr_perc =</span> <span class="fu">as.numeric</span>(Nitr_perc)) <span class="sc">%&gt;%</span> </span>
<span id="cb735-33"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-33" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sex =</span> <span class="fu">as.factor</span>(sex))<span class="sc">%&gt;%</span> </span>
<span id="cb735-34"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb735-34" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">pov_perc =</span> <span class="fu">as.factor</span>(pov_perc))</span></code></pre></div>
</div>
<div id="fit-adjusted-logistic-regression-models-for-each-metal-for-each-categorical-variable" class="section level4 hasAnchor">
<h4>Fit adjusted logistic regression models for each metal, for each categorical variable<a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#fit-adjusted-logistic-regression-models-for-each-metal-for-each-categorical-variable" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>First, we will fit an adjusted logistic regression model for each metal, for each categorical variable, to demonstrate a variable by variable approach before diving into mixtures methods. Note that there are different regression techniques (linear and logistic are covered in another TAME module) and that here we will start with using percentage variables.</p>
<div class="sourceCode" id="cb736"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb736-1"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-1" tabindex="-1"></a>metals <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Arsenic&quot;</span>,<span class="st">&quot;Cadmium&quot;</span>,<span class="st">&quot;Chromium&quot;</span>, <span class="st">&quot;Copper&quot;</span>,<span class="st">&quot;Lead&quot;</span>,<span class="st">&quot;Manganese&quot;</span>,<span class="st">&quot;Zinc&quot;</span>)</span>
<span id="cb736-2"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-2" tabindex="-1"></a></span>
<span id="cb736-3"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-3" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(metals)) {</span>
<span id="cb736-4"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-4" tabindex="-1"></a>  metal <span class="ot">&lt;-</span> metals[[i]]</span>
<span id="cb736-5"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-5" tabindex="-1"></a>  metal <span class="ot">&lt;-</span> <span class="fu">as.name</span>(metal)</span>
<span id="cb736-6"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-6" tabindex="-1"></a>  <span class="fu">print</span>(metal)</span>
<span id="cb736-7"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-7" tabindex="-1"></a>  </span>
<span id="cb736-8"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-8" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">is.factor</span>(<span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> <span class="fu">paste0</span>(<span class="st">&quot;cohort$&quot;</span>,metal,<span class="st">&quot;_perc&quot;</span>))))) <span class="co">#check that metal var is a factor</span></span>
<span id="cb736-9"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-9" tabindex="-1"></a></span>
<span id="cb736-10"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-10" tabindex="-1"></a>  mod <span class="ot">&lt;-</span> <span class="fu">glm</span>(preterm <span class="sc">~</span> <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> <span class="fu">paste0</span>(metal,<span class="st">&quot;_perc&quot;</span>))) <span class="sc">+</span> mage <span class="sc">+</span> mage_sq<span class="sc">+</span> racegp <span class="sc">+</span> smoke <span class="sc">+</span> season_concep <span class="sc">+</span> mothed <span class="sc">+</span> Nitr_perc <span class="sc">+</span> pov_perc, <span class="at">family=</span>binomial, <span class="at">data=</span>cohort)</span>
<span id="cb736-11"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-11" tabindex="-1"></a>  </span>
<span id="cb736-12"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-12" tabindex="-1"></a>  mod_tid <span class="ot">&lt;-</span> <span class="fu">tidy</span>(mod, <span class="at">conf.int=</span><span class="cn">TRUE</span>, <span class="at">conf.level=</span><span class="fl">0.95</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb736-13"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-13" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">model_name=</span><span class="fu">paste0</span>(metal,<span class="st">&quot;_adj_perc&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb736-14"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-14" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">OR =</span> <span class="fu">exp</span>(estimate)) <span class="sc">%&gt;%</span> </span>
<span id="cb736-15"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-15" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">OR.conf.high =</span> <span class="fu">exp</span>(conf.high)) <span class="sc">%&gt;%</span> </span>
<span id="cb736-16"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-16" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">OR.conf.low =</span> <span class="fu">exp</span>(conf.low)) </span>
<span id="cb736-17"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-17" tabindex="-1"></a></span>
<span id="cb736-18"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-18" tabindex="-1"></a>  mod_tid[<span class="dv">2</span>,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">paste0</span>(metal,<span class="st">&quot;_perc_50to90&quot;</span>)</span>
<span id="cb736-19"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-19" tabindex="-1"></a>  mod_tid[<span class="dv">3</span>,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">paste0</span>(metal,<span class="st">&quot;_perc_over90&quot;</span>)</span>
<span id="cb736-20"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-20" tabindex="-1"></a>  </span>
<span id="cb736-21"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-21" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> mod_tid <span class="sc">%&gt;%</span> </span>
<span id="cb736-22"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-22" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&#39;perc_&#39;</span>, term))<span class="sc">%&gt;%</span> </span>
<span id="cb736-23"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-23" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(OR, term, <span class="at">xmin =</span> OR.conf.low, <span class="at">xmax =</span> OR.conf.high, <span class="at">height =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb736-24"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-24" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb736-25"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-25" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">trans=</span><span class="st">&quot;log10&quot;</span>)<span class="sc">+</span></span>
<span id="cb736-26"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-26" tabindex="-1"></a>  <span class="fu">geom_errorbarh</span>()</span>
<span id="cb736-27"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-27" tabindex="-1"></a>  </span>
<span id="cb736-28"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-28" tabindex="-1"></a>  <span class="fu">assign</span>(<span class="fu">paste0</span>(metal,<span class="st">&quot;_adj_perc&quot;</span>),mod_tid) </span>
<span id="cb736-29"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-29" tabindex="-1"></a>  <span class="fu">assign</span>(<span class="fu">paste0</span>(metal,<span class="st">&quot;_adj_perc_plot&quot;</span>),plot) </span>
<span id="cb736-30"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-30" tabindex="-1"></a>  </span>
<span id="cb736-31"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb736-31" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## Arsenic
## [1] TRUE
## Cadmium
## [1] TRUE
## Chromium
## [1] TRUE
## Copper
## [1] TRUE
## Lead
## [1] TRUE
## Manganese
## [1] TRUE
## Zinc
## [1] TRUE</code></pre>
<p>Plot the results:</p>
<div class="sourceCode" id="cb738"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb738-1"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb738-1" tabindex="-1"></a>perc_plots <span class="ot">&lt;-</span> <span class="fu">ggarrange</span>(Arsenic_adj_perc_plot,</span>
<span id="cb738-2"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb738-2" tabindex="-1"></a>                        Cadmium_adj_perc_plot,</span>
<span id="cb738-3"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb738-3" tabindex="-1"></a>                        Chromium_adj_perc_plot,</span>
<span id="cb738-4"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb738-4" tabindex="-1"></a>                        Copper_adj_perc_plot)</span>
<span id="cb738-5"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb738-5" tabindex="-1"></a><span class="fu">plot</span>(perc_plots)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-554-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb739"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb739-1"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb739-1" tabindex="-1"></a>perc_plots1 <span class="ot">&lt;-</span> <span class="fu">ggarrange</span>(Lead_adj_perc_plot,</span>
<span id="cb739-2"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb739-2" tabindex="-1"></a>                        Manganese_adj_perc_plot,</span>
<span id="cb739-3"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb739-3" tabindex="-1"></a>                        Zinc_adj_perc_plot)</span>
<span id="cb739-4"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb739-4" tabindex="-1"></a><span class="fu">plot</span>(perc_plots1)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-554-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Save the plots:</p>
<div class="sourceCode" id="cb740"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb740-1"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb740-1" tabindex="-1"></a><span class="fu">tiff</span>(<span class="at">file =</span> (<span class="fu">paste0</span>(Output_Folder,<span class="st">&quot;/&quot;</span>, cur_date, <span class="st">&quot;_NCbirths_pretermbirth_singlemetal_adjusted_models_percplots_1.tiff&quot;</span>)), <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">8</span>, <span class="at">units =</span> <span class="st">&quot;in&quot;</span>, <span class="at">pointsize =</span> <span class="dv">12</span>, <span class="at">res =</span> <span class="dv">600</span>)</span>
<span id="cb740-2"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb740-2" tabindex="-1"></a><span class="fu">plot</span>(perc_plots)</span>
<span id="cb740-3"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb740-3" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb740-4"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb740-4" tabindex="-1"></a></span>
<span id="cb740-5"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb740-5" tabindex="-1"></a><span class="fu">tiff</span>(<span class="at">file =</span> (<span class="fu">paste0</span>(Output_Folder,<span class="st">&quot;/&quot;</span>, cur_date, <span class="st">&quot;_NCbirths_pretermbirth_singlemetal_adjusted_models_percplots_2.tiff&quot;</span>)), <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">8</span>, <span class="at">units =</span> <span class="st">&quot;in&quot;</span>, <span class="at">pointsize =</span> <span class="dv">12</span>, <span class="at">res =</span> <span class="dv">600</span>)</span>
<span id="cb740-6"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb740-6" tabindex="-1"></a><span class="fu">plot</span>(perc_plots1)</span>
<span id="cb740-7"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb740-7" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<p>We can also run the analysis using limit variables:</p>
<div class="sourceCode" id="cb741"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb741-1"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-1" tabindex="-1"></a>  <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(metals)) {</span>
<span id="cb741-2"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-2" tabindex="-1"></a>  metal <span class="ot">&lt;-</span> metals[[i]]</span>
<span id="cb741-3"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-3" tabindex="-1"></a>  metal <span class="ot">&lt;-</span> <span class="fu">as.name</span>(metal)</span>
<span id="cb741-4"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-4" tabindex="-1"></a>  <span class="fu">print</span>(metal)</span>
<span id="cb741-5"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-5" tabindex="-1"></a>  </span>
<span id="cb741-6"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-6" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">is.factor</span>(<span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> <span class="fu">paste0</span>(<span class="st">&quot;cohort$&quot;</span>,metal,<span class="st">&quot;_limit&quot;</span>))))) <span class="co">#check that metal var is a factor</span></span>
<span id="cb741-7"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-7" tabindex="-1"></a></span>
<span id="cb741-8"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-8" tabindex="-1"></a>  mod <span class="ot">&lt;-</span> <span class="fu">glm</span>(preterm <span class="sc">~</span> <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> <span class="fu">paste0</span>(metal,<span class="st">&quot;_limit&quot;</span>)))<span class="sc">+</span> mage <span class="sc">+</span> mage_sq<span class="sc">+</span> racegp <span class="sc">+</span> smoke <span class="sc">+</span> season_concep <span class="sc">+</span> mothed <span class="sc">+</span> Nitr_perc <span class="sc">+</span> pov_perc, <span class="at">family=</span>binomial, <span class="at">data=</span>cohort)</span>
<span id="cb741-9"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-9" tabindex="-1"></a>  </span>
<span id="cb741-10"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-10" tabindex="-1"></a>  mod_tid <span class="ot">&lt;-</span> <span class="fu">tidy</span>(mod, <span class="at">conf.int=</span><span class="cn">TRUE</span>, <span class="at">conf.level=</span><span class="fl">0.95</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb741-11"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-11" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">model_name=</span><span class="fu">paste0</span>(metal,<span class="st">&quot;_adj_limit&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb741-12"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-12" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">OR =</span> <span class="fu">exp</span>(estimate)) <span class="sc">%&gt;%</span> </span>
<span id="cb741-13"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-13" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">OR.conf.high =</span> <span class="fu">exp</span>(conf.high)) <span class="sc">%&gt;%</span> </span>
<span id="cb741-14"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-14" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">OR.conf.low =</span> <span class="fu">exp</span>(conf.low)) </span>
<span id="cb741-15"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-15" tabindex="-1"></a></span>
<span id="cb741-16"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-16" tabindex="-1"></a>  mod_tid[<span class="dv">2</span>,<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="fu">paste0</span>(metal,<span class="st">&quot;_limit_over25perc&quot;</span>)</span>
<span id="cb741-17"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-17" tabindex="-1"></a>  </span>
<span id="cb741-18"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-18" tabindex="-1"></a>  plot <span class="ot">&lt;-</span> mod_tid <span class="sc">%&gt;%</span> </span>
<span id="cb741-19"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-19" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&#39;limit&#39;</span>, term))<span class="sc">%&gt;%</span></span>
<span id="cb741-20"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-20" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(OR, term, <span class="at">xmin =</span> OR.conf.low, <span class="at">xmax =</span> OR.conf.high, <span class="at">height =</span> <span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb741-21"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-21" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb741-22"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-22" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">trans=</span><span class="st">&quot;log10&quot;</span>)<span class="sc">+</span></span>
<span id="cb741-23"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-23" tabindex="-1"></a>  <span class="fu">geom_errorbarh</span>()</span>
<span id="cb741-24"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-24" tabindex="-1"></a>  </span>
<span id="cb741-25"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-25" tabindex="-1"></a>  <span class="fu">assign</span>(<span class="fu">paste0</span>(metal,<span class="st">&quot;_adj_limit&quot;</span>),mod_tid) </span>
<span id="cb741-26"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-26" tabindex="-1"></a>  <span class="fu">assign</span>(<span class="fu">paste0</span>(metal,<span class="st">&quot;_adj_limit_plot&quot;</span>),plot) </span>
<span id="cb741-27"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-27" tabindex="-1"></a>  </span>
<span id="cb741-28"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb741-28" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## Arsenic
## [1] TRUE
## Cadmium
## [1] TRUE
## Chromium
## [1] TRUE
## Copper
## [1] TRUE</code></pre>
<pre><code>## Lead
## [1] TRUE
## Manganese
## [1] TRUE
## Zinc
## [1] TRUE</code></pre>
<p>Note: you will get this warning for some of the models:<br />
“Warning: glm.fit: fitted probabilities numerically 0 or 1”.</p>
<p>This is because for the variability in the exposure data, ideally the sample size would be larger (as noted above the analysis this draws from was completed on &gt;1.3million observations).</p>
<p>Plot the results:</p>
<div class="sourceCode" id="cb744"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb744-1"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb744-1" tabindex="-1"></a>limit_plots <span class="ot">&lt;-</span> <span class="fu">ggarrange</span>(Arsenic_adj_limit_plot,</span>
<span id="cb744-2"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb744-2" tabindex="-1"></a>                        Cadmium_adj_limit_plot,</span>
<span id="cb744-3"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb744-3" tabindex="-1"></a>                        Chromium_adj_limit_plot,</span>
<span id="cb744-4"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb744-4" tabindex="-1"></a>                        Copper_adj_limit_plot)</span>
<span id="cb744-5"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb744-5" tabindex="-1"></a></span>
<span id="cb744-6"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb744-6" tabindex="-1"></a><span class="fu">plot</span>(limit_plots)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-557-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb745"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb745-1"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb745-1" tabindex="-1"></a>limit_plots1 <span class="ot">&lt;-</span> <span class="fu">ggarrange</span>(Lead_adj_limit_plot,</span>
<span id="cb745-2"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb745-2" tabindex="-1"></a>                        Manganese_adj_limit_plot,</span>
<span id="cb745-3"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb745-3" tabindex="-1"></a>                        Zinc_adj_limit_plot)</span>
<span id="cb745-4"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb745-4" tabindex="-1"></a></span>
<span id="cb745-5"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb745-5" tabindex="-1"></a><span class="fu">plot</span>(limit_plots1)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-557-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>Save the plots:</p>
<div class="sourceCode" id="cb746"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb746-1"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb746-1" tabindex="-1"></a><span class="fu">tiff</span>(<span class="at">file =</span> (<span class="fu">paste0</span>(Output_Folder,<span class="st">&quot;/&quot;</span>, cur_date, <span class="st">&quot;_NCbirths_pretermbirth_singlemetal_adjusted_models_limitplots1.tiff&quot;</span>)), <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">8</span>, <span class="at">units =</span> <span class="st">&quot;in&quot;</span>, <span class="at">pointsize =</span> <span class="dv">12</span>, <span class="at">res =</span> <span class="dv">600</span>)</span>
<span id="cb746-2"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb746-2" tabindex="-1"></a><span class="fu">plot</span>(limit_plots)</span>
<span id="cb746-3"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb746-3" tabindex="-1"></a><span class="fu">dev.off</span>()</span>
<span id="cb746-4"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb746-4" tabindex="-1"></a></span>
<span id="cb746-5"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb746-5" tabindex="-1"></a><span class="fu">tiff</span>(<span class="at">file =</span> (<span class="fu">paste0</span>(Output_Folder,<span class="st">&quot;/&quot;</span>, cur_date, <span class="st">&quot;_NCbirths_pretermbirth_singlemetal_adjusted_models_limitplots2.tiff&quot;</span>)), <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">8</span>, <span class="at">units =</span> <span class="st">&quot;in&quot;</span>, <span class="at">pointsize =</span> <span class="dv">12</span>, <span class="at">res =</span> <span class="dv">600</span>)</span>
<span id="cb746-6"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb746-6" tabindex="-1"></a><span class="fu">plot</span>(limit_plots1)</span>
<span id="cb746-7"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb746-7" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<p>Merge all of the logistic regression model results. This is the data frame that you could export for supplementary material or to view the results in Excel.</p>
<div class="sourceCode" id="cb747"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb747-1"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb747-1" tabindex="-1"></a><span class="co">#merge all model output </span></span>
<span id="cb747-2"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb747-2" tabindex="-1"></a>results_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(Arsenic_adj_perc, Arsenic_adj_limit,</span>
<span id="cb747-3"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb747-3" tabindex="-1"></a>                    Cadmium_adj_perc, Cadmium_adj_limit,</span>
<span id="cb747-4"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb747-4" tabindex="-1"></a>                    Chromium_adj_perc, Chromium_adj_limit,</span>
<span id="cb747-5"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb747-5" tabindex="-1"></a>                    Copper_adj_perc, Copper_adj_limit,</span>
<span id="cb747-6"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb747-6" tabindex="-1"></a>                    Lead_adj_perc, Lead_adj_limit,</span>
<span id="cb747-7"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb747-7" tabindex="-1"></a>                    Manganese_adj_perc, Manganese_adj_limit,</span>
<span id="cb747-8"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb747-8" tabindex="-1"></a>                    Zinc_adj_perc, Zinc_adj_limit)</span></code></pre></div>
<p>To select only the coefficients related to the primary exposures:</p>
<div class="sourceCode" id="cb748"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb748-1"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb748-1" tabindex="-1"></a>results_df <span class="ot">&lt;-</span> results_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">str_detect</span>(term, <span class="st">&#39;limit|50to90|over90&#39;</span>)) </span></code></pre></div>
<p>This file outputs the coefficients and the odds ratios (OR) of the logistic regression models all together.
+ The ORs in associated with [metal]_perc_50to90 can be interpreted as the OR comparing the odds of preterm birth among individuals in the 50th to 90th percentile of [metal] exposure compared to those below the 50th.
+ The ORs in associated with [metal]_perc_over90 can be interpreted as the OR comparing the odds of preterm birth among individuals above the 90th percentile of [metal] exposure compared to those below the 50th.
+ The ORs in associated with [metal]_limit_over25perc can be interpreted as the OR comparing the odds of preterm birth among individuals living in census tracts in with tests exceeding the an EPA standard for [metal] in 25% or more tests versus tracts with less that 25% of tests exceeding the standard</p>
</div>
</div>
<div id="answer-to-environmental-health-question-1-14" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 1<a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#answer-to-environmental-health-question-1-14" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer also <strong>Environmental Health Question #1</strong></em>: What is the risk of preterm birth associated with exposure to each of arsenic, lead, cadmium, chromium, manganese, copper and zinc via private well water individually?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: Using the interpretation guides described in the prior paragraph and the “_NCbirths_pretermbirth_singlemetal_adjusted_models.csv” file, you can answer this question. For example, for cadmium, compared to individuals residing in census tracts with cadmium below the 50th percentile, those residing in tracts with lead between the 50th and 90th percentile had a 7% increase in the adjusted odds of PTB (aOR 1.07 (95% CI: 1.00,1.14)) and those in tracts with cadmium above the 90th percentile had a 8% increased adjusted odds of PTB (aOR 1.08 (95% CI: 0.97,1.20). Compared to individuals in tracts with less than 25% of tests exceeding the standard for lead (note this is the EPA treatment technique action level=15 ppb), individuals residing in census tracts where 25% or more of tests exceeded the MCL had 1.23 (95% CI: 0.81,1.81) times the adjusted odds of preterm birth. IMPORTANT NOTE: as described above, these results differ from the publication (Eaves et al. 2023) because this scripted example is conducted on a smaller subsetted dataset.</p>
</div>
<p>While the single contaminant models provide useful information, they cannot inform us of the effect of multiple simultaneous exposures or account for co-occurring contaminant confounding. Therefore, we want to utilize quantile g-compuation to assess mixtures.</p>
</div>
</div>
<div id="mixtures-model-with-standard-qqcomp" class="section level2 hasAnchor">
<h2>Mixtures Model with Standard qqcomp<a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#mixtures-model-with-standard-qqcomp" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb749"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb749-1"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb749-1" tabindex="-1"></a><span class="co">#list of exposure variables</span></span>
<span id="cb749-2"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb749-2" tabindex="-1"></a>Xnm <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&#39;Arsenic.Mean_avg&#39;</span>, <span class="st">&#39;Cadmium.Mean_avg&#39;</span>, <span class="st">&#39;Lead.Mean_avg&#39;</span>, <span class="st">&#39;Manganese.Mean_avg&#39;</span>, <span class="st">&#39;Chromium.Mean_avg&#39;</span>, <span class="st">&#39;Copper.Mean_avg&#39;</span>, <span class="st">&#39;Zinc.Mean_avg&#39;</span>)</span>
<span id="cb749-3"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb749-3" tabindex="-1"></a><span class="co">#list of covariates </span></span>
<span id="cb749-4"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb749-4" tabindex="-1"></a>covars <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&#39;mage&#39;</span>,<span class="st">&#39;mage_sq&#39;</span>,<span class="st">&#39;racegp&#39;</span>,<span class="st">&#39;smoke&#39;</span>,<span class="st">&#39;season_concep&#39;</span>,<span class="st">&#39;mothed&#39;</span>,<span class="st">&#39;Nitr_perc&#39;</span>,<span class="st">&#39;pov_perc&#39;</span>)</span>
<span id="cb749-5"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb749-5" tabindex="-1"></a></span>
<span id="cb749-6"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb749-6" tabindex="-1"></a><span class="co">#fit adjusted model</span></span>
<span id="cb749-7"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb749-7" tabindex="-1"></a>PTB_adj_ppb <span class="ot">&lt;-</span> <span class="fu">qgcomp.noboot</span>(preterm<span class="sc">~</span>.,</span>
<span id="cb749-8"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb749-8" tabindex="-1"></a>                          <span class="at">expnms=</span>Xnm, <span class="at">dat=</span>cohort[,<span class="fu">c</span>(Xnm,covars,<span class="st">&#39;preterm&#39;</span>)], <span class="at">family=</span><span class="fu">binomial</span>(), <span class="at">q=</span><span class="dv">4</span>)</span></code></pre></div>
<p>In English, <code>preterm~.</code> is saying fit a model that has preterm (1/0) as the dependent variable and then the independent variables (exposures and covariates) are all other variables in the dataset (<code>.</code>). <code>expnms=Xnm</code> is saying that the mixture of exposures is given by the vector <code>Xnm,</code> defined above. <code>dat=cohort[,c(Xnm,covars,'preterm')]</code> is saying that the dataset to be used to fit this model includes all columns in the cohort dataset that are listed in the <code>Xnm</code> and <code>covars</code> vectors and also the <code>preterm</code> variable. <code>family=binomial()</code> is saying that the outcome is a binary outcome and therefore the model will fit a logistic regression model. <code>q=4</code> is saying break the exposures into quartiles, other options would be q=3 for teriltes, q=5 for quintiles and so forth.</p>
<p>This is a summary of the qgcomp model output</p>
<div class="sourceCode" id="cb750"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb750-1"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb750-1" tabindex="-1"></a>PTB_adj_ppb</span></code></pre></div>
<pre><code>## Scaled effect size (positive direction, sum of positive coefficients = 0.0969)
##   Cadmium.Mean_avg  Chromium.Mean_avg Manganese.Mean_avg      Zinc.Mean_avg 
##             0.4556             0.4027             0.1006             0.0412 
## 
## Scaled effect size (negative direction, sum of negative coefficients = -0.0532)
## Arsenic.Mean_avg  Copper.Mean_avg    Lead.Mean_avg 
##            0.452            0.396            0.152 
## 
## Mixture log(OR) (Delta method CI):
## 
##              Estimate Std. Error  Lower CI Upper CI Z value  Pr(&gt;|z|)
## (Intercept) -2.189951   0.270100 -2.719337 -1.66056 -8.1079 4.441e-16
## psi1         0.043705   0.025982 -0.007219  0.09463  1.6821   0.09255</code></pre>
<p>This output can be interpreted as:</p>
<ul>
<li>Cadmium, chromium, manganese and zinc had positive effects, as in they increased the risk of preterm birth. Arsenic, coppper and lead had negative effects, as in they reduced the risk of preterm birth.</li>
<li>The total effect of all positive acting mixture components is given by the sum of positive coefficients = 0.0969, total effect of all negative acting mixture components is given by the sum of negative coefficients = -0.0532.</li>
<li>The numbers underneath each of the individual mixture component are the weights assigned to each component. These sum to 1 in each direction. They represent the relative contribution of each component to the effect in that direction. If only one components was acting in the positive or negative direction, it would have a weight of 1. A component’s weight multiplied by the sum of the coefficient’s in the relevant direction is that individual component’s coefficient and represents the independent effect of that component (e.g. cadmium log(OR) = 0.0969*0.4556=0.0441).</li>
<li>The overall mixture effect (i.e. the log(OR) when all exposures are increased by one quartile) is given by psi1. Here it equals 0.0437. Note that this value is equal to combining the sum of coefficients in the positive direction adn the sum in the negative direction (ie. 0.0969-0.0532= 0.0437)</li>
</ul>
<p>IMPORTANT NOTE: as described above, these results differ from the publication (Eaves et al. 2023) because this scripted example is conducted on a smaller subsetted dataset.</p>
<p>This is the plot that gives you the weights of the components</p>
<div class="sourceCode" id="cb752"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb752-1"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb752-1" tabindex="-1"></a><span class="fu">plot</span>(PTB_adj_ppb)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-563-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>To save the plot:</p>
<div class="sourceCode" id="cb753"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb753-1"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb753-1" tabindex="-1"></a><span class="fu">tiff</span>(<span class="at">file =</span> (<span class="fu">paste0</span>(Output_Folder,<span class="st">&quot;/&quot;</span>, cur_date, <span class="st">&quot;_NCbirths_pretermbirth_qgcomp_weights.tiff&quot;</span>)), <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">8</span>, <span class="at">units =</span> <span class="st">&quot;in&quot;</span>, <span class="at">pointsize =</span> <span class="dv">12</span>, <span class="at">res =</span> <span class="dv">600</span>)</span>
<span id="cb753-2"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb753-2" tabindex="-1"></a><span class="fu">plot</span>(PTB_adj_ppb)</span>
<span id="cb753-3"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb753-3" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<div id="answer-to-environmental-health-question-2-15" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 2<a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#answer-to-environmental-health-question-2-15" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #2</strong></em>: What is the risk of preterm birth associated with combined exposure to arsenic, lead, cadmium, chromium, manganese, copper and zinc (ie. a mixture) via private well water?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: When all exposures (arsenic, lead, cadmium, chromium, manganese, copper and zinc) are increased in concentration by one quartile the odds ratio is 1.044 (exp(0.043705)). IMPORTANT NOTE: as described above, these results differ from the publication (Eaves et al. 2023) because this scripted example is conducted on a smaller subsetted dataset.</p>
</div>
</div>
<div id="answer-to-environmental-health-question-3-9" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 3<a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#answer-to-environmental-health-question-3-9" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer also <strong>Environmental Health Question #3</strong></em>: Which of these chemicals has the strongest effect on preterm birth risk?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: The mixture component with the strongest effect is the one that has the largest independent effect given my the component’s coefficient (which can be calculated by (sum of coefficients in relevant direction)<em>(component weight), and, as shown below can also be generated to ouput into results files). In this case, the components with the largest independent effect is cadmium (0.0969</em>0.4556=0.0441). IMPORTANT NOTE: as described above, these results differ from the publication (Eaves et al. 2023) because this scripted example is conducted on a smaller subsetted dataset.</p>
</div>
</div>
<div id="answer-to-environmental-health-question-4-6" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 4<a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#answer-to-environmental-health-question-4-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer also <strong>Environmental Health Question #4</strong></em>: Which of these chemicals increases the risk of preterm birth and which decreases the risk of preterm birth?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: This is indicated by the direction of effect for each component. Thus, the mixture components that increase the risk of preterm birth are cadmium, chromium, manganese and zinc, while the mixture components that decrease the risk of preterm birth are arsenic, copper and lead. IMPORTANT NOTE: as described above, these results differ from the publication (Eaves et al. 2023) because this scripted example is conducted on a smaller subsetted dataset.</p>
</div>
<p>We can export the mixtures modeling results using the following code, which stores the data in three different files:
+ Results_SlopeParams outputs the overall mixture effect results
+ Results_MetalCoeffs outputs the individual mixture components (metals) coefficients. Note that this will also output coefficient for covariates included in the model.
+ Results_MetalWeights outputs the individual mixture components (metals) weights</p>
<div class="sourceCode" id="cb754"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb754-1"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-1" tabindex="-1"></a>allmodels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;PTB_adj_ppb&quot;</span>) <span class="co">#if you run more than one qgcomp model, list them here and the following code can output the results in clean format all together</span></span>
<span id="cb754-2"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-2" tabindex="-1"></a></span>
<span id="cb754-3"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-3" tabindex="-1"></a>clean_print <span class="ot">&lt;-</span> <span class="cf">function</span>(x){</span>
<span id="cb754-4"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-4" tabindex="-1"></a>  output  <span class="ot">=</span> <span class="fu">data.frame</span>(</span>
<span id="cb754-5"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-5" tabindex="-1"></a>    x<span class="sc">$</span>coef,</span>
<span id="cb754-6"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-6" tabindex="-1"></a>    <span class="fu">sqrt</span>(x<span class="sc">$</span>var.coef),</span>
<span id="cb754-7"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-7" tabindex="-1"></a>    x<span class="sc">$</span>ci.coef,</span>
<span id="cb754-8"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-8" tabindex="-1"></a>    x<span class="sc">$</span>pval</span>
<span id="cb754-9"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-9" tabindex="-1"></a>  )</span>
<span id="cb754-10"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-10" tabindex="-1"></a>  <span class="fu">names</span>(output) <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;Estimate&quot;</span>, <span class="st">&quot;Std. Error&quot;</span>, <span class="st">&quot;Lower CI&quot;</span>, <span class="st">&quot;Upper CI&quot;</span>, <span class="st">&quot;p value&quot;</span>)</span>
<span id="cb754-11"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-11" tabindex="-1"></a>  <span class="fu">return</span>(output)</span>
<span id="cb754-12"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-12" tabindex="-1"></a>}</span>
<span id="cb754-13"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-13" tabindex="-1"></a></span>
<span id="cb754-14"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-14" tabindex="-1"></a>Results_SlopeParams <span class="ot">&lt;-</span> <span class="fu">data.frame</span>() <span class="co">#empty vector to append dfs to</span></span>
<span id="cb754-15"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-15" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> allmodels){</span>
<span id="cb754-16"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-16" tabindex="-1"></a>  <span class="fu">print</span>(i)</span>
<span id="cb754-17"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-17" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> <span class="fu">paste0</span>(<span class="st">&quot;clean_print(&quot;</span>,i,<span class="st">&quot;)&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb754-18"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-18" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="st">&quot;Parameter&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb754-19"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-19" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">&quot;Model&quot;</span> <span class="ot">=</span> i) </span>
<span id="cb754-20"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-20" tabindex="-1"></a>  Results_SlopeParams <span class="ot">&lt;-</span> <span class="fu">rbind</span>(Results_SlopeParams,df)</span>
<span id="cb754-21"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-21" tabindex="-1"></a>}</span>
<span id="cb754-22"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-22" tabindex="-1"></a>Results_SlopeParams <span class="ot">&lt;-</span> Results_SlopeParams <span class="sc">%&gt;%</span> </span>
<span id="cb754-23"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-23" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">OR=</span><span class="fu">exp</span>(Estimate)) <span class="sc">%&gt;%</span> </span>
<span id="cb754-24"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-24" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">UpperCI_OR=</span><span class="fu">exp</span>(<span class="st">`</span><span class="at">Upper CI</span><span class="st">`</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb754-25"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-25" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">LowerCI_OR=</span><span class="fu">exp</span>(<span class="st">`</span><span class="at">Lower CI</span><span class="st">`</span>))</span>
<span id="cb754-26"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-26" tabindex="-1"></a></span>
<span id="cb754-27"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-27" tabindex="-1"></a>Results_MetalCoeffs <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb754-28"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-28" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> allmodels){</span>
<span id="cb754-29"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-29" tabindex="-1"></a>  <span class="fu">print</span>(i)</span>
<span id="cb754-30"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-30" tabindex="-1"></a>  df <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> <span class="fu">paste0</span>(<span class="st">&quot;as.data.frame(summary(&quot;</span>,i,<span class="st">&quot;$fit)$coefficients[,])&quot;</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb754-31"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-31" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">&quot;Model&quot;</span> <span class="ot">=</span> i)</span>
<span id="cb754-32"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-32" tabindex="-1"></a>  df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="at">var=</span><span class="st">&quot;variable&quot;</span>)  </span>
<span id="cb754-33"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-33" tabindex="-1"></a>  Results_MetalCoeffs<span class="ot">&lt;-</span> <span class="fu">rbind</span>(Results_MetalCoeffs,df)</span>
<span id="cb754-34"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-34" tabindex="-1"></a>}</span>
<span id="cb754-35"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-35" tabindex="-1"></a></span>
<span id="cb754-36"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-36" tabindex="-1"></a>Results_MetalWeights <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb754-37"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-37" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> allmodels){</span>
<span id="cb754-38"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-38" tabindex="-1"></a>  Results_PWeights <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> <span class="fu">paste0</span>(<span class="st">&quot;as.data.frame(&quot;</span>,i,<span class="st">&quot;$pos.weights)&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb754-39"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-39" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="st">&quot;Metal&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb754-40"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-40" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;Weight&quot;</span> <span class="ot">=</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb754-41"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-41" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">&quot;Weight Direction&quot;</span> <span class="ot">=</span> <span class="st">&quot;Positive&quot;</span>)</span>
<span id="cb754-42"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-42" tabindex="-1"></a>  Results_NWeights <span class="ot">&lt;-</span> <span class="fu">eval</span>(<span class="fu">parse</span>(<span class="at">text =</span> <span class="fu">paste0</span>(<span class="st">&quot;as.data.frame(&quot;</span>,i,<span class="st">&quot;$neg.weights)&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb754-43"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-43" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="st">&quot;Metal&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb754-44"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-44" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">&quot;Weight&quot;</span> <span class="ot">=</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb754-45"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-45" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">&quot;Weight Direction&quot;</span> <span class="ot">=</span> <span class="st">&quot;Negative&quot;</span>)</span>
<span id="cb754-46"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-46" tabindex="-1"></a>  Results_Weights <span class="ot">&lt;-</span> <span class="fu">rbind</span>(Results_PWeights, Results_NWeights) <span class="sc">%&gt;%</span></span>
<span id="cb754-47"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-47" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">&quot;Model&quot;</span> <span class="ot">=</span> i) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb754-48"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-48" tabindex="-1"></a>  Results_MetalWeights <span class="ot">&lt;-</span> <span class="fu">rbind</span>(Results_MetalWeights, Results_Weights)</span>
<span id="cb754-49"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-49" tabindex="-1"></a>}</span>
<span id="cb754-50"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-50" tabindex="-1"></a></span>
<span id="cb754-51"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-51" tabindex="-1"></a><span class="fu">write.csv</span>(Results_SlopeParams, <span class="fu">paste0</span>(Output_Folder,<span class="st">&quot;/&quot;</span>, cur_date, <span class="st">&quot;_qgcomp_Results_SlopeParams.csv&quot;</span>), <span class="at">row.names=</span><span class="cn">TRUE</span>)</span>
<span id="cb754-52"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-52" tabindex="-1"></a><span class="fu">write.csv</span>(Results_MetalCoeffs, <span class="fu">paste0</span>(Output_Folder,<span class="st">&quot;/&quot;</span>, cur_date, <span class="st">&quot;_qgcomp_Results_MetalCoeffs.csv&quot;</span>), <span class="at">row.names=</span><span class="cn">TRUE</span>)</span>
<span id="cb754-53"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#cb754-53" tabindex="-1"></a><span class="fu">write.csv</span>(Results_MetalWeights, <span class="fu">paste0</span>(Output_Folder,<span class="st">&quot;/&quot;</span>, cur_date, <span class="st">&quot;_qgcomp_Results_MetalWeights.csv&quot;</span>), <span class="at">row.names=</span><span class="cn">TRUE</span>)</span></code></pre></div>
</div>
</div>
<div id="concluding-remarks-25" class="section level2 hasAnchor">
<h2>Concluding Remarks<a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#concluding-remarks-25" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In conclusion, this module reviews a suite of methodologies researches can use to answer different questions relevant to environmental mixtures and their relationships to health outcomes. In this specific scripted example we utilized a large epidemiological dataset (for educational purposes, subsetted to a reduced sample size), to demonstrate using logistic regression to assess single contaminant associations with a health outcome (preterm birth) and quantile g computation to assess mixture effects with a health outcome.</p>
</div>
<div id="additional-resources-11" class="section level2 hasAnchor">
<h2>Additional Resources<a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#additional-resources-11" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The field of mixtures is vast, with many different approaches and example studies to learn from as analysts lead in their own analyses. Some resources that can be helpful include the following reviews:</p>
<ul>
<li>Our recent review on mixtures methodologies, particularly in the field of sufficient similarity, titled <a href="https://www.sciencedirect.com/science/article/abs/pii/S2468202023000323?via%3Dihub">Wrangling whole mixtures risk assessment: Recent advances in determining sufficient similarity</a></li>
<li>Two more general, epidemiology-focused reviews on mixtures questions and methodologies, titled <a href="https://link.springer.com/article/10.1007/s40572-019-00229-5">Complex Mixtures, Complex Analyses: an Emphasis on Interpretable Results</a> and <a href="https://pubmed.ncbi.nlm.nih.gov/30643709/">Environmental exposure mixtures: questions and methods to address them</a></li>
<li><a href="https://bookdown.org/andreabellavia/mixtures/preface.html">A helpful online toolkit</a> for mixtures analyses generated by Andrea Bellavia, PhD</li>
</ul>
<p>Some helpful mixtures case studies include the following:</p>
<ul>
<li>Our recent study that implemented quantile g-computation statistics to identify chemicals present in wildfire smoke emissions that impact toxicity, published as the following: Rager JE, Clark J, Eaves LA, Avula V, Niehoff NM, Kim YH, Jaspers I, Gilmour MI. Mixtures modeling identifies chemical inducers versus repressors of toxicity associated with wildfire smoke. Sci Total Environ. 2021 Jun 25;775:145759. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/33611182/">33611182</a>.</li>
<li>Another study from our group that implemented quantile g-computation identify placental gene networks that had altered expression in response to cord tissue mixtures of metals, published as the following: Eaves LA, Bulka CM, Rager JE, Galusha AL, Parsons PJ, O’Shea TM and Fry RC. Metals mixtures modeling identifies birth weight-associated gene networks in the placentas of children born extremely preterm. Chemosphere. 2022;137469.PMID:<a href="https://pubmed.ncbi.nlm.nih.gov/36493891/">36493891</a></li>
</ul>
<p>Many other groups also leverage quantile g-computation, with the following as exemplar case studies:</p>
<ul>
<li><a href="https://link.springer.com/article/10.1186/s12940-021-00724-z">Prenatal exposure to consumer product chemical mixtures and size for gestational age at delivery</a></li>
<li><a href="https://www.sciencedirect.com/science/article/pii/S0160412023005718">Use of personal care product mixtures and incident hormone-sensitive cancers in the Sister Study: A U.S.-wide prospective cohort</a></li>
</ul>
<p><label class="tykfont">
Test Your Knowledge
</label></p>
<div class="tyk">
<p>Using the metals dataset within the <em>qgcomp</em> package (see the <a href="https://cran.r-project.org/web/packages/qgcomp/vignettes/qgcomp-vignette.html">package vignette</a> for how to access), answer the following three mixtures-related environmental health questions using quantile g-computation, focusing on a mixture of arsenic, copper, zinc and lead:</p>
<ol style="list-style-type: decimal">
<li>What is the risk of disease associated with combined exposure to each of the chemicals?</li>
<li>Which of these chemicals has the strongest effect on disease?</li>
<li>Which of these chemicals increases the risk of disease and which decreases the risk of disease?</li>
</ol>
<p>Note that disease is given by the variable <code>disease_state</code> (1 = case, 0 = non-case).</p>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="omics-and-system-biology-transcriptomic-applications.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
