<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6.5 Mixtures Analysis Methods Part 3: Sufficient Similarity | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="6.5 Mixtures Analysis Methods Part 3: Sufficient Similarity | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6.5 Mixtures Analysis Methods Part 3: Sufficient Similarity | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  <meta name="twitter:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  

<meta name="author" content="Rager Lab" />


<meta name="date" content="2024-10-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="Icons_Used_Throughout/Favicon.png" type="image/x-icon" />
<link rel="prev" href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"/>
<link rel="next" href="toxicokinetic-modeling.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>
<link href="libs/table1-1.0/table1_defaults.css" rel="stylesheet" />
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-FGFKVX2X3G"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-FGFKVX2X3G');
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="Icons_Used_Throughout/TAME 2.0 Logo.png"><br></a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Chapter 1 Introductory <br>Data Science</b></span></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html"><i class="fa fa-check"></i>1.1 FAIR Data Management Practices</a>
<ul>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-training-module"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-fair"><i class="fa fa-check"></i>Introduction to FAIR</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#breaking-down-fair-letter-by-letter"><i class="fa fa-check"></i>Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#what-does-this-mean-for-you"><i class="fa fa-check"></i>What Does This Mean for You?</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#data-repositories-for-sharing-of-data"><i class="fa fa-check"></i>Data Repositories for Sharing of Data</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#recent-shifts-in-regulatory-policies-for-data-sharing"><i class="fa fa-check"></i>Recent Shifts in Regulatory Policies for Data Sharing</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#additional-training-resources-on-fair"><i class="fa fa-check"></i>Additional Training Resources on FAIR</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html"><i class="fa fa-check"></i>1.2 Data Sharing through Online Repositories</a>
<ul>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#an-overview-and-example-with-the-dataverse-repository"><i class="fa fa-check"></i>An Overview and Example with the Dataverse Repository</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#introduction-to-training-module-1"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#data-repositories"><i class="fa fa-check"></i>Data Repositories</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#the-dataverse-project"><i class="fa fa-check"></i>The Dataverse Project</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#what-is-a-dataverse"><i class="fa fa-check"></i>What is a Dataverse?</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#metadata"><i class="fa fa-check"></i>Metadata</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataverse"><i class="fa fa-check"></i>Creating a Dataverse</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataset"><i class="fa fa-check"></i>Creating a Dataset</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#concluding-remarks-1"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html"><i class="fa fa-check"></i>1.3 File Management using Github</a>
<ul>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#introduction-to-training-module-2"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-an-account"><i class="fa fa-check"></i>Creating an Account</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-a-repository"><i class="fa fa-check"></i>Creating a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#uploading-code"><i class="fa fa-check"></i>Uploading Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#adding-subfolders-in-a-repository"><i class="fa fa-check"></i>Adding Subfolders in a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-code"><i class="fa fa-check"></i>Updating Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-repository-titles-and-structure-to-support-a-manuscript"><i class="fa fa-check"></i>Updating Repository Titles and Structure to Support a Manuscript</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#tracking-code-changes-using-github-branches"><i class="fa fa-check"></i>Tracking Code Changes using Github Branches</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#concluding-remarks-2"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html"><i class="fa fa-check"></i>1.4 Data Wrangling in Excel</a>
<ul>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#introduction-to-training-module-3"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#save-a-copy-of-the-soon-to-be-organized-and-cleaned-dataset-as-a-new-file"><i class="fa fa-check"></i>Save a Copy of the Soon-To-Be Organized and Cleaned Dataset as a New File</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-extraneous-white-space"><i class="fa fa-check"></i>Remove Extraneous White Space</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#replace-missing-data-with-na"><i class="fa fa-check"></i>Replace Missing Data with “NA”</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-metadata-tab"><i class="fa fa-check"></i>Create a Metadata Tab</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#abbreviate-and-capitalize-categorical-data"><i class="fa fa-check"></i>Abbreviate and Capitalize Categorical Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#alphabetize-sort-the-data-by-the-categorical-variable-of-interest"><i class="fa fa-check"></i>Alphabetize (Sort) the Data by the Categorical Variable of Interest</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-new-subject-number-column"><i class="fa fa-check"></i>Create a New Subject Number Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-special-symbols-and-dashes"><i class="fa fa-check"></i>Remove Special Symbols and Dashes</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#bold-all-column-names-and-center-all-data"><i class="fa fa-check"></i>Bold all Column Names and Center all Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-subject-identifier-column"><i class="fa fa-check"></i>Create a Subject Identifier Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#separate-subject-demographic-data-from-experimental-measurements"><i class="fa fa-check"></i>Separate Subject Demographic Data from Experimental Measurements</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#convert-data-from-wide-to-long-format"><i class="fa fa-check"></i>Convert Data from Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#pivoting-data-from-a-wide-to-long-format"><i class="fa fa-check"></i>Pivoting Data from a Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#generating-summary-level-statistics-with-pivot-tables"><i class="fa fa-check"></i>Generating Summary-Level Statistics with Pivot Tables</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#excel-vs.-r-which-should-you-use"><i class="fa fa-check"></i>Excel vs. R: Which Should You Use?</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#concluding-remarks-3"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 2 Coding in R</b></span></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html"><i class="fa fa-check"></i>2.1 Downloading and Programming in R</a>
<ul>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-training-module-4"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#general-introduction-and-installation-of-r-and-rstudio"><i class="fa fa-check"></i>General Introduction and Installation of R and RStudio</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-r-packages"><i class="fa fa-check"></i>Introduction to R Packages</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#scripting-basics"><i class="fa fa-check"></i>Scripting Basics</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#code-troubleshooting"><i class="fa fa-check"></i>Code Troubleshooting</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#concluding-remarks-4"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html"><i class="fa fa-check"></i>2.2 Coding “Best” Practices</a>
<ul>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#introduction-to-training-module-5"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#scripting-file-types"><i class="fa fa-check"></i>Scripting File Types</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-headers-and-annotation"><i class="fa fa-check"></i>Script Headers and Annotation</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#coding-style"><i class="fa fa-check"></i>Coding Style</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-organization"><i class="fa fa-check"></i>Script Organization</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#concluding-remarks-5"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html"><i class="fa fa-check"></i>2.3 Data Manipulation and Reshaping</a>
<ul>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-training-module-6"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#data-manipulation-using-base-r"><i class="fa fa-check"></i>Data Manipulation Using Base R</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-tidyverse"><i class="fa fa-check"></i>Introduction to Tidyverse</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#concluding-remarks-6"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html"><i class="fa fa-check"></i>2.4 Improving Coding Efficiencies</a>
<ul>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#introduction-to-training-module-7"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#loops"><i class="fa fa-check"></i>Loops</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#functions"><i class="fa fa-check"></i>Functions</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#list-operations"><i class="fa fa-check"></i>List operations</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#concluding-remarks-7"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#additional-resources-2"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="part"><span><b>Chapter 3 Basics of <br>Data Analysis and Visualizations</b></span></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html"><i class="fa fa-check"></i>3.1 Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-data-visualizations"><i class="fa fa-check"></i>Introduction to Data Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-training-module-8"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#density-plot-visualization"><i class="fa fa-check"></i>Density Plot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#boxplot-visualization"><i class="fa fa-check"></i>Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#correlation-visualizations"><i class="fa fa-check"></i>Correlation Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#heatmap-visualization"><i class="fa fa-check"></i>Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#concluding-remarks-8"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html"><i class="fa fa-check"></i>3.2 Improving Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-data-visulization-conventions"><i class="fa fa-check"></i>Introduction to Data Visulization Conventions</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-training-module-9"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-boxplot-visualization"><i class="fa fa-check"></i>Creating an Improved Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-heatmap-visualization"><i class="fa fa-check"></i>Creating an Improved Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-multi-plot-figures"><i class="fa fa-check"></i>Creating Multi-Plot Figures</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#concluding-remarks-9"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html"><i class="fa fa-check"></i>3.3 Normality Tests and Data Transformations</a>
<ul>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#introduction-to-training-module-10"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#what-is-a-normal-distribution"><i class="fa fa-check"></i>What is a Normal Distribution?</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#qualitative-assessment-of-normality"><i class="fa fa-check"></i>Qualitative Assessment of Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#quantitative-normality-assessment"><i class="fa fa-check"></i>Quantitative Normality Assessment</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#data-transformation"><i class="fa fa-check"></i>Data Transformation</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#additional-considerations-regarding-normality"><i class="fa fa-check"></i>Additional Considerations Regarding Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#concluding-remarks-10"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html"><i class="fa fa-check"></i>3.4 Intoduction to Statistical Tests</a>
<ul>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#introduction-to-training-module-11"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#assessing-normality-homogeneity-of-variance"><i class="fa fa-check"></i>Assessing Normality &amp; Homogeneity of Variance</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#two-group-visualizations-and-statistical-comparisons-using-the-t-test"><i class="fa fa-check"></i>Two-Group Visualizations and Statistical Comparisons using the T-Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#three-group-visualizations-and-statistical-comparisons-using-an-anova"><i class="fa fa-check"></i>Three-Group Visualizations and Statistical Comparisons using an ANOVA</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#regression-modeling-and-visualization-linear-and-logistic-regressions"><i class="fa fa-check"></i>Regression Modeling and Visualization: Linear and Logistic Regressions</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#statistical-evaluations-of-categorical-data-using-the-chi-squared-test-and-fishers-exact-test"><i class="fa fa-check"></i>Statistical Evaluations of Categorical Data using the Chi-Squared Test and Fisher’s Exact Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#concluding-remarks-11"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 4 Converting Wet Lab Data into Dry Lab Analyses</b></span></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html"><i class="fa fa-check"></i>4.1 Overview of Experimental Design and Example Data</a>
<ul>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#introduction-to-training-module-12"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#replicates"><i class="fa fa-check"></i>Replicates</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#orientation-to-example-data-for-chapter-4"><i class="fa fa-check"></i>Orientation to Example Data for Chapter 4</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#concluding-remarks-12"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html"><i class="fa fa-check"></i>4.2 Data Import, Processing, and Summary Statistics</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#introduction-to-training-module-13"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#data-import"><i class="fa fa-check"></i>Data Import</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#handling-missing-values"><i class="fa fa-check"></i>Handling Missing Values</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#averaging-replicates"><i class="fa fa-check"></i>Averaging Replicates</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#descriptive-statistics"><i class="fa fa-check"></i>Descriptive Statistics</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#normality-assessment-and-data-transformation"><i class="fa fa-check"></i>Normality Assessment and Data Transformation</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#concluding-remarks-13"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html"><i class="fa fa-check"></i>4.3 Data Import from PDF Sources</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#introduction-to-training-module-14"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-from-many-single-pdfs-with-the-same-formatting"><i class="fa fa-check"></i>Importing Data from Many Single PDFs with the Same Formatting</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-stored-in-pdf-tables"><i class="fa fa-check"></i>Importing Data Stored in PDF Tables</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#concluding-remarks-14"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html"><i class="fa fa-check"></i>4.4 Two Group Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#introduction-to-training-module-15"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#overview-of-two-group-statistical-tests"><i class="fa fa-check"></i>Overview of Two Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#statistical-vs.-biological-significance"><i class="fa fa-check"></i>Statistical vs. Biological Significance</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#unpaired-test-example"><i class="fa fa-check"></i>Unpaired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#paired-test-example"><i class="fa fa-check"></i>Paired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#visualizing-results"><i class="fa fa-check"></i>Visualizing Results</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#concluding-remarks-15"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html"><i class="fa fa-check"></i>4.5 Multi-Group and Multi-Variable Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#introduction-to-training-module-16"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#overview-of-multi-group-statistical-tests"><i class="fa fa-check"></i>Overview of Multi-Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#multi-group-analysis-example"><i class="fa fa-check"></i>Multi-Group Analysis Example</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#visualization-of-multi-group-statistical-results"><i class="fa fa-check"></i>Visualization of Multi-Group Statistical Results</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#concluding-remarks-16"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html"><i class="fa fa-check"></i>4.6 Advanced Multi-Group Comparisons</a>
<ul>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#introduction-to-training-module-17"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#advanced-multi-group-comparisons-1"><i class="fa fa-check"></i>Advanced Multi-Group Comparisons</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#two-way-anova-example"><i class="fa fa-check"></i>Two-way ANOVA Example</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#ancova-example"><i class="fa fa-check"></i>ANCOVA Example</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#concluding-remarks-17"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 5 Machine Learning &amp; Artificial Intelligence</b></span></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><i class="fa fa-check"></i>5.1 Introduction to Artificial Intelligence, Machine Learning, and Predictive Modeling for Environmental Health</a>
<ul>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#introduction-to-training-module-18"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#general-historical-context-and-taxonomy-of-modern-aiml"><i class="fa fa-check"></i>General Historical Context and Taxonomy of Modern AI/ML</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#application-of-machine-learning-in-environmental-health-science"><i class="fa fa-check"></i>Application of Machine Learning in Environmental Health Science</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#concluding-remarks-18"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html"><i class="fa fa-check"></i>5.2 Supervised Machine Learning</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-training-module-19"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-machine-learning"><i class="fa fa-check"></i>Types of Machine Learning</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-supervised-machine-learning-algorithms"><i class="fa fa-check"></i>Types of Supervised Machine Learning Algorithms</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#training-supervised-machine-learning-models"><i class="fa fa-check"></i>Training Supervised Machine Learning Models</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#assessing-classification-based-model-performance"><i class="fa fa-check"></i>Assessing Classification-Based Model Performance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-activity-and-example-dataset"><i class="fa fa-check"></i>Introduction to Activity and Example Dataset</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#testing-for-differences-in-predictor-variables-across-the-outcome-classes"><i class="fa fa-check"></i>Testing for Differences in Predictor Variables across the Outcome Classes</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#predicting-ias-detection-with-a-random-forest-rf-model"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#class-imbalance"><i class="fa fa-check"></i>Class Imbalance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#concluding-remarks-19"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html"><i class="fa fa-check"></i>5.3 Supervised Machine Learning Model Interpretation</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-training-module-20"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance"><i class="fa fa-check"></i>Variable Importance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary"><i class="fa fa-check"></i>Decision Boundary</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-example-dataset-and-activity"><i class="fa fa-check"></i>Introduction to Example Dataset and Activity</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#predicting-ias-detection-with-a-random-forest-rf-model-1"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance-plot"><i class="fa fa-check"></i>Variable Importance Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot"><i class="fa fa-check"></i>Decision Boundary Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot-incorporating-smote"><i class="fa fa-check"></i>Decision Boundary Plot Incorporating SMOTE</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#concluding-remarks-20"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><i class="fa fa-check"></i>5.4 Unsupervised Machine Learning Part 1: K-Means Clustering &amp; PCA</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-training-module-21"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-unsupervised-machine-learning"><i class="fa fa-check"></i>Introduction to Unsupervised Machine Learning</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-example-data"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#identifying-clusters-of-chemicals-through-k-means"><i class="fa fa-check"></i>Identifying Clusters of Chemicals through <em>K</em>-Means</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#principal-component-analysis-pca-1"><i class="fa fa-check"></i>Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#incorporating-k-means-into-pca-for-predictive-modeling"><i class="fa fa-check"></i>Incorporating <em>K</em>-Means into PCA for Predictive Modeling</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#concluding-remarks-21"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><i class="fa fa-check"></i>5.5 Unsupervised Machine Learning Part 2: Additional Clustering Applications</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-training-module-22"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#k-means-clustering-1"><i class="fa fa-check"></i><em>K</em>-Means Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-example-data-1"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-1"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#clustering-plot"><i class="fa fa-check"></i>Clustering Plot</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-visualization"><i class="fa fa-check"></i>Hierarchical Clustering Visualization</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#variable-contributions"><i class="fa fa-check"></i>Variable Contributions</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#concluding-remarks-22"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 6 Applications in Toxicology &amp; Exposure Science</b></span></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html"><i class="fa fa-check"></i>6.1 Descriptive Cohort Analyses</a>
<ul>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#introduction-to-training-module-23"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#participant-exploration"><i class="fa fa-check"></i>Participant Exploration</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#chemical-detection-filtering"><i class="fa fa-check"></i>Chemical Detection Filtering</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#outlier-identification"><i class="fa fa-check"></i>Outlier Identification</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#summary-statistics-tables"><i class="fa fa-check"></i>Summary Statistics Tables</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#demographics-table"><i class="fa fa-check"></i>Demographics Table</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#concluding-remarks-23"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html"><i class="fa fa-check"></i>6.2 -Omics and System Biology: Transcriptomic Applications</a>
<ul>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-training-module-24"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-the-field-of--omics"><i class="fa fa-check"></i>Introduction to the Field of “-Omics”</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-transcriptomics"><i class="fa fa-check"></i>Introduction to Transcriptomics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#formatting-data-for-downstream-statistics"><i class="fa fa-check"></i>Formatting Data for Downstream Statistics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#transcriptomics-data-qaqc"><i class="fa fa-check"></i>Transcriptomics Data QA/QC</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#controlling-for-sources-of-sample-heterogeneity"><i class="fa fa-check"></i>Controlling for Sources of Sample Heterogeneity</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e.g.-biomass-smoke-exposure"><i class="fa fa-check"></i>Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure)</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-ma-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using MA Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-volcano-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using Volcano Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis"><i class="fa fa-check"></i>Interpretting Findings at the Systems Level through Pathway Enrichment Analysis</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#concluding-remarks-24"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><i class="fa fa-check"></i>6.3 Mixtures Analysis Methods Part 1: Overview and Example with Quantile G-Computation</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-training-module-25"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#overview-of-mixtures-analysis"><i class="fa fa-check"></i>Overview of Mixtures Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-example-data-2"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#mixtures-model-with-standard-qqcomp"><i class="fa fa-check"></i>Mixtures Model with Standard qqcomp</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#concluding-remarks-25"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#additional-resources-11"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><i class="fa fa-check"></i>6.4 Mixtures Analysis Methods Part 2: Bayesian Kernel Machine Regression</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-training-module-26"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-example-data-3"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-bkmr"><i class="fa fa-check"></i>Introduction to BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#run-bkmr"><i class="fa fa-check"></i>Run BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#concluding-remarks-26"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#additional-resources-12"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html"><i class="fa fa-check"></i>6.5 Mixtures Analysis Methods Part 3: Sufficient Similarity</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-training-module-27"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-toxicant-and-dataset"><i class="fa fa-check"></i>Introduction to Toxicant and Dataset</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#workspace-preparation-and-data-import-9"><i class="fa fa-check"></i>Workspace Preparation and Data Import</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#chemistry-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Chemistry-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#toxicity-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Toxicity-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#comparing-chemistry-vs.-toxicity-sufficient-similarity-analyses"><i class="fa fa-check"></i>Comparing Chemistry vs. Toxicity Sufficient Similarity Analyses</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#concluding-remarks-27"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i>6.6 Toxicokinetic Modeling</a>
<ul>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-training-module-28"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-toxicokinetic-modeling"><i class="fa fa-check"></i>Introduction to Toxicokinetic Modeling</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#data-and-models-used-in-toxicokinetic-modeling-tk"><i class="fa fa-check"></i>Data and Models used in Toxicokinetic Modeling (TK)</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-toxicokinetics"><i class="fa fa-check"></i>Reverse Toxicokinetics</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#capturing-population-variability-in-toxicokinetics-and-uncertainty-in-chemical-specific-parameters"><i class="fa fa-check"></i>Capturing Population Variability in Toxicokinetics, and Uncertainty in Chemical-Specific Parameters</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-tk-calculating-administered-equivalent-doses-for-toxcast-bioactive-concentrations"><i class="fa fa-check"></i>Reverse TK: Calculating Administered Equivalent Doses for ToxCast Bioactive Concentrations</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#comparing-equivalent-doses-estimated-to-elicit-toxicity-hazard-to-external-exposure-estimates-exposure-for-chemical-prioritization-by-bioactivity-exposure-ratios-bers"><i class="fa fa-check"></i>Comparing Equivalent Doses Estimated to Elicit Toxicity (Hazard) to External Exposure Estimates (Exposure), for Chemical Prioritization by Bioactivity-Exposure Ratios (BERs)</a></li>
</ul></li>
<li class="part"><span><b>Chapter 7 Environmental Health Database Mining</b></span></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html"><i class="fa fa-check"></i>7.1 Comparative Toxicogenomics Database</a>
<ul>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#introduction-to-training-module-30"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#ctd-data-in-r"><i class="fa fa-check"></i>CTD Data in R</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#identifying-genes-under-epigenetic-control"><i class="fa fa-check"></i>Identifying Genes Under Epigenetic Control</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#concluding-remarks-30"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html"><i class="fa fa-check"></i>7.2 Gene Expression Omnibus</a>
<ul>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-training-module-31"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-geo"><i class="fa fa-check"></i>Introduction to GEO</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-example-data-4"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#geo-data-in-r"><i class="fa fa-check"></i>GEO Data in R</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#visualizing-data"><i class="fa fa-check"></i>Visualizing Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#statistical-analyses"><i class="fa fa-check"></i>Statistical Analyses</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#concluding-remarks-31"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html"><i class="fa fa-check"></i>7.3 CompTox Dashboard Data through APIs</a>
<ul>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#introduction-to-training-module-32"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#training-modules-environmental-health-questions-24"><i class="fa fa-check"></i>Training Module’s Environmental Health Questions</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#script-preparations-12"><i class="fa fa-check"></i>Script Preparations</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#introduction-to-comptox-chemicals-dashboard"><i class="fa fa-check"></i>Introduction to CompTox Chemicals Dashboard</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#cctes-ctx-application-programming-interfaces-apis-for-automated-batch-search-of-the-ccd"><i class="fa fa-check"></i>CCTE’s CTX Application Programming Interfaces (APIs) for Automated Batch Search of the CCD</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#comparing-physico-chemical-properties-between-two-important-environmental-contaminant-lists"><i class="fa fa-check"></i>Comparing Physico-chemical Properties between Two Important Environmental Contaminant Lists</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#hazard-data-genotoxicity"><i class="fa fa-check"></i>Hazard Data: Genotoxicity</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#hazard-resource"><i class="fa fa-check"></i>Hazard Resource</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#concluding-remarks-32"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html"><i class="fa fa-check"></i>7.4 Database Integration: Air Quality, Mortality, and Environmental Justice Data</a>
<ul>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#introduction-to-training-module-33"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#introduction-to-exposure-and-health-databases"><i class="fa fa-check"></i>Introduction to Exposure and Health Databases</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#analyzing-relationships-between-pm2.5-and-mortality"><i class="fa fa-check"></i>Analyzing Relationships between PM<sub>2.5</sub> and Mortality</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#environmental-justice-considerations"><i class="fa fa-check"></i>Environmental Justice Considerations</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#concluding-remarks-33"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mixtures-analysis-methods-part-3-sufficient-similarity" class="section level1 hasAnchor">
<h1>6.5 Mixtures Analysis Methods Part 3: Sufficient Similarity<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#mixtures-analysis-methods-part-3-sufficient-similarity" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This training module was developed by Cynthia Rider, with contributions from Lauren E. Koval and Julia E. Rager.</p>
<p>All input files (script, data, and figures) can be downloaded from the <a href="https://github.com/UNCSRP/TAME2">UNC-SRP TAME2 GitHub website</a>.</p>
<div id="introduction-to-training-module-27" class="section level2 hasAnchor">
<h2>Introduction to Training Module<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-training-module-27" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Humans are rarely, if ever, exposed to single chemicals at a time. Instead, humans are often exposed to multiple stressors in their everyday environments in the form of mixtures. These stressors can include environmental chemicals and pharmaceuticals, and they can also include other types of stressors such as socioeconomic factors and other attributes that can place individuals at increased risk of acquiring disease. Because it is not possible to test every possible combination of exposure that an individual might experience in their lifetime, approaches that take into account variable and complex exposure conditions through mixtures modeling are needed.</p>
<p>There are different computational approaches that can be implemented to address this research topic. In this training module, we will demonstrate how to use <strong>sufficient similarity</strong> to determine which groups of exposure conditions are chemically/biologically similar enough to be regulated for safety together, based on the same set of regulatory criteria. Here, our example mixtures analysis will focus on characterizing the nutritional supplement <em>Ginkgo biloba</em>.</p>
<div id="training-modules-environmental-health-questions-19" class="section level3 hasAnchor">
<h3>Training Module’s Environmental Health Questions<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#training-modules-environmental-health-questions-19" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This training module was specifically developed to answer the following environmental health questions:</p>
<ol style="list-style-type: decimal">
<li>Based on the chemical analysis, which <em>Ginkgo biloba</em> extract looks the most different?</li>
<li>When viewing the variability between chemical profiles, how many groupings of potentially ‘sufficiently similar’ <em>Ginkgo biloba</em> samples do you see?</li>
<li>Based on the chemical analysis, which chemicals do you think are important in differentiating between the different <em>Ginkgo biloba</em> samples?</li>
<li>After removing two samples that have the most different chemical profiles (and are thus, potential outliers), do we obtain similar chemical groupings?</li>
<li>When viewing the variability between toxicity profiles, how many groupings of potentially ‘sufficiently similar’ <em>Ginkgo biloba</em> samples do you see?</li>
<li>Based on the toxicity analysis, which genes do you think are important in differentiating between the different <em>Ginkgo biloba</em> samples?</li>
<li>Were similar chemical groups identified when looking at just the chemistry vs. just the toxicity? How could this impact regulatory decisions, if we only had one of these datasets?</li>
</ol>
</div>
</div>
<div id="introduction-to-toxicant-and-dataset" class="section level2 hasAnchor">
<h2>Introduction to Toxicant and Dataset<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-toxicant-and-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><em>Ginkgo biloba</em> represents a popular type of botanical supplement currently on the market. People take <em>Ginkgo biloba</em> to improve brain function, but there is conflicting data on its efficacy. Like other botanicals, <em>Ginkgo biloba</em> is a complex mixture with 100s-1000s of constituents. Here, the variability in chemical and toxicological profiles across samples of <em>Ginkgo biloba</em> purchased from different commercial sources is evaluated. We can use data from a well-characterized sample (reference sample) to evaluate the safety of other samples that are ‘sufficiently similar’ to the reference sample. Samples that are different (i.e., do not meet the standards of sufficient similarity) from the reference sample would require additional safety data.</p>
<p>A total of 29 <em>Ginkgo biloba</em> extract samples were analyzed. These samples are abbreviated as “GbE_” followed by a unique sample identifier (GbE = <em>Ginkgo biloba</em> Extract). These data have been previously published:</p>
<ul>
<li><p>Catlin NR, Collins BJ, Auerbach SS, Ferguson SS, Harnly JM, Gennings C, Waidyanatha S, Rice GE, Smith-Roe SL, Witt KL, Rider CV. How similar is similar enough? A sufficient similarity case study with Ginkgo biloba extract. Food Chem Toxicol. 2018 Aug;118:328-339. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/29752982/">29752982</a>.</p></li>
<li><p>Collins BJ, Kerns SP, Aillon K, Mueller G, Rider CV, DeRose EF, London RE, Harnly JM, Waidyanatha S. Comparison of phytochemical composition of Ginkgo biloba extracts using a combination of non-targeted and targeted analytical approaches. Anal Bioanal Chem. 2020 Oct;412(25):6789-6809. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/32865633/">32865633</a>.</p></li>
</ul>
<div id="ginkgo-biloba-chemistry-dataset-overview" class="section level3 hasAnchor">
<h3><em>Ginkgo biloba</em> Chemistry Dataset Overview<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#ginkgo-biloba-chemistry-dataset-overview" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The chemical profiles of these sample extracts were first analyzed using targeted mass spectrometry-based approaches. The concentrations of 12 <em>Ginkgo biloba</em> marker compounds were measured in units of mean weight as a ratio [g chemical / g sample]. Note that in this dataset, non-detects have been replaced with values of zero for simplicity; though there are more advanced methods to impute values for non-detects. Script is provided to evaluate how <em>Ginkgo biloba</em> extracts group together, based on chemical profiles.</p>
</div>
<div id="ginkgo-biloba-toxicity-dataset-overview" class="section level3 hasAnchor">
<h3><em>Ginkgo biloba</em> Toxicity Dataset Overview<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#ginkgo-biloba-toxicity-dataset-overview" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The toxicological profiles of these samples were also analyzed using <em>in vitro</em> test methods. These data represent area under the curve (AUC) values indicating changes in gene expression across various concentrations of the <em>Ginkgo biloba</em> extract samples. Positive AUC values indicate a gene that was collectively increased in expression as concentration increased, and a negative AUC value indicates a gene that was collectively decreased in expression as exposure concentration increased. Script is provided to evaluate how <em>Ginkgo biloba</em> extracts group together, based on toxicity profiles.</p>
</div>
</div>
<div id="workspace-preparation-and-data-import-9" class="section level2 hasAnchor">
<h2>Workspace Preparation and Data Import<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#workspace-preparation-and-data-import-9" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="install-required-r-packages" class="section level4 hasAnchor">
<h4>Install required R packages<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#install-required-r-packages" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>If you already have these packages installed, you can skip this step, or you can run the below code which checks installation status for you</p>
<div class="sourceCode" id="cb785"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb785-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb785-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;tidyverse&quot;</span>))</span>
<span id="cb785-2"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb785-2" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;tidyverse&quot;</span>);</span>
<span id="cb785-3"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb785-3" tabindex="-1"></a></span>
<span id="cb785-4"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb785-4" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;readxl&quot;</span>))</span>
<span id="cb785-5"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb785-5" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;readxl&quot;</span>);</span>
<span id="cb785-6"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb785-6" tabindex="-1"></a></span>
<span id="cb785-7"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb785-7" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;factoextra&quot;</span>))</span>
<span id="cb785-8"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb785-8" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;factoextra&quot;</span>);</span>
<span id="cb785-9"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb785-9" tabindex="-1"></a></span>
<span id="cb785-10"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb785-10" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;pheatmap&quot;</span>))</span>
<span id="cb785-11"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb785-11" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;pheatmap&quot;</span>);</span>
<span id="cb785-12"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb785-12" tabindex="-1"></a></span>
<span id="cb785-13"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb785-13" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;gridExtra&quot;</span>))</span>
<span id="cb785-14"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb785-14" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;gridExtra&quot;</span>);</span>
<span id="cb785-15"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb785-15" tabindex="-1"></a></span>
<span id="cb785-16"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb785-16" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;ggplotify&quot;</span>))</span>
<span id="cb785-17"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb785-17" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;ggplotify&quot;</span>)</span></code></pre></div>
</div>
<div id="loading-required-packages-1" class="section level4 hasAnchor">
<h4>Loading required packages<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#loading-required-packages-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb786"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb786-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb786-1" tabindex="-1"></a><span class="fu">library</span>(readxl) <span class="co">#used to read in and work with excel files</span></span>
<span id="cb786-2"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb786-2" tabindex="-1"></a><span class="fu">library</span>(factoextra) <span class="co">#used to run and visualize multivariate analyses, here PCA</span></span>
<span id="cb786-3"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb786-3" tabindex="-1"></a><span class="fu">library</span>(pheatmap) <span class="co">#used to make heatmaps. This can be done in ggplot2 but pheatmap is easier and nicer</span></span>
<span id="cb786-4"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb786-4" tabindex="-1"></a><span class="fu">library</span>(gridExtra) <span class="co">#used to arrange and visualize multiple figures at once</span></span>
<span id="cb786-5"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb786-5" tabindex="-1"></a><span class="fu">library</span>(ggplotify) <span class="co">#used to make non ggplot figures (like a pheatmap) gg compatible</span></span>
<span id="cb786-6"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb786-6" tabindex="-1"></a><span class="fu">library</span>(tidyverse) <span class="co">#all tidyverse packages, including dplyr and ggplot2</span></span></code></pre></div>
</div>
<div id="set-your-working-directory-12" class="section level4 hasAnchor">
<h4>Set your working directory<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#set-your-working-directory-12" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb787"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb787-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb787-1" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/filepath to where your input files are&quot;</span>)</span></code></pre></div>
</div>
<div id="import-example-ginkgo-biloba-dataset" class="section level4 hasAnchor">
<h4>Import example <em>Ginkgo biloba</em> dataset<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#import-example-ginkgo-biloba-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>We need to first read in the chemistry and toxicity data from the provided excel file. Here, data were originally organized such that the actual observations start on row 2 (dataset descriptions were in the first row). So let’s implement skip=1, which skips reading in the first row.</p>
<div class="sourceCode" id="cb788"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb788-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb788-1" tabindex="-1"></a>chem <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">&quot;Module6_5_Input/Module6_5_InputData.xlsx&quot;</span> , <span class="at">sheet =</span> <span class="st">&quot;chemistry data&quot;</span>, <span class="at">skip=</span><span class="dv">1</span>) <span class="co"># loads the chemistry data tab</span></span>
<span id="cb788-2"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb788-2" tabindex="-1"></a>tox  <span class="ot">&lt;-</span> <span class="fu">read_xlsx</span>(<span class="st">&quot;Module6_5_Input/Module6_5_InputData.xlsx&quot;</span> , <span class="at">sheet =</span> <span class="st">&quot;in vitro data&quot;</span>, <span class="at">skip=</span><span class="dv">1</span>) <span class="co"># loads the toxicity data tab</span></span></code></pre></div>
</div>
<div id="view-example-dataset" class="section level3 hasAnchor">
<h3>View example dataset<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#view-example-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let’s first see how many rows and columns of data are present in both datasets:</p>
<div class="sourceCode" id="cb789"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb789-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb789-1" tabindex="-1"></a><span class="fu">dim</span>(chem)</span></code></pre></div>
<pre><code>## [1] 29 13</code></pre>
<p>The chemistry dataset contains information on 29 samples (rows); and 1 sample identifier + 12 chemicals (total of 13 columns).</p>
<div class="sourceCode" id="cb791"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb791-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb791-1" tabindex="-1"></a><span class="fu">dim</span>(tox)</span></code></pre></div>
<pre><code>## [1] 29  6</code></pre>
<p>The tox dataset contains information on 29 samples (rows); and 1 sample identifier + 5 genes (total of 6 columns).</p>
<p>Let’s also see what kind of data are organized within the datasets:</p>
<div class="sourceCode" id="cb793"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb793-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb793-1" tabindex="-1"></a><span class="fu">colnames</span>(chem)</span></code></pre></div>
<pre><code>##  [1] &quot;Sample&quot;             &quot;Bilobalide&quot;         &quot;Ginkgolide_A&quot;      
##  [4] &quot;Ginkgolide_B&quot;       &quot;Ginkgolide_C&quot;       &quot;Ginkgolide_J&quot;      
##  [7] &quot;Rutin&quot;              &quot;Quercetin&quot;          &quot;Kaempferol&quot;        
## [10] &quot;Isorhamnetin&quot;       &quot;Ginkgolic_Acid_C15&quot; &quot;Ginkgolic_Acid_C17&quot;
## [13] &quot;Ginkgotoxin&quot;</code></pre>
<div class="sourceCode" id="cb795"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb795-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb795-1" tabindex="-1"></a><span class="fu">head</span>(chem)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 13
##   Sample Bilobalide Ginkgolide_A Ginkgolide_B Ginkgolide_C Ginkgolide_J Rutin
##   &lt;chr&gt;       &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt; &lt;dbl&gt;
## 1 GbE_A        1.28         0            0            0            1.77  0   
## 2 GbE_B        0            0            0            0            0     0.05
## 3 GbE_C        0            0            0            0            0     0.01
## 4 GbE_D        1.28         2.6          1.6          2.79         1.18  1.11
## 5 GbE_E        1.5          2.13         1.46         2.6          1.21  1.21
## 6 GbE_F        0            0            0            0            0     0.04
## # ℹ 6 more variables: Quercetin &lt;dbl&gt;, Kaempferol &lt;dbl&gt;, Isorhamnetin &lt;dbl&gt;,
## #   Ginkgolic_Acid_C15 &lt;dbl&gt;, Ginkgolic_Acid_C17 &lt;dbl&gt;, Ginkgotoxin &lt;dbl&gt;</code></pre>
<div class="sourceCode" id="cb797"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb797-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb797-1" tabindex="-1"></a><span class="fu">colnames</span>(tox)</span></code></pre></div>
<pre><code>## [1] &quot;Sample&quot; &quot;ABCB11&quot; &quot;CYP1A2&quot; &quot;CYP2B6&quot; &quot;CYP3A4&quot; &quot;HMGCS2&quot;</code></pre>
<div class="sourceCode" id="cb799"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb799-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb799-1" tabindex="-1"></a><span class="fu">head</span>(tox)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 6
##   Sample  ABCB11 CYP1A2 CYP2B6 CYP3A4  HMGCS2
##   &lt;chr&gt;    &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;   &lt;dbl&gt;
## 1 GbE_A  -0.450   0.778  0.124 -1.39  -0.0208
## 2 GbE_B  -0.210   0.902  0.456 -1.22  -0.149 
## 3 GbE_C  -1.10    1.32   1.58  -1.62   0.195 
## 4 GbE_D  -0.818   1.61   2.46   0.935 -1.62  
## 5 GbE_E  -0.963   2.27   2.44   1.33  -1.54  
## 6 GbE_F  -0.0828  1.2    0.587 -1.29  -0.147</code></pre>
</div>
</div>
<div id="chemistry-based-sufficient-similarity-analysis" class="section level2 hasAnchor">
<h2>Chemistry-Based Sufficient Similarity Analysis<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#chemistry-based-sufficient-similarity-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The first method employed in this Sufficient Similarity analysis is Principal Component Analysis (PCA). PCA is a very common dimensionality reduction technique, as detailed in <strong>TAME 2.0 Module 5.4 Unsupervised Machine Learning</strong>.</p>
<p>In summary, PCA finds dimensions (eigenvectors) in the higher dimensional original data that capture as much of the variation as possible, which you can then plot. This allows you to project higher dimensional data, in this case 12 dimensions (representing 12 measured chemicals), in fewer dimensions (we’ll use 2). These dimensions, or components, capture the “essence” of the original dataset.</p>
<p>Before we can run PCA on this chemistry dataset, we first need to scale the data across samples. We do this here for the chemistry dataset, because we specifically want to evaluate and potentially highlight/emphasize chemicals that may be at relatively low abundance. These low-abundance chemicals may actually be contaminants that drive toxicological effects.</p>
<p>Let’s first re-save the original chemistry dataset to compare off of:</p>
<div class="sourceCode" id="cb801"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb801-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb801-1" tabindex="-1"></a>chem_original <span class="ot">&lt;-</span> chem</span></code></pre></div>
<p>Then, we’ll make a scaled version to carry forward in this analysis. To do this, we move the sample column the row names and then scale and center the data.</p>
<div class="sourceCode" id="cb802"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb802-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb802-1" tabindex="-1"></a>chem <span class="ot">&lt;-</span> chem <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="st">&quot;Sample&quot;</span>)</span>
<span id="cb802-2"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb802-2" tabindex="-1"></a>chem <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">scale</span>(<span class="fu">as.matrix</span>(chem)))</span></code></pre></div>
<p>Let’s now compare one of the rows of data (here, sample GbE_E) to see what scaling did:</p>
<div class="sourceCode" id="cb803"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb803-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb803-1" tabindex="-1"></a>chem_original[<span class="dv">5</span>,]</span></code></pre></div>
<pre><code>## # A tibble: 1 × 13
##   Sample Bilobalide Ginkgolide_A Ginkgolide_B Ginkgolide_C Ginkgolide_J Rutin
##   &lt;chr&gt;       &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt;        &lt;dbl&gt; &lt;dbl&gt;
## 1 GbE_E         1.5         2.13         1.46          2.6         1.21  1.21
## # ℹ 6 more variables: Quercetin &lt;dbl&gt;, Kaempferol &lt;dbl&gt;, Isorhamnetin &lt;dbl&gt;,
## #   Ginkgolic_Acid_C15 &lt;dbl&gt;, Ginkgolic_Acid_C17 &lt;dbl&gt;, Ginkgotoxin &lt;dbl&gt;</code></pre>
<div class="sourceCode" id="cb805"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb805-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb805-1" tabindex="-1"></a>chem[<span class="dv">5</span>,]</span></code></pre></div>
<pre><code>##       Bilobalide Ginkgolide_A Ginkgolide_B Ginkgolide_C Ginkgolide_J     Rutin
## GbE_E -0.4996768   0.02749315    0.2559249      1.36305    0.3842836 0.5758008
##       Quercetin Kaempferol Isorhamnetin Ginkgolic_Acid_C15 Ginkgolic_Acid_C17
## GbE_E 0.2059607 -0.2420023     -0.61695         -0.2737699         -0.2317415
##       Ginkgotoxin
## GbE_E  -0.8114437</code></pre>
<p>You can see that scaling made the concentrations distributed across each chemical center around 0.</p>
<p>Now, we can run PCA on the scaled data:</p>
<div class="sourceCode" id="cb807"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb807-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb807-1" tabindex="-1"></a>chem_pca <span class="ot">&lt;-</span> <span class="fu">princomp</span>(chem)</span></code></pre></div>
<p>Looking at the scree plot, we see the first two principal components capture most of the variance in the data (~64%):</p>
<div class="sourceCode" id="cb808"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb808-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb808-1" tabindex="-1"></a><span class="fu">fviz_eig</span>(chem_pca)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-599-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Here are the resulting PCA scores for each sample, for each principal component (shown here as components 1-12):</p>
<div class="sourceCode" id="cb809"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb809-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb809-1" tabindex="-1"></a><span class="fu">head</span>(chem_pca<span class="sc">$</span>scores)</span></code></pre></div>
<pre><code>##           Comp.1      Comp.2     Comp.3     Comp.4      Comp.5      Comp.6
## GbE_A -2.6256689 -0.53470486 0.38196672  1.5147505  1.34638144 -0.24020735
## GbE_B -3.7343187 -1.07490206 0.07766108  0.2529618 -0.01731847 -0.02446797
## GbE_C -3.8019563 -1.06232680 0.08335445  0.2588084 -0.01912736 -0.03423438
## GbE_D  0.7175782 -0.09566345 0.90082998 -0.7762090 -0.40507924 -0.78361086
## GbE_E  0.4157644 -0.14794948 1.16402759 -0.4856176 -0.15497152 -0.64206760
## GbE_F -3.7621819 -1.04784203 0.08687721  0.2503188 -0.02195005 -0.04030373
##           Comp.7     Comp.8      Comp.9     Comp.10     Comp.11      Comp.12
## GbE_A -0.8890488  0.8145597  0.15021776 -0.54318277 -0.32353295  0.049538024
## GbE_B  0.3988596 -0.3227102 -0.10344907 -0.12911495  0.11127631  0.008306532
## GbE_C  0.3817061 -0.2788579 -0.13057528 -0.02613584  0.08148776  0.011485316
## GbE_D -1.1916851 -0.4306198  0.08460588  0.26115540  0.01065657 -0.053819603
## GbE_E -1.1412900 -0.5632547  0.12309347 -0.02872126  0.24882196  0.047691048
## GbE_F  0.3948245 -0.3105324 -0.10539998 -0.11015645  0.10607314  0.012066512</code></pre>
<p>And the resulting loading factors of each chemical’s contribution towards each principal component. Results are arranged by a chemical’s contribution to PC1, the component accounting for most of the variation in the data.</p>
<div class="sourceCode" id="cb811"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb811-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb811-1" tabindex="-1"></a><span class="fu">head</span>(chem_pca<span class="sc">$</span>loadings)</span></code></pre></div>
<pre><code>##                 Comp.1      Comp.2     Comp.3      Comp.4      Comp.5
## Bilobalide   0.3260729 0.085117226 0.24541927  0.36968834  0.03575030
## Ginkgolide_A 0.3732546 0.184101347 0.08586157  0.02446723 -0.03299212
## Ginkgolide_B 0.3611463 0.201280307 0.05996588 -0.19928568 -0.05965128
## Ginkgolide_C 0.3616497 0.001365838 0.12157943 -0.24092774 -0.10470516
## Ginkgolide_J 0.2995761 0.178689209 0.09144978  0.35348565  0.47372240
## Rutin        0.1763429 0.012553984 0.59496115 -0.47712128  0.08063637
##                    Comp.6      Comp.7      Comp.8      Comp.9     Comp.10
## Bilobalide    0.004975969  0.55079870  0.35062913  0.23584371  0.25226337
## Ginkgolide_A -0.247529927  0.30284388 -0.30206946 -0.01601813  0.20620248
## Ginkgolide_B -0.068401213 -0.00687696  0.09497565 -0.83752355  0.02062813
## Ginkgolide_C -0.185456111 -0.45474675 -0.34421625  0.32067335  0.45797955
## Ginkgolide_J -0.062170758 -0.55785820  0.36066655  0.03820900 -0.14858754
## Rutin         0.517141873  0.05060803  0.03111005  0.15083257 -0.27469825
##                  Comp.11     Comp.12
## Bilobalide    0.29295985  0.23837797
## Ginkgolide_A -0.60099943 -0.41172878
## Ginkgolide_B  0.24851189  0.06938891
## Ginkgolide_C  0.28752202  0.17463609
## Ginkgolide_J -0.22581449 -0.03024110
## Rutin        -0.09884752 -0.04092322</code></pre>
<p>We can save the chemical-specific loadings into a separate matrix and view them from highest to lowest values for PC1.</p>
<div class="sourceCode" id="cb813"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb813-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb813-1" tabindex="-1"></a>loadings <span class="ot">&lt;-</span> <span class="fu">as.data.frame.matrix</span>(chem_pca<span class="sc">$</span>loadings)  </span>
<span id="cb813-2"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb813-2" tabindex="-1"></a>loadings <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(Comp<span class="fl">.1</span>))</span></code></pre></div>
<pre><code>##                        Comp.1       Comp.2      Comp.3      Comp.4       Comp.5
## Quercetin           0.3801459 -0.001945021 -0.19193647  0.04697879  0.007656212
## Ginkgolide_A        0.3732546  0.184101347  0.08586157  0.02446723 -0.032992122
## Ginkgolide_C        0.3616497  0.001365838  0.12157943 -0.24092774 -0.104705164
## Ginkgolide_B        0.3611463  0.201280307  0.05996588 -0.19928568 -0.059651275
## Bilobalide          0.3260729  0.085117226  0.24541927  0.36968834  0.035750299
## Kaempferol          0.3001354 -0.085004317 -0.29667523 -0.15611039 -0.655825688
## Ginkgolide_J        0.2995761  0.178689209  0.09144978  0.35348565  0.473722400
## Isorhamnetin        0.2740348 -0.075119327 -0.29665890  0.39008587 -0.060880190
## Rutin               0.1763429  0.012553984  0.59496115 -0.47712128  0.080636368
## Ginkgotoxin         0.1675373 -0.034318656 -0.56232119 -0.47935782  0.551341021
## Ginkgolic_Acid_C15 -0.1201265  0.667543042 -0.11607308 -0.05802100 -0.085384063
## Ginkgolic_Acid_C17 -0.1418140  0.653142232 -0.09559022 -0.02189315 -0.084006824
##                          Comp.6      Comp.7      Comp.8      Comp.9     Comp.10
## Quercetin          -0.249799236  0.13263831 -0.30855302  0.18612332 -0.72929122
## Ginkgolide_A       -0.247529927  0.30284388 -0.30206946 -0.01601813  0.20620248
## Ginkgolide_C       -0.185456111 -0.45474675 -0.34421625  0.32067335  0.45797955
## Ginkgolide_B       -0.068401213 -0.00687696  0.09497565 -0.83752355  0.02062813
## Bilobalide          0.004975969  0.55079870  0.35062913  0.23584371  0.25226337
## Kaempferol          0.050018874 -0.12464461  0.51491286  0.16055155 -0.07828551
## Ginkgolide_J       -0.062170758 -0.55785820  0.36066655  0.03820900 -0.14858754
## Isorhamnetin        0.730543567 -0.06658953 -0.34052044 -0.10456587  0.10158173
## Rutin               0.517141873  0.05060803  0.03111005  0.15083257 -0.27469825
## Ginkgotoxin         0.092918281  0.20744490  0.16087302  0.11145659  0.17463719
## Ginkgolic_Acid_C15  0.057775937  0.03440329 -0.13417826  0.06575084 -0.07780733
## Ginkgolic_Acid_C17  0.133278823 -0.05017155  0.08415192  0.16593739  0.03564092
##                         Comp.11      Comp.12
## Quercetin           0.273123642  0.008854815
## Ginkgolide_A       -0.600999427 -0.411728782
## Ginkgolide_C        0.287522018  0.174636086
## Ginkgolide_B        0.248511890  0.069388910
## Bilobalide          0.292959851  0.238377968
## Kaempferol         -0.211380567 -0.020939233
## Ginkgolide_J       -0.225814490 -0.030241100
## Isorhamnetin        0.002690835 -0.006305513
## Rutin              -0.098847524 -0.040923217
## Ginkgotoxin        -0.005807642  0.016904160
## Ginkgolic_Acid_C15 -0.285797465  0.633437667
## Ginkgolic_Acid_C17  0.383124914 -0.577639931</code></pre>
<p>These resulting loading factors allow us to identify which constituents (of the 12 total) contribute to the principal components explaining data variabilities. For instance, we can see here that <strong>Quercetin</strong> is listed at the top, with the largest loading value for principal component 1. Thus, Quercetin represents the constituents that contributes to the overall variability in the dataset to the greatest extent. The next three chemicals are all <strong>Ginkgolide</strong> constituents, followed by <strong>Bilobalide</strong> and <strong>Kaempferol</strong>, and so forth.</p>
<p>If we look at principal component 2 (PC2), we can now see a different set of chemicals contributing to the variability captured in this component:</p>
<div class="sourceCode" id="cb815"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb815-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb815-1" tabindex="-1"></a>loadings <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(<span class="fu">desc</span>(Comp<span class="fl">.2</span>))</span></code></pre></div>
<pre><code>##                        Comp.1       Comp.2      Comp.3      Comp.4       Comp.5
## Ginkgolic_Acid_C15 -0.1201265  0.667543042 -0.11607308 -0.05802100 -0.085384063
## Ginkgolic_Acid_C17 -0.1418140  0.653142232 -0.09559022 -0.02189315 -0.084006824
## Ginkgolide_B        0.3611463  0.201280307  0.05996588 -0.19928568 -0.059651275
## Ginkgolide_A        0.3732546  0.184101347  0.08586157  0.02446723 -0.032992122
## Ginkgolide_J        0.2995761  0.178689209  0.09144978  0.35348565  0.473722400
## Bilobalide          0.3260729  0.085117226  0.24541927  0.36968834  0.035750299
## Rutin               0.1763429  0.012553984  0.59496115 -0.47712128  0.080636368
## Ginkgolide_C        0.3616497  0.001365838  0.12157943 -0.24092774 -0.104705164
## Quercetin           0.3801459 -0.001945021 -0.19193647  0.04697879  0.007656212
## Ginkgotoxin         0.1675373 -0.034318656 -0.56232119 -0.47935782  0.551341021
## Isorhamnetin        0.2740348 -0.075119327 -0.29665890  0.39008587 -0.060880190
## Kaempferol          0.3001354 -0.085004317 -0.29667523 -0.15611039 -0.655825688
##                          Comp.6      Comp.7      Comp.8      Comp.9     Comp.10
## Ginkgolic_Acid_C15  0.057775937  0.03440329 -0.13417826  0.06575084 -0.07780733
## Ginkgolic_Acid_C17  0.133278823 -0.05017155  0.08415192  0.16593739  0.03564092
## Ginkgolide_B       -0.068401213 -0.00687696  0.09497565 -0.83752355  0.02062813
## Ginkgolide_A       -0.247529927  0.30284388 -0.30206946 -0.01601813  0.20620248
## Ginkgolide_J       -0.062170758 -0.55785820  0.36066655  0.03820900 -0.14858754
## Bilobalide          0.004975969  0.55079870  0.35062913  0.23584371  0.25226337
## Rutin               0.517141873  0.05060803  0.03111005  0.15083257 -0.27469825
## Ginkgolide_C       -0.185456111 -0.45474675 -0.34421625  0.32067335  0.45797955
## Quercetin          -0.249799236  0.13263831 -0.30855302  0.18612332 -0.72929122
## Ginkgotoxin         0.092918281  0.20744490  0.16087302  0.11145659  0.17463719
## Isorhamnetin        0.730543567 -0.06658953 -0.34052044 -0.10456587  0.10158173
## Kaempferol          0.050018874 -0.12464461  0.51491286  0.16055155 -0.07828551
##                         Comp.11      Comp.12
## Ginkgolic_Acid_C15 -0.285797465  0.633437667
## Ginkgolic_Acid_C17  0.383124914 -0.577639931
## Ginkgolide_B        0.248511890  0.069388910
## Ginkgolide_A       -0.600999427 -0.411728782
## Ginkgolide_J       -0.225814490 -0.030241100
## Bilobalide          0.292959851  0.238377968
## Rutin              -0.098847524 -0.040923217
## Ginkgolide_C        0.287522018  0.174636086
## Quercetin           0.273123642  0.008854815
## Ginkgotoxin        -0.005807642  0.016904160
## Isorhamnetin        0.002690835 -0.006305513
## Kaempferol         -0.211380567 -0.020939233</code></pre>
<p>Here, <strong>Ginkgolic Acids</strong> are listed first.</p>
<p>We can also visualize sample groupings based on these principal components 1 &amp; 2:</p>
<div class="sourceCode" id="cb817"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb817-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-1" tabindex="-1"></a><span class="co"># First pull the percent variation captured by each component</span></span>
<span id="cb817-2"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-2" tabindex="-1"></a>pca_percent <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span>chem_pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span><span class="fu">sum</span>(chem_pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>),<span class="dv">1</span>)</span>
<span id="cb817-3"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-3" tabindex="-1"></a></span>
<span id="cb817-4"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-4" tabindex="-1"></a><span class="co"># Then make a dataframe for the PCA plot generation script using first three components</span></span>
<span id="cb817-5"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-5" tabindex="-1"></a>pca_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC1 =</span> chem_pca<span class="sc">$</span>scores[,<span class="dv">1</span>], <span class="at">PC2 =</span> chem_pca<span class="sc">$</span>scores[,<span class="dv">2</span>])</span>
<span id="cb817-6"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-6" tabindex="-1"></a></span>
<span id="cb817-7"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-7" tabindex="-1"></a><span class="co"># Plot this dataframe</span></span>
<span id="cb817-8"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-8" tabindex="-1"></a>chem_pca_plt <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_df, <span class="fu">aes</span>(PC1,PC2))<span class="sc">+</span></span>
<span id="cb817-9"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-9" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">size=</span><span class="fl">0.3</span>)<span class="sc">+</span></span>
<span id="cb817-10"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-10" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">size=</span><span class="fl">0.3</span>)<span class="sc">+</span></span>
<span id="cb817-11"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-11" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">3</span>, <span class="at">color=</span><span class="st">&quot;deepskyblue3&quot;</span>) <span class="sc">+</span></span>
<span id="cb817-12"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-12" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label=</span><span class="fu">rownames</span>(pca_df)), <span class="at">fontface=</span><span class="st">&quot;bold&quot;</span>, <span class="at">position=</span><span class="fu">position_jitter</span>(<span class="at">width=</span><span class="fl">0.4</span>,<span class="at">height=</span><span class="fl">0.4</span>))<span class="sc">+</span></span>
<span id="cb817-13"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-13" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="fu">paste0</span>(<span class="st">&quot;PC1 (&quot;</span>,pca_percent[<span class="dv">1</span>],<span class="st">&quot;%)&quot;</span>), <span class="at">y=</span><span class="fu">paste0</span>(<span class="st">&quot;PC2 (&quot;</span>,pca_percent[<span class="dv">2</span>],<span class="st">&quot;%)&quot;</span>))<span class="sc">+</span></span>
<span id="cb817-14"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-14" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;GbE Sample PCA by Chemistry Profiles&quot;</span>)</span>
<span id="cb817-15"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-15" tabindex="-1"></a></span>
<span id="cb817-16"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-16" tabindex="-1"></a></span>
<span id="cb817-17"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-17" tabindex="-1"></a><span class="co"># Changing the colors of the titles and axis text</span></span>
<span id="cb817-18"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-18" tabindex="-1"></a>chem_pca_plt <span class="ot">&lt;-</span> chem_pca_plt <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.title=</span><span class="fu">element_text</span>(<span class="at">color=</span><span class="st">&quot;deepskyblue3&quot;</span>, <span class="at">face=</span><span class="st">&quot;bold&quot;</span>),</span>
<span id="cb817-19"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-19" tabindex="-1"></a>                    <span class="at">axis.title.x=</span><span class="fu">element_text</span>(<span class="at">color=</span><span class="st">&quot;deepskyblue3&quot;</span>, <span class="at">face=</span><span class="st">&quot;bold&quot;</span>),</span>
<span id="cb817-20"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-20" tabindex="-1"></a>                    <span class="at">axis.title.y=</span><span class="fu">element_text</span>(<span class="at">color=</span><span class="st">&quot;deepskyblue3&quot;</span>, <span class="at">face=</span><span class="st">&quot;bold&quot;</span>))</span>
<span id="cb817-21"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-21" tabindex="-1"></a></span>
<span id="cb817-22"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-22" tabindex="-1"></a><span class="co"># Viewing this resulting plot</span></span>
<span id="cb817-23"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb817-23" tabindex="-1"></a>chem_pca_plt</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-604-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>This plot tells us a lot about sample groupings based on chemical profiles!</p>
<div id="answer-to-environmental-health-question-1-16" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 1<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#answer-to-environmental-health-question-1-16" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><i>With this, we can answer <strong>Environmental Health Question 1</strong>:</i> Based on the chemical analysis, which <em>Ginkgo biloba</em> extract looks the most different?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> GbE_G</p>
</div>
</div>
<div id="answer-to-environmental-health-question-2-17" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 2<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#answer-to-environmental-health-question-2-17" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><i> We can also answer <strong>Environmental Health Question 2</strong>:</i> When viewing the variability between chemical profiles, how many groupings of potentially ‘sufficiently similar’ <em>Ginkgo biloba</em> samples do you see?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> Approximately 4 (though could argue +1/-1): bottom left group; bottom right group; and two completely separate samples of GbE_G and GbE_N</p>
</div>
<p>As an alternative way of viewing the chemical profile data, we can make a heat map of the scaled chemistry data. We concurrently run hierarchical clustering that shows us how closely samples are related to each other, based on different algorithms than data reduction-based PCA. Samples that fall on nearby branches are more similar. Samples that don’t share branches with many/any others are often considered outliers.</p>
<p>By default, pheatmap uses a Euclidean distance to cluster the observations, which is a very common clustering algorithm.
For more details, see the following description of <a href="https://en.wikipedia.org/wiki/Euclidean_distance">Euclidean distance</a>.</p>
<div class="sourceCode" id="cb818"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb818-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb818-1" tabindex="-1"></a>chem_hm <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(chem, <span class="at">main=</span><span class="st">&quot;GbE Sample Heatmap by Chemistry Profiles&quot;</span>,</span>
<span id="cb818-2"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb818-2" tabindex="-1"></a>                    <span class="at">cluster_rows=</span><span class="cn">TRUE</span>, <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb818-3"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb818-3" tabindex="-1"></a>                    <span class="at">angle_col =</span> <span class="dv">45</span>, <span class="at">fontsize_col =</span> <span class="dv">7</span>, <span class="at">treeheight_row =</span> <span class="dv">60</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-605-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>This plot tells us a lot about the individual chemicals that differentiate the sample groupings.</p>
</div>
<div id="answer-to-environmental-health-question-3-11" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 3<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#answer-to-environmental-health-question-3-11" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><i>With this, we can answer <strong>Environmental Health Question 3</strong>:</i> Based on the chemical analysis, which chemicals do you think are important in differentiating between the different <em>Ginkgo biloba</em> samples?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> All of the chemicals technically contribute to these sample patterns, but here are some that stand out: (i) Ginkgolic_Acid_C15 and Ginkgolic_Acid_C17 appear to drive the clustering of one particular GbE sample, GbE_G, as well as potentially GbE_N; (ii) Isorhamnetin influences the clustering of GbE_T; (iii) Bilobalide, Ginkgolides A &amp; B, and Quercetin are also important because they show a general cluster of abundance at decreased levels at the bottom and increased levels at the top.</p>
</div>
<p>Let’s now revisit the PCA plot:</p>
<div class="sourceCode" id="cb819"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb819-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb819-1" tabindex="-1"></a>chem_pca_plt</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-606-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>GbE_G and GbE_N look so different from the rest of the samples that they could be outliers and potentially influencing overall data trends. Let’s make sure that, if we remove these two samples, our sample groupings still look the same.</p>
<p>First, we remove those two samples from the dataframe:</p>
<div class="sourceCode" id="cb820"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb820-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb820-1" tabindex="-1"></a>chem_filt <span class="ot">&lt;-</span> chem <span class="sc">%&gt;%</span></span>
<span id="cb820-2"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb820-2" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;Sample&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb820-3"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb820-3" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>Sample <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;GbE_G&quot;</span>,<span class="st">&quot;GbE_N&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb820-4"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb820-4" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&quot;Sample&quot;</span>)</span></code></pre></div>
<p>Then, we can re-run PCA and generate a heatmap of the chemical data with these outlier samples removed:</p>
<div class="sourceCode" id="cb821"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb821-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-1" tabindex="-1"></a>chem_filt_pca <span class="ot">&lt;-</span> <span class="fu">princomp</span>(chem_filt)</span>
<span id="cb821-2"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-2" tabindex="-1"></a></span>
<span id="cb821-3"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-3" tabindex="-1"></a><span class="co"># Get the percent variation captured by each component</span></span>
<span id="cb821-4"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-4" tabindex="-1"></a>pca_percent_filt <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span>chem_filt_pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span><span class="fu">sum</span>(chem_filt_pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>),<span class="dv">1</span>)</span>
<span id="cb821-5"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-5" tabindex="-1"></a></span>
<span id="cb821-6"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-6" tabindex="-1"></a><span class="co"># Make dataframe for PCA plot generation using first three components</span></span>
<span id="cb821-7"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-7" tabindex="-1"></a>pca_df_filt <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC1 =</span> chem_filt_pca<span class="sc">$</span>scores[,<span class="dv">1</span>], <span class="at">PC2 =</span> chem_filt_pca<span class="sc">$</span>scores[,<span class="dv">2</span>])</span>
<span id="cb821-8"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-8" tabindex="-1"></a></span>
<span id="cb821-9"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-9" tabindex="-1"></a><span class="co"># Plot this dataframe</span></span>
<span id="cb821-10"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-10" tabindex="-1"></a>chem_filt_pca_plt <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_df_filt, <span class="fu">aes</span>(PC1,PC2))<span class="sc">+</span></span>
<span id="cb821-11"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-11" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">size=</span><span class="fl">0.3</span>)<span class="sc">+</span></span>
<span id="cb821-12"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-12" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">size=</span><span class="fl">0.3</span>)<span class="sc">+</span></span>
<span id="cb821-13"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-13" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">3</span>, <span class="at">color=</span><span class="st">&quot;aquamarine2&quot;</span>) <span class="sc">+</span></span>
<span id="cb821-14"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-14" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label=</span><span class="fu">rownames</span>(pca_df_filt)), <span class="at">fontface=</span><span class="st">&quot;bold&quot;</span>, <span class="at">position=</span><span class="fu">position_jitter</span>(<span class="at">width=</span><span class="fl">0.5</span>,<span class="at">height=</span><span class="fl">0.5</span>))<span class="sc">+</span></span>
<span id="cb821-15"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-15" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="fu">paste0</span>(<span class="st">&quot;PC1 (&quot;</span>,pca_percent[<span class="dv">1</span>],<span class="st">&quot;%)&quot;</span>), <span class="at">y=</span><span class="fu">paste0</span>(<span class="st">&quot;PC2 (&quot;</span>,pca_percent[<span class="dv">2</span>],<span class="st">&quot;%)&quot;</span>))<span class="sc">+</span></span>
<span id="cb821-16"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-16" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;GbE Sample PCA by Chemistry Profiles excluding Potential Outliers&quot;</span>)</span>
<span id="cb821-17"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-17" tabindex="-1"></a></span>
<span id="cb821-18"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-18" tabindex="-1"></a><span class="co"># Changing the colors of the titles and axis text</span></span>
<span id="cb821-19"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-19" tabindex="-1"></a>chem_filt_pca_plt <span class="ot">&lt;-</span> chem_filt_pca_plt <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.title=</span><span class="fu">element_text</span>(<span class="at">color=</span><span class="st">&quot;aquamarine2&quot;</span>, <span class="at">face=</span><span class="st">&quot;bold&quot;</span>),</span>
<span id="cb821-20"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-20" tabindex="-1"></a>                    <span class="at">axis.title.x=</span><span class="fu">element_text</span>(<span class="at">color=</span><span class="st">&quot;aquamarine2&quot;</span>, <span class="at">face=</span><span class="st">&quot;bold&quot;</span>),</span>
<span id="cb821-21"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-21" tabindex="-1"></a>                    <span class="at">axis.title.y=</span><span class="fu">element_text</span>(<span class="at">color=</span><span class="st">&quot;aquamarine2&quot;</span>, <span class="at">face=</span><span class="st">&quot;bold&quot;</span>))</span>
<span id="cb821-22"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-22" tabindex="-1"></a></span>
<span id="cb821-23"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-23" tabindex="-1"></a><span class="co"># Viewing this resulting plot</span></span>
<span id="cb821-24"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb821-24" tabindex="-1"></a>chem_filt_pca_plt</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-608-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>To view the PCA plots of all samples vs filtered samples:</p>
<div class="sourceCode" id="cb822"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb822-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb822-1" tabindex="-1"></a><span class="fu">grid.arrange</span>(chem_pca_plt, chem_filt_pca_plt)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-609-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="answer-to-environmental-health-question-4-8" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 4<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#answer-to-environmental-health-question-4-8" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><i>With this, we can answer <strong>Environmental Health Question 4</strong>:</i> After removing two samples that have the most different chemical profiles (and are thus, potential outliers), do we obtain similar chemical groupings?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> Yes! Removal of the potential outliers basically spreads the rest of the remaining data points out, since there is less variance in the overall dataset, and thus, more room to show variance amongst the remaining samples. The general locations of the samples on the PCA plot, however, remain consistent. We now feel confident that our similarity analysis is producing consistent grouping results.</p>
</div>
</div>
</div>
<div id="toxicity-based-sufficient-similarity-analysis" class="section level2 hasAnchor">
<h2>Toxicity-Based Sufficient Similarity Analysis<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#toxicity-based-sufficient-similarity-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Now, we will perform sufficient similarity analysis using the toxicity data.Unlike the chemistry dataset, we can use the toxicity dataset as is without scaling because we want to focus on genes that are showing a large response. Similarly, we want to de-emphasize genes that are showing a strong response to the exposure condition. If we scale these data, we will reduce this needed variability.</p>
<p>Here, we first move the sample column to row names:</p>
<div class="sourceCode" id="cb823"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb823-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb823-1" tabindex="-1"></a>tox <span class="ot">&lt;-</span> tox <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="st">&quot;Sample&quot;</span>)</span></code></pre></div>
<p>Then, we can run PCA on this tox dataframe:</p>
<div class="sourceCode" id="cb824"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb824-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb824-1" tabindex="-1"></a>tox_pca <span class="ot">&lt;-</span> <span class="fu">princomp</span>(tox)</span></code></pre></div>
<p>Looking at the scree plot, we see the first two principal components capture most of the variation (~93%):</p>
<div class="sourceCode" id="cb825"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb825-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb825-1" tabindex="-1"></a><span class="fu">fviz_eig</span>(tox_pca)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-612-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>We can then create a plot of the samples by principal components:</p>
<div class="sourceCode" id="cb826"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb826-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-1" tabindex="-1"></a><span class="co"># Get the percent variation captured by each component</span></span>
<span id="cb826-2"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-2" tabindex="-1"></a>pca_percent <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span>tox_pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span><span class="fu">sum</span>(tox_pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>),<span class="dv">1</span>)</span>
<span id="cb826-3"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-3" tabindex="-1"></a></span>
<span id="cb826-4"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-4" tabindex="-1"></a><span class="co"># Make dataframe for PCA plot generation using first three components</span></span>
<span id="cb826-5"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-5" tabindex="-1"></a>tox_pca_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC1 =</span> tox_pca<span class="sc">$</span>scores[,<span class="dv">1</span>], <span class="at">PC2 =</span> tox_pca<span class="sc">$</span>scores[,<span class="dv">2</span>])</span>
<span id="cb826-6"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-6" tabindex="-1"></a></span>
<span id="cb826-7"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-7" tabindex="-1"></a><span class="co"># Plot the first two components</span></span>
<span id="cb826-8"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-8" tabindex="-1"></a>tox_pca_plt <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(tox_pca_df, <span class="fu">aes</span>(PC1,PC2))<span class="sc">+</span></span>
<span id="cb826-9"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-9" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">size=</span><span class="fl">0.3</span>)<span class="sc">+</span></span>
<span id="cb826-10"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-10" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">size=</span><span class="fl">0.3</span>)<span class="sc">+</span></span>
<span id="cb826-11"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-11" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">3</span>, <span class="at">color=</span><span class="st">&quot;deeppink3&quot;</span>) <span class="sc">+</span></span>
<span id="cb826-12"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-12" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label=</span><span class="fu">rownames</span>(pca_df)), <span class="at">fontface=</span><span class="st">&quot;bold&quot;</span>, <span class="at">position=</span><span class="fu">position_jitter</span>(<span class="at">width=</span><span class="fl">0.25</span>,<span class="at">height=</span><span class="fl">0.25</span>))<span class="sc">+</span></span>
<span id="cb826-13"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-13" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="fu">paste0</span>(<span class="st">&quot;PC1 (&quot;</span>,pca_percent[<span class="dv">1</span>],<span class="st">&quot;%)&quot;</span>), <span class="at">y=</span><span class="fu">paste0</span>(<span class="st">&quot;PC2 (&quot;</span>,pca_percent[<span class="dv">2</span>],<span class="st">&quot;%)&quot;</span>))<span class="sc">+</span></span>
<span id="cb826-14"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-14" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;GbE Sample PCA by Toxicity Profiles&quot;</span>)</span>
<span id="cb826-15"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-15" tabindex="-1"></a></span>
<span id="cb826-16"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-16" tabindex="-1"></a><span class="co"># Changing the colors of the titles and axis text</span></span>
<span id="cb826-17"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-17" tabindex="-1"></a>tox_pca_plt <span class="ot">&lt;-</span> tox_pca_plt <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">plot.title=</span><span class="fu">element_text</span>(<span class="at">color=</span><span class="st">&quot;deeppink3&quot;</span>, <span class="at">face=</span><span class="st">&quot;bold&quot;</span>),</span>
<span id="cb826-18"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-18" tabindex="-1"></a>                    <span class="at">axis.title.x=</span><span class="fu">element_text</span>(<span class="at">color=</span><span class="st">&quot;deeppink3&quot;</span>, <span class="at">face=</span><span class="st">&quot;bold&quot;</span>),</span>
<span id="cb826-19"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-19" tabindex="-1"></a>                    <span class="at">axis.title.y=</span><span class="fu">element_text</span>(<span class="at">color=</span><span class="st">&quot;deeppink3&quot;</span>, <span class="at">face=</span><span class="st">&quot;bold&quot;</span>))</span>
<span id="cb826-20"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-20" tabindex="-1"></a></span>
<span id="cb826-21"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb826-21" tabindex="-1"></a>tox_pca_plt</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-613-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>This plot tells us a lot about sample groupings based on toxicity profiles!</p>
<div id="answer-to-environmental-health-question-5-6" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 5<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#answer-to-environmental-health-question-5-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><i>With this, we can answer <strong>Environmental Health Question 5</strong>:</i> When viewing the variability between toxicity profiles, how many groupings of potentially ‘sufficiently similar’ <em>Ginkgo biloba</em> samples do you see?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> Approximately 3 (though could argue +1/-1): top left group; top right group; GbE_M and GbE_W.</p>
</div>
<p>Similar to the chemistry data, as an alternative way of viewing the toxicity profile data, we can make a heatmap of the toxicity data:</p>
<div class="sourceCode" id="cb827"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb827-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb827-1" tabindex="-1"></a>tox_hm <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(tox, <span class="at">main=</span><span class="st">&quot;GbE Sample Heatmap by Toxicity Profiles&quot;</span>,</span>
<span id="cb827-2"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb827-2" tabindex="-1"></a>               <span class="at">cluster_rows=</span><span class="cn">TRUE</span>, <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb827-3"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb827-3" tabindex="-1"></a>               <span class="at">angle_col =</span> <span class="dv">45</span>, <span class="at">fontsize_col =</span> <span class="dv">7</span>, <span class="at">treeheight_row =</span> <span class="dv">60</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-614-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>This plot tells us a lot about the individual genes that differentiate the sample groupings!</p>
</div>
<div id="answer-to-environmental-health-question-6-3" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 6<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#answer-to-environmental-health-question-6-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><i>With this, we can answer <strong>Environmental Health Question 6</strong>:</i> Based on the toxicity analysis, which genes do you think are important in differentiating between the different <em>Ginkgo biloba</em> samples?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> It looks like the CYP enzyme genes, particularly CYP2B6, are highly up-regulated in response to several of these sample exposures, and thus dictate a lot of these groupings.</p>
</div>
</div>
</div>
<div id="comparing-chemistry-vs.-toxicity-sufficient-similarity-analyses" class="section level2 hasAnchor">
<h2>Comparing Chemistry vs. Toxicity Sufficient Similarity Analyses<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#comparing-chemistry-vs.-toxicity-sufficient-similarity-analyses" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Let’s view the PCA plots for both datasets together, side-by-side:</p>
<div class="sourceCode" id="cb828"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb828-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb828-1" tabindex="-1"></a>pca_compare <span class="ot">&lt;-</span> <span class="fu">grid.arrange</span>(chem_pca_plt,tox_pca_plt, <span class="at">nrow=</span><span class="dv">1</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-615-1.png" width="1056" style="display: block; margin: auto;" /></p>
<p>Let’s also view the PCA plots for both datasets together, top-to-bottom, to visualize the trends along both axes better between these two views:</p>
<div class="sourceCode" id="cb829"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb829-1"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#cb829-1" tabindex="-1"></a>pca_compare <span class="ot">&lt;-</span> <span class="fu">grid.arrange</span>(chem_pca_plt,tox_pca_plt)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-616-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Here is an edited version of the above figures, highlighting with colored circles some chemical groups of interest identified through chemistry vs toxicity-based sufficient similarity analyses:</p>
<p><img src="Module6_5_Input/Module6_5_Image1.png" width="1119" style="display: block; margin: auto;" /></p>
<div id="answer-to-environmental-health-question-7-1" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 7<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#answer-to-environmental-health-question-7-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><i>With this, we can answer <strong>Environmental Health Question 7</strong>:</i> Were similar chemical groups identified when looking at just the chemistry vs. just the toxicity? How could this impact regulatory action, if we only had one of these datasets?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> There are some similarities between groupings, though there are also notable differences. For example, samples GbE_A, GbE_B, GbE_C, GbE_F, and GbE_H group together from the chemistry and toxicity similarity analyses. Though samples GbE_G, GbE_W, GbE_N, and others clearly demonstrate differences in grouping assignments. These differences could impact the accuracy of how regulatory decisions are made, where if regulation was dictated solely on the chemistry (without toxicity data) and/or vice versa, we may miss important information that could aid in accurate health risk evaluations.</p>
</div>
</div>
<div id="additional-methods" class="section level3 hasAnchor">
<h3>Additional Methods<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#additional-methods" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Although we focused on sufficient similarity for this module, a number of other approaches exist to evaluate mixutres. For example, <strong>relative potency factors</strong> is another component-based approach that can be used to evalaute mixtures. Component-based approaches use data from individual chemicals (components of the mixture) and additivity models to estimate the effects of the mixture. For other methods, also see <strong>TAME 2.0 Module 6.3 Mixtures I: Overview and Quantile G-Computation Application</strong> and <strong>TAME 2.0 Module 6.4 Mixtures II: BKMR Application</strong>.</p>
</div>
</div>
<div id="concluding-remarks-27" class="section level2 hasAnchor">
<h2>Concluding Remarks<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#concluding-remarks-27" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this module, we evaluated the similarity between variable lots of <em>Ginkgo biloba</em> and identified sample groupings that could be used for chemical risk assessment purposes. Together, this example highlights the utility of sufficient similarity analyses to address environmental health research questions.</p>
<div id="additional-resources-13" class="section level3 hasAnchor">
<h3>Additional Resources<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#additional-resources-13" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Some helpful resources that provide further background on the topic of mixtures toxicology and mixtures modeling include the following:</p>
<ul>
<li><p>Carlin DJ, Rider CV, Woychik R, Birnbaum LS. Unraveling the health effects of environmental mixtures: an NIEHS priority. Environ Health Perspect. 2013 Jan;121(1):A6-8. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/23409283/">23409283</a>.</p></li>
<li><p>Drakvik E, Altenburger R, Aoki Y, Backhaus T, Bahadori T, Barouki R, Brack W, Cronin MTD, Demeneix B, Hougaard Bennekou S, van Klaveren J, Kneuer C, Kolossa-Gehring M, Lebret E, Posthuma L, Reiber L, Rider C, Rüegg J, Testa G, van der Burg B, van der Voet H, Warhurst AM, van de Water B, Yamazaki K, Öberg M, Bergman Å. Statement on advancing the assessment of chemical mixtures and their risks for human health and the environment. Environ Int. 2020 Jan;134:105267. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/31704565/">31704565</a>.</p></li>
<li><p>Rider CV, McHale CM, Webster TF, Lowe L, Goodson WH 3rd, La Merrill MA, Rice G, Zeise L, Zhang L, Smith MT. Using the Key Characteristics of Carcinogens to Develop Research on Chemical Mixtures and Cancer. Environ Health Perspect. 2021 Mar;129(3):35003. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/33784186/">33784186</a>.</p></li>
<li><p>Taylor KW, Joubert BR, Braun JM, Dilworth C, Gennings C, Hauser R, Heindel JJ, Rider CV, Webster TF, Carlin DJ. Statistical Approaches for Assessing Health Effects of Environmental Chemical Mixtures in Epidemiology: Lessons from an Innovative Workshop. Environ Health Perspect. 2016 Dec 1;124(12):A227-A229. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/27905274/">27905274</a>.</p></li>
</ul>
<p>For more information and additional examples in environmental health research, see the following relevant publications implementing sufficient similarity methods to address complex mixtures:</p>
<ul>
<li><p>Catlin NR, Collins BJ, Auerbach SS, Ferguson SS, Harnly JM, Gennings C, Waidyanatha S, Rice GE, Smith-Roe SL, Witt KL, Rider CV. How similar is similar enough? A sufficient similarity case study with Ginkgo biloba extract. Food Chem Toxicol. 2018 Aug;118:328-339. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/29752982/">29752982</a>.</p></li>
<li><p>Collins BJ, Kerns SP, Aillon K, Mueller G, Rider CV, DeRose EF, London RE, Harnly JM, Waidyanatha S. Comparison of phytochemical composition of Ginkgo biloba extracts using a combination of non-targeted and targeted analytical approaches. Anal Bioanal Chem. 2020 Oct;412(25):6789-6809. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/32865633/">32865633</a>.</p></li>
<li><p>Ryan KR, Huang MC, Ferguson SS, Waidyanatha S, Ramaiahgari S, Rice JR, Dunlap PE, Auerbach SS, Mutlu E, Cristy T, Peirfelice J, DeVito MJ, Smith-Roe SL, Rider CV. Evaluating Sufficient Similarity of Botanical Dietary Supplements: Combining Chemical and In Vitro Biological Data. Toxicol Sci. 2019 Dec 1;172(2):316-329. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/31504990/">31504990</a>.</p></li>
<li><p>Rice GE, Teuschler LK, Bull RJ, Simmons JE, Feder PI. Evaluating the similarity of complex drinking-water disinfection by-product mixtures: overview of the issues. J Toxicol Environ Health A. 2009;72(7):429-36. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/19267305/">19267305</a>.</p></li>
</ul>
<p><br></p>
<p><label class="tykfont">
Test Your Knowledge
</label></p>
<div class="tyk">
<p>We recently published a study evaluating similarities across wildfire chemistry profiles using a more advanced analysis approach than described in this module (PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/36399130/">36399130</a>). For this test your knowledge box, let’s implement the more simple, PCA-based sufficient similarity analysis to identify groups of biomass smoke exposure signatures using chemical profiles. The relevant dataset is included in the file <em>Module6_5_TYKInput.csv</em>. Specifically:</p>
<ol style="list-style-type: decimal">
<li><p>Perform a PCA on the chemistry data and visualize the proximity of each chemical signature to other signatures according to the first two principal components.</p></li>
<li><p>Identify major groupings of biomass smoke exposure signatures.</p></li>
</ol>
</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="toxicokinetic-modeling.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
