<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4.5 Multi-Group and Multi-Variable Comparisons and Visualizations | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="generator" content="bookdown 0.46 and GitBook 2.6.7" />

  <meta property="og:title" content="4.5 Multi-Group and Multi-Variable Comparisons and Visualizations | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4.5 Multi-Group and Multi-Variable Comparisons and Visualizations | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  <meta name="twitter:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  

<meta name="author" content="Rager Lab" />


<meta name="date" content="2026-02-09" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="images/icons/Favicon.png" type="image/x-icon" />
<link rel="prev" href="two-group-comparisons-and-visualizations.html"/>
<link rel="next" href="advanced-multi-group-comparisons.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.9/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.34.0/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.2/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.2/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>
<link href="libs/table1-1.0/table1_defaults.css" rel="stylesheet" />
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-FGFKVX2X3G"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-FGFKVX2X3G');
</script>


<style type="text/css">
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="images/icons/TAME 2.0 Logo.png"><br></a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Chapter 1 Introductory <br>Data Science</b></span></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html"><i class="fa fa-check"></i>1.1 FAIR Data Management Practices</a>
<ul>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-training-module"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-fair"><i class="fa fa-check"></i>Introduction to FAIR</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#breaking-down-fair-letter-by-letter"><i class="fa fa-check"></i>Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#what-does-this-mean-for-you"><i class="fa fa-check"></i>What Does This Mean for You?</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#data-repositories-for-sharing-of-data"><i class="fa fa-check"></i>Data Repositories for Sharing of Data</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#recent-shifts-in-regulatory-policies-for-data-sharing"><i class="fa fa-check"></i>Recent Shifts in Regulatory Policies for Data Sharing</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#additional-training-resources-on-fair"><i class="fa fa-check"></i>Additional Training Resources on FAIR</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html"><i class="fa fa-check"></i>1.2 Data Sharing through Online Repositories</a>
<ul>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#an-overview-and-example-with-the-dataverse-repository"><i class="fa fa-check"></i>An Overview and Example with the Dataverse Repository</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#introduction-to-training-module-1"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#data-repositories"><i class="fa fa-check"></i>Data Repositories</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#the-dataverse-project"><i class="fa fa-check"></i>The Dataverse Project</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#what-is-a-dataverse"><i class="fa fa-check"></i>What is a Dataverse?</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#metadata"><i class="fa fa-check"></i>Metadata</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataverse"><i class="fa fa-check"></i>Creating a Dataverse</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataset"><i class="fa fa-check"></i>Creating a Dataset</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#concluding-remarks-1"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html"><i class="fa fa-check"></i>1.3 File Management using Github</a>
<ul>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#introduction-to-training-module-2"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-an-account"><i class="fa fa-check"></i>Creating an Account</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-a-repository"><i class="fa fa-check"></i>Creating a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#uploading-code"><i class="fa fa-check"></i>Uploading Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#adding-subfolders-in-a-repository"><i class="fa fa-check"></i>Adding Subfolders in a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-code"><i class="fa fa-check"></i>Updating Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-repository-titles-and-structure-to-support-a-manuscript"><i class="fa fa-check"></i>Updating Repository Titles and Structure to Support a Manuscript</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#tracking-code-changes-using-github-branches"><i class="fa fa-check"></i>Tracking Code Changes using Github Branches</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#concluding-remarks-2"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html"><i class="fa fa-check"></i>1.4 Data Wrangling in Excel</a>
<ul>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#introduction-to-training-module-3"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#save-a-copy-of-the-soon-to-be-organized-and-cleaned-dataset-as-a-new-file"><i class="fa fa-check"></i>Save a Copy of the Soon-To-Be Organized and Cleaned Dataset as a New File</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-extraneous-white-space"><i class="fa fa-check"></i>Remove Extraneous White Space</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#replace-missing-data-with-na"><i class="fa fa-check"></i>Replace Missing Data with “NA”</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-metadata-tab"><i class="fa fa-check"></i>Create a Metadata Tab</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#abbreviate-and-capitalize-categorical-data"><i class="fa fa-check"></i>Abbreviate and Capitalize Categorical Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#alphabetize-sort-the-data-by-the-categorical-variable-of-interest"><i class="fa fa-check"></i>Alphabetize (Sort) the Data by the Categorical Variable of Interest</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-new-subject-number-column"><i class="fa fa-check"></i>Create a New Subject Number Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-special-symbols-and-dashes"><i class="fa fa-check"></i>Remove Special Symbols and Dashes</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#bold-all-column-names-and-center-all-data"><i class="fa fa-check"></i>Bold all Column Names and Center all Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-subject-identifier-column"><i class="fa fa-check"></i>Create a Subject Identifier Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#separate-subject-demographic-data-from-experimental-measurements"><i class="fa fa-check"></i>Separate Subject Demographic Data from Experimental Measurements</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#convert-data-from-wide-to-long-format"><i class="fa fa-check"></i>Convert Data from Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#pivoting-data-from-a-wide-to-long-format"><i class="fa fa-check"></i>Pivoting Data from a Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#generating-summary-level-statistics-with-pivot-tables"><i class="fa fa-check"></i>Generating Summary-Level Statistics with Pivot Tables</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#excel-vs.-r-which-should-you-use"><i class="fa fa-check"></i>Excel vs. R: Which Should You Use?</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#concluding-remarks-3"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 2 Coding in R</b></span></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html"><i class="fa fa-check"></i>2.1 Downloading and Programming in R</a>
<ul>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-training-module-4"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#general-introduction-and-installation-of-r-and-rstudio"><i class="fa fa-check"></i>General Introduction and Installation of R and RStudio</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-r-packages"><i class="fa fa-check"></i>Introduction to R Packages</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#scripting-basics"><i class="fa fa-check"></i>Scripting Basics</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#code-troubleshooting"><i class="fa fa-check"></i>Code Troubleshooting</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#concluding-remarks-4"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html"><i class="fa fa-check"></i>2.2 Coding “Best” Practices</a>
<ul>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#introduction-to-training-module-5"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#scripting-file-types"><i class="fa fa-check"></i>Scripting File Types</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-headers-and-annotation"><i class="fa fa-check"></i>Script Headers and Annotation</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#coding-style"><i class="fa fa-check"></i>Coding Style</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-organization"><i class="fa fa-check"></i>Script Organization</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#concluding-remarks-5"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html"><i class="fa fa-check"></i>2.3 Data Manipulation and Reshaping</a>
<ul>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-training-module-6"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#data-manipulation-using-base-r"><i class="fa fa-check"></i>Data Manipulation Using Base R</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-tidyverse"><i class="fa fa-check"></i>Introduction to Tidyverse</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#concluding-remarks-6"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html"><i class="fa fa-check"></i>2.4 Improving Coding Efficiencies</a>
<ul>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#introduction-to-training-module-7"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#loops"><i class="fa fa-check"></i>Loops</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#functions"><i class="fa fa-check"></i>Functions</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#list-operations"><i class="fa fa-check"></i>List operations</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#concluding-remarks-7"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#additional-resources-2"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="part"><span><b>Chapter 3 Basics of Data <br>Analysis and Visualizations</b></span></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html"><i class="fa fa-check"></i>3.1 Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-data-visualizations"><i class="fa fa-check"></i>Introduction to Data Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-training-module-8"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#density-plot-visualization"><i class="fa fa-check"></i>Density Plot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#boxplot-visualization"><i class="fa fa-check"></i>Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#correlation-visualizations"><i class="fa fa-check"></i>Correlation Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#heatmap-visualization"><i class="fa fa-check"></i>Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#concluding-remarks-8"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html"><i class="fa fa-check"></i>3.2 Improving Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-data-visulization-conventions"><i class="fa fa-check"></i>Introduction to Data Visulization Conventions</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-training-module-9"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-boxplot-visualization"><i class="fa fa-check"></i>Creating an Improved Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-heatmap-visualization"><i class="fa fa-check"></i>Creating an Improved Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-multi-plot-figures"><i class="fa fa-check"></i>Creating Multi-Plot Figures</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#concluding-remarks-9"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html"><i class="fa fa-check"></i>3.3 Normality Tests and Data Transformations</a>
<ul>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#introduction-to-training-module-10"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#what-is-a-normal-distribution"><i class="fa fa-check"></i>What is a Normal Distribution?</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#qualitative-assessment-of-normality"><i class="fa fa-check"></i>Qualitative Assessment of Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#quantitative-normality-assessment"><i class="fa fa-check"></i>Quantitative Normality Assessment</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#data-transformation"><i class="fa fa-check"></i>Data Transformation</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#additional-considerations-regarding-normality"><i class="fa fa-check"></i>Additional Considerations Regarding Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#concluding-remarks-10"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="introduction-to-statistical-tests.html"><a href="introduction-to-statistical-tests.html"><i class="fa fa-check"></i>3.4 Introduction to Statistical Tests</a>
<ul>
<li class="chapter" data-level="" data-path="introduction-to-statistical-tests.html"><a href="introduction-to-statistical-tests.html#introduction-to-training-module-11"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="introduction-to-statistical-tests.html"><a href="introduction-to-statistical-tests.html#assessing-normality-homogeneity-of-variance"><i class="fa fa-check"></i>Assessing Normality &amp; Homogeneity of Variance</a></li>
<li class="chapter" data-level="" data-path="introduction-to-statistical-tests.html"><a href="introduction-to-statistical-tests.html#two-group-visualizations-and-statistical-comparisons-using-the-t-test"><i class="fa fa-check"></i>Two-Group Visualizations and Statistical Comparisons using the T-Test</a></li>
<li class="chapter" data-level="" data-path="introduction-to-statistical-tests.html"><a href="introduction-to-statistical-tests.html#three-group-visualizations-and-statistical-comparisons-using-an-anova"><i class="fa fa-check"></i>Three-Group Visualizations and Statistical Comparisons using an ANOVA</a></li>
<li class="chapter" data-level="" data-path="introduction-to-statistical-tests.html"><a href="introduction-to-statistical-tests.html#regression-modeling-and-visualization-linear-and-logistic-regressions"><i class="fa fa-check"></i>Regression Modeling and Visualization: Linear and Logistic Regressions</a></li>
<li class="chapter" data-level="" data-path="introduction-to-statistical-tests.html"><a href="introduction-to-statistical-tests.html#statistical-evaluations-of-categorical-data-using-the-chi-squared-test-and-fishers-exact-test"><i class="fa fa-check"></i>Statistical Evaluations of Categorical Data using the Chi-Squared Test and Fisher’s Exact Test</a></li>
<li class="chapter" data-level="" data-path="introduction-to-statistical-tests.html"><a href="introduction-to-statistical-tests.html#concluding-remarks-11"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 4 Converting Wet <br>Lab Data into Dry Lab <br>Analyses</b></span></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html"><i class="fa fa-check"></i>4.1 Overview of Experimental Design and Example Data</a>
<ul>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#introduction-to-training-module-12"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#replicates"><i class="fa fa-check"></i>Replicates</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#orientation-to-example-data-for-chapter-4"><i class="fa fa-check"></i>Orientation to Example Data for Chapter 4</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#concluding-remarks-12"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html"><i class="fa fa-check"></i>4.2 Data Import, Processing, and Summary Statistics</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#introduction-to-training-module-13"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#data-import"><i class="fa fa-check"></i>Data Import</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#handling-missing-values"><i class="fa fa-check"></i>Handling Missing Values</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#averaging-replicates"><i class="fa fa-check"></i>Averaging Replicates</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#descriptive-statistics"><i class="fa fa-check"></i>Descriptive Statistics</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#normality-assessment-and-data-transformation"><i class="fa fa-check"></i>Normality Assessment and Data Transformation</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#concluding-remarks-13"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html"><i class="fa fa-check"></i>4.3 Data Import from PDF Sources</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#introduction-to-training-module-14"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-from-many-single-pdfs-with-the-same-formatting"><i class="fa fa-check"></i>Importing Data from Many Single PDFs with the Same Formatting</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-stored-in-pdf-tables"><i class="fa fa-check"></i>Importing Data Stored in PDF Tables</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#concluding-remarks-14"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html"><i class="fa fa-check"></i>4.4 Two Group Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#introduction-to-training-module-15"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#overview-of-two-group-statistical-tests"><i class="fa fa-check"></i>Overview of Two Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#statistical-vs.-biological-significance"><i class="fa fa-check"></i>Statistical vs. Biological Significance</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#unpaired-test-example"><i class="fa fa-check"></i>Unpaired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#paired-test-example"><i class="fa fa-check"></i>Paired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#visualizing-results"><i class="fa fa-check"></i>Visualizing Results</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#concluding-remarks-15"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html"><i class="fa fa-check"></i>4.5 Multi-Group and Multi-Variable Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#introduction-to-training-module-16"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#overview-of-multi-group-statistical-tests"><i class="fa fa-check"></i>Overview of Multi-Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#multi-group-analysis-example"><i class="fa fa-check"></i>Multi-Group Analysis Example</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#visualization-of-multi-group-statistical-results"><i class="fa fa-check"></i>Visualization of Multi-Group Statistical Results</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#concluding-remarks-16"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html"><i class="fa fa-check"></i>4.6 Advanced Multi-Group Comparisons</a>
<ul>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#introduction-to-training-module-17"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#advanced-multi-group-comparisons-1"><i class="fa fa-check"></i>Advanced Multi-Group Comparisons</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#two-way-anova-example"><i class="fa fa-check"></i>Two-way ANOVA Example</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#ancova-example"><i class="fa fa-check"></i>ANCOVA Example</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#concluding-remarks-17"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 5 Machine Learning <br>&amp; Artificial Intelligence</b></span></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><i class="fa fa-check"></i>5.1 Introduction to Artificial Intelligence, Machine Learning, and Predictive Modeling for Environmental Health</a>
<ul>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#introduction-to-training-module-18"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#general-historical-context-and-taxonomy-of-modern-aiml"><i class="fa fa-check"></i>General Historical Context and Taxonomy of Modern AI/ML</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#application-of-machine-learning-in-environmental-health-science"><i class="fa fa-check"></i>Application of Machine Learning in Environmental Health Science</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#concluding-remarks-18"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html"><i class="fa fa-check"></i>5.2 Supervised Machine Learning</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-training-module-19"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-machine-learning"><i class="fa fa-check"></i>Types of Machine Learning</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-supervised-machine-learning-algorithms"><i class="fa fa-check"></i>Types of Supervised Machine Learning Algorithms</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#training-supervised-machine-learning-models"><i class="fa fa-check"></i>Training Supervised Machine Learning Models</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#assessing-classification-based-model-performance"><i class="fa fa-check"></i>Assessing Classification-Based Model Performance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-activity-and-example-dataset"><i class="fa fa-check"></i>Introduction to Activity and Example Dataset</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#testing-for-differences-in-predictor-variables-across-the-outcome-classes"><i class="fa fa-check"></i>Testing for Differences in Predictor Variables across the Outcome Classes</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#predicting-ias-detection-with-a-random-forest-rf-model"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#class-imbalance"><i class="fa fa-check"></i>Class Imbalance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#concluding-remarks-19"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html"><i class="fa fa-check"></i>5.3 Supervised Machine Learning Model Interpretation</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-training-module-20"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance"><i class="fa fa-check"></i>Variable Importance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary"><i class="fa fa-check"></i>Decision Boundary</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-example-dataset-and-activity"><i class="fa fa-check"></i>Introduction to Example Dataset and Activity</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#predicting-ias-detection-with-a-random-forest-rf-model-1"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance-plot"><i class="fa fa-check"></i>Variable Importance Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot"><i class="fa fa-check"></i>Decision Boundary Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot-incorporating-smote"><i class="fa fa-check"></i>Decision Boundary Plot Incorporating SMOTE</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#concluding-remarks-20"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><i class="fa fa-check"></i>5.4 Unsupervised Machine Learning Part 1: K-Means Clustering &amp; PCA</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-training-module-21"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-unsupervised-machine-learning"><i class="fa fa-check"></i>Introduction to Unsupervised Machine Learning</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-example-data"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#identifying-clusters-of-chemicals-through-k-means"><i class="fa fa-check"></i>Identifying Clusters of Chemicals through <em>K</em>-Means</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#principal-component-analysis-pca-1"><i class="fa fa-check"></i>Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#incorporating-k-means-into-pca-for-predictive-modeling"><i class="fa fa-check"></i>Incorporating <em>K</em>-Means into PCA for Predictive Modeling</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#concluding-remarks-21"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><i class="fa fa-check"></i>5.5 Unsupervised Machine Learning Part 2: Additional Clustering Applications</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-training-module-22"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#k-means-clustering-1"><i class="fa fa-check"></i><em>K</em>-Means Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-example-data-1"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-1"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#clustering-plot"><i class="fa fa-check"></i>Clustering Plot</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-visualization"><i class="fa fa-check"></i>Hierarchical Clustering Visualization</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#variable-contributions"><i class="fa fa-check"></i>Variable Contributions</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#concluding-remarks-22"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 6 Applications in <br>Toxicology &amp; Exposure <br>Science</b></span></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html"><i class="fa fa-check"></i>6.1 Descriptive Cohort Analyses</a>
<ul>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#introduction-to-training-module-23"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#participant-exploration"><i class="fa fa-check"></i>Participant Exploration</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#chemical-detection-filtering"><i class="fa fa-check"></i>Chemical Detection Filtering</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#outlier-identification"><i class="fa fa-check"></i>Outlier Identification</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#summary-statistics-tables"><i class="fa fa-check"></i>Summary Statistics Tables</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#demographics-table"><i class="fa fa-check"></i>Demographics Table</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#concluding-remarks-23"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html"><i class="fa fa-check"></i>6.2 -Omics and System Biology: Transcriptomic Applications</a>
<ul>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-training-module-24"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-the-field-of--omics"><i class="fa fa-check"></i>Introduction to the Field of “-Omics”</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-transcriptomics"><i class="fa fa-check"></i>Introduction to Transcriptomics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#formatting-data-for-downstream-statistics"><i class="fa fa-check"></i>Formatting Data for Downstream Statistics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#transcriptomics-data-qaqc"><i class="fa fa-check"></i>Transcriptomics Data QA/QC</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#controlling-for-sources-of-sample-heterogeneity"><i class="fa fa-check"></i>Controlling for Sources of Sample Heterogeneity</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e.g.-biomass-smoke-exposure"><i class="fa fa-check"></i>Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure)</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-ma-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using MA Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-volcano-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using Volcano Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis"><i class="fa fa-check"></i>Interpretting Findings at the Systems Level through Pathway Enrichment Analysis</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#concluding-remarks-24"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><i class="fa fa-check"></i>6.3 Mixtures Analysis Methods Part 1: Overview and Example with Quantile G-Computation</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-training-module-25"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#overview-of-mixtures-analysis"><i class="fa fa-check"></i>Overview of Mixtures Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-example-data-2"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#mixtures-model-with-standard-qqcomp"><i class="fa fa-check"></i>Mixtures Model with Standard qqcomp</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#concluding-remarks-25"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#additional-resources-11"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><i class="fa fa-check"></i>6.4 Mixtures Analysis Methods Part 2: Bayesian Kernel Machine Regression</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-training-module-26"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-example-data-3"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-bkmr"><i class="fa fa-check"></i>Introduction to BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#run-bkmr"><i class="fa fa-check"></i>Run BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#concluding-remarks-26"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#additional-resources-12"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html"><i class="fa fa-check"></i>6.5 Mixtures Analysis Methods Part 3: Sufficient Similarity</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-training-module-27"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-toxicant-and-dataset"><i class="fa fa-check"></i>Introduction to Toxicant and Dataset</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#workspace-preparation-and-data-import-9"><i class="fa fa-check"></i>Workspace Preparation and Data Import</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#chemistry-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Chemistry-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#toxicity-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Toxicity-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#comparing-chemistry-vs.-toxicity-sufficient-similarity-analyses"><i class="fa fa-check"></i>Comparing Chemistry vs. Toxicity Sufficient Similarity Analyses</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#concluding-remarks-27"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i>6.6 Toxicokinetic Modeling</a>
<ul>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-training-module-28"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-toxicokinetic-modeling"><i class="fa fa-check"></i>Introduction to Toxicokinetic Modeling</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#data-and-models-used-in-toxicokinetic-modeling-tk"><i class="fa fa-check"></i>Data and Models used in Toxicokinetic Modeling (TK)</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-toxicokinetics"><i class="fa fa-check"></i>Reverse Toxicokinetics</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#capturing-population-variability-in-toxicokinetics-and-uncertainty-in-chemical-specific-parameters"><i class="fa fa-check"></i>Capturing Population Variability in Toxicokinetics, and Uncertainty in Chemical-Specific Parameters</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-tk-calculating-administered-equivalent-doses-for-toxcast-bioactive-concentrations"><i class="fa fa-check"></i>Reverse TK: Calculating Administered Equivalent Doses for ToxCast Bioactive Concentrations</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#comparing-equivalent-doses-estimated-to-elicit-toxicity-hazard-to-external-exposure-estimates-exposure-for-chemical-prioritization-by-bioactivity-exposure-ratios-bers"><i class="fa fa-check"></i>Comparing Equivalent Doses Estimated to Elicit Toxicity (Hazard) to External Exposure Estimates (Exposure), for Chemical Prioritization by Bioactivity-Exposure Ratios (BERs)</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#concluding-remarks-28"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html"><i class="fa fa-check"></i>6.7 Chemical Read-Across for Toxicity Predictions</a>
<ul>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#introduction-to-training-module-29"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#introduction-to-activity"><i class="fa fa-check"></i>Introduction to Activity</a></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#read-across-example-analysis"><i class="fa fa-check"></i>Read-Across Example Analysis</a></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#calculating-chemical-similarities"><i class="fa fa-check"></i>Calculating Chemical Similarities</a></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#chemical-read-across-to-predict-acute-toxicity"><i class="fa fa-check"></i>Chemical Read-Across to Predict Acute Toxicity</a></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#concluding-remarks-29"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 7 Environmental<br> Health Database Mining</b></span></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html"><i class="fa fa-check"></i>7.1 Comparative Toxicogenomics Database</a>
<ul>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#introduction-to-training-module-30"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#ctd-data-in-r"><i class="fa fa-check"></i>CTD Data in R</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#identifying-genes-under-epigenetic-control"><i class="fa fa-check"></i>Identifying Genes Under Epigenetic Control</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#concluding-remarks-30"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html"><i class="fa fa-check"></i>7.2 Gene Expression Omnibus</a>
<ul>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-training-module-31"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-geo"><i class="fa fa-check"></i>Introduction to GEO</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-example-data-4"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#geo-data-in-r"><i class="fa fa-check"></i>GEO Data in R</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#visualizing-data"><i class="fa fa-check"></i>Visualizing Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#statistical-analyses"><i class="fa fa-check"></i>Statistical Analyses</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#concluding-remarks-31"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html"><i class="fa fa-check"></i>7.3 CompTox Dashboard Data through APIs</a>
<ul>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#introduction-to-training-module-32"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#training-modules-environmental-health-questions-24"><i class="fa fa-check"></i>Training Module’s Environmental Health Questions</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#script-preparations-12"><i class="fa fa-check"></i>Script Preparations</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#introduction-to-comptox-chemicals-dashboard"><i class="fa fa-check"></i>Introduction to CompTox Chemicals Dashboard</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#cctes-ctx-application-programming-interfaces-apis-for-automated-batch-search-of-the-ccd"><i class="fa fa-check"></i>CCTE’s CTX Application Programming Interfaces (APIs) for Automated Batch Search of the CCD</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#comparing-physico-chemical-properties-between-two-important-environmental-contaminant-lists"><i class="fa fa-check"></i>Comparing Physico-chemical Properties between Two Important Environmental Contaminant Lists</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#hazard-data-genotoxicity"><i class="fa fa-check"></i>Hazard Data: Genotoxicity</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#hazard-resource"><i class="fa fa-check"></i>Hazard Resource</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#concluding-remarks-32"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="multi-group-and-multi-variable-comparisons-and-visualizations" class="section level1 hasAnchor">
<h1>4.5 Multi-Group and Multi-Variable Comparisons and Visualizations<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#multi-group-and-multi-variable-comparisons-and-visualizations" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This training module was developed by Elise Hickman, Alexis Payton, and Julia E. Rager.</p>
<p>All input files (script, data, and figures) can be downloaded from the <a href="https://github.com/UNCSRP/TAME2">UNC-SRP TAME2 GitHub website</a>.</p>
<div id="introduction-to-training-module-16" class="section level2 hasAnchor">
<h2>Introduction to Training Module<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#introduction-to-training-module-16" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In the previous module, we covered how to apply two-group statistical testing, one of the most basic types of statistical tests. In this module, we will build on the concepts introduced previously to apply statistical testing to datasets with more than two groups, which are also very common in environmental health research. We will review common multi-group overall effects tests and post-hoc tests, and we will demonstrate how to apply these tests and how to graph the results using the same example dataset as in previous modules in this chapter, which represents concentrations of inflammatory biomarkers secreted by airway epithelial cells after exposure to different concentrations of acrolein.</p>
<div id="training-modules-environmental-health-questions-9" class="section level3 hasAnchor">
<h3>Training Module’s Environmental Health Questions<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#training-modules-environmental-health-questions-9" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This training module was specifically developed to answer the following environmental health questions:</p>
<ol style="list-style-type: decimal">
<li>Are there significant differences in inflammatory biomarker concentrations between different doses of acrolein?</li>
<li>Do TNF-<span class="math inline">\(\alpha\)</span> concentrations significantly increase with increasing dose of acrolein?</li>
</ol>
</div>
<div id="workspace-preparation-and-data-import-5" class="section level3 hasAnchor">
<h3>Workspace Preparation and Data Import<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#workspace-preparation-and-data-import-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Here, we will import the processed data that we generated at the end of TAME 2.0 Module 4.2, introduced in <strong>TAME 2.0 Module 4.1 Overview of Experimental Design and Example Data</strong> and the associated demographic data. These data represent log<sub>2</sub> concentrations of inflammatory biomarkers secreted by airway epithelial cells after exposure to four different concentrations of acrolein (plus filtered air as a control). We will also load packages that will be needed for the analysis, including previously introduced packages such as <em>openxlsx</em>, <em>tidyverse</em>, <em>DT</em>, <em>ggpubr</em>, and <em>rstatix</em>.</p>
<div id="cleaning-the-global-environment-4" class="section level4 hasAnchor">
<h4>Cleaning the global environment<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cleaning-the-global-environment-4" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb461"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb461-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb461-1" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list=</span><span class="fu">ls</span>())</span></code></pre></div>
</div>
<div id="loading-r-packages-required-for-this-session-3" class="section level4 hasAnchor">
<h4>Loading R packages required for this session<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#loading-r-packages-required-for-this-session-3" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb462"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb462-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb462-1" tabindex="-1"></a><span class="fu">library</span>(openxlsx)</span>
<span id="cb462-2"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb462-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb462-3"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb462-3" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb462-4"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb462-4" tabindex="-1"></a><span class="fu">library</span>(rstatix)</span>
<span id="cb462-5"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb462-5" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span></code></pre></div>
</div>
<div id="set-your-working-directory-5" class="section level4 hasAnchor">
<h4>Set your working directory<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#set-your-working-directory-5" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb463"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb463-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb463-1" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/filepath to where your input files are&quot;</span>)</span></code></pre></div>
</div>
<div id="importing-example-dataset-4" class="section level4 hasAnchor">
<h4>Importing example dataset<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#importing-example-dataset-4" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb464"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb464-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb464-1" tabindex="-1"></a>biomarker_data <span class="ot">&lt;-</span> <span class="fu">read.xlsx</span>(<span class="st">&quot;Chapter_4/Module4_5_Input/Module4_5_InputData1.xlsx&quot;</span>)</span>
<span id="cb464-2"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb464-2" tabindex="-1"></a>demographic_data <span class="ot">&lt;-</span> <span class="fu">read.xlsx</span>(<span class="st">&quot;Chapter_4/Module4_5_Input/Module4_5_InputData2.xlsx&quot;</span>)</span>
<span id="cb464-3"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb464-3" tabindex="-1"></a></span>
<span id="cb464-4"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb464-4" tabindex="-1"></a><span class="co"># View data</span></span>
<span id="cb464-5"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb464-5" tabindex="-1"></a><span class="fu">datatable</span>(biomarker_data)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-9656dcf29d4d2b11cb3a" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-9656dcf29d4d2b11cb3a">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80"],["D10","D10","D10","D10","D10","D11","D11","D11","D11","D11","D12","D12","D12","D12","D12","D13","D13","D13","D13","D13","D14","D14","D14","D14","D14","D15","D15","D15","D15","D15","D16","D16","D16","D16","D16","D1","D1","D1","D1","D1","D2","D2","D2","D2","D2","D3","D3","D3","D3","D3","D4","D4","D4","D4","D4","D5","D5","D5","D5","D5","D6","D6","D6","D6","D6","D7","D7","D7","D7","D7","D8","D8","D8","D8","D8","D9","D9","D9","D9","D9"],["0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4","0","0.6","1","2","4"],[3.71808758396052,3.7235585614721,3.70597790168252,3.76553474636298,3.6724253419715,3.83491342970104,3.93073733756289,3.92884403671257,3.80632405739003,3.81762325751143,3.36036427654347,3.27649666564036,3.57167680997093,3.54101915313356,3.10097764772482,3.35755200461808,3.33628338786443,3.31469652565629,3.50589092972996,3.69041709553247,3.31614574229336,3.43562859405209,3.57773093149008,3.87578006306849,3.83086375675176,3.42894634476948,3.49825086752783,3.57288966842058,3.63807383718072,3.6915341649192,3.38681094647222,3.49825086752783,3.58856473740135,3.66220549965362,3.69376571221778,3.76871365703048,3.77820857639809,3.74953426766926,3.84096670448742,3.50589092972996,3.53480866123175,3.60880924267552,3.58856473740135,3.2585189247113,3.75167794568758,3.58014548442338,3.73876783680019,3.61823865559545,3.58736499093646,3.39643353125099,3.52732060787705,3.50842865253186,3.62175885670973,3.23266075679027,4.13996056954546,3.16671544496642,3.18269229751619,3.13422093976063,3.7279204545632,3.799087306074,2.86195536414487,3.08746284125034,3.64961545906341,3.66220549965362,3.64616265715789,3.04614178164472,3.1811025507538,3.27202318906105,3.54101915313356,3.42088657497553,3.37712374912949,3.57531233068744,3.58976348698498,3.83289001416474,3.8972404255748,3.21878116778407,3.30888505741176,3.51853513898218,3.5765221379205,3.7279204545632],[8.68422263328295,8.773963368433559,8.561326136453919,9.710324307072939,7.55028504908841,10.2222394489088,10.1590009172826,9.94747622373011,9.81610366284939,8.03815081026938,9.27052894238072,9.05077375519723,8.99740532549677,8.27635707925924,7.61889934790406,8.15790268278964,8.089053172156429,7.41515020542639,7.37920505996061,4.10265813136374,7.56094417806009,7.5894638929561,7.9047851297316,7.83857365266832,6.50969584193339,9.254131033541951,9.422148885880191,9.165032591502291,9.09402488708305,8.202809492125009,8.68422263328295,8.773963368433559,8.77360079528254,8.60099074240823,7.71163256182322,10.1099481540673,10.3357141430951,9.686097776967401,10.0151641270811,10.1498741117182,8.88962564051876,9.317005526387209,9.03875441393145,8.60099074240823,4.87479696589866,9.23160520814165,9.677085350559841,9.23208509872283,9.583628142323651,9.560179957764181,8.34837407519116,8.351469324267081,8.77360079528254,9.33695280016051,7.70355756192103,3.9429835981871,4.12018602753186,4.24868662687383,5.4110872267986,5.17871464117544,7.43212427264156,6.85112436288675,7.38076407498511,6.96220195703608,8.36382813184704,8.12861303870471,8.0503384381673,7.86530024272912,7.3639595068907,8.90926307937532,11.4238722598684,11.7288813640812,11.4570986742783,10.878112237463,8.52575569282949,7.01077983875324,7.32804655627074,7.03452387533439,7.27537959625362,4.70099449416827],[15.0940690235359,15.1807289234454,15.1191518828101,16.3458996676138,16.7241460980745,15.0028356680746,15.5465951732949,15.4894995973234,16.169128959283,16.0996069083821,13.9520295507825,14.0492691434224,13.1502883461063,14.8168226706095,16.3560077655342,14.30563571243,14.2750718138596,14.2603087281481,14.3170903457307,16.6038859325635,13.9567554773633,14.1371656727442,14.4431119040824,15.1950634099335,16.3613300730261,14.4305440239947,14.6030328264996,14.6583565484435,15.3569594653045,16.5368999542995,14.3291992954213,14.4239037658366,14.5587322032289,15.1950634099335,16.3717050181953,15.1620702745062,15.5056676741312,15.1429673906615,15.4079647745043,16.9008155214109,14.6079451777076,14.6753941971444,13.1497550568455,14.9206710618732,17.2934204566841,14.5987126147926,14.8967320984425,14.6970451993658,15.0929012215711,17.1264130896721,14.542155252148,14.4584290430224,14.8614654668171,15.9922404459517,15.3426552970282,14.1001424218123,13.9841362654046,14.0738746130993,15.2370937904406,16.3510586187167,13.4034767756747,13.4654490662827,13.7572472746771,15.5483372770406,16.2848127161883,13.5861396108168,13.8384840329337,13.8699889688218,14.4670316946933,17.0361832734254,14.1156143059506,14.4239037658366,14.4497857465895,15.0182445232662,15.1484821432167,14.3291992954213,14.1946037445951,13.1529914376826,14.708543993667,16.3449160294487],[0.189033824390017,-0.3040061868901,0.286881147788162,0.298658315564515,1.0703893278914,2.3305584000308,1.37851162325373,1.30451104180995,0.978195629681652,1.64154602908752,0.526068811667588,0.238786859587116,0.713695814843359,1.50589092972996,0.333423733725192,1.24488705912353,-0.3040061868901,1.56559717585423,1.09761079662642,0.855989697308481,1.06350294230616,0.20163386116965,0.895302621333307,1.1243281350022,1.37295209791183,1.27500704749987,0.632268215499513,1.16992500144231,0.910732661902913,1.16992500144231,0.8155754288625729,0.333423733725192,0.748461233004036,0.903038270112912,1.13750352374994,0.367371065648529,0.137503523749935,0.464668267003444,0.910732661902913,0.454175893185802,1.35614381022528,0.298658315564515,0.632268215499513,0.748461233004036,1.40599235967584,0.31034012061215,0.124328135002202,0.53605290024021,0.495695162624069,0.321928094887362,0.879705766282288,0.298658315564515,1.01435529297707,1.19534759832222,2.03210084316702,0.933572638261024,0.20163386116965,2.68706068833989,1.13750352374994,0.84799690655495,1.86393845042397,1.41142624572647,0.992768430768924,1.0071955014042,0.443606651475615,0.650764559116902,0.8718436485093179,1.19534759832222,0.16349873228288,1.17632277264046,1.01435529297707,0.941106310946431,1.04264433740849,0.8718436485093179,1.57531233068744,2.51349074558812,0.604071323668861,0.411426245726465,0.584962500721156,1.47508488294878],[1.73552217729654,1.55581615506164,1.93734439215023,2.08406426478847,2.83390207666916,3.23572705983806,3.24031432933371,3.15218341905505,3.16510798514454,3.48284828306847,1.63226821549951,1.91073266190291,2.35049724708413,3.24031432933371,1.57046293102604,1.32192809488736,1.20789285164133,1.20789285164133,2.75274859140713,2.59931779369823,1.91456452349394,1.72246602447109,1.96347412397489,2.81352468929781,3.27202318906105,2.08746284125034,2.02502879449152,2.37016428054021,2.63923216324928,2.94485844580754,1.83187724119167,1.91073266190291,2.0496307677246,2.64154602908752,2.93168305705981,1.86393845042397,1.790772037862,1.86393845042397,1.88362081628567,2.88362081628567,1.83187724119167,2.0071955014042,2.02147972741045,2.48542682717024,2.98913900736823,2.52606881166759,1.49569516262407,1.85598969730848,1.50080205305716,2.06695024392463,2.47508488294878,2.31034012061215,2.39231742277876,2.99457972421575,3.73226919950145,2.20476675065461,2.34482849699744,1.96347412397489,2.40053792958373,2.85798099512757,1.790772037862,2.11103131238874,2.39506279951758,3.0232553523003,2.16671544496642,1.56559717585423,1.32192809488736,3.59693514238723,1.73552217729654,2.75702324650746,2.53106949272595,2.48284828306847,2.7158933705476,2.99819550315325,3.47118746038699,1.0703893278914,1.33342373372519,1.97819562968165,2.58496250072116,3.08406426478847],[10.4475275409266,10.4320198468125,10.4523853997561,10.7211590148677,11.2458081856384,9.02901108665907,9.22583403553784,9.412739192116421,9.91233549843777,9.316688823436939,9.428025110276749,9.412146396540679,9.620146497339491,10.9260809830959,7.3039633562453,9.247642742898019,9.32424806586749,10.2153509830024,11.2601080099627,10.8557639471917,10.6537676442859,10.6473145095471,10.8149028512154,11.0078962674403,10.9341110109735,9.933439857942711,10.009954550333,10.2267734234482,10.7809497987497,10.646036617491,9.797969484059131,9.8490610025615,10.2090268302306,10.7211590148677,10.8557639471917,9.696480751820189,9.72723977748481,9.49325503512067,10.0461417816447,11.0193335777127,10.534828110721,10.8215188973143,10.8050102485124,11.7900226112658,11.031563370024,9.35887093968562,9.18935511254853,9.033670771393091,9.36430978186441,11.0217850685075,8.283551423174311,8.22211866974429,8.617467465294199,9.414283550503709,8.973180744513019,9.425320822759881,9.308885057411761,9.61402881967061,10.4624204732526,10.7035402591663,10.7662472988462,10.9813603523445,11.5188793273505,11.3227900730377,9.89590846480624,10.3333342781343,10.3711558797312,10.374854737178,10.7157046495212,11.7319596151152,9.92478281821723,9.8490610025615,10.0477374043091,10.6360535917989,9.40224465525408,9.797969484059131,10.0853263897027,10.4360035526972,11.191225812124,10.9299462991725]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Donor<\/th>\n      <th>Dose<\/th>\n      <th>IL1B<\/th>\n      <th>IL6<\/th>\n      <th>IL8<\/th>\n      <th>IL10<\/th>\n      <th>TNFa<\/th>\n      <th>VEGF<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4,5,6,7,8]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Donor","targets":1},{"name":"Dose","targets":2},{"name":"IL1B","targets":3},{"name":"IL6","targets":4},{"name":"IL8","targets":5},{"name":"IL10","targets":6},{"name":"TNFa","targets":7},{"name":"VEGF","targets":8}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb465"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb465-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb465-1" tabindex="-1"></a><span class="fu">datatable</span>(demographic_data)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-e6c2951cd62acf70012f" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-e6c2951cd62acf70012f">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16"],["D1","D2","D3","D4","D5","D6","D7","D8","D9","D10","D11","D12","D13","D14","D15","D16"],[22,35,27,60,43,38,26,36,40,44,46,52,58,65,37,25],["M","F","M","F","F","M","M","F","F","M","M","M","M","M","F","F"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Donor<\/th>\n      <th>Age_yr<\/th>\n      <th>Sex<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Donor","targets":1},{"name":"Age_yr","targets":2},{"name":"Sex","targets":3}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p><br></p>
</div>
</div>
</div>
<div id="overview-of-multi-group-statistical-tests" class="section level2 hasAnchor">
<h2>Overview of Multi-Group Statistical Tests<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#overview-of-multi-group-statistical-tests" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Before applying statistical tests to our data, let’s first review the mechanics of multi-group statistical tests, including overall effects tests and post-hoc tests.
<img src="Chapter_4/Module4_5_Input/Module4_5_Image1.png" alt="" width="600px" style="display: block; margin: auto;" /></p>
<div id="overall-effects-tests" class="section level3 hasAnchor">
<h3>Overall Effects Tests<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#overall-effects-tests" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The first step for multi-group statistical testing is to run an overall effects test. The null hypothesis for the overall effects test is that there are no differences among group means. A significant p-value rejects the null hypothesis that the groups are drawn from populations with the same mean and indicates that at least one group differs significantly from the overall mean. Similar to two-group statistical testing, choice of the specific overall statistical test to run depends on whether the data are normally or non-normally distributed and whether the experimental design is paired:</p>
<p><img src="Chapter_4/Module4_5_Input/Module4_5_Image2.png" alt="" width="700px" style="display: block; margin: auto;" /></p>
<p>Importantly, overall effects tests return <strong>one</strong> p-value regardless of the number of groups being compared. To determine which pairwise comparisons are significant, post-hoc testing is needed.</p>
</div>
<div id="post-hoc-testing" class="section level3 hasAnchor">
<h3>Post-Hoc Testing<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#post-hoc-testing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>If significance is obtained with an overall effects test, we can use post-hoc testing to determine which specific pairs of groups are significantly different from each other. Just as with two group statistical tests and overall effects multi-group statistical tests, choosing the appropriate post-hoc test depends on the data’s normality and whether the experimental design is paired:
<img src="Chapter_4/Module4_5_Input/Module4_5_Image3.png" alt="" width="700px" style="display: block; margin: auto;" /></p>
<p>Note that the above diagram represents commonly selected post-hoc tests; others may also be appropriate depending on your specific experimental design. As with other aspects of the analysis, be sure to report which post-hoc test(s) you performed!</p>
</div>
<div id="correcting-for-multiple-hypothesis-testing" class="section level3 hasAnchor">
<h3>Correcting for Multiple Hypothesis Testing<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#correcting-for-multiple-hypothesis-testing" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Correcting for multiple hypothesis testing is important for both the overall effects test (if you are running it over many endpoints) and post-hoc tests; however, it is particularly important for post-hoc tests. This is because even an analysis of a relatively small number of experimental groups results in quite a few pairwise comparisons. Comparing each of our five dose groups to each other in our example data, there are 10 separate statistical tests being performed! Therefore, it is generally advisable to adjust pairwise post-hoc testing p-values. The Tukey’s HSD function within <em>rstatix</em> does this automatically, while pairwise t-tests, pairwise Wilcoxon tests, and Dunn’s test do not. P-value adjustment can be added to their respective <em>rstatix</em> functions using the <code>p.adjust.method =</code> argument.</p>
<p>When applying a post-hoc test, you may choose to compare every group to every other group, or you may only be interested in significant differences between specific groups (e.g., treatment groups vs. a control). This choice will be governed by your hypothesis. Statistical testing functions will typically default to comparing all groups to each other, but the comparisons can be defined using the <code>comparisons =</code> argument if you want to restrict the test to specific comparisons. It is important to decide at the beginning of your analysis which comparisons are relevant to your hypothesis because the number of pairwise tests performed in the post-hoc analysis will influence how much the resulting p-values will be adjusted for multiple hypothesis testing.</p>
</div>
<div id="which-test-should-i-choose-1" class="section level3 hasAnchor">
<h3>Which test should I choose?<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#which-test-should-i-choose-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Use the following flowchart to help guide your choice of statistical test to compare multiple groups:
<img src="Chapter_4/Module4_5_Input/Module4_5_Image4.png" alt="" width="900px" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="multi-group-analysis-example" class="section level2 hasAnchor">
<h2>Multi-Group Analysis Example<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#multi-group-analysis-example" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To determine whether there are significant differences across all of our doses, the Friedman test is the most appropriate due to our matched experimental design and non-normally distributed data. The <code>friedman_test()</code> function is part of the <a href="https://github.com/kassambara/rstatix">rstatix</a> package. This package also has many other helpful functions for statistical tests that are pipe/tidyverse friendly. To demonstrate how this test works, we will first perform the test on one variable:</p>
<div class="sourceCode" id="cb466"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb466-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb466-1" tabindex="-1"></a>biomarker_data <span class="sc">%&gt;%</span> <span class="fu">friedman_test</span>(IL1B <span class="sc">~</span> Dose <span class="sc">|</span> Donor)</span></code></pre></div>
<pre><code>## # A tibble: 1 × 6
##   .y.       n statistic    df      p method       
## * &lt;chr&gt; &lt;int&gt;     &lt;dbl&gt; &lt;dbl&gt;  &lt;dbl&gt; &lt;chr&gt;        
## 1 IL1B     16      12.5     4 0.0140 Friedman test</code></pre>
<p>A p-value of 0.01 indicates that we can reject the null hypothesis that all of our data are drawn from groups that have equivalent means.</p>
<p>Now, we can run a <code>for</code> loop similar to our two-group comparisons in <strong>TAME 2.0 Module 4.4 Two Group Comparisons and Visualizations</strong> to determine the overall p-value for each endpoint:</p>
<div class="sourceCode" id="cb468"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb468-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-1" tabindex="-1"></a><span class="co"># Create a vector with the names of the variables you want to run the test on</span></span>
<span id="cb468-2"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-2" tabindex="-1"></a>endpoints <span class="ot">&lt;-</span> <span class="fu">colnames</span>(biomarker_data <span class="sc">%&gt;%</span> <span class="fu">select</span>(IL1B<span class="sc">:</span>VEGF))</span>
<span id="cb468-3"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-3" tabindex="-1"></a></span>
<span id="cb468-4"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-4" tabindex="-1"></a><span class="co"># Create data frame to store results</span></span>
<span id="cb468-5"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-5" tabindex="-1"></a>dose_friedmanres <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb468-6"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-6" tabindex="-1"></a></span>
<span id="cb468-7"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-7" tabindex="-1"></a><span class="co"># Run for loop</span></span>
<span id="cb468-8"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-8" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(endpoints)) {</span>
<span id="cb468-9"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-9" tabindex="-1"></a>  </span>
<span id="cb468-10"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-10" tabindex="-1"></a>  <span class="co"># Assign a name to the endpoint variable.</span></span>
<span id="cb468-11"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-11" tabindex="-1"></a>  endpoint <span class="ot">&lt;-</span> endpoints[i]</span>
<span id="cb468-12"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-12" tabindex="-1"></a>  </span>
<span id="cb468-13"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-13" tabindex="-1"></a>  <span class="co"># Run wilcox test and store in results data frame.</span></span>
<span id="cb468-14"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-14" tabindex="-1"></a>  res <span class="ot">&lt;-</span> biomarker_data <span class="sc">%&gt;%</span></span>
<span id="cb468-15"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-15" tabindex="-1"></a>    <span class="fu">friedman_test</span>(<span class="fu">as.formula</span>(<span class="fu">paste0</span>(endpoint, <span class="st">&quot;~ Dose | Donor&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb468-16"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-16" tabindex="-1"></a>    <span class="fu">select</span>(<span class="fu">c</span>(.y., p))</span>
<span id="cb468-17"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-17" tabindex="-1"></a>  </span>
<span id="cb468-18"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-18" tabindex="-1"></a>  dose_friedmanres <span class="ot">&lt;-</span> <span class="fu">rbind</span>(dose_friedmanres, res)</span>
<span id="cb468-19"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-19" tabindex="-1"></a>}</span>
<span id="cb468-20"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-20" tabindex="-1"></a></span>
<span id="cb468-21"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-21" tabindex="-1"></a><span class="co"># View results</span></span>
<span id="cb468-22"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb468-22" tabindex="-1"></a><span class="fu">datatable</span>(dose_friedmanres)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-b1e3a138061a75b07b6d" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-b1e3a138061a75b07b6d">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6"],["IL1B","IL6","IL8","IL10","TNFa","VEGF"],[0.01399579248765089,0.01558587421705306,4.071975170150775e-10,0.002839556743989798,8.904966446852206e-07,2.338026427265107e-07]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>.y.<\/th>\n      <th>p<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":2},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":".y.","targets":1},{"name":"p","targets":2}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>These results demonstrate that all of our endpoints have significant overall differences across doses (p &lt; 0.05). To determine which pairwise comparisons are significant, we next need to apply a post-hoc test. We will apply a pairwise, paired Wilcoxon test due to our experimental design and data distribution, with the Benjamini-Hochberg (BH) correction for multiple testing:</p>
<div class="sourceCode" id="cb469"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb469-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb469-1" tabindex="-1"></a>dose_wilcox_posthoc_IL1B <span class="ot">&lt;-</span> biomarker_data <span class="sc">%&gt;%</span> </span>
<span id="cb469-2"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb469-2" tabindex="-1"></a>  <span class="fu">pairwise_wilcox_test</span>(IL1B <span class="sc">~</span> Dose, <span class="at">paired =</span> <span class="cn">TRUE</span>, <span class="at">p.adjust.method =</span> <span class="st">&quot;BH&quot;</span>)</span>
<span id="cb469-3"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb469-3" tabindex="-1"></a></span>
<span id="cb469-4"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb469-4" tabindex="-1"></a>dose_wilcox_posthoc_IL1B </span></code></pre></div>
<pre><code>## # A tibble: 10 × 9
##    .y.   group1 group2    n1    n2 statistic     p p.adj p.adj.signif
##  * &lt;chr&gt; &lt;chr&gt;  &lt;chr&gt;  &lt;int&gt; &lt;int&gt;     &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;chr&gt;       
##  1 IL1B  0      0.6       16    16        17 0.006 0.025 *           
##  2 IL1B  0      1         16    16        11 0.002 0.017 *           
##  3 IL1B  0      2         16    16        21 0.013 0.033 *           
##  4 IL1B  0      4         16    16        18 0.008 0.025 *           
##  5 IL1B  0.6    1         16    16        38 0.13  0.186 ns          
##  6 IL1B  0.6    2         16    16        31 0.058 0.096 ns          
##  7 IL1B  0.6    4         16    16        25 0.025 0.05  *           
##  8 IL1B  1      2         16    16        43 0.211 0.234 ns          
##  9 IL1B  1      4         16    16        40 0.159 0.199 ns          
## 10 IL1B  2      4         16    16        62 0.782 0.782 ns</code></pre>
<p>Here, we can now see whether there are statistically significant differences in IL-1<span class="math inline">\(\beta\)</span> secretion between each of our doses. To generate pairwise comparison results for each of our inflammatory biomarkers, we can run a for loop similar to the one we ran for our overall test:</p>
<div class="sourceCode" id="cb471"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb471-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-1" tabindex="-1"></a><span class="co"># Create a vector with the names of the variables you want to run the test on</span></span>
<span id="cb471-2"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-2" tabindex="-1"></a>endpoints <span class="ot">&lt;-</span> <span class="fu">colnames</span>(biomarker_data <span class="sc">%&gt;%</span> <span class="fu">select</span>(IL1B<span class="sc">:</span>VEGF))</span>
<span id="cb471-3"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-3" tabindex="-1"></a></span>
<span id="cb471-4"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-4" tabindex="-1"></a><span class="co"># Create data frame to store results</span></span>
<span id="cb471-5"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-5" tabindex="-1"></a>dose_wilcox_posthoc <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb471-6"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-6" tabindex="-1"></a></span>
<span id="cb471-7"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-7" tabindex="-1"></a><span class="co"># Run for loop</span></span>
<span id="cb471-8"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-8" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(endpoints)) {</span>
<span id="cb471-9"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-9" tabindex="-1"></a>  </span>
<span id="cb471-10"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-10" tabindex="-1"></a>  <span class="co"># Assign a name to the endpoint variable.</span></span>
<span id="cb471-11"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-11" tabindex="-1"></a>  endpoint <span class="ot">&lt;-</span> endpoints[i]</span>
<span id="cb471-12"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-12" tabindex="-1"></a>  </span>
<span id="cb471-13"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-13" tabindex="-1"></a>  <span class="co"># Run wilcox test and store in results data frame.</span></span>
<span id="cb471-14"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-14" tabindex="-1"></a>  res <span class="ot">&lt;-</span> biomarker_data <span class="sc">%&gt;%</span></span>
<span id="cb471-15"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-15" tabindex="-1"></a>    <span class="fu">pairwise_wilcox_test</span>(<span class="fu">as.formula</span>(<span class="fu">paste0</span>(endpoint, <span class="st">&quot;~ Dose&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>)), <span class="at">paired =</span> <span class="cn">TRUE</span>, <span class="at">p.adjust.method =</span> <span class="st">&quot;BH&quot;</span>)</span>
<span id="cb471-16"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-16" tabindex="-1"></a>  </span>
<span id="cb471-17"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-17" tabindex="-1"></a>  dose_wilcox_posthoc <span class="ot">&lt;-</span> <span class="fu">rbind</span>(dose_wilcox_posthoc, res)</span>
<span id="cb471-18"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-18" tabindex="-1"></a>}</span>
<span id="cb471-19"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-19" tabindex="-1"></a></span>
<span id="cb471-20"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-20" tabindex="-1"></a><span class="co"># View results</span></span>
<span id="cb471-21"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb471-21" tabindex="-1"></a><span class="fu">datatable</span>(dose_wilcox_posthoc)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-ab213ebda19739278874" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ab213ebda19739278874">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60"],["IL1B","IL1B","IL1B","IL1B","IL1B","IL1B","IL1B","IL1B","IL1B","IL1B","IL6","IL6","IL6","IL6","IL6","IL6","IL6","IL6","IL6","IL6","IL8","IL8","IL8","IL8","IL8","IL8","IL8","IL8","IL8","IL8","IL10","IL10","IL10","IL10","IL10","IL10","IL10","IL10","IL10","IL10","TNFa","TNFa","TNFa","TNFa","TNFa","TNFa","TNFa","TNFa","TNFa","TNFa","VEGF","VEGF","VEGF","VEGF","VEGF","VEGF","VEGF","VEGF","VEGF","VEGF"],["0","0","0","0","0.6","0.6","0.6","1","1","2","0","0","0","0","0.6","0.6","0.6","1","1","2","0","0","0","0","0.6","0.6","0.6","1","1","2","0","0","0","0","0.6","0.6","0.6","1","1","2","0","0","0","0","0.6","0.6","0.6","1","1","2","0","0","0","0","0.6","0.6","0.6","1","1","2"],["0.6","1","2","4","1","2","4","2","4","4","0.6","1","2","4","1","2","4","2","4","4","0.6","1","2","4","1","2","4","2","4","4","0.6","1","2","4","1","2","4","2","4","4","0.6","1","2","4","1","2","4","2","4","4","0.6","1","2","4","1","2","4","2","4","4"],[16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16],[16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16],[17,11,21,18,38,31,25,43,40,62,38,76,76,114,96,83,116,72,118,117,22,53,0,0,78,2,0,0,0,5,133,69,84,65,13,24,11,74,45,41,72,30,14,5,15,3,3,22,15,30,44,17,0,28,16,0,30,2,39,84],[0.006,0.002,0.013,0.008,0.13,0.058,0.025,0.211,0.159,0.782,0.127,0.706,0.706,0.016,0.159,0.464,0.011,0.86,0.008,0.008999999999999999,0.016,0.464,3.05e-05,3.05e-05,0.632,9.16e-05,3.05e-05,3.05e-05,3.05e-05,0.000305,0.000153,0.98,0.433,0.9,0.003,0.021,0.002,0.782,0.41,0.175,0.86,0.094,0.003,0.000305,0.012,0.000153,0.000153,0.016,0.004,0.051,0.231,0.006,3.05e-05,0.039,0.005,3.05e-05,0.051,9.16e-05,0.144,0.433],[0.025,0.017,0.033,0.025,0.186,0.096,0.05,0.234,0.199,0.782,0.254,0.784,0.784,0.039,0.265,0.663,0.037,0.86,0.037,0.037,0.019,0.516,6.1e-05,6.1e-05,0.632,0.000153,6.1e-05,6.1e-05,6.1e-05,0.000436,0.002,0.98,0.619,0.98,0.008999999999999999,0.054,0.008,0.978,0.619,0.35,0.86,0.104,0.008,0.001,0.019,0.0007649999999999999,0.0007649999999999999,0.022,0.008,0.063,0.257,0.013,0.000152,0.064,0.013,0.000152,0.07199999999999999,0.000305,0.18,0.433],["*","*","*","*","ns","ns","*","ns","ns","ns","ns","ns","ns","*","ns","ns","*","ns","*","*","*","ns","****","****","ns","***","****","****","****","***","**","ns","ns","ns","**","ns","**","ns","ns","ns","ns","ns","**","**","*","***","***","*","**","ns","ns","*","***","ns","*","***","ns","***","ns","ns"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>.y.<\/th>\n      <th>group1<\/th>\n      <th>group2<\/th>\n      <th>n1<\/th>\n      <th>n2<\/th>\n      <th>statistic<\/th>\n      <th>p<\/th>\n      <th>p.adj<\/th>\n      <th>p.adj.signif<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[4,5,6,7,8]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":".y.","targets":1},{"name":"group1","targets":2},{"name":"group2","targets":3},{"name":"n1","targets":4},{"name":"n2","targets":5},{"name":"statistic","targets":6},{"name":"p","targets":7},{"name":"p.adj","targets":8},{"name":"p.adj.signif","targets":9}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>We now have a dataframe storing all of our pairwise comparison results. However, this is a lot to scroll through, making it hard to interpret. We can generate a publication-quality table by manipulating the table and joining it with the overall test data.</p>
<div class="sourceCode" id="cb472"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb472-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb472-1" tabindex="-1"></a>dose_results_cleaned <span class="ot">&lt;-</span> dose_wilcox_posthoc <span class="sc">%&gt;%</span></span>
<span id="cb472-2"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb472-2" tabindex="-1"></a>  <span class="fu">unite</span>(comparison, group1, group2, <span class="at">sep =</span> <span class="st">&quot; vs. &quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb472-3"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb472-3" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(.y., comparison, p.adj)) <span class="sc">%&gt;%</span></span>
<span id="cb472-4"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb472-4" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> <span class="st">&quot;.y.&quot;</span>, <span class="at">names_from =</span> <span class="st">&quot;comparison&quot;</span>, <span class="at">values_from =</span> <span class="st">&quot;p.adj&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb472-5"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb472-5" tabindex="-1"></a>  <span class="fu">left_join</span>(dose_friedmanres, <span class="at">by =</span> <span class="st">&quot;.y.&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb472-6"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb472-6" tabindex="-1"></a>  <span class="fu">relocate</span>(p, <span class="at">.after =</span> <span class="st">&quot;.y.&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb472-7"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb472-7" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">&quot;Variable&quot;</span> <span class="ot">=</span> <span class="st">&quot;.y.&quot;</span>, <span class="st">&quot;Overall&quot;</span> <span class="ot">=</span> <span class="st">&quot;p&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb472-8"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb472-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="st">&#39;Overall&#39;</span><span class="sc">:</span><span class="st">&#39;2 vs. 4&#39;</span>, \(x) <span class="fu">format</span>(x, <span class="at">scientific =</span> <span class="cn">TRUE</span>, <span class="at">digits =</span> <span class="dv">3</span>)))</span>
<span id="cb472-9"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb472-9" tabindex="-1"></a></span>
<span id="cb472-10"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb472-10" tabindex="-1"></a><span class="fu">datatable</span>(dose_results_cleaned)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-31737f180aa90ce38144" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-31737f180aa90ce38144">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6"],["IL1B","IL6","IL8","IL10","TNFa","VEGF"],["1.40e-02","1.56e-02","4.07e-10","2.84e-03","8.90e-07","2.34e-07"],["2.50e-02","2.54e-01","1.90e-02","2.00e-03","8.60e-01","2.57e-01"],["1.70e-02","7.84e-01","5.16e-01","9.80e-01","1.04e-01","1.30e-02"],["3.30e-02","7.84e-01","6.10e-05","6.19e-01","8.00e-03","1.52e-04"],["2.5e-02","3.9e-02","6.1e-05","9.8e-01","1.0e-03","6.4e-02"],["1.86e-01","2.65e-01","6.32e-01","9.00e-03","1.90e-02","1.30e-02"],["9.60e-02","6.63e-01","1.53e-04","5.40e-02","7.65e-04","1.52e-04"],["5.00e-02","3.70e-02","6.10e-05","8.00e-03","7.65e-04","7.20e-02"],["2.34e-01","8.60e-01","6.10e-05","9.78e-01","2.20e-02","3.05e-04"],["1.99e-01","3.70e-02","6.10e-05","6.19e-01","8.00e-03","1.80e-01"],["7.82e-01","3.70e-02","4.36e-04","3.50e-01","6.30e-02","4.33e-01"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Variable<\/th>\n      <th>Overall<\/th>\n      <th>0 vs. 0.6<\/th>\n      <th>0 vs. 1<\/th>\n      <th>0 vs. 2<\/th>\n      <th>0 vs. 4<\/th>\n      <th>0.6 vs. 1<\/th>\n      <th>0.6 vs. 2<\/th>\n      <th>0.6 vs. 4<\/th>\n      <th>1 vs. 2<\/th>\n      <th>1 vs. 4<\/th>\n      <th>2 vs. 4<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Variable","targets":1},{"name":"Overall","targets":2},{"name":"0 vs. 0.6","targets":3},{"name":"0 vs. 1","targets":4},{"name":"0 vs. 2","targets":5},{"name":"0 vs. 4","targets":6},{"name":"0.6 vs. 1","targets":7},{"name":"0.6 vs. 2","targets":8},{"name":"0.6 vs. 4","targets":9},{"name":"1 vs. 2","targets":10},{"name":"1 vs. 4","targets":11},{"name":"2 vs. 4","targets":12}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>To more easily see overall significance patterns, we could also make the same table but with significance stars instead of p-values by keeping the <code>p.adjust.signif</code> column instead of the <code>p.adj</code> column in our post-hoc test results dataframe:</p>
<div class="sourceCode" id="cb473"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb473-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb473-1" tabindex="-1"></a>dose_results_cleaned_2 <span class="ot">&lt;-</span> dose_wilcox_posthoc <span class="sc">%&gt;%</span></span>
<span id="cb473-2"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb473-2" tabindex="-1"></a>  <span class="fu">unite</span>(comparison, group1, group2, <span class="at">sep =</span> <span class="st">&quot; vs. &quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb473-3"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb473-3" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(.y., comparison, p.adj.signif)) <span class="sc">%&gt;%</span></span>
<span id="cb473-4"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb473-4" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">id_cols =</span> <span class="st">&quot;.y.&quot;</span>, <span class="at">names_from =</span> <span class="st">&quot;comparison&quot;</span>, <span class="at">values_from =</span> <span class="st">&quot;p.adj.signif&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb473-5"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb473-5" tabindex="-1"></a>  <span class="fu">left_join</span>(dose_friedmanres, <span class="at">by =</span> <span class="st">&quot;.y.&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb473-6"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb473-6" tabindex="-1"></a>  <span class="fu">relocate</span>(p, <span class="at">.after =</span> <span class="st">&quot;.y.&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb473-7"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb473-7" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">&quot;Variable&quot;</span> <span class="ot">=</span> <span class="st">&quot;.y.&quot;</span>, <span class="st">&quot;Overall&quot;</span> <span class="ot">=</span> <span class="st">&quot;p&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb473-8"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb473-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="st">&#39;Overall&#39;</span><span class="sc">:</span><span class="st">&#39;2 vs. 4&#39;</span>, \(x) <span class="fu">format</span>(x, <span class="at">scientific =</span> <span class="cn">TRUE</span>, <span class="at">digits =</span> <span class="dv">3</span>)))</span>
<span id="cb473-9"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb473-9" tabindex="-1"></a></span>
<span id="cb473-10"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb473-10" tabindex="-1"></a><span class="fu">datatable</span>(dose_results_cleaned_2)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-e55974bd037526bf441d" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-e55974bd037526bf441d">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6"],["IL1B","IL6","IL8","IL10","TNFa","VEGF"],["1.40e-02","1.56e-02","4.07e-10","2.84e-03","8.90e-07","2.34e-07"],["* ","ns","* ","**","ns","ns"],["* ","ns","ns","ns","ns","* "],["*   ","ns  ","****","ns  ","**  ","*** "],["*   ","*   ","****","ns  ","**  ","ns  "],["ns","ns","ns","**","* ","* "],["ns ","ns ","***","ns ","***","***"],["*   ","*   ","****","**  ","*** ","ns  "],["ns  ","ns  ","****","ns  ","*   ","*** "],["ns  ","*   ","****","ns  ","**  ","ns  "],["ns ","*  ","***","ns ","ns ","ns "]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Variable<\/th>\n      <th>Overall<\/th>\n      <th>0 vs. 0.6<\/th>\n      <th>0 vs. 1<\/th>\n      <th>0 vs. 2<\/th>\n      <th>0 vs. 4<\/th>\n      <th>0.6 vs. 1<\/th>\n      <th>0.6 vs. 2<\/th>\n      <th>0.6 vs. 4<\/th>\n      <th>1 vs. 2<\/th>\n      <th>1 vs. 4<\/th>\n      <th>2 vs. 4<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"Variable","targets":1},{"name":"Overall","targets":2},{"name":"0 vs. 0.6","targets":3},{"name":"0 vs. 1","targets":4},{"name":"0 vs. 2","targets":5},{"name":"0 vs. 4","targets":6},{"name":"0.6 vs. 1","targets":7},{"name":"0.6 vs. 2","targets":8},{"name":"0.6 vs. 4","targets":9},{"name":"1 vs. 2","targets":10},{"name":"1 vs. 4","targets":11},{"name":"2 vs. 4","targets":12}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="answer-to-environmental-health-question-1-6" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 1<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#answer-to-environmental-health-question-1-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><i> With this, we can answer <strong>Environmental Health Question #1 </strong></i>: Are there significant differences in inflammatory biomarker concentrations between different doses of acrolein?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: Yes, there are significant differences in inflammatory biomarker concentrations between different doses of acrolein. The overall p-values for all biomarkers are significant. Within each biomarker, at least one pairwise comparison was significant between doses, with a majority of these significant comparisons being with the highest dose (4 ppm).</p>
</div>
<p><br></p>
</div>
</div>
<div id="visualization-of-multi-group-statistical-results" class="section level2 hasAnchor">
<h2>Visualization of Multi-Group Statistical Results<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#visualization-of-multi-group-statistical-results" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The statistical results we generated are a lot to digest in table format, so it can be helpful to graph the results. As our statistical testing becomes more complicated, so does the code used to generate results. The <em>ggpubr</em> package can perform statistical testing and overlay the results onto graphs for a specific set of tests, such as overall effects tests and unpaired t-tests or Wilcoxon tests. However, for tests that aren’t available by default, the package also contains the helpful <code>stat_pvalue_manual()</code> function that can be added to plots. This is what we will need to use to add the results of our pairwise, paired Wilcoxon test with BH correction, as there is no option for BH correction within the default function we might otherwise use (<code>stat_compare_means()</code>). We will first work through an example of this using one of our endpoints, and then we will demonstrate how to apply it to facet plotting.</p>
<div id="single-plot" class="section level3 hasAnchor">
<h3>Single Plot<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#single-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We first need to format our existing statistical results so that they match the format that the function needs as input. Specifically, the dataframe needs to contain the following columns:</p>
<ul>
<li><code>group1</code> and <code>group2</code>: the groups being compared</li>
<li>A column containing the results you want displayed (<code>p</code>, <code>p.adj</code>, or <code>p.adj.signif</code> typically)</li>
<li><code>y.position</code>, which tells the function where to plot the significance markers</li>
</ul>
<p>Our results dataframe for IL-1<span class="math inline">\(\beta\)</span> already contains our groups and p-values:</p>
<div class="sourceCode" id="cb474"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb474-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb474-1" tabindex="-1"></a><span class="fu">datatable</span>(dose_wilcox_posthoc_IL1B)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-881d67060d1746536de8" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-881d67060d1746536de8">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10"],["IL1B","IL1B","IL1B","IL1B","IL1B","IL1B","IL1B","IL1B","IL1B","IL1B"],["0","0","0","0","0.6","0.6","0.6","1","1","2"],["0.6","1","2","4","1","2","4","2","4","4"],[16,16,16,16,16,16,16,16,16,16],[16,16,16,16,16,16,16,16,16,16],[17,11,21,18,38,31,25,43,40,62],[0.006,0.002,0.013,0.008,0.13,0.058,0.025,0.211,0.159,0.782],[0.025,0.017,0.033,0.025,0.186,0.096,0.05,0.234,0.199,0.782],["*","*","*","*","ns","ns","*","ns","ns","ns"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>.y.<\/th>\n      <th>group1<\/th>\n      <th>group2<\/th>\n      <th>n1<\/th>\n      <th>n2<\/th>\n      <th>statistic<\/th>\n      <th>p<\/th>\n      <th>p.adj<\/th>\n      <th>p.adj.signif<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[4,5,6,7,8]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":".y.","targets":1},{"name":"group1","targets":2},{"name":"group2","targets":3},{"name":"n1","targets":4},{"name":"n2","targets":5},{"name":"statistic","targets":6},{"name":"p","targets":7},{"name":"p.adj","targets":8},{"name":"p.adj.signif","targets":9}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>We can add the position columns using the function <code>add_xy_position()</code>:</p>
<div class="sourceCode" id="cb475"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb475-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb475-1" tabindex="-1"></a>dose_wilcox_posthoc_IL1B <span class="ot">&lt;-</span> dose_wilcox_posthoc_IL1B <span class="sc">%&gt;%</span></span>
<span id="cb475-2"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb475-2" tabindex="-1"></a>  <span class="fu">add_xy_position</span>(<span class="at">x =</span> <span class="st">&quot;Dose&quot;</span>, <span class="at">step.increase =</span> <span class="dv">2</span>)</span>
<span id="cb475-3"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb475-3" tabindex="-1"></a></span>
<span id="cb475-4"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb475-4" tabindex="-1"></a><span class="fu">datatable</span>(dose_wilcox_posthoc_IL1B)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-3910b698e8037327dec7" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-3910b698e8037327dec7">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10"],["IL1B","IL1B","IL1B","IL1B","IL1B","IL1B","IL1B","IL1B","IL1B","IL1B"],["0","0","0","0","0.6","0.6","0.6","1","1","2"],["0.6","1","2","4","1","2","4","2","4","4"],[16,16,16,16,16,16,16,16,16,16],[16,16,16,16,16,16,16,16,16,16],[17,11,21,18,38,31,25,43,40,62],[0.006,0.002,0.013,0.008,0.13,0.058,0.025,0.211,0.159,0.782],[0.025,0.017,0.033,0.025,0.186,0.096,0.05,0.234,0.199,0.782],["*","*","*","*","ns","ns","*","ns","ns","ns"],[4.749999999999999,5.427777777777776,6.105555555555553,6.783333333333331,7.461111111111109,8.138888888888886,8.816666666666663,9.49444444444444,10.17222222222222,10.84999999999999],[["0","0.6"],["0","1"],["0","2"],["0","4"],["0.6","1"],["0.6","2"],["0.6","4"],["1","2"],["1","4"],["2","4"]],[1,1,1,1,2,2,2,3,3,4],[2,3,4,5,3,4,5,4,5,5]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>.y.<\/th>\n      <th>group1<\/th>\n      <th>group2<\/th>\n      <th>n1<\/th>\n      <th>n2<\/th>\n      <th>statistic<\/th>\n      <th>p<\/th>\n      <th>p.adj<\/th>\n      <th>p.adj.signif<\/th>\n      <th>y.position<\/th>\n      <th>groups<\/th>\n      <th>xmin<\/th>\n      <th>xmax<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[4,5,6,7,8,10,12,13]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":".y.","targets":1},{"name":"group1","targets":2},{"name":"group2","targets":3},{"name":"n1","targets":4},{"name":"n2","targets":5},{"name":"statistic","targets":6},{"name":"p","targets":7},{"name":"p.adj","targets":8},{"name":"p.adj.signif","targets":9},{"name":"y.position","targets":10},{"name":"groups","targets":11},{"name":"xmin","targets":12},{"name":"xmax","targets":13}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Now, we are ready to make a graph of our results. We will use <code>stat_friedman_test()</code> to add our overall p-value and <code>stat_pvalue_manual()</code> to add our pairwise values.</p>
<div class="sourceCode" id="cb476"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb476-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb476-1" tabindex="-1"></a><span class="co"># Set graphing theme</span></span>
<span id="cb476-2"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb476-2" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>())</span>
<span id="cb476-3"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb476-3" tabindex="-1"></a></span>
<span id="cb476-4"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb476-4" tabindex="-1"></a><span class="co"># Make plot</span></span>
<span id="cb476-5"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb476-5" tabindex="-1"></a><span class="fu">ggplot</span>(biomarker_data, <span class="fu">aes</span>(<span class="at">x =</span> Dose, <span class="at">y =</span> IL1B)) <span class="sc">+</span></span>
<span id="cb476-6"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb476-6" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> Dose), <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb476-7"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb476-7" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;#BFBFBF&quot;</span>, <span class="st">&quot;#D5A298&quot;</span>, <span class="st">&quot;#E38273&quot;</span>, <span class="st">&quot;#EB5F4E&quot;</span>, <span class="st">&quot;#EE2B2B&quot;</span>)) <span class="sc">+</span></span>
<span id="cb476-8"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb476-8" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="fl">0.15</span>)) <span class="sc">+</span></span>
<span id="cb476-9"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb476-9" tabindex="-1"></a>  <span class="fu">stat_friedman_test</span>(<span class="at">wid =</span> <span class="st">&quot;Donor&quot;</span>, <span class="at">p.adjust.method =</span> <span class="st">&quot;none&quot;</span>, <span class="at">label =</span> <span class="st">&quot;p = {p.format}&quot;</span>, </span>
<span id="cb476-10"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb476-10" tabindex="-1"></a>                     <span class="at">label.x.npc =</span> <span class="st">&quot;left&quot;</span>, <span class="at">label.y =</span> <span class="fl">9.5</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">6</span>) <span class="sc">+</span></span>
<span id="cb476-11"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb476-11" tabindex="-1"></a>  <span class="fu">stat_pvalue_manual</span>(dose_wilcox_posthoc_IL1B, <span class="at">label =</span> <span class="st">&quot;p.adj.signif&quot;</span>, <span class="at">size =</span> <span class="dv">12</span>, <span class="at">hide.ns =</span> <span class="cn">TRUE</span>) <span class="sc">+</span></span>
<span id="cb476-12"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb476-12" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="fl">2.5</span>, <span class="dv">10</span>) <span class="sc">+</span></span>
<span id="cb476-13"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb476-13" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Log2(IL-1\u03B2  (pg/mL))&quot;</span>, <span class="at">x =</span> <span class="st">&quot;Acrolein (ppm)&quot;</span>) <span class="sc">+</span></span>
<span id="cb476-14"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb476-14" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb476-15"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb476-15" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">15</span>),</span>
<span id="cb476-16"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb476-16" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.75</span>),</span>
<span id="cb476-17"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb476-17" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="dv">2</span>),</span>
<span id="cb476-18"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb476-18" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">12</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-357-1.png" alt="" width="600px" style="display: block; margin: auto;" /></p>
<p>However, to make room for all of our annotations, our data become compressed, and it makes it difficult to see our data. Although presentation of statistical results is largely a matter of personal preference, we could clean up this plot by making our annotations appear on top of the bars, with indication in the figure legend that the comparison is with a specific dose. We will do this by:</p>
<ol style="list-style-type: decimal">
<li>Filtering our results to those that are significant.</li>
<li>Changing the symbol for comparisons that are not to the 0 dose.</li>
<li>Layering this text onto the plot with <code>geom_text()</code> rather than <code>stat_pvalue_manual()</code>.</li>
</ol>
<p>First, let’s filter our results to significant results and change the symbol for comparisons that are not to the 0 dose to a caret (^) instead of stars. We can do this by creating a new column called label that keeps the existing label if <code>group1</code> is 0, and if not, changes the label to a caret of the same length. We then use the summarize function to paste the labels for each of the groups together, resulting in a final dataframe containing our annotations for our plot.</p>
<div class="sourceCode" id="cb477"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb477-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-1" tabindex="-1"></a>dose_wilcox_posthoc_IL1B_2 <span class="ot">&lt;-</span> dose_wilcox_posthoc_IL1B <span class="sc">%&gt;%</span></span>
<span id="cb477-2"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-2" tabindex="-1"></a>  </span>
<span id="cb477-3"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-3" tabindex="-1"></a>  <span class="co"># Filter results to those that are significant</span></span>
<span id="cb477-4"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-4" tabindex="-1"></a>  <span class="fu">filter</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb477-5"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-5" tabindex="-1"></a>  </span>
<span id="cb477-6"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-6" tabindex="-1"></a>  <span class="co"># Make new symbol</span></span>
<span id="cb477-7"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">ifelse</span>(group1 <span class="sc">==</span> <span class="st">&quot;0&quot;</span>, p.adj.signif, <span class="fu">strrep</span>(<span class="st">&quot;^&quot;</span>, <span class="fu">nchar</span>(p.adj.signif)))) <span class="sc">%&gt;%</span></span>
<span id="cb477-8"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-8" tabindex="-1"></a>  </span>
<span id="cb477-9"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-9" tabindex="-1"></a>  <span class="co"># Select only the columns we need</span></span>
<span id="cb477-10"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-10" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(group1, group2, label)) <span class="sc">%&gt;%</span></span>
<span id="cb477-11"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-11" tabindex="-1"></a>  </span>
<span id="cb477-12"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-12" tabindex="-1"></a>  <span class="co"># Combine symbols for the same group</span></span>
<span id="cb477-13"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-13" tabindex="-1"></a>  <span class="fu">group_by</span>(group2) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">label =</span> <span class="fu">paste</span>(label, <span class="at">collapse=</span><span class="st">&quot; &quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb477-14"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-14" tabindex="-1"></a>  </span>
<span id="cb477-15"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-15" tabindex="-1"></a>  <span class="co"># Remove duplicate row</span></span>
<span id="cb477-16"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-16" tabindex="-1"></a>  <span class="fu">distinct</span>(group2, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb477-17"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-17" tabindex="-1"></a>  </span>
<span id="cb477-18"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-18" tabindex="-1"></a>  <span class="co"># Rename group2 to dose</span></span>
<span id="cb477-19"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-19" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">&quot;Dose&quot;</span> <span class="ot">=</span> <span class="st">&quot;group2&quot;</span>)</span>
<span id="cb477-20"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-20" tabindex="-1"></a></span>
<span id="cb477-21"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb477-21" tabindex="-1"></a>dose_wilcox_posthoc_IL1B_2</span></code></pre></div>
<pre><code>## # A tibble: 4 × 2
##   Dose  label
##   &lt;chr&gt; &lt;chr&gt;
## 1 0.6   *    
## 2 1     *    
## 3 2     *    
## 4 4     * ^</code></pre>
<p>Then, we can use the same code as for our previous plot, but instead of using <code>stat_pvalue_manual()</code>, we will use <code>geom_text()</code> in combination with the dataframe we just created.</p>
<div class="sourceCode" id="cb479"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb479-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb479-1" tabindex="-1"></a><span class="fu">ggplot</span>(biomarker_data, <span class="fu">aes</span>(<span class="at">x =</span> Dose, <span class="at">y =</span> IL1B)) <span class="sc">+</span></span>
<span id="cb479-2"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb479-2" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> Dose), <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb479-3"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb479-3" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;#BFBFBF&quot;</span>, <span class="st">&quot;#D5A298&quot;</span>, <span class="st">&quot;#E38273&quot;</span>, <span class="st">&quot;#EB5F4E&quot;</span>, <span class="st">&quot;#EE2B2B&quot;</span>)) <span class="sc">+</span></span>
<span id="cb479-4"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb479-4" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="dv">3</span>, <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="fl">0.15</span>)) <span class="sc">+</span></span>
<span id="cb479-5"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb479-5" tabindex="-1"></a>  <span class="fu">stat_friedman_test</span>(<span class="at">wid =</span> <span class="st">&quot;Donor&quot;</span>, <span class="at">p.adjust.method =</span> <span class="st">&quot;none&quot;</span>, <span class="at">label =</span> <span class="st">&quot;p = {p.format}&quot;</span>, </span>
<span id="cb479-6"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb479-6" tabindex="-1"></a>                     <span class="at">label.x.npc =</span> <span class="st">&quot;left&quot;</span>, <span class="at">label.y =</span> <span class="fl">4.85</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> <span class="dv">6</span>) <span class="sc">+</span></span>
<span id="cb479-7"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb479-7" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> dose_wilcox_posthoc_IL1B_2, <span class="fu">aes</span>(<span class="at">x =</span> Dose, <span class="at">y =</span> <span class="fl">4.5</span>, </span>
<span id="cb479-8"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb479-8" tabindex="-1"></a>                                       <span class="at">label =</span> <span class="fu">paste0</span>(label)), <span class="at">size =</span> <span class="dv">10</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb479-9"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb479-9" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="fl">2.5</span>, <span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb479-10"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb479-10" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Log2(IL-1\u03B2  (pg/mL))&quot;</span>, <span class="at">x =</span> <span class="st">&quot;Acrolein (ppm)&quot;</span>) <span class="sc">+</span></span>
<span id="cb479-11"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb479-11" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb479-12"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb479-12" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">15</span>),</span>
<span id="cb479-13"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb479-13" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.75</span>),</span>
<span id="cb479-14"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb479-14" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="dv">2</span>),</span>
<span id="cb479-15"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb479-15" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">12</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-359-1.png" alt="" width="600px" style="display: block; margin: auto;" /></p>
<p>An appropriate title for this figure could be:</p>
<p>“<strong>Figure X. Exposure to 0.6-4 ppm acrolein increases IL-1<span class="math inline">\(\beta\)</span> secretion in primary human bronchial epithelial cells.</strong> Groups were compared using the Friedman test to obtain overall p-value and Wilcoxon signed rank test for post-hoc testing. * p &lt; 0.05 in comparison with 0 ppm, ^ p &lt; 0.05 in comparison with 0.6 ppm, n = 16 per group (paired).”</p>
</div>
<div id="faceted-plot" class="section level3 hasAnchor">
<h3>Faceted Plot<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#faceted-plot" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Ideally, we would extend this sort of graphical approach to our faceted plot showing all of our endpoints. However, there are quite a few statistically significant comparisons to graph, including comparisons that are significant between different pairs of doses (not just back to the control). While we could attempt to graph all of them, ultimately, this will lead to a cluttered figure panel. When thinking about how to simplify our plots, some options are:</p>
<ol style="list-style-type: decimal">
<li><p>Instead of using the number of symbols to represent p-values, we could use a single symbol to represent any comparison with a p-value with at least p &lt; 0.05, and that symbol could be different depending on which group the significance is in comparison to. Symbols can be difficult to parse in R, so we could use letters or even the group names above the column of interest. For example, if the concentration of an endpoint at 2 ppm was significant in comparison with both 0 and 0.6 ppm, we could annotate “0, 0.6” above the 2 ppm column, or we could choose a letter (“a, b”) or symbol (“*, ^“) to convey these results.</p></li>
<li><p>If the pattern is the same across many of the endpoints measured, we could graph a subset of the endpoints with the most notable data trends or the most biological meaning for the main body of the manuscript, with data for additional endpoints referred to in the text and shown in the supplemental figures or tables.</p></li>
<li><p>If most of the significant comparisons are back to the control group, we could choose to only show comparisons with the control group, with textual description of the other significant comparisons and indication that those specific p-values can be viewed in the supplemental table of results.</p></li>
</ol>
<p>Which approach you decide to take (or maybe another approach altogether) is a matter of both personal preference and your specific study goals. You may also decide that it is important to you to show all significant comparisons, which will require more careful formatting of the plots to ensure that all text and annotations are legible. For this module, we will proceed with option #3 because many of our comparisons to the control dose (0) are significant, and we have enough groups that there likely will not be space to annotate all of them above our data.</p>
<p>We will take similar steps here that we did when constructing our single endpoint graph, with a couple of small differences. Specifically, we need to:</p>
<ol style="list-style-type: decimal">
<li>Create a dataframe of labels/annotations as we did above, but now filtered to only significant comparisons with the 0 group.</li>
<li>Add to the label/annotation dataframe what we want the y position for each of the labels to be, which will be different for each endpoint.</li>
</ol>
<p>First, let’s create our annotations dataframe. We will start with the results dataframe from our posthoc testing:</p>
<div class="sourceCode" id="cb480"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb480-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb480-1" tabindex="-1"></a><span class="fu">datatable</span>(dose_wilcox_posthoc)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-9bcae5a61ca2c0b82a38" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-9bcae5a61ca2c0b82a38">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60"],["IL1B","IL1B","IL1B","IL1B","IL1B","IL1B","IL1B","IL1B","IL1B","IL1B","IL6","IL6","IL6","IL6","IL6","IL6","IL6","IL6","IL6","IL6","IL8","IL8","IL8","IL8","IL8","IL8","IL8","IL8","IL8","IL8","IL10","IL10","IL10","IL10","IL10","IL10","IL10","IL10","IL10","IL10","TNFa","TNFa","TNFa","TNFa","TNFa","TNFa","TNFa","TNFa","TNFa","TNFa","VEGF","VEGF","VEGF","VEGF","VEGF","VEGF","VEGF","VEGF","VEGF","VEGF"],["0","0","0","0","0.6","0.6","0.6","1","1","2","0","0","0","0","0.6","0.6","0.6","1","1","2","0","0","0","0","0.6","0.6","0.6","1","1","2","0","0","0","0","0.6","0.6","0.6","1","1","2","0","0","0","0","0.6","0.6","0.6","1","1","2","0","0","0","0","0.6","0.6","0.6","1","1","2"],["0.6","1","2","4","1","2","4","2","4","4","0.6","1","2","4","1","2","4","2","4","4","0.6","1","2","4","1","2","4","2","4","4","0.6","1","2","4","1","2","4","2","4","4","0.6","1","2","4","1","2","4","2","4","4","0.6","1","2","4","1","2","4","2","4","4"],[16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16],[16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16,16],[17,11,21,18,38,31,25,43,40,62,38,76,76,114,96,83,116,72,118,117,22,53,0,0,78,2,0,0,0,5,133,69,84,65,13,24,11,74,45,41,72,30,14,5,15,3,3,22,15,30,44,17,0,28,16,0,30,2,39,84],[0.006,0.002,0.013,0.008,0.13,0.058,0.025,0.211,0.159,0.782,0.127,0.706,0.706,0.016,0.159,0.464,0.011,0.86,0.008,0.008999999999999999,0.016,0.464,3.05e-05,3.05e-05,0.632,9.16e-05,3.05e-05,3.05e-05,3.05e-05,0.000305,0.000153,0.98,0.433,0.9,0.003,0.021,0.002,0.782,0.41,0.175,0.86,0.094,0.003,0.000305,0.012,0.000153,0.000153,0.016,0.004,0.051,0.231,0.006,3.05e-05,0.039,0.005,3.05e-05,0.051,9.16e-05,0.144,0.433],[0.025,0.017,0.033,0.025,0.186,0.096,0.05,0.234,0.199,0.782,0.254,0.784,0.784,0.039,0.265,0.663,0.037,0.86,0.037,0.037,0.019,0.516,6.1e-05,6.1e-05,0.632,0.000153,6.1e-05,6.1e-05,6.1e-05,0.000436,0.002,0.98,0.619,0.98,0.008999999999999999,0.054,0.008,0.978,0.619,0.35,0.86,0.104,0.008,0.001,0.019,0.0007649999999999999,0.0007649999999999999,0.022,0.008,0.063,0.257,0.013,0.000152,0.064,0.013,0.000152,0.07199999999999999,0.000305,0.18,0.433],["*","*","*","*","ns","ns","*","ns","ns","ns","ns","ns","ns","*","ns","ns","*","ns","*","*","*","ns","****","****","ns","***","****","****","****","***","**","ns","ns","ns","**","ns","**","ns","ns","ns","ns","ns","**","**","*","***","***","*","**","ns","ns","*","***","ns","*","***","ns","***","ns","ns"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>.y.<\/th>\n      <th>group1<\/th>\n      <th>group2<\/th>\n      <th>n1<\/th>\n      <th>n2<\/th>\n      <th>statistic<\/th>\n      <th>p<\/th>\n      <th>p.adj<\/th>\n      <th>p.adj.signif<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[4,5,6,7,8]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":".y.","targets":1},{"name":"group1","targets":2},{"name":"group2","targets":3},{"name":"n1","targets":4},{"name":"n2","targets":5},{"name":"statistic","targets":6},{"name":"p","targets":7},{"name":"p.adj","targets":8},{"name":"p.adj.signif","targets":9}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb481"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb481-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb481-1" tabindex="-1"></a>dose_wilcox_posthoc_forgraph <span class="ot">&lt;-</span> dose_wilcox_posthoc <span class="sc">%&gt;%</span></span>
<span id="cb481-2"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb481-2" tabindex="-1"></a>  </span>
<span id="cb481-3"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb481-3" tabindex="-1"></a>  <span class="fu">filter</span>(p.adj <span class="sc">&lt;=</span> <span class="fl">0.05</span>) <span class="sc">%&gt;%</span></span>
<span id="cb481-4"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb481-4" tabindex="-1"></a>  </span>
<span id="cb481-5"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb481-5" tabindex="-1"></a>  <span class="co"># Filter for only comparisons to 0</span></span>
<span id="cb481-6"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb481-6" tabindex="-1"></a>  <span class="fu">filter</span>(group1 <span class="sc">==</span> <span class="st">&quot;0&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb481-7"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb481-7" tabindex="-1"></a>  </span>
<span id="cb481-8"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb481-8" tabindex="-1"></a>  <span class="co"># Rename columns</span></span>
<span id="cb481-9"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb481-9" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">&quot;variable&quot;</span> <span class="ot">=</span> <span class="st">&quot;.y.&quot;</span>, <span class="st">&quot;Dose&quot;</span> <span class="ot">=</span> <span class="st">&quot;group2&quot;</span>)</span>
<span id="cb481-10"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb481-10" tabindex="-1"></a>  </span>
<span id="cb481-11"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb481-11" tabindex="-1"></a><span class="fu">datatable</span>(dose_wilcox_posthoc_forgraph)</span></code></pre></div>
<div class="datatables html-widget html-fill-item" id="htmlwidget-64ab383b77f9dd7893db" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-64ab383b77f9dd7893db">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13"],["IL1B","IL1B","IL1B","IL1B","IL6","IL8","IL8","IL8","IL10","TNFa","TNFa","VEGF","VEGF"],["0","0","0","0","0","0","0","0","0","0","0","0","0"],["0.6","1","2","4","4","0.6","2","4","0.6","2","4","1","2"],[16,16,16,16,16,16,16,16,16,16,16,16,16],[16,16,16,16,16,16,16,16,16,16,16,16,16],[17,11,21,18,114,22,0,0,133,14,5,17,0],[0.006,0.002,0.013,0.008,0.016,0.016,3.05e-05,3.05e-05,0.000153,0.003,0.000305,0.006,3.05e-05],[0.025,0.017,0.033,0.025,0.039,0.019,6.1e-05,6.1e-05,0.002,0.008,0.001,0.013,0.000152],["*","*","*","*","*","*","****","****","**","**","**","*","***"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>variable<\/th>\n      <th>group1<\/th>\n      <th>Dose<\/th>\n      <th>n1<\/th>\n      <th>n2<\/th>\n      <th>statistic<\/th>\n      <th>p<\/th>\n      <th>p.adj<\/th>\n      <th>p.adj.signif<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[4,5,6,7,8]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"variable","targets":1},{"name":"group1","targets":2},{"name":"Dose","targets":3},{"name":"n1","targets":4},{"name":"n2","targets":5},{"name":"statistic","targets":6},{"name":"p","targets":7},{"name":"p.adj","targets":8},{"name":"p.adj.signif","targets":9}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>The <code>Dose</code> column will be used to tell <em>ggplot2</em> where to place the annotations on the x axis, but we need to also specify where to add the annotations on the y axis. This will be different for each variable because each variable is on a different scale. We can approach this by computing the maximum value of each variable, then increasing that by 20% to add some space on top of the points.</p>
<div class="sourceCode" id="cb482"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb482-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb482-1" tabindex="-1"></a>sig_labs_y <span class="ot">&lt;-</span> biomarker_data <span class="sc">%&gt;%</span></span>
<span id="cb482-2"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb482-2" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="fu">across</span>(IL1B<span class="sc">:</span>VEGF, \(x) <span class="fu">max</span>(x))) <span class="sc">%&gt;%</span></span>
<span id="cb482-3"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb482-3" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb482-4"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb482-4" tabindex="-1"></a>  <span class="fu">rownames_to_column</span>(<span class="st">&quot;variable&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb482-5"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb482-5" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">&quot;y_pos&quot;</span> <span class="ot">=</span> <span class="st">&quot;V1&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb482-6"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb482-6" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">y_pos =</span> y_pos<span class="sc">*</span><span class="fl">1.2</span>)</span>
<span id="cb482-7"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb482-7" tabindex="-1"></a></span>
<span id="cb482-8"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb482-8" tabindex="-1"></a>sig_labs_y</span></code></pre></div>
<pre><code>##   variable     y_pos
## 1     IL1B  4.967953
## 2      IL6 14.074658
## 3      IL8 20.752105
## 4     IL10  3.224473
## 5     TNFa  4.478723
## 6     VEGF 14.148027</code></pre>
<p>Then, we can join these data to our labeling dataframe to complete what we need to make the annotations.</p>
<div class="sourceCode" id="cb484"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb484-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb484-1" tabindex="-1"></a>dose_wilcox_posthoc_forgraph <span class="ot">&lt;-</span> dose_wilcox_posthoc_forgraph <span class="sc">%&gt;%</span></span>
<span id="cb484-2"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb484-2" tabindex="-1"></a>  <span class="fu">left_join</span>(sig_labs_y, <span class="at">by =</span> <span class="st">&quot;variable&quot;</span>)</span></code></pre></div>
<p>Now, it’s time to graph! Keep in mind that although the plotting script can get long and unweildy, each line is just a new instruction to ggplot about a formatting element or an additional layer to add to the graph.</p>
<div class="sourceCode" id="cb485"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb485-1"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-1" tabindex="-1"></a><span class="co"># Pivot data longer</span></span>
<span id="cb485-2"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-2" tabindex="-1"></a>biomarker_data_long <span class="ot">&lt;-</span> biomarker_data <span class="sc">%&gt;%</span></span>
<span id="cb485-3"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-3" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="sc">-</span><span class="fu">c</span>(Donor, Dose), <span class="at">names_to =</span> <span class="st">&quot;variable&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;value&quot;</span>)</span>
<span id="cb485-4"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-4" tabindex="-1"></a></span>
<span id="cb485-5"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-5" tabindex="-1"></a><span class="co"># Create clean labels for the graph titles</span></span>
<span id="cb485-6"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-6" tabindex="-1"></a>new_labels <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;IL10&quot;</span> <span class="ot">=</span> <span class="st">&quot;IL-10&quot;</span>, <span class="st">&quot;IL1B&quot;</span> <span class="ot">=</span> <span class="st">&quot;IL-1\u03B2 &quot;</span>, <span class="st">&quot;IL6&quot;</span> <span class="ot">=</span> <span class="st">&quot;IL-6&quot;</span>, <span class="st">&quot;IL8&quot;</span> <span class="ot">=</span> <span class="st">&quot;IL-8&quot;</span>, </span>
<span id="cb485-7"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-7" tabindex="-1"></a>                <span class="st">&quot;TNFa&quot;</span> <span class="ot">=</span> <span class="st">&quot;TNF-\u03b1&quot;</span>, <span class="st">&quot;VEGF&quot;</span> <span class="ot">=</span> <span class="st">&quot;VEGF&quot;</span>)</span>
<span id="cb485-8"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-8" tabindex="-1"></a></span>
<span id="cb485-9"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-9" tabindex="-1"></a><span class="co"># Make graph</span></span>
<span id="cb485-10"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-10" tabindex="-1"></a><span class="fu">ggplot</span>(biomarker_data_long, <span class="fu">aes</span>(<span class="at">x =</span> Dose, <span class="at">y =</span> value)) <span class="sc">+</span></span>
<span id="cb485-11"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-11" tabindex="-1"></a>  <span class="co"># outlier.shape = NA removes outliers</span></span>
<span id="cb485-12"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-12" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">fill =</span> Dose), <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span></span>
<span id="cb485-13"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-13" tabindex="-1"></a>  <span class="co"># Changing box plot colors</span></span>
<span id="cb485-14"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-14" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;#BFBFBF&quot;</span>, <span class="st">&quot;#D5A298&quot;</span>, <span class="st">&quot;#E38273&quot;</span>, <span class="st">&quot;#EB5F4E&quot;</span>, <span class="st">&quot;#EE2B2B&quot;</span>)) <span class="sc">+</span></span>
<span id="cb485-15"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-15" tabindex="-1"></a>  <span class="fu">geom_jitter</span>(<span class="at">size =</span> <span class="fl">1.5</span>, <span class="at">position =</span> <span class="fu">position_jitter</span>(<span class="fl">0.15</span>)) <span class="sc">+</span></span>
<span id="cb485-16"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-16" tabindex="-1"></a>  <span class="co"># Adding a p value from Friedman test</span></span>
<span id="cb485-17"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-17" tabindex="-1"></a>  <span class="fu">stat_friedman_test</span>(<span class="at">wid =</span> <span class="st">&quot;Donor&quot;</span>, <span class="at">p.adjust.method =</span> <span class="st">&quot;none&quot;</span>, <span class="at">label =</span> <span class="st">&quot;p = {p.format}&quot;</span>, </span>
<span id="cb485-18"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-18" tabindex="-1"></a>                     <span class="at">label.x.npc =</span> <span class="st">&quot;left&quot;</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">3.5</span>, <span class="at">hjust =</span> <span class="fl">0.1</span>, <span class="at">size =</span> <span class="fl">3.5</span>) <span class="sc">+</span> </span>
<span id="cb485-19"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-19" tabindex="-1"></a>  <span class="co"># Add label</span></span>
<span id="cb485-20"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-20" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">data =</span> dose_wilcox_posthoc_forgraph, <span class="fu">aes</span>(<span class="at">x =</span> Dose, <span class="at">y =</span> y_pos, <span class="at">label =</span> p.adj.signif,</span>
<span id="cb485-21"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-21" tabindex="-1"></a>                                       <span class="at">size =</span> <span class="dv">5</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span></span>
<span id="cb485-22"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-22" tabindex="-1"></a>  <span class="co"># Adding padding y axis </span></span>
<span id="cb485-23"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-23" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.6</span>))) <span class="sc">+</span></span>
<span id="cb485-24"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-24" tabindex="-1"></a>  <span class="co"># Changing y axis label</span></span>
<span id="cb485-25"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-25" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="fu">expression</span>(Log[<span class="dv">2</span>]<span class="sc">*</span><span class="st">&quot;(Concentration (pg/ml))&quot;</span>)) <span class="sc">+</span></span>
<span id="cb485-26"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-26" tabindex="-1"></a>  <span class="co"># Changing x axis label</span></span>
<span id="cb485-27"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-27" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Acrolein (ppm)&quot;</span>) <span class="sc">+</span></span>
<span id="cb485-28"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-28" tabindex="-1"></a>  <span class="co"># Faceting by each biomarker</span></span>
<span id="cb485-29"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-29" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span> variable, <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">scales =</span> <span class="st">&quot;free_y&quot;</span>, <span class="at">labeller =</span> <span class="fu">labeller</span>(<span class="at">variable =</span> new_labels)) <span class="sc">+</span></span>
<span id="cb485-30"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-30" tabindex="-1"></a>  <span class="co"># Removing legend</span></span>
<span id="cb485-31"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-31" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb485-32"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-32" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">12</span>),</span>
<span id="cb485-33"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-33" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.75</span>),</span>
<span id="cb485-34"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-34" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="dv">2</span>),</span>
<span id="cb485-35"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-35" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">size =</span> <span class="dv">10</span>),</span>
<span id="cb485-36"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#cb485-36" tabindex="-1"></a>        <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">12</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-364-1.png" alt="" width="800px" style="display: block; margin: auto;" /></p>
<p>An appropriate title for this figure could be:</p>
<p>“<strong>Figure X. Exposure to acrolein increases secretion of proinflammatory biomarkers in primary human bronchial epithelial cells.</strong> Groups were compared using the Friedman test to obtain overall p-value and Wilcoxon signed rank test for post-hoc testing. * p &lt; 0.05, ** p &lt; 0.01, *** p &lt; 0.001, **** p &lt; 0.0001 for comparison with control. For additional significant comparisons, see Supplemental Table X. n = 16 per group (paired).”</p>
</div>
<div id="answer-to-environmental-health-question-2-7" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 2<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#answer-to-environmental-health-question-2-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><i> With this, we can answer <strong>Environmental Health Question #2 </strong></i>: Do TNF-<span class="math inline">\(\alpha\)</span> concentrations significantly increase with increasing dose of acrolein?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: Yes, TNF-<span class="math inline">\(\alpha\)</span> concentrations significantly increase with increasing dose of acrolein, which we were able to visualize, along with other mediators, in our facet plot.</p>
</div>
<p><br></p>
</div>
</div>
<div id="concluding-remarks-16" class="section level2 hasAnchor">
<h2>Concluding Remarks<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#concluding-remarks-16" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this module, we introduced common multi-group statistical tests, including both overall effects tests and post-hoc testing. We applied these tests to our example dataset and demonstrated how to produce publication-quality tables and figures of our results. Implementing a workflow such as this enables efficient analysis of wet-bench generated data and customization of output figures and tables suited to your personal preferences.</p>
<div id="additional-resources-5" class="section level3 hasAnchor">
<h3>Additional Resources<a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#additional-resources-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><a href="http://www.sthda.com/english/articles/24-ggpubr-publication-ready-plots/76-add-p-values-and-significance-levels-to-ggplots/">STHDA: How to Add P-Values and Significance Levels to ggplots using <em>ggpubr</em></a></li>
<li><a href="https://cran.r-project.org/web/packages/ggprism/vignettes/pvalues.html">Adding p-values with <em>ggprism</em></a></li>
<li><a href="https://const-ae.github.io/ggsignif/">Overview of <em>ggsignif</em></a></li>
</ul>
<p><br></p>
<p><label class="tykfont">
Test Your Knowledge
</label></p>
<div class="tyk">
<p>Functional endpoints from these cultures were also measured. These endpoints were: 1) Membrane Permeability (MemPerm), 2) Trans-Epithelial Electrical Resistance (TEER), 3) Ciliary Beat Frequency (CBF), and 4) Expression of Mucin (MUC5AC). These data were already processed and tested for normality (see Test Your Knowledge for <strong>TAME 2.0 Module 4.2 Data Import, Processing, and Summary Statistics</strong>), with results indicating that two of the endpoints are normally distributed and two non-normally distributed.</p>
<p>Use the same processes demonstrated in this module and the provided data (“Module4_5_TYKInput.xlsx” (functional data)) to run analyses and make a publication-quality figure panel and table to answer the following question: Are there significant differences in functional endpoints between cells treated with different concentrations of acrolein?</p>
<p>For an extra challenge, try also making your faceted plot in the style of option #1 above, with different symbols, letters, or group names above columns to indicate which group that column in significant in comparison with.</p>
</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="two-group-comparisons-and-visualizations.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="advanced-multi-group-comparisons.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
  "sharing": {
    "github": false,
    "facebook": true,
    "twitter": true,
    "linkedin": false,
    "weibo": false,
    "instapaper": false,
    "vk": false,
    "whatsapp": false,
    "all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
  },
  "fontsettings": {
    "theme": "white",
    "family": "sans",
    "size": 2
  },
  "edit": {
    "link": null,
    "text": null
  },
  "history": {
    "link": null,
    "text": null
  },
  "view": {
    "link": null,
    "text": null
  },
  "download": ["_main.pdf", "_main.epub"],
  "search": {
    "engine": "fuse",
    "options": null
  },
  "toc": {
    "collapse": "section"
  }
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
