<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6.2 -Omics and System Biology: Transcriptomic Applications | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="6.2 -Omics and System Biology: Transcriptomic Applications | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6.2 -Omics and System Biology: Transcriptomic Applications | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  <meta name="twitter:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  

<meta name="author" content="Rager Lab" />


<meta name="date" content="2024-10-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="Icons_Used_Throughout/Favicon.png" type="image/x-icon" />
<link rel="prev" href="descriptive-cohort-analyses.html"/>
<link rel="next" href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>
<link href="libs/table1-1.0/table1_defaults.css" rel="stylesheet" />
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-FGFKVX2X3G"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-FGFKVX2X3G');
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="Icons_Used_Throughout/TAME 2.0 Logo.png"><br></a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Chapter 1 Introductory <br>Data Science</b></span></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html"><i class="fa fa-check"></i>1.1 FAIR Data Management Practices</a>
<ul>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-training-module"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-fair"><i class="fa fa-check"></i>Introduction to FAIR</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#breaking-down-fair-letter-by-letter"><i class="fa fa-check"></i>Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#what-does-this-mean-for-you"><i class="fa fa-check"></i>What Does This Mean for You?</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#data-repositories-for-sharing-of-data"><i class="fa fa-check"></i>Data Repositories for Sharing of Data</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#recent-shifts-in-regulatory-policies-for-data-sharing"><i class="fa fa-check"></i>Recent Shifts in Regulatory Policies for Data Sharing</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#additional-training-resources-on-fair"><i class="fa fa-check"></i>Additional Training Resources on FAIR</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html"><i class="fa fa-check"></i>1.2 Data Sharing through Online Repositories</a>
<ul>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#an-overview-and-example-with-the-dataverse-repository"><i class="fa fa-check"></i>An Overview and Example with the Dataverse Repository</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#introduction-to-training-module-1"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#data-repositories"><i class="fa fa-check"></i>Data Repositories</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#the-dataverse-project"><i class="fa fa-check"></i>The Dataverse Project</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#what-is-a-dataverse"><i class="fa fa-check"></i>What is a Dataverse?</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#metadata"><i class="fa fa-check"></i>Metadata</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataverse"><i class="fa fa-check"></i>Creating a Dataverse</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataset"><i class="fa fa-check"></i>Creating a Dataset</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#concluding-remarks-1"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html"><i class="fa fa-check"></i>1.3 File Management using Github</a>
<ul>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#introduction-to-training-module-2"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-an-account"><i class="fa fa-check"></i>Creating an Account</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-a-repository"><i class="fa fa-check"></i>Creating a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#uploading-code"><i class="fa fa-check"></i>Uploading Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#adding-subfolders-in-a-repository"><i class="fa fa-check"></i>Adding Subfolders in a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-code"><i class="fa fa-check"></i>Updating Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-repository-titles-and-structure-to-support-a-manuscript"><i class="fa fa-check"></i>Updating Repository Titles and Structure to Support a Manuscript</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#tracking-code-changes-using-github-branches"><i class="fa fa-check"></i>Tracking Code Changes using Github Branches</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#concluding-remarks-2"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html"><i class="fa fa-check"></i>1.4 Data Wrangling in Excel</a>
<ul>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#introduction-to-training-module-3"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#save-a-copy-of-the-soon-to-be-organized-and-cleaned-dataset-as-a-new-file"><i class="fa fa-check"></i>Save a Copy of the Soon-To-Be Organized and Cleaned Dataset as a New File</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-extraneous-white-space"><i class="fa fa-check"></i>Remove Extraneous White Space</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#replace-missing-data-with-na"><i class="fa fa-check"></i>Replace Missing Data with “NA”</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-metadata-tab"><i class="fa fa-check"></i>Create a Metadata Tab</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#abbreviate-and-capitalize-categorical-data"><i class="fa fa-check"></i>Abbreviate and Capitalize Categorical Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#alphabetize-sort-the-data-by-the-categorical-variable-of-interest"><i class="fa fa-check"></i>Alphabetize (Sort) the Data by the Categorical Variable of Interest</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-new-subject-number-column"><i class="fa fa-check"></i>Create a New Subject Number Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-special-symbols-and-dashes"><i class="fa fa-check"></i>Remove Special Symbols and Dashes</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#bold-all-column-names-and-center-all-data"><i class="fa fa-check"></i>Bold all Column Names and Center all Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-subject-identifier-column"><i class="fa fa-check"></i>Create a Subject Identifier Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#separate-subject-demographic-data-from-experimental-measurements"><i class="fa fa-check"></i>Separate Subject Demographic Data from Experimental Measurements</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#convert-data-from-wide-to-long-format"><i class="fa fa-check"></i>Convert Data from Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#pivoting-data-from-a-wide-to-long-format"><i class="fa fa-check"></i>Pivoting Data from a Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#generating-summary-level-statistics-with-pivot-tables"><i class="fa fa-check"></i>Generating Summary-Level Statistics with Pivot Tables</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#excel-vs.-r-which-should-you-use"><i class="fa fa-check"></i>Excel vs. R: Which Should You Use?</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#concluding-remarks-3"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 2 Coding in R</b></span></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html"><i class="fa fa-check"></i>2.1 Downloading and Programming in R</a>
<ul>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-training-module-4"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#general-introduction-and-installation-of-r-and-rstudio"><i class="fa fa-check"></i>General Introduction and Installation of R and RStudio</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-r-packages"><i class="fa fa-check"></i>Introduction to R Packages</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#scripting-basics"><i class="fa fa-check"></i>Scripting Basics</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#code-troubleshooting"><i class="fa fa-check"></i>Code Troubleshooting</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#concluding-remarks-4"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html"><i class="fa fa-check"></i>2.2 Coding “Best” Practices</a>
<ul>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#introduction-to-training-module-5"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#scripting-file-types"><i class="fa fa-check"></i>Scripting File Types</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-headers-and-annotation"><i class="fa fa-check"></i>Script Headers and Annotation</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#coding-style"><i class="fa fa-check"></i>Coding Style</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-organization"><i class="fa fa-check"></i>Script Organization</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#concluding-remarks-5"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html"><i class="fa fa-check"></i>2.3 Data Manipulation and Reshaping</a>
<ul>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-training-module-6"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#data-manipulation-using-base-r"><i class="fa fa-check"></i>Data Manipulation Using Base R</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-tidyverse"><i class="fa fa-check"></i>Introduction to Tidyverse</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#concluding-remarks-6"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html"><i class="fa fa-check"></i>2.4 Improving Coding Efficiencies</a>
<ul>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#introduction-to-training-module-7"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#loops"><i class="fa fa-check"></i>Loops</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#functions"><i class="fa fa-check"></i>Functions</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#list-operations"><i class="fa fa-check"></i>List operations</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#concluding-remarks-7"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#additional-resources-2"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="part"><span><b>Chapter 3 Basics of <br>Data Analysis and Visualizations</b></span></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html"><i class="fa fa-check"></i>3.1 Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-data-visualizations"><i class="fa fa-check"></i>Introduction to Data Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-training-module-8"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#density-plot-visualization"><i class="fa fa-check"></i>Density Plot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#boxplot-visualization"><i class="fa fa-check"></i>Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#correlation-visualizations"><i class="fa fa-check"></i>Correlation Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#heatmap-visualization"><i class="fa fa-check"></i>Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#concluding-remarks-8"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html"><i class="fa fa-check"></i>3.2 Improving Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-data-visulization-conventions"><i class="fa fa-check"></i>Introduction to Data Visulization Conventions</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-training-module-9"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-boxplot-visualization"><i class="fa fa-check"></i>Creating an Improved Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-heatmap-visualization"><i class="fa fa-check"></i>Creating an Improved Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-multi-plot-figures"><i class="fa fa-check"></i>Creating Multi-Plot Figures</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#concluding-remarks-9"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html"><i class="fa fa-check"></i>3.3 Normality Tests and Data Transformations</a>
<ul>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#introduction-to-training-module-10"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#what-is-a-normal-distribution"><i class="fa fa-check"></i>What is a Normal Distribution?</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#qualitative-assessment-of-normality"><i class="fa fa-check"></i>Qualitative Assessment of Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#quantitative-normality-assessment"><i class="fa fa-check"></i>Quantitative Normality Assessment</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#data-transformation"><i class="fa fa-check"></i>Data Transformation</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#additional-considerations-regarding-normality"><i class="fa fa-check"></i>Additional Considerations Regarding Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#concluding-remarks-10"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html"><i class="fa fa-check"></i>3.4 Intoduction to Statistical Tests</a>
<ul>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#introduction-to-training-module-11"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#assessing-normality-homogeneity-of-variance"><i class="fa fa-check"></i>Assessing Normality &amp; Homogeneity of Variance</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#two-group-visualizations-and-statistical-comparisons-using-the-t-test"><i class="fa fa-check"></i>Two-Group Visualizations and Statistical Comparisons using the T-Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#three-group-visualizations-and-statistical-comparisons-using-an-anova"><i class="fa fa-check"></i>Three-Group Visualizations and Statistical Comparisons using an ANOVA</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#regression-modeling-and-visualization-linear-and-logistic-regressions"><i class="fa fa-check"></i>Regression Modeling and Visualization: Linear and Logistic Regressions</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#statistical-evaluations-of-categorical-data-using-the-chi-squared-test-and-fishers-exact-test"><i class="fa fa-check"></i>Statistical Evaluations of Categorical Data using the Chi-Squared Test and Fisher’s Exact Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#concluding-remarks-11"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 4 Converting Wet Lab Data into Dry Lab Analyses</b></span></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html"><i class="fa fa-check"></i>4.1 Overview of Experimental Design and Example Data</a>
<ul>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#introduction-to-training-module-12"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#replicates"><i class="fa fa-check"></i>Replicates</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#orientation-to-example-data-for-chapter-4"><i class="fa fa-check"></i>Orientation to Example Data for Chapter 4</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#concluding-remarks-12"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html"><i class="fa fa-check"></i>4.2 Data Import, Processing, and Summary Statistics</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#introduction-to-training-module-13"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#data-import"><i class="fa fa-check"></i>Data Import</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#handling-missing-values"><i class="fa fa-check"></i>Handling Missing Values</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#averaging-replicates"><i class="fa fa-check"></i>Averaging Replicates</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#descriptive-statistics"><i class="fa fa-check"></i>Descriptive Statistics</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#normality-assessment-and-data-transformation"><i class="fa fa-check"></i>Normality Assessment and Data Transformation</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#concluding-remarks-13"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html"><i class="fa fa-check"></i>4.3 Data Import from PDF Sources</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#introduction-to-training-module-14"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-from-many-single-pdfs-with-the-same-formatting"><i class="fa fa-check"></i>Importing Data from Many Single PDFs with the Same Formatting</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-stored-in-pdf-tables"><i class="fa fa-check"></i>Importing Data Stored in PDF Tables</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#concluding-remarks-14"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html"><i class="fa fa-check"></i>4.4 Two Group Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#introduction-to-training-module-15"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#overview-of-two-group-statistical-tests"><i class="fa fa-check"></i>Overview of Two Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#statistical-vs.-biological-significance"><i class="fa fa-check"></i>Statistical vs. Biological Significance</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#unpaired-test-example"><i class="fa fa-check"></i>Unpaired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#paired-test-example"><i class="fa fa-check"></i>Paired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#visualizing-results"><i class="fa fa-check"></i>Visualizing Results</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#concluding-remarks-15"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html"><i class="fa fa-check"></i>4.5 Multi-Group and Multi-Variable Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#introduction-to-training-module-16"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#overview-of-multi-group-statistical-tests"><i class="fa fa-check"></i>Overview of Multi-Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#multi-group-analysis-example"><i class="fa fa-check"></i>Multi-Group Analysis Example</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#visualization-of-multi-group-statistical-results"><i class="fa fa-check"></i>Visualization of Multi-Group Statistical Results</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#concluding-remarks-16"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html"><i class="fa fa-check"></i>4.6 Advanced Multi-Group Comparisons</a>
<ul>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#introduction-to-training-module-17"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#advanced-multi-group-comparisons-1"><i class="fa fa-check"></i>Advanced Multi-Group Comparisons</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#two-way-anova-example"><i class="fa fa-check"></i>Two-way ANOVA Example</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#ancova-example"><i class="fa fa-check"></i>ANCOVA Example</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#concluding-remarks-17"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 5 Machine Learning &amp; Artificial Intelligence</b></span></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><i class="fa fa-check"></i>5.1 Introduction to Artificial Intelligence, Machine Learning, and Predictive Modeling for Environmental Health</a>
<ul>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#introduction-to-training-module-18"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#general-historical-context-and-taxonomy-of-modern-aiml"><i class="fa fa-check"></i>General Historical Context and Taxonomy of Modern AI/ML</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#application-of-machine-learning-in-environmental-health-science"><i class="fa fa-check"></i>Application of Machine Learning in Environmental Health Science</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#concluding-remarks-18"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html"><i class="fa fa-check"></i>5.2 Supervised Machine Learning</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-training-module-19"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-machine-learning"><i class="fa fa-check"></i>Types of Machine Learning</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-supervised-machine-learning-algorithms"><i class="fa fa-check"></i>Types of Supervised Machine Learning Algorithms</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#training-supervised-machine-learning-models"><i class="fa fa-check"></i>Training Supervised Machine Learning Models</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#assessing-classification-based-model-performance"><i class="fa fa-check"></i>Assessing Classification-Based Model Performance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-activity-and-example-dataset"><i class="fa fa-check"></i>Introduction to Activity and Example Dataset</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#testing-for-differences-in-predictor-variables-across-the-outcome-classes"><i class="fa fa-check"></i>Testing for Differences in Predictor Variables across the Outcome Classes</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#predicting-ias-detection-with-a-random-forest-rf-model"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#class-imbalance"><i class="fa fa-check"></i>Class Imbalance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#concluding-remarks-19"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html"><i class="fa fa-check"></i>5.3 Supervised Machine Learning Model Interpretation</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-training-module-20"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance"><i class="fa fa-check"></i>Variable Importance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary"><i class="fa fa-check"></i>Decision Boundary</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-example-dataset-and-activity"><i class="fa fa-check"></i>Introduction to Example Dataset and Activity</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#predicting-ias-detection-with-a-random-forest-rf-model-1"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance-plot"><i class="fa fa-check"></i>Variable Importance Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot"><i class="fa fa-check"></i>Decision Boundary Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot-incorporating-smote"><i class="fa fa-check"></i>Decision Boundary Plot Incorporating SMOTE</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#concluding-remarks-20"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><i class="fa fa-check"></i>5.4 Unsupervised Machine Learning Part 1: K-Means Clustering &amp; PCA</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-training-module-21"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-unsupervised-machine-learning"><i class="fa fa-check"></i>Introduction to Unsupervised Machine Learning</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-example-data"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#identifying-clusters-of-chemicals-through-k-means"><i class="fa fa-check"></i>Identifying Clusters of Chemicals through <em>K</em>-Means</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#principal-component-analysis-pca-1"><i class="fa fa-check"></i>Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#incorporating-k-means-into-pca-for-predictive-modeling"><i class="fa fa-check"></i>Incorporating <em>K</em>-Means into PCA for Predictive Modeling</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#concluding-remarks-21"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><i class="fa fa-check"></i>5.5 Unsupervised Machine Learning Part 2: Additional Clustering Applications</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-training-module-22"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#k-means-clustering-1"><i class="fa fa-check"></i><em>K</em>-Means Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-example-data-1"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-1"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#clustering-plot"><i class="fa fa-check"></i>Clustering Plot</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-visualization"><i class="fa fa-check"></i>Hierarchical Clustering Visualization</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#variable-contributions"><i class="fa fa-check"></i>Variable Contributions</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#concluding-remarks-22"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 6 Applications in Toxicology &amp; Exposure Science</b></span></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html"><i class="fa fa-check"></i>6.1 Descriptive Cohort Analyses</a>
<ul>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#introduction-to-training-module-23"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#participant-exploration"><i class="fa fa-check"></i>Participant Exploration</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#chemical-detection-filtering"><i class="fa fa-check"></i>Chemical Detection Filtering</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#outlier-identification"><i class="fa fa-check"></i>Outlier Identification</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#summary-statistics-tables"><i class="fa fa-check"></i>Summary Statistics Tables</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#demographics-table"><i class="fa fa-check"></i>Demographics Table</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#concluding-remarks-23"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html"><i class="fa fa-check"></i>6.2 -Omics and System Biology: Transcriptomic Applications</a>
<ul>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-training-module-24"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-the-field-of--omics"><i class="fa fa-check"></i>Introduction to the Field of “-Omics”</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-transcriptomics"><i class="fa fa-check"></i>Introduction to Transcriptomics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#formatting-data-for-downstream-statistics"><i class="fa fa-check"></i>Formatting Data for Downstream Statistics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#transcriptomics-data-qaqc"><i class="fa fa-check"></i>Transcriptomics Data QA/QC</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#controlling-for-sources-of-sample-heterogeneity"><i class="fa fa-check"></i>Controlling for Sources of Sample Heterogeneity</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e.g.-biomass-smoke-exposure"><i class="fa fa-check"></i>Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure)</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-ma-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using MA Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-volcano-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using Volcano Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis"><i class="fa fa-check"></i>Interpretting Findings at the Systems Level through Pathway Enrichment Analysis</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#concluding-remarks-24"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><i class="fa fa-check"></i>6.3 Mixtures Analysis Methods Part 1: Overview and Example with Quantile G-Computation</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-training-module-25"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#overview-of-mixtures-analysis"><i class="fa fa-check"></i>Overview of Mixtures Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-example-data-2"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#mixtures-model-with-standard-qqcomp"><i class="fa fa-check"></i>Mixtures Model with Standard qqcomp</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#concluding-remarks-25"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#additional-resources-11"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><i class="fa fa-check"></i>6.4 Mixtures Analysis Methods Part 2: Bayesian Kernel Machine Regression</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-training-module-26"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-example-data-3"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-bkmr"><i class="fa fa-check"></i>Introduction to BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#run-bkmr"><i class="fa fa-check"></i>Run BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#concluding-remarks-26"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#additional-resources-12"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html"><i class="fa fa-check"></i>6.5 Mixtures Analysis Methods Part 3: Sufficient Similarity</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-training-module-27"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-toxicant-and-dataset"><i class="fa fa-check"></i>Introduction to Toxicant and Dataset</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#workspace-preparation-and-data-import-9"><i class="fa fa-check"></i>Workspace Preparation and Data Import</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#chemistry-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Chemistry-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#toxicity-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Toxicity-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#comparing-chemistry-vs.-toxicity-sufficient-similarity-analyses"><i class="fa fa-check"></i>Comparing Chemistry vs. Toxicity Sufficient Similarity Analyses</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#concluding-remarks-27"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i>6.6 Toxicokinetic Modeling</a>
<ul>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-training-module-28"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-toxicokinetic-modeling"><i class="fa fa-check"></i>Introduction to Toxicokinetic Modeling</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#data-and-models-used-in-toxicokinetic-modeling-tk"><i class="fa fa-check"></i>Data and Models used in Toxicokinetic Modeling (TK)</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-toxicokinetics"><i class="fa fa-check"></i>Reverse Toxicokinetics</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#capturing-population-variability-in-toxicokinetics-and-uncertainty-in-chemical-specific-parameters"><i class="fa fa-check"></i>Capturing Population Variability in Toxicokinetics, and Uncertainty in Chemical-Specific Parameters</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-tk-calculating-administered-equivalent-doses-for-toxcast-bioactive-concentrations"><i class="fa fa-check"></i>Reverse TK: Calculating Administered Equivalent Doses for ToxCast Bioactive Concentrations</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#comparing-equivalent-doses-estimated-to-elicit-toxicity-hazard-to-external-exposure-estimates-exposure-for-chemical-prioritization-by-bioactivity-exposure-ratios-bers"><i class="fa fa-check"></i>Comparing Equivalent Doses Estimated to Elicit Toxicity (Hazard) to External Exposure Estimates (Exposure), for Chemical Prioritization by Bioactivity-Exposure Ratios (BERs)</a></li>
</ul></li>
<li class="part"><span><b>Chapter 7 Environmental Health Database Mining</b></span></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html"><i class="fa fa-check"></i>7.1 Comparative Toxicogenomics Database</a>
<ul>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#introduction-to-training-module-30"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#ctd-data-in-r"><i class="fa fa-check"></i>CTD Data in R</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#identifying-genes-under-epigenetic-control"><i class="fa fa-check"></i>Identifying Genes Under Epigenetic Control</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#concluding-remarks-30"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html"><i class="fa fa-check"></i>7.2 Gene Expression Omnibus</a>
<ul>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-training-module-31"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-geo"><i class="fa fa-check"></i>Introduction to GEO</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-example-data-4"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#geo-data-in-r"><i class="fa fa-check"></i>GEO Data in R</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#visualizing-data"><i class="fa fa-check"></i>Visualizing Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#statistical-analyses"><i class="fa fa-check"></i>Statistical Analyses</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#concluding-remarks-31"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html"><i class="fa fa-check"></i>7.3 CompTox Dashboard Data through APIs</a>
<ul>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#introduction-to-training-module-32"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#training-modules-environmental-health-questions-24"><i class="fa fa-check"></i>Training Module’s Environmental Health Questions</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#script-preparations-12"><i class="fa fa-check"></i>Script Preparations</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#introduction-to-comptox-chemicals-dashboard"><i class="fa fa-check"></i>Introduction to CompTox Chemicals Dashboard</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#cctes-ctx-application-programming-interfaces-apis-for-automated-batch-search-of-the-ccd"><i class="fa fa-check"></i>CCTE’s CTX Application Programming Interfaces (APIs) for Automated Batch Search of the CCD</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#comparing-physico-chemical-properties-between-two-important-environmental-contaminant-lists"><i class="fa fa-check"></i>Comparing Physico-chemical Properties between Two Important Environmental Contaminant Lists</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#hazard-data-genotoxicity"><i class="fa fa-check"></i>Hazard Data: Genotoxicity</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#hazard-resource"><i class="fa fa-check"></i>Hazard Resource</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#concluding-remarks-32"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html"><i class="fa fa-check"></i>7.4 Database Integration: Air Quality, Mortality, and Environmental Justice Data</a>
<ul>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#introduction-to-training-module-33"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#introduction-to-exposure-and-health-databases"><i class="fa fa-check"></i>Introduction to Exposure and Health Databases</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#analyzing-relationships-between-pm2.5-and-mortality"><i class="fa fa-check"></i>Analyzing Relationships between PM<sub>2.5</sub> and Mortality</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#environmental-justice-considerations"><i class="fa fa-check"></i>Environmental Justice Considerations</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#concluding-remarks-33"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="omics-and-system-biology-transcriptomic-applications" class="section level1 hasAnchor">
<h1>6.2 -Omics and System Biology: Transcriptomic Applications<a href="omics-and-system-biology-transcriptomic-applications.html#omics-and-system-biology-transcriptomic-applications" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This training module was developed by Lauren Koval, Dr. Kyle Roell, and Dr. Julia E. Rager.</p>
<p>All input files (script, data, and figures) can be downloaded from the <a href="https://github.com/UNCSRP/TAME2">UNC-SRP TAME2 GitHub website</a>.</p>
<div id="introduction-to-training-module-24" class="section level2 hasAnchor">
<h2>Introduction to Training Module<a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-training-module-24" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This training module incorporates the highly relevant example of RNA sequencing to evaluate the impacts of environmental exposures on cellular responses and general human health. <strong>RNA sequencing</strong> is the most common method that is currently implemented to measure the transcriptome. Results from an RNA sequencing platform are often summarized as count data, representing the number of relative times a gene (or other annotated portion of the genome) was ‘read’ in a given sample. For more details surrounding the methodological underpinnings of RNA sequencing, see the following recent review:</p>
<ul>
<li>Stark R, Grzelak M, Hadfield J. RNA sequencing: the teenage years. Nat Rev Genet. 2019 Nov;20(11):631-656. doi: 10.1038/s41576-019-0150-2. Epub 2019 Jul 24. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/31341269/">31341269</a>.</li>
</ul>
<p>In this training module, we guide participants through an example RNA sequencing analysis. Here, we analyze RNA sequencing data collected in a toxicology study evaluating the effects of biomass smoke exposure, representing wildfire-relevant exposure conditions. This study has been previously been described in the following publications:</p>
<ul>
<li><p>Rager JE, Clark J, Eaves LA, Avula V, Niehoff NM, Kim YH, Jaspers I, Gilmour MI. Mixtures modeling identifies chemical inducers versus repressors of toxicity associated with wildfire smoke. Sci Total Environ. 2021 Jun 25;775:145759. doi: 10.1016/j.scitotenv.2021.145759. Epub 2021 Feb 10. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/33611182/">33611182</a>.</p></li>
<li><p>Kim YH, Warren SH, Krantz QT, King C, Jaskot R, Preston WT, George BJ, Hays MD, Landis MS, Higuchi M, DeMarini DM, Gilmour MI. Mutagenicity and Lung Toxicity of Smoldering vs. Flaming Emissions from Various Biomass Fuels: Implications for Health Effects from Wildland Fires. Environ Health Perspect. 2018 Jan 24;126(1):017011. doi: 10.1289/EHP2200. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/29373863/">29373863</a>.</p></li>
</ul>
<p>Here, we specifically analyze mRNA sequencing profiles collected in mouse lung tissues. These mice were exposed to two different biomass burn scenarios: smoldering pine needles and flaming pine needles, representing certain wildfire smoke exposure scenarios that can occur. The goal of these analyses is to identify which genes demonstrate altered expression in response to these wildfire-relevant exposures, and identify which biological pathways these genes influence to evaluate findings at the systems biology level.</p>
<p>This training module begins by guiding users through the loading, viewing, and formatting of the example transcriptomics datasets and associated metadata. Methods to carry out quality assurance (QA) / quality control (QC) of the transcriptomics data are then described, which are advantageous to ensure high quality data are included in the final statistical analysis. Because these transcriptomic data were derived from bulk lung tissue samples, consisting of mixed cell populations that could have shifted in response to exposures, data are then adjusted for potential sources of heterogeneity using the R package <a href="https://bioconductor.org/packages/release/bioc/html/RUVSeq.html">RUVseq</a>.</p>
<p>Statistical models are then implemented to identify genes that were significantly differentially expressed between exposed vs unexposed samples. Models are implemented using algorithms within the commonly implemented R package <a href="https://doi.org/10.1186/s13059-014-0550-8">DESeq2</a>. This package is very convenient, well written, and widely used. The main advantage of this package is that is allows you to perform differential expression analyses and easily obtain various statistics and results with minimal script development on the user-end.</p>
<p>After obtaining results from differential gene expression analyses, we visualize these results using both MA and volcano plots. Finally, we carry out a systems level analysis through pathway enrichment using the R package <a href="https://doi.org/10.1093/nar/gkt111">PIANO</a> to identify which biological pathways were altered in response to these wildfire-relevant exposure scenarios.</p>
</div>
<div id="introduction-to-the-field-of--omics" class="section level2 hasAnchor">
<h2>Introduction to the Field of “-Omics”<a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-the-field-of--omics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The field of “-omics” has rapidly evolved since its inception in the mid-1990’s, initiated from information obtained through sequencing of the human genome (see the <a href="https://www.genome.gov/human-genome-project">Human Genome Project</a>) as well as the advent of high-content technologies. High-content technologies have allowed the rapid and economical assessment of genome-wide, or ‘omics’-based, endpoints.</p>
<p>Traditional molecular biology techniques typically evaluate the function(s) of individual genes and gene products. Omics-based methods, on the other hand, utilize non-targeted methods to identify many to all genes or gene products in a given environmental/biological sample. These non-targeted approaches allow for the unbiased investigation of potentially unknown or understudied molecular mediators involved in regulating cell health and disease. These molecular profiles have the potential of being altered in response to toxicant exposures and/or during disease initiation/progression.</p>
<p>To further understand the molecular consequences of -omics-based alterations, molecules can be overlaid onto molecular networks to uncover biological pathways and molecular functions that are perturbed at the systems biology level. An overview of these generally methods, starting with high-content technologies and ending of systems biology, is provided in the below figure (created with BioRender.com).</p>
<p><img src="Module6_2_Input/Module6_2_Image1.png" width="1566" style="display: block; margin: auto;" /></p>
<p><br>
A helpful introduction to the field of -omics in relation to environmental health, as well as methods used to relate -omic-level alterations to systems biology, is provided in the following book chapter:</p>
<ul>
<li>Rager JE, Fry RC. Systems Biology and Environmental Exposures. Chpt 4 of ‘Network Biology’ edited by WenJun Zhang. 2013. ISBN: 978-1-62618-941-3. Nova Science Publishers, Inc. Available at: <a href="https://www.novapublishers.com/wp-content/uploads/2019/07/978-1-62618-942-3_ch4.pdf" class="uri">https://www.novapublishers.com/wp-content/uploads/2019/07/978-1-62618-942-3_ch4.pdf</a>.</li>
</ul>
<p>An additional helpful resource describing computational methods that can be used in systems level analyses is the following book chapter:</p>
<ul>
<li>Meisner M, Reif DM. Computational Methods Used in Systems Biology. Chpt 5 of ‘Systems Biology in Toxicology and Environmental Health’ edited by Fry RC. 2015: 85-115. ISBN 9780128015643. Academic Press. Available at: <a href="https://www.sciencedirect.com/science/article/pii/B9780128015643000055" class="uri">https://www.sciencedirect.com/science/article/pii/B9780128015643000055</a>.</li>
</ul>
<p>Parallel to human genomics/epigenomics-based research is the newer “-omics” topic of the <strong>exposome</strong>. The exposome was originally conceptualized as ’all life-course environmental exposures (including lifestyle factors), from the prenatal period onwards (<a href="https://cebp.aacrjournals.org/content/14/8/1847.long">Wild et al. 2005</a>). Since then, this concept has received much attention and additional associated definitions. We like to think of the exposome as including anything in ones environment that may impact the overall health of an individual, excluding the individual’s genome/epigenome. Common elements evaluated as part of the exposome include environmental exposures, such as chemicals and other substances that may impart toxicity. Additional potential stressors include lifestyle factors, socioeconomic factors, infectious agents, therapeutics, and other stressors that may be altered internally (e.g., microbiome). A helpful review of this research field is provided as the following publication:</p>
<ul>
<li>Wild CP. The exposome: from concept to utility. Int J Epidemiol. 2012 Feb;41(1):24-32. doi: 10.1093/ije/dyr236. Epub 2012 Jan 31. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/22296988/">22296988</a>.</li>
</ul>
</div>
<div id="introduction-to-transcriptomics" class="section level2 hasAnchor">
<h2>Introduction to Transcriptomics<a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-transcriptomics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>One of the most widely evaluated -omics endpoints is messenger RNA (mRNA) expression (also termed gene expression). As a reminder, mRNA molecules are a major type of RNA produced as the “middle step” in the <a href="https://en.wikipedia.org/wiki/Central_dogma_of_molecular_biology#:~:text=The%20central%20dogma%20of%20molecular,The%20Central%20Dogma">Central Dogma Theory</a>, which describes how genetic DNA is first transcribed into RNA and then translated into protein. Protein molecules are ultimately the major regulators of cellular processes and overall health. Therefore, any perturbations to this process (including changes to mRNA expression levels) can have tremendous consequences on overall cell function and health. A visualization of these steps in the Central Dogma theory are included below.</p>
<p><img src="Module6_2_Input/Module6_2_Image2.png" width="540" style="display: block; margin: auto;" /></p>
<p>mRNA expression can be evaluated in a high-throughout/high-content manner, across the genome, and is referred to as the <strong>transcriptome</strong> when doing so. Transcriptomics can be measured using a variety of technologies, including high-density nucleic acid arrays (e.g., DNA microarrays or GeneChip arrays), high-throughput PCR technologies, or RNA sequencing technologies. These methods are used to obtain relative measures of genes that are being expressed or transcribed from DNA by measuring the abundance of mRNA molecules. Results of these methods are often termed as providing gene expression signatures or ‘transcriptomes’ of a sample under evaluation.</p>
<div id="training-modules-environmental-health-questions-16" class="section level3 hasAnchor">
<h3>Training Module’s <strong>Environmental Health Questions</strong><a href="omics-and-system-biology-transcriptomic-applications.html#training-modules-environmental-health-questions-16" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This training module was specifically developed to answer the following environmental health questions:</p>
<ol style="list-style-type: decimal">
<li><p>What two types of data are commonly needed in the analysis of transcriptomics data?</p></li>
<li><p>When preparing transcriptomics data for statistical analyses, what are three common data filtering steps that are completed during the data QA/QC process?</p></li>
<li><p>When identifying potential sample outliers in a typical transcriptomics dataset, what two types of approaches are commonly employed to identify samples with outlying data distributions?</p></li>
<li><p>What is an approach that analysts can use when evaluating transcriptomic data from tissues of mixed cellular composition to aid in controlling for sources of sample heterogeneity?</p></li>
<li><p>How many genes showed significant differential expression associated with flaming pine needles exposure in the mouse lung, based on a statistical filter of a multiple test corrected p-value (padj) &lt; 0.05?</p></li>
<li><p>How many genes showed significant differential expression associated with smoldering pine needles exposure in the mouse lung, based on a statistical filter of a multiple test corrected p-value (padj) &lt; 0.05?</p></li>
<li><p>How many genes showed significant differential expression associated with lipopolysaccharide (LPS) exposure in the mouse lung, based on a statistical filter of a multiple test corrected p-value (padj) &lt; 0.05?</p></li>
<li><p>What biological pathways are disrupted in association with flaming pine needles exposure in the lung, identified through systems level analyses?</p></li>
</ol>
</div>
<div id="workspace-preparation-and-data-import-8" class="section level3 hasAnchor">
<h3>Workspace Preparation and Data Import<a href="omics-and-system-biology-transcriptomic-applications.html#workspace-preparation-and-data-import-8" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><br></p>
<div id="installing-required-r-packages-9" class="section level4 hasAnchor">
<h4>Installing required R packages<a href="omics-and-system-biology-transcriptomic-applications.html#installing-required-r-packages-9" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>If you already have these packages installed, you can skip this step, or you can run the below code which checks installation status for you</p>
<div class="sourceCode" id="cb646"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb646-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;tidyverse&quot;</span>))</span>
<span id="cb646-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-2" tabindex="-1"></a>   <span class="fu">install.packages</span>(<span class="st">&quot;tidyverse&quot;</span>);</span>
<span id="cb646-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-3" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>))</span>
<span id="cb646-4"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-4" tabindex="-1"></a>   BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;BiocManager&quot;</span>);</span>
<span id="cb646-5"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-5" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;DESeq2&quot;</span>))</span>
<span id="cb646-6"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-6" tabindex="-1"></a>   BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;DESeq2&quot;</span>);</span>
<span id="cb646-7"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-7" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;edgeR&quot;</span>))</span>
<span id="cb646-8"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-8" tabindex="-1"></a>   BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;edgeR&quot;</span>);</span>
<span id="cb646-9"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-9" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;RUVSeq&quot;</span>))</span>
<span id="cb646-10"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-10" tabindex="-1"></a>   BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;RUVSeq&quot;</span>);</span>
<span id="cb646-11"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-11" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;janitor&quot;</span>))</span>
<span id="cb646-12"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-12" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;janitor&quot;</span>);</span>
<span id="cb646-13"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-13" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;pheatmap&quot;</span>))</span>
<span id="cb646-14"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-14" tabindex="-1"></a>   <span class="fu">install.packages</span>(<span class="st">&quot;pheatmap&quot;</span>);</span>
<span id="cb646-15"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-15" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;factoextra&quot;</span>))</span>
<span id="cb646-16"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-16" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;factoextra&quot;</span>);</span>
<span id="cb646-17"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-17" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;RColorBrewer&quot;</span>))</span>
<span id="cb646-18"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-18" tabindex="-1"></a>     <span class="fu">install.packages</span>(<span class="st">&quot;RColorBrewer&quot;</span>);</span>
<span id="cb646-19"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-19" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;data.table&quot;</span>))</span>
<span id="cb646-20"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-20" tabindex="-1"></a>    <span class="fu">install.packages</span>(<span class="st">&quot;data.table&quot;</span>);</span>
<span id="cb646-21"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-21" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;EnhancedVolcano&quot;</span>))</span>
<span id="cb646-22"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-22" tabindex="-1"></a>   BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;EnhancedVolcano&quot;</span>);</span>
<span id="cb646-23"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-23" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;piano&quot;</span>))</span>
<span id="cb646-24"><a href="omics-and-system-biology-transcriptomic-applications.html#cb646-24" tabindex="-1"></a>   BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">&quot;piano&quot;</span>);</span></code></pre></div>
</div>
<div id="loading-r-packages-required-for-this-session-7" class="section level4 hasAnchor">
<h4>Loading R packages required for this session<a href="omics-and-system-biology-transcriptomic-applications.html#loading-r-packages-required-for-this-session-7" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb647"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb647-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb647-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb647-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb647-2" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb647-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb647-3" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb647-4"><a href="omics-and-system-biology-transcriptomic-applications.html#cb647-4" tabindex="-1"></a><span class="fu">library</span>(RUVSeq)</span>
<span id="cb647-5"><a href="omics-and-system-biology-transcriptomic-applications.html#cb647-5" tabindex="-1"></a><span class="fu">library</span>(janitor)</span>
<span id="cb647-6"><a href="omics-and-system-biology-transcriptomic-applications.html#cb647-6" tabindex="-1"></a><span class="fu">library</span>(factoextra)</span>
<span id="cb647-7"><a href="omics-and-system-biology-transcriptomic-applications.html#cb647-7" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb647-8"><a href="omics-and-system-biology-transcriptomic-applications.html#cb647-8" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb647-9"><a href="omics-and-system-biology-transcriptomic-applications.html#cb647-9" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb647-10"><a href="omics-and-system-biology-transcriptomic-applications.html#cb647-10" tabindex="-1"></a><span class="fu">library</span>(EnhancedVolcano)</span>
<span id="cb647-11"><a href="omics-and-system-biology-transcriptomic-applications.html#cb647-11" tabindex="-1"></a><span class="fu">library</span>(piano)</span></code></pre></div>
</div>
<div id="set-your-working-directory-11" class="section level4 hasAnchor">
<h4>Set your working directory<a href="omics-and-system-biology-transcriptomic-applications.html#set-your-working-directory-11" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb648"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb648-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb648-1" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/filepath to where your input files are&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="loading-the-example-transcriptomic-dataset-and-metadata" class="section level3 hasAnchor">
<h3>Loading the Example Transcriptomic Dataset and Metadata<a href="omics-and-system-biology-transcriptomic-applications.html#loading-the-example-transcriptomic-dataset-and-metadata" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First, let’s read in the transcriptional signature data, previously summarized as number of sequence reads per gene (also simply referred to as ‘count data’) and its associated metadata file:</p>
<div class="sourceCode" id="cb649"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb649-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb649-1" tabindex="-1"></a><span class="co"># Read in the count data</span></span>
<span id="cb649-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb649-2" tabindex="-1"></a>countdata <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">&#39;Module6_2_Input/Module6_2_InputData1_GeneCounts.csv&#39;</span>, <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb649-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb649-3" tabindex="-1"></a></span>
<span id="cb649-4"><a href="omics-and-system-biology-transcriptomic-applications.html#cb649-4" tabindex="-1"></a><span class="co"># Read in the metadata (describing information on each sample)</span></span>
<span id="cb649-5"><a href="omics-and-system-biology-transcriptomic-applications.html#cb649-5" tabindex="-1"></a>sampleinfo <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="at">file =</span> <span class="st">&quot;Module6_2_Input/Module6_2_InputData2_SampleInfo.csv&quot;</span>, <span class="at">check.names =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
</div>
<div id="data-viewing-1" class="section level3 hasAnchor">
<h3>Data Viewing<a href="omics-and-system-biology-transcriptomic-applications.html#data-viewing-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="lets-see-how-many-rows-and-columns-of-data-are-present-in-the-countdata-dataframe" class="section level4 hasAnchor">
<h4>Let’s see how many rows and columns of data are present in the countdata dataframe<a href="omics-and-system-biology-transcriptomic-applications.html#lets-see-how-many-rows-and-columns-of-data-are-present-in-the-countdata-dataframe" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb650"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb650-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb650-1" tabindex="-1"></a><span class="fu">dim</span>(countdata)</span></code></pre></div>
<pre><code>## [1] 30146    23</code></pre>
</div>
<div id="lets-also-view-the-column-headers" class="section level4 hasAnchor">
<h4>Let’s also view the column headers<a href="omics-and-system-biology-transcriptomic-applications.html#lets-also-view-the-column-headers" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb652"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb652-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb652-1" tabindex="-1"></a><span class="fu">colnames</span>(countdata)</span></code></pre></div>
<pre><code>##  [1] &quot;Gene&quot;           &quot;Plate1-m13-RNA&quot; &quot;Plate2-m14-RNA&quot; &quot;Plate2-m15-RNA&quot;
##  [5] &quot;Plate2-m16-RNA&quot; &quot;Plate2-m17-RNA&quot; &quot;Plate2-m18-RNA&quot; &quot;Plate1-m49-RNA&quot;
##  [9] &quot;Plate1-m50-RNA&quot; &quot;Plate1-m51-RNA&quot; &quot;Plate1-m52-RNA&quot; &quot;Plate1-m53-RNA&quot;
## [13] &quot;Plate1-m54-RNA&quot; &quot;Plate1-m31-RNA&quot; &quot;Plate1-m32-RNA&quot; &quot;Plate1-m33-RNA&quot;
## [17] &quot;Plate1-m34-RNA&quot; &quot;Plate1-m35-RNA&quot; &quot;Plate1-m36-RNA&quot; &quot;Plate1-m67-RNA&quot;
## [21] &quot;Plate1-m68-RNA&quot; &quot;Plate1-m69-RNA&quot; &quot;Plate1-m70-RNA&quot;</code></pre>
</div>
<div id="and-finally-lets-view-the-top-few-rows-of-data" class="section level4 hasAnchor">
<h4>And finally let’s view the top few rows of data<a href="omics-and-system-biology-transcriptomic-applications.html#and-finally-lets-view-the-top-few-rows-of-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb654"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb654-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb654-1" tabindex="-1"></a><span class="fu">head</span>(countdata)</span></code></pre></div>
<pre><code>##           Gene Plate1-m13-RNA Plate2-m14-RNA Plate2-m15-RNA Plate2-m16-RNA
## 1  Bcat1_29039              0              0              0              0
## 2 Laptm5_29040              0              2              0              1
## 3   Skp2_29041              0              0              0              0
## 4   Nme1_29042            765           1041           1028            856
## 5   Ctsd_29043             25             30             38             24
## 6  Camk1_29044            443            445            433            628
##   Plate2-m17-RNA Plate2-m18-RNA Plate1-m49-RNA Plate1-m50-RNA Plate1-m51-RNA
## 1              0              0              0              0              0
## 2              2              1              0              1              2
## 3              0              0              0              0              0
## 4            888            967            816            569            854
## 5             17             19             38             38             34
## 6            497            684            574             71            356
##   Plate1-m52-RNA Plate1-m53-RNA Plate1-m54-RNA Plate1-m31-RNA Plate1-m32-RNA
## 1              0              2              0              0              3
## 2              2              7              1              2              1
## 3              0              0              0              0              0
## 4            736            887            930            563            776
## 5             30             24             23             29             21
## 6            403            179            302            172            487
##   Plate1-m33-RNA Plate1-m34-RNA Plate1-m35-RNA Plate1-m36-RNA Plate1-m67-RNA
## 1              0              0              0              0              1
## 2              5              0              0              0              1
## 3              0              0              0              0              0
## 4            604           1008            724            632            950
## 5             27             26             18             22             33
## 6            487            429            599            683            147
##   Plate1-m68-RNA Plate1-m69-RNA Plate1-m70-RNA
## 1              0              0              0
## 2              3              2              2
## 3              0              0              0
## 4            782            879            860
## 5             19             12             14
## 6            265            335            454</code></pre>
<p>Together, this dataframe contains information across 30146 mRNA identifiers, that are labeled according to “Gene name” followed by an underscore and probe number assigned by the platform used in this analysis, BioSpyder TempoSeq Technologies.</p>
<p>A total of 23 columns are included in this dataframe, the first of which represents the gene identifier, followed by gene count data across 22 samples.</p>
</div>
<div id="lets-also-see-what-the-metadata-dataframe-looks-like" class="section level4 hasAnchor">
<h4>Let’s also see what the metadata dataframe looks like<a href="omics-and-system-biology-transcriptomic-applications.html#lets-also-see-what-the-metadata-dataframe-looks-like" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb656"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb656-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb656-1" tabindex="-1"></a><span class="fu">dim</span>(sampleinfo)</span></code></pre></div>
<pre><code>## [1] 22  9</code></pre>
</div>
<div id="lets-also-view-the-column-headers-1" class="section level4 hasAnchor">
<h4>Let’s also view the column headers<a href="omics-and-system-biology-transcriptomic-applications.html#lets-also-view-the-column-headers-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb658"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb658-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb658-1" tabindex="-1"></a><span class="fu">colnames</span>(sampleinfo)</span></code></pre></div>
<pre><code>## [1] &quot;SampleID_BioSpyderCountFile&quot; &quot;PlateBatch&quot;                 
## [3] &quot;MouseID&quot;                     &quot;NumericID&quot;                  
## [5] &quot;Treatment&quot;                   &quot;ID&quot;                         
## [7] &quot;Timepoint&quot;                   &quot;Tissue&quot;                     
## [9] &quot;Group&quot;</code></pre>
</div>
<div id="and-finally-lets-view-the-top-few-rows-of-data-1" class="section level4 hasAnchor">
<h4>And finally let’s view the top few rows of data<a href="omics-and-system-biology-transcriptomic-applications.html#and-finally-lets-view-the-top-few-rows-of-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb660"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb660-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb660-1" tabindex="-1"></a><span class="fu">head</span>(sampleinfo)</span></code></pre></div>
<pre><code>##   SampleID_BioSpyderCountFile PlateBatch MouseID NumericID          Treatment
## 1              Plate1-m13-RNA     Plate1     M13        13 PineNeedlesSmolder
## 2              Plate2-m14-RNA     Plate2     M14        14 PineNeedlesSmolder
## 3              Plate2-m15-RNA     Plate2     M15        15 PineNeedlesSmolder
## 4              Plate2-m16-RNA     Plate2     M16        16 PineNeedlesSmolder
## 5              Plate2-m17-RNA     Plate2     M17        17 PineNeedlesSmolder
## 6              Plate2-m18-RNA     Plate2     M18        18 PineNeedlesSmolder
##                       ID Timepoint Tissue                      Group
## 1 M13_PineNeedlesSmolder        4h   Lung PineNeedlesSmolder_4h_Lung
## 2 M14_PineNeedlesSmolder        4h   Lung PineNeedlesSmolder_4h_Lung
## 3 M15_PineNeedlesSmolder        4h   Lung PineNeedlesSmolder_4h_Lung
## 4 M16_PineNeedlesSmolder        4h   Lung PineNeedlesSmolder_4h_Lung
## 5 M17_PineNeedlesSmolder        4h   Lung PineNeedlesSmolder_4h_Lung
## 6 M18_PineNeedlesSmolder        4h   Lung PineNeedlesSmolder_4h_Lung</code></pre>
<p>Together, this dataframe contains information across the 22 total samples, that are labeled according to “SampleID_BioSpyderCountFile” header. These identifiers match those used as column headers in the countdata dataframe.</p>
<p>A total of 9 columns are included in this dataframe, including the following:</p>
<ul>
<li>‘SampleID_BioSpyderCountFile’: The unique sample identifers (total n=22)</li>
<li>‘PlateBatch’: The plate number that was used in the generation of these data.</li>
<li>‘MouseID’: The unique identifier, that starts with “M” followed by a number, for each mouse used in this study</li>
<li>‘NumericID’: The unique numeric identifier for each mouse.</li>
<li>‘Treatment’: The type of exposure condition that each mouse was administered. These include smoldering pine needles, flaming pine needles, vehicle control (saline), and positive inflammation control (LPS, or lipopolysaccharide)</li>
<li>‘ID’: Another form of identifier that combines the mouse identifier with the exposure condition</li>
<li>‘Timepoint’: The timepoint at which samples were collected (here, all 4h post-exposure)</li>
<li>‘Tissue’: The type of tissue that was collected and analyzed (here, all lung tissue)</li>
<li>‘Group’: The higher level identifier that groups samples together based on exposure condition, timepoint, and tissue</li>
</ul>
</div>
<div id="one-common-qcpreparation-step-that-is-helpful-when-organizing-transcriptomics-data-is-to-check-for-potential-duplicate-mrna-ids-in-the-countdata." class="section level4 hasAnchor">
<h4>One common QC/preparation step that is helpful when organizing transcriptomics data is to check for potential duplicate mRNA IDs in the countdata.<a href="omics-and-system-biology-transcriptomic-applications.html#one-common-qcpreparation-step-that-is-helpful-when-organizing-transcriptomics-data-is-to-check-for-potential-duplicate-mrna-ids-in-the-countdata." class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb662"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb662-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb662-1" tabindex="-1"></a><span class="co"># Visualize this data quickly by viewing top left corner, to check where ID column is located:</span></span>
<span id="cb662-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb662-2" tabindex="-1"></a>countdata[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##           Gene Plate1-m13-RNA Plate2-m14-RNA Plate2-m15-RNA Plate2-m16-RNA
## 1  Bcat1_29039              0              0              0              0
## 2 Laptm5_29040              0              2              0              1
## 3   Skp2_29041              0              0              0              0</code></pre>
<div class="sourceCode" id="cb664"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb664-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb664-1" tabindex="-1"></a><span class="co"># Then check for duplicates within column 1 (where the ID column is located):</span></span>
<span id="cb664-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb664-2" tabindex="-1"></a>Dups <span class="ot">&lt;-</span> <span class="fu">duplicated</span>(countdata[,<span class="dv">1</span>])</span>
<span id="cb664-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb664-3" tabindex="-1"></a><span class="fu">summary</span>(Dups)</span></code></pre></div>
<pre><code>##    Mode   FALSE 
## logical   30146</code></pre>
<p>In this case, because all potential duplicate checks turn up “FALSE”, these data do not contain duplicate mRNA identifiers in its current organized format.</p>
</div>
</div>
<div id="answer-to-environmental-health-question-1-13" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 1<a href="omics-and-system-biology-transcriptomic-applications.html#answer-to-environmental-health-question-1-13" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can now answer <strong>Environmental Health Question #1</strong></em>: What two types of data are commonly needed in the analysis of transcriptomics data?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> A file containing the raw -omics signatures are needed (in this case, the count data summarized per gene acquired from RNA sequencing technologies), and a file containing the associated metadata describing the actual samples, where they were derived from, what they represent, etc, is needed.</p>
</div>
</div>
</div>
<div id="formatting-data-for-downstream-statistics" class="section level2 hasAnchor">
<h2>Formatting Data for Downstream Statistics<a href="omics-and-system-biology-transcriptomic-applications.html#formatting-data-for-downstream-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Most of the statistical analyses included in this training module will be carried out using the DESeq2 pipeline. This package requires that the count data and sample information data be formatted in a certain manner, which will expedite the downstream coding needed to carry out the statistics. Here, we will walk users through these initial formatting steps.</p>
<p>DESeq2 first requires a ‘coldata’ dataframe, which includes the sample information (i.e., metadata). Let’s create this new dataframe based on the original ‘sampleinfo’ dataframe:</p>
<div class="sourceCode" id="cb666"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb666-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb666-1" tabindex="-1"></a>coldata <span class="ot">&lt;-</span> sampleinfo</span></code></pre></div>
<p>DESeq2 also requires a ‘countdata’ dataframe, which we’ve previously created; however, this dataframe requires some minor formatting before it can be used as input for downstream script.</p>
<p>First, the gene identifiers need to be converted into row names:</p>
<div class="sourceCode" id="cb667"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb667-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb667-1" tabindex="-1"></a>countdata <span class="ot">&lt;-</span> countdata <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="st">&quot;Gene&quot;</span>)</span></code></pre></div>
<p>Then, the column names need to be edited. Let’s remind ourselves what the column names are currently:</p>
<div class="sourceCode" id="cb668"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb668-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb668-1" tabindex="-1"></a><span class="fu">colnames</span>(countdata)</span></code></pre></div>
<pre><code>##  [1] &quot;Plate1-m13-RNA&quot; &quot;Plate2-m14-RNA&quot; &quot;Plate2-m15-RNA&quot; &quot;Plate2-m16-RNA&quot;
##  [5] &quot;Plate2-m17-RNA&quot; &quot;Plate2-m18-RNA&quot; &quot;Plate1-m49-RNA&quot; &quot;Plate1-m50-RNA&quot;
##  [9] &quot;Plate1-m51-RNA&quot; &quot;Plate1-m52-RNA&quot; &quot;Plate1-m53-RNA&quot; &quot;Plate1-m54-RNA&quot;
## [13] &quot;Plate1-m31-RNA&quot; &quot;Plate1-m32-RNA&quot; &quot;Plate1-m33-RNA&quot; &quot;Plate1-m34-RNA&quot;
## [17] &quot;Plate1-m35-RNA&quot; &quot;Plate1-m36-RNA&quot; &quot;Plate1-m67-RNA&quot; &quot;Plate1-m68-RNA&quot;
## [21] &quot;Plate1-m69-RNA&quot; &quot;Plate1-m70-RNA&quot;</code></pre>
<p>These column identifiers need to be converted into more intuitive sample IDs, that also indicate treatment. This information can be found in the coldata dataframe. Specifically, information in the column labeled ‘SampleID_BioSpyderCountFile’ will be helpful for these purposes.</p>
<p>To replace these original column identifiers with these more helpful sample identifiers, let’s first make sure the order of the countdata columns are in the same order as the coldata column of ‘SampleID_BioSpyderCountFile’:</p>
<div class="sourceCode" id="cb670"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb670-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb670-1" tabindex="-1"></a>countdata <span class="ot">&lt;-</span> <span class="fu">setcolorder</span>(countdata, <span class="fu">as.character</span>(coldata<span class="sc">$</span>SampleID_BioSpyderCountFile))</span></code></pre></div>
<p>Now, we can rename the column names within the countdata dataframe with these more helpful identifiers, since both dataframes are now arranged in the same order:</p>
<div class="sourceCode" id="cb671"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb671-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb671-1" tabindex="-1"></a><span class="fu">colnames</span>(countdata) <span class="ot">&lt;-</span> coldata<span class="sc">$</span>ID <span class="co"># Rename the countdata column names with the treatment IDs. </span></span>
<span id="cb671-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb671-2" tabindex="-1"></a><span class="fu">colnames</span>(countdata) <span class="co"># Viewing these new column names</span></span></code></pre></div>
<pre><code>##  [1] &quot;M13_PineNeedlesSmolder&quot; &quot;M14_PineNeedlesSmolder&quot; &quot;M15_PineNeedlesSmolder&quot;
##  [4] &quot;M16_PineNeedlesSmolder&quot; &quot;M17_PineNeedlesSmolder&quot; &quot;M18_PineNeedlesSmolder&quot;
##  [7] &quot;M49_PineNeedlesFlame&quot;   &quot;M50_PineNeedlesFlame&quot;   &quot;M51_PineNeedlesFlame&quot;  
## [10] &quot;M52_PineNeedlesFlame&quot;   &quot;M53_PineNeedlesFlame&quot;   &quot;M54_PineNeedlesFlame&quot;  
## [13] &quot;M31_Saline&quot;             &quot;M32_Saline&quot;             &quot;M33_Saline&quot;            
## [16] &quot;M34_Saline&quot;             &quot;M35_Saline&quot;             &quot;M36_Saline&quot;            
## [19] &quot;M67_LPS&quot;                &quot;M68_LPS&quot;                &quot;M69_LPS&quot;               
## [22] &quot;M70_LPS&quot;</code></pre>
<p>These new column identifiers look much better, and can better inform downstream statistical analysis script. Remember that these identifiers indicate that these are mouse samples (“M”), with unique numbers, followed by an underscore and the exposure condition.</p>
<p>When relabeling dataframes, it’s always important to triple check any of these major edits. For example, here, let’s double check that the same samples appear in the same order between the two working dataframes required for dowstream DESeq2 code:</p>
<div class="sourceCode" id="cb673"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb673-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb673-1" tabindex="-1"></a><span class="fu">setequal</span>(<span class="fu">as.character</span>(coldata<span class="sc">$</span>ID), <span class="fu">colnames</span>(countdata))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<div class="sourceCode" id="cb675"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb675-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb675-1" tabindex="-1"></a><span class="fu">identical</span>(<span class="fu">as.character</span>(coldata<span class="sc">$</span>ID), <span class="fu">colnames</span>(countdata))</span></code></pre></div>
<pre><code>## [1] TRUE</code></pre>
<p><br></p>
</div>
<div id="transcriptomics-data-qaqc" class="section level2 hasAnchor">
<h2>Transcriptomics Data QA/QC<a href="omics-and-system-biology-transcriptomic-applications.html#transcriptomics-data-qaqc" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>After preparing your transcriptomic data and sample information dataframes for statistical analyses, it is very important to carry out QA/QC on your organized datasets, prior to including all samples and all genes in the actual statistical model. It is critical to only include high quality data that inform underlying biology of exposure responses/disease etiology, rather than data that may contribute noise to the overall data distributions. Some common QA/QC steps and associated data pre-filters carried out in transcriptomics analyses are detailed below.</p>
<div id="background-filter" class="section level3 hasAnchor">
<h3>Background Filter<a href="omics-and-system-biology-transcriptomic-applications.html#background-filter" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>It is very common to perform a background filter step when preparing transcriptomic data for statistical analyses. The goal of this step is to remove genes that are very lowly expressed across the majority of samples, and thus are referred to as universally lowly expressed. Signals from these genes can mute the overall signals that may be identified in -omics analyses. The specific threshold that you may want to apply as the background filter to your dataset will depend on the distribution of your dataset and analysis goal(s).</p>
<p>For this example, we apply a background threshold, to remove genes that are lowly expressed across the majority of samples, specifically defined as genes that have expression levels across at least 20% of the samples that are less than (or equal to) the median expression of all genes across all samples. This will result in including only genes that are expressed above background, that have expression levels in at least 20% of samples that are greater than the overall median expression. Script to apply this filter is detailed below:</p>
<div class="sourceCode" id="cb677"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb677-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-1" tabindex="-1"></a><span class="co"># First count the total number of samples, and save it as a value in the global environment</span></span>
<span id="cb677-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-2" tabindex="-1"></a>nsamp <span class="ot">&lt;-</span> <span class="fu">ncol</span>(countdata)</span>
<span id="cb677-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-3" tabindex="-1"></a></span>
<span id="cb677-4"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-4" tabindex="-1"></a><span class="co"># Then, calculate the median expression level across all genes and all samples, and save it as a value</span></span>
<span id="cb677-5"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-5" tabindex="-1"></a>total_median <span class="ot">&lt;-</span> <span class="fu">median</span>(<span class="fu">as.matrix</span>(countdata))</span>
<span id="cb677-6"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-6" tabindex="-1"></a></span>
<span id="cb677-7"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-7" tabindex="-1"></a></span>
<span id="cb677-8"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-8" tabindex="-1"></a><span class="co"># We need to temporarily add back in the Gene column to the countdata so we can filter for genes that pass the background filter</span></span>
<span id="cb677-9"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-9" tabindex="-1"></a>countdata <span class="ot">&lt;-</span> countdata <span class="sc">%&gt;%</span> <span class="fu">rownames_to_column</span>(<span class="st">&quot;Gene&quot;</span>)</span>
<span id="cb677-10"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-10" tabindex="-1"></a></span>
<span id="cb677-11"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-11" tabindex="-1"></a><span class="co"># Then we can apply a set of filters and organization steps (using the tidyverse) to result in a list of genes that have an expression greater than the total median in at least 20% of the samples</span></span>
<span id="cb677-12"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-12" tabindex="-1"></a>genes_above_background <span class="ot">&lt;-</span> countdata <span class="sc">%&gt;%</span> <span class="co"># Start from the &#39;countdata&#39; dataframe</span></span>
<span id="cb677-13"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-13" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="sc">!</span>Gene, <span class="at">names_to =</span> <span class="st">&quot;sampleID&quot;</span>, <span class="at">values_to=</span><span class="st">&quot;expression&quot;</span>) <span class="sc">%&gt;%</span> <span class="co"># Melt the data so that we have three columns: gene, exposure condition, and expression counts</span></span>
<span id="cb677-14"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-14" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">above_median=</span><span class="fu">ifelse</span>(expression<span class="sc">&gt;</span>total_median,<span class="dv">1</span>,<span class="dv">0</span>)) <span class="sc">%&gt;%</span> <span class="co"># Add a column that indicates whether the expression of a gene for the corresponding exposure condition is above (1) or not above (0) the median of all count data</span></span>
<span id="cb677-15"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-15" tabindex="-1"></a>  <span class="fu">group_by</span>(Gene) <span class="sc">%&gt;%</span> <span class="co"># Group the dataframe by the gene</span></span>
<span id="cb677-16"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-16" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">total_above_median=</span><span class="fu">sum</span>(above_median)) <span class="sc">%&gt;%</span> <span class="co"># For each gene, count the number of exposure conditions where the expression was greater than the median of all count data</span></span>
<span id="cb677-17"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-17" tabindex="-1"></a>  <span class="fu">filter</span>(total_above_median<span class="sc">&gt;=</span>.<span class="dv">2</span><span class="sc">*</span>nsamp) <span class="sc">%&gt;%</span> <span class="co"># Filter for genes that have expression above the median in at least 20% of the samples</span></span>
<span id="cb677-18"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-18" tabindex="-1"></a>  <span class="fu">select</span>(Gene) <span class="co"># Select just the genes that pass the filter</span></span>
<span id="cb677-19"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-19" tabindex="-1"></a></span>
<span id="cb677-20"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-20" tabindex="-1"></a><span class="co"># Then filter the original &#39;countdata&#39; dataframe for only the genes above background. </span></span>
<span id="cb677-21"><a href="omics-and-system-biology-transcriptomic-applications.html#cb677-21" tabindex="-1"></a>countdata <span class="ot">&lt;-</span> <span class="fu">left_join</span>(genes_above_background, countdata, <span class="at">by=</span><span class="st">&quot;Gene&quot;</span>)</span></code></pre></div>
<p>Here, the ‘countdata’ dataframe went from having 30,146 rows of data (representing genes) to 16,664 rows of data (representing genes with expression levels that passed this background filter)</p>
</div>
<div id="sample-filtering" class="section level3 hasAnchor">
<h3>Sample Filtering<a href="omics-and-system-biology-transcriptomic-applications.html#sample-filtering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Another common QA/QC check is to evaluate whether there are any samples that did not produce adequate RNA material to be measured using the technology employed. Thus, a sample filter can be applied to remove samples that have inadequate data. Here, we demonstrate this filter by checking to see whether there were any samples that resulted in mRNA expression values of zero across all genes. If any sample demonstrates this issue, it should be removed prior to any statistical analysis. Note, there are other filter cut-offs you can use depending on your specific study.</p>
<p>Below is example script that checks for the presence of samples that meet the above criteria:</p>
<div class="sourceCode" id="cb678"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb678-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb678-1" tabindex="-1"></a><span class="co"># Transpose filtered &#39;countdata&#39;, while keeping data in dataframe format, to allow for script that easily sums the total expression levels per sample</span></span>
<span id="cb678-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb678-2" tabindex="-1"></a>countdata_T <span class="ot">&lt;-</span> countdata <span class="sc">%&gt;%</span> </span>
<span id="cb678-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb678-3" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="sc">!</span>Gene, <span class="at">names_to=</span><span class="st">&quot;sampleID&quot;</span>,<span class="at">values_to=</span><span class="st">&quot;expression&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb678-4"><a href="omics-and-system-biology-transcriptomic-applications.html#cb678-4" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from=</span>Gene, <span class="at">values_from=</span>expression)</span>
<span id="cb678-5"><a href="omics-and-system-biology-transcriptomic-applications.html#cb678-5" tabindex="-1"></a></span>
<span id="cb678-6"><a href="omics-and-system-biology-transcriptomic-applications.html#cb678-6" tabindex="-1"></a><span class="co"># Then add in a column to the transposed countdata dataframe that sums expression across all genes for each exposure condition</span></span>
<span id="cb678-7"><a href="omics-and-system-biology-transcriptomic-applications.html#cb678-7" tabindex="-1"></a>countdata_T<span class="sc">$</span>rowsum <span class="ot">&lt;-</span> <span class="fu">rowSums</span>(countdata_T[<span class="dv">2</span><span class="sc">:</span><span class="fu">ncol</span>(countdata_T)])</span>
<span id="cb678-8"><a href="omics-and-system-biology-transcriptomic-applications.html#cb678-8" tabindex="-1"></a></span>
<span id="cb678-9"><a href="omics-and-system-biology-transcriptomic-applications.html#cb678-9" tabindex="-1"></a><span class="co"># Remove samples that have no expression. All samples have some expression in this example, so all samples are retained.</span></span>
<span id="cb678-10"><a href="omics-and-system-biology-transcriptomic-applications.html#cb678-10" tabindex="-1"></a>countdata_T <span class="ot">&lt;-</span> countdata_T <span class="sc">%&gt;%</span> <span class="fu">filter</span>(rowsum<span class="sc">!=</span><span class="dv">0</span>)</span>
<span id="cb678-11"><a href="omics-and-system-biology-transcriptomic-applications.html#cb678-11" tabindex="-1"></a></span>
<span id="cb678-12"><a href="omics-and-system-biology-transcriptomic-applications.html#cb678-12" tabindex="-1"></a><span class="co"># Take the count data filtered for correct samples, remove the &#39;rowsums&#39; column</span></span>
<span id="cb678-13"><a href="omics-and-system-biology-transcriptomic-applications.html#cb678-13" tabindex="-1"></a>countdata_T <span class="ot">&lt;-</span> countdata_T <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">!</span>rowsum) </span>
<span id="cb678-14"><a href="omics-and-system-biology-transcriptomic-applications.html#cb678-14" tabindex="-1"></a></span>
<span id="cb678-15"><a href="omics-and-system-biology-transcriptomic-applications.html#cb678-15" tabindex="-1"></a><span class="co"># Then, transpose it back to the correct format for analysis</span></span>
<span id="cb678-16"><a href="omics-and-system-biology-transcriptomic-applications.html#cb678-16" tabindex="-1"></a>countdata <span class="ot">&lt;-</span> countdata_T <span class="sc">%&gt;%</span></span>
<span id="cb678-17"><a href="omics-and-system-biology-transcriptomic-applications.html#cb678-17" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols=</span><span class="sc">!</span>sampleID, <span class="at">names_to =</span> <span class="st">&quot;Gene&quot;</span>,<span class="at">values_to=</span><span class="st">&quot;expression&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb678-18"><a href="omics-and-system-biology-transcriptomic-applications.html#cb678-18" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> sampleID, <span class="at">values_from =</span> <span class="st">&quot;expression&quot;</span>)</span></code></pre></div>
</div>
<div id="identifying-removing-sample-outliers" class="section level3 hasAnchor">
<h3>Identifying &amp; Removing Sample Outliers<a href="omics-and-system-biology-transcriptomic-applications.html#identifying-removing-sample-outliers" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Prior to final statistical analysis, raw transcriptomic data are commonly evaluated for the presence of potential sample outliers. Outliers can result from experimental error, technical error/measurement error, and/or huge sources of variation in biology. For many analyses, it is beneficial to remove such outliers to enhance computational abilities to identify biologically meaningful signals across data. Here, we present two methods to check for the presence of sample outliers:</p>
<p><strong>1. Principal component analysis (PCA)</strong> can be used to identify potential outliers in a dataset through visualization of summary-level values illustrating reduced representations of the entire dataset. Note that a more detailed description of PCA is provided in <strong>Training Module 2.2</strong>. Here, PCA is run on the raw count data and further analyzed using scree plots, assessing principal components (PCs), and visualized using biplots displaying the first two principal components as a scatter plot.</p>
<p><strong>2. Hierarchical clustering</strong> is another approach that can be used to identify potential outliers. Hierarchical clustering aims to cluster data based on a similarity measure, defined by the function and/or specified by the user. There are several R packages and functions that will run hierarchical clustering, but it is often helpful visually to do this in conjuction with a heatmap. Here, we use the package pheatmap (introduced in <strong>Training Module 1.4</strong>) with hierarchical clustering across samples to identify potential outliers.</p>
<p>Let’s start by using PCA to identify potential outliers, while providing a visualization of potential sources of variation across the dataset.</p>
<div id="first-we-need-to-move-the-gene-column-back-to-the-rownames-so-our-dataframe-is-numeric-and-we-can-run-the-pca-script" class="section level4 hasAnchor">
<h4>First we need to move the Gene column back to the rownames so our dataframe is numeric and we can run the PCA script<a href="omics-and-system-biology-transcriptomic-applications.html#first-we-need-to-move-the-gene-column-back-to-the-rownames-so-our-dataframe-is-numeric-and-we-can-run-the-pca-script" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb679"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb679-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb679-1" tabindex="-1"></a>countdata <span class="ot">&lt;-</span> countdata <span class="sc">%&gt;%</span> <span class="fu">column_to_rownames</span>(<span class="st">&quot;Gene&quot;</span>)</span>
<span id="cb679-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb679-2" tabindex="-1"></a></span>
<span id="cb679-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb679-3" tabindex="-1"></a><span class="co"># Let&#39;s remind ourselves what these data look like</span></span>
<span id="cb679-4"><a href="omics-and-system-biology-transcriptomic-applications.html#cb679-4" tabindex="-1"></a>countdata[<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>] <span class="co">#viewing first 10 rows and 5 columns</span></span></code></pre></div>
<pre><code>##                     M13_PineNeedlesSmolder M14_PineNeedlesSmolder
## 0610009B22Rik_56046                    123                    118
## 0610010F05Rik_69119                   1378                   1511
## 0610010F05Rik_74637                     83                     55
## 0610010K14Rik_31619                    462                    580
## 0610010K14Rik_31873                    560                    814
## 0610010K14Rik_68949                    207                    182
## 0610012G03Rik_58446                    243                    299
## 0610030E20Rik_65200                    152                    105
## 0610040J01Rik_55628                    125                    106
## 1110004F10Rik_77756                    324                    377
##                     M15_PineNeedlesSmolder M16_PineNeedlesSmolder
## 0610009B22Rik_56046                    110                    167
## 0610010F05Rik_69119                   1534                    936
## 0610010F05Rik_74637                     57                     13
## 0610010K14Rik_31619                    625                    560
## 0610010K14Rik_31873                    680                    799
## 0610010K14Rik_68949                    104                    202
## 0610012G03Rik_58446                    294                    327
## 0610030E20Rik_65200                     92                     87
## 0610040J01Rik_55628                    127                    107
## 1110004F10Rik_77756                    356                    233
##                     M17_PineNeedlesSmolder
## 0610009B22Rik_56046                    147
## 0610010F05Rik_69119                   1301
## 0610010F05Rik_74637                     54
## 0610010K14Rik_31619                    604
## 0610010K14Rik_31873                    707
## 0610010K14Rik_68949                    166
## 0610012G03Rik_58446                    262
## 0610030E20Rik_65200                    150
## 0610040J01Rik_55628                     93
## 1110004F10Rik_77756                    276</code></pre>
</div>
<div id="then-we-can-calculate-principal-components-using-transposed-count-data" class="section level4 hasAnchor">
<h4>Then we can calculate principal components using transposed count data<a href="omics-and-system-biology-transcriptomic-applications.html#then-we-can-calculate-principal-components-using-transposed-count-data" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb681"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb681-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb681-1" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(countdata))</span></code></pre></div>
</div>
<div id="and-visualize-the-percent-variation-captured-by-each-principal-component-pc-with-a-scree-plot." class="section level4 hasAnchor">
<h4>And visualize the percent variation captured by each principal component (PC) with a scree plot.<a href="omics-and-system-biology-transcriptomic-applications.html#and-visualize-the-percent-variation-captured-by-each-principal-component-pc-with-a-scree-plot." class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb682"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb682-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb682-1" tabindex="-1"></a><span class="co"># We can generate a scree plot that shows the eigenvalues of each component, indicating how much of the total variation is captured by each component</span></span>
<span id="cb682-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb682-2" tabindex="-1"></a><span class="fu">fviz_eig</span>(pca)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-523-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>This scree plot indicates that nearly all variation is explained in PC1 and PC2, so we are comfortable with viewing these first two PCs when evaluating whether or not potential outliers exist in this dataset.</p>
</div>
<div id="further-visualization-of-how-these-transcriptomic-data-appear-through-pca-can-be-produced-through-a-scatter-plot-showing-the-data-reduced-values-per-sample" class="section level4 hasAnchor">
<h4>Further visualization of how these transcriptomic data appear through PCA can be produced through a scatter plot showing the data reduced values per sample:<a href="omics-and-system-biology-transcriptomic-applications.html#further-visualization-of-how-these-transcriptomic-data-appear-through-pca-can-be-produced-through-a-scatter-plot-showing-the-data-reduced-values-per-sample" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb683"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb683-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb683-1" tabindex="-1"></a><span class="co"># Calculate the percent variation captured by each PC</span></span>
<span id="cb683-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb683-2" tabindex="-1"></a>pca_percent <span class="ot">&lt;-</span> <span class="fu">round</span>(<span class="dv">100</span><span class="sc">*</span>pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span><span class="sc">/</span><span class="fu">sum</span>(pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span>),<span class="dv">1</span>)</span>
<span id="cb683-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb683-3" tabindex="-1"></a></span>
<span id="cb683-4"><a href="omics-and-system-biology-transcriptomic-applications.html#cb683-4" tabindex="-1"></a><span class="co"># Make dataframe for PCA plot generation using first two components and the sample name</span></span>
<span id="cb683-5"><a href="omics-and-system-biology-transcriptomic-applications.html#cb683-5" tabindex="-1"></a>pca_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">PC1 =</span> pca<span class="sc">$</span>x[,<span class="dv">1</span>], <span class="at">PC2 =</span> pca<span class="sc">$</span>x[,<span class="dv">2</span>], <span class="at">Sample=</span><span class="fu">colnames</span>(countdata))</span>
<span id="cb683-6"><a href="omics-and-system-biology-transcriptomic-applications.html#cb683-6" tabindex="-1"></a></span>
<span id="cb683-7"><a href="omics-and-system-biology-transcriptomic-applications.html#cb683-7" tabindex="-1"></a><span class="co"># Organize dataframe so we can color our points by the exposure condition</span></span>
<span id="cb683-8"><a href="omics-and-system-biology-transcriptomic-applications.html#cb683-8" tabindex="-1"></a>pca_df <span class="ot">&lt;-</span> pca_df <span class="sc">%&gt;%</span> <span class="fu">separate</span>(Sample, <span class="at">into =</span> <span class="fu">c</span>(<span class="st">&quot;mouse_num&quot;</span>, <span class="st">&quot;expo_cond&quot;</span>), <span class="at">sep=</span><span class="st">&quot;_&quot;</span>)</span>
<span id="cb683-9"><a href="omics-and-system-biology-transcriptomic-applications.html#cb683-9" tabindex="-1"></a></span>
<span id="cb683-10"><a href="omics-and-system-biology-transcriptomic-applications.html#cb683-10" tabindex="-1"></a><span class="co"># Plot PC1 and PC2 for each sample and color the point by the exposure condition</span></span>
<span id="cb683-11"><a href="omics-and-system-biology-transcriptomic-applications.html#cb683-11" tabindex="-1"></a><span class="fu">ggplot</span>(pca_df, <span class="fu">aes</span>(PC1,PC2, <span class="at">color =</span> expo_cond))<span class="sc">+</span></span>
<span id="cb683-12"><a href="omics-and-system-biology-transcriptomic-applications.html#cb683-12" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">size=</span><span class="fl">0.3</span>)<span class="sc">+</span></span>
<span id="cb683-13"><a href="omics-and-system-biology-transcriptomic-applications.html#cb683-13" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">size=</span><span class="fl">0.3</span>)<span class="sc">+</span></span>
<span id="cb683-14"><a href="omics-and-system-biology-transcriptomic-applications.html#cb683-14" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb683-15"><a href="omics-and-system-biology-transcriptomic-applications.html#cb683-15" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label=</span>mouse_num), <span class="at">vjust =</span><span class="sc">-</span><span class="dv">1</span>, <span class="at">size=</span><span class="dv">4</span>)<span class="sc">+</span></span>
<span id="cb683-16"><a href="omics-and-system-biology-transcriptomic-applications.html#cb683-16" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="fu">paste0</span>(<span class="st">&quot;PC1 (&quot;</span>,pca_percent[<span class="dv">1</span>],<span class="st">&quot;%)&quot;</span>), <span class="at">y=</span><span class="fu">paste0</span>(<span class="st">&quot;PC2 (&quot;</span>,pca_percent[<span class="dv">2</span>],<span class="st">&quot;%)&quot;</span>))<span class="sc">+</span></span>
<span id="cb683-17"><a href="omics-and-system-biology-transcriptomic-applications.html#cb683-17" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;PCA for 4h Lung Pine Needles &amp; Control Exposure Conditions&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
## ℹ Please use `linewidth` instead.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this
## warning was generated.</code></pre>
<p><img src="_main_files/figure-html/unnamed-chunk-524-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>With this plot, we can see that samples do not demonstrate obvious groupings, where certain samples group far apart from others. Therefore, our PCA analysis indicates that there are unlikely any sample outliers in this dataset.</p>
</div>
<div id="now-lets-implement-hierarchical-clustering-to-identify-potential-outliers" class="section level4 hasAnchor">
<h4>Now lets implement hierarchical clustering to identify potential outliers<a href="omics-and-system-biology-transcriptomic-applications.html#now-lets-implement-hierarchical-clustering-to-identify-potential-outliers" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>First we need to create a dataframe of our transposed ‘countdata’ such that samples are rows and genes are columns to input into the clustering algorithm.</p>
<div class="sourceCode" id="cb685"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb685-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb685-1" tabindex="-1"></a>countdata_for_clustering <span class="ot">&lt;-</span> <span class="fu">t</span>(countdata)</span>
<span id="cb685-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb685-2" tabindex="-1"></a>countdata_for_clustering[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>] <span class="co"># Viewing what this transposed dataframe looks like</span></span></code></pre></div>
<pre><code>##                        0610009B22Rik_56046 0610010F05Rik_69119
## M13_PineNeedlesSmolder                 123                1378
## M14_PineNeedlesSmolder                 118                1511
## M15_PineNeedlesSmolder                 110                1534
## M16_PineNeedlesSmolder                 167                 936
## M17_PineNeedlesSmolder                 147                1301
##                        0610010F05Rik_74637 0610010K14Rik_31619
## M13_PineNeedlesSmolder                  83                 462
## M14_PineNeedlesSmolder                  55                 580
## M15_PineNeedlesSmolder                  57                 625
## M16_PineNeedlesSmolder                  13                 560
## M17_PineNeedlesSmolder                  54                 604
##                        0610010K14Rik_31873 0610010K14Rik_68949
## M13_PineNeedlesSmolder                 560                 207
## M14_PineNeedlesSmolder                 814                 182
## M15_PineNeedlesSmolder                 680                 104
## M16_PineNeedlesSmolder                 799                 202
## M17_PineNeedlesSmolder                 707                 166
##                        0610012G03Rik_58446 0610030E20Rik_65200
## M13_PineNeedlesSmolder                 243                 152
## M14_PineNeedlesSmolder                 299                 105
## M15_PineNeedlesSmolder                 294                  92
## M16_PineNeedlesSmolder                 327                  87
## M17_PineNeedlesSmolder                 262                 150
##                        0610040J01Rik_55628 1110004F10Rik_77756
## M13_PineNeedlesSmolder                 125                 324
## M14_PineNeedlesSmolder                 106                 377
## M15_PineNeedlesSmolder                 127                 356
## M16_PineNeedlesSmolder                 107                 233
## M17_PineNeedlesSmolder                  93                 276</code></pre>
<p>Next we can run hierarchical clustering in conjunction with the generation of a heatmap. Note that we scale these data for improved visualization.</p>
<div class="sourceCode" id="cb687"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb687-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb687-1" tabindex="-1"></a><span class="fu">pheatmap</span>(<span class="fu">scale</span>(countdata_for_clustering), <span class="at">main=</span><span class="st">&quot;Hierarchical Clustering&quot;</span>,</span>
<span id="cb687-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb687-2" tabindex="-1"></a>         <span class="at">cluster_rows=</span><span class="cn">TRUE</span>, <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>,</span>
<span id="cb687-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb687-3" tabindex="-1"></a>         <span class="at">fontsize_col =</span> <span class="dv">7</span>, <span class="at">treeheight_row =</span> <span class="dv">60</span>, <span class="at">show_colnames =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-526-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Like the PCA findings, heirarchical clustering demonstrated an overall lack of potential sample outliers because there were no obvious sample(s) that grouped separately from the rest along the clustering dendograms.<br />
Therefore, <em>neither approach points to outliers that should be removed in this analysis.</em></p>
<p><br></p>
</div>
</div>
<div id="answer-to-environmental-health-question-2-14" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 2<a href="omics-and-system-biology-transcriptomic-applications.html#answer-to-environmental-health-question-2-14" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can now answer <strong>Environmental Health Question #2</strong></em>: When preparing transcriptomics data for statistical analyses, what are three common data filtering steps that are completed during the data QA/QC process?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> (1) Background filter to remove genes that are universally lowly expressed; (2) Sample filter to remove samples that may be not have any detectable mRNA; (3) Sample outlier filter to remove samples with underlying data distributions outside of the overall, collective dataset.*</p>
</div>
</div>
<div id="answer-to-environmental-health-question-3-8" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 3<a href="omics-and-system-biology-transcriptomic-applications.html#answer-to-environmental-health-question-3-8" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can now also answer <strong>Environmental Health Question #3</strong></em>: When identifying potential sample outliers in a typical transcriptomics dataset, what two types of approaches are commonly employed to identify samples with outlying data distributions?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> Principal component analysis (PCA) and hierarchical clustering.</p>
</div>
<p><br></p>
</div>
</div>
<div id="controlling-for-sources-of-sample-heterogeneity" class="section level2 hasAnchor">
<h2>Controlling for Sources of Sample Heterogeneity<a href="omics-and-system-biology-transcriptomic-applications.html#controlling-for-sources-of-sample-heterogeneity" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Because these transcriptomic data were generated from mouse lung tissues, there is potential for these samples to show heterogeneity based on underlying shifts in cell populations (e.g., neutrophil influx) or other aspects of sample heterogeneity (e.g., batch effects from plating, among other sources of heterogeneity that we may want to control for). For these kinds of complex samples, there are data processing methods that can be leveraged to minimize the influence of these sources of heterogeneity. Example methods include Remove Unwanted Variable (RUV), which is discussed here, as well as others (e.g., <a href="https://academic.oup.com/nar/article/42/21/e161/2903156">Surrogate Variable Analysis (SVA)</a>).</p>
<p>Here, we leverage the package called <strong>RUVseq</strong> to employ RUV on this sequencing dataset. Script was developed based off <a href="https://bioconductor.org/packages/release/bioc/html/RUVSeq.html">Bioconductor website</a>, <a href="http://bioconductor.org/packages/release/bioc/vignettes/RUVSeq/inst/doc/RUVSeq.pdf">vignette</a>, and original <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4404308/">publication</a>.</p>
<div id="steps-in-carrying-out-ruv-using-ruvseq-on-this-example-dataset" class="section level4 hasAnchor">
<h4>Steps in carrying out RUV using RUVseq on this example dataset:<a href="omics-and-system-biology-transcriptomic-applications.html#steps-in-carrying-out-ruv-using-ruvseq-on-this-example-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb688"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb688-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb688-1" tabindex="-1"></a><span class="co"># First we store the treatment IDs and exposure conditions as a separate vector</span></span>
<span id="cb688-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb688-2" tabindex="-1"></a>ID <span class="ot">&lt;-</span> coldata<span class="sc">$</span>ID</span>
<span id="cb688-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb688-3" tabindex="-1"></a></span>
<span id="cb688-4"><a href="omics-and-system-biology-transcriptomic-applications.html#cb688-4" tabindex="-1"></a><span class="co"># And differentiate our treatments and control conditions, first by grabbing the groups associated with each sample</span></span>
<span id="cb688-5"><a href="omics-and-system-biology-transcriptomic-applications.html#cb688-5" tabindex="-1"></a>groups <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(coldata<span class="sc">$</span>Group)</span>
<span id="cb688-6"><a href="omics-and-system-biology-transcriptomic-applications.html#cb688-6" tabindex="-1"></a></span>
<span id="cb688-7"><a href="omics-and-system-biology-transcriptomic-applications.html#cb688-7" tabindex="-1"></a><span class="co"># Let&#39;s view all the groups</span></span>
<span id="cb688-8"><a href="omics-and-system-biology-transcriptomic-applications.html#cb688-8" tabindex="-1"></a>groups</span></code></pre></div>
<pre><code>##  [1] PineNeedlesSmolder_4h_Lung PineNeedlesSmolder_4h_Lung
##  [3] PineNeedlesSmolder_4h_Lung PineNeedlesSmolder_4h_Lung
##  [5] PineNeedlesSmolder_4h_Lung PineNeedlesSmolder_4h_Lung
##  [7] PineNeedlesFlame_4h_Lung   PineNeedlesFlame_4h_Lung  
##  [9] PineNeedlesFlame_4h_Lung   PineNeedlesFlame_4h_Lung  
## [11] PineNeedlesFlame_4h_Lung   PineNeedlesFlame_4h_Lung  
## [13] Saline_4h_Lung             Saline_4h_Lung            
## [15] Saline_4h_Lung             Saline_4h_Lung            
## [17] Saline_4h_Lung             Saline_4h_Lung            
## [19] LPS_4h_Lung                LPS_4h_Lung               
## [21] LPS_4h_Lung                LPS_4h_Lung               
## 4 Levels: LPS_4h_Lung PineNeedlesFlame_4h_Lung ... Saline_4h_Lung</code></pre>
<div class="sourceCode" id="cb690"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb690-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb690-1" tabindex="-1"></a><span class="co"># then setting a control label</span></span>
<span id="cb690-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb690-2" tabindex="-1"></a>ctrl <span class="ot">&lt;-</span> <span class="st">&quot;Saline_4h_Lung&quot;</span></span>
<span id="cb690-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb690-3" tabindex="-1"></a></span>
<span id="cb690-4"><a href="omics-and-system-biology-transcriptomic-applications.html#cb690-4" tabindex="-1"></a><span class="co"># and extracting a vector of just our treatment groups</span></span>
<span id="cb690-5"><a href="omics-and-system-biology-transcriptomic-applications.html#cb690-5" tabindex="-1"></a>trt_groups <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(groups,ctrl)</span>
<span id="cb690-6"><a href="omics-and-system-biology-transcriptomic-applications.html#cb690-6" tabindex="-1"></a></span>
<span id="cb690-7"><a href="omics-and-system-biology-transcriptomic-applications.html#cb690-7" tabindex="-1"></a><span class="co"># let&#39;s view this vector</span></span>
<span id="cb690-8"><a href="omics-and-system-biology-transcriptomic-applications.html#cb690-8" tabindex="-1"></a>trt_groups</span></code></pre></div>
<pre><code>## [1] &quot;PineNeedlesSmolder_4h_Lung&quot; &quot;PineNeedlesFlame_4h_Lung&quot;  
## [3] &quot;LPS_4h_Lung&quot;</code></pre>
<p>RUVseq contains its own set of plotting and normalization functions, though requires input of what’s called an object of S4 class SeqExpressionSet. Let’s go ahead and make this object, using the RUVseq function ‘newSeqExpressionSet’:</p>
<div class="sourceCode" id="cb692"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb692-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb692-1" tabindex="-1"></a>exprSet <span class="ot">&lt;-</span> <span class="fu">newSeqExpressionSet</span>(<span class="fu">as.matrix</span>(countdata),<span class="at">phenoData =</span> <span class="fu">data.frame</span>(groups,<span class="at">row.names=</span><span class="fu">colnames</span>(countdata)))</span></code></pre></div>
<p>And then use this object to generate some exploratory plots using built-in tools within RUVseq.
First starting with some bar charts summarizing overall data distributions per sample:</p>
<div class="sourceCode" id="cb693"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb693-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb693-1" tabindex="-1"></a>colors <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">4</span>, <span class="st">&quot;Set2&quot;</span>)</span>
<span id="cb693-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb693-2" tabindex="-1"></a><span class="fu">plotRLE</span>(exprSet, <span class="at">outline=</span><span class="cn">FALSE</span>, <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">4</span>, <span class="dv">4</span>), <span class="at">col=</span>colors[groups])</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-529-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>We can see from this plot that some of the samples show distributions that may vary from the overall - for instance, one of the flaming pine needles-exposed samples (in orange) is far lower than the rest.</p>
<p>Then viewing a PCA plot of these samples:</p>
<div class="sourceCode" id="cb694"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb694-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb694-1" tabindex="-1"></a>colors <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">4</span>, <span class="st">&quot;Set2&quot;</span>)</span>
<span id="cb694-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb694-2" tabindex="-1"></a><span class="fu">plotPCA</span>(exprSet, <span class="at">col=</span>colors[groups], <span class="at">cex=</span><span class="fl">1.2</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-530-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>This PCA plot shows pretty good data distributions, with samples mainly showing groupings based upon exposure condition (e.g., LPS), which is to be expected. With this, we can conclude that there may be some sources of unwanted variation, but not a huge amount. Let’s see what the data look like after running RUV.</p>
<p>Now to actually run the RUVseq algorithm, to control for potential sources of sample heterogeneity, we need to first construct a matrix specifying the replicates (samples of the same exposure condition):</p>
<div class="sourceCode" id="cb695"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb695-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb695-1" tabindex="-1"></a><span class="co"># Construct a matrix specifying the replicates (samples of the same exposure condition) for running RUV</span></span>
<span id="cb695-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb695-2" tabindex="-1"></a>differences <span class="ot">&lt;-</span> <span class="fu">makeGroups</span>(groups)</span>
<span id="cb695-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb695-3" tabindex="-1"></a></span>
<span id="cb695-4"><a href="omics-and-system-biology-transcriptomic-applications.html#cb695-4" tabindex="-1"></a><span class="co"># Viewing this new matrix</span></span>
<span id="cb695-5"><a href="omics-and-system-biology-transcriptomic-applications.html#cb695-5" tabindex="-1"></a><span class="fu">head</span>(differences)</span></code></pre></div>
<pre><code>##      [,1] [,2] [,3] [,4] [,5] [,6]
## [1,]   19   20   21   22   -1   -1
## [2,]    7    8    9   10   11   12
## [3,]    1    2    3    4    5    6
## [4,]   13   14   15   16   17   18</code></pre>
<p>This matrix groups the samples by exposure condition. Here, each of the four rows represents one of the four exposure conditions, and each of the six columns represents a possible sample. Since the LPS exposure condition only had four samples, instead of six like the rest of the exposure conditions, a value of -1 is automatically used as a place holder to fill out the matrix. The samples in the matrix are identified by the index of the sample in the previously defined ‘groups’ factor that was used to generate the matrix. For example, the PineNeedlesSmolder_4h_Lung samples are the the first six samples contained in the ‘groups’ factor, so in the matrix, samples of this exposure condition are identified as ‘1’,‘2’,‘3’,‘4’,‘5’, and ‘6’.</p>
<p>Let’s now implement the RUVseq algorithm and, for this example, capture one factor (k=1) of unwanted variation. Note that the k parameter can be modified to capture additional factors, if necessary.</p>
<div class="sourceCode" id="cb697"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb697-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb697-1" tabindex="-1"></a><span class="co"># Now capture 1 factor (k=1) of unwanted variation</span></span>
<span id="cb697-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb697-2" tabindex="-1"></a>ruv_set <span class="ot">&lt;-</span> <span class="fu">RUVs</span>(exprSet, <span class="fu">rownames</span>(countdata), <span class="at">k=</span><span class="dv">1</span>, differences) </span></code></pre></div>
<p>This results in a list of objects within ‘ruv_set’, which include the following important pieces of information:</p>
<ol style="list-style-type: decimal">
<li>Estimated factors of unwanted variation are provided in the phenoData object, as viewed using the following:</li>
</ol>
<div class="sourceCode" id="cb698"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb698-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb698-1" tabindex="-1"></a><span class="co"># viewing the estimated factors of unwanted variation in the column W_1</span></span>
<span id="cb698-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb698-2" tabindex="-1"></a><span class="fu">pData</span>(ruv_set)</span></code></pre></div>
<pre><code>##                                            groups       W_1
## M13_PineNeedlesSmolder PineNeedlesSmolder_4h_Lung 1.3777900
## M14_PineNeedlesSmolder PineNeedlesSmolder_4h_Lung 1.2392662
## M15_PineNeedlesSmolder PineNeedlesSmolder_4h_Lung 1.0687702
## M16_PineNeedlesSmolder PineNeedlesSmolder_4h_Lung 1.2448747
## M17_PineNeedlesSmolder PineNeedlesSmolder_4h_Lung 1.3156003
## M18_PineNeedlesSmolder PineNeedlesSmolder_4h_Lung 1.4136600
## M49_PineNeedlesFlame     PineNeedlesFlame_4h_Lung 1.3343115
## M50_PineNeedlesFlame     PineNeedlesFlame_4h_Lung 0.4446084
## M51_PineNeedlesFlame     PineNeedlesFlame_4h_Lung 1.0172990
## M52_PineNeedlesFlame     PineNeedlesFlame_4h_Lung 1.0332373
## M53_PineNeedlesFlame     PineNeedlesFlame_4h_Lung 0.7746752
## M54_PineNeedlesFlame     PineNeedlesFlame_4h_Lung 1.0011086
## M31_Saline                         Saline_4h_Lung 0.6582117
## M32_Saline                         Saline_4h_Lung 1.2518829
## M33_Saline                         Saline_4h_Lung 1.1892456
## M34_Saline                         Saline_4h_Lung 1.2607852
## M35_Saline                         Saline_4h_Lung 1.3141520
## M36_Saline                         Saline_4h_Lung 1.4335200
## M67_LPS                               LPS_4h_Lung 1.0341947
## M68_LPS                               LPS_4h_Lung 1.2823240
## M69_LPS                               LPS_4h_Lung 1.4460347
## M70_LPS                               LPS_4h_Lung 1.4376930</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>Normalized counts obtained by regressing the original counts on the unwanted factors (normalizedCounts object within ‘ruv_set’). Note that the normalized counts should only used for exploratory purposes and not subsequent differential expression analyses. For additional information on this topic, please refer official RUVSeq documentation. The normalized counts can be viewed using the following:</li>
</ol>
<div class="sourceCode" id="cb700"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb700-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb700-1" tabindex="-1"></a><span class="co"># Viewing the head of the normalized count data, accounting for unwanted variation</span></span>
<span id="cb700-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb700-2" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">normCounts</span>(ruv_set))</span></code></pre></div>
<pre><code>##                     M13_PineNeedlesSmolder M14_PineNeedlesSmolder
## 0610009B22Rik_56046                      4                      5
## 0610010F05Rik_69119                   2552                   2630
## 0610010F05Rik_74637                     17                     13
## 0610010K14Rik_31619                   1602                   1774
## 0610010K14Rik_31873                    711                   1009
## 0610010K14Rik_68949                     23                     25
##                     M15_PineNeedlesSmolder M16_PineNeedlesSmolder
## 0610009B22Rik_56046                      8                      8
## 0610010F05Rik_69119                   2474                   1634
## 0610010F05Rik_74637                     17                      3
## 0610010K14Rik_31619                   1639                   1722
## 0610010K14Rik_31873                    819                    992
## 0610010K14Rik_68949                     19                     28
##                     M17_PineNeedlesSmolder M18_PineNeedlesSmolder
## 0610009B22Rik_56046                      6                      4
## 0610010F05Rik_69119                   2343                   2352
## 0610010F05Rik_74637                     12                      7
## 0610010K14Rik_31619                   1979                   1726
## 0610010K14Rik_31873                    889                    888
## 0610010K14Rik_68949                     20                     30
##                     M49_PineNeedlesFlame M50_PineNeedlesFlame
## 0610009B22Rik_56046                    4                    5
## 0610010F05Rik_69119                 1767                 1563
## 0610010F05Rik_74637                   12                    6
## 0610010K14Rik_31619                 1480                 1134
## 0610010K14Rik_31873                  586                  599
## 0610010K14Rik_68949                   23                   27
##                     M51_PineNeedlesFlame M52_PineNeedlesFlame
## 0610009B22Rik_56046                    4                    4
## 0610010F05Rik_69119                 1918                 1711
## 0610010F05Rik_74637                    4                    9
## 0610010K14Rik_31619                 1390                 1763
## 0610010K14Rik_31873                  683                  797
## 0610010K14Rik_68949                   26                   27
##                     M53_PineNeedlesFlame M54_PineNeedlesFlame M31_Saline
## 0610009B22Rik_56046                    5                    5          3
## 0610010F05Rik_69119                 1751                 1582       1891
## 0610010F05Rik_74637                    6                    3         10
## 0610010K14Rik_31619                 1370                 1419       1340
## 0610010K14Rik_31873                  593                  647        561
## 0610010K14Rik_68949                   20                   24         19
##                     M32_Saline M33_Saline M34_Saline M35_Saline M36_Saline
## 0610009B22Rik_56046          2          4          3          4          4
## 0610010F05Rik_69119       2507       2518       2524       1831       1878
## 0610010F05Rik_74637          7          7          5          7          8
## 0610010K14Rik_31619       1563       1632       1438       1280        967
## 0610010K14Rik_31873        685        815        703        646        526
## 0610010K14Rik_68949         16         21         19         22         20
##                     M67_LPS M68_LPS M69_LPS M70_LPS
## 0610009B22Rik_56046       3       2       4       5
## 0610010F05Rik_69119    1906    1597    1461    1201
## 0610010F05Rik_74637       6       6       7       5
## 0610010K14Rik_31619     964    1012     776     638
## 0610010K14Rik_31873     622     702     522     496
## 0610010K14Rik_68949      21      21      18      15</code></pre>
<p>Let’s again generate an exploratory plot using this updated dataset, focusing on the bar chart view since that was the most informative pre-RUV. Here are the updated bar charts summarizing overall data distributions per sample:</p>
<div class="sourceCode" id="cb702"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb702-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb702-1" tabindex="-1"></a>colors <span class="ot">&lt;-</span> <span class="fu">brewer.pal</span>(<span class="dv">4</span>, <span class="st">&quot;Set2&quot;</span>)</span>
<span id="cb702-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb702-2" tabindex="-1"></a><span class="fu">plotRLE</span>(ruv_set, <span class="at">outline=</span><span class="cn">FALSE</span>, <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">4</span>, <span class="dv">4</span>), <span class="at">col=</span>colors[groups])</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-535-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>This plot shows overall tighter data that are more similarly distributed across samples. Therefore, it is looking like this RUV addition improved the overall distribution of this dataset. It is important not to over-correct/over-smooth your datasets, so implement these types of pre-processing steps with caution. One strategy that we commonly employ to gage whether data smoothing is needed/applied correctly is to run the statistical models with and without correction of potential sources of heterogeneity, and critically evaluate similarities vs differences produced in the results.</p>
</div>
<div id="answer-to-environmental-health-question-4-5" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 4<a href="omics-and-system-biology-transcriptomic-applications.html#answer-to-environmental-health-question-4-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can now answer <strong>Environmental Health Question #4</strong></em>: What is an approach that analysts can use when evaluating transcriptomic data from tissues of mixed cellular composition to aid in controlling for sources of sample heterogeneity?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> Remove unwanted variation (RUV), among other approaches, including surrogate variable analysis (SVA).</p>
</div>
<p><br></p>
</div>
</div>
<div id="identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e.g.-biomass-smoke-exposure" class="section level2 hasAnchor">
<h2>Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure)<a href="omics-and-system-biology-transcriptomic-applications.html#identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e.g.-biomass-smoke-exposure" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>At this point, we have completed several data pre-processing, QA/QC, and additional steps to prepare our example transcriptomics data for statistical analysis. And finally, we are ready to run the overall statistical model to identify genes that are altered in expression in association with different biomass burn conditions.</p>
<p>Here we leverage the DESeq2 package to carry out these statistical comparisons. This package is now the most commonly implemented analysis pipeline used for transcriptomic data, including sequencing data as well as transcriptomic data produced via other technologies (e.g., Nanostring, Fluidigm, and other gene expression technologies). This package is extremely well-documented and we encourage trainees to leverage these resources in parallel with the current training module when carrying out their own transcriptomics analyses in R:</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/bioc/html/DESeq2.html">Bioconductor website</a><br />
</li>
<li><a href="http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">Vignette</a><br />
</li>
<li><a href="https://bioconductor.org/packages/devel/bioc/manuals/DESeq2/man/DESeq2.pdf">Manual</a><br />
</li>
<li>Primary citation: Love MI, Huber W, Anders S. Moderated estimation of fold change and dispersion for RNA-seq data with DESeq2. Genome Biol. 2014;15(12):550. doi: 10.1186/s13059-014-0550-8. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/25516281/">25516281</a></li>
</ul>
<p>In brief, the basic calculations employed within the DESeq2 underlying algorithms include the following:</p>
<p><strong>1. Estimate size factors.</strong>
In the first step, size factors are estimated to help account for potential differences in the sequencing depth across samples. It is similar to a normalization parameter in the model.</p>
<p><strong>2. Normalize count data.</strong>
DESeq2 employs different normalization algorithms depending on the parameters selected / stage of analysis. The most commonly employed method is called the <strong>median of ratios</strong>, which takes into account sequencing depth and RNA composition, as described <a href="https://hbctraining.github.io/DGE_workshop/lessons/02_DGE_count_normalization.html">here</a>. Specifically, these normalized values are calculated as counts divided by sample-specific size factors determined by median ratio of gene counts relative to geometric mean per gene. DESeq2 then transforms these data using variance stabilization within the final statistical model. Because of these two steps, we prefer to export both the median of ratios normalized data as well as the variance stabilization transformed data, to save in our records and use when generating plots of expression levels for specific genes we are interested in. These steps are detailed below.</p>
<p><strong>3. Estimate dispersion.</strong>
The dispersion estimate takes into account the relationship between the variance of an observed count and its mean value. It is similar to a variance parameter. In DESeq2, dispersion is estimated using a maximum likelihood and empirical bayes approach.</p>
<p><strong>4. Fit negative binomial generalized linear model (GLM).</strong>
Finally, a negative binomial model is fit for each gene using the design formula that will be described within the proceeding code. The Wald test is performed to test if log fold changes in expression (typically calculated as log(average exposed / average unexposed)) significantly differ from zero. Statistical p-values are reported from this test and also adjusted for multiple testing using the Benjamini and Hochberg procedure.</p>
<p>Note that these calculations, among others, are embedded within the DESeq2 functions, so we do not need to code for them ourselves. Instead, we just need to make sure that we set-up the DESeq2 functions correctly, such that these calculations are carried out appropriately in our final transcriptomics analyses.</p>
<div id="setting-up-the-deseq2-experiment" class="section level4 hasAnchor">
<h4>Setting up the DESeq2 experiment<a href="omics-and-system-biology-transcriptomic-applications.html#setting-up-the-deseq2-experiment" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Here we provide example script that is used to identify which genes are significantly differentially expressed in association with the example biomass smoke exposures, smoldering pine needles and flaming pine needles, as well as a positive inflammation control, LPS.</p>
<p>First, we need to set-up the DESeq2 experiment:</p>
<div class="sourceCode" id="cb703"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb703-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb703-1" tabindex="-1"></a><span class="co"># Set up our experiment using our RUV adjusted count and phenotype data.</span></span>
<span id="cb703-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb703-2" tabindex="-1"></a><span class="co"># Our design indicates that our count data is dependent on the exposure condition (groups variable) and our factor of unwanted variation, and we have specified that there not be an intercept term through the use of &#39;~0&#39;</span></span>
<span id="cb703-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb703-3" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> <span class="fu">counts</span>(ruv_set), <span class="co"># Grabbing count data from the &#39;ruv_set&#39; object</span></span>
<span id="cb703-4"><a href="omics-and-system-biology-transcriptomic-applications.html#cb703-4" tabindex="-1"></a>                                 <span class="at">colData =</span> <span class="fu">pData</span>(ruv_set), <span class="co"># Grabbing the phenotype data and corresponding factor of unwanted variation from the &#39;ruv_set&#39; object</span></span>
<span id="cb703-5"><a href="omics-and-system-biology-transcriptomic-applications.html#cb703-5" tabindex="-1"></a>                                 <span class="at">design =</span> <span class="sc">~</span><span class="dv">0</span><span class="sc">+</span>groups<span class="sc">+</span>W_1) <span class="co"># Setting up the statistical formula (see below)</span></span></code></pre></div>
<p>For the formula design, we use a ‘~0’ at the front to not include an intercept term, and then also account for the exposure condition (groups) and the previously calculated factors of unwanted variation (W_1) of the samples. Formula design is an important step and should be carefully considered for each individual analysis. Other resources, including official DESeq2 documentation, are availble for consultation regarding formula design, as the specifics of formula design are beyond the scope of this training module.</p>
<p>It is worth noting that, by default, DESeq2 will use the last variable in the design formula (“W_1”) in this case, as the default variable to be output from the “results” function. Additionally, if the variable is categorical, it will display results comparing the reference level to the last level of that variable. To get results for other variables or to see other comparisions within a categorical variable, we can use the “contrast” parameter, which will be demonstrated below.</p>
</div>
<div id="estimating-size-factors" class="section level4 hasAnchor">
<h4>Estimating size factors<a href="omics-and-system-biology-transcriptomic-applications.html#estimating-size-factors" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb704"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb704-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb704-1" tabindex="-1"></a><span class="co"># Estimate size factors from the dds object that was just created as the experiment above</span></span>
<span id="cb704-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb704-2" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">estimateSizeFactors</span>(dds)</span>
<span id="cb704-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb704-3" tabindex="-1"></a><span class="fu">sizeFactors</span>(dds)  <span class="co"># viewing the size factors</span></span></code></pre></div>
<pre><code>## M13_PineNeedlesSmolder M14_PineNeedlesSmolder M15_PineNeedlesSmolder 
##              1.2668895              1.3040762              1.1782071 
## M16_PineNeedlesSmolder M17_PineNeedlesSmolder M18_PineNeedlesSmolder 
##              1.1723749              1.2904950              1.3405500 
##   M49_PineNeedlesFlame   M50_PineNeedlesFlame   M51_PineNeedlesFlame 
##              1.1313237              0.6313099              0.9328970 
##   M52_PineNeedlesFlame   M53_PineNeedlesFlame   M54_PineNeedlesFlame 
##              1.0050414              0.7649632              0.8747854 
##             M31_Saline             M32_Saline             M33_Saline 
##              0.8050127              1.0480725              1.0389623 
##             M34_Saline             M35_Saline             M36_Saline 
##              1.0282261              0.9519266              1.0409103 
##                M67_LPS                M68_LPS                M69_LPS 
##              0.9308588              1.1002076              0.9793051 
##                M70_LPS 
##              0.8575718</code></pre>
</div>
<div id="calculating-and-exporting-normalized-counts" class="section level4 hasAnchor">
<h4>Calculating and exporting normalized counts<a href="omics-and-system-biology-transcriptomic-applications.html#calculating-and-exporting-normalized-counts" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Here, we extract normalized counts and variance stablized counts.</p>
<div class="sourceCode" id="cb706"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb706-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb706-1" tabindex="-1"></a><span class="co"># Extract normalized count data</span></span>
<span id="cb706-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb706-2" tabindex="-1"></a>normcounts <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">counts</span>(dds, <span class="at">normalized=</span><span class="cn">TRUE</span>))</span>
<span id="cb706-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb706-3" tabindex="-1"></a></span>
<span id="cb706-4"><a href="omics-and-system-biology-transcriptomic-applications.html#cb706-4" tabindex="-1"></a><span class="co"># Transforming normalized counts through variance stabilization</span></span>
<span id="cb706-5"><a href="omics-and-system-biology-transcriptomic-applications.html#cb706-5" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">varianceStabilizingTransformation</span>(dds, <span class="at">blind=</span><span class="cn">FALSE</span>)</span>
<span id="cb706-6"><a href="omics-and-system-biology-transcriptomic-applications.html#cb706-6" tabindex="-1"></a>vsd_matrix <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(<span class="fu">assay</span>(vsd))</span></code></pre></div>
<p>We could also export them using code such as:</p>
<div class="sourceCode" id="cb707"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb707-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb707-1" tabindex="-1"></a><span class="co"># Export data</span></span>
<span id="cb707-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb707-2" tabindex="-1"></a><span class="fu">write.csv</span>(normcounts, <span class="st">&quot;Module6_2_Input/Module6_2_Output_NormalizedCounts.csv&quot;</span>)</span>
<span id="cb707-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb707-3" tabindex="-1"></a><span class="fu">write.csv</span>(vsd_matrix, <span class="st">&quot;Module6_2_Input/Module6_2_Output_VSDCounts.csv&quot;</span>, <span class="at">row.names=</span><span class="cn">TRUE</span>)</span></code></pre></div>
</div>
<div id="running-the-final-deseq2-experiment" class="section level4 hasAnchor">
<h4>Running the final DESeq2 experiment<a href="omics-and-system-biology-transcriptomic-applications.html#running-the-final-deseq2-experiment" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Here, we are finally ready to run the actual statistical comparisons (exposed vs control samples) to calculate fold changes and p-values that describe the degree to which each gene may or may not be altered at the expression level in association with treatment.</p>
<p>For this example, we would like to run three different comparisons:<br />
(1) Smoldering Pine Needles vs. Control<br />
(2) Flaming Pine Needles vs. Control<br />
(3) LPS vs. Control<br />
which we can easily code for using a loop function, as detailed below.</p>
<p>Note that we have commented out the line of code for writing out the CSV because we do not need it for the rest of the module, but this could be used if you need to write out and view results in an external application such as Excel for supplementary materials.</p>
<div class="sourceCode" id="cb708"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb708-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb708-1" tabindex="-1"></a><span class="co"># Run experiment</span></span>
<span id="cb708-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb708-2" tabindex="-1"></a>dds_run <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(dds, <span class="at">betaPrior=</span><span class="cn">FALSE</span>) </span>
<span id="cb708-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb708-3" tabindex="-1"></a></span>
<span id="cb708-4"><a href="omics-and-system-biology-transcriptomic-applications.html#cb708-4" tabindex="-1"></a><span class="co"># Loop through and extract and export results for all contrasts (treatments vs. control)</span></span>
<span id="cb708-5"><a href="omics-and-system-biology-transcriptomic-applications.html#cb708-5" tabindex="-1"></a><span class="cf">for</span> (trt <span class="cf">in</span> trt_groups){ <span class="co"># Iterate for each of the treatments listed in &#39;trt_groups&#39;</span></span>
<span id="cb708-6"><a href="omics-and-system-biology-transcriptomic-applications.html#cb708-6" tabindex="-1"></a>  <span class="fu">cat</span>(trt) <span class="co"># Print which treatment group we are on in the loop </span></span>
<span id="cb708-7"><a href="omics-and-system-biology-transcriptomic-applications.html#cb708-7" tabindex="-1"></a>  res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_run, <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">&quot;groups&quot;</span>,trt,ctrl)) <span class="co"># Extract the results of the DESeq2 analysis specifically for the comparison of the treatment group for the current iteration of the loop with the control group</span></span>
<span id="cb708-8"><a href="omics-and-system-biology-transcriptomic-applications.html#cb708-8" tabindex="-1"></a>  <span class="fu">summary</span>(res) <span class="co"># Print out a high-level summary of the results</span></span>
<span id="cb708-9"><a href="omics-and-system-biology-transcriptomic-applications.html#cb708-9" tabindex="-1"></a>  ordered <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(res[<span class="fu">order</span>(res<span class="sc">$</span>padj),]) <span class="co"># Make a dataframe of the results and order them by adjusted p-value from lowest to highest</span></span>
<span id="cb708-10"><a href="omics-and-system-biology-transcriptomic-applications.html#cb708-10" tabindex="-1"></a>  top10 <span class="ot">&lt;-</span> <span class="fu">head</span>(ordered, <span class="at">n=</span><span class="dv">10</span>) <span class="co"># Make dataframe of the first ten rows of the ordered results</span></span>
<span id="cb708-11"><a href="omics-and-system-biology-transcriptomic-applications.html#cb708-11" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">The 10 most significantly differentially expressed genes by adjusted p-value:</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb708-12"><a href="omics-and-system-biology-transcriptomic-applications.html#cb708-12" tabindex="-1"></a>  <span class="fu">print</span>(top10) <span class="co"># View the first ten rows of the ordered results</span></span>
<span id="cb708-13"><a href="omics-and-system-biology-transcriptomic-applications.html#cb708-13" tabindex="-1"></a>  pfilt<span class="fl">.05</span> <span class="ot">&lt;-</span> <span class="fu">nrow</span>(ordered <span class="sc">%&gt;%</span> <span class="fu">filter</span>(padj<span class="sc">&lt;</span><span class="fl">0.05</span>)) <span class="co"># Get the number of genes that are significantly differentially expressed where padj &lt; 0.05</span></span>
<span id="cb708-14"><a href="omics-and-system-biology-transcriptomic-applications.html#cb708-14" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">The number of genes showing significant differential expression where padj &lt; 0.05 is &quot;</span>, pfilt<span class="fl">.05</span>)</span>
<span id="cb708-15"><a href="omics-and-system-biology-transcriptomic-applications.html#cb708-15" tabindex="-1"></a>  pfilt<span class="fl">.10</span> <span class="ot">&lt;-</span> <span class="fu">nrow</span>(ordered <span class="sc">%&gt;%</span> <span class="fu">filter</span>(padj<span class="sc">&lt;</span><span class="fl">0.1</span>)) <span class="co"># Get the number of genes that are significantly differentially expressed where padj &lt; 0.10</span></span>
<span id="cb708-16"><a href="omics-and-system-biology-transcriptomic-applications.html#cb708-16" tabindex="-1"></a>  <span class="fu">cat</span>(<span class="st">&quot;</span><span class="sc">\n</span><span class="st">The number of genes showing significant differential expression where padj &lt; 0.10 is &quot;</span>, pfilt<span class="fl">.10</span>,<span class="st">&quot;</span><span class="sc">\n\n</span><span class="st">&quot;</span>)</span>
<span id="cb708-17"><a href="omics-and-system-biology-transcriptomic-applications.html#cb708-17" tabindex="-1"></a>  <span class="co"># write.csv(ordered, paste0(&quot;Module6_2_Output_StatisticalResults_&quot;,trt ,&quot;.csv&quot;)) ## Export the full dataframe of ordered results as a csv</span></span>
<span id="cb708-18"><a href="omics-and-system-biology-transcriptomic-applications.html#cb708-18" tabindex="-1"></a>}</span></code></pre></div>
<pre><code>## PineNeedlesSmolder_4h_Lung
## out of 16664 with nonzero total read count
## adjusted p-value &lt; 0.1
## LFC &gt; 0 (up)       : 457, 2.7%
## LFC &lt; 0 (down)     : 697, 4.2%
## outliers [1]       : 0, 0%
## low counts [2]     : 1293, 7.8%
## (mean count &lt; 21)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results
## 
## 
## The 10 most significantly differentially expressed genes by adjusted p-value:
## 
##                baseMean log2FoldChange      lfcSE      stat       pvalue
## Hbs1l_72436   367.64772     -1.1465230 0.13614949 -8.421060 3.730933e-17
## Cetn3_30290  2350.31984     -0.7197604 0.09379276 -7.673944 1.667868e-14
## Grasp_56900   277.43936      0.9759274 0.14177298  6.883734 5.830390e-12
## Pcsk7_57804   180.24040      0.8261941 0.13529825  6.106465 1.018617e-09
## Plscr4_55764  178.95216      1.3193654 0.21751536  6.065620 1.314459e-09
## Socs3_79479   183.28293      1.2535533 0.20740514  6.043983 1.503550e-09
## Kat2b_64780   444.97603     -0.7197105 0.12034633 -5.980327 2.226896e-09
## Ptp4a2_62762 1840.75247     -0.7802857 0.13076704 -5.966990 2.416698e-09
## Rmi1_56327    377.57936     -0.8548903 0.14288443 -5.983089 2.189449e-09
## Abcd2_62271    57.46455     -1.2504459 0.21464501 -5.825646 5.689193e-09
##                      padj
## Hbs1l_72436  5.734817e-13
## Cetn3_30290  1.281840e-10
## Grasp_56900  2.987297e-08
## Pcsk7_57804  3.851843e-06
## Plscr4_55764 3.851843e-06
## Socs3_79479  3.851843e-06
## Kat2b_64780  4.127451e-06
## Ptp4a2_62762 4.127451e-06
## Rmi1_56327   4.127451e-06
## Abcd2_62271  8.744859e-06
## 
## The number of genes showing significant differential expression where padj &lt; 0.05 is  679
## The number of genes showing significant differential expression where padj &lt; 0.10 is  1154 
## 
## PineNeedlesFlame_4h_Lung
## out of 16664 with nonzero total read count
## adjusted p-value &lt; 0.1
## LFC &gt; 0 (up)       : 411, 2.5%
## LFC &lt; 0 (down)     : 575, 3.5%
## outliers [1]       : 0, 0%
## low counts [2]     : 0, 0%
## (mean count &lt; 10)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results
## 
## 
## The 10 most significantly differentially expressed genes by adjusted p-value:
## 
##                 baseMean log2FoldChange     lfcSE      stat       pvalue
## Tmem109_70229  351.49267      1.2681226 0.1637785  7.742913 9.716419e-15
## Rhof_64396      58.57258      1.4767798 0.2210640  6.680328 2.384081e-11
## Plscr4_55764   178.95216      1.5349460 0.2393976  6.411701 1.439049e-10
## Pip4k2a_67992  796.64597      0.8262520 0.1299839  6.356571 2.063073e-10
## Zranb2_68671   739.91379      0.8131628 0.1330277  6.112735 9.793822e-10
## Cul1_31957     896.52286     -0.7061150 0.1203618 -5.866602 4.448172e-09
## Ppfibp1_75630  655.07197      1.4148222 0.2481843  5.700692 1.193223e-08
## Mbnl1_30670    851.18586     -1.0159639 0.1811247 -5.609197 2.032670e-08
## Atp5k_68198   1096.49479      1.1003470 0.1983469  5.547588 2.896380e-08
## Gclm_29258     138.54250      1.0993904 0.1985396  5.537387 3.070173e-08
##                       padj
## Tmem109_70229 1.619144e-10
## Rhof_64396    1.986416e-07
## Plscr4_55764  7.993439e-07
## Pip4k2a_67992 8.594761e-07
## Zranb2_68671  3.264085e-06
## Cul1_31957    1.235406e-05
## Ppfibp1_75630 2.840553e-05
## Mbnl1_30670   4.234052e-05
## Atp5k_68198   5.116136e-05
## Gclm_29258    5.116136e-05
## 
## The number of genes showing significant differential expression where padj &lt; 0.05 is  515
## The number of genes showing significant differential expression where padj &lt; 0.10 is  986 
## 
## LPS_4h_Lung
## out of 16664 with nonzero total read count
## adjusted p-value &lt; 0.1
## LFC &gt; 0 (up)       : 2960, 18%
## LFC &lt; 0 (down)     : 2947, 18%
## outliers [1]       : 0, 0%
## low counts [2]     : 0, 0%
## (mean count &lt; 10)
## [1] see &#39;cooksCutoff&#39; argument of ?results
## [2] see &#39;independentFiltering&#39; argument of ?results
## 
## 
## The 10 most significantly differentially expressed genes by adjusted p-value:
## 
##                baseMean log2FoldChange     lfcSE     stat        pvalue
## Usp18_32270   1319.2259       3.829526 0.1459140 26.24508 8.133515e-152
## Usp18_67659    414.5959       3.739864 0.1679851 22.26307 8.428171e-110
## Ifit1_29859    862.3286       5.000365 0.2259553 22.12989 1.629877e-108
## Oasl2_67419   1437.6182       3.710164 0.1904583 19.48019  1.616902e-84
## Gbp5_77887     600.7219       5.272287 0.2769982 19.03365  8.977555e-81
## Oasl2_67043    292.4482       3.898484 0.2096435 18.59578  3.476310e-77
## Rnf213_57840  1052.0454       3.168052 0.1731212 18.29962  8.332621e-75
## Cmpk2_70664    429.4232       4.592488 0.2518916 18.23200  2.875888e-74
## Cmpk2_29189    695.1997       4.264136 0.2386279 17.86939  2.042134e-71
## Eif2ak2_30238 1585.6071       2.671974 0.1495526 17.86645  2.152596e-71
##                        padj
## Usp18_32270   1.355369e-147
## Usp18_67659   7.022352e-106
## Ifit1_29859   9.053424e-105
## Oasl2_67419    6.736013e-81
## Gbp5_77887     2.992039e-77
## Oasl2_67043    9.654872e-74
## Rnf213_57840   1.983640e-71
## Cmpk2_70664    5.990475e-71
## Cmpk2_29189    3.587085e-68
## Eif2ak2_30238  3.587085e-68
## 
## The number of genes showing significant differential expression where padj &lt; 0.05 is  4813
## The number of genes showing significant differential expression where padj &lt; 0.10 is  5907</code></pre>
</div>
<div id="answer-to-environmental-health-question-5-4" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 5<a href="omics-and-system-biology-transcriptomic-applications.html#answer-to-environmental-health-question-5-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can now answer <strong>Environmental Health Question #5</strong></em>: How many genes showed significant differential expression associated with flaming pine needles exposure in the mouse lung, based on a statistical filter of a multiple test corrected p-value (padj) &lt; 0.05?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> 515 genes</p>
</div>
</div>
<div id="answer-to-environmental-health-question-6-2" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 6<a href="omics-and-system-biology-transcriptomic-applications.html#answer-to-environmental-health-question-6-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can also now answer <strong>Environmental Health Question #6</strong></em>: How many genes showed significant differential expression associated with smoldering pine needles exposure in the mouse lung, based on a statistical filter of a multiple test corrected p-value (padj) &lt; 0.05?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> 679 genes</p>
</div>
</div>
<div id="answer-to-environmental-health-question-7" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 7<a href="omics-and-system-biology-transcriptomic-applications.html#answer-to-environmental-health-question-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>And, we can answer <strong>Environmental Health Question #7</strong></em>: How many genes showed significant differential expression associated with lipopolysaccharide (LPS) exposure in the mouse lung, based on a statistical filter of a multiple test corrected p-value (padj) &lt; 0.05?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> 4813 genes</p>
</div>
<p><em>Together, we find that exposure to both flaming and smoldering of pine needles caused substantial disruptions in gene expression profiles. LPS serves as a positive control for inflammation and produced the greatest transcriptomic response.</em></p>
<p><br></p>
</div>
</div>
<div id="visualizing-statistical-results-using-ma-plots" class="section level2 hasAnchor">
<h2>Visualizing Statistical Results using MA Plots<a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-ma-plots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="https://en.wikipedia.org/wiki/MA_plot">MA plots</a> represent a common method of visualization that illustrates differences between measurements taken in two samples, by transforming the data onto M (log ratio) and A (mean average) scales, then plotting these values.</p>
<p>Here, we leverage MA plots to show how log fold changes relate to expression levels. In these plots, the log fold change is plotted on the y-axis and expression values are plotted along the x-axis, and dots are colored according to statistical significance (using padj&lt;0.05 as the statistical filter). Here we will generate an MA plot for Flaming Pine Needles.</p>
<div class="sourceCode" id="cb710"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb710-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-1" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(dds_run, <span class="at">pAdjustMethod =</span> <span class="st">&quot;BH&quot;</span>, <span class="at">contrast =</span> <span class="fu">c</span>(<span class="st">&quot;groups&quot;</span>,<span class="st">&quot;PineNeedlesFlame_4h_Lung&quot;</span>,ctrl)) <span class="co"># Re-extract the DESeq2 results for the flamming pine needles </span></span>
<span id="cb710-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-2" tabindex="-1"></a>MA <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(res) <span class="co"># Make a prelimiary dataframe of the flaming pine needle results</span></span>
<span id="cb710-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-3" tabindex="-1"></a>MA_ns <span class="ot">&lt;-</span> MA[ <span class="fu">which</span>(MA<span class="sc">$</span>padj<span class="sc">&gt;=</span><span class="fl">0.05</span>),]  <span class="co"># Non-significant genes to plot</span></span>
<span id="cb710-4"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-4" tabindex="-1"></a>MA_up <span class="ot">&lt;-</span> MA[ <span class="fu">which</span>(MA<span class="sc">$</span>padj<span class="sc">&lt;</span><span class="fl">0.05</span> <span class="sc">&amp;</span> MA<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span>),]   <span class="co"># Significant up-regulated genes to plot</span></span>
<span id="cb710-5"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-5" tabindex="-1"></a>MA_down <span class="ot">&lt;-</span> MA[ <span class="fu">which</span>(MA<span class="sc">$</span>padj<span class="sc">&lt;</span><span class="fl">0.05</span> <span class="sc">&amp;</span> MA<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span>),]  <span class="co">#Significant down-regulated genes to plot</span></span>
<span id="cb710-6"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-6" tabindex="-1"></a></span>
<span id="cb710-7"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-7" tabindex="-1"></a><span class="fu">ggplot</span>(MA_ns, <span class="fu">aes</span>(<span class="at">x =</span> baseMean, <span class="at">y =</span> log2FoldChange)) <span class="sc">+</span> <span class="co"># Plot data with counts on x-axis and log2 fold change on y-axis</span></span>
<span id="cb710-8"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-8" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">color=</span><span class="st">&quot;gray75&quot;</span>, <span class="at">size =</span> .<span class="dv">5</span>) <span class="sc">+</span> <span class="co"># Set point size and color</span></span>
<span id="cb710-9"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-9" tabindex="-1"></a>  </span>
<span id="cb710-10"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-10" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> MA_up, <span class="at">color=</span><span class="st">&quot;firebrick&quot;</span>, <span class="at">size=</span><span class="dv">1</span>, <span class="at">show.legend =</span> <span class="cn">TRUE</span>) <span class="sc">+</span> <span class="co"># Plot the up-regulated significant genes</span></span>
<span id="cb710-11"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-11" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> MA_down, <span class="at">color=</span><span class="st">&quot;dodgerblue2&quot;</span>, <span class="at">size=</span><span class="dv">1</span>, <span class="at">show.legend =</span> <span class="cn">TRUE</span>) <span class="sc">+</span> <span class="co"># Plot down-regulated significant genes</span></span>
<span id="cb710-12"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-12" tabindex="-1"></a>  </span>
<span id="cb710-13"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-13" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="co"># Change theme of plot from gray to black and white</span></span>
<span id="cb710-14"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-14" tabindex="-1"></a>  </span>
<span id="cb710-15"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-15" tabindex="-1"></a>  <span class="co"># We want to log10 transform x-axis for better visualizations</span></span>
<span id="cb710-16"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-16" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">trans =</span> <span class="st">&quot;log10&quot;</span>, <span class="at">breaks=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">10</span>,<span class="dv">100</span>, <span class="dv">1000</span>, <span class="dv">10000</span>, <span class="dv">100000</span>, <span class="dv">1000000</span>), <span class="at">labels=</span><span class="fu">c</span>(<span class="st">&quot;1&quot;</span>,<span class="st">&quot;10&quot;</span>,<span class="st">&quot;100&quot;</span>, <span class="st">&quot;1000&quot;</span>, <span class="st">&quot;10000&quot;</span>, <span class="st">&quot;100000&quot;</span>, <span class="st">&quot;1000000&quot;</span>)) <span class="sc">+</span> </span>
<span id="cb710-17"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-17" tabindex="-1"></a>  <span class="co"># We will bound y axis as well to better fit data while not leaving out too many points</span></span>
<span id="cb710-18"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-18" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>, <span class="dv">2</span>)) <span class="sc">+</span></span>
<span id="cb710-19"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-19" tabindex="-1"></a>  </span>
<span id="cb710-20"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-20" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Expression (Normalized Count)&quot;</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Fold Change (log2)&quot;</span>) <span class="sc">+</span> <span class="co"># Add labels for axes</span></span>
<span id="cb710-21"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-21" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title=</span><span class="st">&quot;MA Plot Flaming Pine Needles 4h Lung&quot;</span>) <span class="sc">+</span>   <span class="co"># Add title</span></span>
<span id="cb710-22"><a href="omics-and-system-biology-transcriptomic-applications.html#cb710-22" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">0</span>) <span class="co"># Add horizontal line at 0</span></span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-541-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="visualizing-statistical-results-using-volcano-plots" class="section level2 hasAnchor">
<h2>Visualizing Statistical Results using Volcano Plots<a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-volcano-plots" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Similar to MA plots, volcano plots provide visualizations of fold changes in expression from transcriptomic data. However, instead of plotting these values against expression, log fold change is plotted against (adjusted) p-values in volcano plots. Here, we use functions within the <a href="https://www.rdocumentation.org/packages/EnhancedVolcano/versions/1.11.3/topics/EnhancedVolcano">EnhancedVolcano package</a> to generate a volcano plot for Flaming Pine Needles.</p>
<p>Running the ‘EnhancedVolcano’ function to generate an example volcano plot:</p>
<div class="sourceCode" id="cb711"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb711-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb711-1" tabindex="-1"></a>Vol <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(res) <span class="co"># Dataset to use for plotting</span></span>
<span id="cb711-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb711-2" tabindex="-1"></a><span class="fu">EnhancedVolcano</span>(Vol, </span>
<span id="cb711-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb711-3" tabindex="-1"></a>                <span class="at">lab =</span> <span class="fu">rownames</span>(res), <span class="co"># Label information from dataset (can be a column name)</span></span>
<span id="cb711-4"><a href="omics-and-system-biology-transcriptomic-applications.html#cb711-4" tabindex="-1"></a>                <span class="at">x =</span> <span class="st">&#39;log2FoldChange&#39;</span>, <span class="co"># Column name in dataset with l2fc information</span></span>
<span id="cb711-5"><a href="omics-and-system-biology-transcriptomic-applications.html#cb711-5" tabindex="-1"></a>                <span class="at">y =</span> <span class="st">&#39;padj&#39;</span>, <span class="co"># Column name in dataset with adjusted p-value information</span></span>
<span id="cb711-6"><a href="omics-and-system-biology-transcriptomic-applications.html#cb711-6" tabindex="-1"></a>                <span class="at">ylab =</span> <span class="st">&quot;-Log(FDR-adjusted p)&quot;</span>, <span class="co"># Y-axis label</span></span>
<span id="cb711-7"><a href="omics-and-system-biology-transcriptomic-applications.html#cb711-7" tabindex="-1"></a>                <span class="at">pCutoff=</span> <span class="fl">0.05</span>, <span class="co"># Set p-value cutoff</span></span>
<span id="cb711-8"><a href="omics-and-system-biology-transcriptomic-applications.html#cb711-8" tabindex="-1"></a>                <span class="at">ylim=</span><span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">5</span>), <span class="co"># Limit y-axis for better plot visuals</span></span>
<span id="cb711-9"><a href="omics-and-system-biology-transcriptomic-applications.html#cb711-9" tabindex="-1"></a>                <span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">2</span>,<span class="dv">2</span>), <span class="co"># Limit x-axis (similar to in MA plot y-axis)</span></span>
<span id="cb711-10"><a href="omics-and-system-biology-transcriptomic-applications.html#cb711-10" tabindex="-1"></a>                <span class="at">title=</span><span class="st">&quot;Volcano Plot&quot;</span>, <span class="co"># Set title</span></span>
<span id="cb711-11"><a href="omics-and-system-biology-transcriptomic-applications.html#cb711-11" tabindex="-1"></a>                <span class="at">subtitle =</span> <span class="st">&quot;Flaming Pine Needles 4h Lung&quot;</span>, <span class="co"># Set subtitle</span></span>
<span id="cb711-12"><a href="omics-and-system-biology-transcriptomic-applications.html#cb711-12" tabindex="-1"></a>                <span class="at">legendPosition =</span> <span class="st">&#39;bottom&#39;</span>) <span class="co"># Put legend on bottom</span></span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-542-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
<div id="interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis" class="section level2 hasAnchor">
<h2>Interpretting Findings at the Systems Level through Pathway Enrichment Analysis<a href="omics-and-system-biology-transcriptomic-applications.html#interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Pathway enrichment analysis is a very helpful tool that can be applied to interpret transcriptomic changes of interest in terms of systems biology. In these types of analyses, gene lists of interest are used to identify biological pathways that include genes present in your dataset more often than expected by chance alone. There are many tools that can be used to carry out pathway enrichment analyses. Here, we are using the R package, PIANO, to carry out the statistical enrichment analysis based on the lists of genes we previously identified with differential expression associated with flaming pine needles exposure.</p>
<p>To detail, the following input data are required to run PIANO:<br />
(1) Your background gene sets, which represent all genes queried from your experiment (aka your ‘gene universe’)
(2) The list of genes you are interested in evaluating pathway enrichment of; here, this represents the genes identified with significant differential expression associated with flaming pine needles<br />
(3) A underlying pathway dataset; here, we’re using the KEGG PATHWAY Database (<a href="https://www.genome.jp/kegg/pathway.html">KEGG</a>), summarized through the Molecular Signature Database (<a href="https://www.gsea-msigdb.org/gsea/msigdb/">MSigDB</a>) into pre-formatted input files (.gmt) ready for PIANO.</p>
<p><em>Let’s organize these three required data inputs.</em></p>
<ol style="list-style-type: decimal">
<li>Background gene set:</li>
</ol>
<div class="sourceCode" id="cb712"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb712-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb712-1" tabindex="-1"></a><span class="co"># First grab the rownames of the &#39;res&#39; object, which was redefined as the DESeq2 results for flaming pine needles prior to MA plot generation, and remove the BioSpyder numeric identifier using a sub function, while maintaining the gene symbol and place these IDs into a new list within the &#39;res&#39; object (saved as &#39;id&#39;)</span></span>
<span id="cb712-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb712-2" tabindex="-1"></a>res<span class="sc">$</span>id <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(res));</span>
<span id="cb712-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb712-3" tabindex="-1"></a></span>
<span id="cb712-4"><a href="omics-and-system-biology-transcriptomic-applications.html#cb712-4" tabindex="-1"></a><span class="co"># Because these IDs now contain duplicate gene symbols, we need to remove duplicates</span></span>
<span id="cb712-5"><a href="omics-and-system-biology-transcriptomic-applications.html#cb712-5" tabindex="-1"></a><span class="co"># One way to do this is to preferentially retain rows of data with the largest fold change (it doesn&#39;t really matter here, because we&#39;re just identifying unique genes within the background set)</span></span>
<span id="cb712-6"><a href="omics-and-system-biology-transcriptomic-applications.html#cb712-6" tabindex="-1"></a>res.ordered <span class="ot">&lt;-</span> res[<span class="fu">order</span>(res<span class="sc">$</span>id, <span class="sc">-</span><span class="fu">abs</span>(res<span class="sc">$</span>log2FoldChange) ), ] <span class="co"># sort by id and reverse of abs(log2foldchange)</span></span>
<span id="cb712-7"><a href="omics-and-system-biology-transcriptomic-applications.html#cb712-7" tabindex="-1"></a>res.ordered <span class="ot">&lt;-</span> res.ordered[ <span class="sc">!</span><span class="fu">duplicated</span>(res.ordered<span class="sc">$</span>id), ]     <span class="co"># removing gene duplicates</span></span>
<span id="cb712-8"><a href="omics-and-system-biology-transcriptomic-applications.html#cb712-8" tabindex="-1"></a></span>
<span id="cb712-9"><a href="omics-and-system-biology-transcriptomic-applications.html#cb712-9" tabindex="-1"></a><span class="co"># Setting this as the background list</span></span>
<span id="cb712-10"><a href="omics-and-system-biology-transcriptomic-applications.html#cb712-10" tabindex="-1"></a>Background <span class="ot">&lt;-</span> <span class="fu">toupper</span>(<span class="fu">as.character</span>(res.ordered<span class="sc">$</span>id))</span>
<span id="cb712-11"><a href="omics-and-system-biology-transcriptomic-applications.html#cb712-11" tabindex="-1"></a>Background[<span class="dv">1</span><span class="sc">:</span><span class="dv">200</span>] <span class="co"># viewing the first 200 genes in this background list</span></span></code></pre></div>
<pre><code>##   [1] &quot;0610009B22RIK&quot;  &quot;0610010F05RIK&quot;  &quot;0610010K14RIK&quot;  &quot;0610012G03RIK&quot; 
##   [5] &quot;0610030E20RIK&quot;  &quot;0610040J01RIK&quot;  &quot;1110004F10RIK&quot;  &quot;1110008P14RIK&quot; 
##   [9] &quot;1110012L19RIK&quot;  &quot;1110017D15RIK&quot;  &quot;1110032A03RIK&quot;  &quot;1110038F14RIK&quot; 
##  [13] &quot;1110059E24RIK&quot;  &quot;1110059G10RIK&quot;  &quot;1110065P20RIK&quot;  &quot;1190002N15RIK&quot; 
##  [17] &quot;1190007I07RIK&quot;  &quot;1500011B03RIK&quot;  &quot;1500011K16RIK&quot;  &quot;1600002K03RIK&quot; 
##  [21] &quot;1600012H06RIK&quot;  &quot;1600014C10RIK&quot;  &quot;1700001L19RIK&quot;  &quot;1700003E16RIK&quot; 
##  [25] &quot;1700007K13RIK&quot;  &quot;1700012B09RIK&quot;  &quot;1700013F07RIK&quot;  &quot;1700016K19RIK&quot; 
##  [29] &quot;1700017B05RIK&quot;  &quot;1700020D05RIK&quot;  &quot;1700024G13RIK&quot;  &quot;1700025G04RIK&quot; 
##  [33] &quot;1700028P14RIK&quot;  &quot;1700029I15RIK&quot;  &quot;1700029J07RIK&quot;  &quot;1700030K09RIK&quot; 
##  [37] &quot;1700037C18RIK&quot;  &quot;1700037H04RIK&quot;  &quot;1700086D15RIK&quot;  &quot;1700088E04RIK&quot; 
##  [41] &quot;1700102P08RIK&quot;  &quot;1700109H08RIK&quot;  &quot;1700123O20RIK&quot;  &quot;1810009A15RIK&quot; 
##  [45] &quot;1810010H24RIK&quot;  &quot;1810013L24RIK&quot;  &quot;1810022K09RIK&quot;  &quot;1810030O07RIK&quot; 
##  [49] &quot;1810037I17RIK&quot;  &quot;1810043G02RIK&quot;  &quot;1810055G02RIK&quot;  &quot;2010005H15RIK&quot; 
##  [53] &quot;2010111I01RIK&quot;  &quot;2010300C02RIK&quot;  &quot;2010309G21RIK&quot;  &quot;2200002D01RIK&quot; 
##  [57] &quot;2210011C24RIK&quot;  &quot;2210016H18RIK&quot;  &quot;2210016L21RIK&quot;  &quot;2210408I21RIK&quot; 
##  [61] &quot;2300009A05RIK&quot;  &quot;2310007B03RIK&quot;  &quot;2310009B15RIK&quot;  &quot;2310011J03RIK&quot; 
##  [65] &quot;2310022A10RIK&quot;  &quot;2310022B05RIK&quot;  &quot;2310030G06RIK&quot;  &quot;2310033P09RIK&quot; 
##  [69] &quot;2310057M21RIK&quot;  &quot;2310061I04RIK&quot;  &quot;2410002F23RIK&quot;  &quot;2410004B18RIK&quot; 
##  [73] &quot;2410015M20RIK&quot;  &quot;2410131K14RIK&quot;  &quot;2510002D24RIK&quot;  &quot;2510009E07RIK&quot; 
##  [77] &quot;2600001M11RIK&quot;  &quot;2610001J05RIK&quot;  &quot;2610002M06RIK&quot;  &quot;2610008E11RIK&quot; 
##  [81] &quot;2610028H24RIK&quot;  &quot;2610301B20RIK&quot;  &quot;2610507B11RIK&quot;  &quot;2700049A03RIK&quot; 
##  [85] &quot;2700062C07RIK&quot;  &quot;2700081O15RIK&quot;  &quot;2700097O09RIK&quot;  &quot;2810004N23RIK&quot; 
##  [89] &quot;2810021J22RIK&quot;  &quot;2810408A11RIK&quot;  &quot;2900026A02RIK&quot;  &quot;3110001I22RIK&quot; 
##  [93] &quot;3110040N11RIK&quot;  &quot;3110082I17RIK&quot;  &quot;3830406C13RIK&quot;  &quot;4430402I18RIK&quot; 
##  [97] &quot;4732423E21RIK&quot;  &quot;4833414E09RIK&quot;  &quot;4833420G17RIK&quot;  &quot;4833427G06RIK&quot; 
## [101] &quot;4833439L19RIK&quot;  &quot;4921524J17RIK&quot;  &quot;4930402H24RIK&quot;  &quot;4930430F08RIK&quot; 
## [105] &quot;4930451G09RIK&quot;  &quot;4930453N24RIK&quot;  &quot;4930486L24RIK&quot;  &quot;4930503L19RIK&quot; 
## [109] &quot;4930523C07RIK&quot;  &quot;4930550C14RIK&quot;  &quot;4930562C15RIK&quot;  &quot;4931406C07RIK&quot; 
## [113] &quot;4931406P16RIK&quot;  &quot;4931414P19RIK&quot;  &quot;4932438A13RIK&quot;  &quot;4933408B17RIK&quot; 
## [117] &quot;4933415F23RIK&quot;  &quot;4933427D14RIK&quot;  &quot;4933434E20RIK&quot;  &quot;5330417C22RIK&quot; 
## [121] &quot;5430427O19RIK&quot;  &quot;5730480H06RIK&quot;  &quot;5830417I10RIK&quot;  &quot;6030458C11RIK&quot; 
## [125] &quot;6030468B19RIK&quot;  &quot;6330403K07RIK&quot;  &quot;6330417A16RIK&quot;  &quot;6430531B16RIK&quot; 
## [129] &quot;6430548M08RIK&quot;  &quot;6430550D23RIK&quot;  &quot;6720489N17RIK&quot;  &quot;8030462N17RIK&quot; 
## [133] &quot;9030624G23RIK&quot;  &quot;9130008F23RIK&quot;  &quot;9130019O22RIK&quot;  &quot;9130023H24RIK&quot; 
## [137] &quot;9130230L23RIK&quot;  &quot;9230104L09RIK&quot;  &quot;9330182L06RIK&quot;  &quot;9530068E07RIK&quot; 
## [141] &quot;9930012K11RIK&quot;  &quot;9930021J03RIK&quot;  &quot;9930111J21RIK1&quot; &quot;9930111J21RIK2&quot;
## [145] &quot;A130071D04RIK&quot;  &quot;A230050P20RIK&quot;  &quot;A2ML1&quot;          &quot;A430005L14RIK&quot; 
## [149] &quot;A430033K04RIK&quot;  &quot;A430078G23RIK&quot;  &quot;A530032D15RIK&quot;  &quot;A630001G21RIK&quot; 
## [153] &quot;A730034C02&quot;     &quot;A730049H05RIK&quot;  &quot;A830018L16RIK&quot;  &quot;A930002H24RIK&quot; 
## [157] &quot;A930004D18RIK&quot;  &quot;AA415038&quot;       &quot;AA986860&quot;       &quot;AACS&quot;          
## [161] &quot;AAED1&quot;          &quot;AAGAB&quot;          &quot;AAK1&quot;           &quot;AAMDC&quot;         
## [165] &quot;AAMP&quot;           &quot;AAR2&quot;           &quot;AARS&quot;           &quot;AARS2&quot;         
## [169] &quot;AARSD1&quot;         &quot;AASDH&quot;          &quot;AASS&quot;           &quot;AATF&quot;          
## [173] &quot;AATK&quot;           &quot;AB124611&quot;       &quot;ABCA1&quot;          &quot;ABCA17&quot;        
## [177] &quot;ABCA2&quot;          &quot;ABCA3&quot;          &quot;ABCA5&quot;          &quot;ABCA7&quot;         
## [181] &quot;ABCA8A&quot;         &quot;ABCA8B&quot;         &quot;ABCA9&quot;          &quot;ABCB10&quot;        
## [185] &quot;ABCB1A&quot;         &quot;ABCB1B&quot;         &quot;ABCB6&quot;          &quot;ABCB7&quot;         
## [189] &quot;ABCB8&quot;          &quot;ABCC1&quot;          &quot;ABCC3&quot;          &quot;ABCC4&quot;         
## [193] &quot;ABCC5&quot;          &quot;ABCD1&quot;          &quot;ABCD2&quot;          &quot;ABCD3&quot;         
## [197] &quot;ABCD4&quot;          &quot;ABCE1&quot;          &quot;ABCF1&quot;          &quot;ABCF3&quot;</code></pre>
<ol start="2" style="list-style-type: decimal">
<li>The list of genes identified with significant differential expression associated with flaming pine needles:</li>
</ol>
<div class="sourceCode" id="cb714"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb714-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb714-1" tabindex="-1"></a><span class="co"># Similar to the above script, but starting with the res$id object</span></span>
<span id="cb714-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb714-2" tabindex="-1"></a><span class="co"># and filtering for genes with padj &lt; 0.05</span></span>
<span id="cb714-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb714-3" tabindex="-1"></a></span>
<span id="cb714-4"><a href="omics-and-system-biology-transcriptomic-applications.html#cb714-4" tabindex="-1"></a>res.ordered <span class="ot">&lt;-</span> res[<span class="fu">order</span>(res<span class="sc">$</span>id, <span class="sc">-</span><span class="fu">abs</span>(res<span class="sc">$</span>log2FoldChange) ), ] <span class="co">#sort by id and reverse of abs(log2FC)</span></span>
<span id="cb714-5"><a href="omics-and-system-biology-transcriptomic-applications.html#cb714-5" tabindex="-1"></a>SigGenes <span class="ot">&lt;-</span> <span class="fu">toupper</span>(<span class="fu">as.character</span>(res.ordered[<span class="fu">which</span>(res.ordered<span class="sc">$</span>padj<span class="sc">&lt;</span>.<span class="dv">05</span>),<span class="st">&quot;id&quot;</span>])) <span class="co"># pulling the genes with padj &lt; 0.05</span></span>
<span id="cb714-6"><a href="omics-and-system-biology-transcriptomic-applications.html#cb714-6" tabindex="-1"></a>SigGenes <span class="ot">&lt;-</span> SigGenes[ <span class="sc">!</span><span class="fu">duplicated</span>(SigGenes)] <span class="co"># removing gene duplicates</span></span>
<span id="cb714-7"><a href="omics-and-system-biology-transcriptomic-applications.html#cb714-7" tabindex="-1"></a></span>
<span id="cb714-8"><a href="omics-and-system-biology-transcriptomic-applications.html#cb714-8" tabindex="-1"></a><span class="fu">length</span>(SigGenes) <span class="co"># viewing the length of this significant gene list</span></span></code></pre></div>
<pre><code>## [1] 488</code></pre>
<p>Therefore, this gene set includes 488 <em>unique</em> genes significantly associated with the Flaming Pine Needles condition, based on padj&lt;0.05.</p>
<ol start="3" style="list-style-type: decimal">
<li>The underlying KEGG pathway dataset.
Note that this file was simply downloaded from <a href="https://www.gsea-msigdb.org/gsea/msigdb/">MSigDB</a>, ready for upload as a .gmt file. Here, we use the ‘loadGSC’ function enabled through the PIANO package to upload and organize these pathways.</li>
</ol>
<div class="sourceCode" id="cb716"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb716-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb716-1" tabindex="-1"></a>KEGG_Pathways <span class="ot">&lt;-</span> <span class="fu">loadGSC</span>(<span class="at">file=</span><span class="st">&quot;Module6_2_Input/Module6_2_InputData3_KEGGv7.gmt&quot;</span>, <span class="at">type=</span><span class="st">&quot;gmt&quot;</span>)</span>
<span id="cb716-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb716-2" tabindex="-1"></a></span>
<span id="cb716-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb716-3" tabindex="-1"></a><span class="fu">length</span>(KEGG_Pathways<span class="sc">$</span>gsc) <span class="co"># viewing the number of biological pathways contained in the database</span></span></code></pre></div>
<pre><code>## [1] 186</code></pre>
<p>This KEGG pathway database therefore includes 186 biological pathways available to query</p>
<p>With these data inputs ready, we can now run the pathway enrichment analysis. The enrichment statistic that is commonly employed through the PIANO package is based of a hypergeometric test, run through the ‘runGSAhyper’ function. This returns a p-value for each gene set from which you can determine enrichment status.</p>
<div class="sourceCode" id="cb718"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb718-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb718-1" tabindex="-1"></a><span class="co"># Running the piano function based on the hypergeometric statistic</span></span>
<span id="cb718-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb718-2" tabindex="-1"></a>Results_GSA <span class="ot">&lt;-</span> piano<span class="sc">::</span><span class="fu">runGSAhyper</span>(<span class="at">genes=</span>SigGenes, <span class="at">universe=</span>Background,<span class="at">gsc=</span>KEGG_Pathways, <span class="at">gsSizeLim=</span><span class="fu">c</span>(<span class="dv">1</span>,<span class="cn">Inf</span>), <span class="at">adjMethod =</span> <span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb718-3"><a href="omics-and-system-biology-transcriptomic-applications.html#cb718-3" tabindex="-1"></a></span>
<span id="cb718-4"><a href="omics-and-system-biology-transcriptomic-applications.html#cb718-4" tabindex="-1"></a><span class="co"># Pulling the pathway enrichment results into a separate dataframe</span></span>
<span id="cb718-5"><a href="omics-and-system-biology-transcriptomic-applications.html#cb718-5" tabindex="-1"></a>PathwayResults <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(Results_GSA<span class="sc">$</span>resTab)</span>
<span id="cb718-6"><a href="omics-and-system-biology-transcriptomic-applications.html#cb718-6" tabindex="-1"></a></span>
<span id="cb718-7"><a href="omics-and-system-biology-transcriptomic-applications.html#cb718-7" tabindex="-1"></a><span class="co"># Viewing the top of these pathway enrichment results (which are not ordered at the moment)</span></span>
<span id="cb718-8"><a href="omics-and-system-biology-transcriptomic-applications.html#cb718-8" tabindex="-1"></a><span class="fu">head</span>(PathwayResults)</span></code></pre></div>
<pre><code>##                                                        p-value Adjusted p-value
## KEGG_N_GLYCAN_BIOSYNTHESIS                          0.77021314        1.0000000
## KEGG_OTHER_GLYCAN_DEGRADATION                       1.00000000        1.0000000
## KEGG_O_GLYCAN_BIOSYNTHESIS                          0.03553158        0.6656139
## KEGG_GLYCOSAMINOGLYCAN_DEGRADATION                  0.42056787        1.0000000
## KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE 1.00000000        1.0000000
## KEGG_GLYCEROLIPID_METABOLISM                        0.16241736        1.0000000
##                                                     Significant (in gene set)
## KEGG_N_GLYCAN_BIOSYNTHESIS                                                  1
## KEGG_OTHER_GLYCAN_DEGRADATION                                               0
## KEGG_O_GLYCAN_BIOSYNTHESIS                                                  3
## KEGG_GLYCOSAMINOGLYCAN_DEGRADATION                                          1
## KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE                         0
## KEGG_GLYCEROLIPID_METABOLISM                                                3
##                                                     Non-significant (in gene set)
## KEGG_N_GLYCAN_BIOSYNTHESIS                                                     34
## KEGG_OTHER_GLYCAN_DEGRADATION                                                  13
## KEGG_O_GLYCAN_BIOSYNTHESIS                                                     15
## KEGG_GLYCOSAMINOGLYCAN_DEGRADATION                                             12
## KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE                            12
## KEGG_GLYCEROLIPID_METABOLISM                                                   31
##                                                     Significant (not in gene set)
## KEGG_N_GLYCAN_BIOSYNTHESIS                                                    487
## KEGG_OTHER_GLYCAN_DEGRADATION                                                 488
## KEGG_O_GLYCAN_BIOSYNTHESIS                                                    485
## KEGG_GLYCOSAMINOGLYCAN_DEGRADATION                                            487
## KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE                           488
## KEGG_GLYCEROLIPID_METABOLISM                                                  485
##                                                     Non-significant (not in gene set)
## KEGG_N_GLYCAN_BIOSYNTHESIS                                                      11355
## KEGG_OTHER_GLYCAN_DEGRADATION                                                   11376
## KEGG_O_GLYCAN_BIOSYNTHESIS                                                      11374
## KEGG_GLYCOSAMINOGLYCAN_DEGRADATION                                              11377
## KEGG_GLYCOSAMINOGLYCAN_BIOSYNTHESIS_KERATAN_SULFATE                             11377
## KEGG_GLYCEROLIPID_METABOLISM                                                    11358</code></pre>
<p>This dataframe therefore summarizes the enrichment p-value for each pathway, FDR adjusted p-value, number of significant genes in the gene set that intersect with genes in the pathway, etc</p>
<p>With these results, let’s identify which pathways meet a statistical enrichment p-value filter of 0.05:</p>
<div class="sourceCode" id="cb720"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb720-1"><a href="omics-and-system-biology-transcriptomic-applications.html#cb720-1" tabindex="-1"></a>SigPathways <span class="ot">&lt;-</span> PathwayResults[<span class="fu">which</span>(PathwayResults<span class="sc">$</span><span class="st">`</span><span class="at">p-value</span><span class="st">`</span> <span class="sc">&lt;</span> <span class="fl">0.05</span>),]</span>
<span id="cb720-2"><a href="omics-and-system-biology-transcriptomic-applications.html#cb720-2" tabindex="-1"></a><span class="fu">rownames</span>(SigPathways)</span></code></pre></div>
<pre><code>##  [1] &quot;KEGG_O_GLYCAN_BIOSYNTHESIS&quot;                               
##  [2] &quot;KEGG_HYPERTROPHIC_CARDIOMYOPATHY_HCM&quot;                     
##  [3] &quot;KEGG_ARRHYTHMOGENIC_RIGHT_VENTRICULAR_CARDIOMYOPATHY_ARVC&quot;
##  [4] &quot;KEGG_PROTEASOME&quot;                                          
##  [5] &quot;KEGG_OOCYTE_MEIOSIS&quot;                                      
##  [6] &quot;KEGG_VASCULAR_SMOOTH_MUSCLE_CONTRACTION&quot;                  
##  [7] &quot;KEGG_WNT_SIGNALING_PATHWAY&quot;                               
##  [8] &quot;KEGG_HEDGEHOG_SIGNALING_PATHWAY&quot;                          
##  [9] &quot;KEGG_FOCAL_ADHESION&quot;                                      
## [10] &quot;KEGG_ECM_RECEPTOR_INTERACTION&quot;                            
## [11] &quot;KEGG_COMPLEMENT_AND_COAGULATION_CASCADES&quot;                 
## [12] &quot;KEGG_GNRH_SIGNALING_PATHWAY&quot;</code></pre>
<div id="answer-to-environmental-health-question-8" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 8<a href="omics-and-system-biology-transcriptomic-applications.html#answer-to-environmental-health-question-8" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can now answer <strong>Environmental Health Question #8</strong></em>: What biological pathways are disrupted in association with flaming pine needles exposure in the lung, identified through systems level analyses?</p>
</div>
<div class="answer">
<p><strong>Answer:</strong> Biological pathways involved in cardiopulmonary function (e.g., arrhythmogenic right ventricular cardiomyopathy, hypertrophic cardiomyopathy, vascular smooth muscle contraction), carcinogenesis signaling (e.g., Wnt signaling pathway, hedgehog signaling pathway), and hormone signaling (e.g., Gnrh signaling pathway), among others.</p>
</div>
<p><br></p>
</div>
</div>
<div id="concluding-remarks-24" class="section level2 hasAnchor">
<h2>Concluding Remarks<a href="omics-and-system-biology-transcriptomic-applications.html#concluding-remarks-24" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this module, users are guided through the uploading, organization, QA/QC, statistical analysis, and systems level analysis of an example -omics dataset based on transcriptomic responses to biomass burn scenarios, representing environmental exposure scenarios of growing concern worldwide. It is worth noting that the methods described herein represent a fraction of the approaches and tools that can be leveraged in the analysis of -omics datasets, and methods should be tailored to the purposes of each individual analysis’ goal. For additional example research projects that have leveraged -omics and systems biology to address environmental health questions, see the following select relevant publications:</p>
<p><strong>Genomic publications evaluating gene-environment interactions and relations to disease etiology:</strong></p>
<ul>
<li><p>Balik-Meisner M, Truong L, Scholl EH, La Du JK, Tanguay RL, Reif DM. Elucidating Gene-by-Environment Interactions Associated with Differential Susceptibility to Chemical Exposure. Environ Health Perspect. 2018 Jun 28;126(6):067010. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/29968567/">29968567</a>.</p></li>
<li><p>Ward-Caviness CK, Neas LM, Blach C, Haynes CS, LaRocque-Abramson K, Grass E, Dowdy ZE, Devlin RB, Diaz-Sanchez D, Cascio WE, Miranda ML, Gregory SG, Shah SH, Kraus WE, Hauser ER. A genome-wide trans-ethnic interaction study links the PIGR-FCAMR locus to coronary atherosclerosis via interactions between genetic variants and residential exposure to traffic. PLoS One. 2017 Mar 29;12(3):e0173880. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/28355232/">28355232</a>.</p></li>
</ul>
<p><strong>Transcriptomic publications evaluating gene expression responses to environmental exposures and relations to disease etiology:</strong></p>
<ul>
<li><p>Chang Y, Rager JE, Tilton SC. Linking Coregulated Gene Modules with Polycyclic Aromatic Hydrocarbon-Related Cancer Risk in the 3D Human Bronchial Epithelium. Chem Res Toxicol. 2021 Jun 21;34(6):1445-1455. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/34048650/">34048650</a>.</p></li>
<li><p>Chappell GA, Rager JE, Wolf J, Babic M, LeBlanc KJ, Ring CL, Harris MA, Thompson CM. Comparison of Gene Expression Responses in the Small Intestine of Mice Following Exposure to 3 Carcinogens Using the S1500+ Gene Set Informs a Potential Common Adverse Outcome Pathway. Toxicol Pathol. 2019 Oct;47(7):851-864. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/31558096/">31558096</a>.</p></li>
<li><p>Manuck TA, Eaves LA, Rager JE, Fry RC. Mid-pregnancy maternal blood nitric oxide-related gene and miRNA expression are associated with preterm birth. Epigenomics. 2021 May;13(9):667-682. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/33890487/">33890487</a>.</p></li>
</ul>
<p><strong>Epigenomic publications</strong> evaluating microRNA, CpG methylation, and/or histone methylation responses to environmental exposures and relations to disease etiology:</p>
<ul>
<li><p>Chappell GA, Rager JE. Epigenetics in chemical-induced genotoxic carcinogenesis. Curr Opinion Toxicol. <a href="https://www.sciencedirect.com/science/article/abs/pii/S2468202017300396">2017 Oct; 6:10-17</a>.</p></li>
<li><p>Rager JE, Bailey KA, Smeester L, Miller SK, Parker JS, Laine JE, Drobná Z, Currier J, Douillet C, Olshan AF, Rubio-Andrade M, Stýblo M, García-Vargas G, Fry RC. Prenatal arsenic exposure and the epigenome: altered microRNAs associated with innate and adaptive immune signaling in newborn cord blood. Environ Mol Mutagen. 2014 Apr;55(3):196-208. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/24327377/">24327377</a>.</p></li>
<li><p>Rager JE, Bauer RN, Müller LL, Smeester L, Carson JL, Brighton LE, Fry RC, Jaspers I. DNA methylation in nasal epithelial cells from smokers: identification of ULBP3-related effects. Am J Physiol Lung Cell Mol Physiol. 2013 Sep 15;305(6):L432-8. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/23831618/">23831618</a>.</p></li>
<li><p>Smeester L, Rager JE, Bailey KA, Guan X, Smith N, García-Vargas G, Del Razo LM, Drobná Z, Kelkar H, Stýblo M, Fry RC. Epigenetic changes in individuals with arsenicosis. Chem Res Toxicol. 2011 Feb 18;24(2):165-7. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/21291286/">21291286</a>.</p></li>
</ul>
<p><strong>Metabolomic publications</strong> evaluating changes in the metabolome in response to environmental exposures and involved in disease etiology:</p>
<ul>
<li><p>Lu K, Abo RP, Schlieper KA, Graffam ME, Levine S, Wishnok JS, Swenberg JA, Tannenbaum SR, Fox JG. Arsenic exposure perturbs the gut microbiome and its metabolic profile in mice: an integrated metagenomics and metabolomics analysis. Environ Health Perspect. 2014 Mar;122(3):284-91. PMID: 24413286; PMCID: <a href="https://pubmed.ncbi.nlm.nih.gov/24413286/">PMC3948040</a>.</p></li>
<li><p>Manuck TA, Lai Y, Ru H, Glover AV, Rager JE, Fry RC, Lu K. Metabolites from midtrimester plasma of pregnant patients at high risk for preterm birth. Am J Obstet Gynecol MFM. 2021 Jul;3(4):100393. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/33991707/">33991707</a>.</p></li>
</ul>
<p><strong>Microbiome publications</strong> evaluating changes in microbiome profiles in relation to the environment and human disease:</p>
<ul>
<li><p>Chi L, Bian X, Gao B, Ru H, Tu P, Lu K. Sex-Specific Effects of Arsenic Exposure on the Trajectory and Function of the Gut Microbiome. Chem Res Toxicol. 2016 Jun 20;29(6):949-51.PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/27268458/">27268458</a>.</p></li>
<li><p>Cho I, Blaser MJ. The human microbiome: at the interface of health and disease. Nat Rev Genet. 2012 Mar 13;13(4):260-70. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/22411464/">22411464</a>.</p></li>
<li><p>Lu K, Abo RP, Schlieper KA, Graffam ME, Levine S, Wishnok JS, Swenberg JA, Tannenbaum SR, Fox JG. Arsenic exposure perturbs the gut microbiome and its metabolic profile in mice: an integrated metagenomics and metabolomics analysis. Environ Health Perspect. 2014 Mar;122(3):284-91. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/24413286/">24413286</a>.</p></li>
</ul>
<p><strong>Exposome publications</strong> evaluating changes in chemical signatures in relation to the environment and human disease:</p>
<ul>
<li><p>Rager JE, Strynar MJ, Liang S, McMahen RL, Richard AM, Grulke CM, Wambaugh JF, Isaacs KK, Judson R, Williams AJ, Sobus JR. Linking high resolution mass spectrometry data with exposure and toxicity forecasts to advance high-throughput environmental monitoring. Environ Int. 2016 Mar;88:269-280. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/26812473/">26812473</a>.</p></li>
<li><p>Rappaport SM, Barupal DK, Wishart D, Vineis P, Scalbert A. The blood exposome and its role in discovering causes of disease. Environ Health Perspect. 2014 Aug;122(8):769-74. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/24659601/">24659601</a>.</p></li>
<li><p>Viet SM, Falman JC, Merrill LS, Faustman EM, Savitz DA, Mervish N, Barr DB, Peterson LA, Wright R, Balshaw D, O’Brien B. Human Health Exposure Analysis Resource (HHEAR): A model for incorporating the exposome into health studies. Int J Hyg Environ Health. 2021 Jun;235:113768. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/34034040/">34034040</a>.</p></li>
</ul>
<p><br> <br></p>
<p><label class="tykfont">
Test Your Knowledge
</label></p>
<p>:::tyk</p>
<p>Using “Module6_2_TYKInput1.csv” (gene counts) and “Module6_2_TYKInput2.csv” (sample info) data sets, which have already been run through the QC process described in this module and are ready for analysis:</p>
<ol style="list-style-type: decimal">
<li>Conduct a differential expression analysis associated with “Season” using DESeq2. (Don’t worry about including any covariates or using RUV).</li>
<li>Find the number of significant differentially expressed genes associated with “Season”, at the .05 level.</li>
</ol>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="descriptive-cohort-analyses.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
