<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>6.6 Toxicokinetic Modeling | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="6.6 Toxicokinetic Modeling | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="6.6 Toxicokinetic Modeling | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  <meta name="twitter:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  

<meta name="author" content="Rager Lab" />


<meta name="date" content="2024-10-16" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="Icons_Used_Throughout/Favicon.png" type="image/x-icon" />
<link rel="prev" href="mixtures-analysis-methods-part-3-sufficient-similarity.html"/>
<link rel="next" href="chemical-read-across-for-toxicity-predictions.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>
<link href="libs/table1-1.0/table1_defaults.css" rel="stylesheet" />
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-FGFKVX2X3G"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-FGFKVX2X3G');
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="Icons_Used_Throughout/TAME 2.0 Logo.png"><br></a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Chapter 1 Introductory <br>Data Science</b></span></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html"><i class="fa fa-check"></i>FAIR Data Management Practices</a>
<ul>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-training-module"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-fair"><i class="fa fa-check"></i>Introduction to FAIR</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#breaking-down-fair-letter-by-letter"><i class="fa fa-check"></i>Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#what-does-this-mean-for-you"><i class="fa fa-check"></i>What Does This Mean for You?</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#data-repositories-for-sharing-of-data"><i class="fa fa-check"></i>Data Repositories for Sharing of Data</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#recent-shifts-in-regulatory-policies-for-data-sharing"><i class="fa fa-check"></i>Recent Shifts in Regulatory Policies for Data Sharing</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#additional-training-resources-on-fair"><i class="fa fa-check"></i>Additional Training Resources on FAIR</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html"><i class="fa fa-check"></i>Data Sharing through Online Repositories</a>
<ul>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#an-overview-and-example-with-the-dataverse-repository"><i class="fa fa-check"></i>An Overview and Example with the Dataverse Repository</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#introduction-to-training-module-1"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#data-repositories"><i class="fa fa-check"></i>Data Repositories</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#the-dataverse-project"><i class="fa fa-check"></i>The Dataverse Project</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#what-is-a-dataverse"><i class="fa fa-check"></i>What is a Dataverse?</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#metadata"><i class="fa fa-check"></i>Metadata</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataverse"><i class="fa fa-check"></i>Creating a Dataverse</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataset"><i class="fa fa-check"></i>Creating a Dataset</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#concluding-remarks-1"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html"><i class="fa fa-check"></i>File Management using Github</a>
<ul>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#introduction-to-training-module-2"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-an-account"><i class="fa fa-check"></i>Creating an Account</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-a-repository"><i class="fa fa-check"></i>Creating a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#uploading-code"><i class="fa fa-check"></i>Uploading Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#adding-subfolders-in-a-repository"><i class="fa fa-check"></i>Adding Subfolders in a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-code"><i class="fa fa-check"></i>Updating Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-repository-titles-and-structure-to-support-a-manuscript"><i class="fa fa-check"></i>Updating Repository Titles and Structure to Support a Manuscript</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#tracking-code-changes-using-github-branches"><i class="fa fa-check"></i>Tracking Code Changes using Github Branches</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#concluding-remarks-2"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html"><i class="fa fa-check"></i>Data Wrangling in Excel</a>
<ul>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#introduction-to-training-module-3"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#save-a-copy-of-the-soon-to-be-organized-and-cleaned-dataset-as-a-new-file"><i class="fa fa-check"></i>Save a Copy of the Soon-To-Be Organized and Cleaned Dataset as a New File</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-extraneous-white-space"><i class="fa fa-check"></i>Remove Extraneous White Space</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#replace-missing-data-with-na"><i class="fa fa-check"></i>Replace Missing Data with “NA”</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-metadata-tab"><i class="fa fa-check"></i>Create a Metadata Tab</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#abbreviate-and-capitalize-categorical-data"><i class="fa fa-check"></i>Abbreviate and Capitalize Categorical Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#alphabetize-sort-the-data-by-the-categorical-variable-of-interest"><i class="fa fa-check"></i>Alphabetize (Sort) the Data by the Categorical Variable of Interest</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-new-subject-number-column"><i class="fa fa-check"></i>Create a New Subject Number Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-special-symbols-and-dashes"><i class="fa fa-check"></i>Remove Special Symbols and Dashes</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#bold-all-column-names-and-center-all-data"><i class="fa fa-check"></i>Bold all Column Names and Center all Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-subject-identifier-column"><i class="fa fa-check"></i>Create a Subject Identifier Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#separate-subject-demographic-data-from-experimental-measurements"><i class="fa fa-check"></i>Separate Subject Demographic Data from Experimental Measurements</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#convert-data-from-wide-to-long-format"><i class="fa fa-check"></i>Convert Data from Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#pivoting-data-from-a-wide-to-long-format"><i class="fa fa-check"></i>Pivoting Data from a Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#generating-summary-level-statistics-with-pivot-tables"><i class="fa fa-check"></i>Generating Summary-Level Statistics with Pivot Tables</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#excel-vs.-r-which-should-you-use"><i class="fa fa-check"></i>Excel vs. R: Which Should You Use?</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#concluding-remarks-3"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 2 Coding in R</b></span></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html"><i class="fa fa-check"></i>Downloading and Programming in R</a>
<ul>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-training-module-4"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#general-introduction-and-installation-of-r-and-rstudio"><i class="fa fa-check"></i>General Introduction and Installation of R and RStudio</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-r-packages"><i class="fa fa-check"></i>Introduction to R Packages</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#scripting-basics"><i class="fa fa-check"></i>Scripting Basics</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#code-troubleshooting"><i class="fa fa-check"></i>Code Troubleshooting</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#concluding-remarks-4"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html"><i class="fa fa-check"></i>Coding “Best” Practices</a>
<ul>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#introduction-to-training-module-5"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#scripting-file-types"><i class="fa fa-check"></i>Scripting File Types</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-headers-and-annotation"><i class="fa fa-check"></i>Script Headers and Annotation</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#coding-style"><i class="fa fa-check"></i>Coding Style</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-organization"><i class="fa fa-check"></i>Script Organization</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#concluding-remarks-5"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html"><i class="fa fa-check"></i>Data Manipulation and Reshaping</a>
<ul>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-training-module-6"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#data-manipulation-using-base-r"><i class="fa fa-check"></i>Data Manipulation Using Base R</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-tidyverse"><i class="fa fa-check"></i>Introduction to Tidyverse</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#concluding-remarks-6"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html"><i class="fa fa-check"></i>Improving Coding Efficiencies</a>
<ul>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#introduction-to-training-module-7"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#loops"><i class="fa fa-check"></i>Loops</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#functions"><i class="fa fa-check"></i>Functions</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#list-operations"><i class="fa fa-check"></i>List operations</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#concluding-remarks-7"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#additional-resources-2"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="part"><span><b>Chapter 3 Basics of <br>Data Analysis and Visualizations</b></span></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html"><i class="fa fa-check"></i>Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-data-visualizations"><i class="fa fa-check"></i>Introduction to Data Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-training-module-8"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#density-plot-visualization"><i class="fa fa-check"></i>Density Plot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#boxplot-visualization"><i class="fa fa-check"></i>Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#correlation-visualizations"><i class="fa fa-check"></i>Correlation Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#heatmap-visualization"><i class="fa fa-check"></i>Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#concluding-remarks-8"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html"><i class="fa fa-check"></i>Improving Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-data-visulization-conventions"><i class="fa fa-check"></i>Introduction to Data Visulization Conventions</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-training-module-9"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-boxplot-visualization"><i class="fa fa-check"></i>Creating an Improved Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-heatmap-visualization"><i class="fa fa-check"></i>Creating an Improved Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-multi-plot-figures"><i class="fa fa-check"></i>Creating Multi-Plot Figures</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#concluding-remarks-9"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html"><i class="fa fa-check"></i>Normality Tests and Data Transformations</a>
<ul>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#introduction-to-training-module-10"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#what-is-a-normal-distribution"><i class="fa fa-check"></i>What is a Normal Distribution?</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#qualitative-assessment-of-normality"><i class="fa fa-check"></i>Qualitative Assessment of Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#quantitative-normality-assessment"><i class="fa fa-check"></i>Quantitative Normality Assessment</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#data-transformation"><i class="fa fa-check"></i>Data Transformation</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#additional-considerations-regarding-normality"><i class="fa fa-check"></i>Additional Considerations Regarding Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#concluding-remarks-10"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html"><i class="fa fa-check"></i>Intoduction to Statistical Tests</a>
<ul>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#introduction-to-training-module-11"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#assessing-normality-homogeneity-of-variance"><i class="fa fa-check"></i>Assessing Normality &amp; Homogeneity of Variance</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#two-group-visualizations-and-statistical-comparisons-using-the-t-test"><i class="fa fa-check"></i>Two-Group Visualizations and Statistical Comparisons using the T-Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#three-group-visualizations-and-statistical-comparisons-using-an-anova"><i class="fa fa-check"></i>Three-Group Visualizations and Statistical Comparisons using an ANOVA</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#regression-modeling-and-visualization-linear-and-logistic-regressions"><i class="fa fa-check"></i>Regression Modeling and Visualization: Linear and Logistic Regressions</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#statistical-evaluations-of-categorical-data-using-the-chi-squared-test-and-fishers-exact-test"><i class="fa fa-check"></i>Statistical Evaluations of Categorical Data using the Chi-Squared Test and Fisher’s Exact Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#concluding-remarks-11"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 4 Converting Wet Lab Data into Dry Lab Analyses</b></span></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html"><i class="fa fa-check"></i>Overview of Experimental Design and Example Data</a>
<ul>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#introduction-to-training-module-12"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#replicates"><i class="fa fa-check"></i>Replicates</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#orientation-to-example-data-for-chapter-4"><i class="fa fa-check"></i>Orientation to Example Data for Chapter 4</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#concluding-remarks-12"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html"><i class="fa fa-check"></i>Data Import, Processing, and Summary Statistics</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#introduction-to-training-module-13"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#data-import"><i class="fa fa-check"></i>Data Import</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#handling-missing-values"><i class="fa fa-check"></i>Handling Missing Values</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#averaging-replicates"><i class="fa fa-check"></i>Averaging Replicates</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#descriptive-statistics"><i class="fa fa-check"></i>Descriptive Statistics</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#normality-assessment-and-data-transformation"><i class="fa fa-check"></i>Normality Assessment and Data Transformation</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#concluding-remarks-13"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html"><i class="fa fa-check"></i>Data Import from PDF Sources</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#introduction-to-training-module-14"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-from-many-single-pdfs-with-the-same-formatting"><i class="fa fa-check"></i>Importing Data from Many Single PDFs with the Same Formatting</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-stored-in-pdf-tables"><i class="fa fa-check"></i>Importing Data Stored in PDF Tables</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#concluding-remarks-14"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html"><i class="fa fa-check"></i>Two-Group Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#introduction-to-training-module-15"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#overview-of-two-group-statistical-tests"><i class="fa fa-check"></i>Overview of Two Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#statistical-vs-biological-significance"><i class="fa fa-check"></i>Statistical vs Biological Significance</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#unpaired-test-example"><i class="fa fa-check"></i>Unpaired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#paired-test-example"><i class="fa fa-check"></i>Paired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#visualizing-results"><i class="fa fa-check"></i>Visualizing Results</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#concluding-remarks-15"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html"><i class="fa fa-check"></i>Multi-Group and Multi-Variable Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#introduction-to-training-module-16"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#overview-of-multi-group-statistical-tests"><i class="fa fa-check"></i>Overview of Multi-Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#multi-group-analysis-example"><i class="fa fa-check"></i>Multi-Group Analysis Example</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#visualization-of-multi-group-statistical-results"><i class="fa fa-check"></i>Visualization of Multi-Group Statistical Results</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#concluding-remarks-16"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 5 Machine Learning &amp; Artificial Intelligence</b></span></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><i class="fa fa-check"></i>Introduction to Artificial Intelligence, Machine Learning, and Predictive Modeling for Environmental Health</a>
<ul>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#introduction-to-training-module-17"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#general-historical-context-and-taxonomy-of-modern-aiml"><i class="fa fa-check"></i>General Historical Context and Taxonomy of Modern AI/ML</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#application-of-machine-learning-in-environmental-health-science"><i class="fa fa-check"></i>Application of Machine Learning in Environmental Health Science</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#concluding-remarks-17"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html"><i class="fa fa-check"></i>Supervised Machine Learning</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-training-module-18"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-machine-learning"><i class="fa fa-check"></i>Types of Machine Learning</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-supervised-machine-learning-algorithms"><i class="fa fa-check"></i>Types of Supervised Machine Learning Algorithms</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#training-supervised-machine-learning-models"><i class="fa fa-check"></i>Training Supervised Machine Learning Models</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#assessing-classification-based-model-performance"><i class="fa fa-check"></i>Assessing Classification-Based Model Performance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-activity-and-example-dataset"><i class="fa fa-check"></i>Introduction to Activity and Example Dataset</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#testing-for-differences-in-predictor-variables-across-the-outcome-classes"><i class="fa fa-check"></i>Testing for Differences in Predictor Variables across the Outcome Classes</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#predicting-ias-detection-with-a-random-forest-rf-model"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#class-imbalance"><i class="fa fa-check"></i>Class Imbalance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#concluding-remarks-18"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html"><i class="fa fa-check"></i>Supervised Machine Learning Model Interpretation</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-training-module-19"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance"><i class="fa fa-check"></i>Variable Importance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary"><i class="fa fa-check"></i>Decision Boundary</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-example-dataset-and-activity"><i class="fa fa-check"></i>Introduction to Example Dataset and Activity</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#predicting-ias-detection-with-a-random-forest-rf-model-1"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance-plot"><i class="fa fa-check"></i>Variable Importance Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot"><i class="fa fa-check"></i>Decision Boundary Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot-incorporating-smote"><i class="fa fa-check"></i>Decision Boundary Plot Incorporating SMOTE</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#concluding-remarks-19"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><i class="fa fa-check"></i>Unsupervised Machine Learning Part 1: K-Means Clustering &amp; PCA</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-training-module-20"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-unsupervised-machine-learning"><i class="fa fa-check"></i>Introduction to Unsupervised Machine Learning</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-example-data"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#identifying-clusters-of-chemicals-through-k-means"><i class="fa fa-check"></i>Identifying Clusters of Chemicals through <em>K</em>-Means</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#principal-component-analysis-pca-1"><i class="fa fa-check"></i>Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#incorporating-k-means-into-pca-for-predictive-modeling"><i class="fa fa-check"></i>Incorporating <em>K</em>-Means into PCA for Predictive Modeling</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#concluding-remarks-20"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><i class="fa fa-check"></i>Unsupervised Machine Learning Part 2: Additional Clustering Applications</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-training-module-21"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#k-means-clustering-1"><i class="fa fa-check"></i><em>K</em>-Means Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-example-data-1"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-1"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#clustering-plot"><i class="fa fa-check"></i>Clustering Plot</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-visualization"><i class="fa fa-check"></i>Hierarchical Clustering Visualization</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#variable-contributions"><i class="fa fa-check"></i>Variable Contributions</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#concluding-remarks-21"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 6 Applications in Toxicology &amp; Exposure Science</b></span></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html"><i class="fa fa-check"></i>6.1 Descriptive Cohort Analyses</a>
<ul>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#introduction-to-training-module-22"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#participant-exploration"><i class="fa fa-check"></i>Participant Exploration</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#chemical-detection-filtering"><i class="fa fa-check"></i>Chemical Detection Filtering</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#outlier-identification"><i class="fa fa-check"></i>Outlier Identification</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#summary-statistics-tables"><i class="fa fa-check"></i>Summary Statistics Tables</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#demographics-table"><i class="fa fa-check"></i>Demographics Table</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#concluding-remarks-22"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html"><i class="fa fa-check"></i>6.2 -Omics and System Biology: Transcriptomic Applications</a>
<ul>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-training-module-23"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-the-field-of--omics"><i class="fa fa-check"></i>Introduction to the Field of “-Omics”</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-transcriptomics"><i class="fa fa-check"></i>Introduction to Transcriptomics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#formatting-data-for-downstream-statistics"><i class="fa fa-check"></i>Formatting Data for Downstream Statistics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#transcriptomics-data-qaqc"><i class="fa fa-check"></i>Transcriptomics Data QA/QC</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#controlling-for-sources-of-sample-heterogeneity"><i class="fa fa-check"></i>Controlling for Sources of Sample Heterogeneity</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e.g.-biomass-smoke-exposure"><i class="fa fa-check"></i>Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure)</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-ma-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using MA Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-volcano-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using Volcano Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis"><i class="fa fa-check"></i>Interpretting Findings at the Systems Level through Pathway Enrichment Analysis</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#concluding-remarks-23"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><i class="fa fa-check"></i>6.3 Mixtures Analysis Methods Part 1: Overview and Example with Quantile G-Computation</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-training-module-24"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#overview-of-mixtures-analysis"><i class="fa fa-check"></i>Overview of Mixtures Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-example-data-2"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#mixtures-model-with-standard-qqcomp"><i class="fa fa-check"></i>Mixtures Model with Standard qqcomp</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#concluding-remarks-24"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#additional-resources-10"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><i class="fa fa-check"></i>6.4 Mixtures Analysis Methods Part 2: Bayesian Kernel Machine Regression</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-training-module-25"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-example-data-3"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-bkmr"><i class="fa fa-check"></i>Introduction to BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#run-bkmr"><i class="fa fa-check"></i>Run BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#concluding-remarks-25"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#additional-resources-11"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html"><i class="fa fa-check"></i>6.5 Mixtures Analysis Methods Part 3: Sufficient Similarity</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-training-module-26"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-toxicant-and-dataset"><i class="fa fa-check"></i>Introduction to Toxicant and Dataset</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#workspace-preparation-and-data-import-8"><i class="fa fa-check"></i>Workspace Preparation and Data Import</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#chemistry-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Chemistry-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#toxicity-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Toxicity-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#comparing-chemistry-vs.-toxicity-sufficient-similarity-analyses"><i class="fa fa-check"></i>Comparing Chemistry vs. Toxicity Sufficient Similarity Analyses</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#concluding-remarks-26"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i>6.6 Toxicokinetic Modeling</a>
<ul>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-training-module-27"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-toxicokinetic-modeling"><i class="fa fa-check"></i>Introduction to Toxicokinetic Modeling</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#data-and-models-used-in-toxicokinetic-modeling-tk"><i class="fa fa-check"></i>Data and Models used in Toxicokinetic Modeling (TK)</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-toxicokinetics"><i class="fa fa-check"></i>Reverse Toxicokinetics</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#capturing-population-variability-in-toxicokinetics-and-uncertainty-in-chemical-specific-parameters"><i class="fa fa-check"></i>Capturing Population Variability in Toxicokinetics, and Uncertainty in Chemical-Specific Parameters</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-tk-calculating-administered-equivalent-doses-for-toxcast-bioactive-concentrations"><i class="fa fa-check"></i>Reverse TK: Calculating Administered Equivalent Doses for ToxCast Bioactive Concentrations</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#comparing-equivalent-doses-estimated-to-elicit-toxicity-hazard-to-external-exposure-estimates-exposure-for-chemical-prioritization-by-bioactivity-exposure-ratios-bers"><i class="fa fa-check"></i>Comparing Equivalent Doses Estimated to Elicit Toxicity (Hazard) to External Exposure Estimates (Exposure), for Chemical Prioritization by Bioactivity-Exposure Ratios (BERs)</a></li>
</ul></li>
<li class="part"><span><b>Chapter 7 Environmental Health Database Mining</b></span></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html"><i class="fa fa-check"></i>7.1 Comparative Toxicogenomics Database</a>
<ul>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#introduction-to-training-module-29"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#ctd-data-in-r"><i class="fa fa-check"></i>CTD Data in R</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#identifying-genes-under-epigenetic-control"><i class="fa fa-check"></i>Identifying Genes Under Epigenetic Control</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#concluding-remarks-29"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html"><i class="fa fa-check"></i>7.2 Gene Expression Omnibus</a>
<ul>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-training-module-30"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-geo"><i class="fa fa-check"></i>Introduction to GEO</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-example-data-4"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#geo-data-in-r"><i class="fa fa-check"></i>GEO Data in R</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#visualizing-data"><i class="fa fa-check"></i>Visualizing Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#statistical-analyses"><i class="fa fa-check"></i>Statistical Analyses</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#concluding-remarks-30"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus-1.html"><a href="gene-expression-omnibus-1.html"><i class="fa fa-check"></i>7.2 Gene Expression Omnibus</a>
<ul>
<li class="chapter" data-level="" data-path="gene-expression-omnibus-1.html"><a href="gene-expression-omnibus-1.html#introduction-to-training-module-31"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus-1.html"><a href="gene-expression-omnibus-1.html#introduction-to-geo-1"><i class="fa fa-check"></i>Introduction to GEO</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus-1.html"><a href="gene-expression-omnibus-1.html#introduction-to-example-data-5"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus-1.html"><a href="gene-expression-omnibus-1.html#geo-data-in-r-1"><i class="fa fa-check"></i>GEO Data in R</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus-1.html"><a href="gene-expression-omnibus-1.html#visualizing-data-1"><i class="fa fa-check"></i>Visualizing Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus-1.html"><a href="gene-expression-omnibus-1.html#statistical-analyses-1"><i class="fa fa-check"></i>Statistical Analyses</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus-1.html"><a href="gene-expression-omnibus-1.html#concluding-remarks-31"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus-2.html"><a href="gene-expression-omnibus-2.html"><i class="fa fa-check"></i>7.2 Gene Expression Omnibus</a>
<ul>
<li class="chapter" data-level="" data-path="gene-expression-omnibus-2.html"><a href="gene-expression-omnibus-2.html#introduction-to-training-module-32"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus-2.html"><a href="gene-expression-omnibus-2.html#introduction-to-geo-2"><i class="fa fa-check"></i>Introduction to GEO</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus-2.html"><a href="gene-expression-omnibus-2.html#introduction-to-example-data-6"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus-2.html"><a href="gene-expression-omnibus-2.html#geo-data-in-r-2"><i class="fa fa-check"></i>GEO Data in R</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus-2.html"><a href="gene-expression-omnibus-2.html#visualizing-data-2"><i class="fa fa-check"></i>Visualizing Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus-2.html"><a href="gene-expression-omnibus-2.html#statistical-analyses-2"><i class="fa fa-check"></i>Statistical Analyses</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus-2.html"><a href="gene-expression-omnibus-2.html#concluding-remarks-32"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="toxicokinetic-modeling" class="section level1 hasAnchor">
<h1>6.6 Toxicokinetic Modeling<a href="toxicokinetic-modeling.html#toxicokinetic-modeling" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This training module was developed by Caroline Ring and Lauren Koval.</p>
<p>All input files (script, data, and figures) can be downloaded from the <a href="https://github.com/UNCSRP/TAME2">UNC-SRP TAME2 GitHub website</a>.</p>
<p><em>Disclaimer: The views expressed in this document are those of the author and do not necessarily reflect the views or policies of the U.S. EPA.</em></p>
<div id="introduction-to-training-module-27" class="section level2 hasAnchor">
<h2>Introduction to Training Module<a href="toxicokinetic-modeling.html#introduction-to-training-module-27" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This module serves as an example to guide trainees through the basics of toxicokinetic (TK) modeling and how this type of modeling can be used in the high-throughput setting for environmental health research applications.</p>
<p>In this activity, the capabilities of a high-throughput toxicokinetic modeling package titled ‘httk’ are demonstrated on a suite of environmentally relevant chemicals. The httk R package implements high-throughput toxicokinetic modeling (hence, ‘httk’), including a generic physiologically based toxicokinetic (PBTK) model as well as tables of chemical-specific parameters needed to solve the model for hundreds of chemicals. In this activity, the capabilities of ‘httk’ are demonstrated and explored. Example modeling estimates are produced for the high interest environmental chemical, bisphenol-A. Then, an example script is provided to derive the plasma concentration at steady state for an example environmental chemical, bisphenol-A.</p>
<p>The concept of reverse toxicokinetics is explained and demonstrated, again using bisphenol-A as an example chemical.</p>
<p>This module then demonstrates the derivation of the bioactivity-exposure ratio (BER) across many chemicals leveraging the capabilities of httk, while incorporating exposure measures. BERs are particularly useful in the evaluation of chemical risk, as they take into account both toxicity (i.e., <em>in vitro</em> potency) and exposure rates, the two essential components used in risk calculations for chemical safety and prioritization evaluations. Therefore, the estimates of both potency and exposure and needed to calculate BERs, which are described in this training module.</p>
<p>For potency estimates, the ToxCast high-throughput screening library is introduced as an example high-throughput dataset to carry out in vitro to in vivo extrapolation (IVIVE) modeling through httk. ToxCast activity concentrations that elicit 50% maximal bioactivity (AC50) are uploaded and organized as inputs, and then the tenth percentile ToxCast AC50 is calculated for each chemical (in other words, across all ToxCast screening assays, the tenth percentile of AC50 values were carried forward). These concentration estimates then serve as concentration estimates for potency. For exposure estimates, previously generated exposure estimates that have been inferred from CDC NHANES urinary biomonitoring data are used.</p>
<p>The bioactivity-exposure ratio (BER) is then calculated across chemicals with both potency and exposure estimate information. This ratio is simply calculated as the ratio of the lower-end equivalent dose (for the most-sensitive 5% of the population) divided by the upper-end estimated exposure (here, the upper bound on the inferred population median exposure). Chemicals are then ranked based on resulting BERs and visualized through plots. The importance of these chemical prioritization are then discussed in relation to environmental health research and corresponding regulatory decisions.</p>
</div>
<div id="introduction-to-toxicokinetic-modeling" class="section level2 hasAnchor">
<h2>Introduction to Toxicokinetic Modeling<a href="toxicokinetic-modeling.html#introduction-to-toxicokinetic-modeling" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To understand what toxicokinetic modeling is, consider the following scenario:</p>
<p><img src="Module6_6_Input/Module6_6_Image1.png" width="1292" style="display: block; margin: auto;" /></p>
<p>Simply put, toxicokinetics answers these questions by describing “what the body does to the chemical” after an exposure scenario.</p>
<p>More technically, <strong>toxicokinetic modeling</strong> refers to the evaluation of the uptake and disposition of a chemical in the body.</p>
<div id="notes-on-terminology" class="section level3 hasAnchor">
<h3>Notes on terminology<a href="toxicokinetic-modeling.html#notes-on-terminology" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Pharmacokinetics (PK) is a synonym for toxicokinetics (TK). They are often used interchangeably. PK connotes pharmaceuticals; TK connotes environmental chemicals – but those connotations are weak.</p>
<p>A common abbreviation that you will also see in this research field is <strong>ADME</strong>, which stands for:<br />
<strong>Absorption:</strong> How does the chemical get absorbed into the body tissues?<br />
<strong>Distribution:</strong> Where does the chemical go inside the body?<br />
<strong>Metabolism:</strong> How do enzymes in the body break apart the chemical molecules?<br />
<strong>Excretion:</strong> How does the chemical leave the body?<br />
To place this term into the context of TK, TK models describe ADME mathematically by representing the body as compartments and flows.</p>
</div>
<div id="types-of-tk-models" class="section level3 hasAnchor">
<h3>Types of TK models<a href="toxicokinetic-modeling.html#types-of-tk-models" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>TK models describe the body mathematically as one or more “compartments” connected by “flows.” The compartments represent organs or tissues. Using mass balance equations, the amount or concentration of chemical in each compartment is described as a function of time.</p>
<p>Types of models discussed throughout this training module are described here.</p>
<div id="compartment-model" class="section level4 hasAnchor">
<h4>1 Compartment Model<a href="toxicokinetic-modeling.html#compartment-model" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>The simplest TK model is a 1-compartment model, where the body is assumed to be one big well-mixed compartment.</p>
</div>
<div id="compartment-model-1" class="section level4 hasAnchor">
<h4>3 Compartment Model<a href="toxicokinetic-modeling.html#compartment-model-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>A 3-compartment model mathematically incorporates three distinct body compartments, that can exhibit different parameters contributing to their individual mass balance. Commonly used compartments in 3-compartment modeling can include tissues like blood plasma, liver, gut, kidney, and/or ‘rest of body’ terms; though the specific compartments included depend on the chemical under evaluation, exposure scenario, and modeling assumptions.</p>
</div>
<div id="pbtk-model" class="section level4 hasAnchor">
<h4>PBTK Model<a href="toxicokinetic-modeling.html#pbtk-model" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>A physiologically-based TK (PBTK) model incorporates compartments and flows that represent real physiological quantities (as opposed to the aforementioned empirical 1- and 3-compartment models). PBTK models have more parameters overall, including parameters representing physiological quantities that are known <em>a priori</em> based on studies of anatomy. The only PBTK model parameters that need to be estimated for each new chemical are parameters representing chemical-body interactions, which can include the following:</p>
<ul>
<li>Rate of hepatic metabolism of chemical: How fast does liver break down chemical?</li>
<li>Plasma protein binding: How tightly does the chemical bind to proteins in blood plasma? Liver may not be able to break down chemical that is bound to plasma protein.</li>
<li>Blood:tissue partition coefficients: Assuming chemical diffuses between blood and other tissues very fast compared to the rate of blood flow, the ratio of concentration in blood to concentration in each tissue is approximately constant = partition coefficient.</li>
<li>Rate of active transport into/out of a tissue: If chemical moves between blood and tissues not just by passive diffusion, but by cells actively transporting it in or out of the tissue</li>
<li>Binding to other tissues: Some chemical may be bound inside a tissue and not available for diffusion or transport in/out</li>
</ul>
<p>Types of TK modeling can also fall into the following major categories:<br />
1. <strong>Forward TK Modeling:</strong> Where external exposure doses are converted into internal doses (or concentrations of chemicals/drugs in one or more body tissues of interest)<br />
2. <strong>Reverse TK Modeling:</strong> The reverse of the above, where internal doses are converted into external exposure doses.</p>
</div>
</div>
<div id="other-tk-modeling-resources" class="section level3 hasAnchor">
<h3>Other TK modeling resources<a href="toxicokinetic-modeling.html#other-tk-modeling-resources" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For further information on TK modeling background, math, and example models, there are additional resources online including a helpful course website on <a href="https://www.boomer.org/c/p4/">Basic Pharmacokinetics</a> by Dr. Bourne.</p>
</div>
<div id="script-preparations-8" class="section level3 hasAnchor">
<h3>Script Preparations<a href="toxicokinetic-modeling.html#script-preparations-8" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="cleaning-the-global-environment-8" class="section level4 hasAnchor">
<h4>Cleaning the global environment<a href="toxicokinetic-modeling.html#cleaning-the-global-environment-8" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb800"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb800-1"><a href="toxicokinetic-modeling.html#cb800-1" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list=</span><span class="fu">ls</span>())</span></code></pre></div>
</div>
<div id="installing-required-r-packages-9" class="section level4 hasAnchor">
<h4>Installing required R packages<a href="toxicokinetic-modeling.html#installing-required-r-packages-9" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>If you already have these packages installed, you can skip this step, or you can run the below code which checks installation status for you</p>
<div class="sourceCode" id="cb801"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb801-1"><a href="toxicokinetic-modeling.html#cb801-1" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">nzchar</span>(<span class="fu">system.file</span>(<span class="at">package =</span> <span class="st">&quot;ggplot2&quot;</span>))){</span>
<span id="cb801-2"><a href="toxicokinetic-modeling.html#cb801-2" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;ggplot2&quot;</span>)}</span>
<span id="cb801-3"><a href="toxicokinetic-modeling.html#cb801-3" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">nzchar</span>(<span class="fu">system.file</span>(<span class="at">package =</span> <span class="st">&quot;reshape2&quot;</span>))){</span>
<span id="cb801-4"><a href="toxicokinetic-modeling.html#cb801-4" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;reshape2&quot;</span>)}</span>
<span id="cb801-5"><a href="toxicokinetic-modeling.html#cb801-5" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">nzchar</span>(<span class="fu">system.file</span>(<span class="at">package =</span> <span class="st">&quot;stringr&quot;</span>))){</span>
<span id="cb801-6"><a href="toxicokinetic-modeling.html#cb801-6" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;stringr&quot;</span>)}</span>
<span id="cb801-7"><a href="toxicokinetic-modeling.html#cb801-7" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">nzchar</span>(<span class="fu">system.file</span>(<span class="at">package =</span> <span class="st">&quot;httk&quot;</span>))){</span>
<span id="cb801-8"><a href="toxicokinetic-modeling.html#cb801-8" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;httk&quot;</span>)}</span>
<span id="cb801-9"><a href="toxicokinetic-modeling.html#cb801-9" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">nzchar</span>(<span class="fu">system.file</span>(<span class="at">package =</span> <span class="st">&quot;eulerr&quot;</span>))){</span>
<span id="cb801-10"><a href="toxicokinetic-modeling.html#cb801-10" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;eulerr&quot;</span>)}</span></code></pre></div>
</div>
<div id="loading-r-packages-required-for-this-session-6" class="section level4 hasAnchor">
<h4>Loading R packages required for this session<a href="toxicokinetic-modeling.html#loading-r-packages-required-for-this-session-6" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb802"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb802-1"><a href="toxicokinetic-modeling.html#cb802-1" tabindex="-1"></a><span class="fu">library</span>(ggplot2) <span class="co"># ggplot2 will be used to generate associated graphics</span></span>
<span id="cb802-2"><a href="toxicokinetic-modeling.html#cb802-2" tabindex="-1"></a><span class="fu">library</span>(reshape2) <span class="co"># reshape2 will be used to organize and transform datasets</span></span>
<span id="cb802-3"><a href="toxicokinetic-modeling.html#cb802-3" tabindex="-1"></a><span class="fu">library</span>(stringr) <span class="co"># stringr will be used to aid in various data manipulation steps through this module</span></span>
<span id="cb802-4"><a href="toxicokinetic-modeling.html#cb802-4" tabindex="-1"></a><span class="fu">library</span>(httk) <span class="co"># httk package will be used to carry out all toxicokinetic modeling steps</span></span>
<span id="cb802-5"><a href="toxicokinetic-modeling.html#cb802-5" tabindex="-1"></a><span class="fu">library</span>(eulerr) <span class="co">#eulerr package will be used to generate Venn/Euler diagram graphics</span></span></code></pre></div>
<p>For more information on the <strong>ggplot2 package</strong>, see its associated <a href="https://cran.r-project.org/web/packages/ggplot2/index.html">CRAN webpage</a> and <a href="https://www.rdocumentation.org/packages/ggplot2/versions/3.3.5">RDocumentation webpage</a>.</p>
<p>For more information on the <strong>reshape2 package</strong>, see its associated <a href="https://cran.r-project.org/web/packages/reshape2/index.html">CRAN webpage</a> and <a href="https://www.rdocumentation.org/packages/reshape2/versions/1.4.4">RDocumentation webpage</a>.</p>
<p>For more information on the <strong>stringr package</strong>, see its associated <a href="https://cran.r-project.org/web/packages/stringr/index.html">CRAN webpage</a> and <a href="https://www.rdocumentation.org/packages/stringr/versions/1.4.0">RDocumentation webpage</a>.</p>
<p>For more information on the <strong>httk package</strong>, see its associated <a href="https://cran.r-project.org/web/packages/httk/index.html">CRAN webpage</a> and parent publication by <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6134854/">Pearce et al. (2017)</a>.</p>
</div>
<div id="more-information-on-the-httk-package" class="section level4 hasAnchor">
<h4>More information on the httk package<a href="toxicokinetic-modeling.html#more-information-on-the-httk-package" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>You can see an overview of the <code>httk</code> package by typing <code>?httk</code> at the R command line.</p>
<p>You can see a browsable index of all functions in the <code>httk</code> package by typing <code>help(package="httk")</code> at the R command line.</p>
<p>You can see a browsable list of vignettes by typing <code>browseVignettes("httk")</code> at the R command line. (Please note that some of these vignettes were written using older versions of the package and may no longer work as written – specifically the Ring (2017) vignette, which I wrote back in 2016. The <code>httk</code> team is actively working on updating these.)</p>
<p>You can get information about any function in <code>httk</code>, or indeed any function in any R package, by typing <code>help()</code> and placing the function name in quotation marks inside the parentheses. For example, to get information about the <code>httk</code> function <code>solve_model()</code>, type this:</p>
<div class="sourceCode" id="cb803"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb803-1"><a href="toxicokinetic-modeling.html#cb803-1" tabindex="-1"></a><span class="fu">help</span>(<span class="st">&quot;solve_model&quot;</span>)</span></code></pre></div>
<p>Note that this module was run with <code>httk</code> version 2.4.0.</p>
</div>
<div id="set-your-working-directory-11" class="section level4 hasAnchor">
<h4>Set your working directory<a href="toxicokinetic-modeling.html#set-your-working-directory-11" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb804"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb804-1"><a href="toxicokinetic-modeling.html#cb804-1" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/filepath to where your input files are&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="training-modules-environmental-health-questions-19" class="section level3 hasAnchor">
<h3>Training Module’s Environmental Health Questions<a href="toxicokinetic-modeling.html#training-modules-environmental-health-questions-19" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This training module was specifically developed to answer the following environmental health questions:</p>
<ol style="list-style-type: decimal">
<li><p>After solving the TK model that evaluates bisphenol-A, what is the maximum concentration of bisphenol-A estimated to occur in human plasma, after 1 exposure dose of 1 mg/kg/day?</p></li>
<li><p>After solving the TK model that evaluates bisphenol-A, what is the steady-state concentration of bisphenol-A estimated to occur in human plasma, for a long-term oral infusion dose of 1 mg/kg/day?</p></li>
<li><p>What is the predicted range of bisphenol-A concentrations in plasma that can occur in a human population, assuming a long-term exposure rate of 1 mg/kg/day and steady-state conditions? Provide estimates at the 5th, 50th, and 95th percentile?</p></li>
<li><p>Considering the chemicals evaluated in the above TK modeling example, do the <span class="math inline">\(C_{ss}\)</span>-dose slope distributions become wider as the median <span class="math inline">\(C_{ss}\)</span>-dose slope increases?</p></li>
<li><p>How many chemicals have available AC50 values to evaluate in the current ToxCast/Tox21 high-throughput screening database?</p></li>
<li><p>What are the chemicals with the three lowest predicted equivalent doses (for tenth-percentile ToxCast AC50s), for the most-sensitive 5% of the population?</p></li>
<li><p>Based on httk modeling estimates, are chemicals with higher bioactivity-exposure ratios always less potent than chemicals with lower bioactivity-exposure ratios?</p></li>
<li><p>Based on httk modeling estimates, do chemicals with higher bioactivity-exposure ratios always have lower estimated exposures than chemicals with lower bioactivity-exposure ratios?</p></li>
<li><p>How are chemical prioritization results different when using only hazard information vs. only exposure information vs. bioactivity-exposure ratios?</p></li>
<li><p>Of the three data sets used in this training module – bioactivity from ToxCast, TK data from <code>httk</code>, and exposure inferred from NHANES urinary biomonitoring – which one most limits the number of chemicals that can be prioritized using BERs?</p></li>
</ol>
</div>
</div>
<div id="data-and-models-used-in-toxicokinetic-modeling-tk" class="section level2 hasAnchor">
<h2>Data and Models used in Toxicokinetic Modeling (TK)<a href="toxicokinetic-modeling.html#data-and-models-used-in-toxicokinetic-modeling-tk" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="common-models-used-in-tk-modeling-that-are-provided-as-built-in-models-in-httk" class="section level3 hasAnchor">
<h3>Common Models used in TK Modeling, that are Provided as Built-in Models in ‘httk’<a href="toxicokinetic-modeling.html#common-models-used-in-tk-modeling-that-are-provided-as-built-in-models-in-httk" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>There are five TK models currently built into <code>httk</code>. They are:</p>
<ul>
<li><code>pbtk</code>: A physiologically-based TK model with oral absorption. Contains the following compartments: gutlumen, gut, liver, kidneys, veins, arteries, lungs, and the rest of the body. Chemical is metabolized by the liver and excreted by the kidneys via glomerular filtration.</li>
<li><code>gas_pbtk</code>: A PBTK model with absorption via inhalation. Contains the same compartments as <code>pbtk</code>.</li>
<li><code>1compartment</code>: A simple one-compartment TK model with oral absorption.</li>
<li><code>3compartment</code>: A three-compartment TK model with oral absorption. Compartments are gut, liver, and rest of body.</li>
<li><code>3compartmentss</code>: The steady-state solution to the 3-compartment model under an assumption of constant infusion dosing, without considering tissue partitioning. This was the first <code>httk</code> model (see Wambaugh et al. 2015, Wetmore et al. 2012, Rotroff et al. 2010).</li>
</ul>
</div>
<div id="chemical-specific-tk-data-built-into-httk" class="section level3 hasAnchor">
<h3>Chemical-Specific TK Data Built Into ‘httk’<a href="toxicokinetic-modeling.html#chemical-specific-tk-data-built-into-httk" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Each of these TK models has chemical-specific parameters. The chemical-specific TK information needed to parameterize these models is built into <code>httk</code>, in the form of a built-in lookup table in a data.frame called <code>chem.physical_and_invitro.data</code>. This lookup table means that in order to run a TK model for a particular chemical, you only need to specify the chemical.</p>
<p>Look at the first few rows of this data.frame to see everything that’s in there (it is a lot of information).</p>
<div class="sourceCode" id="cb805"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb805-1"><a href="toxicokinetic-modeling.html#cb805-1" tabindex="-1"></a><span class="fu">head</span>(chem.physical_and_invitro.data)</span></code></pre></div>
<pre><code>##                                                         Compound        CAS
## 2971-36-0  2,2-bis(4-hydroxyphenyl)-1,1,1-trichloroethane (hpte)  2971-36-0
## 94-75-7                                                    2,4-d    94-75-7
## 94-82-6                                                   2,4-db    94-82-6
## 90-43-7                                           2-phenylphenol    90-43-7
## 1007-28-9                                 6-desisopropylatrazine  1007-28-9
## 71751-41-2                                             Abamectin 71751-41-2
##            CAS.Checksum        DTXSID     Formula
## 2971-36-0          TRUE DTXSID8022325 C14H11Cl3O2
## 94-75-7            TRUE DTXSID0020442   C8H6Cl2O3
## 94-82-6            TRUE DTXSID7024035 C10H10Cl2O3
## 90-43-7            TRUE DTXSID2021151     C12H10O
## 1007-28-9          TRUE DTXSID0037495    C5H8ClN5
## 71751-41-2         TRUE DTXSID8023892        &lt;NA&gt;
##                                                                                                                      All.Compound.Names
## 2971-36-0  2,2-bis(4-hydroxyphenyl)-1,1,1-trichloroethane (hpte)|2,2-bis(4-hydroxyphenyl)-1,1,1-trichloroethane|Dtxsid8022325|2971-36-0
## 94-75-7                                                      2,4-d|Dichlorophenoxy|2,4-dichlorophenoxyacetic acid|94-75-7|Dtxsid0020442
## 94-82-6                                                                    2,4-db|2,4-dichlorophenoxybutyric acid|Dtxsid7024035|94-82-6
## 90-43-7                                                                                            2-phenylphenol|Dtxsid2021151|90-43-7
## 1007-28-9                                                            6-desisopropylatrazine|Deisopropylatrazine|Dtxsid0037495|1007-28-9
## 71751-41-2                                                                                                         Abamectin|71751-41-2
##            logHenry logHenry.Reference logMA logMA.Reference  logP
## 2971-36-0    -7.179      EPA-CCD-OPERA    NA            &lt;NA&gt; 4.622
## 94-75-7      -8.529      EPA-CCD-OPERA    NA            &lt;NA&gt; 2.809
## 94-82-6      -8.833      EPA-CCD-OPERA    NA            &lt;NA&gt; 3.528
## 90-43-7      -7.143      EPA-CCD-OPERA  3.46       Endo 2011 3.091
## 1007-28-9    -8.003      EPA-CCD-OPERA    NA            &lt;NA&gt; 1.150
## 71751-41-2       NA               &lt;NA&gt;    NA            &lt;NA&gt; 4.480
##            logP.Reference logPwa logPwa.Reference logWSol logWSol.Reference
## 2971-36-0   EPA-CCD-OPERA  4.528    EPA-CCD-OPERA  -3.707     EPA-CCD-OPERA
## 94-75-7     EPA-CCD-OPERA  5.840    EPA-CCD-OPERA  -2.165     EPA-CCD-OPERA
## 94-82-6     EPA-CCD-OPERA  4.998    EPA-CCD-OPERA  -3.202     EPA-CCD-OPERA
## 90-43-7     EPA-CCD-OPERA  6.108    EPA-CCD-OPERA  -1.812     EPA-CCD-OPERA
## 1007-28-9   EPA-CCD-OPERA  6.989    EPA-CCD-OPERA  -2.413     EPA-CCD-OPERA
## 71751-41-2 Tonnelier 2012     NA             &lt;NA&gt;      NA              &lt;NA&gt;
##                MP  MP.Reference    MW   MW.Reference pKa_Accept
## 2971-36-0  171.40 EPA-CCD-OPERA 317.6  EPA-CCD-OPERA       &lt;NA&gt;
## 94-75-7    140.60 EPA-CCD-OPERA 221.0  EPA-CCD-OPERA       &lt;NA&gt;
## 94-82-6    118.10 EPA-CCD-OPERA 249.1  EPA-CCD-OPERA       &lt;NA&gt;
## 90-43-7     59.03 EPA-CCD-OPERA 170.2  EPA-CCD-OPERA       &lt;NA&gt;
## 1007-28-9  155.00 EPA-CCD-OPERA 173.6  EPA-CCD-OPERA       3.43
## 71751-41-2     NA          &lt;NA&gt; 819.0 Tonnelier 2012       &lt;NA&gt;
##            pKa_Accept.Reference         pKa_Donor pKa_Donor.Reference
## 2971-36-0                  &lt;NA&gt;              8.33           OPERAv2.9
## 94-75-7                    &lt;NA&gt;              2.42           OPERAv2.9
## 94-82-6                    &lt;NA&gt;              3.11           OPERAv2.9
## 90-43-7                    &lt;NA&gt;              9.35           OPERAv2.9
## 1007-28-9             OPERAv2.9              &lt;NA&gt;                &lt;NA&gt;
## 71751-41-2                 &lt;NA&gt; 12.47,13.17,13.80         Strope 2018
##            All.Species Dog.Foral Dog.Foral.Reference DTXSID.Reference
## 2971-36-0        Human        NA                &lt;NA&gt;    EPA-CCD-OPERA
## 94-75-7      Human|Rat        NA                &lt;NA&gt;    EPA-CCD-OPERA
## 94-82-6          Human        NA                &lt;NA&gt;    EPA-CCD-OPERA
## 90-43-7          Human        NA                &lt;NA&gt;    EPA-CCD-OPERA
## 1007-28-9        Human        NA                &lt;NA&gt;    EPA-CCD-OPERA
## 71751-41-2       Human        NA                &lt;NA&gt;    EPA-CCD-OPERA
##            Formula.Reference Human.Caco2.Pab Human.Caco2.Pab.Reference
## 2971-36-0      EPA-CCD-OPERA            &lt;NA&gt;                      &lt;NA&gt;
## 94-75-7        EPA-CCD-OPERA  13.4,7.44,24.1          HondaUnpublished
## 94-82-6        EPA-CCD-OPERA            &lt;NA&gt;                      &lt;NA&gt;
## 90-43-7        EPA-CCD-OPERA            &lt;NA&gt;                      &lt;NA&gt;
## 1007-28-9      EPA-CCD-OPERA  52.4,29.2,94.3          HondaUnpublished
## 71751-41-2              &lt;NA&gt;            &lt;NA&gt;                      &lt;NA&gt;
##            Human.Clint Human.Clint.pValue Human.Clint.pValue.Reference
## 2971-36-0        136.5          0.0000357                 Wetmore 2012
## 94-75-7              0          0.1488000                 Wetmore 2012
## 94-82-6              0          0.1038000                 Wetmore 2012
## 90-43-7          2.077          0.1635000                 Wetmore 2012
## 1007-28-9            0          0.5387000                 Wetmore 2012
## 71751-41-2        5.24          0.0009170                 Wetmore 2012
##            Human.Clint.Reference Human.Fabs Human.Fabs.Reference Human.Fgut
## 2971-36-0           Wetmore 2012         NA                 &lt;NA&gt;         NA
## 94-75-7             Wetmore 2012         NA                 &lt;NA&gt;         NA
## 94-82-6             Wetmore 2012         NA                 &lt;NA&gt;         NA
## 90-43-7             Wetmore 2012         NA                 &lt;NA&gt;         NA
## 1007-28-9           Wetmore 2012         NA                 &lt;NA&gt;         NA
## 71751-41-2          Wetmore 2012         NA                 &lt;NA&gt;         NA
##            Human.Fgut.Reference Human.Fhep Human.Fhep.Reference Human.Foral
## 2971-36-0                  &lt;NA&gt;         NA                 &lt;NA&gt;          NA
## 94-75-7                    &lt;NA&gt;         NA                 &lt;NA&gt;          NA
## 94-82-6                    &lt;NA&gt;         NA                 &lt;NA&gt;          NA
## 90-43-7                    &lt;NA&gt;         NA                 &lt;NA&gt;          NA
## 1007-28-9                  &lt;NA&gt;         NA                 &lt;NA&gt;          NA
## 71751-41-2                 &lt;NA&gt;         NA                 &lt;NA&gt;          NA
##            Human.Foral.Reference Human.Funbound.plasma
## 2971-36-0                   &lt;NA&gt;                     0
## 94-75-7                     &lt;NA&gt;               0.04001
## 94-82-6                     &lt;NA&gt;              0.006623
## 90-43-7                     &lt;NA&gt;               0.04105
## 1007-28-9                   &lt;NA&gt;                0.4588
## 71751-41-2                  &lt;NA&gt;               0.06687
##            Human.Funbound.plasma.Reference Human.Rblood2plasma
## 2971-36-0                     Wetmore 2012                  NA
## 94-75-7                       Wetmore 2012                2.11
## 94-82-6                       Wetmore 2012                  NA
## 90-43-7                       Wetmore 2012                  NA
## 1007-28-9                     Wetmore 2012                  NA
## 71751-41-2                    Wetmore 2012                  NA
##            Human.Rblood2plasma.Reference Monkey.Foral Monkey.Foral.Reference
## 2971-36-0                           &lt;NA&gt;           NA                   &lt;NA&gt;
## 94-75-7                              TNO           NA                   &lt;NA&gt;
## 94-82-6                             &lt;NA&gt;           NA                   &lt;NA&gt;
## 90-43-7                             &lt;NA&gt;           NA                   &lt;NA&gt;
## 1007-28-9                           &lt;NA&gt;           NA                   &lt;NA&gt;
## 71751-41-2                          &lt;NA&gt;           NA                   &lt;NA&gt;
##            Mouse.Foral Mouse.Foral.Reference Mouse.Funbound.plasma
## 2971-36-0           NA                  &lt;NA&gt;                  &lt;NA&gt;
## 94-75-7             NA                  &lt;NA&gt;                  &lt;NA&gt;
## 94-82-6             NA                  &lt;NA&gt;                  &lt;NA&gt;
## 90-43-7             NA                  &lt;NA&gt;                  &lt;NA&gt;
## 1007-28-9           NA                  &lt;NA&gt;                  &lt;NA&gt;
## 71751-41-2          NA                  &lt;NA&gt;                  &lt;NA&gt;
##            Mouse.Funbound.plasma.Reference Rabbit.Funbound.plasma
## 2971-36-0                             &lt;NA&gt;                   &lt;NA&gt;
## 94-75-7                               &lt;NA&gt;                   &lt;NA&gt;
## 94-82-6                               &lt;NA&gt;                   &lt;NA&gt;
## 90-43-7                               &lt;NA&gt;                   &lt;NA&gt;
## 1007-28-9                             &lt;NA&gt;                   &lt;NA&gt;
## 71751-41-2                            &lt;NA&gt;                   &lt;NA&gt;
##            Rabbit.Funbound.plasma.Reference Rat.Clint Rat.Clint.pValue
## 2971-36-0                              &lt;NA&gt;      &lt;NA&gt;               NA
## 94-75-7                                &lt;NA&gt;         0           0.1365
## 94-82-6                                &lt;NA&gt;      &lt;NA&gt;               NA
## 90-43-7                                &lt;NA&gt;      &lt;NA&gt;               NA
## 1007-28-9                              &lt;NA&gt;      &lt;NA&gt;               NA
## 71751-41-2                             &lt;NA&gt;      &lt;NA&gt;               NA
##            Rat.Clint.pValue.Reference Rat.Clint.Reference Rat.Foral
## 2971-36-0                        &lt;NA&gt;                &lt;NA&gt;        NA
## 94-75-7                  Wetmore 2013        Wetmore 2013         1
## 94-82-6                          &lt;NA&gt;                &lt;NA&gt;        NA
## 90-43-7                          &lt;NA&gt;                &lt;NA&gt;        NA
## 1007-28-9                        &lt;NA&gt;                &lt;NA&gt;        NA
## 71751-41-2                       &lt;NA&gt;                &lt;NA&gt;        NA
##            Rat.Foral.Reference Rat.Funbound.plasma
## 2971-36-0                 &lt;NA&gt;                &lt;NA&gt;
## 94-75-7          Wambaugh 2018             0.02976
## 94-82-6                   &lt;NA&gt;                &lt;NA&gt;
## 90-43-7                   &lt;NA&gt;                &lt;NA&gt;
## 1007-28-9                 &lt;NA&gt;                &lt;NA&gt;
## 71751-41-2                &lt;NA&gt;                &lt;NA&gt;
##            Rat.Funbound.plasma.Reference Rat.Rblood2plasma
## 2971-36-0                           &lt;NA&gt;                NA
## 94-75-7                     Wetmore 2013                NA
## 94-82-6                             &lt;NA&gt;                NA
## 90-43-7                             &lt;NA&gt;                NA
## 1007-28-9                           &lt;NA&gt;                NA
## 71751-41-2                          &lt;NA&gt;                NA
##            Rat.Rblood2plasma.Reference Chemical.Class
## 2971-36-0                         &lt;NA&gt;               
## 94-75-7                           &lt;NA&gt;               
## 94-82-6                           &lt;NA&gt;               
## 90-43-7                           &lt;NA&gt;               
## 1007-28-9                         &lt;NA&gt;               
## 71751-41-2                        &lt;NA&gt;</code></pre>
<p>The table contains chemical identifiers: name, CASRN (Chemical Abstract Service Registry Number), and DTXSID (DSSTox ID, a chemical identifier from the EPA Distributed Structure-Searchable Toxicity Database, DSSTox for short – more information can be found at <a href="https://www.epa.gov/chemical-research/distributed-structure-searchable-toxicity-dsstox-database" class="uri">https://www.epa.gov/chemical-research/distributed-structure-searchable-toxicity-dsstox-database</a>). The table also contains physical-chemical properties for each chemical. These are used in predicting tissue partitioning.</p>
<p>The table contains <em>in vitro</em> measured chemical-specific TK parameters, if available. These chemical-specific parameters include intrinsic hepatic clearance (<code>Clint</code>) and fraction unbound to plasma protein (<code>Funbound.plasma</code>) for each chemical. It also contains measured values for oral absorption fraction <code>Fgutabs</code>, and for the partition coefficient between blood and plasma <code>Rblood2plasma</code>, if these values have been measured for a given chemical. If available, there may be chemical-specific TK values for multiple species.</p>
<div id="listing-chemicals-for-which-a-tk-model-can-be-parameterized" class="section level4 hasAnchor">
<h4>Listing chemicals for which a TK model can be parameterized<a href="toxicokinetic-modeling.html#listing-chemicals-for-which-a-tk-model-can-be-parameterized" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>You can easily get a list of all the chemicals for which a specific TK model can be parameterized (for a given species, if needed) using the function <code>get_cheminfo()</code>.</p>
<p>For example, here is how you get a list of all the chemicals for which the PBTK model can be parameterized for humans.</p>
<div class="sourceCode" id="cb807"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb807-1"><a href="toxicokinetic-modeling.html#cb807-1" tabindex="-1"></a>chems_pbtk <span class="ot">&lt;-</span> <span class="fu">get_cheminfo</span>(<span class="at">info =</span> <span class="fu">c</span>(<span class="st">&quot;Compound&quot;</span>, <span class="st">&quot;CAS&quot;</span>, <span class="st">&quot;DTXSID&quot;</span>),</span>
<span id="cb807-2"><a href="toxicokinetic-modeling.html#cb807-2" tabindex="-1"></a>                           <span class="at">model =</span> <span class="st">&quot;pbtk&quot;</span>,</span>
<span id="cb807-3"><a href="toxicokinetic-modeling.html#cb807-3" tabindex="-1"></a>                           <span class="at">species =</span> <span class="st">&quot;Human&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in get_cheminfo(info = c(&quot;Compound&quot;, &quot;CAS&quot;, &quot;DTXSID&quot;), model = &quot;pbtk&quot;, : Excluding compounds that have one or more needed parameters missing in chem.physical_and_invitro.table.
## 
## For model pbtk each chemical must have non-NA values for:Human.Clint, Human.Funbound.plasma, logP, MW</code></pre>
<pre><code>## Warning in get_cheminfo(info = c(&quot;Compound&quot;, &quot;CAS&quot;, &quot;DTXSID&quot;), model = &quot;pbtk&quot;,
## : Excluding compounds without a &#39;fup&#39; value (i.e. fup value = NA) or a &#39;fup&#39;
## value of 0.</code></pre>
<pre><code>## Warning in get_cheminfo(info = c(&quot;Compound&quot;, &quot;CAS&quot;, &quot;DTXSID&quot;), model = &quot;pbtk&quot;,
## : Excluding compounds with uncertain &#39;fup&#39; confidence/credible intervals.</code></pre>
<pre><code>## Warning in get_cheminfo(info = c(&quot;Compound&quot;, &quot;CAS&quot;, &quot;DTXSID&quot;), model = &quot;pbtk&quot;,
## : Excluding compounds that do not have a clint value or distribution of clint
## values.</code></pre>
<pre><code>## Warning in get_cheminfo(info = c(&quot;Compound&quot;, &quot;CAS&quot;, &quot;DTXSID&quot;), model = &quot;pbtk&quot;,
## : Excluding volatile compounds defined as log.Henry &gt;= -4.5.</code></pre>
<pre><code>## Warning in get_cheminfo(info = c(&quot;Compound&quot;, &quot;CAS&quot;, &quot;DTXSID&quot;), model = &quot;pbtk&quot;,
## : Excluding compounds that are categorized in one or more of the following
## chemical classes: PFAS.</code></pre>
<div class="sourceCode" id="cb814"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb814-1"><a href="toxicokinetic-modeling.html#cb814-1" tabindex="-1"></a><span class="fu">head</span>(chems_pbtk) <span class="co">#first few rows</span></span></code></pre></div>
<pre><code>##                 Compound        CAS        DTXSID
## 1                  2,4-d    94-75-7 DTXSID0020442
## 2                 2,4-db    94-82-6 DTXSID7024035
## 3         2-phenylphenol    90-43-7 DTXSID2021151
## 4 6-desisopropylatrazine  1007-28-9 DTXSID0037495
## 5              Abamectin 71751-41-2 DTXSID8023892
## 6               Acephate 30560-19-1 DTXSID8023846</code></pre>
<p>How many such chemicals have parameter data to run a PBTK model in this package?</p>
<div class="sourceCode" id="cb816"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb816-1"><a href="toxicokinetic-modeling.html#cb816-1" tabindex="-1"></a><span class="fu">nrow</span>(chems_pbtk)</span></code></pre></div>
<pre><code>## [1] 957</code></pre>
<p>Here is how you get all the chemicals for which the 3-compartment steady-state model can be parameterized for humans.</p>
<div class="sourceCode" id="cb818"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb818-1"><a href="toxicokinetic-modeling.html#cb818-1" tabindex="-1"></a>chems_3compss <span class="ot">&lt;-</span> <span class="fu">get_cheminfo</span>(<span class="at">info =</span> <span class="fu">c</span>(<span class="st">&quot;Compound&quot;</span>, <span class="st">&quot;CAS&quot;</span>, <span class="st">&quot;DTXSID&quot;</span>),</span>
<span id="cb818-2"><a href="toxicokinetic-modeling.html#cb818-2" tabindex="-1"></a>                           <span class="at">model =</span> <span class="st">&quot;3compartmentss&quot;</span>,</span>
<span id="cb818-3"><a href="toxicokinetic-modeling.html#cb818-3" tabindex="-1"></a>                           <span class="at">species =</span> <span class="st">&quot;Human&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in get_cheminfo(info = c(&quot;Compound&quot;, &quot;CAS&quot;, &quot;DTXSID&quot;), model = &quot;3compartmentss&quot;, : Excluding compounds that have one or more needed parameters missing in chem.physical_and_invitro.table.
## 
## For model 3compartmentss each chemical must have non-NA values for:Human.Clint, Human.Funbound.plasma, logP, MW</code></pre>
<pre><code>## Warning in get_cheminfo(info = c(&quot;Compound&quot;, &quot;CAS&quot;, &quot;DTXSID&quot;), model =
## &quot;3compartmentss&quot;, : Excluding compounds without a &#39;fup&#39; value (i.e. fup value =
## NA).</code></pre>
<pre><code>## Warning in get_cheminfo(info = c(&quot;Compound&quot;, &quot;CAS&quot;, &quot;DTXSID&quot;), model =
## &quot;3compartmentss&quot;, : Excluding compounds with uncertain &#39;fup&#39;
## confidence/credible intervals.</code></pre>
<pre><code>## Warning in get_cheminfo(info = c(&quot;Compound&quot;, &quot;CAS&quot;, &quot;DTXSID&quot;), model =
## &quot;3compartmentss&quot;, : Excluding compounds that do not have a clint value or
## distribution of clint values.</code></pre>
<pre><code>## Warning in get_cheminfo(info = c(&quot;Compound&quot;, &quot;CAS&quot;, &quot;DTXSID&quot;), model =
## &quot;3compartmentss&quot;, : Excluding volatile compounds defined as log.Henry &gt;= -4.5.</code></pre>
<pre><code>## Warning in get_cheminfo(info = c(&quot;Compound&quot;, &quot;CAS&quot;, &quot;DTXSID&quot;), model =
## &quot;3compartmentss&quot;, : Excluding compounds that are categorized in one or more of
## the following chemical classes: PFAS.</code></pre>
<p>How many such chemicals have parameter data to run a 3-compartment steady-state model in this package?</p>
<div class="sourceCode" id="cb825"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb825-1"><a href="toxicokinetic-modeling.html#cb825-1" tabindex="-1"></a><span class="fu">nrow</span>(chems_3compss)</span></code></pre></div>
<pre><code>## [1] 1021</code></pre>
<p>The 3-compartment steady-state model can be parameterized for a few more chemicals than the PBTK model, because it is a simpler model and requires less data to parameterize. Specifically, the 3-compartment steady-state model does not require estimating tissue partition coefficients, unlike the PBTK model.</p>
</div>
</div>
<div id="solving-toxicokinetic-models-to-obtain-internal-chemical-concentration-vs.-time-predictions" class="section level3 hasAnchor">
<h3>Solving Toxicokinetic Models to Obtain Internal Chemical Concentration vs. Time Predictions<a href="toxicokinetic-modeling.html#solving-toxicokinetic-models-to-obtain-internal-chemical-concentration-vs.-time-predictions" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>You can solve any of the models for a specified chemical and specified dosing protocol, and get concentration vs. time predictions, using the function <code>solve_model()</code>. For example:</p>
<div class="sourceCode" id="cb827"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb827-1"><a href="toxicokinetic-modeling.html#cb827-1" tabindex="-1"></a>sol_pbtk <span class="ot">&lt;-</span> <span class="fu">solve_model</span>(<span class="at">chem.name =</span> <span class="st">&quot;Bisphenol-A&quot;</span>, <span class="co">#chemical to simulate</span></span>
<span id="cb827-2"><a href="toxicokinetic-modeling.html#cb827-2" tabindex="-1"></a>            <span class="at">model =</span> <span class="st">&quot;pbtk&quot;</span>, <span class="co">#TK model to use</span></span>
<span id="cb827-3"><a href="toxicokinetic-modeling.html#cb827-3" tabindex="-1"></a>            <span class="at">dosing =</span> <span class="fu">list</span>(<span class="at">initial.dose =</span> <span class="cn">NULL</span>, <span class="co">#for repeated dosing, if first dose is different from the rest, specify first dose here</span></span>
<span id="cb827-4"><a href="toxicokinetic-modeling.html#cb827-4" tabindex="-1"></a>                          <span class="at">doses.per.day =</span> <span class="dv">1</span>, <span class="co">#number of doses per day</span></span>
<span id="cb827-5"><a href="toxicokinetic-modeling.html#cb827-5" tabindex="-1"></a>                          <span class="at">daily.dose =</span> <span class="dv">1</span>, <span class="co">#total daily dose in mg/kg units</span></span>
<span id="cb827-6"><a href="toxicokinetic-modeling.html#cb827-6" tabindex="-1"></a>                          <span class="at">dosing.matrix =</span> <span class="cn">NULL</span>), <span class="co">#used to specify more complicated dosing protocols</span></span>
<span id="cb827-7"><a href="toxicokinetic-modeling.html#cb827-7" tabindex="-1"></a>            <span class="at">days =</span> <span class="dv">1</span>) <span class="co">#number of days to simulate</span></span></code></pre></div>
<pre><code>## None of the monitored components undergo unit conversions  (i.e. conversion factor of 1).
## 
## AUC is area under the plasma concentration curve in uM*days units with Rblood2plasma = 0.795.
## The model outputs are provided in the following units:
##  umol: Agutlumen, Atubules, Ametabolized
##  uM: Cgut, Cliver, Cven, Clung, Cart, Crest, Ckidney, Cplasma
##  uM*days: AUC</code></pre>
<p>There are some cryptic-sounding warnings that can safely be ignored. (They are providing information about certain assumptions that were made while solving the model). Then there is a final message providing the units of the output.</p>
<p>The output, assigned to <code>sol_pbtk</code>, is a matrix with concentration vs. time data for each of the compartments in the <code>pbtk</code> model. Time is in units of days. Additionally, the output traces the amount excreted via passive renal filtration (<code>Atubules</code>), the amount metabolized in the liver (<code>Ametabolized</code>), and the cumulative area under the curve for plasma concentration vs. time (<code>AUC</code>). Here are the first few rows of <code>sol_pbtk</code> so you can see the format.</p>
<div class="sourceCode" id="cb829"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb829-1"><a href="toxicokinetic-modeling.html#cb829-1" tabindex="-1"></a><span class="fu">head</span>(sol_pbtk)</span></code></pre></div>
<pre><code>##        time Agutlumen    Cgut    Cliver     Cven  Clung    Cart    Crest
## [1,] 0.0000       0.0  0.0000  0.000000 0.000000 0.0000 0.00000 0.000000
## [2,] 0.0001     197.3  0.1582  0.000479 0.000001 0.0000 0.00000 0.000000
## [3,] 0.0104     180.0 10.1300  3.063000 0.036390 0.2967 0.03130 0.007976
## [4,] 0.0208     164.1 13.6200  7.596000 0.102500 0.8846 0.09679 0.056220
## [5,] 0.0312     149.6 14.7600 11.110000 0.162000 1.4230 0.15740 0.149600
## [6,] 0.0416     136.3 14.9800 13.380000 0.206900 1.8320 0.20360 0.275500
##      Ckidney  Cplasma Atubules Ametabolized      AUC
## [1,]  0.0000 0.000000 0.000000     0.000000 0.000000
## [2,]  0.0000 0.000001 0.000000     0.000000 0.000000
## [3,]  0.3831 0.045780 0.000583     0.003302 0.000170
## [4,]  1.7530 0.128900 0.004221     0.018490 0.001074
## [5,]  3.2900 0.203700 0.011570     0.045260 0.002818
## [6,]  4.5510 0.260300 0.021980     0.080160 0.005247</code></pre>
<p>You can plot the results, for example plasma concentration vs. time.</p>
<div class="sourceCode" id="cb831"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb831-1"><a href="toxicokinetic-modeling.html#cb831-1" tabindex="-1"></a>sol_pbtk <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(sol_pbtk) <span class="co">#because ggplot2 requires data.frame input, not matrix</span></span>
<span id="cb831-2"><a href="toxicokinetic-modeling.html#cb831-2" tabindex="-1"></a></span>
<span id="cb831-3"><a href="toxicokinetic-modeling.html#cb831-3" tabindex="-1"></a><span class="fu">ggplot</span>(sol_pbtk) <span class="sc">+</span></span>
<span id="cb831-4"><a href="toxicokinetic-modeling.html#cb831-4" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x =</span> time,</span>
<span id="cb831-5"><a href="toxicokinetic-modeling.html#cb831-5" tabindex="-1"></a>                <span class="at">y =</span> Cplasma)) <span class="sc">+</span></span>
<span id="cb831-6"><a href="toxicokinetic-modeling.html#cb831-6" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb831-7"><a href="toxicokinetic-modeling.html#cb831-7" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Time, days&quot;</span>) <span class="sc">+</span></span>
<span id="cb831-8"><a href="toxicokinetic-modeling.html#cb831-8" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Cplasma, uM&quot;</span>) <span class="sc">+</span></span>
<span id="cb831-9"><a href="toxicokinetic-modeling.html#cb831-9" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;Plasma concentration vs. time for single dose 1 mg/kg Bisphenol-A&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-617-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="calculating-summary-metrics-of-internal-dose-produced-from-tk-models" class="section level3 hasAnchor">
<h3>Calculating summary metrics of internal dose produced from TK models<a href="toxicokinetic-modeling.html#calculating-summary-metrics-of-internal-dose-produced-from-tk-models" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We can calculate summary metrics of internal dose – peak concentration, average concentration, and AUC – using the function <code>calc_tkstats()</code>. We have to specify the dosing protocol and length of simulation. Here, we use the same dosing protocol and simulation length as in the plot above.</p>
<div class="sourceCode" id="cb832"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb832-1"><a href="toxicokinetic-modeling.html#cb832-1" tabindex="-1"></a>tkstats <span class="ot">&lt;-</span> <span class="fu">calc_tkstats</span>(<span class="at">chem.name =</span> <span class="st">&quot;Bisphenol-A&quot;</span>, <span class="co">#chemical to simulate</span></span>
<span id="cb832-2"><a href="toxicokinetic-modeling.html#cb832-2" tabindex="-1"></a>             <span class="at">stats =</span> <span class="fu">c</span>(<span class="st">&quot;AUC&quot;</span>, <span class="st">&quot;peak&quot;</span>, <span class="st">&quot;mean&quot;</span>), <span class="co">#which metrics to return (these are the only three choices)</span></span>
<span id="cb832-3"><a href="toxicokinetic-modeling.html#cb832-3" tabindex="-1"></a>             <span class="at">model =</span> <span class="st">&quot;pbtk&quot;</span>, <span class="co">#model to use</span></span>
<span id="cb832-4"><a href="toxicokinetic-modeling.html#cb832-4" tabindex="-1"></a>             <span class="at">tissue =</span> <span class="st">&quot;plasma&quot;</span>, <span class="co">#tissue for which to return internal dose metrics</span></span>
<span id="cb832-5"><a href="toxicokinetic-modeling.html#cb832-5" tabindex="-1"></a>             <span class="at">days =</span> <span class="dv">1</span>, <span class="co">#length of simulation</span></span>
<span id="cb832-6"><a href="toxicokinetic-modeling.html#cb832-6" tabindex="-1"></a>             <span class="at">daily.dose =</span> <span class="dv">1</span>, <span class="co">#total daily dose in mg/kg/day</span></span>
<span id="cb832-7"><a href="toxicokinetic-modeling.html#cb832-7" tabindex="-1"></a>             <span class="at">doses.per.day =</span> <span class="dv">1</span>) <span class="co">#number of doses per day</span></span></code></pre></div>
<pre><code>## Human plasma concentrations returned in uM units.
## AUC is area under plasma concentration curve in uM * days units with Rblood2plasma = 0.7949 .</code></pre>
<div class="sourceCode" id="cb834"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb834-1"><a href="toxicokinetic-modeling.html#cb834-1" tabindex="-1"></a><span class="fu">print</span>(tkstats)</span></code></pre></div>
<pre><code>## $AUC
## [1] 0.3163
## 
## $peak
## [1] 0.3779
## 
## $mean
## [1] 0.3163</code></pre>
</div>
<div id="answer-to-environmental-health-question-1-15" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 1<a href="toxicokinetic-modeling.html#answer-to-environmental-health-question-1-15" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #1</strong></em>: After solving the TK model that evaluates bisphenol-A, what is the maximum concentration of bisphenol-A estimated to occur in human plasma, after 1 exposure dose of 1 mg/kg/day?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: The peak plasma concentration estimate for bisphenol-A, under the conditions tested, is 0.3779 uM.</p>
</div>
</div>
<div id="calculating-steady-state-concentration" class="section level3 hasAnchor">
<h3>Calculating steady-state concentration<a href="toxicokinetic-modeling.html#calculating-steady-state-concentration" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Another summary metric is the steady-state concentration: If the same dose is given repeatedly over many days, the body concentration will (usually) reach a steady state after some time. The value of this steady-state concentration, and the time needed to achieve steady state, are different for different chemicals. Steady-state concentrations are useful when considering long-term, low-level exposures, which is frequently the situation in environmental health.</p>
<p>For example, here is a plot of plasma concentration vs. time for 1 mg/kg/day Bisphenol-A, administered for 12 days. You can see how the average plasma concentration reaches a steady state around 1.5 uM. Each peak represents one day’s dose.</p>
<div class="sourceCode" id="cb836"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb836-1"><a href="toxicokinetic-modeling.html#cb836-1" tabindex="-1"></a>foo <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">solve_pbtk</span>(</span>
<span id="cb836-2"><a href="toxicokinetic-modeling.html#cb836-2" tabindex="-1"></a>    <span class="at">chem.name=</span><span class="st">&#39;Bisphenol-A&#39;</span>,</span>
<span id="cb836-3"><a href="toxicokinetic-modeling.html#cb836-3" tabindex="-1"></a>    <span class="at">daily.dose=</span><span class="dv">1</span>,</span>
<span id="cb836-4"><a href="toxicokinetic-modeling.html#cb836-4" tabindex="-1"></a>    <span class="at">days=</span><span class="dv">12</span>,</span>
<span id="cb836-5"><a href="toxicokinetic-modeling.html#cb836-5" tabindex="-1"></a>    <span class="at">doses.per.day=</span><span class="dv">1</span>,</span>
<span id="cb836-6"><a href="toxicokinetic-modeling.html#cb836-6" tabindex="-1"></a>    <span class="at">tsteps=</span><span class="dv">2</span>))</span></code></pre></div>
<pre><code>## None of the monitored components undergo unit conversions  (i.e. conversion factor of 1).
## 
## AUC is area under the plasma concentration curve in uM*days units with Rblood2plasma = 0.795.
## The model outputs are provided in the following units:
##  umol: Agutlumen, Atubules, Ametabolized
##  uM: Cgut, Cliver, Cven, Clung, Cart, Crest, Ckidney, Cplasma
##  uM*days: AUC</code></pre>
<div class="sourceCode" id="cb838"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb838-1"><a href="toxicokinetic-modeling.html#cb838-1" tabindex="-1"></a><span class="fu">ggplot</span>(foo) <span class="sc">+</span></span>
<span id="cb838-2"><a href="toxicokinetic-modeling.html#cb838-2" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x =</span> time,</span>
<span id="cb838-3"><a href="toxicokinetic-modeling.html#cb838-3" tabindex="-1"></a>                <span class="at">y=</span> Cplasma)) <span class="sc">+</span></span>
<span id="cb838-4"><a href="toxicokinetic-modeling.html#cb838-4" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="dv">12</span>)) <span class="sc">+</span></span>
<span id="cb838-5"><a href="toxicokinetic-modeling.html#cb838-5" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Time, days&quot;</span>) <span class="sc">+</span></span>
<span id="cb838-6"><a href="toxicokinetic-modeling.html#cb838-6" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Cplasma, uM&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-619-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><code>httk</code> includes a function <code>calc_analytic_css()</code> to calculate the steady-state plasma concentration (<span class="math inline">\(C_{ss}\)</span> for short) analytically for each model, for a specified chemical and daily oral dose. This function assumes that the daily oral dose is administered as an oral infusion, rather than a single oral bolus dose – in effect, that the daily dose is divided into many small doses over the day. Therefore, the result of <code>calc_analytic_css()</code> may be slightly different than our previous estimate based on the concentration vs. time plot from a single oral bolus dose every day.</p>
<p>Here is the result of <code>calc_analytic_css()</code> for a 1 mg/kg/day dose of bisphenol-A.</p>
<div class="sourceCode" id="cb839"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb839-1"><a href="toxicokinetic-modeling.html#cb839-1" tabindex="-1"></a><span class="fu">calc_analytic_css</span>(<span class="at">chem.name =</span> <span class="st">&quot;Bisphenol-A&quot;</span>,</span>
<span id="cb839-2"><a href="toxicokinetic-modeling.html#cb839-2" tabindex="-1"></a>                  <span class="at">daily.dose =</span> <span class="dv">1</span>,</span>
<span id="cb839-3"><a href="toxicokinetic-modeling.html#cb839-3" tabindex="-1"></a>                  <span class="at">output.units =</span> <span class="st">&quot;uM&quot;</span>,</span>
<span id="cb839-4"><a href="toxicokinetic-modeling.html#cb839-4" tabindex="-1"></a>                  <span class="at">model =</span> <span class="st">&quot;pbtk&quot;</span>,</span>
<span id="cb839-5"><a href="toxicokinetic-modeling.html#cb839-5" tabindex="-1"></a>                  <span class="at">concentration =</span> <span class="st">&quot;plasma&quot;</span>)</span></code></pre></div>
<pre><code>## Plasma concentration returned in uM units.</code></pre>
<pre><code>## [1] 0.9417</code></pre>
</div>
<div id="answer-to-environmental-health-question-2-16" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 2<a href="toxicokinetic-modeling.html#answer-to-environmental-health-question-2-16" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #2</strong></em>: After solving the TK model that evaluates bisphenol-A, what is the steady-state concentration of bisphenol-A estimated to occur in human plasma, for a long-term oral infusion dose of 1 mg/kg/day?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: The steady-state plasma concentration estimate for bisphenol-A, under the conditions tested, is 0.9417 uM.</p>
</div>
</div>
<div id="steady-state-concentration-is-linear-with-dose-for-httk-models" class="section level3 hasAnchor">
<h3>Steady-state concentration is linear with dose for <code>httk</code> models<a href="toxicokinetic-modeling.html#steady-state-concentration-is-linear-with-dose-for-httk-models" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For the TK models included in the <code>httk</code> package, steady-state concentration is linear with dose for a given chemical. The slope of the line is simply the steady-state concentration for a dose of 1 mg/kg/day. This can be shown by solving <code>calc_analytic_css()</code> for several doses, and plotting the dose-<span class="math inline">\(C_{ss}\)</span> points along a line whose slope is equal to <span class="math inline">\(C_{ss}\)</span> for 1 mg/kg/day.</p>
<div class="sourceCode" id="cb842"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb842-1"><a href="toxicokinetic-modeling.html#cb842-1" tabindex="-1"></a><span class="co">#choose five doses at which to find the Css</span></span>
<span id="cb842-2"><a href="toxicokinetic-modeling.html#cb842-2" tabindex="-1"></a>doses <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="co">#all mg/kg/day</span></span>
<span id="cb842-3"><a href="toxicokinetic-modeling.html#cb842-3" tabindex="-1"></a>           <span class="fl">0.5</span>,</span>
<span id="cb842-4"><a href="toxicokinetic-modeling.html#cb842-4" tabindex="-1"></a>           <span class="fl">1.0</span>,</span>
<span id="cb842-5"><a href="toxicokinetic-modeling.html#cb842-5" tabindex="-1"></a>           <span class="fl">1.5</span>,</span>
<span id="cb842-6"><a href="toxicokinetic-modeling.html#cb842-6" tabindex="-1"></a>           <span class="fl">2.0</span>)</span>
<span id="cb842-7"><a href="toxicokinetic-modeling.html#cb842-7" tabindex="-1"></a><span class="fu">suppressWarnings</span>(bpa_css <span class="ot">&lt;-</span> <span class="fu">sapply</span>(doses,</span>
<span id="cb842-8"><a href="toxicokinetic-modeling.html#cb842-8" tabindex="-1"></a>                  <span class="cf">function</span>(dose) <span class="fu">calc_analytic_css</span>(<span class="at">chem.name =</span> <span class="st">&quot;Bisphenol-A&quot;</span>,</span>
<span id="cb842-9"><a href="toxicokinetic-modeling.html#cb842-9" tabindex="-1"></a>                  <span class="at">daily.dose =</span> dose,</span>
<span id="cb842-10"><a href="toxicokinetic-modeling.html#cb842-10" tabindex="-1"></a>                  <span class="at">output.units =</span> <span class="st">&quot;uM&quot;</span>,</span>
<span id="cb842-11"><a href="toxicokinetic-modeling.html#cb842-11" tabindex="-1"></a>                  <span class="at">model =</span> <span class="st">&quot;pbtk&quot;</span>,</span>
<span id="cb842-12"><a href="toxicokinetic-modeling.html#cb842-12" tabindex="-1"></a>                  <span class="at">concentration =</span> <span class="st">&quot;plasma&quot;</span>,</span>
<span id="cb842-13"><a href="toxicokinetic-modeling.html#cb842-13" tabindex="-1"></a>                  <span class="at">suppress.messages =</span> <span class="cn">TRUE</span>)))</span>
<span id="cb842-14"><a href="toxicokinetic-modeling.html#cb842-14" tabindex="-1"></a></span>
<span id="cb842-15"><a href="toxicokinetic-modeling.html#cb842-15" tabindex="-1"></a>DF <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">dose =</span> doses,</span>
<span id="cb842-16"><a href="toxicokinetic-modeling.html#cb842-16" tabindex="-1"></a>                 <span class="at">Css =</span> bpa_css)</span>
<span id="cb842-17"><a href="toxicokinetic-modeling.html#cb842-17" tabindex="-1"></a></span>
<span id="cb842-18"><a href="toxicokinetic-modeling.html#cb842-18" tabindex="-1"></a><span class="co">#Plot the results</span></span>
<span id="cb842-19"><a href="toxicokinetic-modeling.html#cb842-19" tabindex="-1"></a>Cssdosefig  <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(DF) <span class="sc">+</span></span>
<span id="cb842-20"><a href="toxicokinetic-modeling.html#cb842-20" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> dose,</span>
<span id="cb842-21"><a href="toxicokinetic-modeling.html#cb842-21" tabindex="-1"></a>                 <span class="at">y =</span> Css),</span>
<span id="cb842-22"><a href="toxicokinetic-modeling.html#cb842-22" tabindex="-1"></a>             <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb842-23"><a href="toxicokinetic-modeling.html#cb842-23" tabindex="-1"></a>  <span class="fu">geom_abline</span>( <span class="co">#plot a straight line</span></span>
<span id="cb842-24"><a href="toxicokinetic-modeling.html#cb842-24" tabindex="-1"></a>    <span class="at">intercept =</span> <span class="dv">0</span>, <span class="co">#intercept 0</span></span>
<span id="cb842-25"><a href="toxicokinetic-modeling.html#cb842-25" tabindex="-1"></a>              <span class="at">slope =</span> DF[DF<span class="sc">$</span>dose<span class="sc">==</span><span class="dv">1</span>, <span class="co">#slope = Css for 1 mg/kg/day</span></span>
<span id="cb842-26"><a href="toxicokinetic-modeling.html#cb842-26" tabindex="-1"></a>                         <span class="st">&quot;Css&quot;</span>],</span>
<span id="cb842-27"><a href="toxicokinetic-modeling.html#cb842-27" tabindex="-1"></a>    <span class="at">linetype =</span> <span class="dv">2</span></span>
<span id="cb842-28"><a href="toxicokinetic-modeling.html#cb842-28" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb842-29"><a href="toxicokinetic-modeling.html#cb842-29" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Daily dose, mg/kg/day&quot;</span>) <span class="sc">+</span></span>
<span id="cb842-30"><a href="toxicokinetic-modeling.html#cb842-30" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Css, uM&quot;</span>)</span>
<span id="cb842-31"><a href="toxicokinetic-modeling.html#cb842-31" tabindex="-1"></a></span>
<span id="cb842-32"><a href="toxicokinetic-modeling.html#cb842-32" tabindex="-1"></a><span class="fu">print</span>(Cssdosefig)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-621-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="reverse-toxicokinetics" class="section level2 hasAnchor">
<h2>Reverse Toxicokinetics<a href="toxicokinetic-modeling.html#reverse-toxicokinetics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In the previous TK examples, we started with a specified dosing protocol, then solved the TK models to find the resulting concentration in the body (e.g., in plasma). This allows us to convert from external exposure metrics to internal exposure metrics. However, many environmental health questions require the reverse: converting from internal exposure metrics to external exposure metrics.</p>
<p>For example, when health effects of environmental chemicals are studied in epidemiological cohorts, adverse health effects are often related to <em>internal</em> exposure metrics, such as blood or plasma concentration of a chemical. Similarly, <em>in vitro</em> studies of chemical bioactivity (for example, the ToxCast program) relate bioactivity to <em>in vitro</em> concentration, which can be consdered analogous to internal exposure or body concentration. So we may know the <em>internal</em> exposure level associated with some adverse health effect of a chemical.</p>
<p>However, risk assessors and risk managers typically control <em>external</em> exposure to reduce the risk of adverse health effects. They need some way to start from an internal exposure associated with adverse health effects, and convert to the corresponding external exposure.</p>
<p>The solution is <em>reverse toxicokinetics</em> (reverse TK). Starting with a specified internal exposure metric (body concentration), solve the TK model <em>in reverse</em> to find the corresponding external exposure that produced that concentration.</p>
<p>When exposures are long-term and low-level (as environmental exposures often are), then the relevant internal exposure metric is the steady-state concentration. In this case, it is useful to remember the linear relationship between <span class="math inline">\(C_{ss}\)</span> and dose for the <code>httk</code> TK models. It gives you a quick and easy way to perform reverse TK for the steady-state case.</p>
<p>The procedure is illustrated graphically below.</p>
<ol style="list-style-type: decimal">
<li>Begin with a “target” concentration on the y-axis (labeled <span class="math inline">\(C_{\textrm{target}}\)</span>). For example, <span class="math inline">\(C_{\textrm{target}}\)</span> may be the <em>in vitro</em> concentration associated with bioactivity in a ToxCast assay, or the plasma concentration associated with an adverse health effect in an epidemiological study.</li>
<li>Draw a horizontal line over to the <span class="math inline">\(C_{ss}\)</span>-dose line.</li>
<li>Drop down vertically to the x-axis and read off the corresponding dose. This is the <em>administered equivalent dose</em> (AED): the the external dose or exposure rate, in mg/kg/day, that would produce an internal steady-state plasma concentration equal to the target concentration.</li>
</ol>
<p><img src="_main_files/figure-html/unnamed-chunk-622-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Mathematically, the relation is very simple:</p>
<p><span class="math display">\[ AED = \frac{C_{\textrm{target}}}{C_{ss}\textrm{-dose slope}} \]</span></p>
<p>Since the <span class="math inline">\(C_{ss}\)</span>-dose slope is simply <span class="math inline">\(C_{ss}\)</span> for a daily dose of 1 mg/kg/day, this equation can be rewritten as</p>
<p><span class="math display">\[ AED = \frac{C_{\textrm{target}}}{C_{ss}\textrm{ for 1 mg/kg/day}} \]</span></p>
</div>
<div id="capturing-population-variability-in-toxicokinetics-and-uncertainty-in-chemical-specific-parameters" class="section level2 hasAnchor">
<h2>Capturing Population Variability in Toxicokinetics, and Uncertainty in Chemical-Specific Parameters<a href="toxicokinetic-modeling.html#capturing-population-variability-in-toxicokinetics-and-uncertainty-in-chemical-specific-parameters" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For a given dose, <span class="math inline">\(C_{ss}\)</span> is determined by the values of the parameters of the TK model. These parameters describe absorption, distribution, metabolism, and excretion (ADME) of each chemical. They include both chemical-specific parameters, describing hepatic clearance and protein binding, and chemical-independent parameters, describing physiology. A table of these parameters is presented below.</p>
<table>
<colgroup>
<col width="21%" />
<col width="38%" />
<col width="27%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Parameter</th>
<th align="left">Details</th>
<th align="left">Estimated</th>
<th align="left">Type</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">Intrinsic hepatic clearance rate</td>
<td align="left">Rate at which liver removes chemical from blood</td>
<td align="left">Measured <em>in vitro</em></td>
<td align="left">Chemical-specific</td>
</tr>
<tr class="even">
<td align="left">Fraction unbound to plasma protein</td>
<td align="left">Free fraction of chemical in plasma</td>
<td align="left">Measured <em>in vitro</em></td>
<td align="left">Chemical-specific</td>
</tr>
<tr class="odd">
<td align="left">Tissue:plasma partition coefficients</td>
<td align="left">Ratio of concentration in body tissues to concentration in plasma</td>
<td align="left">Estimated from chemical and tissue properties</td>
<td align="left">Chemical-specific</td>
</tr>
<tr class="even">
<td align="left">Tissue masses</td>
<td align="left">Mass of each body tissue (including total body weight)</td>
<td align="left">From anatomical literature</td>
<td align="left">Chemical-independent</td>
</tr>
<tr class="odd">
<td align="left">Tissue blood flows</td>
<td align="left">Blood flow rate to each body tissue</td>
<td align="left">From anatomical literature</td>
<td align="left">Chemical-independent</td>
</tr>
<tr class="even">
<td align="left">Glomerular filtration rate</td>
<td align="left">Rate at which kidneys remove chemical from blood</td>
<td align="left">From anatomical literature</td>
<td align="left">Chemical-independent</td>
</tr>
<tr class="odd">
<td align="left">Hepatocellularity</td>
<td align="left">Number of cells per mg liver</td>
<td align="left">From anatomical literature</td>
<td align="left">Chemical-independent</td>
</tr>
</tbody>
</table>
<p>Because these parameters represent physiology and chemical-body interactions, their exact values will vary across individuals in a population, reflecting population physiological variability. Additionally, parameters are subject to measurement uncertainty.</p>
<p>Since the <span class="math inline">\(C_{ss}\)</span>-dose relation is determined by these parameters, variability and uncertainty in the TK parameters translates directly into variability and uncertainty in <span class="math inline">\(C_{ss}\)</span> for a given dose. In other words, there is a distribution of <span class="math inline">\(C_{ss}\)</span> values for each daily dose level of a chemical.</p>
<p>The <span class="math inline">\(C_{ss}\)</span>-dose relationship is still linear when variability and uncertainty are taken into account. However, rather than a single <span class="math inline">\(C_{ss}\)</span>-dose slope, there is a distribution of <span class="math inline">\(C_{ss}\)</span>-dose slopes. Because the <span class="math inline">\(C_{ss}\)</span>-dose slope is simply the <span class="math inline">\(C_{ss}\)</span> value for an exposure rate of 1 mg/kg/day, the distribution of the <span class="math inline">\(C_{ss}\)</span>-dose slope is the same as the <span class="math inline">\(C_{ss}\)</span> distribution for an exposure rate of 1 mg/kg/day.</p>
<p>A distribution of <span class="math inline">\(C_{ss}\)</span>-dose slopes is illustrated in the figure below, along with boxplots illustrating the distributions for <span class="math inline">\(C_{ss}\)</span> itself at five different dose levels: 0.05, 0.25, 0.5, 0.75, and 0.95 mg/kg/day.</p>
<pre><code>## Warning: A numeric `legend.position` argument in `theme()` was deprecated in
## ggplot2 3.5.0.
## ℹ Please use the `legend.position.inside` argument of `theme()`
##   instead.
## This warning is displayed once every 8 hours.
## Call `lifecycle::last_lifecycle_warnings()` to see where this warning
## was generated.</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-624"></span>
<img src="_main_files/figure-html/unnamed-chunk-624-1.png" alt="Boxplots: Distributions of Css for five daily dose levels of Bisphenol-A. Boxes extend from 25th to 75th percentile. Lower whisker = 5th percentile; upper whisker = 95th percentile. Lines: Css-dose relations for each quantile." width="672" />
<p class="caption">
Figure 15: Boxplots: Distributions of Css for five daily dose levels of Bisphenol-A. Boxes extend from 25th to 75th percentile. Lower whisker = 5th percentile; upper whisker = 95th percentile. Lines: Css-dose relations for each quantile.
</p>
</div>
<div id="variability-and-uncertainty-in-reverse-toxicokinetics" class="section level3 hasAnchor">
<h3>Variability and Uncertainty in Reverse Toxicokinetics<a href="toxicokinetic-modeling.html#variability-and-uncertainty-in-reverse-toxicokinetics" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Earlier, we found that with a linear <span class="math inline">\(C_{ss}\)</span>-dose relation, reverse toxicokinetics became a matter of a simple linear equation. For a given target concentration – for example, a plasma concentration associated with adverse health effects <em>in vivo</em>, or a concentration associated with bioactivity <em>in vitro</em> – we could predict an AED (administered equivalent dose), the external exposure rate in mg/kg/day that would produce the target concentration at steady state.</p>
<p><span class="math display">\[ AED = \frac{C_{\textrm{target}}}{C_{ss}\textrm{-dose slope}} \]</span></p>
<p>Since AED depends on the <span class="math inline">\(C_{ss}\)</span>-dose slope, variability and uncertainty in that slope will induce variability and uncertainty in the AED. A distribution of slopes will lead to a distribution of AEDs for the same target concentration.</p>
<p>For example, a graphical representation of finding the AED distribution for a target concentration of 1 uM looks like this, for the same arbitrary example chemical used to illustrate the distribution of <span class="math inline">\(C_{ss}\)</span>-dose slopes above. (The lines shown in this plot are the same as the previous plot, but the plot has been “zoomed in” on the y-axis.)</p>
<p>The steps are the same as before:</p>
<ol style="list-style-type: decimal">
<li>Begin with a “target” concentration on the y-axis, here 1 uM.</li>
<li>Draw a horizontal line over to intersect each <span class="math inline">\(C_{ss}\)</span>-dose line.</li>
<li>Where the horizontal line intersects each <span class="math inline">\(C_{ss}\)</span>-dose line, drop down vertically to the x-axis and read off each corresponding AED (marked with colored circles matching the color of each <span class="math inline">\(C_{ss}\)</span>-dose line).</li>
</ol>
<p><img src="_main_files/figure-html/unnamed-chunk-625-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Notice that the line with the steepest, 95th-percentile slope (the purple line) yields the lowest AED (the purple dot, approximately 0.07 mg/kg/day for this example chemical), and the line with the shallowest, 5th-percentile slope (the turquoise blue line) yields the highest AED (the turquoise dot, approximately 2 mg/kg/day for this example chemical).</p>
<p>In general, the 95th-percentile <span class="math inline">\(C_{ss}\)</span>-dose slope represents the most-sensitive 5% of the population – individuals who will reach the target concentration in their body with the smallest daily doses. Therefore, using the AED for the 95th-percentile <span class="math inline">\(C_{ss}\)</span>-dose slope is a conservative choice, health-protective for 95% of the estimated population.</p>
</div>
<div id="monte-carlo-approach-to-simulating-variability-and-uncertainty" class="section level3 hasAnchor">
<h3>Monte Carlo approach to simulating variability and uncertainty<a href="toxicokinetic-modeling.html#monte-carlo-approach-to-simulating-variability-and-uncertainty" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The <code>httk</code> package implements a Monte Carlo approach for simulating variability and uncertainty in TK.</p>
<p><code>httk</code> first defines distributions for the TK model parameters, representing population variabilty. These distributions are defined based on real data about U.S. population demographics and physiology collected as part of the Centers for Disease Control’s National Health and Nutrition Examination Survey (NHANES) <a href="https://pubmed.ncbi.nlm.nih.gov/28628784/">(Ring et al., 2017)</a>. TK parameters with known measurement uncertainty (intrinsic hepatic clearance rate and fraction of chemical unbound in plasma) additionally have distributions defined to represent their uncertainty <a href="https://pubmed.ncbi.nlm.nih.gov/31532498/">(Wambaugh et al., 2019)</a>.</p>
<p>Then, <code>httk</code> samples sets of TK parameter values from these distributions (including appropriate correlations: for example, liver mass is correlated with body weight). Each sampled set of TK parameter values represents one “simulated individual.”</p>
<p>Next, <code>httk</code> calculates the <span class="math inline">\(C_{ss}\)</span>-dose slope for each “simulated individual.” The resulting sample of <span class="math inline">\(C_{ss}\)</span>-dose slopes can be used to characterize the distribution of <span class="math inline">\(C_{ss}\)</span>-dose slopes – for example, by calculating percentiles.</p>
<p><code>httk</code> makes this whole Monte Carlo process simple and transparent for the user, You just need to call one function, <code>calc_mc_css()</code>, specifying the chemical whose <span class="math inline">\(C_{ss}\)</span>-dose slope distribution you want to calculate. Behind the scenes, <code>httk</code> will perform all the Monte Carlo calculations. It will return percentiles of the <span class="math inline">\(C_{ss}\)</span>-dose slope (by default), or it can return all individual samples of <span class="math inline">\(C_{ss}\)</span>-dose slope (if you want to do some calculations of your own).</p>
</div>
<div id="chemical-specific-example-capturing-population-variability-for-bisphenol-a-plasma-concentration-estimates" class="section level3 hasAnchor">
<h3>Chemical-Specific Example Capturing Population Variability for Bisphenol-A Plasma Concentration Estimates<a href="toxicokinetic-modeling.html#chemical-specific-example-capturing-population-variability-for-bisphenol-a-plasma-concentration-estimates" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The following code estimates the 5th percentile, 50th percentile, and 95th percentile of the <span class="math inline">\(C_{ss}\)</span>-dose slope for the chemical bisphenol-A. For the sake of simplicity, we will use the 3-compartment steady-state model (rather than the PBTK model used in the previous examples).</p>
<div class="sourceCode" id="cb844"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb844-1"><a href="toxicokinetic-modeling.html#cb844-1" tabindex="-1"></a>css_examp <span class="ot">&lt;-</span> <span class="fu">calc_mc_css</span>(<span class="at">chem.name =</span> <span class="st">&quot;Bisphenol-A&quot;</span>,</span>
<span id="cb844-2"><a href="toxicokinetic-modeling.html#cb844-2" tabindex="-1"></a>            <span class="at">which.quantile =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="co">#specify which quantiles to return</span></span>
<span id="cb844-3"><a href="toxicokinetic-modeling.html#cb844-3" tabindex="-1"></a>                               <span class="fl">0.5</span>,</span>
<span id="cb844-4"><a href="toxicokinetic-modeling.html#cb844-4" tabindex="-1"></a>                               <span class="fl">0.95</span>),</span>
<span id="cb844-5"><a href="toxicokinetic-modeling.html#cb844-5" tabindex="-1"></a>            <span class="at">model =</span> <span class="st">&quot;3compartmentss&quot;</span>, <span class="co">#which model to use to calculate Css</span></span>
<span id="cb844-6"><a href="toxicokinetic-modeling.html#cb844-6" tabindex="-1"></a>            <span class="at">output.units =</span> <span class="st">&quot;uM&quot;</span>) <span class="co">#could also choose mg/Lo</span></span></code></pre></div>
<pre><code>## Human plasma concentration returned in uM units for 0.05 0.5 0.95 quantile.</code></pre>
<div class="sourceCode" id="cb846"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb846-1"><a href="toxicokinetic-modeling.html#cb846-1" tabindex="-1"></a><span class="fu">print</span>(css_examp)</span></code></pre></div>
<pre><code>##     5%    50%    95% 
## 0.2974 1.3460 8.5100</code></pre>
<p>Recall that the <span class="math inline">\(C_{ss}\)</span>-dose slope is the same as <span class="math inline">\(C_{ss}\)</span> for a daily dose of 1 mg/kg/day. The function <code>calc_mc_css()</code> therefore assumes a dose of 1 mg/kg/day and calculates the resulting <span class="math inline">\(C_{ss}\)</span> distribution. If you need to calculate the <span class="math inline">\(C_{ss}\)</span> distribution for a different dose, e.g. 2 mg/kg/day, you can simply multiply the <span class="math inline">\(C_{ss}\)</span> percentiles from <code>calc_mc_css()</code> by your desired dose.</p>
<p>The steady-state plasma concentration for 1 mg/kg/day dose is returned in units of uM. The three requested quantiles are returned as a named numeric vector (whose names in this case are <code>5%</code>, <code>50%</code>, and <code>95%</code>).</p>
</div>
<div id="answer-to-environmental-health-question-3-12" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 3<a href="toxicokinetic-modeling.html#answer-to-environmental-health-question-3-12" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #3</strong></em>: What is the predicted range of bisphenol-A concentrations in plasma that can occur in a human population, assuming a long-term exposure rate of 1 mg/kg/day and steady-state conditions? Provide estimates at the 5th, 50th, and 95th percentile?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: For a human population exposed to 1 mg/kg/day bisphenol-A, plasma concentrations are estimated to be 0.2974 uM at the 5th percentile, 1.346 uM at the 50th percentile, and 8.51 uM at the 95th percentile.</p>
</div>
</div>
<div id="high-throughput-example-capturing-population-variability-for-1000-chemicals" class="section level3 hasAnchor">
<h3>High-Throughput Example Capturing Population Variability for ~1000 Chemicals<a href="toxicokinetic-modeling.html#high-throughput-example-capturing-population-variability-for-1000-chemicals" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We can easily and (fairly) quickly do this for all 998 chemicals for which the 3-compartment steady-state model can be parameterized, using <code>sapply()</code> to loop over the chemicals. This will take a few minutes to run (for example, it takes about 10-15 minutes on a Dell Latitude with an Intel i7 processor).</p>
<p>In order to make the Monte Carlo sampling reproducible, set a seed for the random number generator. It doesn’t matter what seed you choose – it can be any integer. Here, the seed is set to 42, because it’s the answer to the ultimate question of life, the universe, and everything <a href="https://en.wikipedia.org/wiki/The_Hitchhiker%27s_Guide_to_the_Galaxy_(novel)">(Adams, 1979)</a>.</p>
<div class="sourceCode" id="cb848"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb848-1"><a href="toxicokinetic-modeling.html#cb848-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb848-2"><a href="toxicokinetic-modeling.html#cb848-2" tabindex="-1"></a></span>
<span id="cb848-3"><a href="toxicokinetic-modeling.html#cb848-3" tabindex="-1"></a><span class="fu">system.time</span>(</span>
<span id="cb848-4"><a href="toxicokinetic-modeling.html#cb848-4" tabindex="-1"></a>  <span class="fu">suppressWarnings</span>(</span>
<span id="cb848-5"><a href="toxicokinetic-modeling.html#cb848-5" tabindex="-1"></a>    css_3compss <span class="ot">&lt;-</span> <span class="fu">sapply</span>(chems_3compss<span class="sc">$</span>CAS,</span>
<span id="cb848-6"><a href="toxicokinetic-modeling.html#cb848-6" tabindex="-1"></a>              calc_mc_css,</span>
<span id="cb848-7"><a href="toxicokinetic-modeling.html#cb848-7" tabindex="-1"></a>              <span class="co">#additional arguments to calc_mc_css()</span></span>
<span id="cb848-8"><a href="toxicokinetic-modeling.html#cb848-8" tabindex="-1"></a>              <span class="at">model =</span> <span class="st">&quot;3compartmentss&quot;</span>,</span>
<span id="cb848-9"><a href="toxicokinetic-modeling.html#cb848-9" tabindex="-1"></a>              <span class="at">which.quantile =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.5</span>, <span class="fl">0.95</span>),</span>
<span id="cb848-10"><a href="toxicokinetic-modeling.html#cb848-10" tabindex="-1"></a>              <span class="at">output.units =</span> <span class="st">&quot;uM&quot;</span>,</span>
<span id="cb848-11"><a href="toxicokinetic-modeling.html#cb848-11" tabindex="-1"></a>              <span class="at">suppress.messages =</span> <span class="cn">TRUE</span>)</span>
<span id="cb848-12"><a href="toxicokinetic-modeling.html#cb848-12" tabindex="-1"></a>              )</span>
<span id="cb848-13"><a href="toxicokinetic-modeling.html#cb848-13" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>##    user  system elapsed 
##  825.89   11.86  897.49</code></pre>
<p>Organizing the results:</p>
<div class="sourceCode" id="cb850"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb850-1"><a href="toxicokinetic-modeling.html#cb850-1" tabindex="-1"></a><span class="co">#css_3compss comes out as a 3 x 998 array,</span></span>
<span id="cb850-2"><a href="toxicokinetic-modeling.html#cb850-2" tabindex="-1"></a><span class="co">#where rows are quantiles and columns are chemicals</span></span>
<span id="cb850-3"><a href="toxicokinetic-modeling.html#cb850-3" tabindex="-1"></a><span class="co">#transpose it so that rows are chemicals and columns are quantiles</span></span>
<span id="cb850-4"><a href="toxicokinetic-modeling.html#cb850-4" tabindex="-1"></a>css_3compss <span class="ot">&lt;-</span> <span class="fu">t</span>(css_3compss)</span>
<span id="cb850-5"><a href="toxicokinetic-modeling.html#cb850-5" tabindex="-1"></a><span class="co">#convert to data.frame</span></span>
<span id="cb850-6"><a href="toxicokinetic-modeling.html#cb850-6" tabindex="-1"></a>css_3compss <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(css_3compss)</span>
<span id="cb850-7"><a href="toxicokinetic-modeling.html#cb850-7" tabindex="-1"></a><span class="co">#make a column for CAS, rather than just leaving it as the row names</span></span>
<span id="cb850-8"><a href="toxicokinetic-modeling.html#cb850-8" tabindex="-1"></a>css_3compss<span class="sc">$</span>CAS <span class="ot">&lt;-</span> <span class="fu">row.names</span>(css_3compss)</span>
<span id="cb850-9"><a href="toxicokinetic-modeling.html#cb850-9" tabindex="-1"></a></span>
<span id="cb850-10"><a href="toxicokinetic-modeling.html#cb850-10" tabindex="-1"></a><span class="fu">head</span>(css_3compss) <span class="co">#View first few rows</span></span></code></pre></div>
<pre><code>##                 5%     50%      95%        CAS
## 2971-36-0   0.1449   0.678    6.102  2971-36-0
## 94-75-7    18.1500  68.390  431.300    94-75-7
## 94-82-6    56.9200 232.100 1783.000    94-82-6
## 90-43-7    23.5200  76.800  344.100    90-43-7
## 1007-28-9   2.6670   7.203   29.320  1007-28-9
## 71751-41-2  0.7825   3.198   18.720 71751-41-2</code></pre>
</div>
<div id="plotting-the-c_ss-dose-slope-distribution-quantiles-across-these-1000-chemicals" class="section level3 hasAnchor">
<h3>Plotting the <span class="math inline">\(C_{ss}\)</span>-dose slope distribution quantiles across these ~1000 chemicals<a href="toxicokinetic-modeling.html#plotting-the-c_ss-dose-slope-distribution-quantiles-across-these-1000-chemicals" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Here, we will plot the resulting concentration distribution quantiles for each chemical, while sorting the chemicals from lowest to highest median value.</p>
<p>By default, <code>ggplot2</code> will plot the chemical CASRNs in alphabetically-sorted order. To force it to plot them in another order, we have to explicitly specify the desired order. The easiest way to do this is to add a column in the data.frame that contains the chemical names as a factor (categorical) variable, whose levels (categories) are explicitly set to be the CASRNs in our desired plotting order. Then we can tell <code>ggplot2</code> to plot that factor variable on the x-axis, rather than the original CASRN variable.</p>
<p>Set the ordering of the chemical CASRNs from lowest to highest median value</p>
<div class="sourceCode" id="cb852"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb852-1"><a href="toxicokinetic-modeling.html#cb852-1" tabindex="-1"></a>chemical_order <span class="ot">&lt;-</span> <span class="fu">order</span>(css_3compss<span class="sc">$</span><span class="st">`</span><span class="at">50%</span><span class="st">`</span>)</span></code></pre></div>
<p>Create a factor (categorical) CAS column where the factor levels are given by the CASRNs with this ordering.</p>
<div class="sourceCode" id="cb853"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb853-1"><a href="toxicokinetic-modeling.html#cb853-1" tabindex="-1"></a>css_3compss<span class="sc">$</span>CAS_factor <span class="ot">&lt;-</span> <span class="fu">factor</span>(css_3compss<span class="sc">$</span>CAS, <span class="at">levels =</span> css_3compss<span class="sc">$</span>CAS[chemical_order])</span></code></pre></div>
<p>For plotting ease, reshape the data.frame into “long” format – rather than having one column for each quantile of the <span class="math inline">\(C_{ss}\)</span> distribution, have a row for each chemical/quantile combination. We use the <code>melt</code> function from the <code>reshape2</code> package.</p>
<div class="sourceCode" id="cb854"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb854-1"><a href="toxicokinetic-modeling.html#cb854-1" tabindex="-1"></a>css_3compss_melt <span class="ot">&lt;-</span> <span class="fu">melt</span>(css_3compss,</span>
<span id="cb854-2"><a href="toxicokinetic-modeling.html#cb854-2" tabindex="-1"></a>                                   <span class="at">id.vars =</span> <span class="st">&quot;CAS_factor&quot;</span>,</span>
<span id="cb854-3"><a href="toxicokinetic-modeling.html#cb854-3" tabindex="-1"></a>                                   <span class="at">measure.vars =</span> <span class="fu">c</span>(<span class="st">&quot;5%&quot;</span>, <span class="st">&quot;50%&quot;</span>, <span class="st">&quot;95%&quot;</span>),</span>
<span id="cb854-4"><a href="toxicokinetic-modeling.html#cb854-4" tabindex="-1"></a>                                   <span class="at">variable.name =</span> <span class="st">&quot;Percentile&quot;</span>,</span>
<span id="cb854-5"><a href="toxicokinetic-modeling.html#cb854-5" tabindex="-1"></a>                                   <span class="at">value.name =</span> <span class="st">&quot;Css_slope&quot;</span>)</span></code></pre></div>
<pre><code>## Warning in melt.default(css_3compss, id.vars = &quot;CAS_factor&quot;, measure.vars =
## c(&quot;5%&quot;, : The melt generic in data.table has been passed a data.frame and will
## attempt to redirect to the relevant reshape2 method; please note that reshape2
## is superseded and is no longer actively developed, and this redirection is now
## deprecated. To continue using melt methods from reshape2 while both libraries
## are attached, e.g. melt.list, you can prepend the namespace, i.e.
## reshape2::melt(css_3compss). In the next version, this warning will become an
## error.</code></pre>
<div class="sourceCode" id="cb856"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb856-1"><a href="toxicokinetic-modeling.html#cb856-1" tabindex="-1"></a><span class="fu">head</span>(css_3compss_melt)</span></code></pre></div>
<pre><code>##   CAS_factor Percentile Css_slope
## 1  2971-36-0         5%    0.1449
## 2    94-75-7         5%   18.1500
## 3    94-82-6         5%   56.9200
## 4    90-43-7         5%   23.5200
## 5  1007-28-9         5%    2.6670
## 6 71751-41-2         5%    0.7825</code></pre>
<p>Plot the slope percentiles. Use a log scale for the y-axis because the slopes span orders of magnitude. Suppress the x-axis labels (the CASRNs) because they are not readable anyway.</p>
<div class="sourceCode" id="cb858"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb858-1"><a href="toxicokinetic-modeling.html#cb858-1" tabindex="-1"></a><span class="fu">ggplot</span>(css_3compss_melt) <span class="sc">+</span></span>
<span id="cb858-2"><a href="toxicokinetic-modeling.html#cb858-2" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x=</span>CAS_factor,</span>
<span id="cb858-3"><a href="toxicokinetic-modeling.html#cb858-3" tabindex="-1"></a>                 <span class="at">y =</span> Css_slope,</span>
<span id="cb858-4"><a href="toxicokinetic-modeling.html#cb858-4" tabindex="-1"></a>                 <span class="at">color =</span> Percentile)) <span class="sc">+</span></span>
<span id="cb858-5"><a href="toxicokinetic-modeling.html#cb858-5" tabindex="-1"></a>  <span class="fu">scale_color_brewer</span>(<span class="at">palette =</span> <span class="st">&quot;Set2&quot;</span>) <span class="sc">+</span> <span class="co">#use better color scheme than default</span></span>
<span id="cb858-6"><a href="toxicokinetic-modeling.html#cb858-6" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span> <span class="co">#use log scale for y axis</span></span>
<span id="cb858-7"><a href="toxicokinetic-modeling.html#cb858-7" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Chemical&quot;</span>) <span class="sc">+</span></span>
<span id="cb858-8"><a href="toxicokinetic-modeling.html#cb858-8" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Css-dose slope (uM per mg/kg/day)&quot;</span>) <span class="sc">+</span></span>
<span id="cb858-9"><a href="toxicokinetic-modeling.html#cb858-9" tabindex="-1"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">&quot;l&quot;</span>) <span class="sc">+</span> <span class="co">#add log ticks to y axis</span></span>
<span id="cb858-10"><a href="toxicokinetic-modeling.html#cb858-10" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="co">#plot with white plot background instead of gray</span></span>
<span id="cb858-11"><a href="toxicokinetic-modeling.html#cb858-11" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), <span class="co">#suppress x-axis labels</span></span>
<span id="cb858-12"><a href="toxicokinetic-modeling.html#cb858-12" tabindex="-1"></a>        <span class="at">panel.grid.major.x =</span> <span class="fu">element_blank</span>(), <span class="co">#suppress vertical grid lines</span></span>
<span id="cb858-13"><a href="toxicokinetic-modeling.html#cb858-13" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="fu">c</span>(<span class="fl">0.1</span>,<span class="fl">0.8</span>) <span class="co">#place legend in lower right corner</span></span>
<span id="cb858-14"><a href="toxicokinetic-modeling.html#cb858-14" tabindex="-1"></a>        ) </span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-632-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Chemicals along the x-axis are in order from lowest to highest median (50th percentile) predicted <span class="math inline">\(C_{ss}\)</span>-dose slope. The orange points represent that 50th percentile <span class="math inline">\(C_{ss}\)</span>-dose slope for each chemical. The green points represent the 5th percentile <span class="math inline">\(C_{ss}\)</span>-dose slopes, and the purple points represent the 95th percentile <span class="math inline">\(C_{ss}\)</span>-dose slope for each chemical. Each chemical has one orange point (50th percentile), one green point (5th percentile), and one purple point (95th percentile), characterizing the distribution of <span class="math inline">\(C_{ss}\)</span>-dose slopes across the U.S. population for that chemical. The width of the distribution for each chemical is roughly represented by the vertical distance between the green and purple points for that chemical.</p>
</div>
<div id="answer-to-environmental-health-question-4-9" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 4<a href="toxicokinetic-modeling.html#answer-to-environmental-health-question-4-9" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #4</strong></em>: Considering the chemicals evaluated in the above TK modeling example, do the <span class="math inline">\(C_{ss}\)</span>-dose slope distributions become wider as the median <span class="math inline">\(C_{ss}\)</span>-dose slope increases?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: No – the <span class="math inline">\(C_{ss}\)</span>-dose slope distributions generally become narrower as the median <span class="math inline">\(C_{ss}\)</span>-dose slope increases. This can be seen by looking at the right end of the plot, where the highest-median chemicals are located – the distance between the green points and purple points, representing the 5th and 95th percentiles, are much smaller for these higher-median chemicals.</p>
</div>
</div>
</div>
<div id="reverse-tk-calculating-administered-equivalent-doses-for-toxcast-bioactive-concentrations" class="section level2 hasAnchor">
<h2>Reverse TK: Calculating Administered Equivalent Doses for ToxCast Bioactive Concentrations<a href="toxicokinetic-modeling.html#reverse-tk-calculating-administered-equivalent-doses-for-toxcast-bioactive-concentrations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As described in an earlier section of this document, the slope defining the linear relation between <span class="math inline">\(C_{ss}\)</span> and dose is useful for reverse toxicokinetics: converting an internal dose metric to an external dose metric. The internal dose metric may, for example, be a concentration associated with an <em>in vivo</em> health effect, or <em>in vitro</em> bioactivity. Here, we will consider <em>in vitro</em> bioactivity – specifically, from the ToxCast program. ToxCast tests chemicals in multiple concentration-response format across a battery of <em>in vitro</em> assays that measure activity in a wide variety of biological endpoints. If a chemical showed any activity in an assay at any of its tested concentrations, then one metric of concentration associated with bioactivity is AC50 – the concentration at which the assay response is halfway between its minimum and its maximum.</p>
<p>The module won’t address the details of how ToxCast determines assay activity and AC50s from raw concentration-response data. There is an entire R package for the ToxCast data processing workflow, called <code>tcpl</code>. If you want to learn more about those details, <a href="https://www.epa.gov/chemical-research/toxcast-data-generation-toxcast-pipeline-tcpl">start here</a>. Lots of information is available if you install the <code>tcpl</code> R package and look at the package vignette; it essentially walks you through the full ToxCast data processing workflow.</p>
<p>In this module, we will begin with pre-computed ToxCast AC50 values for various chemicals and assays. We will use <code>httk</code> to convert ToxCast AC50 values into administered equivalent doses (AEDs).</p>
<div id="loading-toxcast-ac50s" class="section level3 hasAnchor">
<h3>Loading ToxCast AC50s<a href="toxicokinetic-modeling.html#loading-toxcast-ac50s" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The latest public release of ToxCast high-throughput screening assay data can be downloaded <a href="https://www.epa.gov/chemical-research/exploring-toxcast-data-downloadable-data">here</a>. Previous public releases of ToxCast data included a matrix of AC50s by chemical and assay. The data format of the latest public release does not contain this kind of matrix. So this dataset was pre-processed to prepare a simple data.frame of AC50s for each chemical/assay combination for the purposes of this training module.</p>
<p>Read in the pre-processed data set and view the first few rows.</p>
<div class="sourceCode" id="cb859"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb859-1"><a href="toxicokinetic-modeling.html#cb859-1" tabindex="-1"></a>toxcast <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Module6_6_Input/Module6_6_InputData1.csv&quot;</span>)</span>
<span id="cb859-2"><a href="toxicokinetic-modeling.html#cb859-2" tabindex="-1"></a><span class="fu">head</span>(toxcast)</span></code></pre></div>
<pre><code>##                         Compound        CAS        DTXSID         aenm
## 1                  Acetohexamide   968-81-0 DTXSID7020007 ACEA_ER_80hr
## 2 2-Methoxyaniline hydrochloride   134-29-2 DTXSID8020092 ACEA_ER_80hr
## 3             Sodium L-ascorbate   134-03-2 DTXSID0020105 ACEA_ER_80hr
## 4                   Sodium azide 26628-22-8 DTXSID8020121 ACEA_ER_80hr
## 5               Benzotrichloride    98-07-7 DTXSID1020148 ACEA_ER_80hr
## 6                 Benzyl acetate   140-11-4 DTXSID0020151 ACEA_ER_80hr
##   log10_ac50
## 1  0.6524155
## 2 -1.3141432
## 3  0.8248535
## 4  1.9839338
## 5  1.8370790
## 6 -0.3299611</code></pre>
<p>The columns of this data frame are:</p>
<ul>
<li><p><code>Compound</code>: The compound name.</p></li>
<li><p><code>CAS</code>: The compound’s CASRN.</p></li>
<li><p><code>DTXSID</code>: The compound’s DSSTox Substance ID.</p></li>
<li><p><code>aenm</code>: Assay identifier. “aenm” stands for “Assay Endpoint Name.” More information about the ToxCast assays is available on the <a href="https://www.epa.gov/chemical-research/exploring-toxcast-data-downloadable-data">ToxCast data download page</a>.</p></li>
<li><p><code>log10_ac50</code>: The AC50 for the chemical/assay combination on each row, in log10 uM units.</p></li>
</ul>
<p>How many ToxCast chemicals are in this data set?</p>
<div class="sourceCode" id="cb861"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb861-1"><a href="toxicokinetic-modeling.html#cb861-1" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(toxcast<span class="sc">$</span>DTXSID))</span></code></pre></div>
<pre><code>## [1] 7863</code></pre>
</div>
<div id="answer-to-environmental-health-question-5-7" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 5<a href="toxicokinetic-modeling.html#answer-to-environmental-health-question-5-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #5</strong></em>: How many chemicals have available AC50 values to evaluate in the current ToxCast/Tox21 high-throughput screening database?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: 7863 chemicals.</p>
</div>
</div>
<div id="subsetting-the-toxcast-chemicals-to-include-those-that-are-also-in-httk" class="section level3 hasAnchor">
<h3>Subsetting the ToxCast Chemicals to include those that are also in <code>httk</code><a href="toxicokinetic-modeling.html#subsetting-the-toxcast-chemicals-to-include-those-that-are-also-in-httk" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Not all of the ToxCast chemicals have TK data built into <code>httk</code> such that we can perform reverse TK using the HTTK models. Let’s subset the ToxCast data to include only the chemicals for which we can run the 3-compartment steady-state models.</p>
<p>Previously, we used <code>get_cheminfo()</code> to get a list of chemicals for which we could run the 3-compartment steady state model, including the names, CASRNs, and DSSTox IDs of those chemicals. That list is stored in variable <code>chems_3compss</code>, a data.frame with compound name, CASRN, and DTXSID. Now, we can use that chemical list to subset the ToxCast data.</p>
<div class="sourceCode" id="cb863"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb863-1"><a href="toxicokinetic-modeling.html#cb863-1" tabindex="-1"></a>toxcast_httk <span class="ot">&lt;-</span> <span class="fu">subset</span>(toxcast,</span>
<span id="cb863-2"><a href="toxicokinetic-modeling.html#cb863-2" tabindex="-1"></a>                       <span class="at">subset =</span> toxcast<span class="sc">$</span>DTXSID <span class="sc">%in%</span></span>
<span id="cb863-3"><a href="toxicokinetic-modeling.html#cb863-3" tabindex="-1"></a>                         chems_3compss<span class="sc">$</span>DTXSID)</span></code></pre></div>
<p>How many chemicals are in this subset?</p>
<div class="sourceCode" id="cb864"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb864-1"><a href="toxicokinetic-modeling.html#cb864-1" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">unique</span>(toxcast_httk<span class="sc">$</span>DTXSID))</span></code></pre></div>
<pre><code>## [1] 869</code></pre>
<p>There were 869 <code>httk</code> chemicals for which we could run the 3-compartment steady-state model; only 911 of them had ToxCast data. Conversely, most of the 7863 ToxCast chemicals do not have TK data in <code>httk</code> such that we can run the 3-compartment steady state model.</p>
</div>
<div id="identifying-the-lower-bound-in-vitro-ac50-value-per-chemical" class="section level3 hasAnchor">
<h3>Identifying the Lower-Bound <em>In Vitro</em> AC50 Value per Chemical<a href="toxicokinetic-modeling.html#identifying-the-lower-bound-in-vitro-ac50-value-per-chemical" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>ToxCast/Tox21 screens chemicals across multiple assays, such that each chemical has multiple resulting AC50 values, spanning a range of values. For example, here are boxplots of the AC50s for the first 20 chemicals listed in <code>chems_3compss</code>. Note that the chemical identifiers, DTXSID, are used here in these visualizations to represent unique chemicals.</p>
<div class="sourceCode" id="cb866"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb866-1"><a href="toxicokinetic-modeling.html#cb866-1" tabindex="-1"></a><span class="fu">ggplot</span>(toxcast_httk[toxcast_httk<span class="sc">$</span>DTXSID <span class="sc">%in%</span></span>
<span id="cb866-2"><a href="toxicokinetic-modeling.html#cb866-2" tabindex="-1"></a>                      chems_3compss[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>,</span>
<span id="cb866-3"><a href="toxicokinetic-modeling.html#cb866-3" tabindex="-1"></a>                                    <span class="st">&quot;DTXSID&quot;</span>],</span>
<span id="cb866-4"><a href="toxicokinetic-modeling.html#cb866-4" tabindex="-1"></a>                    ]</span>
<span id="cb866-5"><a href="toxicokinetic-modeling.html#cb866-5" tabindex="-1"></a>       ) <span class="sc">+</span></span>
<span id="cb866-6"><a href="toxicokinetic-modeling.html#cb866-6" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>DTXSID, <span class="at">y =</span> log10_ac50)) <span class="sc">+</span></span>
<span id="cb866-7"><a href="toxicokinetic-modeling.html#cb866-7" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;log10 AC50&quot;</span>) <span class="sc">+</span></span>
<span id="cb866-8"><a href="toxicokinetic-modeling.html#cb866-8" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb866-9"><a href="toxicokinetic-modeling.html#cb866-9" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>,</span>
<span id="cb866-10"><a href="toxicokinetic-modeling.html#cb866-10" tabindex="-1"></a>                                   <span class="at">hjust =</span> <span class="dv">1</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-637-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Sometimes we have an interest in getting the equivalent dose for an AC50 for one specific assay. For example, if we happen to be interested in estrogen-receptor activity, we might look specifically at one of the assays that measures estrogen receptor activity.</p>
<p>However, sometimes we just want a general idea of what concentrations showed bioactivity in <em>any</em> of the ToxCast assays, regardless of the specific biological endpoint of each assay. In this case, typically, we are interested in a “reasonable lower bound” of bioactive concentrations across assays for each chemical. Intuitively, we suspect that the very lowest AC50s for each chemical might represent false activity. Therefore, we often select the tenth percentile of ToxCast AC50s for each chemical as that “reasonable lower bound” on bioactive concentrations.</p>
<p>Let’s calculate the tenth percentile ToxCast AC50 for each chemical. Here, we use the base-R function <code>aggregate</code>, which groups a vector (specified in the <code>x</code> argument) by a list of factors (specified in the <code>by</code> argument), and applies a function to each group (specified in the <code>FUN</code> argument). You can add any extra arguments to the <code>FUN</code> function as named arguments to <code>aggregate</code>.</p>
<div class="sourceCode" id="cb867"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb867-1"><a href="toxicokinetic-modeling.html#cb867-1" tabindex="-1"></a>toxcast_httk_P10 <span class="ot">&lt;-</span> <span class="fu">aggregate</span>(<span class="at">x =</span> toxcast_httk<span class="sc">$</span>log10_ac50, <span class="co">#aggregate the AC50s</span></span>
<span id="cb867-2"><a href="toxicokinetic-modeling.html#cb867-2" tabindex="-1"></a>                              <span class="at">by =</span> <span class="fu">list</span>(<span class="at">DTXSID =</span> toxcast_httk<span class="sc">$</span>DTXSID), <span class="co">#group AC50s by DTXSID</span></span>
<span id="cb867-3"><a href="toxicokinetic-modeling.html#cb867-3" tabindex="-1"></a>                              <span class="at">FUN =</span> quantile, <span class="co">#the function to apply to each group</span></span>
<span id="cb867-4"><a href="toxicokinetic-modeling.html#cb867-4" tabindex="-1"></a>                              <span class="at">prob =</span> <span class="fl">0.1</span>) <span class="co">#an argument to the quantile() function</span></span>
<span id="cb867-5"><a href="toxicokinetic-modeling.html#cb867-5" tabindex="-1"></a><span class="co">#by default the names of the output data.frame will be &#39;DTXSID&#39; and &#39;x&#39;</span></span>
<span id="cb867-6"><a href="toxicokinetic-modeling.html#cb867-6" tabindex="-1"></a><span class="co">#let&#39;s change &#39;x&#39; to be a more informative name</span></span>
<span id="cb867-7"><a href="toxicokinetic-modeling.html#cb867-7" tabindex="-1"></a><span class="fu">names</span>(toxcast_httk_P10) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;DTXSID&quot;</span>, <span class="st">&quot;log10_ac50_P10&quot;</span>)</span></code></pre></div>
<p>Let’s transform the tenth-percentile AC50 values back to the natural scale (they are currently on the log10 scale) and put them in a new column <code>AC50</code>. These AC50s will be in uM.</p>
<div class="sourceCode" id="cb868"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb868-1"><a href="toxicokinetic-modeling.html#cb868-1" tabindex="-1"></a>toxcast_httk_P10<span class="sc">$</span>AC50 <span class="ot">&lt;-</span> <span class="dv">10</span><span class="sc">^</span>(toxcast_httk_P10<span class="sc">$</span>log10_ac50_P10)</span></code></pre></div>
<p>View the first few rows:</p>
<div class="sourceCode" id="cb869"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb869-1"><a href="toxicokinetic-modeling.html#cb869-1" tabindex="-1"></a><span class="fu">head</span>(toxcast_httk_P10)</span></code></pre></div>
<pre><code>##          DTXSID log10_ac50_P10        AC50
## 1 DTXSID0020022      0.8932512  7.82079968
## 2 DTXSID0020232      0.2903537  1.95143342
## 3 DTXSID0020286     -1.3763735  0.04203649
## 4 DTXSID0020311      1.1513461 14.16922669
## 5 DTXSID0020319     -0.1934652  0.64052306
## 6 DTXSID0020365      0.4308058  2.69653361</code></pre>
</div>
<div id="calculating-equivalent-doses-for-10th-percentile-toxcast-ac50s" class="section level3 hasAnchor">
<h3>Calculating Equivalent Doses for 10th Percentile ToxCast AC50s<a href="toxicokinetic-modeling.html#calculating-equivalent-doses-for-10th-percentile-toxcast-ac50s" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We can calculate equivalent doses in one line of R code – again including all of the Monte Carlo for TK uncertainty and variability – just by using the <code>httk</code> function <code>calc_mc_oral_equiv()</code>.</p>
<p>Note that in <code>calc_mc_oral_equiv()</code>, the <code>which.quantile</code> argument refers to the quantile of the <span class="math inline">\(C_{ss}\)</span>-dose slope, not the quantile of the equivalent dose itself. So specifying <code>which.quantile = 0.95</code> will yield a <em>lower</em> equivalent dose than <code>which.quantile = 0.05</code>.</p>
<p>Under the hood, <code>calc_mc_oral_equiv()</code> first calls <code>calc_mc_css()</code> to get percentiles of the <span class="math inline">\(C_{ss}\)</span>-dose slope for a chemical. It then divides a user-specified target concentration (specified in argument <code>conc</code>) by each quantile of <span class="math inline">\(C_{ss}\)</span>-dose slope to get the equivalent dose corresponding to that target concentration for each slope quantile.</p>
<p>Here, we’re using the <code>mapply()</code> function in base R to call <code>calc_mc_oral_equiv()</code> in a loop over chemicals. This is because <code>calc_mc_oral_equiv()</code> requires two chemical-specific arguments – the chemical identifier and the concentration for which to compute the equivalent dose. <code>mapply()</code> lets us provide vectors of values for each argument (in the named arguments <code>dtxsid</code> and <code>conc</code>), and will automatically loop over those vectors. We also use the argument <code>MoreArgs</code>, a named list of additional arguments to the function in <code>FUN</code> that will be the same for every iteration of the loop. Note that this line of code takes a few minutes to run.</p>
<div class="sourceCode" id="cb871"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb871-1"><a href="toxicokinetic-modeling.html#cb871-1" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb871-2"><a href="toxicokinetic-modeling.html#cb871-2" tabindex="-1"></a></span>
<span id="cb871-3"><a href="toxicokinetic-modeling.html#cb871-3" tabindex="-1"></a><span class="fu">system.time</span>(</span>
<span id="cb871-4"><a href="toxicokinetic-modeling.html#cb871-4" tabindex="-1"></a>  <span class="fu">suppressWarnings</span>(</span>
<span id="cb871-5"><a href="toxicokinetic-modeling.html#cb871-5" tabindex="-1"></a>  toxcast_equiv_dose <span class="ot">&lt;-</span> <span class="fu">mapply</span>(<span class="at">FUN =</span> calc_mc_oral_equiv,</span>
<span id="cb871-6"><a href="toxicokinetic-modeling.html#cb871-6" tabindex="-1"></a>  <span class="at">conc =</span> toxcast_httk_P10<span class="sc">$</span>AC50,</span>
<span id="cb871-7"><a href="toxicokinetic-modeling.html#cb871-7" tabindex="-1"></a>    <span class="at">dtxsid =</span> toxcast_httk_P10<span class="sc">$</span>DTXSID,</span>
<span id="cb871-8"><a href="toxicokinetic-modeling.html#cb871-8" tabindex="-1"></a>  <span class="at">MoreArgs =</span> <span class="fu">list</span>(<span class="at">model =</span> <span class="st">&quot;3compartmentss&quot;</span>,  <span class="co">#model to use</span></span>
<span id="cb871-9"><a href="toxicokinetic-modeling.html#cb871-9" tabindex="-1"></a>                  <span class="at">which.quantile =</span> <span class="fu">c</span>(<span class="fl">0.05</span>, <span class="fl">0.5</span>, <span class="fl">0.95</span>), <span class="co">#quantiles of Css-dose slope</span></span>
<span id="cb871-10"><a href="toxicokinetic-modeling.html#cb871-10" tabindex="-1"></a>                       <span class="at">suppress.messages =</span> <span class="cn">TRUE</span>)</span>
<span id="cb871-11"><a href="toxicokinetic-modeling.html#cb871-11" tabindex="-1"></a>  )</span>
<span id="cb871-12"><a href="toxicokinetic-modeling.html#cb871-12" tabindex="-1"></a>)</span>
<span id="cb871-13"><a href="toxicokinetic-modeling.html#cb871-13" tabindex="-1"></a>)</span>
<span id="cb871-14"><a href="toxicokinetic-modeling.html#cb871-14" tabindex="-1"></a></span>
<span id="cb871-15"><a href="toxicokinetic-modeling.html#cb871-15" tabindex="-1"></a><span class="co">#by default, the results are a 3 x 869 matrix, where rows are quantiles and columns are chemicals</span></span>
<span id="cb871-16"><a href="toxicokinetic-modeling.html#cb871-16" tabindex="-1"></a></span>
<span id="cb871-17"><a href="toxicokinetic-modeling.html#cb871-17" tabindex="-1"></a>toxcast_equiv_dose <span class="ot">&lt;-</span> <span class="fu">t</span>(toxcast_equiv_dose) <span class="co">#transpose so that rows are chemicals</span></span>
<span id="cb871-18"><a href="toxicokinetic-modeling.html#cb871-18" tabindex="-1"></a>toxcast_equiv_dose <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(toxcast_equiv_dose) <span class="co">#convert to data.frame</span></span>
<span id="cb871-19"><a href="toxicokinetic-modeling.html#cb871-19" tabindex="-1"></a><span class="fu">head</span>(toxcast_equiv_dose) <span class="co">#look at first few rows</span></span></code></pre></div>
<p>Let’s add the DTXSIDs back into this data.frame.</p>
<div class="sourceCode" id="cb872"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb872-1"><a href="toxicokinetic-modeling.html#cb872-1" tabindex="-1"></a>toxcast_equiv_dose<span class="sc">$</span>DTXSID <span class="ot">&lt;-</span> toxcast_httk_P10<span class="sc">$</span>DTXSID</span></code></pre></div>
<p>We can get the names of these chemicals by using the list of chemicals for which the 3-compartment steady-state model can be parameterized, which was stored in the variable `<code>chems_3compss</code>. In that dataframe, we have the compound name and CASRN corresponding to each DTXSID.</p>
<div class="sourceCode" id="cb873"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb873-1"><a href="toxicokinetic-modeling.html#cb873-1" tabindex="-1"></a><span class="fu">head</span>(chems_3compss)</span></code></pre></div>
<pre><code>##                                                Compound        CAS
## 1 2,2-bis(4-hydroxyphenyl)-1,1,1-trichloroethane (hpte)  2971-36-0
## 2                                                 2,4-d    94-75-7
## 3                                                2,4-db    94-82-6
## 4                                        2-phenylphenol    90-43-7
## 5                                6-desisopropylatrazine  1007-28-9
## 6                                             Abamectin 71751-41-2
##          DTXSID
## 1 DTXSID8022325
## 2 DTXSID0020442
## 3 DTXSID7024035
## 4 DTXSID2021151
## 5 DTXSID0037495
## 6 DTXSID8023892</code></pre>
<p>Merge <code>chems_3compss</code> with <code>toxcast_equiv_dose</code>.</p>
<div class="sourceCode" id="cb875"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb875-1"><a href="toxicokinetic-modeling.html#cb875-1" tabindex="-1"></a>toxcast_equiv_dose <span class="ot">&lt;-</span> <span class="fu">merge</span>(chems_3compss,</span>
<span id="cb875-2"><a href="toxicokinetic-modeling.html#cb875-2" tabindex="-1"></a>                            toxcast_equiv_dose,</span>
<span id="cb875-3"><a href="toxicokinetic-modeling.html#cb875-3" tabindex="-1"></a>                            <span class="at">by =</span> <span class="st">&quot;DTXSID&quot;</span>,</span>
<span id="cb875-4"><a href="toxicokinetic-modeling.html#cb875-4" tabindex="-1"></a>                            <span class="at">all.x =</span> <span class="cn">FALSE</span>,</span>
<span id="cb875-5"><a href="toxicokinetic-modeling.html#cb875-5" tabindex="-1"></a>                            <span class="at">all.y =</span> <span class="cn">TRUE</span>)</span>
<span id="cb875-6"><a href="toxicokinetic-modeling.html#cb875-6" tabindex="-1"></a></span>
<span id="cb875-7"><a href="toxicokinetic-modeling.html#cb875-7" tabindex="-1"></a><span class="fu">head</span>(toxcast_equiv_dose)</span></code></pre></div>
<pre><code>##          DTXSID                 Compound        CAS      5%      50%     95%
## 1 DTXSID0020022              Acifluorfen 50594-66-6  0.3224  0.10940 0.03568
## 2 DTXSID0020232                 Caffeine    58-08-2  2.1720  0.89270 0.32510
## 3 DTXSID0020286 3-chloro-4-methylaniline    95-74-9  1.7450  0.30310 0.04205
## 4 DTXSID0020311                  Monuron   150-68-5 46.5300 12.37000 2.69300
## 5 DTXSID0020319           Chlorothalonil  1897-45-6  8.4260  0.05517 0.01892
## 6 DTXSID0020365            Cyclosporin a 59865-13-3  3.1970  0.57990 0.05113</code></pre>
<p>To find the chemicals with the lowest equivalent doses at the 95th percentile level (corresponding to the most-sensitive 5% of the population), sort this data.frame in ascending order on the <code>95%</code> column.</p>
<div class="sourceCode" id="cb877"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb877-1"><a href="toxicokinetic-modeling.html#cb877-1" tabindex="-1"></a>toxcast_equiv_dose <span class="ot">&lt;-</span> toxcast_equiv_dose[<span class="fu">order</span>(toxcast_equiv_dose<span class="sc">$</span><span class="st">`</span><span class="at">95%</span><span class="st">`</span>), ]</span>
<span id="cb877-2"><a href="toxicokinetic-modeling.html#cb877-2" tabindex="-1"></a><span class="fu">head</span>(toxcast_equiv_dose, <span class="dv">10</span>) <span class="co">#first ten rows of sorted table</span></span></code></pre></div>
<pre><code>##            DTXSID                       Compound         CAS        5%
## 8   DTXSID0020442                          2,4-d     94-75-7 3.260e-06
## 771 DTXSID8037594                     Secbumeton  26259-45-0 6.757e-05
## 349 DTXSID4020533                    1,4-dioxane    123-91-1 7.834e-05
## 129 DTXSID1026035 Sodium 2-mercaptobenzothiolate   2492-26-4 1.336e-04
## 727 DTXSID8023214                  Levothyroxine     51-48-9 1.374e-04
## 120 DTXSID1024049                  Diflubenzuron  35367-38-5 4.975e-05
## 726 DTXSID8023187                       Ketamine   6740-88-1 4.916e-04
## 604 DTXSID6046478                      Gestodene  60282-87-3 3.727e-04
## 588 DTXSID6032356                       Cycloate   1134-23-2 3.842e-04
## 696 DTXSID7047306                      Cp-634384 290352-28-2 2.441e-04
##           50%       95%
## 8   8.940e-07 1.790e-07
## 771 1.245e-05 2.045e-06
## 349 1.553e-05 2.281e-06
## 129 3.016e-05 3.533e-06
## 727 1.508e-05 5.051e-06
## 120 1.744e-05 5.702e-06
## 726 7.533e-05 5.728e-06
## 604 6.351e-05 7.311e-06
## 588 8.099e-05 8.092e-06
## 696 4.766e-05 1.073e-05</code></pre>
</div>
<div id="answer-to-environmental-health-question-6-4" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 6<a href="toxicokinetic-modeling.html#answer-to-environmental-health-question-6-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #6</strong></em>: What are the chemicals with the three lowest predicted equivalent doses (for tenth-percentile ToxCast AC50s), for the most-sensitive 5% of the population?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: 2,4-d; secbumeton, and 1,4-dioxane</p>
</div>
</div>
</div>
<div id="comparing-equivalent-doses-estimated-to-elicit-toxicity-hazard-to-external-exposure-estimates-exposure-for-chemical-prioritization-by-bioactivity-exposure-ratios-bers" class="section level2 hasAnchor">
<h2>Comparing Equivalent Doses Estimated to Elicit Toxicity (Hazard) to External Exposure Estimates (Exposure), for Chemical Prioritization by Bioactivity-Exposure Ratios (BERs)<a href="toxicokinetic-modeling.html#comparing-equivalent-doses-estimated-to-elicit-toxicity-hazard-to-external-exposure-estimates-exposure-for-chemical-prioritization-by-bioactivity-exposure-ratios-bers" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To estimate potential risk, hazard – in the form of the equivalent dose for the 10th percentile Toxcast AC50 – now needs to be compared to exposure. A quantitative metric for this comparison is the ratio of the lowest 5% of equivalent doses to the highest 5% of potential exposures. This metric is termed the Bioactivity-Exposure Ratio, or BER. Lower BER corresponds to higher potential risk. With BERs calculated for each chemical, we can ultimately rank all of the chemicals from lowest to highest BER, to achieve a chemical prioritization based on potential risk.</p>
<div id="human-exposure-estimates" class="section level3 hasAnchor">
<h3>Human Exposure Estimates<a href="toxicokinetic-modeling.html#human-exposure-estimates" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Here, we will use exposure estimates that have been inferred from CDC NHANES urinary biomonitoring data (Ring et al., 2019). These estimates consist of an estimated median, and estimated upper and lower 95% credible interval bounds representing uncertainty in that estimated median. These estimates are provided here in the following csv file:</p>
<div class="sourceCode" id="cb879"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb879-1"><a href="toxicokinetic-modeling.html#cb879-1" tabindex="-1"></a>exposure <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Module6_6_Input/Module6_6_InputData2.csv&quot;</span>)</span>
<span id="cb879-2"><a href="toxicokinetic-modeling.html#cb879-2" tabindex="-1"></a><span class="fu">head</span>(exposure) <span class="co">#view first few rows</span></span></code></pre></div>
<pre><code>##                                                                                   Compound
## 1                                        1,2,3,4,5,6-Hexachlorocyclohexane (mixed isomers)
## 2                                                                   1,2,4-Trichlorobenzene
## 3                                                                   1,3,5-Trichlorobenzene
## 4                                                                      1,3-Dichlorobenzene
## 5                                                                      1,4-Dichlorobenzene
## 6 2,3-Dihydro-2,2-dimethyl-7-benzofuryl 2,4-dimethyl-6-oxa-5-oxo-3-thia-2,4-diazadecanoate
##          DTXSID        CAS       Median        low95         up95
## 1 DTXSID7020687   608-73-1 1.237622e-07 1.144743e-10 8.464811e-06
## 2 DTXSID0021965   120-82-1 1.157387e-08 5.005691e-11 2.950528e-07
## 3 DTXSID8026195   108-70-3 8.970557e-08 1.292361e-10 2.563596e-06
## 4 DTXSID6022056   541-73-1 9.802174e-08 9.421797e-11 8.343616e-06
## 5 DTXSID1020431   106-46-7 9.050628e-05 8.456633e-05 9.731353e-05
## 6 DTXSID3052725 65907-30-4 4.245608e-08 1.070856e-10 1.236776e-06</code></pre>
</div>
<div id="merging-exposure-estimates-with-equivalent-dose-estimates-of-toxicity-hazard" class="section level3 hasAnchor">
<h3>Merging Exposure Estimates with Equivalent Dose Estimates of Toxicity (Hazard)<a href="toxicokinetic-modeling.html#merging-exposure-estimates-with-equivalent-dose-estimates-of-toxicity-hazard" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To calculate a BER for a chemical, it needs to have both an equivalent dose and an exposure estimate. Not all of the chemicals for which equivalent doses could be computed (<em>i.e.</em>, chemicals with both ToxCast AC50s and <code>httk</code> data) also have exposure estimates inferred from NHANES. Find out how many do.</p>
<div class="sourceCode" id="cb881"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb881-1"><a href="toxicokinetic-modeling.html#cb881-1" tabindex="-1"></a><span class="fu">length</span>(<span class="fu">intersect</span>(toxcast_equiv_dose<span class="sc">$</span>DTXSID, exposure<span class="sc">$</span>DTXSID))</span></code></pre></div>
<pre><code>## [1] 58</code></pre>
<p>This means that, using the ToxCast AC50 data for bioactive concentrations, the NHANES urinary inference data for exposures, and the <code>httk</code> package to convert bioactive concentrations to equivalent doses, we can compute BERs for 58 chemicals.</p>
<p>Merge together the ToxCast equivalent doses and the exposure data into a single data frame. Keep only the chemicals that have data in both ToxCast equivalent doses and exposure data frames.</p>
<div class="sourceCode" id="cb883"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb883-1"><a href="toxicokinetic-modeling.html#cb883-1" tabindex="-1"></a>hazard_exposure <span class="ot">&lt;-</span> <span class="fu">merge</span>(toxcast_equiv_dose,</span>
<span id="cb883-2"><a href="toxicokinetic-modeling.html#cb883-2" tabindex="-1"></a>                         exposure,</span>
<span id="cb883-3"><a href="toxicokinetic-modeling.html#cb883-3" tabindex="-1"></a>                         <span class="at">by =</span> <span class="st">&quot;DTXSID&quot;</span>,</span>
<span id="cb883-4"><a href="toxicokinetic-modeling.html#cb883-4" tabindex="-1"></a>                         <span class="at">all =</span> <span class="cn">FALSE</span>)</span>
<span id="cb883-5"><a href="toxicokinetic-modeling.html#cb883-5" tabindex="-1"></a><span class="fu">head</span>(hazard_exposure) <span class="co">#view first few rows of result</span></span></code></pre></div>
<pre><code>##          DTXSID           Compound.x      CAS.x        5%       50%       95%
## 1 DTXSID0020442                2,4-d    94-75-7 3.260e-06 8.940e-07 1.790e-07
## 2 DTXSID0021389          Trichlorfon    52-68-6 3.252e+01 5.706e+00 7.463e-01
## 3 DTXSID0024266    Pirimiphos-methyl 29232-93-7 1.128e+01 2.755e+00 4.818e-01
## 4 DTXSID1020855     Methyl parathion   298-00-0 6.342e+00 9.903e-01 1.149e-01
## 5 DTXSID1021956 Di-n-octyl phthalate   117-84-0 6.131e-04 2.551e-04 1.058e-04
## 6 DTXSID1022265             Alachlor 15972-60-8 4.350e+01 1.017e+01 1.563e+00
##                       Compound.y      CAS.y       Median        low95
## 1 2,4-Dichlorophenoxyacetic acid    94-75-7 6.349713e-06 3.100467e-06
## 2                    Trichlorfon    52-68-6 5.021397e-08 8.309014e-11
## 3              Pirimiphos-methyl 29232-93-7 2.569640e-07 1.765961e-10
## 4               Methyl parathion   298-00-0 7.396964e-08 1.559956e-10
## 5              Dioctyl phthalate   117-84-0 8.039695e-05 7.674705e-05
## 6                       Alachlor 15972-60-8 2.249506e-07 1.325551e-07
##           up95
## 1 1.815981e-05
## 2 3.302746e-06
## 3 6.640505e-05
## 4 3.575740e-06
## 5 8.422537e-05
## 6 3.111993e-07</code></pre>
</div>
<div id="plotting-hazard-and-exposure-together" class="section level3 hasAnchor">
<h3>Plotting Hazard and Exposure Together<a href="toxicokinetic-modeling.html#plotting-hazard-and-exposure-together" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>We can visually compare the equivalent doses and the inferred exposure estimates by plotting them together.</p>
<div class="sourceCode" id="cb885"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb885-1"><a href="toxicokinetic-modeling.html#cb885-1" tabindex="-1"></a><span class="fu">ggplot</span>(hazard_exposure) <span class="sc">+</span></span>
<span id="cb885-2"><a href="toxicokinetic-modeling.html#cb885-2" tabindex="-1"></a>  <span class="fu">geom_crossbar</span>(<span class="fu">aes</span>(<span class="at">x =</span> Compound.x, <span class="co">#Boxes for equivalent doses</span></span>
<span id="cb885-3"><a href="toxicokinetic-modeling.html#cb885-3" tabindex="-1"></a>                     <span class="at">y =</span> <span class="st">`</span><span class="at">50%</span><span class="st">`</span>,</span>
<span id="cb885-4"><a href="toxicokinetic-modeling.html#cb885-4" tabindex="-1"></a>                    <span class="at">ymax =</span> <span class="st">`</span><span class="at">5%</span><span class="st">`</span>,</span>
<span id="cb885-5"><a href="toxicokinetic-modeling.html#cb885-5" tabindex="-1"></a>                    <span class="at">ymin =</span> <span class="st">`</span><span class="at">95%</span><span class="st">`</span>,</span>
<span id="cb885-6"><a href="toxicokinetic-modeling.html#cb885-6" tabindex="-1"></a>                    <span class="at">color =</span> <span class="st">&quot;Equiv. dose&quot;</span>)) <span class="sc">+</span></span>
<span id="cb885-7"><a href="toxicokinetic-modeling.html#cb885-7" tabindex="-1"></a>  <span class="fu">geom_crossbar</span>(<span class="fu">aes</span>( <span class="at">x=</span> Compound.x, <span class="co">#Boxes for exposures</span></span>
<span id="cb885-8"><a href="toxicokinetic-modeling.html#cb885-8" tabindex="-1"></a>                     <span class="at">y =</span> Median,</span>
<span id="cb885-9"><a href="toxicokinetic-modeling.html#cb885-9" tabindex="-1"></a>                     <span class="at">ymax =</span> up95,</span>
<span id="cb885-10"><a href="toxicokinetic-modeling.html#cb885-10" tabindex="-1"></a>                     <span class="at">ymin =</span> low95,</span>
<span id="cb885-11"><a href="toxicokinetic-modeling.html#cb885-11" tabindex="-1"></a>                     <span class="at">color =</span> <span class="st">&quot;Exposure&quot;</span>)) <span class="sc">+</span></span>
<span id="cb885-12"><a href="toxicokinetic-modeling.html#cb885-12" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;Equiv. dose&quot;</span> <span class="ot">=</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb885-13"><a href="toxicokinetic-modeling.html#cb885-13" tabindex="-1"></a>                                <span class="st">&quot;Exposure&quot;</span> <span class="ot">=</span> <span class="st">&quot;Orange&quot;</span>),</span>
<span id="cb885-14"><a href="toxicokinetic-modeling.html#cb885-14" tabindex="-1"></a>                     <span class="at">name =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb885-15"><a href="toxicokinetic-modeling.html#cb885-15" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">label =</span> <span class="cf">function</span>(x) <span class="fu">str_trunc</span>(x, <span class="dv">20</span>)</span>
<span id="cb885-16"><a href="toxicokinetic-modeling.html#cb885-16" tabindex="-1"></a>                   ) <span class="sc">+</span> <span class="co">#truncate chemical names to 20 chars</span></span>
<span id="cb885-17"><a href="toxicokinetic-modeling.html#cb885-17" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb885-18"><a href="toxicokinetic-modeling.html#cb885-18" tabindex="-1"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">&quot;l&quot;</span>) <span class="sc">+</span></span>
<span id="cb885-19"><a href="toxicokinetic-modeling.html#cb885-19" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Equiv. dose or Exposure, mg/kg/day&quot;</span>) <span class="sc">+</span></span>
<span id="cb885-20"><a href="toxicokinetic-modeling.html#cb885-20" tabindex="-1"></a> <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb885-21"><a href="toxicokinetic-modeling.html#cb885-21" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>,</span>
<span id="cb885-22"><a href="toxicokinetic-modeling.html#cb885-22" tabindex="-1"></a>                                   <span class="at">hjust =</span> <span class="dv">1</span>,</span>
<span id="cb885-23"><a href="toxicokinetic-modeling.html#cb885-23" tabindex="-1"></a>                                   <span class="at">size =</span> <span class="dv">6</span>),</span>
<span id="cb885-24"><a href="toxicokinetic-modeling.html#cb885-24" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb885-25"><a href="toxicokinetic-modeling.html#cb885-25" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;top&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-649-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="calculating-bioactivity-exposure-ratios-bers" class="section level3 hasAnchor">
<h3>Calculating Bioactivity-Exposure Ratios (BERs)<a href="toxicokinetic-modeling.html#calculating-bioactivity-exposure-ratios-bers" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The bioactivity-exposure ratio (BER) is simply the ratio of the lower-end equivalent dose (for the most-sensitive 5% of the population) divided by the upper-end estimated exposure (here, the upper bound on the inferred population median exposure). In the data frame <code>hazard_exposure</code> containing the hazard and exposure data, the lower-end equivalent dose is in column <code>95%</code> (corresponding to the 95th-percentile <span class="math inline">\(C_{ss}\)</span>-dose slope) and the upper-end exposure is in column <code>up95</code>. Calculate the BER, and assign the result to a new column in the <code>hazard_exposure</code> data frame called <code>BER</code>.</p>
<div class="sourceCode" id="cb886"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb886-1"><a href="toxicokinetic-modeling.html#cb886-1" tabindex="-1"></a>hazard_exposure[[<span class="st">&quot;BER&quot;</span>]] <span class="ot">&lt;-</span> hazard_exposure[[<span class="st">&quot;95%&quot;</span>]]<span class="sc">/</span>hazard_exposure[[<span class="st">&quot;up95&quot;</span>]]</span></code></pre></div>
</div>
<div id="prioritizing-chemicals-by-ber" class="section level3 hasAnchor">
<h3>Prioritizing Chemicals by BER<a href="toxicokinetic-modeling.html#prioritizing-chemicals-by-ber" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To prioritize chemicals according to potential risk, they can be sorted from lowest to highest BER. The lower the BER, the higher the priority.</p>
<p>Sort the rows of the data.frame from lowest to highest BER.</p>
<div class="sourceCode" id="cb887"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb887-1"><a href="toxicokinetic-modeling.html#cb887-1" tabindex="-1"></a>hazard_exposure <span class="ot">&lt;-</span> hazard_exposure[<span class="fu">order</span>(hazard_exposure<span class="sc">$</span>BER), ]</span>
<span id="cb887-2"><a href="toxicokinetic-modeling.html#cb887-2" tabindex="-1"></a><span class="fu">head</span>(hazard_exposure)</span></code></pre></div>
<pre><code>##           DTXSID                    Compound.x    CAS.x        5%       50%
## 1  DTXSID0020442                         2,4-d  94-75-7 3.260e-06 8.940e-07
## 29 DTXSID5020607 Diethylhexyl phthalate (dehp) 117-81-7 1.674e-03 6.497e-04
## 5  DTXSID1021956          Di-n-octyl phthalate 117-84-0 6.131e-04 2.551e-04
## 19 DTXSID3022455            Dimethyl phthalate 131-11-3 1.410e-03 4.971e-04
## 25 DTXSID4022529                 Methylparaben  99-76-3 2.682e+00 4.710e-01
## 27 DTXSID4032613                  Fenitrothion 122-14-5 3.992e-01 6.875e-02
##          95%                     Compound.y    CAS.y       Median        low95
## 1  1.790e-07 2,4-Dichlorophenoxyacetic acid  94-75-7 6.349713e-06 3.100467e-06
## 29 2.774e-04     Di(2-ethylhexyl) phthalate 117-81-7 9.343466e-04 9.133541e-04
## 5  1.058e-04              Dioctyl phthalate 117-84-0 8.039695e-05 7.674705e-05
## 19 1.109e-04             Dimethyl phthalate 131-11-3 1.413887e-05 1.314067e-05
## 25 6.599e-02                  Methylparaben  99-76-3 9.525392e-04 8.949158e-04
## 27 7.191e-03                   Fenitrothion 122-14-5 2.055267e-07 1.277038e-10
##            up95          BER
## 1  1.815981e-05 9.856933e-03
## 29 9.546859e-04 2.905668e-01
## 5  8.422537e-05 1.256154e+00
## 19 1.520612e-05 7.293116e+00
## 25 1.013307e-03 6.512342e+01
## 27 5.930099e-05 1.212627e+02</code></pre>
<p>The hazard-exposure plot above showed chemicals in alphabetical order. It can be revised to show chemicals in order of priority, from lowest to highest BER.</p>
<p>First, create a categorical (factor) variable for the compound names, whose levels are in order of increasing BER. (Since we already sorted the data.frame in order of increasing BER, we can just take the compound names in the order that they appear.)</p>
<div class="sourceCode" id="cb889"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb889-1"><a href="toxicokinetic-modeling.html#cb889-1" tabindex="-1"></a>hazard_exposure<span class="sc">$</span>Compound_factor <span class="ot">&lt;-</span> <span class="fu">factor</span>(hazard_exposure<span class="sc">$</span>Compound.x,</span>
<span id="cb889-2"><a href="toxicokinetic-modeling.html#cb889-2" tabindex="-1"></a>                                          <span class="at">levels =</span> hazard_exposure<span class="sc">$</span>Compound.x)</span></code></pre></div>
<p>Now, make the same plot as before, but use <code>Compound_factor</code> as the x-axis variable instead of <code>Compound</code>.</p>
<div class="sourceCode" id="cb890"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb890-1"><a href="toxicokinetic-modeling.html#cb890-1" tabindex="-1"></a><span class="fu">ggplot</span>(hazard_exposure) <span class="sc">+</span></span>
<span id="cb890-2"><a href="toxicokinetic-modeling.html#cb890-2" tabindex="-1"></a>  <span class="fu">geom_crossbar</span>(<span class="fu">aes</span>(<span class="at">x =</span> Compound_factor, <span class="co">#Boxes for equivalent dose</span></span>
<span id="cb890-3"><a href="toxicokinetic-modeling.html#cb890-3" tabindex="-1"></a>                     <span class="at">y =</span> <span class="st">`</span><span class="at">50%</span><span class="st">`</span>,</span>
<span id="cb890-4"><a href="toxicokinetic-modeling.html#cb890-4" tabindex="-1"></a>                    <span class="at">ymax =</span> <span class="st">`</span><span class="at">5%</span><span class="st">`</span>,</span>
<span id="cb890-5"><a href="toxicokinetic-modeling.html#cb890-5" tabindex="-1"></a>                    <span class="at">ymin =</span> <span class="st">`</span><span class="at">95%</span><span class="st">`</span>,</span>
<span id="cb890-6"><a href="toxicokinetic-modeling.html#cb890-6" tabindex="-1"></a>                    <span class="at">color =</span> <span class="st">&quot;Equiv. dose&quot;</span>)) <span class="sc">+</span></span>
<span id="cb890-7"><a href="toxicokinetic-modeling.html#cb890-7" tabindex="-1"></a>  <span class="fu">geom_crossbar</span>(<span class="fu">aes</span>( <span class="at">x=</span> Compound_factor, <span class="co">#Boxes for exposure</span></span>
<span id="cb890-8"><a href="toxicokinetic-modeling.html#cb890-8" tabindex="-1"></a>                     <span class="at">y =</span> Median,</span>
<span id="cb890-9"><a href="toxicokinetic-modeling.html#cb890-9" tabindex="-1"></a>                     <span class="at">ymax =</span> up95,</span>
<span id="cb890-10"><a href="toxicokinetic-modeling.html#cb890-10" tabindex="-1"></a>                     <span class="at">ymin =</span> low95,</span>
<span id="cb890-11"><a href="toxicokinetic-modeling.html#cb890-11" tabindex="-1"></a>                     <span class="at">color =</span> <span class="st">&quot;Exposure&quot;</span>)) <span class="sc">+</span></span>
<span id="cb890-12"><a href="toxicokinetic-modeling.html#cb890-12" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;Equiv. dose&quot;</span> <span class="ot">=</span> <span class="st">&quot;black&quot;</span>,</span>
<span id="cb890-13"><a href="toxicokinetic-modeling.html#cb890-13" tabindex="-1"></a>                                <span class="st">&quot;Exposure&quot;</span> <span class="ot">=</span> <span class="st">&quot;Orange&quot;</span>),</span>
<span id="cb890-14"><a href="toxicokinetic-modeling.html#cb890-14" tabindex="-1"></a>                     <span class="at">name =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb890-15"><a href="toxicokinetic-modeling.html#cb890-15" tabindex="-1"></a>    <span class="fu">scale_x_discrete</span>(<span class="at">label =</span> <span class="cf">function</span>(x) <span class="fu">str_trunc</span>(x, <span class="dv">20</span>)</span>
<span id="cb890-16"><a href="toxicokinetic-modeling.html#cb890-16" tabindex="-1"></a>                   ) <span class="sc">+</span> <span class="co">#truncate chemical names</span></span>
<span id="cb890-17"><a href="toxicokinetic-modeling.html#cb890-17" tabindex="-1"></a>  <span class="fu">scale_y_log10</span>() <span class="sc">+</span></span>
<span id="cb890-18"><a href="toxicokinetic-modeling.html#cb890-18" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">&quot;Equiv. dose or Exposure, mg/kg/day&quot;</span>) <span class="sc">+</span></span>
<span id="cb890-19"><a href="toxicokinetic-modeling.html#cb890-19" tabindex="-1"></a>  <span class="fu">annotation_logticks</span>(<span class="at">sides =</span> <span class="st">&quot;l&quot;</span>) <span class="sc">+</span></span>
<span id="cb890-20"><a href="toxicokinetic-modeling.html#cb890-20" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb890-21"><a href="toxicokinetic-modeling.html#cb890-21" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>,</span>
<span id="cb890-22"><a href="toxicokinetic-modeling.html#cb890-22" tabindex="-1"></a>                                   <span class="at">hjust =</span> <span class="dv">1</span>,</span>
<span id="cb890-23"><a href="toxicokinetic-modeling.html#cb890-23" tabindex="-1"></a>                                   <span class="at">size =</span> <span class="dv">6</span>),</span>
<span id="cb890-24"><a href="toxicokinetic-modeling.html#cb890-24" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb890-25"><a href="toxicokinetic-modeling.html#cb890-25" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&quot;top&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-653-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now, the chemicals are displayed in order of increasing BER. From left to right, you can visually see the distance increase between the lower bound of equivalent doses (the bottom of the black boxes) and the upper bound of exposure estimates (the top of the orange boxes). Since the y-axis is put on a log10 scale, the distance between the boxes corresponds to the BER. We can gather a lot of information from this plot!</p>
</div>
<div id="answer-to-environmental-health-question-7-2" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 7<a href="toxicokinetic-modeling.html#answer-to-environmental-health-question-7-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #7</strong></em>: Based on httk modeling estimates, are chemicals with higher bioactivity-exposure ratios always less potent than chemicals with lower bioactivity-exposure ratios?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: Answer: No – some chemicals with high potency (low equivalent doses) demonstrate high BERs because they have relatively low human exposure estimates; and vice versa.</p>
</div>
</div>
<div id="answer-to-environmental-health-question-8-1" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 8<a href="toxicokinetic-modeling.html#answer-to-environmental-health-question-8-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can also answer <strong>Environmental Health Question #8</strong></em>: Based on httk modeling estimates, do chemicals with higher bioactivity-exposure ratios always have lower estimated exposures than chemicals with lower bioactivity-exposure ratios?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: No – some chemicals with high estimated exposures have equivalent doses that are higher still, resulting in a high BER despite the higher estimated exposure. Likewise, some chemicals with low estimated exposures also have lower equivalent doses, resulting in a low BER despite the low estimated exposure.</p>
</div>
</div>
<div id="answer-to-environmental-health-question-9" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 9<a href="toxicokinetic-modeling.html#answer-to-environmental-health-question-9" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can also answer <strong>Environmental Health Question #9</strong></em>: How are chemical prioritization results different when using only hazard information vs. only exposure information vs. bioactivity-exposure ratios?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: When chemicals are prioritized solely on the basis of hazard, more-potent chemicals will be highly prioritized. However, if humans are never exposed to these chemicals, or exposure is extremely low compared to potency, then despite the high potency, the potential risk may be low. Conversely, if chemicals are prioritized solely on the basis of exposure, then ubiquitous chemicals will be highly prioritized. However, if these chemicals are inert and do not produce adverse effects, then despite the high exposure, the potential risk may be low. For these reasons, risk-based chemical prioritization efforts consider both hazard (toxicity) and exposure, for instance through bioactivity-exposure ratios.</p>
</div>
</div>
<div id="filling-hazard-and-exposure-data-gaps-to-prioritize-more-chemicals" class="section level3 hasAnchor">
<h3>Filling Hazard and Exposure Data Gaps to Prioritize More Chemicals<a href="toxicokinetic-modeling.html#filling-hazard-and-exposure-data-gaps-to-prioritize-more-chemicals" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>To calculate a BER for a chemical, both bioactivity and exposure data are required, as well as sufficient TK data to perform reverse TK. In this training module, bioactivity data came from ToxCast AC50s; exposure data consisted of exposure inferences made from NHANES urinary biomonitoring data; and TK data consisted of parameter values measured <em>in vitro</em> and built into the <code>httk</code> R package. The intersections are illustrated in an Euler diagram below. BERs can only be calculated for chemicals in the triple intersection.</p>
<div class="sourceCode" id="cb891"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb891-1"><a href="toxicokinetic-modeling.html#cb891-1" tabindex="-1"></a>fit <span class="ot">&lt;-</span> eulerr<span class="sc">::</span><span class="fu">euler</span>(<span class="fu">list</span>(<span class="st">&#39;ToxCast AC50s&#39;</span> <span class="ot">=</span> <span class="fu">unique</span>(toxcast<span class="sc">$</span>DTXSID),</span>
<span id="cb891-2"><a href="toxicokinetic-modeling.html#cb891-2" tabindex="-1"></a>                   <span class="st">&#39;HTTK&#39;</span> <span class="ot">=</span> <span class="fu">unique</span>(chems_3compss<span class="sc">$</span>DTXSID),</span>
<span id="cb891-3"><a href="toxicokinetic-modeling.html#cb891-3" tabindex="-1"></a>                   <span class="st">&#39;NHANES inferred exposure&#39;</span> <span class="ot">=</span> <span class="fu">unique</span>(exposure<span class="sc">$</span>DTXSID)</span>
<span id="cb891-4"><a href="toxicokinetic-modeling.html#cb891-4" tabindex="-1"></a>                   ),</span>
<span id="cb891-5"><a href="toxicokinetic-modeling.html#cb891-5" tabindex="-1"></a>                   <span class="at">shape =</span> <span class="st">&quot;ellipse&quot;</span>)</span>
<span id="cb891-6"><a href="toxicokinetic-modeling.html#cb891-6" tabindex="-1"></a><span class="fu">plot</span>(fit,</span>
<span id="cb891-7"><a href="toxicokinetic-modeling.html#cb891-7" tabindex="-1"></a>     <span class="at">legend =</span> <span class="cn">TRUE</span>,</span>
<span id="cb891-8"><a href="toxicokinetic-modeling.html#cb891-8" tabindex="-1"></a>                   <span class="at">quantities =</span> <span class="cn">TRUE</span></span>
<span id="cb891-9"><a href="toxicokinetic-modeling.html#cb891-9" tabindex="-1"></a>              )</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-654-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Clearly, it would be useful to gather more data to allow calculation of BERs for more chemicals.</p>
</div>
<div id="answer-to-environmental-health-question-10" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 10<a href="toxicokinetic-modeling.html#answer-to-environmental-health-question-10" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can also answer <strong>Environmental Health Question #10</strong></em>: Of the three data sets used in this training module – bioactivity from ToxCast, TK data from <code>httk</code>, and exposure inferred from NHANES urinary biomonitoring – which one most limits the number of chemicals that can be prioritized using BERs?</p>
<div class="answer">
<p><strong>Answer</strong>: The exposure data set includes the fewest chemicals and is therefore the most limiting.</p>
</div>
<p>The exposure data set used in this training module is limited to chemicals for which NHANES did urinary biomonitoring for markers of exposure, which is a fairly small set of chemicals that were of interest to NHANES due to existing concerns about health effects of exposure, and/or other reasons. This data set was chosen because it is a convenient set of exposure estimates to use for demonstration purposes, but it could be expanded by including other sources of exposure data and exposure model predictions. Further discussion is beyond the scope of this training module, but as an example of this kind of high-throughput exposure modeling, see <a href="https://pubmed.ncbi.nlm.nih.gov/30516957/">Ring et al., 2019</a>.</p>
<p>It would additionally be useful to gather TK data for additional chemicals. <em>In vitro</em> measurement efforts are ongoing. Additonally, <em>in silico</em> modeling can produce useful predictions of TK properties to facilitate chemical prioritization. Efforts are ongoing to develop computational models to predict TK parameters from chemical structure and properties.</p>
<div id="concluding-remarks-27" class="section level2 hasAnchor">
<h2>Concluding Remarks<a href="toxicokinetic-modeling.html#concluding-remarks-27" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>This training module provides an overview of toxicokinetic modeling using the <code>httk</code> R package, and its application to <em>in vitro</em>-<em>in vivo</em> extrapolation in the form of placing <em>in vitro</em> data in the context of exposure by calculating equivalent doses for <em>in vitro</em> bioactive concentrations.</p>
<p>We would like to acknowledge the developers of the httk package, as detailed below via the CRAN website:</p>
<p><img src="Module6_6_Input/Module6_6_Image2.png" width="574" style="display: block; margin: auto;" /></p>
<p>This module also summarizes the use of the Bioactivity-Exposure Ratio (BER) for chemical prioritization, and provides examples of calculating the BER and ranking chemicals accordingly.</p>
<p>Together, these approaches can be used to more efficiently identify chemicals present in the environment that pose a potential risk to human health.</p>
<p>For additional case studies that leverage TK and/or httk modeling techniques, see the following publications that also address environmental health questions:</p>
<ul>
<li><p>Breen M, Ring CL, Kreutz A, Goldsmith MR, Wambaugh JF. High-throughput PBTK models for in vitro to in vivo extrapolation. Expert Opin Drug Metab Toxicol. 2021 Aug;17(8):903-921. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/34056988/">34056988</a>.</p></li>
<li><p>Klaren WD, Ring C, Harris MA, Thompson CM, Borghoff S, Sipes NS, Hsieh JH, Auerbach SS, Rager JE. Identifying Attributes That Influence In Vitro-to-In Vivo Concordance by Comparing In Vitro Tox21 Bioactivity Versus In Vivo DrugMatrix Transcriptomic Responses Across 130 Chemicals. Toxicol Sci. 2019 Jan 1;167(1):157-171. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/30202884/">30202884</a>.</p></li>
<li><p>Pearce RG, Setzer RW, Strope CL, Wambaugh JF, Sipes NS. httk: R Package for High-Throughput Toxicokinetics. J Stat Softw. 2017;79(4):1-26. PMID <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6134854/">30220889</a>.</p></li>
<li><p>Ring CL, Pearce RG, Setzer RW, Wetmore BA, Wambaugh JF. Identifying populations sensitive to environmental chemicals by simulating toxicokinetic variability. Environ Int. 2017 Sep;106:105-118. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/28628784/">28628784</a>.</p></li>
<li><p>Ring C, Sipes NS, Hsieh JH, Carberry C, Koval LE, Klaren WD, Harris MA, Auerbach SS, Rager JE. Predictive modeling of biological responses in the rat liver using in vitro Tox21 bioactivity: Benefits from high-throughput toxicokinetics. Comput Toxicol. 2021 May;18:100166. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/34013136/">34013136</a>.</p></li>
<li><p>Rotroff DM, Wetmore BA, Dix DJ, Ferguson SS, Clewell HJ, Houck KA, Lecluyse EL, Andersen ME, Judson RS, Smith CM, Sochaski MA, Kavlock RJ, Boellmann F, Martin MT, Reif DM, Wambaugh JF, Thomas RS. Incorporating human dosimetry and exposure into high-throughput in vitro toxicity screening. Toxicol Sci. 2010 Oct;117(2):348-58. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/20639261/">20639261</a>.</p></li>
<li><p>Wetmore BA, Wambaugh JF, Ferguson SS, Sochaski MA, Rotroff DM, Freeman K, Clewell HJ 3rd, Dix DJ, Andersen ME, Houck KA, Allen B, Judson RS, Singh R, Kavlock RJ, Richard AM, Thomas RS. Integration of dosimetry, exposure, and high-throughput screening data in chemical toxicity assessment. Toxicol Sci. 2012 Jan;125(1):157-74. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/21948869/">21948869</a>.</p></li>
<li><p>Wambaugh JF, Wetmore BA, Pearce R, Strope C, Goldsmith R, Sluka JP, Sedykh A, Tropsha A, Bosgra S, Shah I, Judson R, Thomas RS, Setzer RW. Toxicokinetic Triage for Environmental Chemicals. Toxicol Sci. 2015 Sep;147(1):55-67. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/26085347/">26085347</a>.</p></li>
<li><p>Wambaugh JF, Wetmore BA, Ring CL, Nicolas CI, Pearce RG, Honda GS, Dinallo R, Angus D, Gilbert J, Sierra T, Badrinarayanan A, Snodgrass B, Brockman A, Strock C, Setzer RW, Thomas RS. Assessing Toxicokinetic Uncertainty and Variability in Risk Prioritization. Toxicol Sci. 2019 Dec 1;172(2):235-251. doi: 10.1093/toxsci/kfz205. PMID: <a href="https://pubmed.ncbi.nlm.nih.gov/31532498/">31532498</a>.</p></li>
</ul>
<p><br></p>
<p><label class="tykfont">
Test Your Knowledge
</label></p>
<div class="tyk">
<ol style="list-style-type: decimal">
<li>After exposure to a single daily dose of 1 mg/kg/day methylparaben, what is the maximum concentration of methylparaben estimated to occur in human liver, estimated by the 3-comprtment model implemented in <code>httk</code>?</li>
<li>What is the predicted range of methylparaben concentrations in plasma that can occur in a human population, assuming a long-term exposure rate of 1 mg/kg/day and 3-compartment steady-state conditions? Provide estimates at the 5th, 50th, and 95th percentile.</li>
</ol>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="mixtures-analysis-methods-part-3-sufficient-similarity.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="chemical-read-across-for-toxicity-predictions.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
