<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5.5 Unsupervised Machine Learning Part 2: Additional Clustering Applications | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="5.5 Unsupervised Machine Learning Part 2: Additional Clustering Applications | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5.5 Unsupervised Machine Learning Part 2: Additional Clustering Applications | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  <meta name="twitter:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  

<meta name="author" content="Rager Lab" />


<meta name="date" content="2024-12-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="Icons_Used_Throughout/Favicon.png" type="image/x-icon" />
<link rel="prev" href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"/>
<link rel="next" href="descriptive-cohort-analyses.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>
<link href="libs/table1-1.0/table1_defaults.css" rel="stylesheet" />
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-FGFKVX2X3G"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-FGFKVX2X3G');
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="Icons_Used_Throughout/TAME 2.0 Logo.png"><br></a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Chapter 1 Introductory <br>Data Science</b></span></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html"><i class="fa fa-check"></i>1.1 FAIR Data Management Practices</a>
<ul>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-training-module"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-fair"><i class="fa fa-check"></i>Introduction to FAIR</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#breaking-down-fair-letter-by-letter"><i class="fa fa-check"></i>Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#what-does-this-mean-for-you"><i class="fa fa-check"></i>What Does This Mean for You?</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#data-repositories-for-sharing-of-data"><i class="fa fa-check"></i>Data Repositories for Sharing of Data</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#recent-shifts-in-regulatory-policies-for-data-sharing"><i class="fa fa-check"></i>Recent Shifts in Regulatory Policies for Data Sharing</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#additional-training-resources-on-fair"><i class="fa fa-check"></i>Additional Training Resources on FAIR</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html"><i class="fa fa-check"></i>1.2 Data Sharing through Online Repositories</a>
<ul>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#an-overview-and-example-with-the-dataverse-repository"><i class="fa fa-check"></i>An Overview and Example with the Dataverse Repository</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#introduction-to-training-module-1"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#data-repositories"><i class="fa fa-check"></i>Data Repositories</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#the-dataverse-project"><i class="fa fa-check"></i>The Dataverse Project</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#what-is-a-dataverse"><i class="fa fa-check"></i>What is a Dataverse?</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#metadata"><i class="fa fa-check"></i>Metadata</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataverse"><i class="fa fa-check"></i>Creating a Dataverse</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataset"><i class="fa fa-check"></i>Creating a Dataset</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#concluding-remarks-1"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html"><i class="fa fa-check"></i>1.3 File Management using Github</a>
<ul>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#introduction-to-training-module-2"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-an-account"><i class="fa fa-check"></i>Creating an Account</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-a-repository"><i class="fa fa-check"></i>Creating a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#uploading-code"><i class="fa fa-check"></i>Uploading Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#adding-subfolders-in-a-repository"><i class="fa fa-check"></i>Adding Subfolders in a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-code"><i class="fa fa-check"></i>Updating Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-repository-titles-and-structure-to-support-a-manuscript"><i class="fa fa-check"></i>Updating Repository Titles and Structure to Support a Manuscript</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#tracking-code-changes-using-github-branches"><i class="fa fa-check"></i>Tracking Code Changes using Github Branches</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#concluding-remarks-2"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html"><i class="fa fa-check"></i>1.4 Data Wrangling in Excel</a>
<ul>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#introduction-to-training-module-3"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#save-a-copy-of-the-soon-to-be-organized-and-cleaned-dataset-as-a-new-file"><i class="fa fa-check"></i>Save a Copy of the Soon-To-Be Organized and Cleaned Dataset as a New File</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-extraneous-white-space"><i class="fa fa-check"></i>Remove Extraneous White Space</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#replace-missing-data-with-na"><i class="fa fa-check"></i>Replace Missing Data with “NA”</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-metadata-tab"><i class="fa fa-check"></i>Create a Metadata Tab</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#abbreviate-and-capitalize-categorical-data"><i class="fa fa-check"></i>Abbreviate and Capitalize Categorical Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#alphabetize-sort-the-data-by-the-categorical-variable-of-interest"><i class="fa fa-check"></i>Alphabetize (Sort) the Data by the Categorical Variable of Interest</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-new-subject-number-column"><i class="fa fa-check"></i>Create a New Subject Number Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-special-symbols-and-dashes"><i class="fa fa-check"></i>Remove Special Symbols and Dashes</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#bold-all-column-names-and-center-all-data"><i class="fa fa-check"></i>Bold all Column Names and Center all Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-subject-identifier-column"><i class="fa fa-check"></i>Create a Subject Identifier Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#separate-subject-demographic-data-from-experimental-measurements"><i class="fa fa-check"></i>Separate Subject Demographic Data from Experimental Measurements</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#convert-data-from-wide-to-long-format"><i class="fa fa-check"></i>Convert Data from Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#pivoting-data-from-a-wide-to-long-format"><i class="fa fa-check"></i>Pivoting Data from a Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#generating-summary-level-statistics-with-pivot-tables"><i class="fa fa-check"></i>Generating Summary-Level Statistics with Pivot Tables</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#excel-vs.-r-which-should-you-use"><i class="fa fa-check"></i>Excel vs. R: Which Should You Use?</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#concluding-remarks-3"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 2 Coding in R</b></span></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html"><i class="fa fa-check"></i>2.1 Downloading and Programming in R</a>
<ul>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-training-module-4"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#general-introduction-and-installation-of-r-and-rstudio"><i class="fa fa-check"></i>General Introduction and Installation of R and RStudio</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-r-packages"><i class="fa fa-check"></i>Introduction to R Packages</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#scripting-basics"><i class="fa fa-check"></i>Scripting Basics</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#code-troubleshooting"><i class="fa fa-check"></i>Code Troubleshooting</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#concluding-remarks-4"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html"><i class="fa fa-check"></i>2.2 Coding “Best” Practices</a>
<ul>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#introduction-to-training-module-5"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#scripting-file-types"><i class="fa fa-check"></i>Scripting File Types</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-headers-and-annotation"><i class="fa fa-check"></i>Script Headers and Annotation</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#coding-style"><i class="fa fa-check"></i>Coding Style</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-organization"><i class="fa fa-check"></i>Script Organization</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#concluding-remarks-5"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html"><i class="fa fa-check"></i>2.3 Data Manipulation and Reshaping</a>
<ul>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-training-module-6"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#data-manipulation-using-base-r"><i class="fa fa-check"></i>Data Manipulation Using Base R</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-tidyverse"><i class="fa fa-check"></i>Introduction to Tidyverse</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#concluding-remarks-6"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html"><i class="fa fa-check"></i>2.4 Improving Coding Efficiencies</a>
<ul>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#introduction-to-training-module-7"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#loops"><i class="fa fa-check"></i>Loops</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#functions"><i class="fa fa-check"></i>Functions</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#list-operations"><i class="fa fa-check"></i>List operations</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#concluding-remarks-7"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#additional-resources-2"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="part"><span><b>Chapter 3 Basics of Data <br>Analysis and Visualizations</b></span></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html"><i class="fa fa-check"></i>3.1 Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-data-visualizations"><i class="fa fa-check"></i>Introduction to Data Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-training-module-8"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#density-plot-visualization"><i class="fa fa-check"></i>Density Plot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#boxplot-visualization"><i class="fa fa-check"></i>Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#correlation-visualizations"><i class="fa fa-check"></i>Correlation Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#heatmap-visualization"><i class="fa fa-check"></i>Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#concluding-remarks-8"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html"><i class="fa fa-check"></i>3.2 Improving Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-data-visulization-conventions"><i class="fa fa-check"></i>Introduction to Data Visulization Conventions</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-training-module-9"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-boxplot-visualization"><i class="fa fa-check"></i>Creating an Improved Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-heatmap-visualization"><i class="fa fa-check"></i>Creating an Improved Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-multi-plot-figures"><i class="fa fa-check"></i>Creating Multi-Plot Figures</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#concluding-remarks-9"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html"><i class="fa fa-check"></i>3.3 Normality Tests and Data Transformations</a>
<ul>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#introduction-to-training-module-10"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#what-is-a-normal-distribution"><i class="fa fa-check"></i>What is a Normal Distribution?</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#qualitative-assessment-of-normality"><i class="fa fa-check"></i>Qualitative Assessment of Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#quantitative-normality-assessment"><i class="fa fa-check"></i>Quantitative Normality Assessment</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#data-transformation"><i class="fa fa-check"></i>Data Transformation</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#additional-considerations-regarding-normality"><i class="fa fa-check"></i>Additional Considerations Regarding Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#concluding-remarks-10"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html"><i class="fa fa-check"></i>3.4 Intoduction to Statistical Tests</a>
<ul>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#introduction-to-training-module-11"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#assessing-normality-homogeneity-of-variance"><i class="fa fa-check"></i>Assessing Normality &amp; Homogeneity of Variance</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#two-group-visualizations-and-statistical-comparisons-using-the-t-test"><i class="fa fa-check"></i>Two-Group Visualizations and Statistical Comparisons using the T-Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#three-group-visualizations-and-statistical-comparisons-using-an-anova"><i class="fa fa-check"></i>Three-Group Visualizations and Statistical Comparisons using an ANOVA</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#regression-modeling-and-visualization-linear-and-logistic-regressions"><i class="fa fa-check"></i>Regression Modeling and Visualization: Linear and Logistic Regressions</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#statistical-evaluations-of-categorical-data-using-the-chi-squared-test-and-fishers-exact-test"><i class="fa fa-check"></i>Statistical Evaluations of Categorical Data using the Chi-Squared Test and Fisher’s Exact Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#concluding-remarks-11"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 4 Converting Wet <br>Lab Data into Dry Lab <br>Analyses</b></span></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html"><i class="fa fa-check"></i>4.1 Overview of Experimental Design and Example Data</a>
<ul>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#introduction-to-training-module-12"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#replicates"><i class="fa fa-check"></i>Replicates</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#orientation-to-example-data-for-chapter-4"><i class="fa fa-check"></i>Orientation to Example Data for Chapter 4</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#concluding-remarks-12"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html"><i class="fa fa-check"></i>4.2 Data Import, Processing, and Summary Statistics</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#introduction-to-training-module-13"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#data-import"><i class="fa fa-check"></i>Data Import</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#handling-missing-values"><i class="fa fa-check"></i>Handling Missing Values</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#averaging-replicates"><i class="fa fa-check"></i>Averaging Replicates</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#descriptive-statistics"><i class="fa fa-check"></i>Descriptive Statistics</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#normality-assessment-and-data-transformation"><i class="fa fa-check"></i>Normality Assessment and Data Transformation</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#concluding-remarks-13"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html"><i class="fa fa-check"></i>4.3 Data Import from PDF Sources</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#introduction-to-training-module-14"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-from-many-single-pdfs-with-the-same-formatting"><i class="fa fa-check"></i>Importing Data from Many Single PDFs with the Same Formatting</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-stored-in-pdf-tables"><i class="fa fa-check"></i>Importing Data Stored in PDF Tables</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#concluding-remarks-14"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html"><i class="fa fa-check"></i>4.4 Two Group Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#introduction-to-training-module-15"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#overview-of-two-group-statistical-tests"><i class="fa fa-check"></i>Overview of Two Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#statistical-vs.-biological-significance"><i class="fa fa-check"></i>Statistical vs. Biological Significance</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#unpaired-test-example"><i class="fa fa-check"></i>Unpaired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#paired-test-example"><i class="fa fa-check"></i>Paired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#visualizing-results"><i class="fa fa-check"></i>Visualizing Results</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#concluding-remarks-15"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html"><i class="fa fa-check"></i>4.5 Multi-Group and Multi-Variable Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#introduction-to-training-module-16"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#overview-of-multi-group-statistical-tests"><i class="fa fa-check"></i>Overview of Multi-Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#multi-group-analysis-example"><i class="fa fa-check"></i>Multi-Group Analysis Example</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#visualization-of-multi-group-statistical-results"><i class="fa fa-check"></i>Visualization of Multi-Group Statistical Results</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#concluding-remarks-16"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html"><i class="fa fa-check"></i>4.6 Advanced Multi-Group Comparisons</a>
<ul>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#introduction-to-training-module-17"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#advanced-multi-group-comparisons-1"><i class="fa fa-check"></i>Advanced Multi-Group Comparisons</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#two-way-anova-example"><i class="fa fa-check"></i>Two-way ANOVA Example</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#ancova-example"><i class="fa fa-check"></i>ANCOVA Example</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#concluding-remarks-17"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 5 Machine Learning <br>&amp; Artificial Intelligence</b></span></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><i class="fa fa-check"></i>5.1 Introduction to Artificial Intelligence, Machine Learning, and Predictive Modeling for Environmental Health</a>
<ul>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#introduction-to-training-module-18"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#general-historical-context-and-taxonomy-of-modern-aiml"><i class="fa fa-check"></i>General Historical Context and Taxonomy of Modern AI/ML</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#application-of-machine-learning-in-environmental-health-science"><i class="fa fa-check"></i>Application of Machine Learning in Environmental Health Science</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#concluding-remarks-18"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html"><i class="fa fa-check"></i>5.2 Supervised Machine Learning</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-training-module-19"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-machine-learning"><i class="fa fa-check"></i>Types of Machine Learning</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-supervised-machine-learning-algorithms"><i class="fa fa-check"></i>Types of Supervised Machine Learning Algorithms</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#training-supervised-machine-learning-models"><i class="fa fa-check"></i>Training Supervised Machine Learning Models</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#assessing-classification-based-model-performance"><i class="fa fa-check"></i>Assessing Classification-Based Model Performance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-activity-and-example-dataset"><i class="fa fa-check"></i>Introduction to Activity and Example Dataset</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#testing-for-differences-in-predictor-variables-across-the-outcome-classes"><i class="fa fa-check"></i>Testing for Differences in Predictor Variables across the Outcome Classes</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#predicting-ias-detection-with-a-random-forest-rf-model"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#class-imbalance"><i class="fa fa-check"></i>Class Imbalance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#concluding-remarks-19"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html"><i class="fa fa-check"></i>5.3 Supervised Machine Learning Model Interpretation</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-training-module-20"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance"><i class="fa fa-check"></i>Variable Importance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary"><i class="fa fa-check"></i>Decision Boundary</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-example-dataset-and-activity"><i class="fa fa-check"></i>Introduction to Example Dataset and Activity</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#predicting-ias-detection-with-a-random-forest-rf-model-1"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance-plot"><i class="fa fa-check"></i>Variable Importance Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot"><i class="fa fa-check"></i>Decision Boundary Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot-incorporating-smote"><i class="fa fa-check"></i>Decision Boundary Plot Incorporating SMOTE</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#concluding-remarks-20"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><i class="fa fa-check"></i>5.4 Unsupervised Machine Learning Part 1: K-Means Clustering &amp; PCA</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-training-module-21"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-unsupervised-machine-learning"><i class="fa fa-check"></i>Introduction to Unsupervised Machine Learning</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-example-data"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#identifying-clusters-of-chemicals-through-k-means"><i class="fa fa-check"></i>Identifying Clusters of Chemicals through <em>K</em>-Means</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#principal-component-analysis-pca-1"><i class="fa fa-check"></i>Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#incorporating-k-means-into-pca-for-predictive-modeling"><i class="fa fa-check"></i>Incorporating <em>K</em>-Means into PCA for Predictive Modeling</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#concluding-remarks-21"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><i class="fa fa-check"></i>5.5 Unsupervised Machine Learning Part 2: Additional Clustering Applications</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-training-module-22"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#k-means-clustering-1"><i class="fa fa-check"></i><em>K</em>-Means Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-example-data-1"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-1"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#clustering-plot"><i class="fa fa-check"></i>Clustering Plot</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-visualization"><i class="fa fa-check"></i>Hierarchical Clustering Visualization</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#variable-contributions"><i class="fa fa-check"></i>Variable Contributions</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#concluding-remarks-22"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 6 Applications in <br>Toxicology &amp; Exposure <br>Science</b></span></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html"><i class="fa fa-check"></i>6.1 Descriptive Cohort Analyses</a>
<ul>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#introduction-to-training-module-23"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#participant-exploration"><i class="fa fa-check"></i>Participant Exploration</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#chemical-detection-filtering"><i class="fa fa-check"></i>Chemical Detection Filtering</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#outlier-identification"><i class="fa fa-check"></i>Outlier Identification</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#summary-statistics-tables"><i class="fa fa-check"></i>Summary Statistics Tables</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#demographics-table"><i class="fa fa-check"></i>Demographics Table</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#concluding-remarks-23"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html"><i class="fa fa-check"></i>6.2 -Omics and System Biology: Transcriptomic Applications</a>
<ul>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-training-module-24"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-the-field-of--omics"><i class="fa fa-check"></i>Introduction to the Field of “-Omics”</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-transcriptomics"><i class="fa fa-check"></i>Introduction to Transcriptomics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#formatting-data-for-downstream-statistics"><i class="fa fa-check"></i>Formatting Data for Downstream Statistics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#transcriptomics-data-qaqc"><i class="fa fa-check"></i>Transcriptomics Data QA/QC</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#controlling-for-sources-of-sample-heterogeneity"><i class="fa fa-check"></i>Controlling for Sources of Sample Heterogeneity</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e.g.-biomass-smoke-exposure"><i class="fa fa-check"></i>Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure)</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-ma-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using MA Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-volcano-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using Volcano Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis"><i class="fa fa-check"></i>Interpretting Findings at the Systems Level through Pathway Enrichment Analysis</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#concluding-remarks-24"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><i class="fa fa-check"></i>6.3 Mixtures Analysis Methods Part 1: Overview and Example with Quantile G-Computation</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-training-module-25"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#overview-of-mixtures-analysis"><i class="fa fa-check"></i>Overview of Mixtures Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-example-data-2"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#mixtures-model-with-standard-qqcomp"><i class="fa fa-check"></i>Mixtures Model with Standard qqcomp</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#concluding-remarks-25"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#additional-resources-11"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><i class="fa fa-check"></i>6.4 Mixtures Analysis Methods Part 2: Bayesian Kernel Machine Regression</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-training-module-26"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-example-data-3"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-bkmr"><i class="fa fa-check"></i>Introduction to BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#run-bkmr"><i class="fa fa-check"></i>Run BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#concluding-remarks-26"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#additional-resources-12"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html"><i class="fa fa-check"></i>6.5 Mixtures Analysis Methods Part 3: Sufficient Similarity</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-training-module-27"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-toxicant-and-dataset"><i class="fa fa-check"></i>Introduction to Toxicant and Dataset</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#workspace-preparation-and-data-import-9"><i class="fa fa-check"></i>Workspace Preparation and Data Import</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#chemistry-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Chemistry-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#toxicity-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Toxicity-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#comparing-chemistry-vs.-toxicity-sufficient-similarity-analyses"><i class="fa fa-check"></i>Comparing Chemistry vs. Toxicity Sufficient Similarity Analyses</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#concluding-remarks-27"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i>6.6 Toxicokinetic Modeling</a>
<ul>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-training-module-28"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-toxicokinetic-modeling"><i class="fa fa-check"></i>Introduction to Toxicokinetic Modeling</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#data-and-models-used-in-toxicokinetic-modeling-tk"><i class="fa fa-check"></i>Data and Models used in Toxicokinetic Modeling (TK)</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-toxicokinetics"><i class="fa fa-check"></i>Reverse Toxicokinetics</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#capturing-population-variability-in-toxicokinetics-and-uncertainty-in-chemical-specific-parameters"><i class="fa fa-check"></i>Capturing Population Variability in Toxicokinetics, and Uncertainty in Chemical-Specific Parameters</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-tk-calculating-administered-equivalent-doses-for-toxcast-bioactive-concentrations"><i class="fa fa-check"></i>Reverse TK: Calculating Administered Equivalent Doses for ToxCast Bioactive Concentrations</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#comparing-equivalent-doses-estimated-to-elicit-toxicity-hazard-to-external-exposure-estimates-exposure-for-chemical-prioritization-by-bioactivity-exposure-ratios-bers"><i class="fa fa-check"></i>Comparing Equivalent Doses Estimated to Elicit Toxicity (Hazard) to External Exposure Estimates (Exposure), for Chemical Prioritization by Bioactivity-Exposure Ratios (BERs)</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#concluding-remarks-28"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html"><i class="fa fa-check"></i>Chemical Read-Across for Toxicity Predictions</a>
<ul>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#introduction-to-training-module-29"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#introduction-to-activity"><i class="fa fa-check"></i>Introduction to Activity</a></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#read-across-example-analysis"><i class="fa fa-check"></i>Read-Across Example Analysis</a></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#calculating-chemical-similarities"><i class="fa fa-check"></i>Calculating Chemical Similarities</a></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#chemical-read-across-to-predict-acute-toxicity"><i class="fa fa-check"></i>Chemical Read-Across to Predict Acute Toxicity</a></li>
<li class="chapter" data-level="" data-path="chemical-read-across-for-toxicity-predictions.html"><a href="chemical-read-across-for-toxicity-predictions.html#concluding-remarks-29"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 7 Environmental<br> Health Database Mining</b></span></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html"><i class="fa fa-check"></i>7.1 Comparative Toxicogenomics Database</a>
<ul>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#introduction-to-training-module-30"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#ctd-data-in-r"><i class="fa fa-check"></i>CTD Data in R</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#identifying-genes-under-epigenetic-control"><i class="fa fa-check"></i>Identifying Genes Under Epigenetic Control</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#concluding-remarks-30"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html"><i class="fa fa-check"></i>7.2 Gene Expression Omnibus</a>
<ul>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-training-module-31"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-geo"><i class="fa fa-check"></i>Introduction to GEO</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-example-data-4"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#geo-data-in-r"><i class="fa fa-check"></i>GEO Data in R</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#visualizing-data"><i class="fa fa-check"></i>Visualizing Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#statistical-analyses"><i class="fa fa-check"></i>Statistical Analyses</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#concluding-remarks-31"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html"><i class="fa fa-check"></i>7.3 CompTox Dashboard Data through APIs</a>
<ul>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#introduction-to-training-module-32"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#training-modules-environmental-health-questions-24"><i class="fa fa-check"></i>Training Module’s Environmental Health Questions</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#script-preparations-12"><i class="fa fa-check"></i>Script Preparations</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#introduction-to-comptox-chemicals-dashboard"><i class="fa fa-check"></i>Introduction to CompTox Chemicals Dashboard</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#cctes-ctx-application-programming-interfaces-apis-for-automated-batch-search-of-the-ccd"><i class="fa fa-check"></i>CCTE’s CTX Application Programming Interfaces (APIs) for Automated Batch Search of the CCD</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#comparing-physico-chemical-properties-between-two-important-environmental-contaminant-lists"><i class="fa fa-check"></i>Comparing Physico-chemical Properties between Two Important Environmental Contaminant Lists</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#hazard-data-genotoxicity"><i class="fa fa-check"></i>Hazard Data: Genotoxicity</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#hazard-resource"><i class="fa fa-check"></i>Hazard Resource</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#concluding-remarks-32"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html"><i class="fa fa-check"></i>7.4 Database Integration: Air Quality, Mortality, and Environmental Justice Data</a>
<ul>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#introduction-to-training-module-33"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#introduction-to-exposure-and-health-databases"><i class="fa fa-check"></i>Introduction to Exposure and Health Databases</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#analyzing-relationships-between-pm2.5-and-mortality"><i class="fa fa-check"></i>Analyzing Relationships between PM<sub>2.5</sub> and Mortality</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#environmental-justice-considerations"><i class="fa fa-check"></i>Environmental Justice Considerations</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#concluding-remarks-33"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="unsupervised-machine-learning-part-2-additional-clustering-applications" class="section level1 hasAnchor">
<h1>5.5 Unsupervised Machine Learning Part 2: Additional Clustering Applications<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#unsupervised-machine-learning-part-2-additional-clustering-applications" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This training module was developed by Alexis Payton, Lauren E. Koval, David M. Reif, and Julia E. Rager.</p>
<p>All input files (script, data, and figures) can be downloaded from the <a href="https://github.com/UNCSRP/TAME2">UNC-SRP TAME2 GitHub website</a>.</p>
<div id="introduction-to-training-module-22" class="section level2 hasAnchor">
<h2>Introduction to Training Module<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-training-module-22" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The previous module <strong>TAME 2.0 Module 5.4 Unsupervised Machine Learning Part 1: K-Means Clustering &amp; PCA</strong>, served as introduction to unsupervised machine learning (ML). <strong>Unsupervised ML</strong> involves training a model on a dataset lacking ground truths or response variables. However, in the previous module, the number of clusters was selected based on prior information (i.e., chemical class), but what if you’re in a situation where you don’t know how many clusters to investigate a priori? This commonly occurs, particularly in the field of environmental health research in instances when investigators want to take a more unbiased view of their data and/or do not have information that can be used to inform the optimal number of clusters to select. In these instances,unsupervised ML techniques can be very helpful, and in this module, we’ll explore the following concepts to further understand unsupervised ML:</p>
<ul>
<li><em>K</em>-Means and hierarchical clustering</li>
<li>Deriving the optimal number of clusters</li>
<li>Visualizing clusters through a PCA-based plot, dendrograms, and heatmaps</li>
<li>Determining each variable’s contribution to the clusters</li>
</ul>
<p><br></p>
</div>
<div id="k-means-clustering-1" class="section level2 hasAnchor">
<h2><em>K</em>-Means Clustering<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#k-means-clustering-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>As mentioned in the previous module, <em>K</em>-means is a common clustering algorithm used to partition quantitative data. This algorithm works by first, randomly selecting a pre-specified number of clusters, <em>k</em>, across the data space, with each cluster having a data centroid. When using a standard Euclidean distance metric, the distance is calculated from an observation to each centroid, then the observation is assigned to the cluster of the closest centroid. After all observations have been assigned to one of the <em>k</em> clusters, the average of all observations in a cluster is calculated, and the centroid for the cluster is moved to the location of the mean. The process then repeats, with the distance computed between the observations and the updated centroids. Observations may be reassigned to the same cluster or moved to a different cluster if it is closer to another centroid. These iterations continue until there are no longer changes between cluster assignments for observations, resulting in the final cluster assignments that are then carried forward for analysis/interpretation.</p>
<p>Helpful resources on <em>k</em>-means clustering include the following: <a href="https://web.stanford.edu/~hastie/ElemStatLearn/printings/ESLII_print12_toc.pdf">The Elements of Statistical Learning</a> &amp;
<a href="https://towardsdatascience.com/k-means-clustering-algorithm-applications-evaluation-methods-and-drawbacks-aa03e644b48a">Towards Data Science</a>.</p>
<p><br></p>
</div>
<div id="hierarchical-clustering" class="section level2 hasAnchor">
<h2>Hierarchical Clustering<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Hierarchical clustering</strong> groups objects into clusters by repetitively joining similar observations until there is one large cluster (aka agglomerative or bottom-up) or repetitively splitting one large cluster until each observation stands alone (aka divisive or top-down). Regardless of whether agglomerative or divisive hierarchical clustering is used, the results can be visually represented in a tree-like figure called a dendrogram. The dendrogram below is based on the <code>USArrests</code> dataset available in R. The datset contains statistics on violent crimes rates (murder, assault, and rape) per capita (per 100,000 residents) for each state in the United States in 1973. For more information on the <code>USArrests</code> dataset, check out its associated <a href="https://www.rdocumentation.org/packages/datasets/versions/3.6.2/topics/USArrests">RDocumentation</a>.</p>
<p><img src="Module5_5_Input/Module5_5_Image1.png" width="55%" style="display: block; margin: auto;" /></p>
<p>An appropriate title for this figure could be:</p>
<p>“<strong>Figure X. Hierarchical clustering of states based on violent crime.</strong> The dendogram uses violent crime data including murder, assault, and rape rates per 100,000 residents for each state in 1973.”</p>
<p>Takeaways from this dendogram:</p>
<ul>
<li>The 50 states can be grouped into 4 clusters based on violent crime statistics from 1973</li>
<li>The dendogram can only show us clusters of states but not the data trends that led to the clustering patterns that we see. Yes, it is useful to know what states have similar violent crime patterns overall, but it is important to pinpoint the variables (ie. murder, assault, and rape) that are responsible for the clustering patterns we’re seeing. This idea will be explored later in the module with an environmentally-relevant dataset.</li>
</ul>
<p>Going back to hiearchical clustering, during the repetitive splitting or joining of observations, the similarity between existing clusters is calculated after each iteration. This value informs the formation of subsequent clusters. Different methods, or linkage functions, can be considered when calculating this similarity, particularly for agglomerative clustering which is often the preferred approach. Some example methods include:</p>
<ul>
<li><strong>Complete Linkage</strong>: the maximum distance between two data points located in separate clusters.</li>
<li><strong>Single Linkage</strong>: the minimum distance between two data points located in separate clusters.</li>
<li><strong>Average Linkage</strong>: the average pairwise distance between all pairs of data points in separate clusters.</li>
<li><strong>Centroid Linkage</strong>: the distance between the centroids or centers of each cluster.</li>
<li><strong>Ward Linkage</strong>: seeks to minimize the variance between clusters.</li>
</ul>
<p>Each method has its advantages and disadvantages and more information on all distance calculations between clusters can be found at the following resource: <a href="https://www.learndatasci.com/glossary/hierarchical-clustering/#Hierarchicalclusteringtypes">Hierarchical Clustering</a>.</p>
<div id="deriving-the-optimal-number-of-clusters" class="section level3 hasAnchor">
<h3>Deriving the Optimal Number of Clusters<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#deriving-the-optimal-number-of-clusters" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Before clustering can be performed, the function needs to be informed of the number of clusters to group the objects into. In the previous module, an example was explored to see if <em>k</em>-means clustering would group the chemicals similarly to their chemical class (either a PFAS or statin). Therefore, we told the <em>k</em>-means function to cluster into 2 groups. In situations where there is little to no prior knowledge regarding the “correct” number of clusters to specify, methods exist for deriving the optimal number of clusters. Three common methods to find the optimal <em>k</em>, or number of clusters, for both <em>k</em>-means and hierarchical clustering include: the <strong>elbow method</strong>, <strong>silhouette method</strong>, and the <strong>gap statistic method</strong>. These techniques help us in determining the optimal <em>k</em> using visual inspection.</p>
<ul>
<li><strong>Elbow Method</strong>: uses a plot of the within cluster sum of squares (WCSS) on the y axis and different values of <em>k</em> on the x axis. The location where we no longer observe a significant reduction in WCSS, or where an “elbow” can be seen, is the optimal <em>k</em> value. As we can see, after a certain point, having more clusters does not lead to a significant reduction in WCSS.
<img src="Module5_5_Input/Module5_5_Image2.png" width="75%" style="display: block; margin: auto;" /></li>
</ul>
<p>Looking at the figures above, the elbow point is much clearer in the first plot versus the second, however, elbow curves from real-world datasets typically resemble the second figure. This is why it’s recommended to consider more than one method to determine the optimal number of clusters.</p>
<ul>
<li><strong>Silhouette Method</strong>: uses a plot of the average silhouette width (score) on the y axis and different values of <em>k</em> on the x axis. The silhouette score is measure of each object’s similarity to its own cluster and how dissimilar it is to other clusters. The location where the average silhouette width is <em>maximized</em> is the optimal <em>k</em> value.
<img src="Module5_5_Input/Module5_5_Image3.png" width="65%" style="display: block; margin: auto;" /></li>
</ul>
<p>Based on the figure above, the optimal number of clusters is 2 using the silhouette method.</p>
<ul>
<li><strong>Gap Statistic Method</strong>: uses a plot of the gap statistic on the y axis and different values of <em>k</em> on the x axis. The gap statistic evaluates the intracluster variation in comparison to expected values derived from a Monte Carlo generated, null reference data distribution for varying values of <em>k</em>. The optimal number of clusters is the smallest value where the gap statistic of <em>k</em> is greater than or equal to the gap statistic of <em>k</em>+1 minus the standard deviation of <em>k</em>+1. More details can be found <a href="https://uc-r.github.io/kmeans_clustering#:~:text=The%20gap%20statistic%20compares%20the,simulations%20of%20the%20sampling%20process.">here</a>.
<img src="Module5_5_Input/Module5_5_Image4.png" width="65%" style="display: block; margin: auto;" /></li>
</ul>
<p>Based on the figure above, the optimal number of clusters is 2 using the gap statistic method.</p>
<p>For additional information and code on all three methods, check out <a href="https://www.datanovia.com/en/lessons/determining-the-optimal-number-of-clusters-3-must-know-methods/">Determining the Optimal Number of Clusters: 3 Must Know Methods</a>. It is also worth mentioning that while these methods are useful, further interpreting the result in the context of your problem can be beneficial, for example, checking whether clusters make biological sense when working with a genomic dataset.</p>
<p><br></p>
</div>
</div>
<div id="introduction-to-example-data-1" class="section level2 hasAnchor">
<h2>Introduction to Example Data<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-example-data-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We will apply these techniques using an example dataset from a previously published study where 22 cytokine concentrations were derived from 44 subjects with varying smoking statuses (14 non-smokers, 17 e-cigarette users, and 13 cigarette smokers) from 4 different sampling regions in the body. These samples were derived from nasal lavage fluid (NLF), nasal epithelieum fluid (NELF), sputum, and serum as pictured below. Samples were taken from different regions in the body to compare cytokine expression in the upper respiratory tract, lower respiratory tract, and systemic circulation.
<img src="Module5_5_Input/Module5_5_Image5.png" width="75%" style="display: block; margin: auto;" /></p>
<p>A research question that we had was “Does cytokine expression change based on a subject’s smoking habits? If so, does cigarette smoke or e-cigarette vapor induce cytokine suppression or proliferation?” Traditionally these questions would have been answered by analyzing each biomarker individually using a two-group comparison test like a t test (which we completed in this study). However, biomarkers do not work in isolation in the body, suggesting that individual biomarker statistical approaches may not capture the full biological responses occurring. Therefore we used a clustering approach to group cytokines as an attempt to more closely simulate interactions that occur <em>in vivo</em>. From there, statistical tests were run to assess the effects of smoking status on each cluster.</p>
<p>For the purposes of this training exercise, we will focus solely on the nasal epithelieum lining fluid, or NELF, samples. In addition, we’ll use <em>k</em>-means and hierarchical clustering to compare how cytokines cluster at baseline. Full methods are further described in the publication below:</p>
<ul>
<li>Payton AD, Perryman AN, Hoffman JR, Avula V, Wells H, Robinette C, Alexis NE, Jaspers I, Rager JE, Rebuli ME. Cytokine signature clusters as a tool to compare changes associated with tobacco product use in upper and lower airway samples. American Journal of Physiology-Lung Cellular and Molecular Physiology 2022 322:5, L722-L736. PMID: <a href="https://journals.physiology.org/doi/abs/10.1152/ajplung.00299.2021">35318855</a></li>
</ul>
<p>Let’s read in and view the dataset we’ll be working with.</p>
<div id="script-preparations-7" class="section level3 hasAnchor">
<h3>Script Preparations<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#script-preparations-7" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="cleaning-the-global-environment-9" class="section level4 hasAnchor">
<h4>Cleaning the global environment<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cleaning-the-global-environment-9" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb581"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb581-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb581-1" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list=</span><span class="fu">ls</span>())</span></code></pre></div>
</div>
<div id="installing-required-r-packages-8" class="section level4 hasAnchor">
<h4>Installing required R packages<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#installing-required-r-packages-8" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>If you already have these packages installed, you can skip this step, or you can run the below code which checks installation status for you</p>
<div class="sourceCode" id="cb582"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb582-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb582-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;vegan&quot;</span>))</span>
<span id="cb582-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb582-2" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;vegan&quot;</span>);</span>
<span id="cb582-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb582-3" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;ggrepel&quot;</span>))</span>
<span id="cb582-4"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb582-4" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;ggrepel&quot;</span>);</span>
<span id="cb582-5"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb582-5" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;dendextend&quot;</span>))</span>
<span id="cb582-6"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb582-6" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;dendextend&quot;</span>);</span>
<span id="cb582-7"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb582-7" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;ggsci&quot;</span>))</span>
<span id="cb582-8"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb582-8" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;ggsci&quot;</span>);</span>
<span id="cb582-9"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb582-9" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;FactoMineR&quot;</span>))</span>
<span id="cb582-10"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb582-10" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;FactoMineR&quot;</span>);</span></code></pre></div>
</div>
<div id="loading-required-r-packages-2" class="section level4 hasAnchor">
<h4>Loading required R packages<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#loading-required-r-packages-2" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb583"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb583-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb583-1" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb583-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb583-2" tabindex="-1"></a><span class="fu">library</span>(factoextra)</span>
<span id="cb583-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb583-3" tabindex="-1"></a><span class="fu">library</span>(FactoMineR)</span>
<span id="cb583-4"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb583-4" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb583-5"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb583-5" tabindex="-1"></a><span class="fu">library</span>(vegan)</span>
<span id="cb583-6"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb583-6" tabindex="-1"></a><span class="fu">library</span>(ggrepel)</span>
<span id="cb583-7"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb583-7" tabindex="-1"></a><span class="fu">library</span>(reshape2)</span>
<span id="cb583-8"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb583-8" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb583-9"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb583-9" tabindex="-1"></a><span class="fu">library</span>(ggsci)</span>
<span id="cb583-10"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb583-10" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(dendextend))</span></code></pre></div>
</div>
<div id="set-your-working-directory-10" class="section level4 hasAnchor">
<h4>Set your working directory<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#set-your-working-directory-10" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb584"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb584-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb584-1" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/filepath to where your input files are&quot;</span>)</span></code></pre></div>
</div>
<div id="importing-example-dataset-8" class="section level4 hasAnchor">
<h4>Importing example dataset<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#importing-example-dataset-8" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Then let’s read in our example dataset. As mentioned in the introduction, this example dataset contains cytokine concentrations derived from 44 subjects. Let’s import and view these data:</p>
<div class="sourceCode" id="cb585"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb585-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb585-1" tabindex="-1"></a><span class="co"># Reading in file</span></span>
<span id="cb585-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb585-2" tabindex="-1"></a>cytokines_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">read_excel</span>(<span class="st">&quot;Module5_5_Input/Module5_5_InputData.xlsx&quot;</span>, <span class="at">sheet =</span> <span class="dv">2</span>))</span>
<span id="cb585-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb585-3" tabindex="-1"></a></span>
<span id="cb585-4"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb585-4" tabindex="-1"></a><span class="co"># Viewing data </span></span>
<span id="cb585-5"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb585-5" tabindex="-1"></a><span class="fu">head</span>(cytokines_df)</span></code></pre></div>
<pre><code>##   Original_Identifier Group SubjectNo SubjectID Compartment Protein       Conc
## 1           E_C_F_002    NS         1      NS_1        NELF    IFNg  17.642316
## 2           E_C_F_002    NS         1      NS_1        NELF    IL10   2.873724
## 3           E_C_F_002    NS         1      NS_1        NELF IL12p70   1.625272
## 4           E_C_F_002    NS         1      NS_1        NELF    IL13  36.117692
## 5           E_C_F_002    NS         1      NS_1        NELF    IL1b 104.409217
## 6           E_C_F_002    NS         1      NS_1        NELF     IL6  21.159536
##   Conc_pslog2
## 1    4.220509
## 2    1.953721
## 3    1.392467
## 4    5.214035
## 5    6.719857
## 6    4.469856</code></pre>
<p>These data contain the following information:</p>
<ul>
<li><code>Original_Identifier</code>: initial identifier given to each subject by our wet bench colleagues</li>
<li><code>Group</code>: denotes the smoking status of the subject (“NS” = “non-smoker”, “Ecig” = “E-cigarette user”, “CS” = “cigarette smoker”)</li>
<li><code>SubjectNo</code>: ordinal subject number assigned to each subject after the dataset was wrangled (1-44)</li>
<li><code>SubjectID</code>: unique subject identifier that combines the group and subject number</li>
<li><code>Compartment</code>: region of the body from which the sample was taken (“NLF” = “nasal lavage fluid sample”, “NELF” = “nasal epithelieum lining fluid sample”, “Sputum” = “induced sputum sample”, “Serum” = “blood serum sample”)</li>
<li><code>Protein</code>: cytokine name</li>
<li><code>Conc</code>: concentration (pg/mL)</li>
<li><code>Conc_pslog2</code>: psuedo-log<sub>2</sub> concentration</li>
</ul>
<p>Now that the data has been read in, we can start by asking some initial questions about the data.</p>
</div>
</div>
<div id="training-modules-environmental-health-questions-14" class="section level3 hasAnchor">
<h3>Training Module’s Environmental Health Questions<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#training-modules-environmental-health-questions-14" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This training module was specifically developed to answer the following environmental health questions:</p>
<ol style="list-style-type: decimal">
<li>What is the optimal number of clusters the cytokines can be grouped into that were derived from nasal epithelium fluid (NELF) in non-smokers using <em>k</em>-means clustering?</li>
<li>After selecting a cluster number, which cytokines were assigned to each <em>k</em>-means cluster?</li>
<li>What is the optimal number of clusters the cytokines can be grouped into that were derived from nasal epithelium fluid (NELF) in non-smokers using hierarchical clustering?</li>
<li>How do the hierarchical cluster assignments compare to the <em>k</em>-means cluster assignments?</li>
<li>Which cytokines have the greatest contributions to the first two eigenvectors?</li>
</ol>
<p>To answer the first environmental health question, let’s start by filtering to include only NELF derived samples and non-smokers.</p>
<div class="sourceCode" id="cb587"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb587-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb587-1" tabindex="-1"></a>baseline_df <span class="ot">&lt;-</span> cytokines_df <span class="sc">%&gt;%</span></span>
<span id="cb587-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb587-2" tabindex="-1"></a>    <span class="fu">filter</span>(Group <span class="sc">==</span> <span class="st">&quot;NS&quot;</span>, Compartment <span class="sc">==</span> <span class="st">&quot;NELF&quot;</span>) </span>
<span id="cb587-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb587-3" tabindex="-1"></a></span>
<span id="cb587-4"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb587-4" tabindex="-1"></a><span class="fu">head</span>(baseline_df)</span></code></pre></div>
<pre><code>##   Original_Identifier Group SubjectNo SubjectID Compartment Protein       Conc
## 1           E_C_F_002    NS         1      NS_1        NELF    IFNg  17.642316
## 2           E_C_F_002    NS         1      NS_1        NELF    IL10   2.873724
## 3           E_C_F_002    NS         1      NS_1        NELF IL12p70   1.625272
## 4           E_C_F_002    NS         1      NS_1        NELF    IL13  36.117692
## 5           E_C_F_002    NS         1      NS_1        NELF    IL1b 104.409217
## 6           E_C_F_002    NS         1      NS_1        NELF     IL6  21.159536
##   Conc_pslog2
## 1    4.220509
## 2    1.953721
## 3    1.392467
## 4    5.214035
## 5    6.719857
## 6    4.469856</code></pre>
<p>The functions we use will require us to cast the data wider. We will accomplish this using the <code>dcast()</code> function from the <em>reshape2</em> package.</p>
<div class="sourceCode" id="cb589"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb589-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb589-1" tabindex="-1"></a>wider_baseline_df <span class="ot">&lt;-</span> reshape2<span class="sc">::</span><span class="fu">dcast</span>(baseline_df, Protein <span class="sc">~</span> SubjectID, <span class="at">value.var =</span> <span class="st">&quot;Conc_pslog2&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb589-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb589-2" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&quot;Protein&quot;</span>)</span>
<span id="cb589-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb589-3" tabindex="-1"></a></span>
<span id="cb589-4"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb589-4" tabindex="-1"></a><span class="fu">head</span>(wider_baseline_df)</span></code></pre></div>
<pre><code>##                  NS_1     NS_10     NS_11     NS_12     NS_13     NS_14
## Eotaxin      7.215729  8.016193  6.264252  6.938344  5.152683  7.937892
## Eotaxin3     4.396783  4.006488  5.289509  3.597945  3.051951  3.063025
## Fractalkine 13.584554 13.679882 12.952220 13.464652 12.110742 12.785656
## I309         0.000000  0.000000  0.000000  0.000000  0.000000  1.617372
## IFNg         4.220509  1.990018  2.656691  3.439682  2.318340  3.439850
## IL10         1.953721  1.145162  1.348623  1.801954  1.176498  1.800412
##                  NS_2        NS_3      NS_4        NS_5      NS_6      NS_7
## Eotaxin      9.773407  6.12290913  5.758602  4.19868540  6.832578  7.998467
## Eotaxin3     9.022390  2.50161364  2.721195  2.21084320  3.820379  5.549164
## Fractalkine 14.961872 12.71538464 12.057105 10.38711624 12.917180 14.636190
## I309         1.103140  0.00000000  0.000000  0.00000000  0.000000  0.000000
## IFNg         4.394705  0.00000000  2.638213  0.04830196  1.914092  3.251501
## IL10         2.652120  0.08945805  2.020155  0.84649523  1.119272  2.007941
##                  NS_8      NS_9
## Eotaxin      9.242464  7.127746
## Eotaxin3    13.724108  3.553097
## Fractalkine 14.047382 13.146599
## I309         0.000000  0.000000
## IFNg         4.104820  2.940248
## IL10         2.576814  1.970627</code></pre>
<p>Now we can derive clusters using the <code>fviz_nbclust()</code> function to determine the optimal <em>k</em> based on suggestions from the elbow, silhouette, and gap statistic methods. We can use this code for both <em>k</em>-means and hierarchical clustering by changing the <code>FUNcluster</code> parameter. Lets start with <em>k</em>-means:.</p>
<div class="sourceCode" id="cb591"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb591-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb591-1" tabindex="-1"></a><span class="co"># Elbow method</span></span>
<span id="cb591-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb591-2" tabindex="-1"></a><span class="fu">fviz_nbclust</span>(wider_baseline_df, <span class="at">FUNcluster =</span> kmeans, <span class="at">method =</span> <span class="st">&quot;wss&quot;</span>) <span class="sc">+</span></span>
<span id="cb591-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb591-3" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">&quot;Elbow method&quot;</span>) </span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-463-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb592"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb592-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb592-1" tabindex="-1"></a><span class="co"># Silhouette method</span></span>
<span id="cb592-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb592-2" tabindex="-1"></a><span class="fu">fviz_nbclust</span>(wider_baseline_df, <span class="at">FUNcluster =</span> kmeans, <span class="at">method =</span> <span class="st">&quot;silhouette&quot;</span>) <span class="sc">+</span> </span>
<span id="cb592-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb592-3" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">&quot;Silhouette method&quot;</span>) </span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-463-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb593"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb593-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb593-1" tabindex="-1"></a><span class="co"># Gap statistic method</span></span>
<span id="cb593-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb593-2" tabindex="-1"></a><span class="fu">fviz_nbclust</span>(wider_baseline_df, <span class="at">FUNcluster =</span> kmeans, <span class="at">method =</span> <span class="st">&quot;gap_stat&quot;</span>) <span class="sc">+</span> </span>
<span id="cb593-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb593-3" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">&quot;Gap Statisitc method&quot;</span>) </span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-463-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>The elbow method is suggesting 2 or 3 clusters, the silhouette method is suggesting 2, and the gap statistic method is suggesting 1. Since each of these methods is recommending different <em>k</em> values, we can go ahead and run <em>k</em>-means to visualize the clusters and test those different <em>k</em>’s. <em>K</em>-means clusters will be visualized using the <code>fviz_cluster()</code> function.</p>
<div class="sourceCode" id="cb594"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb594-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb594-1" tabindex="-1"></a><span class="co"># Choosing to iterate through 2 or 3 clusters using i as our iterator</span></span>
<span id="cb594-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb594-2" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>){</span>
<span id="cb594-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb594-3" tabindex="-1"></a>    <span class="co"># nstart = number of random starting partitions, it&#39;s recommended for nstart &gt; 1</span></span>
<span id="cb594-4"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb594-4" tabindex="-1"></a>    cluster_k <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(wider_baseline_df, <span class="at">centers =</span> i, <span class="at">nstart =</span> <span class="dv">25</span>)</span>
<span id="cb594-5"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb594-5" tabindex="-1"></a>    cluster_plot <span class="ot">&lt;-</span> <span class="fu">fviz_cluster</span>(cluster_k, <span class="at">data =</span> wider_baseline_df) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="fu">paste0</span>(<span class="st">&quot;k = &quot;</span>, i))</span>
<span id="cb594-6"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb594-6" tabindex="-1"></a>    <span class="fu">print</span>(cluster_plot)</span>
<span id="cb594-7"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb594-7" tabindex="-1"></a>}</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-464-1.png" width="672" style="display: block; margin: auto;" /><img src="_main_files/figure-html/unnamed-chunk-464-2.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="answer-to-environmental-health-question-1-11" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 1<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#answer-to-environmental-health-question-1-11" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #1</strong></em>: What is the optimal number of clusters the cytokines can be grouped into that were derived from nasal epithelium fluid (NELF) in non-smokers using <em>k</em>-means clustering?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: 2 or 3 clusters can be justified here, based on using the elbow or silhouette method or if <em>k</em>-means happens to group cytokines together that were implicated in similar biological pathways. In the final paper, we moved forward with 3 clusters, because it was justifiable from the methods and provided more granularity in the clusters.</p>
</div>
<p>The final cluster assignments can easily be obtained using the <code>kmeans()</code> function from the <em>stats</em> package.</p>
<div class="sourceCode" id="cb595"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb595-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb595-1" tabindex="-1"></a>cluster_kmeans_3 <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(wider_baseline_df, <span class="at">centers =</span> <span class="dv">3</span>, <span class="at">nstart =</span> <span class="dv">25</span>)</span>
<span id="cb595-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb595-2" tabindex="-1"></a>cluster_kmeans_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(cluster_kmeans_3<span class="sc">$</span>cluster) <span class="sc">%&gt;%</span></span>
<span id="cb595-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb595-3" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="st">&quot;Cytokine&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb595-4"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb595-4" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="st">`</span><span class="at">K-Means Cluster</span><span class="st">`</span> <span class="ot">=</span> cluster_kmeans_3.cluster) <span class="sc">%&gt;%</span></span>
<span id="cb595-5"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb595-5" tabindex="-1"></a>    <span class="co"># Ordering the dataframe for easier comparison</span></span>
<span id="cb595-6"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb595-6" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="st">`</span><span class="at">K-Means Cluster</span><span class="st">`</span>)</span>
<span id="cb595-7"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb595-7" tabindex="-1"></a></span>
<span id="cb595-8"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb595-8" tabindex="-1"></a>cluster_kmeans_df</span></code></pre></div>
<pre><code>##       Cytokine K-Means Cluster
## 1  Fractalkine               1
## 2          IL8               1
## 3         IP10               1
## 4          MIG               1
## 5      Eotaxin               2
## 6     Eotaxin3               2
## 7         IL13               2
## 8        IL17B               2
## 9         IL1b               2
## 10         IL6               2
## 11        ITAC               2
## 12        MCP1               2
## 13       MIP1a               2
## 14       MIP1b               2
## 15        TARC               2
## 16        I309               3
## 17        IFNg               3
## 18        IL10               3
## 19     IL12p70               3
## 20       IL17D               3
## 21         IL4               3
## 22        TNFa               3</code></pre>
</div>
<div id="answer-to-environmental-health-question-2-12" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 2<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#answer-to-environmental-health-question-2-12" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #2</strong></em>: After selecting a cluster number, which cytokines were assigned to each <em>k</em>-means cluster?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: After choosing the number of clusters to be 3, the cluster assignments are as follows:
<img src="Module5_5_Input/Module5_5_Image7.png" width="371" style="display: block; margin: auto;" /></p>
</div>
<p><br></p>
</div>
</div>
<div id="hierarchical-clustering-1" class="section level2 hasAnchor">
<h2>Hierarchical Clustering<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Next, we’ll turn our attention to answering environmental health questions 3 and 4: What is the optimal number of clusters the cytokines can be grouped into that were derived from nasal epithelium fluid (NELF) in non-smokers using hierarchical clustering? How do the hierarchical cluster assignments compare to the <em>k</em>-means cluster assignments?</p>
<p>Just as we used the elbow method, silhouette profile, and gap statistic to determine the optimal number of clusters for <em>k</em>-means, we can leverage the same approaches for hierarchical by changing the <code>FUNcluster</code> parameter.</p>
<div class="sourceCode" id="cb597"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb597-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb597-1" tabindex="-1"></a><span class="co"># Elbow method</span></span>
<span id="cb597-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb597-2" tabindex="-1"></a><span class="fu">fviz_nbclust</span>(wider_baseline_df, <span class="at">FUNcluster =</span> hcut, <span class="at">method =</span> <span class="st">&quot;wss&quot;</span>) <span class="sc">+</span></span>
<span id="cb597-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb597-3" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">&quot;Elbow method&quot;</span>) </span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-467-1.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb598"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb598-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb598-1" tabindex="-1"></a><span class="co"># Silhouette method</span></span>
<span id="cb598-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb598-2" tabindex="-1"></a><span class="fu">fviz_nbclust</span>(wider_baseline_df, <span class="at">FUNcluster =</span> hcut, <span class="at">method =</span> <span class="st">&quot;silhouette&quot;</span>) <span class="sc">+</span> </span>
<span id="cb598-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb598-3" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">&quot;Silhouette method&quot;</span>) </span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-467-2.png" width="672" style="display: block; margin: auto;" /></p>
<div class="sourceCode" id="cb599"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb599-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb599-1" tabindex="-1"></a><span class="co"># Gap statistic method</span></span>
<span id="cb599-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb599-2" tabindex="-1"></a><span class="fu">fviz_nbclust</span>(wider_baseline_df, <span class="at">FUNcluster =</span> hcut, <span class="at">method =</span> <span class="st">&quot;gap_stat&quot;</span>) <span class="sc">+</span> </span>
<span id="cb599-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb599-3" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">subtitle =</span> <span class="st">&quot;Gap Statisitc method&quot;</span>) </span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-467-3.png" width="672" style="display: block; margin: auto;" />
We can see the results are quite similar with 2-3 clusters appearing optimal.</p>
<div id="answer-to-environmental-health-question-3-6" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 3<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#answer-to-environmental-health-question-3-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #3</strong></em>: What is the optimal number of clusters the cytokines can be grouped into that were derived from nasal epithelium fluid (NELF) in non-smokers using hierarchical clustering?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: Again, 2 or 3 clusters can be justified here, but for the same reasons mentioned for the first environmental health question, we landed on 3 clusters.</p>
</div>
<p>Now we can perform the clustering and visualize and extract the results. We’ll start by using the <code>dist()</code> function to calculate the euclidean distance between the clusters followed by the <code>hclust()</code> function to obtain the hierarchical clustering assignments.</p>
<div class="sourceCode" id="cb600"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb600-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb600-1" tabindex="-1"></a><span class="co"># Viewing the wider dataframe we&#39;ll be working with</span></span>
<span id="cb600-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb600-2" tabindex="-1"></a><span class="fu">head</span>(wider_baseline_df)</span></code></pre></div>
<pre><code>##                  NS_1     NS_10     NS_11     NS_12     NS_13     NS_14
## Eotaxin      7.215729  8.016193  6.264252  6.938344  5.152683  7.937892
## Eotaxin3     4.396783  4.006488  5.289509  3.597945  3.051951  3.063025
## Fractalkine 13.584554 13.679882 12.952220 13.464652 12.110742 12.785656
## I309         0.000000  0.000000  0.000000  0.000000  0.000000  1.617372
## IFNg         4.220509  1.990018  2.656691  3.439682  2.318340  3.439850
## IL10         1.953721  1.145162  1.348623  1.801954  1.176498  1.800412
##                  NS_2        NS_3      NS_4        NS_5      NS_6      NS_7
## Eotaxin      9.773407  6.12290913  5.758602  4.19868540  6.832578  7.998467
## Eotaxin3     9.022390  2.50161364  2.721195  2.21084320  3.820379  5.549164
## Fractalkine 14.961872 12.71538464 12.057105 10.38711624 12.917180 14.636190
## I309         1.103140  0.00000000  0.000000  0.00000000  0.000000  0.000000
## IFNg         4.394705  0.00000000  2.638213  0.04830196  1.914092  3.251501
## IL10         2.652120  0.08945805  2.020155  0.84649523  1.119272  2.007941
##                  NS_8      NS_9
## Eotaxin      9.242464  7.127746
## Eotaxin3    13.724108  3.553097
## Fractalkine 14.047382 13.146599
## I309         0.000000  0.000000
## IFNg         4.104820  2.940248
## IL10         2.576814  1.970627</code></pre>
<div class="sourceCode" id="cb602"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb602-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb602-1" tabindex="-1"></a><span class="co"># First scaling data with each subject (down columns)</span></span>
<span id="cb602-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb602-2" tabindex="-1"></a>scaled_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">apply</span>(wider_baseline_df, <span class="dv">2</span>, scale))</span>
<span id="cb602-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb602-3" tabindex="-1"></a><span class="fu">rownames</span>(scaled_df) <span class="ot">=</span> <span class="fu">rownames</span>(wider_baseline_df)</span>
<span id="cb602-4"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb602-4" tabindex="-1"></a></span>
<span id="cb602-5"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb602-5" tabindex="-1"></a><span class="fu">head</span>(scaled_df)</span></code></pre></div>
<pre><code>##                   NS_1      NS_10       NS_11      NS_12       NS_13      NS_14
## Eotaxin      0.1563938  0.5913460  0.20280481  0.3812918  0.04218372  0.3889166
## Eotaxin3    -0.4836295 -0.3751558 -0.03031899 -0.4135856 -0.51644074 -0.7745030
## Fractalkine  1.6023933  1.9565250  1.80232784  1.9342844  1.89246390  1.5458678
## I309        -1.4818900 -1.3408820 -1.29538062 -1.2697485 -1.32801259 -1.1195175
## IFNg        -0.5236512 -0.8612069 -0.65999509 -0.4512458 -0.71152186 -0.6845711
## IL10        -1.0383105 -1.0648515 -0.97283813 -0.8409577 -1.01515934 -1.0758338
##                   NS_2       NS_3       NS_4       NS_5       NS_6       NS_7
## Eotaxin      0.5142129  0.5748344  0.3352251  0.1297382  0.6605088  0.4509596
## Eotaxin3     0.3434700 -0.3390521 -0.4516462 -0.3710044 -0.1481961 -0.1213017
## Fractalkine  1.6938060  2.2385415  1.9669167  1.6886200  2.2940816  2.0018134
## I309        -1.4569648 -0.9703707 -1.1565995 -0.9279215 -1.1738781 -1.4178220
## IFNg        -0.7086302 -0.9703707 -0.4731435 -0.9157541 -0.6599894 -0.6581332
## IL10        -1.1048056 -0.9477947 -0.6332576 -0.7146872 -0.8733797 -0.9486817
##                   NS_8       NS_9
## Eotaxin      0.5504372  0.6258899
## Eotaxin3     1.4999690 -0.3063521
## Fractalkine  1.5684615  2.1955622
## I309        -1.4077759 -1.2329735
## IFNg        -0.5380823 -0.4661787
## IL10        -0.8618228 -0.7190487</code></pre>
<p>The <code>dist()</code> function is initially used to calculate the Euclidean distance between each cytokine. Next, the <code>hclust()</code> function is used to run the hierarchical clustering analysis using the complete method by default. The method can be changed in the function using the method parameter.</p>
<div class="sourceCode" id="cb604"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb604-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb604-1" tabindex="-1"></a><span class="co"># Calculating euclidean dist</span></span>
<span id="cb604-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb604-2" tabindex="-1"></a>dist_matrix <span class="ot">&lt;-</span> <span class="fu">dist</span>(scaled_df, <span class="at">method =</span> <span class="st">&#39;euclidean&#39;</span>)</span>
<span id="cb604-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb604-3" tabindex="-1"></a></span>
<span id="cb604-4"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb604-4" tabindex="-1"></a><span class="co"># Hierarchical clustering</span></span>
<span id="cb604-5"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb604-5" tabindex="-1"></a>cytokines_hc <span class="ot">&lt;-</span> <span class="fu">hclust</span>(dist_matrix)</span></code></pre></div>
<p>Now we can generate a dendrogram to help us evaluate the results using the <code>fviz_dend()</code> function from the <em>factoextra</em> package. We use k=3 to be consistent with the <em>k</em>-means analysis.</p>
<div class="sourceCode" id="cb605"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb605-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb605-1" tabindex="-1"></a>    <span class="fu">fviz_dend</span>(cytokines_hc, <span class="at">k =</span> <span class="dv">3</span>, <span class="co"># Specifying k</span></span>
<span id="cb605-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb605-2" tabindex="-1"></a>              <span class="at">cex =</span> <span class="fl">0.85</span>, <span class="co"># Label size</span></span>
<span id="cb605-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb605-3" tabindex="-1"></a>              <span class="at">palette =</span> <span class="st">&quot;futurama&quot;</span>, <span class="co"># Color palette see ?ggpubr::ggpar</span></span>
<span id="cb605-4"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb605-4" tabindex="-1"></a>              <span class="at">rect =</span> <span class="cn">TRUE</span>, <span class="at">rect_fill =</span> <span class="cn">TRUE</span>, <span class="co"># Add rectangle around groups</span></span>
<span id="cb605-5"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb605-5" tabindex="-1"></a>              <span class="at">horiz =</span> <span class="cn">TRUE</span>, <span class="co"># Changes the orientation of the dendogram </span></span>
<span id="cb605-6"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb605-6" tabindex="-1"></a>              <span class="at">rect_border =</span> <span class="st">&quot;futurama&quot;</span>,  <span class="co"># Rectangle color</span></span>
<span id="cb605-7"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb605-7" tabindex="-1"></a>              <span class="at">labels_track_height =</span> <span class="fl">0.8</span>  <span class="co"># Changes the room for labels</span></span>
<span id="cb605-8"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb605-8" tabindex="-1"></a>             )</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-471-1.png" width="75%" style="display: block; margin: auto;" /></p>
<p>We can also extract those cluster assignments using the <code>cutree()</code> function from the <em>stats</em> package.</p>
<div class="sourceCode" id="cb606"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb606-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb606-1" tabindex="-1"></a>hc_assignments_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">cutree</span>(cytokines_hc, <span class="at">k =</span> <span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb606-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb606-2" tabindex="-1"></a>    <span class="fu">rownames_to_column</span>(<span class="st">&quot;Cytokine&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb606-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb606-3" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="st">`</span><span class="at">Hierarchical Cluster</span><span class="st">`</span> <span class="ot">=</span> cutree.cytokines_hc..k...<span class="fl">3.</span>) <span class="sc">%&gt;%</span></span>
<span id="cb606-4"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb606-4" tabindex="-1"></a>    <span class="co"># Ordering the dataframe for easier comparison</span></span>
<span id="cb606-5"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb606-5" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="st">`</span><span class="at">Hierarchical Cluster</span><span class="st">`</span>)</span>
<span id="cb606-6"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb606-6" tabindex="-1"></a></span>
<span id="cb606-7"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb606-7" tabindex="-1"></a><span class="co"># Combining the dataframes to compare the cluster assignments from each approach</span></span>
<span id="cb606-8"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb606-8" tabindex="-1"></a> comp <span class="ot">&lt;-</span> <span class="fu">full_join</span>(cluster_kmeans_df, hc_assignments_df, <span class="at">by =</span> <span class="st">&quot;Cytokine&quot;</span>)</span>
<span id="cb606-9"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb606-9" tabindex="-1"></a> </span>
<span id="cb606-10"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb606-10" tabindex="-1"></a> comp</span></code></pre></div>
<pre><code>##       Cytokine K-Means Cluster Hierarchical Cluster
## 1  Fractalkine               1                    2
## 2          IL8               1                    2
## 3         IP10               1                    2
## 4          MIG               1                    2
## 5      Eotaxin               2                    1
## 6     Eotaxin3               2                    1
## 7         IL13               2                    1
## 8        IL17B               2                    1
## 9         IL1b               2                    1
## 10         IL6               2                    1
## 11        ITAC               2                    1
## 12        MCP1               2                    1
## 13       MIP1a               2                    1
## 14       MIP1b               2                    1
## 15        TARC               2                    1
## 16        I309               3                    3
## 17        IFNg               3                    3
## 18        IL10               3                    3
## 19     IL12p70               3                    3
## 20       IL17D               3                    3
## 21         IL4               3                    3
## 22        TNFa               3                    3</code></pre>
<p>For additional resources on running hierarchical clustering in R, see <a href="https://agroninfotech.blogspot.com/2020/06/visualizing-clusters-in-r-hierarchical.html">Visualizing Clustering Dendrogram in R</a> and <a href="http://www.sthda.com/english/articles/31-principal-component-methods-in-r-practical-guide/117-hcpc-hierarchical-clustering-on-principal-components-essentials/">Hiearchical Clustering on Principal Components</a>.</p>
</div>
<div id="answer-to-environmental-health-question-4-4" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 4<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#answer-to-environmental-health-question-4-4" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #4</strong></em>: How do the hierarchical cluster assignments compare to the <em>k</em>-means cluster assignments?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: Though this may not always be the case, in this instance, we see that <em>k</em>-means and hierarchical clustering with k=3 clusters yield the same groupings despite the clusters being presented in a different order.</p>
</div>
<p><br></p>
</div>
</div>
<div id="clustering-plot" class="section level2 hasAnchor">
<h2>Clustering Plot<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#clustering-plot" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>One additional way to visualize clustering is to plot the first two principal components on the axes and color the data points based on their corresponding cluster. This visualization can be used for both <em>k</em>-means and hierarchical clustering using the <code>fviz_cluster()</code> function. This figure is essentially a PCA plot with shapes drawn around each cluster to make them distinct from each other.</p>
<div class="sourceCode" id="cb608"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb608-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb608-1" tabindex="-1"></a><span class="fu">fviz_cluster</span>(cluster_kmeans_3, <span class="at">data =</span> wider_baseline_df)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-473-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>Rather than using the <code>fviz_cluster()</code> function as shown in the figure above, we’ll extract the data to recreate the sample figure using <code>ggplot()</code>. For the manuscript this was necessary, since it was important to facet the plots for each compartment (i.e., NLF, NELF, sputum, and serum). For a single plot, this data extraction isn’t required, and the figure above can be further customized within the <code>fviz_cluster()</code> function. However, we’ll go through the steps of obtaining the indices to recreate the same polygons in <code>ggplot()</code> directly.</p>
<p><em>K</em>-means actually uses principal component analysis (PCA) to reduce a dataset’s dimensionality prior to obtaining the cluster assignments and plotting those clusters. Therefore, to obtain the coordinates of each cytokine within their respective clusters, PCA will need to be run first.</p>
<div class="sourceCode" id="cb609"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb609-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb609-1" tabindex="-1"></a><span class="co"># First running PCA</span></span>
<span id="cb609-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb609-2" tabindex="-1"></a>pca_cytokine <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(wider_baseline_df, <span class="at">scale =</span> <span class="cn">TRUE</span>, <span class="at">center =</span> <span class="cn">TRUE</span>)</span>
<span id="cb609-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb609-3" tabindex="-1"></a><span class="co"># Only need PC1 and PC2 for plotting, so selecting the first two columns</span></span>
<span id="cb609-4"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb609-4" tabindex="-1"></a>baseline_scores_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="fu">scores</span>(pca_cytokine)[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]) </span>
<span id="cb609-5"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb609-5" tabindex="-1"></a>baseline_scores_df<span class="sc">$</span>Cluster <span class="ot">&lt;-</span> cluster_kmeans_3<span class="sc">$</span>cluster</span>
<span id="cb609-6"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb609-6" tabindex="-1"></a>baseline_scores_df<span class="sc">$</span>Protein <span class="ot">&lt;-</span> <span class="fu">rownames</span>(baseline_scores_df)</span>
<span id="cb609-7"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb609-7" tabindex="-1"></a></span>
<span id="cb609-8"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb609-8" tabindex="-1"></a><span class="co"># Changing cluster to a character for plotting</span></span>
<span id="cb609-9"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb609-9" tabindex="-1"></a>baseline_scores_df<span class="sc">$</span>Cluster <span class="ot">=</span> <span class="fu">as.character</span>(baseline_scores_df<span class="sc">$</span>Cluster)</span>
<span id="cb609-10"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb609-10" tabindex="-1"></a></span>
<span id="cb609-11"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb609-11" tabindex="-1"></a><span class="fu">head</span>(baseline_scores_df)</span></code></pre></div>
<pre><code>##                    PC1         PC2 Cluster     Protein
## Eotaxin     -1.4837936 -0.37325290       2     Eotaxin
## Eotaxin3     0.7505718 -1.04251406       2    Eotaxin3
## Fractalkine -7.0500961 -0.49034208       1 Fractalkine
## I309         4.6869896  0.42263944       3        I309
## IFNg         2.4935464 -0.21910385       3        IFNg
## IL10         3.4361668 -0.04689913       3        IL10</code></pre>
<p>Within each cluster, the <code>chull()</code> function is used to compute the indices of the points on the convex hull. These are needed for <code>ggplot()</code> to create the polygon shapes of each cluster.</p>
<div class="sourceCode" id="cb611"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb611-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb611-1" tabindex="-1"></a><span class="co"># hull values for cluster 1</span></span>
<span id="cb611-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb611-2" tabindex="-1"></a>cluster_1 <span class="ot">&lt;-</span> baseline_scores_df[baseline_scores_df<span class="sc">$</span>Cluster <span class="sc">==</span> <span class="dv">1</span>, ][<span class="fu">chull</span>(baseline_scores_df <span class="sc">%&gt;%</span> </span>
<span id="cb611-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb611-3" tabindex="-1"></a>                                                                    <span class="fu">filter</span>(Cluster <span class="sc">==</span> <span class="dv">1</span>)),]  </span>
<span id="cb611-4"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb611-4" tabindex="-1"></a><span class="co"># hull values for cluster 2</span></span>
<span id="cb611-5"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb611-5" tabindex="-1"></a>cluster_2 <span class="ot">&lt;-</span> baseline_scores_df[baseline_scores_df<span class="sc">$</span>Cluster <span class="sc">==</span> <span class="dv">2</span>, ][<span class="fu">chull</span>(baseline_scores_df <span class="sc">%&gt;%</span> </span>
<span id="cb611-6"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb611-6" tabindex="-1"></a>                                                                    <span class="fu">filter</span>(Cluster <span class="sc">==</span> <span class="dv">2</span>)),] </span>
<span id="cb611-7"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb611-7" tabindex="-1"></a><span class="co"># hull values for cluster 3</span></span>
<span id="cb611-8"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb611-8" tabindex="-1"></a>cluster_3 <span class="ot">&lt;-</span> baseline_scores_df[baseline_scores_df<span class="sc">$</span>Cluster <span class="sc">==</span> <span class="dv">3</span>, ][<span class="fu">chull</span>(baseline_scores_df <span class="sc">%&gt;%</span> </span>
<span id="cb611-9"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb611-9" tabindex="-1"></a>                                                                    <span class="fu">filter</span>(Cluster <span class="sc">==</span> <span class="dv">3</span>)),]  </span>
<span id="cb611-10"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb611-10" tabindex="-1"></a>all_hulls_baseline <span class="ot">&lt;-</span> <span class="fu">rbind</span>(cluster_1, cluster_2, cluster_3)</span>
<span id="cb611-11"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb611-11" tabindex="-1"></a><span class="co"># Changing cluster to a character for plotting</span></span>
<span id="cb611-12"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb611-12" tabindex="-1"></a>all_hulls_baseline<span class="sc">$</span>Cluster <span class="ot">=</span> <span class="fu">as.character</span>(all_hulls_baseline<span class="sc">$</span>Cluster)</span>
<span id="cb611-13"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb611-13" tabindex="-1"></a></span>
<span id="cb611-14"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb611-14" tabindex="-1"></a><span class="fu">head</span>(all_hulls_baseline)</span></code></pre></div>
<pre><code>##                 PC1        PC2 Cluster  Protein
## MIG      -4.7353654 -0.2498662       1      MIG
## IL8      -7.7200078 -0.8317172       1      IL8
## IP10     -3.8660925  4.2481655       1     IP10
## Eotaxin3  0.7505718 -1.0425141       2 Eotaxin3
## MIP1b    -0.8699488 -0.8834532       2    MIP1b
## MCP1     -2.4837364 -0.3652235       2     MCP1</code></pre>
<p>Now plotting the clusters using <code>ggplot()</code>.</p>
<div class="sourceCode" id="cb613"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb613-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-1" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span> </span>
<span id="cb613-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-2" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">data =</span> baseline_scores_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> Cluster, <span class="at">shape =</span> Cluster), <span class="at">size =</span> <span class="dv">4</span>) <span class="sc">+</span> </span>
<span id="cb613-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-3" tabindex="-1"></a>  <span class="co"># Adding cytokine names</span></span>
<span id="cb613-4"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-4" tabindex="-1"></a>  <span class="fu">geom_text_repel</span>(<span class="at">data =</span> baseline_scores_df, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> Cluster, <span class="at">label =</span> Protein), </span>
<span id="cb613-5"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-5" tabindex="-1"></a>                  <span class="at">show.legend =</span> <span class="cn">FALSE</span>, <span class="at">size =</span> <span class="fl">4.5</span>) <span class="sc">+</span> </span>
<span id="cb613-6"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-6" tabindex="-1"></a>  <span class="co"># Creating polygon shapes of the clusters</span></span>
<span id="cb613-7"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-7" tabindex="-1"></a>  <span class="fu">geom_polygon</span>(<span class="at">data =</span> all_hulls_baseline, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">group =</span> <span class="fu">as.factor</span>(Cluster), <span class="at">fill =</span> Cluster, </span>
<span id="cb613-8"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-8" tabindex="-1"></a>                                        <span class="at">color =</span> Cluster), <span class="at">alpha =</span> <span class="fl">0.25</span>, <span class="at">show.legend =</span> <span class="cn">FALSE</span>) <span class="sc">+</span> </span>
<span id="cb613-9"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-9" tabindex="-1"></a></span>
<span id="cb613-10"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-10" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span> </span>
<span id="cb613-11"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-11" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="fl">0.5</span>), <span class="co">#rotating x labels/ moving x labels slightly to the left</span></span>
<span id="cb613-12"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-12" tabindex="-1"></a>        <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">colour=</span><span class="st">&quot;black&quot;</span>), <span class="co">#making x and y axes black</span></span>
<span id="cb613-13"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-13" tabindex="-1"></a>        <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">13</span>), <span class="co">#changing size of x axis labels</span></span>
<span id="cb613-14"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-14" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">&quot;bold&quot;</span>, <span class="at">size =</span> <span class="fu">rel</span>(<span class="fl">1.7</span>)), <span class="co">#changes axis titles</span></span>
<span id="cb613-15"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-15" tabindex="-1"></a>        <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">&#39;bold&#39;</span>, <span class="at">size =</span> <span class="dv">17</span>), <span class="co">#changes legend title</span></span>
<span id="cb613-16"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-16" tabindex="-1"></a>        <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>), <span class="co">#changes legend text</span></span>
<span id="cb613-17"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-17" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">&#39;bottom&#39;</span>, <span class="co"># moving the legend to the bottom</span></span>
<span id="cb613-18"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-18" tabindex="-1"></a>        <span class="at">legend.background =</span> <span class="fu">element_rect</span>(<span class="at">colour =</span> <span class="st">&#39;black&#39;</span>, <span class="at">fill =</span> <span class="st">&#39;white&#39;</span>, <span class="at">linetype =</span> <span class="st">&#39;solid&#39;</span>), <span class="co">#changes the legend background</span></span>
<span id="cb613-19"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-19" tabindex="-1"></a>        <span class="at">strip.text.x =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>), <span class="co">#changes size of facet x axis </span></span>
<span id="cb613-20"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-20" tabindex="-1"></a>        <span class="at">strip.text.y =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>)) <span class="sc">+</span> <span class="co">#changes size of facet y axis </span></span>
<span id="cb613-21"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-21" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&#39;Dimension 1 (85.1%)&#39;</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&#39;Dimension 2 (7.7%)&#39;</span>) <span class="sc">+</span> <span class="co">#changing axis labels </span></span>
<span id="cb613-22"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-22" tabindex="-1"></a></span>
<span id="cb613-23"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-23" tabindex="-1"></a>  <span class="co"># Using colors from the startrek palette from ggsci</span></span>
<span id="cb613-24"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-24" tabindex="-1"></a>  <span class="fu">scale_color_startrek</span>(<span class="at">name =</span> <span class="st">&#39;Cluster&#39;</span>) <span class="sc">+</span></span>
<span id="cb613-25"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb613-25" tabindex="-1"></a>  <span class="fu">scale_fill_startrek</span>(<span class="at">name =</span> <span class="st">&#39;Cluster&#39;</span>) </span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-476-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>An appropriate title for this figure could be:</p>
<p>“<strong>Figure X. <em>K</em>-means clusters of cytokines at baseline.</strong> Cytokines samples are derived from nasal epithelium (NELF) samples in 14 non-smoking subjects. Cytokine concentration values were transformed using a data reduction technique known as Principal Component Analysis (PCA). The first two eigenvectors plotted on the axes were able to capture a majority of the variance across all samples from the original dataset.”</p>
<p>Takeaways from this clustering plot:</p>
<ul>
<li>PCA was able to capture almost all (~93%) of the variance from the original dataset</li>
<li>The 22 cytokines were able to be clustered into 3 distinct clusters using <em>k</em>-means</li>
</ul>
<p><br></p>
</div>
<div id="hierarchical-clustering-visualization" class="section level2 hasAnchor">
<h2>Hierarchical Clustering Visualization<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-visualization" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We can also build a heatmap using the <code>pheatmap()</code> function that has the capability to display hierarchical clustering dendrograms. To do so, we’ll need to go back and use the <code>wider_baseline_df</code> dataframe.</p>
<div class="sourceCode" id="cb614"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb614-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb614-1" tabindex="-1"></a><span class="fu">pheatmap</span>(wider_baseline_df, </span>
<span id="cb614-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb614-2" tabindex="-1"></a>            <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>, <span class="co"># hierarchical clustering of cytokines</span></span>
<span id="cb614-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb614-3" tabindex="-1"></a>            <span class="at">scale =</span> <span class="st">&quot;column&quot;</span>,    <span class="co"># scaling the data to make differences across cytokines more apparent</span></span>
<span id="cb614-4"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb614-4" tabindex="-1"></a>            <span class="at">cutree_row =</span> <span class="dv">3</span>, <span class="co"># adds a space between the 3 largest clusters</span></span>
<span id="cb614-5"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb614-5" tabindex="-1"></a>            <span class="at">display_numbers =</span> <span class="cn">TRUE</span>, <span class="at">number_color =</span> <span class="st">&quot;black&quot;</span>, <span class="at">fontsize =</span> <span class="dv">12</span>, <span class="co"># adding average concentration values</span></span>
<span id="cb614-6"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb614-6" tabindex="-1"></a>            <span class="at">angle_col =</span> <span class="dv">45</span>, <span class="at">fontsize_col =</span> <span class="dv">12</span>, <span class="at">fontsize_row =</span> <span class="dv">12</span>, <span class="co"># adjusting size/ orientation of axes labels</span></span>
<span id="cb614-7"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb614-7" tabindex="-1"></a>            <span class="at">cellheight =</span> <span class="dv">17</span>, <span class="at">cellwidth =</span> <span class="dv">30</span> <span class="co"># setting height and width for cells</span></span>
<span id="cb614-8"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb614-8" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-477-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>An appropriate title for this figure could be:</p>
<p>“<strong>Figure X. Hierarchical clustering of cytokines at baseline.</strong> Cytokines samples are derived from nasal epithelium (NELF) samples in 14 non-smoking subjects. The heatmap visualizes psuedo log<sub>2</sub> cytokine concentrations that were scaled within each subject.”</p>
<p>It may be helpful to add axes titles like “Subject ID” for the x axis, “Cytokine” for the y axis, and “Scaled pslog<sub>2</sub> Concentration” for the legend after exporting from R. The <code>pheatmap()</code> function does not have the functionality to add those titles.</p>
<p>Nevertheless, let’s identify some key takeaways from this heatmap:</p>
<ul>
<li>The 22 cytokines were able to be clustered into 3 distinct clusters using hierarchical clustering</li>
<li>These clusters are based on cytokine concentration levels with the first cluster having the highest expression, the second cluster having the lowest expression, and the last cluster having average expression</li>
</ul>
<p><br></p>
</div>
<div id="variable-contributions" class="section level2 hasAnchor">
<h2>Variable Contributions<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#variable-contributions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To answer our final environmental health question: Which cytokines have the greatest contributions to the first two eigenvectors, we’ll use the <code>fviz_contrib()</code> function that plots the percentage of each variable’s contribution to the principal component(s). It also displays a red dashed line, and variables that fall above are considered to have significant contributions to those principal components. For a refresher on PCA and variable contributions, see the previous module, <strong>TAME 2.0 Module 5.4 Unsupervised Machine Learning</strong>.</p>
<div class="sourceCode" id="cb615"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb615-1"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb615-1" tabindex="-1"></a><span class="co"># kmeans contributions</span></span>
<span id="cb615-2"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb615-2" tabindex="-1"></a><span class="fu">fviz_contrib</span>(pca_cytokine, </span>
<span id="cb615-3"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb615-3" tabindex="-1"></a>             <span class="at">choice =</span> <span class="st">&quot;ind&quot;</span>, <span class="at">addlabels =</span> <span class="cn">TRUE</span>,</span>
<span id="cb615-4"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#cb615-4" tabindex="-1"></a>             <span class="at">axes =</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>) <span class="co"># specifies to show contribution percentages for first 2 PCs</span></span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-478-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>An appropriate title for this figure could be:</p>
<p>“<strong>Figure X. Cytokine contributions to principal components.</strong> The bar chart displays each cytokine’s contribution to the first two eigenvectors in descending order from left to right. The red dashed line represents the expected contribution of each cytokine if all inputs were uniform, therefore the seven cytokines that fall above this reference line are considered to have significant contributions to the first two principal components.”</p>
<div id="answer-to-environmental-health-question-5-3" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 5<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#answer-to-environmental-health-question-5-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #5</strong></em>: Which cytokines have the greatest contributions to the first two eigenvectors?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: The cytokines that have significant contributions to the first two principal components include IL-8, Fractalkine, IP-10, IL-4, MIG, I309, and IL-12p70.</p>
</div>
<p><br></p>
</div>
</div>
<div id="concluding-remarks-22" class="section level2 hasAnchor">
<h2>Concluding Remarks<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#concluding-remarks-22" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this module, we explored scenarios where clustering would be appropriate but lack contextual details informing the number of clusters that should be considered, thus resulting in the need to derive such a number. In addition, methodology for <em>k</em>-means and hierarchical clustering was presented, along with corresponding visualizations. Lastly, variable contributions to the eigenvectors were introduced as a means to determine the most influential variables on the principal components’ composition.</p>
<div id="additional-resources-10" class="section level3 hasAnchor">
<h3>Additional Resources<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#additional-resources-10" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><a href="https://uc-r.github.io/kmeans_clustering#silo"><em>K</em>-Means Cluster Analysis</a></li>
<li><a href="https://www.datanovia.com/en/lessons/k-means-clustering-in-r-algorith-and-practical-examples/"><em>K</em>-Means Clustering in R</a></li>
<li><a href="https://uc-r.github.io/hc_clustering">Hierarchical Clustering in R</a></li>
</ul>
<p><br></p>
<p><label class="tykfont">
Test Your Knowledge
</label></p>
<div class="tyk">
<p>Using the same dataset, answer the questions below.</p>
<ol style="list-style-type: decimal">
<li>Determine the optimal number of <em>k</em>-means clusters of cytokines derived from the nasal epithelieum lining fluid of <strong>e-cigarette users</strong>.</li>
<li>How do those clusters compare to the ones that were derived at baseline (in non-smokers)?</li>
<li>Which cytokines have the greatest contributions to the first two eigenvectors?</li>
</ol>
</div>

</div>
</div>
</div>



            </section>

          </div>
        </div>
      </div>
<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="descriptive-cohort-analyses.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
