<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>5.4 Unsupervised Machine Learning Part 1: K-Means Clustering &amp; PCA | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</title>
  <meta name="description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="generator" content="bookdown 0.40 and GitBook 2.6.7" />

  <meta property="og:title" content="5.4 Unsupervised Machine Learning Part 1: K-Means Clustering &amp; PCA | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="5.4 Unsupervised Machine Learning Part 1: K-Means Clustering &amp; PCA | TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research" />
  
  <meta name="twitter:description" content="This is an example of how the TAME 2.0 website will look, I hope!" />
  

<meta name="author" content="Rager Lab" />


<meta name="date" content="2024-10-17" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="Icons_Used_Throughout/Favicon.png" type="image/x-icon" />
<link rel="prev" href="supervised-machine-learning-model-interpretation.html"/>
<link rel="next" href="unsupervised-machine-learning-part-2-additional-clustering-applications.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<link href="libs/htmltools-fill-0.5.8.1/fill.css" rel="stylesheet" />
<script src="libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.33/datatables.js"></script>
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<script src="libs/kePrint-0.0.1/kePrint.js"></script>
<link href="libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="libs/tabwid-1.1.3/tabwid.css" rel="stylesheet" />
<script src="libs/tabwid-1.1.3/tabwid.js"></script>
<link href="libs/table1-1.0/table1_defaults.css" rel="stylesheet" />
<!-- Google tag (gtag.js) -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-FGFKVX2X3G"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-FGFKVX2X3G');
</script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><a href="./"><img src="Icons_Used_Throughout/TAME 2.0 Logo.png"><br></a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Preface</a></li>
<li class="part"><span><b>Chapter 1 Introductory <br>Data Science</b></span></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html"><i class="fa fa-check"></i>1.1 FAIR Data Management Practices</a>
<ul>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-training-module"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#introduction-to-fair"><i class="fa fa-check"></i>Introduction to FAIR</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#breaking-down-fair-letter-by-letter"><i class="fa fa-check"></i>Breaking Down FAIR, Letter-by-Letter</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#what-does-this-mean-for-you"><i class="fa fa-check"></i>What Does This Mean for You?</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#data-repositories-for-sharing-of-data"><i class="fa fa-check"></i>Data Repositories for Sharing of Data</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#recent-shifts-in-regulatory-policies-for-data-sharing"><i class="fa fa-check"></i>Recent Shifts in Regulatory Policies for Data Sharing</a></li>
<li class="chapter" data-level="" data-path="fair-data-management-practices.html"><a href="fair-data-management-practices.html#additional-training-resources-on-fair"><i class="fa fa-check"></i>Additional Training Resources on FAIR</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html"><i class="fa fa-check"></i>1.2 Data Sharing through Online Repositories</a>
<ul>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#an-overview-and-example-with-the-dataverse-repository"><i class="fa fa-check"></i>An Overview and Example with the Dataverse Repository</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#introduction-to-training-module-1"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#data-repositories"><i class="fa fa-check"></i>Data Repositories</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#the-dataverse-project"><i class="fa fa-check"></i>The Dataverse Project</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#what-is-a-dataverse"><i class="fa fa-check"></i>What is a Dataverse?</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#metadata"><i class="fa fa-check"></i>Metadata</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataverse"><i class="fa fa-check"></i>Creating a Dataverse</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#creating-a-dataset"><i class="fa fa-check"></i>Creating a Dataset</a></li>
<li class="chapter" data-level="" data-path="data-sharing-through-online-repositories.html"><a href="data-sharing-through-online-repositories.html#concluding-remarks-1"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html"><i class="fa fa-check"></i>1.3 File Management using Github</a>
<ul>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#introduction-to-training-module-2"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-an-account"><i class="fa fa-check"></i>Creating an Account</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#creating-a-repository"><i class="fa fa-check"></i>Creating a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#uploading-code"><i class="fa fa-check"></i>Uploading Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#adding-subfolders-in-a-repository"><i class="fa fa-check"></i>Adding Subfolders in a Repository</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-code"><i class="fa fa-check"></i>Updating Code</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#updating-repository-titles-and-structure-to-support-a-manuscript"><i class="fa fa-check"></i>Updating Repository Titles and Structure to Support a Manuscript</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#tracking-code-changes-using-github-branches"><i class="fa fa-check"></i>Tracking Code Changes using Github Branches</a></li>
<li class="chapter" data-level="" data-path="file-management-using-github.html"><a href="file-management-using-github.html#concluding-remarks-2"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html"><i class="fa fa-check"></i>1.4 Data Wrangling in Excel</a>
<ul>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#introduction-to-training-module-3"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#save-a-copy-of-the-soon-to-be-organized-and-cleaned-dataset-as-a-new-file"><i class="fa fa-check"></i>Save a Copy of the Soon-To-Be Organized and Cleaned Dataset as a New File</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-extraneous-white-space"><i class="fa fa-check"></i>Remove Extraneous White Space</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#replace-missing-data-with-na"><i class="fa fa-check"></i>Replace Missing Data with “NA”</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-metadata-tab"><i class="fa fa-check"></i>Create a Metadata Tab</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#abbreviate-and-capitalize-categorical-data"><i class="fa fa-check"></i>Abbreviate and Capitalize Categorical Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#alphabetize-sort-the-data-by-the-categorical-variable-of-interest"><i class="fa fa-check"></i>Alphabetize (Sort) the Data by the Categorical Variable of Interest</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-new-subject-number-column"><i class="fa fa-check"></i>Create a New Subject Number Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#remove-special-symbols-and-dashes"><i class="fa fa-check"></i>Remove Special Symbols and Dashes</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#bold-all-column-names-and-center-all-data"><i class="fa fa-check"></i>Bold all Column Names and Center all Data</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#create-a-subject-identifier-column"><i class="fa fa-check"></i>Create a Subject Identifier Column</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#separate-subject-demographic-data-from-experimental-measurements"><i class="fa fa-check"></i>Separate Subject Demographic Data from Experimental Measurements</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#convert-data-from-wide-to-long-format"><i class="fa fa-check"></i>Convert Data from Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#pivoting-data-from-a-wide-to-long-format"><i class="fa fa-check"></i>Pivoting Data from a Wide to Long Format</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#generating-summary-level-statistics-with-pivot-tables"><i class="fa fa-check"></i>Generating Summary-Level Statistics with Pivot Tables</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#excel-vs.-r-which-should-you-use"><i class="fa fa-check"></i>Excel vs. R: Which Should You Use?</a></li>
<li class="chapter" data-level="" data-path="data-wrangling-in-excel.html"><a href="data-wrangling-in-excel.html#concluding-remarks-3"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 2 Coding in R</b></span></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html"><i class="fa fa-check"></i>2.1 Downloading and Programming in R</a>
<ul>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-training-module-4"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#general-introduction-and-installation-of-r-and-rstudio"><i class="fa fa-check"></i>General Introduction and Installation of R and RStudio</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#introduction-to-r-packages"><i class="fa fa-check"></i>Introduction to R Packages</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#scripting-basics"><i class="fa fa-check"></i>Scripting Basics</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#code-troubleshooting"><i class="fa fa-check"></i>Code Troubleshooting</a></li>
<li class="chapter" data-level="" data-path="downloading-and-programming-in-r.html"><a href="downloading-and-programming-in-r.html#concluding-remarks-4"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html"><i class="fa fa-check"></i>2.2 Coding “Best” Practices</a>
<ul>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#introduction-to-training-module-5"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#scripting-file-types"><i class="fa fa-check"></i>Scripting File Types</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-headers-and-annotation"><i class="fa fa-check"></i>Script Headers and Annotation</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#coding-style"><i class="fa fa-check"></i>Coding Style</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#script-organization"><i class="fa fa-check"></i>Script Organization</a></li>
<li class="chapter" data-level="" data-path="coding-best-practices.html"><a href="coding-best-practices.html#concluding-remarks-5"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html"><i class="fa fa-check"></i>2.3 Data Manipulation and Reshaping</a>
<ul>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-training-module-6"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#data-manipulation-using-base-r"><i class="fa fa-check"></i>Data Manipulation Using Base R</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#introduction-to-tidyverse"><i class="fa fa-check"></i>Introduction to Tidyverse</a></li>
<li class="chapter" data-level="" data-path="data-manipulation-and-reshaping.html"><a href="data-manipulation-and-reshaping.html#concluding-remarks-6"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html"><i class="fa fa-check"></i>2.4 Improving Coding Efficiencies</a>
<ul>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#introduction-to-training-module-7"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#loops"><i class="fa fa-check"></i>Loops</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#functions"><i class="fa fa-check"></i>Functions</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#list-operations"><i class="fa fa-check"></i>List operations</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#concluding-remarks-7"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="improving-coding-efficiencies.html"><a href="improving-coding-efficiencies.html#additional-resources-2"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="part"><span><b>Chapter 3 Basics of <br>Data Analysis and Visualizations</b></span></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html"><i class="fa fa-check"></i>3.1 Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-data-visualizations"><i class="fa fa-check"></i>Introduction to Data Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#introduction-to-training-module-8"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#density-plot-visualization"><i class="fa fa-check"></i>Density Plot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#boxplot-visualization"><i class="fa fa-check"></i>Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#correlation-visualizations"><i class="fa fa-check"></i>Correlation Visualizations</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#heatmap-visualization"><i class="fa fa-check"></i>Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="data-visualizations.html"><a href="data-visualizations.html#concluding-remarks-8"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html"><i class="fa fa-check"></i>3.2 Improving Data Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-data-visulization-conventions"><i class="fa fa-check"></i>Introduction to Data Visulization Conventions</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#introduction-to-training-module-9"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-boxplot-visualization"><i class="fa fa-check"></i>Creating an Improved Boxplot Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-an-improved-heatmap-visualization"><i class="fa fa-check"></i>Creating an Improved Heatmap Visualization</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#creating-multi-plot-figures"><i class="fa fa-check"></i>Creating Multi-Plot Figures</a></li>
<li class="chapter" data-level="" data-path="improving-data-visualizations.html"><a href="improving-data-visualizations.html#concluding-remarks-9"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html"><i class="fa fa-check"></i>3.3 Normality Tests and Data Transformations</a>
<ul>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#introduction-to-training-module-10"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#what-is-a-normal-distribution"><i class="fa fa-check"></i>What is a Normal Distribution?</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#qualitative-assessment-of-normality"><i class="fa fa-check"></i>Qualitative Assessment of Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#quantitative-normality-assessment"><i class="fa fa-check"></i>Quantitative Normality Assessment</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#data-transformation"><i class="fa fa-check"></i>Data Transformation</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#additional-considerations-regarding-normality"><i class="fa fa-check"></i>Additional Considerations Regarding Normality</a></li>
<li class="chapter" data-level="" data-path="normality-tests-and-data-transformations.html"><a href="normality-tests-and-data-transformations.html#concluding-remarks-10"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html"><i class="fa fa-check"></i>3.4 Intoduction to Statistical Tests</a>
<ul>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#introduction-to-training-module-11"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#assessing-normality-homogeneity-of-variance"><i class="fa fa-check"></i>Assessing Normality &amp; Homogeneity of Variance</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#two-group-visualizations-and-statistical-comparisons-using-the-t-test"><i class="fa fa-check"></i>Two-Group Visualizations and Statistical Comparisons using the T-Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#three-group-visualizations-and-statistical-comparisons-using-an-anova"><i class="fa fa-check"></i>Three-Group Visualizations and Statistical Comparisons using an ANOVA</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#regression-modeling-and-visualization-linear-and-logistic-regressions"><i class="fa fa-check"></i>Regression Modeling and Visualization: Linear and Logistic Regressions</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#statistical-evaluations-of-categorical-data-using-the-chi-squared-test-and-fishers-exact-test"><i class="fa fa-check"></i>Statistical Evaluations of Categorical Data using the Chi-Squared Test and Fisher’s Exact Test</a></li>
<li class="chapter" data-level="" data-path="intoduction-to-statistical-tests.html"><a href="intoduction-to-statistical-tests.html#concluding-remarks-11"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 4 Converting Wet Lab Data into Dry Lab Analyses</b></span></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html"><i class="fa fa-check"></i>4.1 Overview of Experimental Design and Example Data</a>
<ul>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#introduction-to-training-module-12"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#replicates"><i class="fa fa-check"></i>Replicates</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#orientation-to-example-data-for-chapter-4"><i class="fa fa-check"></i>Orientation to Example Data for Chapter 4</a></li>
<li class="chapter" data-level="" data-path="overview-of-experimental-design-and-example-data.html"><a href="overview-of-experimental-design-and-example-data.html#concluding-remarks-12"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html"><i class="fa fa-check"></i>4.2 Data Import, Processing, and Summary Statistics</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#introduction-to-training-module-13"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#data-import"><i class="fa fa-check"></i>Data Import</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#handling-missing-values"><i class="fa fa-check"></i>Handling Missing Values</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#averaging-replicates"><i class="fa fa-check"></i>Averaging Replicates</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#descriptive-statistics"><i class="fa fa-check"></i>Descriptive Statistics</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#normality-assessment-and-data-transformation"><i class="fa fa-check"></i>Normality Assessment and Data Transformation</a></li>
<li class="chapter" data-level="" data-path="data-import-processing-and-summary-statistics.html"><a href="data-import-processing-and-summary-statistics.html#concluding-remarks-13"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html"><i class="fa fa-check"></i>4.3 Data Import from PDF Sources</a>
<ul>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#introduction-to-training-module-14"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-from-many-single-pdfs-with-the-same-formatting"><i class="fa fa-check"></i>Importing Data from Many Single PDFs with the Same Formatting</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#importing-data-stored-in-pdf-tables"><i class="fa fa-check"></i>Importing Data Stored in PDF Tables</a></li>
<li class="chapter" data-level="" data-path="data-import-from-pdf-sources.html"><a href="data-import-from-pdf-sources.html#concluding-remarks-14"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html"><i class="fa fa-check"></i>4.4 Two Group Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#introduction-to-training-module-15"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#overview-of-two-group-statistical-tests"><i class="fa fa-check"></i>Overview of Two Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#statistical-vs.-biological-significance"><i class="fa fa-check"></i>Statistical vs. Biological Significance</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#unpaired-test-example"><i class="fa fa-check"></i>Unpaired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#paired-test-example"><i class="fa fa-check"></i>Paired Test Example</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#visualizing-results"><i class="fa fa-check"></i>Visualizing Results</a></li>
<li class="chapter" data-level="" data-path="two-group-comparisons-and-visualizations.html"><a href="two-group-comparisons-and-visualizations.html#concluding-remarks-15"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html"><i class="fa fa-check"></i>4.5 Multi-Group and Multi-Variable Comparisons and Visualizations</a>
<ul>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#introduction-to-training-module-16"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#overview-of-multi-group-statistical-tests"><i class="fa fa-check"></i>Overview of Multi-Group Statistical Tests</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#multi-group-analysis-example"><i class="fa fa-check"></i>Multi-Group Analysis Example</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#visualization-of-multi-group-statistical-results"><i class="fa fa-check"></i>Visualization of Multi-Group Statistical Results</a></li>
<li class="chapter" data-level="" data-path="multi-group-and-multi-variable-comparisons-and-visualizations.html"><a href="multi-group-and-multi-variable-comparisons-and-visualizations.html#concluding-remarks-16"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html"><i class="fa fa-check"></i>4.6 Advanced Multi-Group Comparisons</a>
<ul>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#introduction-to-training-module-17"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#advanced-multi-group-comparisons-1"><i class="fa fa-check"></i>Advanced Multi-Group Comparisons</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#two-way-anova-example"><i class="fa fa-check"></i>Two-way ANOVA Example</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#ancova-example"><i class="fa fa-check"></i>ANCOVA Example</a></li>
<li class="chapter" data-level="" data-path="advanced-multi-group-comparisons.html"><a href="advanced-multi-group-comparisons.html#concluding-remarks-17"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 5 Machine Learning &amp; Artificial Intelligence</b></span></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><i class="fa fa-check"></i>5.1 Introduction to Artificial Intelligence, Machine Learning, and Predictive Modeling for Environmental Health</a>
<ul>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#introduction-to-training-module-18"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#general-historical-context-and-taxonomy-of-modern-aiml"><i class="fa fa-check"></i>General Historical Context and Taxonomy of Modern AI/ML</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#application-of-machine-learning-in-environmental-health-science"><i class="fa fa-check"></i>Application of Machine Learning in Environmental Health Science</a></li>
<li class="chapter" data-level="" data-path="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html"><a href="introduction-to-artificial-intelligence-machine-learning-and-predictive-modeling-for-environmental-health.html#concluding-remarks-18"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html"><i class="fa fa-check"></i>5.2 Supervised Machine Learning</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-training-module-19"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-machine-learning"><i class="fa fa-check"></i>Types of Machine Learning</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#types-of-supervised-machine-learning-algorithms"><i class="fa fa-check"></i>Types of Supervised Machine Learning Algorithms</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#training-supervised-machine-learning-models"><i class="fa fa-check"></i>Training Supervised Machine Learning Models</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#assessing-classification-based-model-performance"><i class="fa fa-check"></i>Assessing Classification-Based Model Performance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#introduction-to-activity-and-example-dataset"><i class="fa fa-check"></i>Introduction to Activity and Example Dataset</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#testing-for-differences-in-predictor-variables-across-the-outcome-classes"><i class="fa fa-check"></i>Testing for Differences in Predictor Variables across the Outcome Classes</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#predicting-ias-detection-with-a-random-forest-rf-model"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#class-imbalance"><i class="fa fa-check"></i>Class Imbalance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning.html"><a href="supervised-machine-learning.html#concluding-remarks-19"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html"><i class="fa fa-check"></i>5.3 Supervised Machine Learning Model Interpretation</a>
<ul>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-training-module-20"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance"><i class="fa fa-check"></i>Variable Importance</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary"><i class="fa fa-check"></i>Decision Boundary</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#introduction-to-example-dataset-and-activity"><i class="fa fa-check"></i>Introduction to Example Dataset and Activity</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#predicting-ias-detection-with-a-random-forest-rf-model-1"><i class="fa fa-check"></i>Predicting iAs Detection with a Random Forest (RF) Model</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#variable-importance-plot"><i class="fa fa-check"></i>Variable Importance Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot"><i class="fa fa-check"></i>Decision Boundary Plot</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#decision-boundary-plot-incorporating-smote"><i class="fa fa-check"></i>Decision Boundary Plot Incorporating SMOTE</a></li>
<li class="chapter" data-level="" data-path="supervised-machine-learning-model-interpretation.html"><a href="supervised-machine-learning-model-interpretation.html#concluding-remarks-20"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><i class="fa fa-check"></i>5.4 Unsupervised Machine Learning Part 1: K-Means Clustering &amp; PCA</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-training-module-21"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-unsupervised-machine-learning"><i class="fa fa-check"></i>Introduction to Unsupervised Machine Learning</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-example-data"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#identifying-clusters-of-chemicals-through-k-means"><i class="fa fa-check"></i>Identifying Clusters of Chemicals through <em>K</em>-Means</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#principal-component-analysis-pca-1"><i class="fa fa-check"></i>Principal Component Analysis (PCA)</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#incorporating-k-means-into-pca-for-predictive-modeling"><i class="fa fa-check"></i>Incorporating <em>K</em>-Means into PCA for Predictive Modeling</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-1-k-means-clustering-pca.html"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#concluding-remarks-21"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><i class="fa fa-check"></i>5.5 Unsupervised Machine Learning Part 2: Additional Clustering Applications</a>
<ul>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-training-module-22"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#k-means-clustering-1"><i class="fa fa-check"></i><em>K</em>-Means Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#introduction-to-example-data-1"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-1"><i class="fa fa-check"></i>Hierarchical Clustering</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#clustering-plot"><i class="fa fa-check"></i>Clustering Plot</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#hierarchical-clustering-visualization"><i class="fa fa-check"></i>Hierarchical Clustering Visualization</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#variable-contributions"><i class="fa fa-check"></i>Variable Contributions</a></li>
<li class="chapter" data-level="" data-path="unsupervised-machine-learning-part-2-additional-clustering-applications.html"><a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html#concluding-remarks-22"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="part"><span><b>Chapter 6 Applications in Toxicology &amp; Exposure Science</b></span></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html"><i class="fa fa-check"></i>6.1 Descriptive Cohort Analyses</a>
<ul>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#introduction-to-training-module-23"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#participant-exploration"><i class="fa fa-check"></i>Participant Exploration</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#chemical-detection-filtering"><i class="fa fa-check"></i>Chemical Detection Filtering</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#outlier-identification"><i class="fa fa-check"></i>Outlier Identification</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#summary-statistics-tables"><i class="fa fa-check"></i>Summary Statistics Tables</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#demographics-table"><i class="fa fa-check"></i>Demographics Table</a></li>
<li class="chapter" data-level="" data-path="descriptive-cohort-analyses.html"><a href="descriptive-cohort-analyses.html#concluding-remarks-23"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html"><i class="fa fa-check"></i>6.2 -Omics and System Biology: Transcriptomic Applications</a>
<ul>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-training-module-24"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-the-field-of--omics"><i class="fa fa-check"></i>Introduction to the Field of “-Omics”</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#introduction-to-transcriptomics"><i class="fa fa-check"></i>Introduction to Transcriptomics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#formatting-data-for-downstream-statistics"><i class="fa fa-check"></i>Formatting Data for Downstream Statistics</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#transcriptomics-data-qaqc"><i class="fa fa-check"></i>Transcriptomics Data QA/QC</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#controlling-for-sources-of-sample-heterogeneity"><i class="fa fa-check"></i>Controlling for Sources of Sample Heterogeneity</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#identifying-genes-that-are-significantly-differentially-expressed-by-environmental-exposure-conditions-e.g.-biomass-smoke-exposure"><i class="fa fa-check"></i>Identifying Genes that are Significantly Differentially Expressed by Environmental Exposure Conditions (e.g., Biomass Smoke Exposure)</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-ma-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using MA Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#visualizing-statistical-results-using-volcano-plots"><i class="fa fa-check"></i>Visualizing Statistical Results using Volcano Plots</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#interpretting-findings-at-the-systems-level-through-pathway-enrichment-analysis"><i class="fa fa-check"></i>Interpretting Findings at the Systems Level through Pathway Enrichment Analysis</a></li>
<li class="chapter" data-level="" data-path="omics-and-system-biology-transcriptomic-applications.html"><a href="omics-and-system-biology-transcriptomic-applications.html#concluding-remarks-24"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><i class="fa fa-check"></i>6.3 Mixtures Analysis Methods Part 1: Overview and Example with Quantile G-Computation</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-training-module-25"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#overview-of-mixtures-analysis"><i class="fa fa-check"></i>Overview of Mixtures Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#introduction-to-example-data-2"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#mixtures-model-with-standard-qqcomp"><i class="fa fa-check"></i>Mixtures Model with Standard qqcomp</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#concluding-remarks-25"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html"><a href="mixtures-analysis-methods-part-1-overview-and-example-with-quantile-g-computation.html#additional-resources-11"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><i class="fa fa-check"></i>6.4 Mixtures Analysis Methods Part 2: Bayesian Kernel Machine Regression</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-training-module-26"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-example-data-3"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#introduction-to-bkmr"><i class="fa fa-check"></i>Introduction to BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#run-bkmr"><i class="fa fa-check"></i>Run BKMR</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#concluding-remarks-26"><i class="fa fa-check"></i>Concluding Remarks</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html"><a href="mixtures-analysis-methods-part-2-bayesian-kernel-machine-regression.html#additional-resources-12"><i class="fa fa-check"></i>Additional Resources</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html"><i class="fa fa-check"></i>6.5 Mixtures Analysis Methods Part 3: Sufficient Similarity</a>
<ul>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-training-module-27"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#introduction-to-toxicant-and-dataset"><i class="fa fa-check"></i>Introduction to Toxicant and Dataset</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#workspace-preparation-and-data-import-9"><i class="fa fa-check"></i>Workspace Preparation and Data Import</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#chemistry-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Chemistry-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#toxicity-based-sufficient-similarity-analysis"><i class="fa fa-check"></i>Toxicity-Based Sufficient Similarity Analysis</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#comparing-chemistry-vs.-toxicity-sufficient-similarity-analyses"><i class="fa fa-check"></i>Comparing Chemistry vs. Toxicity Sufficient Similarity Analyses</a></li>
<li class="chapter" data-level="" data-path="mixtures-analysis-methods-part-3-sufficient-similarity.html"><a href="mixtures-analysis-methods-part-3-sufficient-similarity.html#concluding-remarks-27"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html"><i class="fa fa-check"></i>6.6 Toxicokinetic Modeling</a>
<ul>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-training-module-28"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#introduction-to-toxicokinetic-modeling"><i class="fa fa-check"></i>Introduction to Toxicokinetic Modeling</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#data-and-models-used-in-toxicokinetic-modeling-tk"><i class="fa fa-check"></i>Data and Models used in Toxicokinetic Modeling (TK)</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-toxicokinetics"><i class="fa fa-check"></i>Reverse Toxicokinetics</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#capturing-population-variability-in-toxicokinetics-and-uncertainty-in-chemical-specific-parameters"><i class="fa fa-check"></i>Capturing Population Variability in Toxicokinetics, and Uncertainty in Chemical-Specific Parameters</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#reverse-tk-calculating-administered-equivalent-doses-for-toxcast-bioactive-concentrations"><i class="fa fa-check"></i>Reverse TK: Calculating Administered Equivalent Doses for ToxCast Bioactive Concentrations</a></li>
<li class="chapter" data-level="" data-path="toxicokinetic-modeling.html"><a href="toxicokinetic-modeling.html#comparing-equivalent-doses-estimated-to-elicit-toxicity-hazard-to-external-exposure-estimates-exposure-for-chemical-prioritization-by-bioactivity-exposure-ratios-bers"><i class="fa fa-check"></i>Comparing Equivalent Doses Estimated to Elicit Toxicity (Hazard) to External Exposure Estimates (Exposure), for Chemical Prioritization by Bioactivity-Exposure Ratios (BERs)</a></li>
</ul></li>
<li class="part"><span><b>Chapter 7 Environmental Health Database Mining</b></span></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html"><i class="fa fa-check"></i>7.1 Comparative Toxicogenomics Database</a>
<ul>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#introduction-to-training-module-30"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#ctd-data-in-r"><i class="fa fa-check"></i>CTD Data in R</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#identifying-genes-under-epigenetic-control"><i class="fa fa-check"></i>Identifying Genes Under Epigenetic Control</a></li>
<li class="chapter" data-level="" data-path="comparative-toxicogenomics-database.html"><a href="comparative-toxicogenomics-database.html#concluding-remarks-30"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html"><i class="fa fa-check"></i>7.2 Gene Expression Omnibus</a>
<ul>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-training-module-31"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-geo"><i class="fa fa-check"></i>Introduction to GEO</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#introduction-to-example-data-4"><i class="fa fa-check"></i>Introduction to Example Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#geo-data-in-r"><i class="fa fa-check"></i>GEO Data in R</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#visualizing-data"><i class="fa fa-check"></i>Visualizing Data</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#statistical-analyses"><i class="fa fa-check"></i>Statistical Analyses</a></li>
<li class="chapter" data-level="" data-path="gene-expression-omnibus.html"><a href="gene-expression-omnibus.html#concluding-remarks-31"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html"><i class="fa fa-check"></i>7.3 CompTox Dashboard Data through APIs</a>
<ul>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#introduction-to-training-module-32"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#training-modules-environmental-health-questions-24"><i class="fa fa-check"></i>Training Module’s Environmental Health Questions</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#script-preparations-12"><i class="fa fa-check"></i>Script Preparations</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#introduction-to-comptox-chemicals-dashboard"><i class="fa fa-check"></i>Introduction to CompTox Chemicals Dashboard</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#cctes-ctx-application-programming-interfaces-apis-for-automated-batch-search-of-the-ccd"><i class="fa fa-check"></i>CCTE’s CTX Application Programming Interfaces (APIs) for Automated Batch Search of the CCD</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#comparing-physico-chemical-properties-between-two-important-environmental-contaminant-lists"><i class="fa fa-check"></i>Comparing Physico-chemical Properties between Two Important Environmental Contaminant Lists</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#hazard-data-genotoxicity"><i class="fa fa-check"></i>Hazard Data: Genotoxicity</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#hazard-resource"><i class="fa fa-check"></i>Hazard Resource</a></li>
<li class="chapter" data-level="" data-path="comptox-dashboard-data-through-apis.html"><a href="comptox-dashboard-data-through-apis.html#concluding-remarks-32"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html"><i class="fa fa-check"></i>7.4 Database Integration: Air Quality, Mortality, and Environmental Justice Data</a>
<ul>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#introduction-to-training-module-33"><i class="fa fa-check"></i>Introduction to Training Module</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#introduction-to-exposure-and-health-databases"><i class="fa fa-check"></i>Introduction to Exposure and Health Databases</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#analyzing-relationships-between-pm2.5-and-mortality"><i class="fa fa-check"></i>Analyzing Relationships between PM<sub>2.5</sub> and Mortality</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#environmental-justice-considerations"><i class="fa fa-check"></i>Environmental Justice Considerations</a></li>
<li class="chapter" data-level="" data-path="database-integration-air-quality-mortality-and-environmental-justice-data.html"><a href="database-integration-air-quality-mortality-and-environmental-justice-data.html#concluding-remarks-33"><i class="fa fa-check"></i>Concluding Remarks</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">TAME 2.0: An Update to the TAME Toolkit for Introductory Data Science, Chemical-Biological Analyses, Machine Learning and Predictive Modeling, and Database Mining for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="unsupervised-machine-learning-part-1-k-means-clustering-pca" class="section level1 hasAnchor">
<h1>5.4 Unsupervised Machine Learning Part 1: K-Means Clustering &amp; PCA<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#unsupervised-machine-learning-part-1-k-means-clustering-pca" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This training module was developed by David M. Reif with contributions from Alexis Payton, Lauren E. Koval, and Julia E. Rager.</p>
<p>All input files (script, data, and figures) can be downloaded from the <a href="https://github.com/UNCSRP/TAME2">UNC-SRP TAME2 GitHub website</a>.</p>
<div id="introduction-to-training-module-21" class="section level2 hasAnchor">
<h2>Introduction to Training Module<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-training-module-21" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>To reiterate what has been discussed in the previous module, machine learning is a field that has great utility in environmental health sciences, often to investigate high-dimensional datasets. The two main classifications of machine learning discussed throughout the TAME Toolkit are supervised and unsupervised machine learning, though additional classifications exist. Previously, we discussed artificial intelligence and supervised machine learning in <strong>TAME 2.0 Module 5.1 Introduction to Machine Learning &amp; Artificial Intelligence</strong>, <strong>TAME 2.0 Module 5.2 Supervised Machine Learning</strong>, and <strong>TAME 2.0 Module 5.3 Supervised Machine Learning Model Interpretation</strong>. In this module, we’ll cover background information on unsupervised machine learning and then work through a scripted example of an unsupervised machine learning analysis.</p>
</div>
<div id="introduction-to-unsupervised-machine-learning" class="section level2 hasAnchor">
<h2>Introduction to Unsupervised Machine Learning<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-unsupervised-machine-learning" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>Unsupervised machine learning</strong>, as opposed to supervised machine learning, involves training a model on a dataset lacking ground truths or response variables. In this regard, unsupervised approaches are often used to identify underlying patterns amongst data in a more unbiased manner. This can provide the analyst with insights into the data that may not otherwise be apparent. Unsupervised machine learning has been used for understanding differences in gene expression patterns of breast cancer patients (<a href="https://link.springer.com/article/10.1186/s13058-015-0550-y">Jezequel et. al, 2015</a>) and evaluating metabolomic signatures of patients with and without cystic fibrosis (<a href="https://onlinelibrary.wiley.com/doi/full/10.1002/ppul.23225?casa_token=Vqlz3JgGm10AAAAA%3A4UFubAP2r97CKl9PK8oYDfgrcjrs_ZySDzDCx1t3qc6XvQRxOqIwjTn_eQxm_lzX8UQLE0zURJu94fI">Laguna et. al, 2015</a>).</p>
<div class="moduletextbox">
<p><strong>Note</strong>: Unsupervised machine learning is used for exploratory purposes, and just because it can find relationships between data points, that doesn’t necessarily mean that those relationships have merit, are indicative of causal relationships, or have direct biological implications. Rather, these methods can be used to find new patterns that can also inform future studies testing direct relationships.</p>
</div>
<img src="Module5_4_Input/Module5_4_Image1.png" width="75%" style="display: block; margin: auto;" />
<center>
Langs, G., Röhrich, S., Hofmanninger, J., Prayer, F., Pan, J., Herold, C., &amp; Prosch, H. (2018). Machine learning: from radiomics to discovery and routine. Der Radiologe, 58(S1), 1–6. PMID: <a href="https://doi.org/10.1007/s00117-018-0407-3">34013136</a>. Figure regenerated here in alignment with its published <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution 4.0 International License</a>
</center>
<p>Unsupervised machine learning includes:</p>
<ul>
<li><strong>Clustering</strong>: Involves grouping elements in a dataset such that the elements in the same group are more similar to each other than to the elements in the other groups.
<ul>
<li>Exclusive (<em>K</em>-means)</li>
<li>Overlapping</li>
<li>Hierarchical</li>
<li>Probabilistic</li>
</ul></li>
<li><strong>Dimensionality reduction</strong>: Focuses on taking high-dimensional data and transforming it into a lower-dimensional space that has fewer features while preserving important information inherent to the original dataset. This is useful because reducing the number of features makes the data easier to visualize while trying to maintain the initial integrity of the dataset.
<ul>
<li>Principal Component Analysis (PCA)</li>
<li>Singular Value Decomposition (SVD)</li>
<li>t-Distributed Stochastic Neighbor Embedding (t-SNE)</li>
<li>Uniform Manifold Approximation and Projection (UMAP)</li>
<li>Partial Least Squares-Discriminant Analysis (PLS-DA)</li>
</ul></li>
</ul>
<p>In this module, we’ll focus on methods for <strong><em>K</em>-means clustering</strong> and <strong>Principal Component Analysis</strong> described in more detail in the following sections. In the next module, <strong>TAME 2.0 Module 5.5 Unsupervised Machine Learning II: Hierarchical Clustering</strong>, we’ll focus on hierarchical clustering. For further information on types of unsupervised machine learning, check out <a href="https://cloud.google.com/discover/what-is-unsupervised-learning#section-3">Unsupervised Learning</a>.</p>
<p><br></p>
<div id="k-means-clustering" class="section level3 hasAnchor">
<h3><em>K</em>-Means Clustering<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#k-means-clustering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p><em>K</em>-means is a common clustering algorithm used to partition quantitative data. This algorithm works by first randomly selecting a pre-specified number of clusters, <em>k</em>, across the data space with each cluster having a data centroid. When using a standard Euclidean distance metric, the distance is calculated from an observation to each centroid, then the observation is assigned to the cluster of the closest centroid. After all observations have been assigned to one of the <em>k</em> clusters, the average of all observations in a cluster is calculated, and the centroid for the cluster is moved to the location of the mean. The process then repeats, with the distance computed between the observations and the updated centroids. Observations may be reassigned to the same cluster or moved to a different cluster if it is closer to another centroid. These iterations continue until there are no longer changes between cluster assignments for observations, resulting in the final cluster assignments that are then carried forward for analysis/interpretation.</p>
<p>Helpful resources on <em>k</em>-means clustering include the following: <a href="https://web.stanford.edu/~hastie/ElemStatLearn/printings/ESLII_print12_toc.pdf">The Elements of Statistical Learning</a> &amp;
<a href="https://towardsdatascience.com/k-means-clustering-algorithm-applications-evaluation-methods-and-drawbacks-aa03e644b48a">Towards Data Science</a>.</p>
<p><br></p>
</div>
<div id="principal-component-analysis-pca" class="section level3 hasAnchor">
<h3>Principal Component Analysis (PCA)<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#principal-component-analysis-pca" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Principal Component Analysis, or PCA, is a dimensionality-reduction technique used to transform high-dimensional data into a lower dimensional space while trying to preserve as much of the variability in the original data as possible. PCA has strong foundations in linear algebra, so background knowledge of eigenvalues and eigenvectors is extremely useful. Though the mathematics of PCA is beyond the scope of this module, a variety of more in-depth resources on PCA exist including this <a href="%22https://towardsdatascience.com/the-mathematics-behind-principal-component-analysis-fff2d7f4b643%22">Towards Data Science Blog</a>, and this <a href="https://www.sartorius.com/en/knowledge/science-snippets/what-is-principal-component-analysis-pca-and-how-it-is-used-507186#:~:text=Principal%20component%20analysis%2C%20or%20PCA,more%20easily%20visualized%20and%20analyzed.">Sartorius Blog</a>. At a higher level, important concepts in PCA include:</p>
<ol style="list-style-type: decimal">
<li><p>PCA partitions variance in a dataset into linearly uncorrelated principal components (PCs), which are weighted combinations of the original features.</p></li>
<li><p>Each PC (starting from the first one) summarizes a decreasing percentage of variance.</p></li>
<li><p>Every instance (e.g. chemical) in the original dataset has a “weight” or score” on each PC.</p></li>
<li><p>Any combination of PCs can be compared to summarize relationships amongst the instances (e.g. chemicals), but typically it’s the first two eigenvectors that capture a majority of the variance.
<img src="Module5_4_Input/Module5_4_Image2.png" width="80%" style="display: block; margin: auto;" /></p></li>
</ol>
<p><br></p>
</div>
</div>
<div id="introduction-to-example-data" class="section level2 hasAnchor">
<h2>Introduction to Example Data<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#introduction-to-example-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In this activity, we are going to analyze an example dataset of physicochemical property information for chemicals spanning <strong>per- and polyfluoroalkyl substances (PFAS) and statins</strong>. PFAS represent a ubiquitous and pervasive class of man-made industrial chemicals that are commonly used in food packaging, commercial household products such as Teflon, cleaning products, and flame retardants. PFAS are recognized as highly stable compounds that, upon entering the environment, can persist for many years and act as harmful sources of exposure. Statins represent a class of lipid-lowering compounds that are commonly used as pharmaceutical treatments for patients at risk of cardiovascular disease. Because of their common use amongst patients, statins can also end up in water and wastewater effluent, making them environmentally relevant as well.</p>
<p>This example analysis was designed to evaluate the chemical space of these diverse compounds and to illustrate the utility of unsupervised machine learning methods to differentiate chemical class and make associations between chemical groupings that can inform a variety of environmental and toxicological applications. The two types of machine learning methods that will be employed are <em>k</em>-means and PCA (as described in the introduction).</p>
<div id="training-modules-environmental-health-questions-13" class="section level3 hasAnchor">
<h3>Training Module’s Environmental Health Questions<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#training-modules-environmental-health-questions-13" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This training module was specifically developed to answer the following environmental health questions:</p>
<ol style="list-style-type: decimal">
<li>Can we differentiate between PFAS and statin chemical classes when considering just the raw physicochemical property variables without applying unsupervised machine learning techniques?</li>
<li>If substances are able to be clustered, what are some of the physicochemical properties that seem to be driving chemical clustering patterns derived through <em>k</em>-means?</li>
<li>How do the data compare when physicochemical properties are reduced using PCA?</li>
<li>Upon reducing the data through PCA, which physicochemical property contributes the most towards informing data variance captured in the primary principal component?</li>
<li>If we did not have information telling us which chemical belonged to which class, could we use PCA and <em>k</em>-means to inform whether a chemical is more similar to a PFAS or a statin?</li>
<li>What kinds of applications/endpoints can be better understood and/or predicted because of these derived chemical groupings?</li>
</ol>
<p><br></p>
</div>
<div id="script-preparations-6" class="section level3 hasAnchor">
<h3>Script Preparations<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#script-preparations-6" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="cleaning-the-global-environment-8" class="section level4 hasAnchor">
<h4>Cleaning the global environment<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cleaning-the-global-environment-8" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb530"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb530-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb530-1" tabindex="-1"></a><span class="fu">rm</span>(<span class="at">list=</span><span class="fu">ls</span>())</span></code></pre></div>
</div>
<div id="installing-required-r-packages-7" class="section level4 hasAnchor">
<h4>Installing required R packages<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#installing-required-r-packages-7" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>If you already have these packages installed, you can skip this step, or you can run the below code which checks installation status for you</p>
<div class="sourceCode" id="cb531"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb531-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb531-1" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;factoextra&quot;</span>))</span>
<span id="cb531-2"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb531-2" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;factoextra&quot;</span>);</span>
<span id="cb531-3"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb531-3" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;pheatmap&quot;</span>))</span>
<span id="cb531-4"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb531-4" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;pheatmap&quot;</span>);</span>
<span id="cb531-5"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb531-5" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">requireNamespace</span>(<span class="st">&quot;cowplot&quot;</span>))</span>
<span id="cb531-6"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb531-6" tabindex="-1"></a>  <span class="fu">install.packages</span>(<span class="st">&quot;cowplot&quot;</span>);</span></code></pre></div>
</div>
<div id="loading-required-r-packages-1" class="section level4 hasAnchor">
<h4>Loading required R packages<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#loading-required-r-packages-1" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb532"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb532-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb532-1" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb532-2"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb532-2" tabindex="-1"></a><span class="fu">library</span>(factoextra)</span>
<span id="cb532-3"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb532-3" tabindex="-1"></a><span class="fu">library</span>(pheatmap) <span class="co">#used to make heatmaps</span></span>
<span id="cb532-4"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb532-4" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span></code></pre></div>
<p>Getting help with packages and functions</p>
<div class="sourceCode" id="cb533"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb533-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb533-1" tabindex="-1"></a>?tidyverse <span class="co"># Package documentation for tidyverse</span></span>
<span id="cb533-2"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb533-2" tabindex="-1"></a>?kmeans <span class="co"># Package documentation for kmeans (a part of the standard stats R package, automatically uploaded)</span></span>
<span id="cb533-3"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb533-3" tabindex="-1"></a>?prcomp <span class="co"># Package documentation for deriving principal components within a PCA (a part of the standard stats R package, automatically uploaded)</span></span>
<span id="cb533-4"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb533-4" tabindex="-1"></a>?pheatmap <span class="co"># Package documentation for pheatmap</span></span></code></pre></div>
</div>
<div id="set-your-working-directory-9" class="section level4 hasAnchor">
<h4>Set your working directory<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#set-your-working-directory-9" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb534"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb534-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb534-1" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/filepath to where your input files are&quot;</span>) </span></code></pre></div>
</div>
<div id="loading-the-example-dataset" class="section level4 hasAnchor">
<h4>Loading the Example Dataset<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#loading-the-example-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Let’s start by loading the datasets needed for this training module. We are going to use a dataset of substances that have a diverse chemical space of PFAS and statin compounds. This list of chemicals will be uploaded alongside physicochemical property data. The chemical lists for ‘PFAS’ and ‘Statins’ were obtained from the EPA’s Computational Toxicology Dashboard <a href="https://comptox.epa.gov/dashboard/chemical-lists">Chemical Lists</a>. The physicochemical properties were obtained by uploading these lists into the National Toxicology Program’s <a href="https://ice.ntp.niehs.nih.gov/">Integrated Chemical Environment (ICE)</a>.</p>
<div class="sourceCode" id="cb535"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb535-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb535-1" tabindex="-1"></a>dat <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;Module5_4_Input/Module5_4_InputData.csv&quot;</span>, <span class="at">fileEncoding =</span> <span class="st">&quot;UTF-8-BOM&quot;</span>)</span></code></pre></div>
</div>
<div id="data-viewing" class="section level4 hasAnchor">
<h4>Data Viewing<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#data-viewing" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Starting with the overall dimensions:</p>
<div class="sourceCode" id="cb536"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb536-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb536-1" tabindex="-1"></a><span class="fu">dim</span>(dat)</span></code></pre></div>
<pre><code>## [1] 144  14</code></pre>
<p>Then looking at the first four rows and five columns of data:</p>
<div class="sourceCode" id="cb538"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb538-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb538-1" tabindex="-1"></a>dat[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code></pre></div>
<pre><code>##   List                                        Substance.Name      CASRN
## 1 PFAS    Perfluoro-2-(trifluoromethyl)propanesulphonic acid 93762-09-5
## 2 PFAS                   Potassium perfluoroheptanesulfonate 60270-55-5
## 3 PFAS Bis(2-hydroxyethyl)ammonium perfluoroheptanesulfonate 70225-15-9
## 4 PFAS       Potassium perfluoro-p-ethylcyclohexanesulfonate   335-24-0
##           DTXSID Molecular.Weight
## 1 DTXSID90239569          300.100
## 2  DTXSID9069392          488.212
## 3 DTXSID60880946          555.258
## 4 DTXSID50880117          500.223</code></pre>
<p>Note that the first column, <code>List</code>, designates the following two larger chemical classes:</p>
<div class="sourceCode" id="cb540"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb540-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb540-1" tabindex="-1"></a><span class="fu">unique</span>(dat<span class="sc">$</span>List)</span></code></pre></div>
<pre><code>## [1] &quot;PFAS&quot;    &quot;Statins&quot;</code></pre>
<p>Let’s lastly view all of the column headers:</p>
<div class="sourceCode" id="cb542"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb542-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb542-1" tabindex="-1"></a><span class="fu">colnames</span>(dat)</span></code></pre></div>
<pre><code>##  [1] &quot;List&quot;                                             
##  [2] &quot;Substance.Name&quot;                                   
##  [3] &quot;CASRN&quot;                                            
##  [4] &quot;DTXSID&quot;                                           
##  [5] &quot;Molecular.Weight&quot;                                 
##  [6] &quot;OPERA..Boiling.Point&quot;                             
##  [7] &quot;OPERA..Henry.s.Law.Constant&quot;                      
##  [8] &quot;OPERA..Melting.Point&quot;                             
##  [9] &quot;OPERA..Negative.Log.of.Acid.Dissociation.Constant&quot;
## [10] &quot;OPERA..Octanol.Air.Partition.Coefficient&quot;         
## [11] &quot;OPERA..Octanol.Water.Distribution.Coefficient&quot;    
## [12] &quot;OPERA..Octanol.Water.Partition.Coefficient&quot;       
## [13] &quot;OPERA..Vapor.Pressure&quot;                            
## [14] &quot;OPERA..Water.Solubility&quot;</code></pre>
<p>In the data file, the first four columns represent chemical identifier information. All remaining columns represent different physicochemical properties derived from OPERA via <a href="https://ice.ntp.niehs.nih.gov/">Integrated Chemical Environment (ICE)</a>. Because the original titles of these physicochemical properties contained commas and spaces, R automatically converted these into periods. Hence, titles like <code>OPERA..Boiling.Point</code>.</p>
<p>For ease of downstream data analyses, let’s create a more focused dataframe option containing only one chemical identifier (CASRN) as row names and then just the physicochemical property columns.</p>
<div class="sourceCode" id="cb544"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb544-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb544-1" tabindex="-1"></a><span class="co"># Creating a new dataframe that contains the physiocochemical properties</span></span>
<span id="cb544-2"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb544-2" tabindex="-1"></a>chemical_prop_df <span class="ot">&lt;-</span> dat[,<span class="dv">5</span><span class="sc">:</span><span class="fu">ncol</span>(dat)]</span>
<span id="cb544-3"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb544-3" tabindex="-1"></a><span class="fu">rownames</span>(chemical_prop_df) <span class="ot">&lt;-</span> dat<span class="sc">$</span>CASRN</span></code></pre></div>
<p>Now explore this data subset:</p>
<div class="sourceCode" id="cb545"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb545-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb545-1" tabindex="-1"></a><span class="fu">dim</span>(chemical_prop_df) <span class="co"># overall dimensions</span></span></code></pre></div>
<pre><code>## [1] 144  10</code></pre>
<div class="sourceCode" id="cb547"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb547-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb547-1" tabindex="-1"></a>chemical_prop_df[<span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>] <span class="co"># viewing the first four rows and five columns</span></span></code></pre></div>
<pre><code>##            Molecular.Weight OPERA..Boiling.Point OPERA..Henry.s.Law.Constant
## 93762-09-5          300.100              213.095                       -3.60
## 60270-55-5          488.212              223.097                       -9.75
## 70225-15-9          555.258              223.097                       -9.75
## 335-24-0            500.223              220.578                       -7.56
##            OPERA..Melting.Point
## 93762-09-5               96.455
## 60270-55-5              273.228
## 70225-15-9              182.152
## 335-24-0                231.827
##            OPERA..Negative.Log.of.Acid.Dissociation.Constant
## 93762-09-5                                             0.175
## 60270-55-5                                            -1.810
## 70225-15-9                                             1.000
## 335-24-0                                               1.000</code></pre>
<div class="sourceCode" id="cb549"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb549-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb549-1" tabindex="-1"></a><span class="fu">colnames</span>(chemical_prop_df)</span></code></pre></div>
<pre><code>##  [1] &quot;Molecular.Weight&quot;                                 
##  [2] &quot;OPERA..Boiling.Point&quot;                             
##  [3] &quot;OPERA..Henry.s.Law.Constant&quot;                      
##  [4] &quot;OPERA..Melting.Point&quot;                             
##  [5] &quot;OPERA..Negative.Log.of.Acid.Dissociation.Constant&quot;
##  [6] &quot;OPERA..Octanol.Air.Partition.Coefficient&quot;         
##  [7] &quot;OPERA..Octanol.Water.Distribution.Coefficient&quot;    
##  [8] &quot;OPERA..Octanol.Water.Partition.Coefficient&quot;       
##  [9] &quot;OPERA..Vapor.Pressure&quot;                            
## [10] &quot;OPERA..Water.Solubility&quot;</code></pre>
<p><br></p>
</div>
</div>
<div id="evaluating-the-original-physicochemical-properties-across-substances" class="section level3 hasAnchor">
<h3>Evaluating the Original Physicochemical Properties across Substances<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#evaluating-the-original-physicochemical-properties-across-substances" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let’s first plot two physicochemical properties to determine if and how substances group together without any fancy data reduction or other machine learning techniques. This will answer <strong>Environmental Health Question #1</strong>: Can we differentiate between PFAS and statin chemical classes when considering just the raw physicochemical property variables without applying unsupervised machine learning techniques?</p>
<p>Let’s put molecular weight (<code>Molecular.Weight</code>) as one axis and boiling point (<code>OPERA..Boiling.Point</code>) on the other. We’ll also color by the chemical classes using the <code>List</code> column from the original dataframe.</p>
<div class="sourceCode" id="cb551"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb551-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb551-1" tabindex="-1"></a><span class="fu">ggplot</span>(chemical_prop_df[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], <span class="fu">aes</span>(<span class="at">x =</span> Molecular.Weight, <span class="at">y =</span> OPERA..Boiling.Point, <span class="at">color =</span> dat<span class="sc">$</span>List)) <span class="sc">+</span> </span>
<span id="cb551-2"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb551-2" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb551-3"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb551-3" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&#39;Version A: Bivariate Plot of Two Original Physchem Variables&#39;</span>) <span class="sc">+</span> </span>
<span id="cb551-4"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb551-4" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Molecular Weight&quot;</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Boiling Point&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-430-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Let’s plot two other physicochemical property variables, Henry’s Law constant (<code>OPERA..Henry.s.Law.Constant</code>) and melting point (<code>OPERA..Melting.Point</code>), to see if the same separation of chemical classes is apparent.</p>
<div class="sourceCode" id="cb552"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb552-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb552-1" tabindex="-1"></a><span class="fu">ggplot</span>(chemical_prop_df[,<span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>], <span class="fu">aes</span>(<span class="at">x =</span> OPERA..Henry.s.Law.Constant, <span class="at">y =</span> OPERA..Melting.Point, </span>
<span id="cb552-2"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb552-2" tabindex="-1"></a>                                                  <span class="at">color =</span> dat<span class="sc">$</span>List)) <span class="sc">+</span> </span>
<span id="cb552-3"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb552-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb552-4"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb552-4" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&#39;Version B: Bivariate Plot of Two Other Original Physchem Variables&#39;</span>) <span class="sc">+</span> </span>
<span id="cb552-5"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb552-5" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;OPERA..Henry.s.Law.Constant&quot;</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;OPERA..Melting.Point&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-431-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="answer-to-environmental-health-question-1-10" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 1<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#answer-to-environmental-health-question-1-10" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With these, we can answer <strong>Environmental Health Question #1</strong></em>: Can we differentiate between PFAS and statin chemical classes when considering just the raw physicochemical property variables without applying machine learning techniques?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: Only in part. From the first plot, we can see that PFAS tend to have lower molecular weight ranges in comparison to the statins, though other property variables clearly overlap in ranges of values making the groupings not entirely clear.</p>
</div>
<p><br></p>
</div>
</div>
<div id="identifying-clusters-of-chemicals-through-k-means" class="section level2 hasAnchor">
<h2>Identifying Clusters of Chemicals through <em>K</em>-Means<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#identifying-clusters-of-chemicals-through-k-means" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Let’s turn our attention to <strong>Environmental Health Question #2</strong>: If substances are able to be clustered, what are some of the physicochemical properties that seem to be driving chemical clustering patterns derived through <em>k</em>-means? This will be done deriving clusters of chemicals based on ALL underlying physicochemical property data using <em>k</em>-means clustering.</p>
<p>For this example, let’s coerce the <em>k</em>-means algorithms to calculate 2 distinct clusters (based on their corresponding mean centered values). Here, we choose to derive two distinct clusters, because we are ultimately going to see if we can use this information to predict each chemical’s classification into two distinct chemical classes (i.e., PFAS vs statins). Note that we can derive more clusters using similar code depending on the question being addressed.</p>
<p>We can give a name to this variable to easily provide the number of clusters in the next lines of code, <code>num.centers</code>:</p>
<div class="sourceCode" id="cb553"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb553-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb553-1" tabindex="-1"></a>num.centers <span class="ot">&lt;-</span> <span class="dv">2</span></span></code></pre></div>
<p>Here we derive chemical clusters using <em>k</em>-means:</p>
<div class="sourceCode" id="cb554"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb554-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb554-1" tabindex="-1"></a>clusters <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(chemical_prop_df,                  <span class="co"># input dataframe</span></span>
<span id="cb554-2"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb554-2" tabindex="-1"></a>                   <span class="at">centers =</span> num.centers,  <span class="co"># number of cluster centers to calculate</span></span>
<span id="cb554-3"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb554-3" tabindex="-1"></a>                   <span class="at">iter.max =</span> <span class="dv">1000</span>,        <span class="co"># the maximum number of iterations allowed</span></span>
<span id="cb554-4"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb554-4" tabindex="-1"></a>                   <span class="at">nstart =</span> <span class="dv">50</span>)            <span class="co"># the number of rows used as the random set for the initial centers (during the first iteration)</span></span></code></pre></div>
<p>The resulting property values that were derived as the final cluster centers can be pulled using:</p>
<div class="sourceCode" id="cb555"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb555-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb555-1" tabindex="-1"></a>clusters<span class="sc">$</span>centers</span></code></pre></div>
<pre><code>##   Molecular.Weight OPERA..Boiling.Point OPERA..Henry.s.Law.Constant
## 1         690.1443             233.0402                   -9.589444
## 2         395.0716             281.4445                   -8.655185
##   OPERA..Melting.Point OPERA..Negative.Log.of.Acid.Dissociation.Constant
## 1             183.7980                                        0.01658333
## 2             157.5036                                        1.33226852
##   OPERA..Octanol.Air.Partition.Coefficient
## 1                                 5.940861
## 2                                 6.629556
##   OPERA..Octanol.Water.Distribution.Coefficient
## 1                                     -2.541750
## 2                                     -1.271315
##   OPERA..Octanol.Water.Partition.Coefficient OPERA..Vapor.Pressure
## 1                                   4.000639             -5.538889
## 2                                   3.010302             -6.762009
##   OPERA..Water.Solubility
## 1               -3.760222
## 2               -3.450750</code></pre>
<p>Let’s add the cluster assignments to the physicochemical data and create a new dataframe, which can then be used in a heatmap visualization to see how these physicochemical data distributions clustered according to <em>k</em>-means.</p>
<p>These cluster assignments can be pulled from the <code>cluster</code> list output, where chemicals are designated to each cluster with either a 1 or 2. You can view these using:</p>
<div class="sourceCode" id="cb557"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb557-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb557-1" tabindex="-1"></a>clusters<span class="sc">$</span>cluster</span></code></pre></div>
<pre><code>##   93762-09-5   60270-55-5   70225-15-9     335-24-0     647-29-0   68259-12-1 
##            2            2            1            2            2            1 
##   68259-09-6   68259-07-4   60453-92-1     357-31-3  441296-91-9  749786-16-1 
##            2            2            2            2            1            1 
##   93762-10-8  135524-36-6   93894-55-4   34642-43-8    2706-91-4  791563-89-8 
##            2            2            1            2            2            1 
##     742-73-4   29420-49-3    3871-99-6   29359-39-5    3872-25-1  126105-34-8 
##            2            2            2            1            2            1 
##  630402-22-1 2274731-07-4   98789-57-2   85963-79-7     375-73-5  108427-53-8 
##            2            1            1            2            2            2 
##    4021-47-0  117806-54-9   67906-42-7   68555-66-8   92982-03-1     375-92-8 
##            2            2            1            2            1            2 
##  175905-36-9  102061-82-5  134615-58-0  174675-49-1   79780-39-5   91036-71-4 
##            2            2            1            1            1            1 
##   70225-17-1    6401-03-2     374-58-3     646-83-3   86525-30-6    3916-24-3 
##            2            2            2            2            1            2 
##   42409-05-2  474511-07-4    2795-39-3   45187-15-3   82382-12-5   79963-95-4 
##            2            1            1            2            2            2 
##   45298-90-6  134615-57-9  927670-12-0    2806-15-7   70225-14-8  131651-65-5 
##            2            2            2            1            1            2 
##  343629-46-9  144797-51-3   29081-56-9   80988-54-1 1379460-39-5  343629-43-6 
##            1            2            2            2            1            1 
##  146689-46-5   29457-72-5     355-46-4    3107-18-4   70259-86-8 1036375-28-6 
##            2            2            2            2            2            2 
##   70225-18-2   70225-16-0   84224-48-6  507453-86-3   40365-28-4  110676-15-8 
##            2            2            2            1            1            2 
##   70259-85-7    2106-55-0 1997344-07-6     423-41-6  115416-68-7   17202-41-4 
##            2            2            2            2            2            1 
##   93894-73-6  134615-56-8  134615-59-1   68259-08-5   68259-10-9     374-62-9 
##            1            2            1            2            2            2 
##   68555-67-9    2806-16-8   36913-91-4   85187-17-3  803688-15-5   55120-77-9 
##            2            1            1            1            2            2 
##     335-77-3  141263-54-9   95465-60-4  130200-44-1  144535-22-8  130468-11-0 
##            1            2            2            2            2            2 
##   93957-54-1  126059-69-6  153463-20-8  154417-69-3  147511-69-1  141263-69-6 
##            2            2            2            2            2            2 
##   77517-29-4   80799-31-1   73390-02-0     503-49-1  117678-63-4  145599-86-6 
##            2            2            2            2            2            2 
##  147098-20-2   85798-96-5  120551-59-9   13552-81-3   90761-31-2   79691-18-2 
##            1            2            2            1            2            2 
##   73573-88-3  114801-27-3  151106-12-6  129443-92-1  134523-03-8  122254-45-9 
##            2            2            2            2            1            2 
##   75330-75-5  137023-81-5  136320-61-1   87770-13-6   85551-06-0  144501-27-9 
##            2            2            2            2            2            2 
##  159014-70-7  153321-50-7  133983-25-2   78366-44-6  148750-02-1   79902-63-9 
##            2            2            2            2            2            2 
##  120185-34-4  120171-12-2  141267-47-2   94061-80-0  141240-46-2   81093-37-0 
##            2            2            2            2            2            2</code></pre>
<p>Because these results are listed in the exact same order as the inputted dataframe, we can simply add these assignments to the <code>chemical_prop_df</code> dataframe.</p>
<div class="sourceCode" id="cb559"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb559-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb559-1" tabindex="-1"></a>dat_wclusters <span class="ot">&lt;-</span> <span class="fu">cbind</span>(chemical_prop_df,clusters<span class="sc">$</span>cluster)</span>
<span id="cb559-2"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb559-2" tabindex="-1"></a><span class="fu">colnames</span>(dat_wclusters)[<span class="dv">11</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Cluster&quot;</span>  <span class="co"># renaming this new column &quot;Custer&quot;</span></span>
<span id="cb559-3"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb559-3" tabindex="-1"></a>dat_wclusters <span class="ot">&lt;-</span> dat_wclusters[<span class="fu">order</span>(dat_wclusters<span class="sc">$</span>Cluster),]  <span class="co"># sorting data by cluster assignments</span></span></code></pre></div>
<p>To generate a heatmap, we need to first create a separate dataframe for the cluster assignments, ordered in the same way as the physicochemical data:</p>
<div class="sourceCode" id="cb560"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb560-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb560-1" tabindex="-1"></a>hm_cluster <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(dat_wclusters<span class="sc">$</span>Cluster, <span class="at">row.names =</span> <span class="fu">row.names</span>(dat_wclusters))  <span class="co"># creating the dataframe</span></span>
<span id="cb560-2"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb560-2" tabindex="-1"></a><span class="fu">colnames</span>(hm_cluster) <span class="ot">&lt;-</span> <span class="st">&quot;Cluster&quot;</span>   <span class="co"># reassigning the column name</span></span>
<span id="cb560-3"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb560-3" tabindex="-1"></a>hm_cluster<span class="sc">$</span>Cluster <span class="ot">&lt;-</span> <span class="fu">as.factor</span>(hm_cluster<span class="sc">$</span>Cluster)   <span class="co"># coercing the cluster numbers into factor variables, to make the heatmap prettier</span></span>
<span id="cb560-4"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb560-4" tabindex="-1"></a></span>
<span id="cb560-5"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb560-5" tabindex="-1"></a><span class="fu">head</span>(hm_cluster)  <span class="co"># viewing this new cluster assignment dataframe</span></span></code></pre></div>
<pre><code>##             Cluster
## 70225-15-9        1
## 68259-12-1        1
## 441296-91-9       1
## 749786-16-1       1
## 93894-55-4        1
## 791563-89-8       1</code></pre>
<p>We’re going to go ahead and clean up the physiocochemical property names to make the heatmap a bit tidier.</p>
<div class="sourceCode" id="cb562"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb562-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb562-1" tabindex="-1"></a>clean_names1 <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;OPERA..&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(dat_wclusters))</span>
<span id="cb562-2"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb562-2" tabindex="-1"></a><span class="co"># &quot;\\.&quot; denotes a period</span></span>
<span id="cb562-3"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb562-3" tabindex="-1"></a>clean_names2 <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>, <span class="st">&quot; &quot;</span>, clean_names1)</span>
<span id="cb562-4"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb562-4" tabindex="-1"></a></span>
<span id="cb562-5"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb562-5" tabindex="-1"></a><span class="co"># Reassigning the cleaner names back to the df</span></span>
<span id="cb562-6"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb562-6" tabindex="-1"></a><span class="fu">colnames</span>(dat_wclusters) <span class="ot">=</span> clean_names2</span>
<span id="cb562-7"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb562-7" tabindex="-1"></a></span>
<span id="cb562-8"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb562-8" tabindex="-1"></a><span class="co"># Going back to add in the apostrophe in &quot;Henry&#39;s Law Constant&quot;</span></span>
<span id="cb562-9"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb562-9" tabindex="-1"></a><span class="fu">colnames</span>(dat_wclusters)[<span class="dv">3</span>] <span class="ot">=</span> <span class="st">&quot;Henry&#39;s Law Constant&quot;</span></span></code></pre></div>
<p>Then we can call this dataframe (<code>data_wclusters</code>) into the following heatmap visualization code leveraging the <code>pheatmap()</code> function. This function was designed specifically to enable clustered heatmap visualizations. Check out <a href="https://www.rdocumentation.org/packages/pheatmap/versions/1.0.12/topics/pheatmap">pheatmap Documenation</a> for additional information.</p>
<p><br></p>
<div id="heatmap-visualization-of-the-resulting-k-means-clusters" class="section level3 hasAnchor">
<h3>Heatmap Visualization of the Resulting <em>K</em>-Means Clusters<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#heatmap-visualization-of-the-resulting-k-means-clusters" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb563"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb563-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb563-1" tabindex="-1"></a><span class="fu">pheatmap</span>(dat_wclusters[,<span class="dv">1</span><span class="sc">:</span><span class="dv">10</span>], </span>
<span id="cb563-2"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb563-2" tabindex="-1"></a>            <span class="at">cluster_rows =</span> <span class="cn">FALSE</span>, <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>, <span class="co"># no further clustering, for simplicity</span></span>
<span id="cb563-3"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb563-3" tabindex="-1"></a>            <span class="at">scale =</span> <span class="st">&quot;column&quot;</span>,    <span class="co"># scaling the data to make differences across chemicals more apparent</span></span>
<span id="cb563-4"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb563-4" tabindex="-1"></a>            <span class="at">annotation_row =</span> hm_cluster, <span class="co"># calling the cluster assignment dataframe as a separate color bar</span></span>
<span id="cb563-5"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb563-5" tabindex="-1"></a>            <span class="at">annotation_names_row =</span> <span class="cn">FALSE</span>, <span class="co"># adding removing the annotation name (&quot;Cluster&quot;) from the x axis</span></span>
<span id="cb563-6"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb563-6" tabindex="-1"></a>            <span class="at">angle_col =</span> <span class="dv">45</span>, <span class="at">fontsize_col =</span> <span class="dv">7</span>, <span class="at">fontsize_row =</span> <span class="dv">3</span>, <span class="co"># adjusting size/ orientation of axes labels</span></span>
<span id="cb563-7"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb563-7" tabindex="-1"></a>            <span class="at">cellheight =</span> <span class="dv">3</span>, <span class="at">cellwidth =</span> <span class="dv">25</span>, <span class="co"># setting height and width for cells</span></span>
<span id="cb563-8"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb563-8" tabindex="-1"></a>            <span class="at">border_color =</span> <span class="cn">FALSE</span> <span class="co"># specify no border surrounding the cells</span></span>
<span id="cb563-9"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb563-9" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-439-1.png" width="960" /></p>
<p>An appropriate title for this figure could be:</p>
<p>“<strong>Figure X. Heatmap of physicochemical properties with <em>k</em>-means cluster assignments.</strong> Shown are the relative values for each physicochemical property labeled on the x axis. Individual chemical names are listed on the y axis. The chemicals are grouped based on their <em>k</em>-means cluster assignment as denoted by the color bar on the left.”</p>
<p>Notice that the <code>pheatmap()</code> function does not add axes or legend titles. Adding those can provide clarity, however those can be added to the figure after exporting from R in MS Powerpoint or Adobe.</p>
</div>
<div id="answer-to-environmental-health-question-2-11" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 2<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#answer-to-environmental-health-question-2-11" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #2</strong></em>: What are some of the physicochemical properties that seem to be driving chemical clustering patterns derived through <em>k</em>-means?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: Properties with values that show obvious differences between resulting clusters including molecular weight, boiling point, negative log of acid dissociation constant, octanol air partition coefficient, and octanol water distribution coefficient.</p>
</div>
<p><br></p>
</div>
</div>
<div id="principal-component-analysis-pca-1" class="section level2 hasAnchor">
<h2>Principal Component Analysis (PCA)<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#principal-component-analysis-pca-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Next, we will run through some example analyses applying the common data reduction technique of PCA. We’ll start by determining how much of the variance is able to be captured within the first two principal components to answer <strong>Environmental Health Question #3</strong>: How do the data compare when physicochemical properties are reduced using PCA?</p>
<p>We can calculate the principal components across ALL physicochemical data across all chemicals using the <code>prcomp()</code> function. Always make sure your data is centered and scaled prior to running to PCA, since it’s sensitive to variables having different scales.</p>
<div class="sourceCode" id="cb564"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb564-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb564-1" tabindex="-1"></a>my.pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(chemical_prop_df,   <span class="co"># input dataframe of physchem data</span></span>
<span id="cb564-2"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb564-2" tabindex="-1"></a>                   <span class="at">scale =</span> <span class="cn">TRUE</span>, <span class="at">center =</span> <span class="cn">TRUE</span>) </span></code></pre></div>
<p>We can see how much of the variance was able to be captured in each of the eigenvectors or dimensions using a scree plot.</p>
<div class="sourceCode" id="cb565"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb565-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb565-1" tabindex="-1"></a><span class="fu">fviz_eig</span>(my.pca, <span class="at">addlabels =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-441-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>We can also calculate these values and pull them into a dataframe for future use. For example, to pull the percentage of variance explained by each principal component, we can run the following calculations, where first eigenvalues (eigs) are calculated and then used to calculate percent of variance per principal component:</p>
<div class="sourceCode" id="cb566"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb566-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb566-1" tabindex="-1"></a>eigs <span class="ot">&lt;-</span> my.pca<span class="sc">$</span>sdev<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb566-2"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb566-2" tabindex="-1"></a>Comp.stats <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(eigs, eigs<span class="sc">/</span><span class="fu">sum</span>(eigs), <span class="at">row.names =</span> <span class="fu">names</span>(eigs))</span>
<span id="cb566-3"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb566-3" tabindex="-1"></a><span class="fu">colnames</span>(Comp.stats) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;Eigen_Values&quot;</span>, <span class="st">&quot;Percent_of_Variance&quot;</span>)</span>
<span id="cb566-4"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb566-4" tabindex="-1"></a></span>
<span id="cb566-5"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb566-5" tabindex="-1"></a><span class="fu">head</span>(Comp.stats)</span></code></pre></div>
<pre><code>##   Eigen_Values Percent_of_Variance
## 1    4.0925925          0.40925925
## 2    2.3840022          0.23840022
## 3    1.5415308          0.15415308
## 4    0.5873057          0.05873057
## 5    0.4769190          0.04769190
## 6    0.3659181          0.03659181</code></pre>
<div id="answer-to-environmental-health-question-3-5" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 3<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#answer-to-environmental-health-question-3-5" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #3</strong></em>: How do the data compare when physicochemical properties are reduced using PCA?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: Principal Component 1 captures ~41% of the variance and Principal Component 2 captures ~24% across all physicochemical property values across all chemicals. These two components together describe ~65% of data.</p>
</div>
<p><br></p>
<p>Next, we’ll use PCA to answer <strong>Environmental Health Question #4</strong>: Upon reducing the data through PCA, which physicochemical property contributes the most towards informing data variance captured in the primary principal component (Comp.1)?</p>
<p>Here are the resulting scores for each chemical’s contribution towards each principal component (shown here as components <code>PC1</code>-<code>PC10</code>).</p>
<div class="sourceCode" id="cb568"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb568-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb568-1" tabindex="-1"></a><span class="fu">head</span>(my.pca<span class="sc">$</span>x)</span></code></pre></div>
<pre><code>##                   PC1        PC2         PC3         PC4         PC5
## 93762-09-5 -2.0354310 -1.4824239  1.29328367  0.04865893  0.25305471
## 60270-55-5 -1.2249015  2.2857092 -0.24261251 -0.40140814  0.62781108
## 70225-15-9 -1.0944361  1.3915069 -0.03340358 -0.90391749  0.05735985
## 335-24-0   -1.1334897  1.0675553  0.14299978 -1.08713268  0.21172964
## 647-29-0   -0.4830620  0.1259827 -1.16147937  1.11383217 -0.29571642
## 68259-12-1 -0.3264762  0.2369031 -1.31984895  0.47512052 -1.17555774
##                    PC6         PC7         PC8          PC9        PC10
## 93762-09-5 -0.67755281 -0.14546495  1.25520979  0.230936852  0.14075496
## 60270-55-5 -1.01515519 -0.11316504 -0.34335060 -0.383682123  0.09849160
## 70225-15-9  0.14334979  0.38355763  0.01717491 -0.006702116 -0.02715723
## 335-24-0   -0.93947357 -0.22877517  0.11457280 -0.107720120  0.17701035
## 647-29-0    0.19174355 -0.18228398 -0.48013543  0.074967840  0.22750496
## 68259-12-1  0.05910151  0.01399123  0.03790455  0.043309249 -0.18032083</code></pre>
<p>And the resulting loading factors of each property’s contribution towards each principal component.</p>
<div class="sourceCode" id="cb570"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb570-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb570-1" tabindex="-1"></a>my.pca<span class="sc">$</span>rotation</span></code></pre></div>
<pre><code>##                                                           PC1          PC2
## Molecular.Weight                                   0.09825313  0.108454961
## OPERA..Boiling.Point                               0.46350428  0.029650863
## OPERA..Henry.s.Law.Constant                       -0.17856542 -0.502116638
## OPERA..Melting.Point                               0.20645719  0.474473487
## OPERA..Negative.Log.of.Acid.Dissociation.Constant  0.32172963 -0.119465105
## OPERA..Octanol.Air.Partition.Coefficient           0.45329804 -0.008918089
## OPERA..Octanol.Water.Distribution.Coefficient      0.33026147 -0.437053508
## OPERA..Octanol.Water.Partition.Coefficient         0.16152270 -0.342905461
## OPERA..Vapor.Pressure                             -0.35205065 -0.349618523
## OPERA..Water.Solubility                           -0.36456684  0.254809066
##                                                            PC3         PC4
## Molecular.Weight                                  -0.679740448 -0.48432419
## OPERA..Boiling.Point                               0.199365885  0.03108544
## OPERA..Henry.s.Law.Constant                        0.179876660 -0.27695374
## OPERA..Melting.Point                              -0.214857945 -0.09449999
## OPERA..Negative.Log.of.Acid.Dissociation.Constant  0.286239544 -0.58268278
## OPERA..Octanol.Air.Partition.Coefficient           0.132157694  0.04820475
## OPERA..Octanol.Water.Distribution.Coefficient     -0.005904541  0.15115855
## OPERA..Octanol.Water.Partition.Coefficient        -0.466833374  0.48460961
## OPERA..Vapor.Pressure                             -0.195008167 -0.24988717
## OPERA..Water.Solubility                            0.253557563  0.12980079
##                                                           PC5          PC6
## Molecular.Weight                                   0.17351578  0.357367954
## OPERA..Boiling.Point                               0.22224554 -0.018507534
## OPERA..Henry.s.Law.Constant                        0.30566003 -0.470666686
## OPERA..Melting.Point                              -0.08063905 -0.686723561
## OPERA..Negative.Log.of.Acid.Dissociation.Constant -0.66998767  0.029248038
## OPERA..Octanol.Air.Partition.Coefficient           0.20778895  0.205757894
## OPERA..Octanol.Water.Distribution.Coefficient      0.06378821  0.169005123
## OPERA..Octanol.Water.Partition.Coefficient        -0.48515616 -0.162279640
## OPERA..Vapor.Pressure                             -0.06782464 -0.007031883
## OPERA..Water.Solubility                           -0.29062543  0.284060941
##                                                           PC7         PC8
## Molecular.Weight                                  -0.11763362  0.32938640
## OPERA..Boiling.Point                              -0.12503355  0.09718690
## OPERA..Henry.s.Law.Constant                       -0.21138163  0.44526650
## OPERA..Melting.Point                              -0.34342931 -0.10233816
## OPERA..Negative.Log.of.Acid.Dissociation.Constant  0.09083446  0.03113686
## OPERA..Octanol.Air.Partition.Coefficient          -0.44434707 -0.29734602
## OPERA..Octanol.Water.Distribution.Coefficient     -0.22039378  0.05052653
## OPERA..Octanol.Water.Partition.Coefficient        -0.08082351  0.17794120
## OPERA..Vapor.Pressure                             -0.34958097 -0.65367959
## OPERA..Water.Solubility                           -0.65183980  0.34989539
##                                                           PC9        PC10
## Molecular.Weight                                   0.03295675 -0.02698233
## OPERA..Boiling.Point                               0.03336277 -0.81709497
## OPERA..Henry.s.Law.Constant                        0.19706729  0.10099077
## OPERA..Melting.Point                              -0.24532148  0.10229774
## OPERA..Negative.Log.of.Acid.Dissociation.Constant  0.02576652  0.03380215
## OPERA..Octanol.Air.Partition.Coefficient           0.49672303  0.39565984
## OPERA..Octanol.Water.Distribution.Coefficient     -0.74417871  0.19866516
## OPERA..Octanol.Water.Partition.Coefficient         0.30648450 -0.09852386
## OPERA..Vapor.Pressure                             -0.03202641 -0.31333091
## OPERA..Water.Solubility                           -0.05123040 -0.07696401</code></pre>
</div>
<div id="answer-to-environmental-health-question-4-3" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 4<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#answer-to-environmental-health-question-4-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With these results, we can answer <strong>Environmental Health Question #4</strong></em>: Upon reducing the data through PCA, which physicochemical property contributes the most towards informing data variance captured in the primary principal component (Comp.1)?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: Boiling point contributes the most towards principal component 1, as it has the largest magnitude (0.464).</p>
</div>
<p><br></p>
</div>
<div id="visualizing-pca-results" class="section level3 hasAnchor">
<h3>Visualizing PCA Results<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#visualizing-pca-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let’s turn our attention to <strong>Environmental Health Question #5</strong>: If we did not have information telling us which chemical belonged to which class, could we use PCA and <em>k</em>-means to inform whether a chemical is more similar to a PFAS or a statin?</p>
<p>We can start by answering this question by visualizing the first two principal components and coloring each chemical according to class (i.e. PFAS vs statins).</p>
<div class="sourceCode" id="cb572"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb572-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb572-1" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">data.frame</span>(my.pca<span class="sc">$</span>x), <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> dat<span class="sc">$</span>List)) <span class="sc">+</span> </span>
<span id="cb572-2"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb572-2" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb572-3"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb572-3" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&#39;Version C: PCA Plot of the First 2 PCs, colored by Chemical Class&#39;</span>) <span class="sc">+</span> </span>
<span id="cb572-4"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb572-4" tabindex="-1"></a>  <span class="co"># it&#39;s good practice to put the percentage of the variance captured in the axes titles</span></span>
<span id="cb572-5"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb572-5" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Principal Component 1 (40.9%)&quot;</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Principal Component 2 (23.8%)&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-445-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="answer-to-environmental-health-question-5-2" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 5<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#answer-to-environmental-health-question-5-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this, we can answer <strong>Environmental Health Question #5</strong></em>: If we did not have information telling us which chemical belonged to which class, could we use PCA and <em>k</em>-means to inform whether a chemical is more similar to a PFAS or a statin?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: Data become more compressed and variables reduce across principal components capturing the majority of the variance from the original dataset (~65%). This results in improved data visualizations, where all dimensions of the physiochemical dataset are compressed and captured across the displayed components. In addition, the figure above shows a clear separation between PFAS and statin chemical when visualizing the reduced dataset.</p>
</div>
<p><br></p>
</div>
</div>
<div id="incorporating-k-means-into-pca-for-predictive-modeling" class="section level2 hasAnchor">
<h2>Incorporating <em>K</em>-Means into PCA for Predictive Modeling<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#incorporating-k-means-into-pca-for-predictive-modeling" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>We can also identify cluster-based trends within data that are reduced after running PCA. This example analysis does so, expanding upon the previously generated PCA results.</p>
<div id="estimate-k-means-clusters-from-pca-results" class="section level3 hasAnchor">
<h3>Estimate <em>K</em>-Means Clusters from PCA Results<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#estimate-k-means-clusters-from-pca-results" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Let’s first run code similar to the previous <em>k</em>-means analysis and associated parameters, though instead here we will use data reduced values from the PCA analysis. Specifically, clusters across PCA “scores” values will be derived, where scores represent the relative amount each chemical contributed to each principal component.</p>
<div class="sourceCode" id="cb573"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb573-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb573-1" tabindex="-1"></a>clusters_PCA <span class="ot">&lt;-</span> <span class="fu">kmeans</span>(my.pca<span class="sc">$</span>x, <span class="at">centers =</span> num.centers, <span class="at">iter.max =</span> <span class="dv">1000</span>, <span class="at">nstart =</span> <span class="dv">50</span>)</span></code></pre></div>
<p>The resulting PCA score values that were derived as the final cluster centers can be pulled using:</p>
<div class="sourceCode" id="cb574"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb574-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb574-1" tabindex="-1"></a>clusters_PCA<span class="sc">$</span>centers</span></code></pre></div>
<pre><code>##         PC1        PC2        PC3          PC4         PC5         PC6
## 1  2.612553 -0.3054641  0.6219575 -0.008818739  0.18490367  0.05041185
## 2 -1.187524  0.1388473 -0.2827080  0.004008518 -0.08404712 -0.02291448
##           PC7         PC8          PC9         PC10
## 1 -0.03417184 -0.05270363 -0.018330999 -0.011904302
## 2  0.01553265  0.02395620  0.008332272  0.005411046</code></pre>
<p>Viewing the final cluster assignment per chemical:</p>
<div class="sourceCode" id="cb576"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb576-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb576-1" tabindex="-1"></a><span class="fu">head</span>(<span class="fu">cbind</span>(<span class="fu">rownames</span>(chemical_prop_df),clusters_PCA<span class="sc">$</span>cluster))</span></code></pre></div>
<pre><code>##            [,1]         [,2]
## 93762-09-5 &quot;93762-09-5&quot; &quot;2&quot; 
## 60270-55-5 &quot;60270-55-5&quot; &quot;2&quot; 
## 70225-15-9 &quot;70225-15-9&quot; &quot;2&quot; 
## 335-24-0   &quot;335-24-0&quot;   &quot;2&quot; 
## 647-29-0   &quot;647-29-0&quot;   &quot;2&quot; 
## 68259-12-1 &quot;68259-12-1&quot; &quot;2&quot;</code></pre>
<p><br></p>
<div id="visualizing-k-means-clusters-from-pca-results" class="section level4 hasAnchor">
<h4>Visualizing <em>K</em>-Means Clusters from PCA Results<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#visualizing-k-means-clusters-from-pca-results" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Let’s now view, again, the results of the main PCA focusing on the first two principal components; though this time let’s color each chemical according to <em>k</em>-means cluster.</p>
<div class="sourceCode" id="cb578"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb578-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb578-1" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="fu">data.frame</span>(my.pca<span class="sc">$</span>x), <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> <span class="fu">as.factor</span>(clusters_PCA<span class="sc">$</span>cluster))) <span class="sc">+</span> </span>
<span id="cb578-2"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb578-2" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb578-3"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb578-3" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&#39;Version D: PCA Plot of the First 2 PCs, colored by k-means Clustering&#39;</span>) <span class="sc">+</span> </span>
<span id="cb578-4"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb578-4" tabindex="-1"></a>  <span class="co"># it&#39;s good practice to put the percentage of the variance capture in the axes titles</span></span>
<span id="cb578-5"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb578-5" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">&quot;Principal Component 1 (40.9%)&quot;</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">&quot;Principal Component 2 (23.8%)&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-449-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Let’s put these two PCA plots side by side to compare them more easily. We’ll also tidy up the figures a bit so they’re closer to publication-ready.</p>
<div class="sourceCode" id="cb579"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb579-1"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-1" tabindex="-1"></a><span class="co"># PCA plot colored by chemical class</span></span>
<span id="cb579-2"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-2" tabindex="-1"></a>pcaplot1 <span class="ot">=</span> <span class="fu">ggplot</span>(<span class="fu">data.frame</span>(my.pca<span class="sc">$</span>x), <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> dat<span class="sc">$</span>List)) <span class="sc">+</span> </span>
<span id="cb579-3"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-3" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span> </span>
<span id="cb579-4"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-4" tabindex="-1"></a>  </span>
<span id="cb579-5"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-5" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span> </span>
<span id="cb579-6"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-6" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">9</span>), <span class="co"># changing size of axis labels</span></span>
<span id="cb579-7"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-7" tabindex="-1"></a>      <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">&quot;bold&quot;</span>, <span class="at">size =</span> <span class="fu">rel</span>(<span class="fl">1.3</span>)), <span class="co"># changes axis titles</span></span>
<span id="cb579-8"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-8" tabindex="-1"></a>      <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">&#39;bold&#39;</span>, <span class="at">size =</span> <span class="dv">10</span>), <span class="co"># changes legend title</span></span>
<span id="cb579-9"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-9" tabindex="-1"></a>      <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">9</span>)) <span class="sc">+</span> <span class="co"># changes legend text</span></span>
<span id="cb579-10"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-10" tabindex="-1"></a></span>
<span id="cb579-11"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-11" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&#39;Principal Component 1 (40.9%)&#39;</span>, <span class="at">y =</span> <span class="st">&#39;Principal Component 2 (23.8%)&#39;</span>, </span>
<span id="cb579-12"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-12" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">&quot;Chemical Class&quot;</span>)  <span class="co"># changing axis labels </span></span>
<span id="cb579-13"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-13" tabindex="-1"></a></span>
<span id="cb579-14"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-14" tabindex="-1"></a><span class="co"># PCA Plot by k means clusters</span></span>
<span id="cb579-15"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-15" tabindex="-1"></a>pcaplot2 <span class="ot">=</span> <span class="fu">ggplot</span>(<span class="fu">data.frame</span>(my.pca<span class="sc">$</span>x), <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> <span class="fu">as.factor</span>(clusters_PCA<span class="sc">$</span>cluster))) <span class="sc">+</span> </span>
<span id="cb579-16"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-16" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span>  </span>
<span id="cb579-17"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-17" tabindex="-1"></a>  </span>
<span id="cb579-18"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-18" tabindex="-1"></a>  <span class="fu">theme_light</span>() <span class="sc">+</span> </span>
<span id="cb579-19"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-19" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">9</span>), <span class="co"># changing size of axis labels</span></span>
<span id="cb579-20"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-20" tabindex="-1"></a>      <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">&quot;bold&quot;</span>, <span class="at">size =</span> <span class="fu">rel</span>(<span class="fl">1.3</span>)), <span class="co"># changes axis titles</span></span>
<span id="cb579-21"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-21" tabindex="-1"></a>      <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">9</span>)) <span class="sc">+</span> <span class="co"># changes legend text</span></span>
<span id="cb579-22"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-22" tabindex="-1"></a></span>
<span id="cb579-23"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-23" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&#39;Principal Component 1 (40.9%)&#39;</span>, <span class="at">y =</span> <span class="st">&#39;Principal Component 2 (23.8%)&#39;</span>, </span>
<span id="cb579-24"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-24" tabindex="-1"></a>       <span class="at">color =</span> <span class="fu">expression</span>(<span class="fu">bold</span>(<span class="fu">bolditalic</span>(K)<span class="sc">-</span>Means<span class="sc">~</span>Cluster))) <span class="co"># changing axis labels </span></span>
<span id="cb579-25"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-25" tabindex="-1"></a></span>
<span id="cb579-26"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-26" tabindex="-1"></a><span class="co"># Creating 1 figure</span></span>
<span id="cb579-27"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-27" tabindex="-1"></a><span class="fu">plot_grid</span>(pcaplot1, pcaplot2,</span>
<span id="cb579-28"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-28" tabindex="-1"></a>                     <span class="co"># Adding labels, changing size their size and position</span></span>
<span id="cb579-29"><a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#cb579-29" tabindex="-1"></a>                     <span class="at">labels =</span> <span class="st">&quot;AUTO&quot;</span>, <span class="at">label_size =</span> <span class="dv">15</span>, <span class="at">label_x =</span> <span class="fl">0.03</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-450-1.png" width="1920" style="display: block; margin: auto;" /></p>
<p>An appropriate title for this figure could be:</p>
<p>“<strong>Figure X. Principal Component Analysis (PCA) plots highlight similarities between chemical class and <em>k</em>-means clusters.</strong> These PCA plots are based on physiochemical properties and compare (A) chemical class categories and the (B) <em>K</em>-means derived cluster assignments.”</p>
</div>
</div>
<div id="answer-to-environmental-health-question-6-1" class="section level3 hasAnchor">
<h3>Answer to Environmental Health Question 6<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#answer-to-environmental-health-question-6-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="question">
<p><em>With this we can answer <strong>Environmental Health Question #6</strong></em>: What kinds of applications/endpoints can be better understood and/or predicted because of these derived chemical groupings?</p>
</div>
<div class="answer">
<p><strong>Answer</strong>: With these well-informed chemical groupings, we can now better understand the variables that attribute to the chemical classifications. We can also use this information to better understand data trends and predict environmental fate and transport for these chemicals. The reduced variables derived through PCA, and/or <em>k</em>-means clustering patterns can also be used as input variables to predict toxicological outcomes.</p>
</div>
<p><br></p>
</div>
</div>
<div id="concluding-remarks-21" class="section level2 hasAnchor">
<h2>Concluding Remarks<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#concluding-remarks-21" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>In conclusion, this training module provides an example exercise on organizing physicochemical data and analyzing trends within these data to determine chemical groupings. Results are compared from those produced using just the original data vs. clustered data from <em>k</em>-means vs. reduced data from PCA. These methods represent common tools that are used in high dimensional data analyses within the field of environmental health sciences.</p>
<div id="additional-resources-9" class="section level3 hasAnchor">
<h3>Additional Resources<a href="unsupervised-machine-learning-part-1-k-means-clustering-pca.html#additional-resources-9" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li><a href="https://f0nzie.github.io/machine_learning_compilation/detailed-study-of-principal-component-analysis.html">Detailed study of Principal Component Analysis</a></li>
<li><a href="https://xsliulab.github.io/Workshop/2021/week10/r-cluster-book.pdf">Practical Guide to Cluster Analysis in R</a></li>
</ul>
<p><br></p>
<p><label class="tykfont">
Test Your Knowledge
</label></p>
<div class="tyk">
<p>In this training module, we presented an unsupervised machine learning example that was based on defining <em>k</em>-means clusters based on chemical class where <em>k</em> = 2. Often times, analyses are conducted to explore potential clustering relationships without a preexisting idea of what <em>k</em> or the number of clusters should be. In this test your knowledge section, we’ll go through an example like that.</p>
<p>Using the accompanying flame retardant and pesticide physicochemical property variables found in the file (“Module5_4_TYKInput.csv”), answer the following questions:</p>
<ol style="list-style-type: decimal">
<li>What are some of the physicochemical properties that seem to be driving chemical clustering patterns derived through <em>k</em>-means?</li>
<li>Upon reducing the data through PCA, which physicochemical property contributes the most towards informing data variance captured in the primary principal component?</li>
<li>If we did not have information telling us which chemical belonged to which class, could we use PCA and <em>k</em>-means to accurately predict whether a chemical is a PFAS or a statin?</li>
</ol>
</div>
</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="supervised-machine-learning-model-interpretation.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="unsupervised-machine-learning-part-2-additional-clustering-applications.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
